<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Methylation Analysis Based on Signal Detection • MethylIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Methylation Analysis Based on Signal Detection">
<meta property="og:description" content="Methyl-IT implements methylation analysis based on signal detection
             theory and machine-learning. Methylation process is a stochastic
             process, particularly, a biochemical-biophysical process which must
             not be reduced to statistic. Methyl-IT includes the information on
             the statistical biophysics of the methylation process with the 
             estimation of the probability distribution of the noise 
             (plus signal), which is expressed in terms of information 
             divergences. ">
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="methyl-it-">Methyl-IT <a href="https://genomaths.github.io/methylit"><img src="reference/figures/logo.png" align="right"></a>
<a class="anchor" aria-label="anchor" href="#methyl-it-"></a>
</h1></div>
<p><br></p>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>We have developed a novel methylome analysis procedure, Methyl-IT, based on signal detection and machine learning. Methylation analysis is confronted in Methyl-IT as a signal detection problem, and the method was designed to discriminate methylation regulatory signal from background noise induced by Brownian motion and thermal fluctuations. Our group has proposed an information thermodynamics approach to investigate genome-wide methylation patterning based on the statistical mechanical effect of methylation on DNA molecules (<a href="#id_1">1</a>-<a href="#id_3">3</a>). The information thermodynamics-based approach is postulated to provide greater sensitivity for resolving true signal from the thermodynamic background within the methylome (<a href="#id_1">1</a>). This theory provides the knowledge on the family of probability distributions that better fit the methylation signals when expressed in terms of information divergences of methylation levels. Because the biological signal created within the dynamic methylome environment characteristic of organisms is not free from background noise, the approach, designated Methyl-IT, includes an application of signal detection theory.</p>
<p>A basic requirement for the application of signal detection is the knowledge of the background noise probability distribution. A generalized gamma (GG) probability distribution model can be deduced on a statistical mechanical/thermodynamics basis for DNA methylation induced by thermal fluctuations (<a href="#id_1">1</a>), which leads to the particular cases of members of GG probability distribution family. For example, assuming that this background methylation variation is consistent with a Poisson process, it can be distinguished from variation associated with methylation regulatory machinery, which is non-independent for all genomic regions (<a href="#id_1">1</a>). An information-theoretic divergence to express the variation in methylation induced by background thermal fluctuations will follow a probability distribution model, member of the GG family, provided that it is proportional to minimum energy dissipated per bit of information from methylation change. The information thermodynamics model was previously verified with more than 150 Arabidopsis and more than 90 human methylome datasets (<a href="#id_3">3</a>).</p>
<p>With Methyl-IT R package we are providing the functions from the R scripts used in the manuscript (<a href="#id_1">1</a>), as well as, additional functions that will be used in further studies. The application of the information thermodynamics of cytosine DNA methylation is not limited to the current methylome analysis, which is only a particular application. The theory permits us the study of plant and animal methylomes in the framework of a communication system (<a href="#id_3">3</a>), where cytosine DNA methylation has the dual roles of stabilizing the DNA molecule and to carry the regulatory signals.</p>
<p>The application of Methyl-T signal detection-machine learning approach to methylation analysis of whole genome bisulfite sequencing (WGBS) data permits a high level of methylation signal resolution in cancer-associated genes and pathways (<a href="#id_4">4</a>), as well as, treatment-associated genes and pathways in plants (<a href="#id_5">5</a>).</p>
<p>Notice that MethylIT pipeline is not limited to bisulfite sequencing experimental data. Any experimental data set on summarized <a href="https://rdrr.io/bioc/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a> or <a href="https://rdrr.io/bioc/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList-class</a> objects carrying the counts of methylated (mC) and unmethylated (uC) cytosines in their metacolumns can be analyzed. As a matter of fact, since cell’s developmental stages in a tissue are not synchronized, but in different ontogenetic development stages, the methylation status at each cytosine site would dynamically vary from cell to cell. Therefore, at tissue level, the methylation status can be quantitaivally represented by a probability, i.e., a number between 0 and 1, also known as methylation level.</p>
</div>
<div class="section level2">
<h2 id="status">Status<a class="anchor" aria-label="anchor" href="#status"></a>
</h2>
<p>Currently, the package is actively used in methylation analyses. Nevertheless, improvements are regularly introduced. Watch this repo or check for updates. THE PACKAGE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED. The beta version is frequently updated. We are using the version notation: 0.3.2.x.y, where digit ‘y’ is frequently changed.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Be sure that both the R and bioconductor packages are up to date. To get the latest version of Bioconductor by starting R and entering the commands:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>)) <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>()</span></code></pre></div>
<p>Install OS packages:</p>
<p>CentOS 7</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>sudo yum <span class="sc">-</span>y install libcurl<span class="sc">-</span>devel.x86_64 openssl<span class="sc">-</span>devel.x86_64 libxml2<span class="sc">-</span>devel mariadb<span class="sc">-</span>devel.x86_64</span></code></pre></div>
<p>Ubuntu</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>sudo apt install libmysqlclient<span class="sc">-</span>dev libcurl4<span class="sc">-</span>openssl<span class="sc">-</span>dev libssl<span class="sc">-</span>dev libxml2<span class="sc">-</span>dev git build<span class="sc">-</span>essential</span></code></pre></div>
<p>Install R dependencies:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="fu">c</span>(<span class="st">"caret"</span>, <span class="st">"Epi"</span>, <span class="st">"e1071"</span>, <span class="st">"minpack.lm"</span>, <span class="st">"nls2"</span>,</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="st">"caTools"</span>,  <span class="st">"RCurl"</span>,<span class="st">'GenomicRanges'</span>,<span class="st">'BiocParallel'</span>, <span class="st">'biovizBase'</span>, </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="st">'genefilter'</span>, <span class="st">'data.table'</span>, <span class="st">'dplyr'</span>, <span class="st">'GenomeInfoDb'</span>, </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="st">'randomForest'</span>, <span class="at">dependencies=</span><span class="cn">TRUE</span>))</span></code></pre></div>
<p>You can install MethylIT from GitHub:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Master stable version (0.3.2.5)</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"genomaths/MethylIT"</span>)</span></code></pre></div>
<p>The Beta (developing) version:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">'genomaths/MethylIT'</span>, </span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">ref =</span> <span class="st">"MethylIT_0.3.2_beta"</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>   </span></code></pre></div>
</div>
<div class="section level2">
<h2 id="flow-diagram">Flow diagram<a class="anchor" aria-label="anchor" href="#flow-diagram"></a>
</h2>
<p><img src="reference/figures/workflow.png" align="center"></p>
</div>
<div class="section level2">
<h2 id="get-started">Get Started<a class="anchor" aria-label="anchor" href="#get-started"></a>
</h2>
<p><a href="https://genomaths.github.io/methylit/articles/MethylIT.html">Fast starting here</a></p>
</div>
<div class="section level2">
<h2 id="tutorials">Tutorials<a class="anchor" aria-label="anchor" href="#tutorials"></a>
</h2>
<ul>
<li><a href="https://genomaths.github.io/methylit/articles/MethylIT.html">Get started-with Methyl-IT</a></li>
<li><a href="https://genomaths.github.io/methylit/articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a></li>
<li><a href="https://genomaths.github.io/methylit/articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a></li>
<li><a href="https://genomaths.github.io/methylit/articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a></li>
<li><a href="https://genomaths.github.io/methylit/articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a></li>
<li><a href="https://genomaths.github.io/methylit/articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a></li>
<li><h2 id="dmr-detection-with-methyl-it"><a href="https://genomaths.github.io/methylit/articles/dmr_at_memory.html">DMR detection with Methyl-IT</a></h2></li>
</ul>
</div>
<div class="section level2">
<h2 id="methyl-it-documentation">Methyl-IT documentation:<a class="anchor" aria-label="anchor" href="#methyl-it-documentation"></a>
</h2>
<p>Whole documentation, manual and several examples, are available at <a href="https://genomaths.github.io/methylit/" target="_blank">Methyl-IT website</a></p>
<hr>
</div>
<div class="section level2">
<h2 id="scripts-and-additional-examples-at">Scripts and additional examples at:<a class="anchor" aria-label="anchor" href="#scripts-and-additional-examples-at"></a>
</h2>
<p><a href="https://git.psu.edu/genomath/MethylIT_examples" class="external-link uri">https://git.psu.edu/genomath/MethylIT_examples</a></p>
<p><a href="https://drive.google.com/open?id=12D0vasaptB2IpA_HFxe6lzP5kxNEJqPF" class="external-link">R script and data for testing after installation</a></p>
<hr>
</div>
<div class="section level2">
<h2 id="report-issues-bugs">Report issues (bugs):<a class="anchor" aria-label="anchor" href="#report-issues-bugs"></a>
</h2>
<p><a href="https://github.com/genomaths/MethylIT/issues" class="external-link">Here</a></p>
<hr>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p><a name="1">1</a>. Sanchez, R.; Yang, X.; Maher, T.; Mackenzie, S.A. Discrimination of DNA Methylation Signal from Background Variation for Clinical Diagnostics. <a href="https://doi.org/10.3390/ijms20215343" class="external-link">Int. J. Mol. Sci. 2019, 20, 5343</a>.</p>
<p><br><a name="2">2</a>. Sanchez R, Mackenzie SA: Genome-Wide Discriminatory Information Patterns of Cytosine DNA Methylation. <a href="https://dx.doi.org/10.3390%2Fijms17060938" class="external-link">Int. J. Mol. Sci. 2016, 17(6), 938</a>.</p>
<p><br><a name="3">3</a>. Sanchez R, Mackenzie SA: Information Thermodynamics of Cytosine DNA Methylation. <a href="https://doi.org/10.1371/journal.pone.0150427" class="external-link">Plos One 2016, 11</a>.<br></p>
<p><br><a name="4">4</a>. Sanchez R, Mackenzie SA: Integrative Network Analysis of Differentially Methylated and Expressed Genes for Biomarker Identification in Leukemia. <a href="https://doi.org/10.1038/s41598-020-58123-2" class="external-link">Sci Rep 2020, 10(1), 2123</a>.<br></p>
<p><br><a name="5">5</a>. Yang, X, Sanchez, R et al.: Segregation of an MSH1 RNAi transgene produces heritable non-genetic memory in association with methylome reprogramming. <a href="https://doi.org/10.1038/s41467-020-16036-8" class="external-link">Nature Communications 2020, 11(1), 2214</a>.<br></p>
</div>
<div class="section level2">
<h2 id="license">License<a class="anchor" aria-label="anchor" href="#license"></a>
</h2>
<p>You are free to copy, distribute and transmit MethylIT for non-commercial purposes. Any use of MethylIT for a commercial purpose is subject to and requires a special license.</p>
</div>
<div class="section level2">
<h2 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h2>
<p>For questions about the MethylIT project, contact <a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a></p>
</div>
<div class="section level2">
<h2 id="contributor-code-of-conduct">Contributor Code of Conduct<a class="anchor" aria-label="anchor" href="#contributor-code-of-conduct"></a>
</h2>
<p>Please note that this project is released with a <a href="CONDUCT.html">Contributor Code of Conduct</a>. By participating in this project you agree to abide by its terms.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/genomaths/MethylIT/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/genomaths/MethylIT/issues" class="external-link">Report a bug</a></li>
<li><a href="https://genomaths.com" class="external-link">Author's website</a></li>
<li><a href="https://bio.psu.edu/directory/rus547" class="external-link">Affiliation (<img src="https://git.psu.edu/genomath/MethylIT_data/raw/master/penn-state-lion.png" height="24"> PennState)</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing MethylIT</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Robersy Sanchez <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-5246-1453" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
