<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Given two codon sequences represented in a given Abelian
group, this function computes the automorphisms describing codon mutational
events. Basically, this function is a wrapping to call the corresponding
function for a specified Abelian group."><title>Compute the Automorphisms of Mutational Events Between two Codon Sequences Represented in a Given Abelian group. — automorphisms • GenomAutomorphism</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Compute the Automorphisms of Mutational Events Between two Codon Sequences Represented in a Given Abelian group. — automorphisms"><meta property="og:description" content="Given two codon sequences represented in a given Abelian
group, this function computes the automorphisms describing codon mutational
events. Basically, this function is a wrapping to call the corresponding
function for a specified Abelian group."><meta property="og:image" content="https://genomaths.github.io/genomautomorphism/logo.png"><meta name="robots" content="noindex"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GenomAutomorphism</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">1.0.2.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/GenomAutomorphism.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Package functions</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/GenomAutomorphism.html">GenomAutomorphism</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_and_decision_tree_II.html">A Short Introduction to Algebraic Taxonomy on Genes Regions II</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/Modular_Matrix_Operation_of_Mutational_Events.html">Modular Matrix Operations of Mutational Events</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Compute the Automorphisms of Mutational Events Between two Codon Sequences Represented in a Given Abelian group.</h1>
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/R/automorphisms.R" class="external-link"><code>R/automorphisms.R</code></a></small>
      <div class="d-none name"><code>automorphisms.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Given two codon sequences represented in a given Abelian
group, this function computes the automorphisms describing codon mutational
events. Basically, this function is a wrapping to call the corresponding
function for a specified Abelian group.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">automorphisms</span><span class="op">(</span>seqs <span class="op">=</span> <span class="cn">NULL</span>, filepath <span class="op">=</span> <span class="cn">NULL</span>, group <span class="op">=</span> <span class="st">"Z4"</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for DNAStringSet_OR_NULL</span></span>
<span><span class="fu">automorphisms</span><span class="op">(</span></span>
<span>  seqs <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  filepath <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Z5"</span>, <span class="st">"Z64"</span>, <span class="st">"Z125"</span>, <span class="st">"Z5^3"</span><span class="op">)</span>,</span>
<span>  cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"TGCA"</span><span class="op">)</span>,</span>
<span>  cube_alt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CATG"</span>, <span class="st">"GTAC"</span><span class="op">)</span>,</span>
<span>  nms <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  start <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  end <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  chr <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  strand <span class="op">=</span> <span class="st">"+"</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fu">multicoreWorkers</span><span class="op">(</span><span class="op">)</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>seqs</dt>
<dd><p>An object from a <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></code> or
<code><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">DNAMultipleAlignment</a></code> class carrying the DNA
pairwise alignment of two sequences. The pairwise alignment provided in
argument <em><strong>seq</strong></em> or the 'fasta' file <em><strong>filepath</strong></em>
must correspond to codon sequences.</p></dd>


<dt>filepath</dt>
<dd><p>A character vector containing the path to a file in
<em><strong>fasta</strong></em> format to be read. This argument must be given if
<em>codon &amp; base</em> arguments are not provided.</p></dd>


<dt>group</dt>
<dd><p>A character string denoting the group representation for the
given base or codon as shown in reference (1).</p></dd>


<dt>...</dt>
<dd><p>Not in use.</p></dd>


<dt>cube, cube_alt</dt>
<dd><p>A character string denoting pairs of the 24
Genetic-code cubes, as given in references (2-3). That is, the base pairs
from the given cubes must be complementary each other. Such a cube pair are
call \(dual cubes\) and, as shown in reference (3), each pair integrates
group.</p></dd>


<dt>nms</dt>
<dd><p>Optional. Only used if the DNA sequence alignment provided
carries more than two sequences. A character string giving short names for
the alignments to be compared. If not given then the automorphisms between
pairwise alignment are named as: 'aln_1', 'aln_2', and so on.</p></dd>


<dt>start, end, chr, strand</dt>
<dd><p>Optional parameters required to build a
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code>. If not provided the default
values given for the function definition will be used.</p></dd>


<dt>num.cores, tasks</dt>
<dd><p>Parameters for parallel computation using package
<code><a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallel-package.html" class="external-link">BiocParallel-package</a></code>: the number of cores to
use, i.e. at most how many child processes will be run simultaneously (see
<code><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></code> and the number of tasks per job (only
for Linux OS).</p></dd>


<dt>verbose</dt>
<dd><p>If TRUE, prints the progress bar.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>This function returns a <code><a href="Automorphism.html">Automorphism-class</a></code> object
with four columns on its metacolumn named: <em>seq1</em>, <em>seq2</em>,
<em>autm</em>, and <em>cube</em>.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Herein, automorphisms are algebraic descriptions of mutational
event observed in codon sequences represented on different Abelian groups.
In particular, as described in references (3-4), for each representation of
the codon set on a defined Abelian group there are 24 possible isomorphic
Abelian groups. These Abelian groups can be labeled based on the DNA
base-order used to generate them. The set of 24 Abelian groups can be
described as a group isomorphic to the symmetric group of degree four
(\(S_4\), see reference (4)). Function <code><a href="automorphismByRanges.html">automorphismByRanges</a></code>
permits the classification of the pairwise alignment of protein-coding
sub-regions based on the mutational events observed on it and on the
genetic-code cubes that describe them.</p>
<p>Automorphisms in Z5, Z64 and Z125 are described as functions
\(f(x) = k x mod 64\) and \(f(x) = k x mod 125\), where k and x are
elements from the set of integers modulo 64 or modulo 125, respectively. If
an automorphisms cannot be found on any of the cubes provided in the
argument \(cube\), then function <code>automorphisms</code> will search
for automorphisms in the cubes provided in the argument \(cube_alt\).</p>
<p>Automorphisms in Z5^3' are described as functions \(f(x) = Ax mod Z5\),
where A is diagonal matrix.</p>
<p>Arguments <em><strong>cube</strong></em> and <em><strong>cube_alt</strong></em> must be
pairs of' dual cubes (see section 2.4 from reference 4).</p>
    </div>
    <div class="section level2">
    <h2 id="methods">Methods<a class="anchor" aria-label="anchor" href="#methods"></a></h2>
    

<div class="section">
<h3 id="automorphismbyranges-"><code><a href="automorphismByRanges.html">automorphismByRanges</a></code>:<a class="anchor" aria-label="anchor" href="#automorphismbyranges-"></a></h3>


<p>This function returns a <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> object.
Consecutive mutational events (on the codon sequence) described by
automorphisms on a same cube are grouped in a range.</p>
</div>

<div class="section">
<h3 id="automorphism-bycoef"><code><a href="automorphism_bycoef.html">automorphism_bycoef</a></code><a class="anchor" aria-label="anchor" href="#automorphism-bycoef"></a></h3>


<p>This function returns a <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> object.
Consecutive mutational events (on the codon sequence) described by
the same automorphisms coefficients are grouped in a range.</p>
</div>

<div class="section">
<h3 id="getautomorphisms"><code><a href="getAutomorphisms.html">getAutomorphisms</a></code><a class="anchor" aria-label="anchor" href="#getautomorphisms"></a></h3>


<p>This function returns an AutomorphismList-class object as a list of
Automorphism-class objects, which inherits from
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> objects.</p>
</div>

<div class="section">
<h3 id="conserved-regions"><code><a href="conserved_regions.html">conserved_regions</a></code><a class="anchor" aria-label="anchor" href="#conserved-regions"></a></h3>


<p>Returns a <code><a href="AutomorphismByCoef.html">AutomorphismByCoef</a></code> class object containing the
requested regions.</p>
</div>

    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    
<ol><li><p>Sanchez R, Morgado E, Grau R. Gene algebra from a genetic code
algebraic structure. J Math Biol. 2005 Oct;51(4):431-57.
doi: 10.1007/s00285-005-0332-8. Epub 2005 Jul 13. PMID: 16012800. (
<a href="https://arxiv.org/pdf/q-bio/0412033.pdf" class="external-link">PDF</a>).</p></li>
<li><p>Robersy Sanchez, Jesus Barreto (2021) Genomic Abelian Finite
Groups.
<a href="https://doi.org/10.1101/2021.06.01.446543" class="external-link">doi:10.1101/2021.06.01.446543</a></p></li>
<li><p>M. V Jose, E.R. Morgado, R. Sanchez, T. Govezensky, The 24 possible
algebraic representations of the standard genetic code in six or in three
dimensions, Adv. Stud. Biol. 4 (2012) 110-152.<a href="https://is.gd/na9eap" class="external-link">PDF</a>.</p></li>
<li><p>R. Sanchez. Symmetric Group of the Genetic-Code Cubes. Effect of the
Genetic-Code Architecture on the Evolutionary Process MATCH Commun. Math.
Comput. Chem. 79 (2018) 527-560. <a href="https://bit.ly/2Z9mjM7" class="external-link">PDF</a></p></li>
</ol></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="autZ64.html">autZ64</a></code>.</p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Robersy Sanchez (<a href="https://genomaths.com" class="external-link">https://genomaths.com</a>).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Load a pairwise alignment</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"aln"</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">aln</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet object of length 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]    51 <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span>---<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2]    51 <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span>------<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Automorphism on "Z5^3"</span></span></span>
<span class="r-in"><span><span class="va">autms</span> <span class="op">&lt;-</span> <span class="fu">automorphisms</span><span class="op">(</span>seqs <span class="op">=</span> <span class="va">aln</span>, group <span class="op">=</span> <span class="st">"Z5^3"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">autms</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Automorphism object with 17 ranges and 8 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        seqnames    ranges strand |        seq1        seq2         aa1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]        1         1      + |         ACC         ATC           T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]        1         2      + |         TAT         TAT           Y</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]        1         3      + |         GTT         GTT           V</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]        1         4      + |         GGT         GGT           G</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]        1         5      + |         ATT         ATT           I</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...      ...       ...    ... .         ...         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [13]        1        13      + |         TCT         TCC           S</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [14]        1        14      + |         AGC         ---           S</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [15]        1        15      + |         TCA         ---           S</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [16]        1        16      + |         CTA         CTC           L</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [17]        1        17      + |         CAT         CTT           H</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                aa2   coord1   coord2        autm        cube</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;character&gt; &lt;matrix&gt; &lt;matrix&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]           I    1:2:2    1:4:2       1,2,1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]           Y    4:1:4    4:1:4       1,1,1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]           V    3:4:4    3:4:4       1,1,1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]           G    3:3:4    3:3:4       1,1,1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]           I    1:4:4    1:4:4       1,1,1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...         ...      ...      ...         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [13]           S    4:2:4    4:2:2       1,1,3        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [14]           -    1:3:2    0:0:0           0        Trnl</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [15]           -    4:2:1    0:0:0           0        Trnl</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [16]           L    2:4:1    2:4:2       1,1,2        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [17]           L    2:1:4    2:4:4       1,4,1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Automorphism on "Z64"</span></span></span>
<span class="r-in"><span><span class="va">autms</span> <span class="op">&lt;-</span> <span class="fu">automorphisms</span><span class="op">(</span>seqs <span class="op">=</span> <span class="va">aln</span>, group <span class="op">=</span> <span class="st">"Z64"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">autms</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Automorphism object with 17 ranges and 8 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        seqnames    ranges strand |        seq1        seq2         aa1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]        1         1      + |         ACC         ATC           T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]        1         2      + |         TAT         TAT           Y</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]        1         3      + |         GTT         GTT           V</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]        1         4      + |         GGT         GGT           G</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]        1         5      + |         ATT         ATT           I</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...      ...       ...    ... .         ...         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [13]        1        13      + |         TCT         TCC           S</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [14]        1        14      + |         AGC         ---           S</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [15]        1        15      + |         TCA         ---           S</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [16]        1        16      - |         CTA         CTC           L</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [17]        1        17      + |         CAT         CTT           H</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                aa2    coord1    coord2      autm        cube</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]           I        17        49        33        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]           Y        15        15         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]           V        59        59         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]           G        43        43         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]           I        51        51         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...         ...       ...       ...       ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [13]           S        31        29         3        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [14]           -        33        NA         0        Trnl</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [15]           -        28        NA         0        Trnl</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [16]           L        52        53        30        TGCA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [17]           L         7        55        17        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Automorphism on "Z64" from position 1 to 33</span></span></span>
<span class="r-in"><span><span class="va">autms</span> <span class="op">&lt;-</span> <span class="fu">automorphisms</span><span class="op">(</span></span></span>
<span class="r-in"><span>    seqs <span class="op">=</span> <span class="va">aln</span>,</span></span>
<span class="r-in"><span>    group <span class="op">=</span> <span class="st">"Z64"</span>,</span></span>
<span class="r-in"><span>    start <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>    end <span class="op">=</span> <span class="fl">33</span>,</span></span>
<span class="r-in"><span>    verbose <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">autms</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Automorphism object with 11 ranges and 8 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        seqnames    ranges strand |        seq1        seq2         aa1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]        1         1      + |         ACC         ATC           T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]        1         2      + |         TAT         TAT           Y</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]        1         3      + |         GTT         GTT           V</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]        1         4      + |         GGT         GGT           G</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]        1         5      + |         ATT         ATT           I</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [6]        1         6      + |         ---         ACG           -</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [7]        1         7      + |         GCG         ACG           A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [8]        1         8      + |         CTC         CTC           L</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [9]        1         9      + |         CAA         CAA           Q</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [10]        1        10      + |         CTC         TTC           L</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [11]        1        11      + |         CTT         CTT           L</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                aa2    coord1    coord2      autm        cube</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]           I        17        49        33        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]           Y        15        15         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]           V        59        59         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]           G        43        43         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]           I        51        51         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [6]           T        NA        18         0        Trnl</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [7]           T        26        18        45        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [8]           L        53        53         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [9]           Q         4         4         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [10]           F        53        61        41        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [11]           L        55        55         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

