<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function applies numerical indices representing various physicochemical
and biochemical properties of amino acids and pairs of amino acids to DNA
protein-coding or to aminoacid sequences. As results, DNA protein-coding or
the aminoacid sequences are represented as numerical vectors which can be
subject of further downstream statistical analysis and digital signal
processing."><title>Amino acid numerical matrix — peptide_phychem_index • GenomAutomorphism</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Amino acid numerical matrix — peptide_phychem_index"><meta property="og:description" content="This function applies numerical indices representing various physicochemical
and biochemical properties of amino acids and pairs of amino acids to DNA
protein-coding or to aminoacid sequences. As results, DNA protein-coding or
the aminoacid sequences are represented as numerical vectors which can be
subject of further downstream statistical analysis and digital signal
processing."><meta property="og:image" content="https://genomaths.github.io/genomautomorphism/logo.png"><meta name="robots" content="noindex"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GenomAutomorphism</a>

    <small class="nav-text text-danger me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">1.0.2.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/GenomAutomorphism.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Package functions</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/GenomAutomorphism.html">GenomAutomorphism</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_and_decision_tree_II.html">A Short Introduction to Algebraic Taxonomy on Genes Regions II</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/Modular_Matrix_Operation_of_Mutational_Events.html">Modular Matrix Operations of Mutational Events</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Amino acid numerical matrix</h1>
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/R/peptide_phychem_index.R" class="external-link"><code>R/peptide_phychem_index.R</code></a></small>
      <div class="d-none name"><code>peptide_phychem_index.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function applies numerical indices representing various physicochemical
and biochemical properties of amino acids and pairs of amino acids to DNA
protein-coding or to aminoacid sequences. As results, DNA protein-coding or
the aminoacid sequences are represented as numerical vectors which can be
subject of further downstream statistical analysis and digital signal
processing.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">peptide_phychem_index</span><span class="op">(</span><span class="va">aa</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for character</span></span>
<span><span class="fu">peptide_phychem_index</span><span class="op">(</span></span>
<span>  <span class="va">aa</span>,</span>
<span>  acc <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  aaindex <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  userindex <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  alphabet <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AA"</span>, <span class="st">"DNA"</span><span class="op">)</span>,</span>
<span>  genetic.code <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/GENETIC_CODE.html" class="external-link">getGeneticCode</a></span><span class="op">(</span><span class="st">"1"</span><span class="op">)</span>,</span>
<span>  no.init.codon <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  if.fuzzy.codon <span class="op">=</span> <span class="st">"error"</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for DNAStringSet_OR_DNAMultipleAlignment</span></span>
<span><span class="fu">peptide_phychem_index</span><span class="op">(</span></span>
<span>  <span class="va">aa</span>,</span>
<span>  acc <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  aaindex <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  userindex <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  alphabet <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AA"</span>, <span class="st">"DNA"</span><span class="op">)</span>,</span>
<span>  genetic.code <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/GENETIC_CODE.html" class="external-link">getGeneticCode</a></span><span class="op">(</span><span class="st">"1"</span><span class="op">)</span>,</span>
<span>  no.init.codon <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  if.fuzzy.codon <span class="op">=</span> <span class="st">"error"</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>aa</dt>
<dd><p>A character string, a <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></code>
or a <code><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">DNAMultipleAlignment</a></code> class object carrying
the DNA pairwise alignment of two sequences.</p></dd>


<dt>...</dt>
<dd><p>Not in use.</p></dd>


<dt>acc</dt>
<dd><p>Accession id for a specified mutation or contact potential
matrix.</p></dd>


<dt>aaindex</dt>
<dd><p>Database where the requested accession id is locate and from
where the aminoacid indices can be obtained. The possible values are:
"aaindex2" or "aaindex3".</p></dd>


<dt>userindex</dt>
<dd><p>User provided aminoacid indices. This can be a numerical
vector or a matrix (20 x 20). If a numerical matrix is provided, then the
aminoacid indices are computes as column averages.</p></dd>


<dt>alphabet</dt>
<dd><p>Whether the alphabet is from the 20 aminoacid (AA) or
four (DNA)/RNA base alphabet. This would prevent mistakes, i.e.,
the strings "ACG" would be a base-triplet on the DNA alphabet or simply
the amino acid sequence of alanine, cysteine, and glutamic acid.</p></dd>


<dt>genetic.code, no.init.codon, if.fuzzy.codon</dt>
<dd><p>The same as given in
function <a href="translation.html">translation</a>.</p></dd>


<dt>num.cores, tasks</dt>
<dd><p>Parameters for parallel computation using package
<code><a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallel-package.html" class="external-link">BiocParallel-package</a></code>: the number of cores to
use, i.e. at most how many child processes will be run simultaneously
(see <code><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></code> and the number of tasks per job
(only for Linux OS).</p></dd>


<dt>verbose</dt>
<dd><p>If TRUE, prints the function log to stdout.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Depending on the user specifications, a mutation or contact
potential matrix, a list of available matrices (indices) ids or index
names can be returned. More specifically:</p>


<dl><dt><strong>aa_mutmat</strong>: </dt>
<dd><p>Returns an aminoacid mutation matrix or
a statistical protein contact potentials matrix.</p></dd>

<dt><strong>aa_index</strong>: </dt>
<dd><p>Returns the specified aminoacid physicochemical
indices.</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>If a DNA sequence is given, then it is assumed that it is a DNA base-triplet
sequence, i.e., the base sequence must be multiple of 3.</p>
<p>Errors can be originated if the given sequences carry letter which are not
from the DNA or aminoacid alphabet.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Robersy Sanchez <a href="https://genomaths.com" class="external-link">https://genomaths.com</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Let's create DNAStringSet-class object</span></span></span>
<span class="r-in"><span><span class="va">base</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span> seq1 <span class="op">=</span><span class="st">'ACGTCATAAAGT'</span>,</span></span>
<span class="r-in"><span>                            seq2 <span class="op">=</span> <span class="st">'GTGTAATACAGT'</span>,</span></span>
<span class="r-in"><span>                            seq3 <span class="op">=</span> <span class="st">'TCCTCATAAGGT'</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The stop condon 'TAA' yields NA</span></span></span>
<span class="r-in"><span><span class="va">aa</span> <span class="op">&lt;-</span> <span class="fu">peptide_phychem_index</span><span class="op">(</span><span class="va">base</span>, acc <span class="op">=</span> <span class="st">"EISD840101"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">aa</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MatrixSeq with 3 rows (sequences) and 4 columns (aminoacids/codons):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       A1    A2   A3    A4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S1 -0.18 -0.26   NA -0.26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S2  0.54    NA 0.02 -0.26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S3 -0.26 -0.26   NA  0.16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slots: 'seqs', 'matrix', 'names', 'aaindex', 'phychem', 'accession</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Description of the physicochemical index</span></span></span>
<span class="r-in"><span><span class="va">aa</span><span class="op">@</span><span class="va">phychem</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "EISD840101 Consensus normalized hydrophobicity scale (Eisenberg, 1984)"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Getthe aminoacid sequences. The stop codon 'TAA' is replaced by '*'.</span></span></span>
<span class="r-in"><span><span class="va">aa</span><span class="op">@</span><span class="va">seqs</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seq1   seq2   seq3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "TS*S" "V*YS" "SS*G" </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">aa</span> <span class="op">&lt;-</span> <span class="fu">peptide_phychem_index</span><span class="op">(</span><span class="va">base</span>, acc <span class="op">=</span> <span class="st">"MIYS850103"</span>, aaindex <span class="op">=</span> <span class="st">"aaindex3"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">aa</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MatrixSeq with 3 rows (sequences) and 4 columns (aminoacids/codons):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         A1     A2      A3     A4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S1 -0.0045 0.0015      NA 0.0015</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S2  0.0895     NA -0.0565 0.0015</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S3  0.0015 0.0015      NA 0.0220</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slots: 'seqs', 'matrix', 'names', 'aaindex', 'phychem', 'accession</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Description of the physicochemical index</span></span></span>
<span class="r-in"><span><span class="va">aa</span><span class="op">@</span><span class="va">phychem</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "MIYS850103 Quasichemical energy of interactions in an average buried environment"</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

