<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Linear density of DMPs at a given genomic region</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="/media/sf_D/MethylIY_utils_html/R.css" />
</head><body>

<table width="100%" summary="page for dmpDensity {<a href='https://genomaths.github.io/MethylIT_utils_HTML_Manual/MethylIT_utils_HTML_Manual.html'>MethylIT.utils</a>}"><tr><td>dmpDensity {<a href='https://genomaths.github.io/MethylIT_utils_HTML_Manual/MethylIT_utils_HTML_Manual.html'>MethylIT.utils</a>}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Linear density of DMPs at a given genomic region</h2>

<h3>Description</h3>

<p>The linear density of DMPs in a given genomic region (GR) is
defined according with the classical terminology in physics, i.e., as the
measure of the physical quantity of any characteristic value per unit of
length. In the current case, as the amount of DIMPs per nucleotide base.
</p>


<h3>Usage</h3>

<pre>
dmpDensity(GR, column = 1, cut.col = 1, cutoff, Chr = NULL,
  start.pos = NULL, end.pos = NULL, int.size1 = NULL,
  int.size2 = NULL, breaks = NULL, scaling = TRUE, plot = FALSE,
  noDMP.dens = TRUE, xlabel = "Coordinate",
  ylabel = "Normalized density", col.dmp = "red", col.ndmp = "blue",
  yintercept = 0.25, col.yintercept = "magenta",
  type.yintercept = "dashed", dig.lab = 3)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>GR</code></td>
<td>
<p>A genomic GRanges object carrying the genomic region where the
estimation of the DMP density will be accomplished.</p>
</td></tr>
<tr valign="top"><td><code>cut.col</code></td>
<td>
<p>Integer denoting the GR metacolumn where the decision variable
about whether a position is DMP is located. Default cut.col = 1.</p>
</td></tr>
<tr valign="top"><td><code>cutoff</code></td>
<td>
<p>Cut value to decide wheter the value of the variable used to
estimate the density is a DMP at each position. If missing, then
cutoff is estimated as the first queantile greater than zero from the
values given in the GR column <em>cut.col</em>.</p>
</td></tr>
<tr valign="top"><td><code>Chr</code></td>
<td>
<p>A character string. Default NULL. If the GR object comprises
several chromosomes, then one chromosome must be specified. Otherwise the
density of first chromosome will be returned.</p>
</td></tr>
<tr valign="top"><td><code>start.pos, end.pos</code></td>
<td>
<p>Start and end positions, respectively, of the GR
where the density of DMPs will be estimated. Default NULL. If NULL
densities will be estimated for the whole GR and the specified
chromosome.</p>
</td></tr>
<tr valign="top"><td><code>int.size1, int.size2</code></td>
<td>
<p>The interval/window size where the density of DMP
and no DMPs are computed. Default Null.</p>
</td></tr>
<tr valign="top"><td><code>breaks</code></td>
<td>
<p>Integer. Number of windows/intervals to split the GR. Deafult
NULL. If provided, then it is applied to compute the densities of DMPs
and no-DMPs. If 'int.size1', 'int.size2', and 'breaks' are NULL, then the
breaks are computed as:
<code>breaks &lt;- min(150, max(start(x))/nclass.FD(start(x)),
na.rm = TRUE)</code>,
where function <em>nclass.FD</em> (<code>nclass</code>) applies
Freedman-Diaconis algorithm.</p>
</td></tr>
<tr valign="top"><td><code>scaling</code></td>
<td>
<p>Logic value to deside whether to perform the scaling of the
estimated density values or not. Default is TRUE.</p>
</td></tr>
<tr valign="top"><td><code>plot</code></td>
<td>
<p>Logic. Whether to produce a grahic or not. Default, plot = TRUE.</p>
</td></tr>
<tr valign="top"><td><code>noDMP.dens</code></td>
<td>
<p>Logic whether to produce the graphics for no-DMP density.
Default is TRUE</p>
</td></tr>
<tr valign="top"><td><code>xlabel</code></td>
<td>
<p>X-axis label. Default <em>xlabel = &quot;Coordinate&quot;</em>.</p>
</td></tr>
<tr valign="top"><td><code>ylabel</code></td>
<td>
<p>Y-axis label. Default <em>ylabel = &quot;Normalized density&quot;</em>.</p>
</td></tr>
<tr valign="top"><td><code>col.dmp</code></td>
<td>
<p>Color for the density of DMPs in the graphic.</p>
</td></tr>
<tr valign="top"><td><code>col.ndmp</code></td>
<td>
<p>Color for the density of no DMPs in the graphic.</p>
</td></tr>
<tr valign="top"><td><code>yintercept</code></td>
<td>
<p>If plot == TRUE, this is the position for an horizantal
line that intercept the y-axis. Default yintercept = 0.25.</p>
</td></tr>
<tr valign="top"><td><code>col.yintercept</code></td>
<td>
<p>Color for the horizantal line 'yintercept'. Default
<em>col.yintercept = 'blue'</em></p>
</td></tr>
<tr valign="top"><td><code>type.yintercept</code></td>
<td>
<p>Line type for the horizantal line 'yintercept'.
Default <em>type.yintercept = &quot;dashed&quot;</em>.</p>
</td></tr>
<tr valign="top"><td><code>dig.lab</code></td>
<td>
<p>integer which is used when labels are not given. It determines
the number of digits used in formatting the break numbers.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Since the number of DIMPs along the DNA sequence vary, the local
density of DMPs <i>&rho;_i</i> at a fixed interval <i>&Delta;</i> l_i is
defined by the quotient <i>&rho;_i = &Delta; DMP_i/&Delta; l_i</i> is the
amount of DIMPs at the fixed interval. Likewise the local density of
non-DIMPs is defined as <i>&rho;_i = &Delta; nonDMP_i/&Delta; l_i</i>.
Notice that for a specified methylation context, e.g., CG,
<i>&Delta; CG_i - &Delta; DMP_i</i>, where <i>&Delta; CG</i> is the
amount CG positions at the given interval. The linear densities are
normalized as <i>&rho;_i/&rho;_max</i>, where <i>&rho;_max</i> is the maximum
of linear density found in a given GR.
</p>


<h3>Value</h3>

<p>If plot is TRUE will return a graphic with the densities of DMPs and
and no DMPs. If plot is FALSE a data frame object with the density of
DMPs and not DMPs will be returned.
</p>


<h3>Author(s)</h3>

<p>Robersy Sanchez
</p>


<h3>Examples</h3>

<pre>
set.seed(349)
## An auxiliary function to generate simulated hypothetical values from a
## variable with normal distribution
hypDT &lt;- function(mean, sd, n, num.pos, noise) {
    h &lt;- hist(rnorm(n, mean = mean, sd = sd), breaks = num.pos, plot = FALSE)
    hyp &lt;- h$density * 60 + runif(length(h$density)) * noise
    return(hyp)
}

## To generate a matrix of values with variations introduced by noise
hyp &lt;- hypDT(mean = 5, sd = 30, n = 10^5, noise = 4, num.pos = 8000)
## A GRanges object is built, which will carries the previous matrix on its
## meta-columns
l &lt;- length(hyp)
starts &lt;- seq(0, 30000, 3)[1:l]
ends &lt;- starts
GR &lt;- GRanges(seqnames = "chr1", ranges = IRanges(start = starts,
                end = ends))
mcols(GR) &lt;- data.frame(signal = hyp)

# If plot is TRUE a grphic is printed. Otherwise data frame is returned.
p &lt;- dmpDensity(GR, plot = FALSE)

# If ggplot2 package is installed, then graphic can customized using
# the returned data frame 'p':

# library(ggplot2)
## Auxiliar function to write scientific notation in the graphics
# fancy_scientific &lt;- function(l) {
#   #'turn in to character string in scientific notation
#   l &lt;- format( l, scientific = TRUE, digits = 1 )
#   l &lt;- gsub("0e\\+00","0",l)
#   #'quote the part before the exponent to keep all the digits
#   l &lt;- gsub("^(.*)e", "'\\1'e", l)
#   #'turn the 'e+' into plotmath format
#   l &lt;- gsub("e", "%*%10^", l)
#   l &lt;- gsub("[+]", "", l )
#   #'return this as an expression
#   parse(text=l)
# }
#
# max.pos = max(p$DMP.coordinate)
# ggplot(data=p) +
#   geom_line(aes(x=DMP.coordinate, y=DMP.density), color="red") +
#   geom_hline(aes(yintercept=0.25), linetype="dashed",
#              colour="blue", show.legend=FALSE ) +
#   geom_line(aes(x=coordinate, y=density), color="blue") +
#   xlab("Coordinate") + ylab("Normalized density") +
#   scale_y_continuous(breaks=c(0.00, 0.25, 0.50, 0.75, 1.00)) +
#   scale_x_continuous(breaks=c(0.00, 0.25 *max.pos, 0.50*max.pos,
#                               0.75*max.pos, max.pos),
#                      labels = fancy_scientific) +
#   expand_limits(y=0)
</pre>

<hr /><div style="text-align: center;">[Package <em><a href='https://genomaths.github.io/MethylIT_utils_HTML_Manual/MethylIT_utils_HTML_Manual.html'>MethylIT.utils</a></em> version 0.3.2 ]</div>
</body></html>
