<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Statistic of Genomic Regions</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="/media/sf_D/<a href='https://genomaths.github.io//MethylIT_utils_HTML_Manual/MethylIT_utils_Manual.html'>MethylIT.utils</a>_utils_html/R.css" />
</head><body>

<table width="100%" summary="page for getGRegionsStat2 {<a href='https://genomaths.github.io//MethylIT_utils_HTML_Manual/MethylIT_utils_Manual.html'>MethylIT.utils</a>}"><tr><td>getGRegionsStat2 {<a href='https://genomaths.github.io//MethylIT_utils_HTML_Manual/MethylIT_utils_Manual.html'>MethylIT.utils</a>}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Statistic of Genomic Regions</h2>

<h3>Description</h3>

<p>A function to estimate the summarized measures of a specified
variable given in a GRanges object (a column from the metacolums of the
GRanges object) after split the GRanges object into intervals.
</p>


<h3>Usage</h3>

<pre>
getGRegionsStat2(GR, win.size = 350, step.size = 350,
  grfeatures = NULL, stat = c("sum", "mean", "gmean", "median",
  "density", "count"), columns = NULL, absolute = FALSE,
  select.strand = NULL, maxgap = -1L, minoverlap = 0L,
  select = "all", ignore.strand = FALSE, type = c("any", "start",
  "end", "within", "equal"), scaling = 1000L, logbase = 2,
  missings = 0, naming = FALSE, na.rm = TRUE, verbose = TRUE, ...)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>GR</code></td>
<td>
<p>A GRange object carying the variables of interest in the 
GRanges metacolumn.</p>
</td></tr>
<tr valign="top"><td><code>win.size</code></td>
<td>
<p>An integer for the size of the windows/regions size of the
intervals of genomics regions.</p>
</td></tr>
<tr valign="top"><td><code>step.size</code></td>
<td>
<p>Interval at which the regions/windows must be defined</p>
</td></tr>
<tr valign="top"><td><code>grfeatures</code></td>
<td>
<p>A GRanges object corresponding to an annotated genomic
feature. For example, gene region, transposable elements, exons,
intergenic region, etc. If provided, then parameters 'win.size' and
step.size are ignored and the statistics are estimated for 'grfeatures'.</p>
</td></tr>
<tr valign="top"><td><code>stat</code></td>
<td>
<p>Statistic used to estimate the summarized value of the variable
of interest in each interval/window. Posible options are: &quot;mean&quot;,
geometric mean (&quot;gmean&quot;), &quot;median&quot;, &quot;density&quot;, &quot;count&quot; and &quot;sum&quot;
(default). Here, we define &quot;density&quot; as the sum of values from the
variable of interest in the given region devided by the length/width of
the region. The option 'count' compute the number/count of positions in
the specified regions with values greater than zero in the selected
'column'.</p>
</td></tr>
<tr valign="top"><td><code>absolute</code></td>
<td>
<p>Optional. Logic (default: FALSE). Whether to use the absolute
values of the variable provided. For example, the difference of
methylation levels could take negative values (TV) and we would be
interested on the sum of abs(TV), which is sum of the total variation
distance.</p>
</td></tr>
<tr valign="top"><td><code>select.strand</code></td>
<td>
<p>Optional. If provided,&quot;+&quot; or &quot;-&quot;, then the summarized
statistic is computed only for the specified DNA chain.</p>
</td></tr>
<tr valign="top"><td><code>maxgap, minoverlap, type</code></td>
<td>
<p>See ?<code>findOverlaps</code> in the
<strong>IRanges</strong> package for a description of these arguments.</p>
</td></tr>
<tr valign="top"><td><code>ignore.strand</code></td>
<td>
<p>When set to TRUE, the strand information is ignored in
the overlap calculations.</p>
</td></tr>
<tr valign="top"><td><code>scaling</code></td>
<td>
<p>integer (default 1). Scaling factor to be used when
stat = &quot;density&quot;. For example, if scaling = 1000, then density * scaling
denotes the sum of values in 1000 bp.</p>
</td></tr>
<tr valign="top"><td><code>logbase</code></td>
<td>
<p>A positive number: the base with respect to which logarithms
are computed when parameter 'entropy = TRUE' (default: logbase = 2).</p>
</td></tr>
<tr valign="top"><td><code>missings</code></td>
<td>
<p>Whether to write '0' or 'NA' on regions where there is not
data to compute the statistic.</p>
</td></tr>
<tr valign="top"><td><code>naming</code></td>
<td>
<p>Logical value. If TRUE, the rows GRanges object will be 
given the names(grfeatures). Default is FALSE.</p>
</td></tr>
<tr valign="top"><td><code>na.rm</code></td>
<td>
<p>Logical value. If TRUE, the NA values will be removed.</p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>Logical. Default is TRUE. If TRUE, then the progress of the
computational tasks is given.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function split a Grange object into intervals genomic regions
(GRs) of fixed size A summarized statistic (mean, median, geometric mean
or sum) is calculated for the specified variable values from each region.
Notice that if win.size == step.size, then non-overlapping windows are
obtained.
</p>


<h3>Value</h3>

<p>A GRanges object with the new genomic regions and their corresponding
summarized statistic.
</p>


<h3>Author(s)</h3>

<p>Robersy Sanchez
</p>


<h3>See Also</h3>

<p><code>getGRegionsStat</code>
</p>


<h3>Examples</h3>

<pre>
library(GenomicRanges)
set.seed(1)
gr &lt;- GRanges(seqnames = Rle( c("chr1", "chr2", "chr3", "chr4"),
            c(5, 5, 5, 5)),
            ranges = IRanges(start = 1:20, end = 1:20),
            strand = rep(c("+", "-"), 10),
            A = seq(1, 0, length = 20))
gr$B &lt;- runif(20)
grs &lt;- getGRegionsStat2(gr, win.size = 4, step.size = 4)
grs

## Selecting the positive strand
grs &lt;- getGRegionsStat2(gr, win.size = 4, step.size = 4, select.strand = "+")
grs

## Selecting the negative strand
grs &lt;- getGRegionsStat2(gr, win.size = 4, step.size = 4, select.strand = "-")
grs

</pre>

<hr /><div style="text-align: center;">[Package <em><a href='https://genomaths.github.io//MethylIT_utils_HTML_Manual/MethylIT_utils_Manual.html'>MethylIT.utils</a></em> version 0.3.1 ]</div>
</body></html>
