<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Amino acid mutation matrix — aa_phychem_index • GenomAutomorphism</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Amino acid mutation matrix — aa_phychem_index"><meta name="description" content="The aminoacid similarity matrices from Amino Acid Index Database
https://www.genome.jp/aaindex/ are provided here.
AAindex (ver.9.2) is a database of
numerical indices representing various physicochemical and biochemical
properties of amino acids and pairs of amino acids.
The similarity of amino acids can be represented numerically, expressed in
terms of observed mutation rate or physicochemical properties. A similarity
matrix, also called a mutation matrix, is a set of 210 numerical values, 20
diagonal and 20x19/2 off-diagonal elements, used for sequence alignments
and similarity searches.
Function aa_phychem_index is wrapper function to call two
other functions: aa_mutmat and aa_index"><meta property="og:description" content="The aminoacid similarity matrices from Amino Acid Index Database
https://www.genome.jp/aaindex/ are provided here.
AAindex (ver.9.2) is a database of
numerical indices representing various physicochemical and biochemical
properties of amino acids and pairs of amino acids.
The similarity of amino acids can be represented numerically, expressed in
terms of observed mutation rate or physicochemical properties. A similarity
matrix, also called a mutation matrix, is a set of 210 numerical values, 20
diagonal and 20x19/2 off-diagonal elements, used for sequence alignments
and similarity searches.
Function aa_phychem_index is wrapper function to call two
other functions: aa_mutmat and aa_index"><meta property="og:image" content="https://genomaths.github.io/genomautomorphism/logo.png"><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomAutomorphism</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.8.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../articles/GenomAutomorphism.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Package functions</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GenomAutomorphism.html">GenomAutomorphism</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree_II.html">A Short Introduction to Algebraic Taxonomy on Genes Regions II</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/Modular_Matrix_Operation_of_Mutational_Events.html">Modular Matrix Operations of Mutational Events</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Amino acid mutation matrix</h1>
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/R/aa_phychem_index.R" class="external-link"><code>R/aa_phychem_index.R</code></a></small>
      <div class="d-none name"><code>aa_phychem_index.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The aminoacid similarity matrices from Amino Acid Index Database
<a href="https://www.genome.jp/aaindex/" class="external-link">https://www.genome.jp/aaindex/</a> are provided here.
<a href="https://www.genome.jp/aaindex/" class="external-link">AAindex</a> (ver.9.2) is a database of
numerical indices representing various physicochemical and biochemical
properties of amino acids and pairs of amino acids.</p>
<p>The similarity of amino acids can be represented numerically, expressed in
terms of observed mutation rate or physicochemical properties. A similarity
matrix, also called a mutation matrix, is a set of 210 numerical values, 20
diagonal and 20x19/2 off-diagonal elements, used for sequence alignments
and similarity searches.</p>
<p>Function <strong><em>aa_phychem_index</em></strong> is wrapper function to call two
other functions: <strong><em>aa_mutmat</em></strong> and <strong><em>aa_index</em></strong></p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">aa_phychem_index</span><span class="op">(</span>acc <span class="op">=</span> <span class="cn">NA</span>, aaindex <span class="op">=</span> <span class="cn">NA</span>, acc_list <span class="op">=</span> <span class="cn">FALSE</span>, info <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">aa_mutmat</span><span class="op">(</span>acc <span class="op">=</span> <span class="cn">NA</span>, aaindex <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"aaindex2"</span>, <span class="st">"aaindex3"</span><span class="op">)</span>, acc_list <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">aa_index</span><span class="op">(</span>acc <span class="op">=</span> <span class="cn">NA</span>, acc_list <span class="op">=</span> <span class="cn">FALSE</span>, info <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-acc">acc<a class="anchor" aria-label="anchor" href="#arg-acc"></a></dt>
<dd><p>Accession id for a specified mutation or contact potential
matrix.</p></dd>


<dt id="arg-aaindex">aaindex<a class="anchor" aria-label="anchor" href="#arg-aaindex"></a></dt>
<dd><p>Database where the requested accession id is locate. The
possible values are:  "aaindex2" or "aaindex3".</p></dd>


<dt id="arg-acc-list">acc_list<a class="anchor" aria-label="anchor" href="#arg-acc-list"></a></dt>
<dd><p>Logical. If TRUE, then the list of available matrices ids
and index names is returned.</p></dd>


<dt id="arg-info">info<a class="anchor" aria-label="anchor" href="#arg-info"></a></dt>
<dd><p>Logical. if TRUE, then whole information for the
physicochemical index will be returned.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Depending on the user specifications, a mutation or contact
potential matrix, a list of available matrices (indices) ids or index
names can be returned. More specifically:</p>
<dl><dt><strong>aa_mutmat</strong>: </dt>
<dd><p>Returns an aminoacid mutation matrix or
a statistical protein contact potentials matrix.</p></dd>

<dt><strong>aa_index</strong>: </dt>
<dd><p>Returns the specified aminoacid physicochemical
indices.</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><a href="aaindex1.html">aaindex1</a>, <a href="aaindex2.html">aaindex2</a>,  <a href="aaindex3.html">aaindex3</a>, and <a href="get_mutscore.html">get_mutscore</a>.</p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Robersy Sanchez <a href="https://genomaths.com" class="external-link">https://genomaths.com</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Load the mutation matrices from database from the packages</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"aaindex1"</span>,<span class="st">"aaindex2"</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span> <span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Get the available mutation matrices</span></span></span>
<span class="r-in"><span><span class="va">mat</span> <span class="op">&lt;-</span> <span class="fu">aa_mutmat</span><span class="op">(</span>aaindex <span class="op">=</span> <span class="st">"aaindex2"</span>, acc_list <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">mat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "List of 94 Amino Acid Matrices in AAindex ver.9.2"                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] ""                                                                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] "The columns correspond to the AAindex accession number and the description of"    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] "each matrix."                                                                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [5] ""                                                                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [6] "ALTS910101 The PAM-120 matrix (Altschul, 1991)"                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] "BENS940101 Log-odds scoring matrix collected in 6.4-8.7 PAM (Benner et al., 1994)"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [8] "BENS940102 Log-odds scoring matrix collected in 22-29 PAM (Benner et al., 1994)"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [9] "BENS940103 Log-odds scoring matrix collected in 74-100 PAM (Benner et al., 1994)" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [10] "BENS940104 Genetic code matrix (Benner et al., 1994)"                             </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Return the 'Base-substitution-protein-stability matrix </span></span></span>
<span class="r-in"><span><span class="co">## (Miyazawa-Jernigan, 1993)'</span></span></span>
<span class="r-in"><span><span class="fu">aa_mutmat</span><span class="op">(</span>acc <span class="op">=</span> <span class="st">"MIYS930101"</span>, aaindex <span class="op">=</span> <span class="st">"aaindex2"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       A     R     N     D     C     Q     E     G     H     I     L     K     M</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A  0.34 -0.08 -0.16  0.04 -0.51 -0.16  0.03  0.19 -0.21 -0.45 -0.45 -0.20 -0.47</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> R -0.08  0.65  0.01 -0.06 -0.35  0.15 -0.02  0.20  0.11 -0.82 -0.74  0.04 -0.83</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> N -0.16  0.01  0.38  0.20 -0.46  0.20  0.19 -0.11  0.21 -0.83 -0.90  0.38 -0.97</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> D  0.04 -0.06  0.20  0.36 -0.41  0.16  0.32  0.15  0.18 -0.78 -0.75  0.16 -0.90</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C -0.51 -0.35 -0.46 -0.41  1.02 -0.74 -0.64 -0.18 -0.29 -0.13 -0.02 -0.81 -0.29</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Q -0.16  0.15  0.20  0.16 -0.74  0.48  0.25 -0.14  0.37 -0.95 -0.74  0.30 -0.97</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> E  0.03 -0.02  0.19  0.32 -0.64  0.25  0.36  0.13  0.13 -0.86 -0.80  0.25 -0.87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> G  0.19  0.20 -0.11  0.15 -0.18 -0.14  0.13  0.43 -0.19 -0.58 -0.56 -0.16 -0.61</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> H -0.21  0.11  0.21  0.18 -0.29  0.37  0.13 -0.19  0.54 -0.69 -0.50  0.14 -0.86</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> I -0.45 -0.82 -0.83 -0.78 -0.13 -0.95 -0.86 -0.58 -0.69  0.75  0.48 -1.01  0.67</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> L -0.45 -0.74 -0.90 -0.75 -0.02 -0.74 -0.80 -0.56 -0.50  0.48  0.61 -1.06  0.50</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> K -0.20  0.04  0.38  0.16 -0.81  0.30  0.25 -0.16  0.14 -1.01 -1.06  0.49 -1.03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M -0.47 -0.83 -0.97 -0.90 -0.29 -0.97 -0.87 -0.61 -0.86  0.67  0.50 -1.03  0.97</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> F -0.47 -0.79 -0.71 -0.62  0.39 -0.85 -0.81 -0.52 -0.43  0.45  0.48 -1.03  0.35</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P  0.17  0.14 -0.12 -0.15 -0.65  0.15 -0.13 -0.14  0.14 -0.78 -0.67 -0.14 -0.82</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S  0.16  0.05 -0.02 -0.14 -0.20 -0.20 -0.19  0.00 -0.15 -0.68 -0.70 -0.14 -0.75</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T  0.18  0.00  0.11 -0.10 -0.62 -0.07 -0.09 -0.09 -0.13 -0.59 -0.77  0.09 -0.61</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> W -0.51 -0.26 -0.79 -0.64  0.82 -0.83 -0.66 -0.15 -0.70 -0.23  0.13 -0.92  0.04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Y -0.34 -0.31  0.12  0.09  0.40 -0.02 -0.11 -0.35  0.35 -0.35 -0.27 -0.12 -0.56</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V -0.06 -0.54 -0.69 -0.39 -0.19 -0.68 -0.42 -0.15 -0.59  0.41  0.41 -0.79  0.41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       F     P     S     T     W     Y     V</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A -0.47  0.17  0.16  0.18 -0.51 -0.34 -0.06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> R -0.79  0.14  0.05  0.00 -0.26 -0.31 -0.54</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> N -0.71 -0.12 -0.02  0.11 -0.79  0.12 -0.69</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> D -0.62 -0.15 -0.14 -0.10 -0.64  0.09 -0.39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C  0.39 -0.65 -0.20 -0.62  0.82  0.40 -0.19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Q -0.85  0.15 -0.20 -0.07 -0.83 -0.02 -0.68</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> E -0.81 -0.13 -0.19 -0.09 -0.66 -0.11 -0.42</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> G -0.52 -0.14  0.00 -0.09 -0.15 -0.35 -0.15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> H -0.43  0.14 -0.15 -0.13 -0.70  0.35 -0.59</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> I  0.45 -0.78 -0.68 -0.59 -0.23 -0.35  0.41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> L  0.48 -0.67 -0.70 -0.77  0.13 -0.27  0.41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> K -1.03 -0.14 -0.14  0.09 -0.92 -0.12 -0.79</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M  0.35 -0.82 -0.75 -0.61  0.04 -0.56  0.41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> F  0.61 -0.76 -0.53 -0.75  0.25  0.14  0.36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P -0.76  0.56  0.24  0.25 -0.71 -0.20 -0.47</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S -0.53  0.24  0.48  0.28 -0.29 -0.04 -0.44</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T -0.75  0.25  0.28  0.45 -0.70 -0.27 -0.44</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> W  0.25 -0.71 -0.29 -0.70  1.42  0.00 -0.17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Y  0.14 -0.20 -0.04 -0.27  0.00  0.84 -0.39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V  0.36 -0.47 -0.44 -0.44 -0.17 -0.39  0.54</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Return the 'BLOSUM80 substitution matrix (Henikoff-Henikoff, 1992)'</span></span></span>
<span class="r-in"><span><span class="fu">aa_mutmat</span><span class="op">(</span>acc <span class="op">=</span> <span class="st">"HENS920103"</span>, aaindex <span class="op">=</span> <span class="st">"aaindex2"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    A  R  N  D  C  Q  E  G  H  I  L  K  M  F  P  S  T  W  Y  V</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A  7 -3 -3 -3 -1 -2 -2  0 -3 -3 -3 -1 -2 -4 -1  2  0 -5 -4 -1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> R -3  9 -1 -3 -6  1 -1 -4  0 -5 -4  3 -3 -5 -3 -2 -2 -5 -4 -4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> N -3 -1  9  2 -5  0 -1 -1  1 -6 -6  0 -4 -6 -4  1  0 -7 -4 -5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> D -3 -3  2 10 -7 -1  2 -3 -2 -7 -7 -2 -6 -6 -3 -1 -2 -8 -6 -6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C -1 -6 -5 -7 13 -5 -7 -6 -7 -2 -3 -6 -3 -4 -6 -2 -2 -5 -5 -2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Q -2  1  0 -1 -5  9  3 -4  1 -5 -4  2 -1 -5 -3 -1 -1 -4 -3 -4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> E -2 -1 -1  2 -7  3  8 -4  0 -6 -6  1 -4 -6 -2 -1 -2 -6 -5 -4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> G  0 -4 -1 -3 -6 -4 -4  9 -4 -7 -7 -3 -5 -6 -5 -1 -3 -6 -6 -6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> H -3  0  1 -2 -7  1  0 -4 12 -6 -5 -1 -4 -2 -4 -2 -3 -4  3 -5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> I -3 -5 -6 -7 -2 -5 -6 -7 -6  7  2 -5  2 -1 -5 -4 -2 -5 -3  4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> L -3 -4 -6 -7 -3 -4 -6 -7 -5  2  6 -4  3  0 -5 -4 -3 -4 -2  1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> K -1  3  0 -2 -6  2  1 -3 -1 -5 -4  8 -3 -5 -2 -1 -1 -6 -4 -4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M -2 -3 -4 -6 -3 -1 -4 -5 -4  2  3 -3  9  0 -4 -3 -1 -3 -3  1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> F -4 -5 -6 -6 -4 -5 -6 -6 -2 -1  0 -5  0 10 -6 -4 -4  0  4 -2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P -1 -3 -4 -3 -6 -3 -2 -5 -4 -5 -5 -2 -4 -6 12 -2 -3 -7 -6 -4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S  2 -2  1 -1 -2 -1 -1 -1 -2 -4 -4 -1 -3 -4 -2  7  2 -6 -3 -3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T  0 -2  0 -2 -2 -1 -2 -3 -3 -2 -3 -1 -1 -4 -3  2  8 -5 -3  0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> W -5 -5 -7 -8 -5 -4 -6 -6 -4 -5 -4 -6 -3  0 -7 -6 -5 16  3 -5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Y -4 -4 -4 -6 -5 -3 -5 -6  3 -3 -2 -4 -3  4 -6 -3 -3  3 11 -3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V -1 -4 -5 -6 -2 -4 -4 -6 -5  4  1 -4  1 -2 -4 -3  0 -5 -3  7</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Using wrapping function</span></span></span>
<span class="r-in"><span><span class="fu">aa_phychem_index</span><span class="op">(</span>acc <span class="op">=</span> <span class="st">"EISD840101"</span>, aaindex <span class="op">=</span> <span class="st">"aaindex1"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     A     R     N     D     C     Q     E     G     H     I     L     K     M </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  0.25 -1.76 -0.64 -0.72  0.04 -0.69 -0.62  0.16 -0.40  0.73  0.53 -1.10  0.26 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     F     P     S     T     W     Y     V </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  0.61 -0.07 -0.26 -0.18  0.37  0.02  0.54 </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Just the info. The information provided after the reference</span></span></span>
<span class="r-in"><span><span class="co">## corresponds to the correlaiton of 'EISD840101' with other indices.</span></span></span>
<span class="r-in"><span><span class="fu">aa_phychem_index</span><span class="op">(</span>acc <span class="op">=</span> <span class="st">"EISD840101"</span>, aaindex <span class="op">=</span> <span class="st">"aaindex1"</span>, info <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] "H EISD840101"                                                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] "D Consensus normalized hydrophobicity scale (Eisenberg, 1984)"                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] "R PMID:6383201"                                                                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] "A Eisenberg, D."                                                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [5] "T Three-dimensional structure of membrane and surface proteins"                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [6] "J Ann. Rev. Biochem. 53, 595-623 (1984) Original references: Eisenberg, D., "    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] "  Weiss, R.M., Terwilliger, T.C. and Wilcox, W. Faraday Symp. Chem. Soc. 17, "   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [8] "  109-120 (1982) Eisenberg, D., Weiss, R.M. and Terwilliger, T.C. The "          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [9] "  hydrophobic moment detects periodicity in protein hydrophobicity Proc. Natl. " </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [10] "  Acad. Sci. USA 81, 140-144 (1984)"                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [11] "C RADA880101    0.968  JACR890101    0.938  RADA880107    0.927"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [12] "  ROSM880105    0.923  WOLR810101    0.914  WOLR790101    0.909"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [13] "  RADA880104    0.908  JANJ790102    0.900  JURD980101    0.895"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [14] "  NADH010102    0.887  CHOC760103    0.885  BLAS910101    0.884"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [15] "  EISD860101    0.884  KYTJ820101    0.878  FAUJ830101    0.875"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16] "  JANJ780102    0.874  OLSK800101    0.869  COWR900101    0.863"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [17] "  NADH010101    0.861  NADH010103    0.840  NAKH900110    0.838"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [18] "  EISD860103    0.837  DESM900102    0.828  RADA880108    0.817"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [19] "  BIOV880102    0.814  BIOV880101    0.811  YUTK870101    0.809"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [20] "  NADH010104    0.809  ROSG850102    0.806  BASU050103    0.806"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [21] "  WOLS870101   -0.820  GRAR740102   -0.823  MEIH800102   -0.829"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [22] "  HOPT810101   -0.846  GUYH850101   -0.849  PUNT030102   -0.854"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [23] "  LEVM760101   -0.859  OOBM770101   -0.878  JANJ780103   -0.881"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [24] "  FAUJ880109   -0.890  GUYH850104   -0.892  CHOC760102   -0.892"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [25] "  KIDA850101   -0.900  JANJ780101   -0.907  KUHL950101   -0.907"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [26] "  PUNT030101   -0.914  VHEG790101   -0.924  ROSM880102   -0.925"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [27] "  ENGD860101   -0.936  PRAM900101   -0.936  ROSM880101   -0.947"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [28] "  GUYH850105   -0.951"                                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [29] "I    A/L     R/K     N/M     D/F     C/P     Q/S     E/T     G/W     H/Y     I/V"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [30] "    0.25   -1.76   -0.64   -0.72    0.04   -0.69   -0.62    0.16   -0.40    0.73"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [31] "    0.53   -1.10    0.26    0.61   -0.07   -0.26   -0.18    0.37    0.02    0.54"</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

