<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Autmorphism Probability — automorphism_prob • GenomAutomorphism</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Autmorphism Probability — automorphism_prob"><meta name="description" content="This function applies a Dirichlet-Multinomial Modelling (in Bayesian
framework) to compute the posterior probability of each
type of mutational event. DNA bases are classified based on the
physicochemical criteria used to ordering the set of codons: number of
hydrogen bonds (strong-weak, S-W), chemical type (purine-pyrimidine, Y-R),
and chemical groups (amino versus keto, M-K) (see reference 4). Preserved
codon positions are labeled with letter “H”.
As a result, mutational events are grouped by type of mutation covering all
the possible combinations of symbols: &quot;Y&quot;, &quot;R&quot;, &quot;M&quot;, &quot;W&quot;, &quot;K&quot;, &quot;S&quot;, and &quot;H&quot;,
for example:  &quot;YSH&quot;, &quot;RSH&quot;, &quot;MSH&quot;, &quot;WSH&quot;, &quot;KSH&quot;, &quot;SSH&quot;, &quot;HSH&quot;, &quot;YHH&quot;, &quot;RHH&quot;,
and so on. Insertion/deletion mutations are not considered.
Maximum Likelihood Estimation (MLE) for Dirichlet Parameters


Given the observed data \(n = (n_1, ..., n_k)\), where \(n_i\) is the
frequency for the mutation type \(i\), we want to estimate the parameters
of the Dirichlet distribution, \(\alpha = (\alpha_1, ..., \alpha_k)\),
that maximize the marginal likelihood.



Marginal Likelihood


The marginal likelihood of the data under the Dirichlet-multinomial model is
given by
$$P\left(n\middle|\alpha\right) = \frac{N!}{\prod_{i=1}^{k}n_i}
\frac{\Gamma\left(\sum_{i=1}^{k}\alpha_i\right)}{\Gamma
\left(N+\sum_{i=1}^{k}\alpha_i\right)}\prod_{i=1}^{k}\frac{\Gamma
\left(n_i+\alpha_i\right)}{\Gamma\left(\alpha_i\right)}$$
where \(N = \sum_{i=1}^{k}n_i\).



Optimization


To perform MLE, we maximize the log of this likelihood:
\(log(P\left(n\middle|\alpha\right))\). That is, we aim to maximize this
log-likelihood with respect to \(\alpha\). This is done numerically
because there's no closed-form solution. Here, we use:
$$Arg\max_{\alpha} {\{log\left(P\left(n\middle|\alpha\right)\right\}}$$
with initial guess set as \(\alpha_i = 1 / (n_i + 1)\).



Posterior Distribution


Let be \(\theta\) the vector of probabilities for each mutation type. It's
the parameter we're estimating, which represents the probabilities of
observing each mutation type in the multinomial distribution. The conjugate
distribution in this context refers to the Dirichlet distribution, which is
the prior distribution for \(\theta\). When we have observed data, the
posterior distribution of \(\theta\) given this data is also a Dirichlet
distribution due to the conjugate property.
The prior distribution, before observing any data, our belief about the
distribution of \(\theta\) is given by:
$$\theta\sim\ Dirichlet\left(\alpha_1,\alpha_k,\ldots,\alpha_k\right)$$
where the \(\alpha_i\) are known as initial concentration parameters,
which represents our initial belief or assumption about the distribution
of the mutation types. These parameters control how the probability mass is
distributed among the mutation types. We might set these initial parameters
based on some prior knowledge or simply to provide a non-informative prior.
Once we have the MLE for \(\alpha\), the posterior distribution of
\(\theta\) given the data updates these parameters to incorporate the
observed data:
$$\theta\mid\ n\sim\ Dirichlet\left(\alpha_1^\ast+n1,\alpha_k^\ast+n_2,
\ldots,\alpha_k^\ast+n_k\right)$$
where \(\alpha^\ast_i\) are the MLE estimates, i.e.,
\(\alpha_i^\ast+n_i\) are our updated belief about the frequencies of
mutation types in the population sample.
The expected values of \(\theta_i\), given the data under the posterior
Dirichlet distribution is:
$$E\left[\theta_i\right]=\frac{\alpha_i^*+n_i}
{\sum_{j=1}^{k}\left(\alpha_j^*+n_j\right)}$$
These expected values directly corresponds to the &quot;posterior probability&quot; of
observing mutation type \(i\).
This approach provides a rigorous estimation of the Dirichlet parameters
under the Dirichlet-multinomial model using MLE.

"><meta property="og:description" content="This function applies a Dirichlet-Multinomial Modelling (in Bayesian
framework) to compute the posterior probability of each
type of mutational event. DNA bases are classified based on the
physicochemical criteria used to ordering the set of codons: number of
hydrogen bonds (strong-weak, S-W), chemical type (purine-pyrimidine, Y-R),
and chemical groups (amino versus keto, M-K) (see reference 4). Preserved
codon positions are labeled with letter “H”.
As a result, mutational events are grouped by type of mutation covering all
the possible combinations of symbols: &quot;Y&quot;, &quot;R&quot;, &quot;M&quot;, &quot;W&quot;, &quot;K&quot;, &quot;S&quot;, and &quot;H&quot;,
for example:  &quot;YSH&quot;, &quot;RSH&quot;, &quot;MSH&quot;, &quot;WSH&quot;, &quot;KSH&quot;, &quot;SSH&quot;, &quot;HSH&quot;, &quot;YHH&quot;, &quot;RHH&quot;,
and so on. Insertion/deletion mutations are not considered.
Maximum Likelihood Estimation (MLE) for Dirichlet Parameters


Given the observed data \(n = (n_1, ..., n_k)\), where \(n_i\) is the
frequency for the mutation type \(i\), we want to estimate the parameters
of the Dirichlet distribution, \(\alpha = (\alpha_1, ..., \alpha_k)\),
that maximize the marginal likelihood.



Marginal Likelihood


The marginal likelihood of the data under the Dirichlet-multinomial model is
given by
$$P\left(n\middle|\alpha\right) = \frac{N!}{\prod_{i=1}^{k}n_i}
\frac{\Gamma\left(\sum_{i=1}^{k}\alpha_i\right)}{\Gamma
\left(N+\sum_{i=1}^{k}\alpha_i\right)}\prod_{i=1}^{k}\frac{\Gamma
\left(n_i+\alpha_i\right)}{\Gamma\left(\alpha_i\right)}$$
where \(N = \sum_{i=1}^{k}n_i\).



Optimization


To perform MLE, we maximize the log of this likelihood:
\(log(P\left(n\middle|\alpha\right))\). That is, we aim to maximize this
log-likelihood with respect to \(\alpha\). This is done numerically
because there's no closed-form solution. Here, we use:
$$Arg\max_{\alpha} {\{log\left(P\left(n\middle|\alpha\right)\right\}}$$
with initial guess set as \(\alpha_i = 1 / (n_i + 1)\).



Posterior Distribution


Let be \(\theta\) the vector of probabilities for each mutation type. It's
the parameter we're estimating, which represents the probabilities of
observing each mutation type in the multinomial distribution. The conjugate
distribution in this context refers to the Dirichlet distribution, which is
the prior distribution for \(\theta\). When we have observed data, the
posterior distribution of \(\theta\) given this data is also a Dirichlet
distribution due to the conjugate property.
The prior distribution, before observing any data, our belief about the
distribution of \(\theta\) is given by:
$$\theta\sim\ Dirichlet\left(\alpha_1,\alpha_k,\ldots,\alpha_k\right)$$
where the \(\alpha_i\) are known as initial concentration parameters,
which represents our initial belief or assumption about the distribution
of the mutation types. These parameters control how the probability mass is
distributed among the mutation types. We might set these initial parameters
based on some prior knowledge or simply to provide a non-informative prior.
Once we have the MLE for \(\alpha\), the posterior distribution of
\(\theta\) given the data updates these parameters to incorporate the
observed data:
$$\theta\mid\ n\sim\ Dirichlet\left(\alpha_1^\ast+n1,\alpha_k^\ast+n_2,
\ldots,\alpha_k^\ast+n_k\right)$$
where \(\alpha^\ast_i\) are the MLE estimates, i.e.,
\(\alpha_i^\ast+n_i\) are our updated belief about the frequencies of
mutation types in the population sample.
The expected values of \(\theta_i\), given the data under the posterior
Dirichlet distribution is:
$$E\left[\theta_i\right]=\frac{\alpha_i^*+n_i}
{\sum_{j=1}^{k}\left(\alpha_j^*+n_j\right)}$$
These expected values directly corresponds to the &quot;posterior probability&quot; of
observing mutation type \(i\).
This approach provides a rigorous estimation of the Dirichlet parameters
under the Dirichlet-multinomial model using MLE.

"><meta property="og:image" content="https://genomaths.github.io/genomautomorphism/logo.png"><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomAutomorphism</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.8.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../articles/GenomAutomorphism.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Package functions</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GenomAutomorphism.html">GenomAutomorphism</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree_II.html">A Short Introduction to Algebraic Taxonomy on Genes Regions II</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/Modular_Matrix_Operation_of_Mutational_Events.html">Modular Matrix Operations of Mutational Events</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Autmorphism Probability</h1>
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/R/automorphism_prob.R" class="external-link"><code>R/automorphism_prob.R</code></a></small>
      <div class="d-none name"><code>automorphism_prob.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function applies a Dirichlet-Multinomial Modelling (in Bayesian
framework) to compute the posterior probability of each
<em>type of mutational event</em>. DNA bases are classified based on the
physicochemical criteria used to ordering the set of codons: number of
hydrogen bonds (strong-weak, S-W), chemical type (purine-pyrimidine, Y-R),
and chemical groups (amino versus keto, M-K) (see reference 4). Preserved
codon positions are labeled with letter “H”.</p>
<p>As a result, mutational events are grouped by type of mutation covering all
the possible combinations of symbols: "Y", "R", "M", "W", "K", "S", and "H",
for example:  "YSH", "RSH", "MSH", "WSH", "KSH", "SSH", "HSH", "YHH", "RHH",
and so on. Insertion/deletion mutations are not considered.</p><div class="section">
<h3 id="maximum-likelihood-estimation-mle-for-dirichlet-parameters">Maximum Likelihood Estimation (MLE) for Dirichlet Parameters<a class="anchor" aria-label="anchor" href="#maximum-likelihood-estimation-mle-for-dirichlet-parameters"></a></h3>


<p>Given the observed data \(n = (n_1, ..., n_k)\), where \(n_i\) is the
frequency for the mutation type \(i\), we want to estimate the parameters
of the Dirichlet distribution, \(\alpha = (\alpha_1, ..., \alpha_k)\),
that maximize the marginal likelihood.</p>
</div>

<div class="section">
<h3 id="marginal-likelihood">Marginal Likelihood<a class="anchor" aria-label="anchor" href="#marginal-likelihood"></a></h3>


<p>The marginal likelihood of the data under the Dirichlet-multinomial model is
given by</p>
<p>$$P\left(n\middle|\alpha\right) = \frac{N!}{\prod_{i=1}^{k}n_i}
\frac{\Gamma\left(\sum_{i=1}^{k}\alpha_i\right)}{\Gamma
\left(N+\sum_{i=1}^{k}\alpha_i\right)}\prod_{i=1}^{k}\frac{\Gamma
\left(n_i+\alpha_i\right)}{\Gamma\left(\alpha_i\right)}$$</p>
<p>where \(N = \sum_{i=1}^{k}n_i\).</p>
</div>

<div class="section">
<h3 id="optimization">Optimization<a class="anchor" aria-label="anchor" href="#optimization"></a></h3>


<p>To perform MLE, we maximize the log of this likelihood:
\(log(P\left(n\middle|\alpha\right))\). That is, we aim to maximize this
log-likelihood with respect to \(\alpha\). This is done numerically
because there's no closed-form solution. Here, we use:</p>
<p>$$Arg\max_{\alpha} {\{log\left(P\left(n\middle|\alpha\right)\right\}}$$</p>
<p>with initial guess set as \(\alpha_i = 1 / (n_i + 1)\).</p>
</div>

<div class="section">
<h3 id="posterior-distribution">Posterior Distribution<a class="anchor" aria-label="anchor" href="#posterior-distribution"></a></h3>


<p>Let be \(\theta\) the vector of probabilities for each mutation type. It's
the parameter we're estimating, which represents the probabilities of
observing each mutation type in the multinomial distribution. The conjugate
distribution in this context refers to the Dirichlet distribution, which is
the prior distribution for \(\theta\). When we have observed data, the
posterior distribution of \(\theta\) given this data is also a Dirichlet
distribution due to the conjugate property.</p>
<p>The prior distribution, before observing any data, our belief about the
distribution of \(\theta\) is given by:</p>
<p>$$\theta\sim\ Dirichlet\left(\alpha_1,\alpha_k,\ldots,\alpha_k\right)$$</p>
<p>where the \(\alpha_i\) are known as initial concentration parameters,
which represents our initial belief or assumption about the distribution
of the mutation types. These parameters control how the probability mass is
distributed among the mutation types. We might set these initial parameters
based on some prior knowledge or simply to provide a non-informative prior.</p>
<p>Once we have the MLE for \(\alpha\), the posterior distribution of
\(\theta\) given the data updates these parameters to incorporate the
observed data:</p>
<p>$$\theta\mid\ n\sim\ Dirichlet\left(\alpha_1^\ast+n1,\alpha_k^\ast+n_2,
\ldots,\alpha_k^\ast+n_k\right)$$</p>
<p>where \(\alpha^\ast_i\) are the MLE estimates, i.e.,
\(\alpha_i^\ast+n_i\) are our updated belief about the frequencies of
mutation types in the population sample.</p>
<p>The expected values of \(\theta_i\), given the data under the posterior
Dirichlet distribution is:</p>
<p>$$E\left[\theta_i\right]=\frac{\alpha_i^*+n_i}
{\sum_{j=1}^{k}\left(\alpha_j^*+n_j\right)}$$</p>
<p>These expected values directly corresponds to the "posterior probability" of
observing mutation type \(i\).</p>
<p>This approach provides a rigorous estimation of the Dirichlet parameters
under the Dirichlet-multinomial model using MLE.</p>
</div>

    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">automorphism_prob</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'AutomorphismByCoef'</span></span>
<span><span class="fu">automorphism_prob</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  initial_alpha <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"L-BFGS-B"</span>, <span class="st">"Nelder-Mead"</span>, <span class="st">"BFGS"</span>, <span class="st">"CG"</span>, <span class="st">"SANN"</span>, <span class="st">"Brent"</span><span class="op">)</span>,</span>
<span>  maxit <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  abstol <span class="op">=</span> <span class="fl">10</span><span class="op">^</span><span class="op">-</span><span class="fl">8</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'AutomorphismByCoefList'</span></span>
<span><span class="fu">automorphism_prob</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  initial_alpha <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"L-BFGS-B"</span>, <span class="st">"Nelder-Mead"</span>, <span class="st">"BFGS"</span>, <span class="st">"CG"</span>, <span class="st">"SANN"</span>, <span class="st">"Brent"</span><span class="op">)</span>,</span>
<span>  maxit <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  abstol <span class="op">=</span> <span class="fl">10</span><span class="op">^</span><span class="op">-</span><span class="fl">8</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>An AutomorphismByCoefList-class object returned by function
<a href="automorphism_bycoef.html">automorphism_bycoef</a>.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Not in use yet.</p></dd>


<dt id="arg-initial-alpha">initial_alpha<a class="anchor" aria-label="anchor" href="#arg-initial-alpha"></a></dt>
<dd><p>A vector of initial guess values for pseudo counts
\(\alpha_i\) (see Description). Default is: \(\alpha_i = 1 / (n_i + 1)\),
which corresponds to \(initial_alpha = NULL\).</p></dd>


<dt id="arg-method-maxit-abstol">method, maxit, abstol<a class="anchor" aria-label="anchor" href="#arg-method-maxit-abstol"></a></dt>
<dd><p>Parameter values to pass into
<code><a href="https://rdrr.io/r/stats/optim.html" class="external-link">optim</a></code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A data frame with the posterior probabilities.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><a href="automorphism_bycoef.html">automorphism_bycoef</a></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Load the data set</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"autby_coef"</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">post_prob</span> <span class="op">&lt;-</span> <span class="fu">automorphism_prob</span><span class="op">(</span><span class="va">autby_coef</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">post_prob</span>,<span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    Codon frequency     alpha Posterior_Probability</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1    RHH        27 1.3362573            0.17597995</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2    HHY        25 1.3362573            0.16355911</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3    HRH        25 1.3362573            0.16355911</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4    HHR         9 1.3362573            0.06419246</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5    HMH         9 1.3362573            0.06419246</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6    SHH         9 1.3362573            0.06419246</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7    HYH         4 1.3362573            0.03314038</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8    HSH         3 1.3362573            0.02692996</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9    MHH         3 1.3362573            0.02692996</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10   YHH         1 0.7934929            0.01113834</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

