<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Compute the Automorphisms of Mutational Events Between two Codon
Sequences Represented in a Given Abelian group. — automorphism • GenomAutomorphism</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Compute the Automorphisms of Mutational Events Between two Codon
Sequences Represented in a Given Abelian group. — automorphism" />
<meta property="og:description" content="Given two codon sequences represented in a given Abelian group,
this function computes the automorphisms describing codon mutational events.
Basically, this function is a wrapping to call the corresponding function
for a specified Abelian group." />
<meta property="og:image" content="https://genomaths.github.io/GenomAutomorphism/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomAutomorphism</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/GenomAutomorphism.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/GenomAutomorphism.html">GenomAutomorphism</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/GenomAutomorphism/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Compute the Automorphisms of Mutational Events Between two Codon
Sequences Represented in a Given Abelian group.</h1>
    <small class="dont-index">Source: <a href='https://github.com/genomaths/GenomAutomorphism/blob/master/R/automorphism.R'><code>R/automorphism.R</code></a></small>
    <div class="hidden name"><code>automorphism.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given two codon sequences represented in a given Abelian group,
this function computes the automorphisms describing codon mutational events.
Basically, this function is a wrapping to call the corresponding function
for a specified Abelian group.</p>
    </div>

    <pre class="usage"><span class='fu'>automorphism</span><span class='op'>(</span>seqs <span class='op'>=</span> <span class='cn'>NULL</span>, filepath <span class='op'>=</span> <span class='cn'>NULL</span>, group <span class='op'>=</span> <span class='st'>"Z4"</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S4 method for DNAStringSet_OR_NULL</span>
<span class='fu'>automorphism</span><span class='op'>(</span>
  seqs <span class='op'>=</span> <span class='cn'>NULL</span>,
  filepath <span class='op'>=</span> <span class='cn'>NULL</span>,
  group <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Z5"</span>, <span class='st'>"Z64"</span>, <span class='st'>"Z125"</span>, <span class='st'>"Z5^3"</span><span class='op'>)</span>,
  cube <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ACGT"</span>, <span class='st'>"TGCA"</span><span class='op'>)</span>,
  cube_alt <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"CATG"</span>, <span class='st'>"GTAC"</span><span class='op'>)</span>,
  nms <span class='op'>=</span> <span class='cn'>NULL</span>,
  start <span class='op'>=</span> <span class='cn'>NA</span>,
  end <span class='op'>=</span> <span class='cn'>NA</span>,
  chr <span class='op'>=</span> <span class='fl'>1L</span>,
  strand <span class='op'>=</span> <span class='st'>"+"</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>filepath</th>
      <td><p>A character vector containing the path to a file in
<em><strong>fasta</strong></em> format to be read. This argument must be given if
<em>codon &amp; base</em> arguments are not provided.</p></td>
    </tr>
    <tr>
      <th>group</th>
      <td><p>A character string denoting the group representation for the
given base or codon as shown in reference (1).</p></td>
    </tr>
    <tr>
      <th>cube, cube_alt</th>
      <td><p>A character string denoting pairs of the 24 Genetic-code
cubes, as given in references (2-3). That is, the base pairs from the given
cubes must be complementary each other. Such a cube pair are call
\(dual cubes\) and, as shown in reference (3), each pair integrates group.</p></td>
    </tr>
    <tr>
      <th>nms</th>
      <td><p>Optional. Only used if the DNA sequence alignment provided carries
more than two sequences. A character string giving short names for the
alignments to be compared. If not given then the automorphisms between
pairwise alignment are named as: "aln_1", "aln_2", and so on.</p></td>
    </tr>
    <tr>
      <th>start, end, chr, strand</th>
      <td><p>Optional parameters required to build a
<code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges-class</a></code>. If not provided the default
values given for the function definition will be used.</p></td>
    </tr>
    <tr>
      <th>seq</th>
      <td><p>An object from a <code><a href='https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html'>DNAStringSet</a></code> or
<code><a href='https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html'>DNAMultipleAlignment</a></code> class carrying the DNA
pairwise alignment of two sequences. The pairwise alignment provided in
argument <em><strong>seq</strong></em> or the 'fasta' file <em><strong>filepath</strong></em>
must correspond to codon sequences.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>This function returns a <code><a href='Automorphism-class.html'>Automorphism-class</a></code> object
with four columns on its metacolumn named: <em>seq1</em>, <em>seq2</em>,
<em>autm</em>, and <em>cube</em>.</p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Methods</h3>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a><code><a href='automorphismByRanges.html'>automorphismByRanges</a></code>:</h4>


<p>This function returns a <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges-class</a></code> object.
Consecutive mutational events (on the codon sequence) described by
automorphisms on a same cube are grouped in a range.</p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a><code><a href='automorphismByCoef.html'>automorphismByCoef</a></code></h4>


<p>This function returns a <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges-class</a></code> object.
Consecutive mutational events (on the codon sequence) described by
the same automorphism coefficients are grouped in a range.</p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a><code><a href='getAutomorphisms.html'>getAutomorphisms</a></code></h4>


<p>This function returns an AutomorphismList-class object as a list of
Automorphism-class objects, which inherits from
<code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges-class</a></code> objects.</p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a><code><a href='conserved_regions.html'>conserved_regions</a></code></h4>


<p>Returns a <code><a href='Automorphism-class.html'>AutomorphismByCoef</a></code> class object containing the
requested regions.</p>



    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Herein, automorphisms are algebraic descriptions of mutational event
observed in codon sequences represented on different Abelian groups. In
particular, as described in references (3-4), for each representation of the
codon set on a defined Abelian group there are 24 possible isomorphic Abelian
groups. These Abelian groups can be labeled based on the DNA base-order used
to generate them. The set of 24 Abelian groups can be described as a group
isomorphic to the symmetric group of degree four (\(S_4\), see reference
(4)). Function <code><a href='automorphismByRanges.html'>automorphismByRanges</a></code> permits the classification
of the pairwise alignment of protein-coding sub-regions based on the
mutational events observed on it and on the genetic-code cubes that describe
them.</p>
<p>Automorphisms in Z5, Z64 and Z125 are described as functions \(f(x) =
k x mod 64\) and \(f(x) = k x mod 125\), where k and x are elements from the
set of integers modulo 64 or modulo 125, respectively. If an automorphism
cannot be found on any of the cubes provided in the argument \(cube\), then
function <code>automorphism</code> will search for automorphisms in the
cubes provided in the argument \(cube_alt\).</p>
<p>Automorphisms in Z5^3' are described as functions \(f(x) = Ax mod Z5\),
where A is diagonal matrix.</p>
<p>Arguments <em><strong>cube</strong></em> and <em><strong>cube_alt</strong></em> must be
pairs of' dual cubes (see section 2.4 from reference 4).</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    
<ol>
<li><p>Sanchez R, Morgado E, Grau R. Gene algebra from a genetic code
algebraic structure. J Math Biol. 2005 Oct;51(4):431-57.
doi: 10.1007/s00285-005-0332-8. Epub 2005 Jul 13. PMID: 16012800. (
<a href='https://arxiv.org/pdf/q-bio/0412033.pdf'>PDF</a>).</p></li>
<li><p>Robersy Sanchez, Jesús Barreto (2021) Genomic Abelian Finite
Groups.
<a href='https://doi.org/10.1101/2021.06.01.446543'>doi: 10.1101/2021.06.01.446543</a>.</p></li>
<li><p>M. V José, E.R. Morgado, R. Sánchez, T. Govezensky, The 24 possible
algebraic representations of the standard genetic code in six or in three
dimensions, Adv. Stud. Biol. 4 (2012) 119–152.<a href='https://is.gd/na9eap'>PDF</a>.</p></li>
<li><p>R. Sanchez. Symmetric Group of the Genetic–Code Cubes. Effect of the
Genetic–Code Architecture on the Evolutionary Process MATCH Commun. Math.
Comput. Chem. 79 (2018) 527-560. <a href='https://bit.ly/2Z9mjM7'>PDF</a></p></li>
</ol>

    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='autZ64.html'>autZ64</a></code>.</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Load a pairwise alignment</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>aln</span><span class='op'>)</span>
<span class='va'>aln</span>
</div><div class='output co'>#&gt; DNAStringSet object of length 2:
#&gt;     width seq
#&gt; [1]    51 <span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFFF;'>GG</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span>---<span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFAFAF;'>AA</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFFF;'>GG</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>T</span>
#&gt; [2]    51 <span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFFF;'>GG</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFAFAF;'>AA</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFFF;'>GGG</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span>------<span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span></div><div class='input'>
<span class='co'>## Automorphism on "Z5^3"</span>
<span class='va'>autms</span> <span class='op'>&lt;-</span> <span class='fu'>automorphism</span><span class='op'>(</span>seq <span class='op'>=</span> <span class='va'>aln</span>, group <span class='op'>=</span> <span class='st'>"Z5^3"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='error'>Error in validObject(.Object): invalid class “CodonGroup” object: </span>
#&gt; <span class='error'>    *** Argument 'x' is not a CodonGroup-class object.The slot 'group' is</span>
#&gt; <span class='error'>    not present or wrong naming.</span></div><div class='input'><span class='va'>autms</span>
</div><div class='output co'>#&gt; Automorphism object with 7100 ranges and 6 metadata columns:
#&gt;          seqnames    ranges strand |        seq1        seq2    coord1
#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt;
#&gt;      [1]        1         1      + |         ATG         ATG        50
#&gt;      [2]        1         2      + |         GAG         GAG        10
#&gt;      [3]        1         3      + |         AGC         AGC        33
#&gt;      [4]        1         4      + |         CTT         CTT        55
#&gt;      [5]        1         5      + |         GTT         GTC        59
#&gt;      ...      ...       ...    ... .         ...         ...       ...
#&gt;   [7096]        1      7096      + |         CTT         CTT        55
#&gt;   [7097]        1      7097      + |         GTT         GTT        59
#&gt;   [7098]        1      7098      + |         AAT         AAC         3
#&gt;   [7099]        1      7099      + |         AAC         AAC         1
#&gt;   [7100]        1      7100      + |         TAA         TAA        12
#&gt;             coord2      autm        cube
#&gt;          &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt;
#&gt;      [1]        50         1        ACGT
#&gt;      [2]        10         1        ACGT
#&gt;      [3]        33         1        ACGT
#&gt;      [4]        55         1        ACGT
#&gt;      [5]        57        27        ACGT
#&gt;      ...       ...       ...         ...
#&gt;   [7096]        55         1        ACGT
#&gt;   [7097]        59         1        ACGT
#&gt;   [7098]         1        43        ACGT
#&gt;   [7099]         1         1        ACGT
#&gt;   [7100]        12         1        ACGT
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>## Automorphism on "Z64"</span>
<span class='va'>autms</span> <span class='op'>&lt;-</span> <span class='fu'>automorphism</span><span class='op'>(</span>seq <span class='op'>=</span> <span class='va'>aln</span>, group <span class='op'>=</span> <span class='st'>"Z64"</span><span class='op'>)</span>
<span class='va'>autms</span>
</div><div class='output co'>#&gt; Automorphism object with 17 ranges and 6 metadata columns:
#&gt;        seqnames    ranges strand |        seq1        seq2    coord1    coord2
#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;
#&gt;    [1]        1         1      + |         ACC         ATC        17        49
#&gt;    [2]        1         2      + |         TAT         TAT        15        15
#&gt;    [3]        1         3      + |         GTT         GTT        59        59
#&gt;    [4]        1         4      + |         GGT         GGT        43        43
#&gt;    [5]        1         5      + |         ATT         ATT        51        51
#&gt;    ...      ...       ...    ... .         ...         ...       ...       ...
#&gt;   [13]        1        13      + |         TCT         TCC        31        29
#&gt;   [14]        1        14      + |         AGC         ---        52        NA
#&gt;   [15]        1        15      + |         TCA         ---         9        NA
#&gt;   [16]        1        16      - |         CTA         CTC        52        53
#&gt;   [17]        1        17      + |         CAT         CTT         7        55
#&gt;             autm        cube
#&gt;        &lt;numeric&gt; &lt;character&gt;
#&gt;    [1]        33        ACGT
#&gt;    [2]         1        ACGT
#&gt;    [3]         1        ACGT
#&gt;    [4]         1        ACGT
#&gt;    [5]         1        ACGT
#&gt;    ...       ...         ...
#&gt;   [13]         3        ACGT
#&gt;   [14]        NA        None
#&gt;   [15]        NA        None
#&gt;   [16]        30        TGCA
#&gt;   [17]        17        ACGT
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>## Grouping into ranges the automorphisms by cubes </span>
<span class='fu'><a href='automorphismByRanges.html'>automorphismByRanges</a></span><span class='op'>(</span><span class='va'>autms</span><span class='op'>)</span>
</div><div class='output co'>#&gt; GRanges object with 6 ranges and 1 metadata column:
#&gt;       seqnames    ranges strand |        cube
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
#&gt;   [1]        1       1-5      + |        ACGT
#&gt;   [2]        1         6      + |        None
#&gt;   [3]        1      7-13      + |        ACGT
#&gt;   [4]        1     14-15      + |        None
#&gt;   [5]        1        16      - |        TGCA
#&gt;   [6]        1        17      + |        ACGT
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>## Automorphism on "Z64" from position 1 to 33</span>
<span class='va'>autms</span> <span class='op'>&lt;-</span> <span class='fu'>automorphism</span><span class='op'>(</span>seq <span class='op'>=</span> <span class='va'>aln</span>,
                      group <span class='op'>=</span> <span class='st'>"Z64"</span>,
                      start <span class='op'>=</span> <span class='fl'>1</span>,
                      end <span class='op'>=</span> <span class='fl'>33</span><span class='op'>)</span>
<span class='va'>autms</span>
</div><div class='output co'>#&gt; Automorphism object with 11 ranges and 6 metadata columns:
#&gt;        seqnames    ranges strand |        seq1        seq2    coord1    coord2
#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;
#&gt;    [1]        1         1      + |         ACC         ATC        17        49
#&gt;    [2]        1         2      + |         TAT         TAT        15        15
#&gt;    [3]        1         3      + |         GTT         GTT        59        59
#&gt;    [4]        1         4      + |         GGT         GGT        43        43
#&gt;    [5]        1         5      + |         ATT         ATT        51        51
#&gt;    [6]        1         6      + |         ---         ACG        NA         7
#&gt;    [7]        1         7      + |         GCG         ACG        26        18
#&gt;    [8]        1         8      + |         CTC         CTC        53        53
#&gt;    [9]        1         9      + |         CAA         CAA         4         4
#&gt;   [10]        1        10      + |         CTC         TTC        53        61
#&gt;   [11]        1        11      + |         CTT         CTT        55        55
#&gt;             autm        cube
#&gt;        &lt;numeric&gt; &lt;character&gt;
#&gt;    [1]        33        ACGT
#&gt;    [2]         1        ACGT
#&gt;    [3]         1        ACGT
#&gt;    [4]         1        ACGT
#&gt;    [5]         1        ACGT
#&gt;    [6]        NA        None
#&gt;    [7]        45        ACGT
#&gt;    [8]         1        ACGT
#&gt;    [9]         1        ACGT
#&gt;   [10]        41        ACGT
#&gt;   [11]         1        ACGT
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>## Grouping into ranges the automorphisms by cubes </span>
<span class='fu'><a href='automorphismByRanges.html'>automorphismByRanges</a></span><span class='op'>(</span><span class='va'>autms</span><span class='op'>)</span>
</div><div class='output co'>#&gt; GRanges object with 3 ranges and 1 metadata column:
#&gt;       seqnames    ranges strand |        cube
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
#&gt;   [1]        1       1-5      + |        ACGT
#&gt;   [2]        1         6      + |        None
#&gt;   [3]        1      7-11      + |        ACGT
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://genomaths.com'>Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


