<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='This function computes the weighted Manhattan distance between
codons from two sequences as given in reference (1). That is, given two
codons \(x\) and \(y\) with coordinates on the set of integers modulo 5
("Z5"): \(x = (x_1, x_2, x_3)\) and  \(x = (y_1, y_2, y_3)\) (see (1)),
the Weighted Manhattan distance between this two codons is defined as:
$$d_w(x,y) = |x_1 - y_1|/5 + |x_2 - y_2| + |x_3 -y_3|/25$$
If the codon coordinates are given on "Z4", then the Weighted Manhattan
distance is define as:
$$d_w(x,y) = |x_1 - y_1|/4 + |x_2 - y_2| + |x_3 -y_3|/16$$
Herein, we move to the generalized version given in reference (3), for
which:
$$d_w(x,y) = |x_1 - y_1| w_1  + |x_2 - y_2| w_2 + |x_3 -y_3| w_3$$
where we use the vector of \(weight = (w_1, w_2, w_3)\).'><title>Weighted Manhattan Distance Between Codons — codon_dist • GenomAutomorphism</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Weighted Manhattan Distance Between Codons — codon_dist"><meta property="og:description" content='This function computes the weighted Manhattan distance between
codons from two sequences as given in reference (1). That is, given two
codons \(x\) and \(y\) with coordinates on the set of integers modulo 5
("Z5"): \(x = (x_1, x_2, x_3)\) and  \(x = (y_1, y_2, y_3)\) (see (1)),
the Weighted Manhattan distance between this two codons is defined as:
$$d_w(x,y) = |x_1 - y_1|/5 + |x_2 - y_2| + |x_3 -y_3|/25$$
If the codon coordinates are given on "Z4", then the Weighted Manhattan
distance is define as:
$$d_w(x,y) = |x_1 - y_1|/4 + |x_2 - y_2| + |x_3 -y_3|/16$$
Herein, we move to the generalized version given in reference (3), for
which:
$$d_w(x,y) = |x_1 - y_1| w_1  + |x_2 - y_2| w_2 + |x_3 -y_3| w_3$$
where we use the vector of \(weight = (w_1, w_2, w_3)\).'><meta property="og:image" content="https://genomaths.github.io/genomautomorphism/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GenomAutomorphism</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/GenomAutomorphism.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Package functions</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/GenomAutomorphism.html">GenomAutomorphism</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_and_decision_tree_II.html">A Short Introduction to Algebraic Taxonomy on Genes Regions II</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Weighted Manhattan Distance Between Codons</h1>
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/R/codon_dist.R" class="external-link"><code>R/codon_dist.R</code></a></small>
      <div class="d-none name"><code>codon_dist.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function computes the weighted Manhattan distance between
codons from two sequences as given in reference (1). That is, given two
codons \(x\) and \(y\) with coordinates on the set of integers modulo 5
("Z5"): \(x = (x_1, x_2, x_3)\) and  \(x = (y_1, y_2, y_3)\) (see (1)),
the Weighted Manhattan distance between this two codons is defined as:</p>
<p>$$d_w(x,y) = |x_1 - y_1|/5 + |x_2 - y_2| + |x_3 -y_3|/25$$</p>
<p>If the codon coordinates are given on "Z4", then the Weighted Manhattan
distance is define as:</p>
<p>$$d_w(x,y) = |x_1 - y_1|/4 + |x_2 - y_2| + |x_3 -y_3|/16$$</p>
<p>Herein, we move to the generalized version given in reference (3), for
which:</p>
<p>$$d_w(x,y) = |x_1 - y_1| w_1  + |x_2 - y_2| w_2 + |x_3 -y_3| w_3$$</p>
<p>where we use the vector of \(weight = (w_1, w_2, w_3)\).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">codon_dist</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for DNAStringSet</span></span>
<span><span class="fu">codon_dist</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  weight <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Z4"</span>, <span class="st">"Z5"</span><span class="op">)</span>,</span>
<span>  cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"AGCT"</span>, <span class="st">"TCGA"</span>, <span class="st">"TGCA"</span>, <span class="st">"CATG"</span>, <span class="st">"GTAC"</span>, <span class="st">"CTAG"</span>, <span class="st">"GATC"</span>, <span class="st">"ACTG"</span>,</span>
<span>    <span class="st">"ATCG"</span>, <span class="st">"GTCA"</span>, <span class="st">"GCTA"</span>, <span class="st">"CAGT"</span>, <span class="st">"TAGC"</span>, <span class="st">"TGAC"</span>, <span class="st">"CGAT"</span>, <span class="st">"AGTC"</span>, <span class="st">"ATGC"</span>, <span class="st">"CGTA"</span>,</span>
<span>    <span class="st">"CTGA"</span>, <span class="st">"GACT"</span>, <span class="st">"GCAT"</span>, <span class="st">"TACG"</span>, <span class="st">"TCAG"</span><span class="op">)</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for character</span></span>
<span><span class="fu">codon_dist</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">y</span>,</span>
<span>  weight <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Z4"</span>, <span class="st">"Z5"</span><span class="op">)</span>,</span>
<span>  cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"AGCT"</span>, <span class="st">"TCGA"</span>, <span class="st">"TGCA"</span>, <span class="st">"CATG"</span>, <span class="st">"GTAC"</span>, <span class="st">"CTAG"</span>, <span class="st">"GATC"</span>, <span class="st">"ACTG"</span>,</span>
<span>    <span class="st">"ATCG"</span>, <span class="st">"GTCA"</span>, <span class="st">"GCTA"</span>, <span class="st">"CAGT"</span>, <span class="st">"TAGC"</span>, <span class="st">"TGAC"</span>, <span class="st">"CGAT"</span>, <span class="st">"AGTC"</span>, <span class="st">"ATGC"</span>, <span class="st">"CGTA"</span>,</span>
<span>    <span class="st">"CTGA"</span>, <span class="st">"GACT"</span>, <span class="st">"GCAT"</span>, <span class="st">"TACG"</span>, <span class="st">"TCAG"</span><span class="op">)</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for CodonGroup_OR_Automorphisms</span></span>
<span><span class="fu">codon_dist</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  weight <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Z4"</span>, <span class="st">"Z5"</span><span class="op">)</span>,</span>
<span>  cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"AGCT"</span>, <span class="st">"TCGA"</span>, <span class="st">"TGCA"</span>, <span class="st">"CATG"</span>, <span class="st">"GTAC"</span>, <span class="st">"CTAG"</span>, <span class="st">"GATC"</span>, <span class="st">"ACTG"</span>,</span>
<span>    <span class="st">"ATCG"</span>, <span class="st">"GTCA"</span>, <span class="st">"GCTA"</span>, <span class="st">"CAGT"</span>, <span class="st">"TAGC"</span>, <span class="st">"TGAC"</span>, <span class="st">"CGAT"</span>, <span class="st">"AGTC"</span>, <span class="st">"ATGC"</span>, <span class="st">"CGTA"</span>,</span>
<span>    <span class="st">"CTGA"</span>, <span class="st">"GACT"</span>, <span class="st">"GCAT"</span>, <span class="st">"TACG"</span>, <span class="st">"TCAG"</span><span class="op">)</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>x, y</dt>
<dd><p>A character string of codon sequences, i.e., sequences of DNA
base-triplets. If only 'x' argument is given, then it must be a
<code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet-class</a></code> object.</p></dd>


<dt>...</dt>
<dd><p>Not in use yet.</p></dd>


<dt>weight</dt>
<dd><p>A numerical vector of  weights to compute weighted Manhattan
distance between codons. If \(weight = NULL\), then
\(weight = (1/4,1,1/16)\) for \(group = "Z4"\) and
\(weight = (1/5,1,1/25)\) for \(group = "Z5"\).</p></dd>


<dt>group</dt>
<dd><p>A character string denoting the group representation for the
given codon sequence as shown in reference (2-3).</p></dd>


<dt>cube</dt>
<dd><p>A character string denoting one of the 24 Genetic-code cubes,
as given in references (2-3).</p></dd>


<dt>num.cores, tasks</dt>
<dd><p>Parameters for parallel computation using package
<code><a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallel-package.html" class="external-link">BiocParallel-package</a></code>: the number of cores to
use, i.e. at most how many child processes will be run simultaneously (see
<code><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></code> and the number of tasks per job (only
for Linux OS).</p></dd>


<dt>verbose</dt>
<dd><p>If TRUE, prints the progress bar.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A numerical vector with the pairwise distances between codons in
sequences 'x' and 'y'.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    
<ol><li><p>Sanchez R. Evolutionary Analysis of DNA-Protein-Coding Regions Based
on a Genetic Code Cube Metric. Curr Top Med Chem. 2014;14: 407–417.
<a href="https://doi.org/10.2174/1568026613666131204110022" class="external-link">https://doi.org/10.2174/1568026613666131204110022</a>.</p></li>
<li><p>M. V Jose, E.R. Morgado, R. Sanchez, T. Govezensky, The 24 possible
algebraic representations of the standard genetic code in six or in three
dimensions, Adv. Stud. Biol. 4 (2012) 119-152.<a href="https://is.gd/na9eap" class="external-link">PDF</a>.</p></li>
<li><p>R. Sanchez. Symmetric Group of the Genetic-Code Cubes. Effect of the
Genetic-Code Architecture on the Evolutionary Process MATCH Commun. Math.
Comput. Chem. 79 (2018) 527-560. <a href="https://is.gd/ZY1Gx8" class="external-link">PDF</a>.</p></li>
</ol></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="codon_dist_matrix.html">codon_dist_matrix</a></code>, <code><a href="automorphisms.html">automorphisms</a></code>,
<code><a href="codon_coord.html">codon_coord</a></code>, and <code><a href="aminoacid_dist.html">aminoacid_dist</a></code>.</p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Let's write two small DNA sequences</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">=</span> <span class="st">"ACGCGTGTACCGTGACTG"</span></span></span>
<span class="r-in"><span><span class="va">y</span> <span class="op">=</span> <span class="st">"TGCGCCCGTGACGCGTGA"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">codon_dist</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, group <span class="op">=</span> <span class="st">"Z5"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 1.64 1.28 1.32 1.24 1.28 1.48</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Alternatively, data can be vectors of codons, i.e., vectors of DNA </span></span></span>
<span class="r-in"><span><span class="co">## base-triplets (including gaps simbol "-").</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACG"</span>,<span class="st">"CGT"</span>,<span class="st">"GTA"</span>,<span class="st">"CCG"</span>,<span class="st">"TGA"</span>,<span class="st">"CTG"</span>,<span class="st">"ACG"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">y</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TGC"</span>,<span class="st">"GCC"</span>,<span class="st">"CGT"</span>,<span class="st">"GAC"</span>,<span class="st">"---"</span>,<span class="st">"TGA"</span>,<span class="st">"A-G"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Gaps are not defined on "Z4"</span></span></span>
<span class="r-in"><span><span class="fu">codon_dist</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, group <span class="op">=</span> <span class="st">"Z4"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.8750 0.4375 0.5000 0.3750     NA 0.6875     NA</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Gaps are considered on "Z5"</span></span></span>
<span class="r-in"><span><span class="fu">codon_dist</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, group <span class="op">=</span> <span class="st">"Z5"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 1.64 1.28 1.32 1.24 3.84 1.48 2.00</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Load an Automorphism-class object</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">autm</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">codon_dist</span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">autm</span>,<span class="fl">20</span><span class="op">)</span>, group <span class="op">=</span> <span class="st">"Z4"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] 0.0000 0.0000 0.0000 0.0000 0.1250 0.1250 0.0000 0.2500 0.0000 0.0000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [11] 0.0000 0.0000 0.0000 0.1875 0.0625 0.0000 0.0000 0.0000 0.0000 0.1250</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Load a pairwise alignment</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">aln</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">aln</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet object of length 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]    51 <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span>---<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2]    51 <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span>------<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">codon_dist</span><span class="op">(</span>x <span class="op">=</span> <span class="va">aln</span>, group <span class="op">=</span> <span class="st">"Z5"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] 2.00 0.00 0.00 0.00 0.00 2.32 0.40 0.00 0.00 0.40 0.00 0.04 0.08 3.28 2.84</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [16] 0.04 3.00</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

