<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Codon Coordinate Matrix — codon_matrix • GenomAutomorphism</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Codon Coordinate Matrix — codon_matrix"><meta name="description" content="This function build the coordinate matrix for each sequence from an aligned
set of DNA codon sequences."><meta property="og:description" content="This function build the coordinate matrix for each sequence from an aligned
set of DNA codon sequences."><meta property="og:image" content="https://genomaths.github.io/genomautomorphism/logo.png"><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomAutomorphism</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.8.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../articles/GenomAutomorphism.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Package functions</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GenomAutomorphism.html">GenomAutomorphism</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree_II.html">A Short Introduction to Algebraic Taxonomy on Genes Regions II</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/Modular_Matrix_Operation_of_Mutational_Events.html">Modular Matrix Operations of Mutational Events</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Codon Coordinate Matrix</h1>
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/R/codon_matrix.R" class="external-link"><code>R/codon_matrix.R</code></a></small>
      <div class="d-none name"><code>codon_matrix.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function build the coordinate matrix for each sequence from an aligned
set of DNA codon sequences.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">codon_matrix</span><span class="op">(</span><span class="va">base</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'BaseSeqMatrix'</span></span>
<span><span class="fu">codon_matrix</span><span class="op">(</span><span class="va">base</span>, num.cores <span class="op">=</span> <span class="fl">1L</span>, tasks <span class="op">=</span> <span class="fl">0L</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'DNAStringSet'</span></span>
<span><span class="fu">codon_matrix</span><span class="op">(</span></span>
<span>  <span class="va">base</span>,</span>
<span>  cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"AGCT"</span>, <span class="st">"TCGA"</span>, <span class="st">"TGCA"</span>, <span class="st">"CATG"</span>, <span class="st">"GTAC"</span>, <span class="st">"CTAG"</span>, <span class="st">"GATC"</span>, <span class="st">"ACTG"</span>,</span>
<span>    <span class="st">"ATCG"</span>, <span class="st">"GTCA"</span>, <span class="st">"GCTA"</span>, <span class="st">"CAGT"</span>, <span class="st">"TAGC"</span>, <span class="st">"TGAC"</span>, <span class="st">"CGAT"</span>, <span class="st">"AGTC"</span>, <span class="st">"ATGC"</span>, <span class="st">"CGTA"</span>,</span>
<span>    <span class="st">"CTGA"</span>, <span class="st">"GACT"</span>, <span class="st">"GCAT"</span>, <span class="st">"TACG"</span>, <span class="st">"TCAG"</span><span class="op">)</span>,</span>
<span>  group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Z4"</span>, <span class="st">"Z5"</span><span class="op">)</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'DNAMultipleAlignment'</span></span>
<span><span class="fu">codon_matrix</span><span class="op">(</span></span>
<span>  <span class="va">base</span>,</span>
<span>  cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"AGCT"</span>, <span class="st">"TCGA"</span>, <span class="st">"TGCA"</span>, <span class="st">"CATG"</span>, <span class="st">"GTAC"</span>, <span class="st">"CTAG"</span>, <span class="st">"GATC"</span>, <span class="st">"ACTG"</span>,</span>
<span>    <span class="st">"ATCG"</span>, <span class="st">"GTCA"</span>, <span class="st">"GCTA"</span>, <span class="st">"CAGT"</span>, <span class="st">"TAGC"</span>, <span class="st">"TGAC"</span>, <span class="st">"CGAT"</span>, <span class="st">"AGTC"</span>, <span class="st">"ATGC"</span>, <span class="st">"CGTA"</span>,</span>
<span>    <span class="st">"CTGA"</span>, <span class="st">"GACT"</span>, <span class="st">"GCAT"</span>, <span class="st">"TACG"</span>, <span class="st">"TCAG"</span><span class="op">)</span>,</span>
<span>  group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Z4"</span>, <span class="st">"Z5"</span><span class="op">)</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-base">base<a class="anchor" aria-label="anchor" href="#arg-base"></a></dt>
<dd><p>A <code><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">DNAMultipleAlignment</a></code>, a
<code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></code>, or a <a href="BaseSeqMatrix.html">BaseSeqMatrix</a>.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Not in use yet.</p></dd>


<dt id="arg-num-cores-tasks">num.cores, tasks<a class="anchor" aria-label="anchor" href="#arg-num-cores-tasks"></a></dt>
<dd><p>Parameters for parallel computation using package
<code><a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallel-package.html" class="external-link">BiocParallel-package</a></code>: the number of cores to
use, i.e. at most how many child processes will be run simultaneously (see
<code><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></code> and the number of tasks per job (only
for Linux OS).</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>If TRUE, prints the function log to stdout</p></dd>


<dt id="arg-cube">cube<a class="anchor" aria-label="anchor" href="#arg-cube"></a></dt>
<dd><p>A character string denoting one of the 24 Genetic-code cubes,
as given in references (3-4).</p></dd>


<dt id="arg-group">group<a class="anchor" aria-label="anchor" href="#arg-group"></a></dt>
<dd><p>A character string denoting the group representation for the
given base or codon as shown in reference (3-4).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <a href="ListCodonMatrix.html">ListCodonMatrix</a> class object with the codon coordinate on its
metacolumns.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The purpose of this function is making the codon coordinates from multiple
sequence alignments (MSA) available for further downstream statistical
analyses, like those reported in references (1) and (2).</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>

<ol><li><p>Lorenzo-Ginori, Juan V., Aníbal Rodríguez-Fuentes, Ricardo Grau
Ábalo, and Robersy Sánchez Rodríguez. "Digital signal processing in the
analysis of genomic sequences." Current Bioinformatics 4, no. 1 (2009):
28-40.</p></li>
<li><p>Sanchez, Robersy. "Evolutionary analysis of DNA-protein-coding
regions based on a genetic code cube metric." Current Topics in Medicinal
Chemistry 14, no. 3 (2014): 407-417.</p></li>
<li><p>Robersy Sanchez, Jesus Barreto (2021) Genomic Abelian Finite
Groups.
<a href="https://doi.org/10.1101/2021.06.01.446543" class="external-link">doi: 10.1101/2021.06.01.446543</a></p></li>
<li><p>M. V Jose, E.R. Morgado, R. Sanchez, T. Govezensky, The 24 possible
algebraic representations of the standard genetic code in six or in three
dimensions, Adv. Stud. Biol. 4 (2012) 119-152.<a href="https://is.gd/na9eap" class="external-link">PDF</a>.</p></li>
<li><p>R. Sanchez. Symmetric Group of the Genetic-Code Cubes. Effect of the
Genetic-Code Architecture on the Evolutionary Process MATCH Commun. Math.
Comput. Chem. 79 (2018) 527-560.</p></li>
</ol><ol><li><p></p></li>
<li><p></p></li>
</ol></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><a href="codon_coord.html">codon_coord</a>, <a href="base_methods.html">base_coord</a> and <a href="base2int.html">base2int</a>.</p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Robersy Sanchez <a href="https://genomaths.com" class="external-link">https://genomaths.com</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Load the MSA of Primate BRCA1 DNA repair genes</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"brca1_aln"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Get the DNAStringSet for the first 33 codons and apply 'codon_matrix'</span></span></span>
<span class="r-in"><span><span class="va">brca1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/MaskedXString-class.html" class="external-link">unmasked</a></span><span class="op">(</span><span class="va">brca1_aln</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">brca1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/XVector/man/XVector-class.html" class="external-link">subseq</a></span><span class="op">(</span><span class="va">brca1</span>, start <span class="op">=</span> <span class="fl">1</span>, end <span class="op">=</span> <span class="fl">33</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">codon_matrix</span><span class="op">(</span><span class="va">brca1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ListCodonMatrix object of length: 11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Seq.Alias: codon.1 codon.2 codon.3 codon.4 codon.5 codon.6 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $codon.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CodonMatrix object with 3 ranges and 20 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        S1        S2        S3        S4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]        1       1-3      + |         0         0         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]        1       1-3      + |         3         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]        1       1-3      + |         2         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              S5        S6        S7        S8        S9       S10       S11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]         0         0         0         0         0         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]         3         3         3         3         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]         2         2         2         2         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             S12       S13       S14       S15       S16       S17       S18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]         0         0         0         0         0         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]         3         3         3         3         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]         2         2         2         2         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             S19       S20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $codon.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CodonMatrix object with 3 ranges and 20 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seqnames    ranges strand |        S1        S2        S3        S4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4        1       4-6      + |         2         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5        1       4-6      + |         0         0         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   6        1       4-6      + |         3         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            S5        S6        S7        S8        S9       S10       S11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4         2         2         2         2         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5         0         0         0         0         0         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   6         3         3         3         3         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           S12       S13       S14       S15       S16       S17       S18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4         2         2         2         2         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5         0         0         0         0         0         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   6         3         3         3         3         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           S19       S20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   6         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;9 more CodonMatrix element(s)&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Three slots: 'DataList', 'group', 'cube' &amp; 'seq_alias'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Get back the alignment object and apply 'codon_matrix' gives us the </span></span></span>
<span class="r-in"><span><span class="co">## same result.</span></span></span>
<span class="r-in"><span><span class="va">brca1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">DNAMultipleAlignment</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">brca1</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">codon_matrix</span><span class="op">(</span><span class="va">brca1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ListCodonMatrix object of length: 11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Seq.Alias: codon.1 codon.2 codon.3 codon.4 codon.5 codon.6 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $codon.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CodonMatrix object with 3 ranges and 20 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        S1        S2        S3        S4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]        1       1-3      + |         0         0         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]        1       1-3      + |         3         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]        1       1-3      + |         2         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              S5        S6        S7        S8        S9       S10       S11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]         0         0         0         0         0         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]         3         3         3         3         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]         2         2         2         2         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             S12       S13       S14       S15       S16       S17       S18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]         0         0         0         0         0         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]         3         3         3         3         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]         2         2         2         2         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             S19       S20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $codon.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CodonMatrix object with 3 ranges and 20 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seqnames    ranges strand |        S1        S2        S3        S4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4        1       4-6      + |         2         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5        1       4-6      + |         0         0         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   6        1       4-6      + |         3         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            S5        S6        S7        S8        S9       S10       S11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4         2         2         2         2         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5         0         0         0         0         0         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   6         3         3         3         3         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           S12       S13       S14       S15       S16       S17       S18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4         2         2         2         2         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5         0         0         0         0         0         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   6         3         3         3         3         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           S19       S20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   4         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   5         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   6         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;9 more CodonMatrix element(s)&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Three slots: 'DataList', 'group', 'cube' &amp; 'seq_alias'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

