<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Conserved and Non-conserved Regions from a MSA — conserved_regions • GenomAutomorphism</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Conserved and Non-conserved Regions from a MSA — conserved_regions"><meta name="description" content="Returns the Conserved or the Non-conserved Regions from a MSA."><meta property="og:description" content="Returns the Conserved or the Non-conserved Regions from a MSA."><meta property="og:image" content="https://genomaths.github.io/genomautomorphism/logo.png"><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomAutomorphism</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.8.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../articles/GenomAutomorphism.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Package functions</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GenomAutomorphism.html">GenomAutomorphism</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree_II.html">A Short Introduction to Algebraic Taxonomy on Genes Regions II</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/Modular_Matrix_Operation_of_Mutational_Events.html">Modular Matrix Operations of Mutational Events</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Conserved and Non-conserved Regions from a MSA</h1>
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/R/conserved_regions.R" class="external-link"><code>R/conserved_regions.R</code></a></small>
      <div class="d-none name"><code>conserved_regions.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Returns the Conserved or the Non-conserved Regions from a MSA.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">conserved_regions</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'Automorphism'</span></span>
<span><span class="fu">conserved_regions</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  conserved <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"all_pairs"</span>, <span class="st">"unique_pairs"</span>, <span class="st">"unique"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'AutomorphismList'</span></span>
<span><span class="fu">conserved_regions</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  conserved <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"all_pairs"</span>, <span class="st">"unique_pairs"</span>, <span class="st">"unique"</span><span class="op">)</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fu">multicoreWorkers</span><span class="op">(</span><span class="op">)</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'AutomorphismByCoef'</span></span>
<span><span class="fu">conserved_regions</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  conserved <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"all_pairs"</span>, <span class="st">"unique_pairs"</span>, <span class="st">"unique"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'AutomorphismByCoefList'</span></span>
<span><span class="fu">conserved_regions</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  conserved <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"all_pairs"</span>, <span class="st">"unique_pairs"</span>, <span class="st">"unique"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>A <code><a href="Automorphism.html">Automorphism-class</a></code>, a
<code><a href="AutomorphismList.html">AutomorphismList-class</a></code>,
a <code><a href="AutomorphismByCoef.html">AutomorphismByCoef</a></code> or a
<code><a href="AutomorphismByCoefList.html">AutomorphismByCoefList</a></code> class object.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Not in use.</p></dd>


<dt id="arg-conserved">conserved<a class="anchor" aria-label="anchor" href="#arg-conserved"></a></dt>
<dd><p>Logical, Whether to return the <em>conserved</em> or the
<em>non-conserved regions</em>.</p></dd>


<dt id="arg-output">output<a class="anchor" aria-label="anchor" href="#arg-output"></a></dt>
<dd><p>A character string. Type of output.</p></dd>


<dt id="arg-num-cores-tasks">num.cores, tasks<a class="anchor" aria-label="anchor" href="#arg-num-cores-tasks"></a></dt>
<dd><p>Integers. Argument <em>num.cores</em> denotes the
number of cores to use, i.e. at most how many child processes will be run
simultaneously (see <code><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></code> function from
BiocParallel package). Argument <em>tasks</em> denotes the number of tasks per
job. value must be a scalar integer &gt;= 0L. In this documentation a job is
defined as a single call to a function, such as
<code><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></code>. A task is the division of the \(X\)
argument into chunks. When tasks == 0 (default), \(X\) is divided as
evenly as possible over the number of workers (see
<code><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html" class="external-link">MulticoreParam</a></code> from BiocParallel package).</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>logic(1). If TRUE, enable progress bar.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <code><a href="AutomorphismByCoef.html">AutomorphismByCoef</a></code> class object containing the
requested regions.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Load dataset</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"autm"</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">conserved_regions</span><span class="op">(</span><span class="va">autm</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> AutomorphismByCoef object with 3 ranges and 7 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        seq1        seq2         aa1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]        1       1-3      + |         ATG         ATG           M</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]        1       1-3      + |         GAG         GAG           E</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]        1       1-3      + |         AGC         AGC           S</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               aa2      autm    mut_type        cube</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;character&gt; &lt;numeric&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]           M         1         HHH        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]           E         1         HHH        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]           S         1         HHH        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span><span class="co">## Load automorphism found COVID datatset</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"covid_autm"</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Conserved regions in the first 100 codons</span></span></span>
<span class="r-in"><span><span class="va">conserv</span> <span class="op">&lt;-</span> <span class="fu">conserved_regions</span><span class="op">(</span><span class="va">covid_autm</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span>, output <span class="op">=</span> <span class="st">"unique"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">conserv</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 7 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |      autm        cube</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]        1      1-19      + |         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]        1     21-44      + |         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]        1     47-52      + |         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]        1     54-60      + |         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]        1     62-80      + |         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [6]        1     83-91      + |         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [7]        1    93-100      + |         1        ACGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

