<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Some Physicochemical Properties of DNA bases — dna_phyche • GenomAutomorphism</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Some Physicochemical Properties of DNA bases — dna_phyche"><meta name="description" content="This data set carries some relevant physicochemical properties of the
DNA bases. Available properties are:
&quot;proton_affinity: &quot;
It is an indicatio of the thermodynamic
gradient between a molecule and the anionic form of that molecule upon
removal of a proton from it
(Wikipedia). The
proton affinity values, given in kJ/mol, were taken from reference (1),
also available in Wolfram Alpha at https://www.wolframalpha.com/ and in
the cell phone App 'Wolfram Alpha'.. Reference (2) provides several
measurements accomplished by several computational and experimental
approaches.

&quot;partition_coef: &quot;
1-octanol/water partition coefficients, logP.
In the physical sciences, a partition coefficient (P) or distribution
coefficient (D) is the ratio of concentrations of a compound in a mixture
of two immiscible solvents at equilibrium (3). The partition coefficient
measures how hydrophilic (&quot;water-loving&quot;) or hydrophobic (&quot;water-fearing&quot;)
a chemical substance is. Partition coefficients are useful in estimating
the distribution of drugs within the body. Hydrophobic drugs with high
octanol-water partition coefficients are mainly distributed to hydrophobic
areas such as lipid bilayers of cells. Conversely, hydrophilic drugs
(low octanol/water partition coefficients) are found primarily in aqueous
regions such as blood serum. The partition coefficient values included
here were taken from reference (1), also available in Wolfram Alpha at
https://www.wolframalpha.com/  and in the cell phone App 'Wolfram Alpha'.

&quot;dipole_moment: &quot;
Dipole-dipole, dipole-induced-dipole and London
force interactions among the bases in the helix are large, and make the
free energy of the helix depend on the base composition and sequence. The
dipole moment values were taken from reference (4). The dipole moment of
DNA bases refers to the measure of polarity in the chemical bonds between
atoms within the nucleobases. Dipole moments arise due to differences in
electronegativity between the bonded atoms. In DNA bases, these dipole
moments can influence the orientation of the bases when interacting with
other molecules or surfaces, such as graphene/h-BN interfaces. The concept
of dipole moments has been applied to analyze the electric moments of
RNA-binding proteins, which can help identify DNA-binding proteins and
provide insights into their mechanisms and prediction.

&quot;tautomerization_energy: &quot;
The term “tautomerism” is usually
defined as structural isomerism with a low barrier to interconversion
between the isomers, for example, the enol/imino forms for cytosine and
guanine. Tautomerization is a process where the chemical structure of a
molecule, such as DNA bases, undergoes a rearrangement of its atoms. This
rearrangement results in the formation of different isomers, called
tautomers, which can exist in solution or in a cell. The DNA bases can
undergo tautomeric shifts, which can potentially contribute to mutagenic
mispairings during DNA replication. The energy required for
tautomerization of DNA bases is known as tautomerization energy. These
values were taken from reference (2) and the value for each base
corresponds to the average of the values estimated from different
measurement approaches.



"><meta property="og:description" content="This data set carries some relevant physicochemical properties of the
DNA bases. Available properties are:
&quot;proton_affinity: &quot;
It is an indicatio of the thermodynamic
gradient between a molecule and the anionic form of that molecule upon
removal of a proton from it
(Wikipedia). The
proton affinity values, given in kJ/mol, were taken from reference (1),
also available in Wolfram Alpha at https://www.wolframalpha.com/ and in
the cell phone App 'Wolfram Alpha'.. Reference (2) provides several
measurements accomplished by several computational and experimental
approaches.

&quot;partition_coef: &quot;
1-octanol/water partition coefficients, logP.
In the physical sciences, a partition coefficient (P) or distribution
coefficient (D) is the ratio of concentrations of a compound in a mixture
of two immiscible solvents at equilibrium (3). The partition coefficient
measures how hydrophilic (&quot;water-loving&quot;) or hydrophobic (&quot;water-fearing&quot;)
a chemical substance is. Partition coefficients are useful in estimating
the distribution of drugs within the body. Hydrophobic drugs with high
octanol-water partition coefficients are mainly distributed to hydrophobic
areas such as lipid bilayers of cells. Conversely, hydrophilic drugs
(low octanol/water partition coefficients) are found primarily in aqueous
regions such as blood serum. The partition coefficient values included
here were taken from reference (1), also available in Wolfram Alpha at
https://www.wolframalpha.com/  and in the cell phone App 'Wolfram Alpha'.

&quot;dipole_moment: &quot;
Dipole-dipole, dipole-induced-dipole and London
force interactions among the bases in the helix are large, and make the
free energy of the helix depend on the base composition and sequence. The
dipole moment values were taken from reference (4). The dipole moment of
DNA bases refers to the measure of polarity in the chemical bonds between
atoms within the nucleobases. Dipole moments arise due to differences in
electronegativity between the bonded atoms. In DNA bases, these dipole
moments can influence the orientation of the bases when interacting with
other molecules or surfaces, such as graphene/h-BN interfaces. The concept
of dipole moments has been applied to analyze the electric moments of
RNA-binding proteins, which can help identify DNA-binding proteins and
provide insights into their mechanisms and prediction.

&quot;tautomerization_energy: &quot;
The term “tautomerism” is usually
defined as structural isomerism with a low barrier to interconversion
between the isomers, for example, the enol/imino forms for cytosine and
guanine. Tautomerization is a process where the chemical structure of a
molecule, such as DNA bases, undergoes a rearrangement of its atoms. This
rearrangement results in the formation of different isomers, called
tautomers, which can exist in solution or in a cell. The DNA bases can
undergo tautomeric shifts, which can potentially contribute to mutagenic
mispairings during DNA replication. The energy required for
tautomerization of DNA bases is known as tautomerization energy. These
values were taken from reference (2) and the value for each base
corresponds to the average of the values estimated from different
measurement approaches.



"><meta property="og:image" content="https://genomaths.github.io/genomautomorphism/logo.png"><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomAutomorphism</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.8.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../articles/GenomAutomorphism.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Package functions</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GenomAutomorphism.html">GenomAutomorphism</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree_II.html">A Short Introduction to Algebraic Taxonomy on Genes Regions II</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/Modular_Matrix_Operation_of_Mutational_Events.html">Modular Matrix Operations of Mutational Events</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Some Physicochemical Properties of DNA bases</h1>
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/R/data_dna_phyche.R" class="external-link"><code>R/data_dna_phyche.R</code></a></small>
      <div class="d-none name"><code>dna_phyche.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This data set carries some relevant physicochemical properties of the
DNA bases. Available properties are:</p><dl><dt>"proton_affinity: "</dt>
<dd><p>It is an indicatio of the thermodynamic
gradient between a molecule and the anionic form of that molecule upon
removal of a proton from it
(<a href="https://en.wikipedia.org/wiki/Proton_affinity" class="external-link">Wikipedia</a>). The
proton affinity values, given in kJ/mol, were taken from reference (1),
also available in Wolfram Alpha at https://www.wolframalpha.com/ and in
the cell phone App 'Wolfram Alpha'.. Reference (2) provides several
measurements accomplished by several computational and experimental
approaches.</p></dd>

<dt>"partition_coef: "</dt>
<dd><p>1-octanol/water partition coefficients, logP.
In the physical sciences, a partition coefficient (P) or distribution
coefficient (D) is the ratio of concentrations of a compound in a mixture
of two immiscible solvents at equilibrium (3). The partition coefficient
measures how hydrophilic ("water-loving") or hydrophobic ("water-fearing")
a chemical substance is. Partition coefficients are useful in estimating
the distribution of drugs within the body. Hydrophobic drugs with high
octanol-water partition coefficients are mainly distributed to hydrophobic
areas such as lipid bilayers of cells. Conversely, hydrophilic drugs
(low octanol/water partition coefficients) are found primarily in aqueous
regions such as blood serum. The partition coefficient values included
here were taken from reference (1), also available in Wolfram Alpha at
https://www.wolframalpha.com/  and in the cell phone App 'Wolfram Alpha'.</p></dd>

<dt>"dipole_moment: "</dt>
<dd><p>Dipole-dipole, dipole-induced-dipole and London
force interactions among the bases in the helix are large, and make the
free energy of the helix depend on the base composition and sequence. The
dipole moment values were taken from reference (4). The dipole moment of
DNA bases refers to the measure of polarity in the chemical bonds between
atoms within the nucleobases. Dipole moments arise due to differences in
electronegativity between the bonded atoms. In DNA bases, these dipole
moments can influence the orientation of the bases when interacting with
other molecules or surfaces, such as graphene/h-BN interfaces. The concept
of dipole moments has been applied to analyze the electric moments of
RNA-binding proteins, which can help identify DNA-binding proteins and
provide insights into their mechanisms and prediction.</p></dd>

<dt>"tautomerization_energy: "</dt>
<dd><p>The term “tautomerism” is usually
defined as structural isomerism with a low barrier to interconversion
between the isomers, for example, the enol/imino forms for cytosine and
guanine. Tautomerization is a process where the chemical structure of a
molecule, such as DNA bases, undergoes a rearrangement of its atoms. This
rearrangement results in the formation of different isomers, called
tautomers, which can exist in solution or in a cell. The DNA bases can
undergo tautomeric shifts, which can potentially contribute to mutagenic
mispairings during DNA replication. The energy required for
tautomerization of DNA bases is known as tautomerization energy. These
values were taken from reference (2) and the value for each base
corresponds to the average of the values estimated from different
measurement approaches.</p></dd>


</dl></div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"dna_phyche"</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="format">Format<a class="anchor" aria-label="anchor" href="#format"></a></h2>
    <p>A data frame object.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>

<ol><li><p>Wolfram Research (2007), ChemicalData, Wolfram Language function,
https://reference.wolfram.com/language/ref/ChemicalData.html (updated 2016).</p></li>
<li><p>Moser A, Range K, York DM. Accurate proton affinity and gas-phase
basicity values for molecules important in biocatalysis. J Phys Chem B.
2010;114: 13911–13921. doi:10.1021/jp107450n.</p></li>
<li><p>Leo A, Hansch C, Elkins D. Partition coefficients and their uses. Chem
Rev. 1971;71: 525–616. doi:10.1021/cr60274a001.</p></li>
<li><p>Vovusha H, Amorim RG, Scheicher RH, Sanyal B. Controlling the orientation
of nucleobases by dipole moment interaction with graphene/h-BN interfaces.
RSC Adv. Royal Society of Chemistry; 2018;8: 6527–6531.
doi:10.1039/c7ra11664k.</p></li>
</ol></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"dna_phyche"</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dna_phyche</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   proton_affinity partition_coef dipole_moment tautomerization_energy</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A           942.8           -0.3          2.51                  12.68</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C           949.9           -1.1          5.58                   2.47</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> G           959.5           -0.9          6.65                   0.76</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> T           880.9           -0.7          4.37                  12.26</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Select DNA base tautomerization energy</span></span></span>
<span class="r-in"><span><span class="va">te</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">dna_phyche</span><span class="op">$</span><span class="va">tautomerization_energy</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">te</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dna_phyche</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Let's create DNAStringSet-class object</span></span></span>
<span class="r-in"><span><span class="va">base</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span> seq1 <span class="op">=</span><span class="st">'ACGTGATCAAGT'</span>,</span></span>
<span class="r-in"><span>                            seq2 <span class="op">=</span> <span class="st">'GTGTGATCCAGT'</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="dna_phychem.html">dna_phychem</a></span><span class="op">(</span>seqs <span class="op">=</span> <span class="va">base</span>, phychem <span class="op">=</span> <span class="va">te</span>,</span></span>
<span class="r-in"><span>            index_name <span class="op">=</span> <span class="st">"Tautomerization-Energy"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MatrixSeq with 2 rows (sequences) and 12 columns (aminoacids/codons):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       B1    B2   B3    B4   B5    B6    B7   B8    B9   B10  B11   B12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S1 12.68  2.47 0.76 12.26 0.76 12.68 12.26 2.47 12.68 12.68 0.76 12.26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S2  0.76 12.26 0.76 12.26 0.76 12.68 12.26 2.47  2.47 12.68 0.76 12.26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slots: 'seqs', 'matrix', 'names', 'aaindex', 'phychem', 'accession</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

