<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>DNA base/codon sequence and coordinates represented on a given Abelian group. — get_coord • GenomAutomorphism</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="DNA base/codon sequence and coordinates represented on a given Abelian group. — get_coord"><meta name="description" content="Given a string denoting a codon or base from the DNA (or RNA)
alphabet and a genetic-code Abelian group as given in reference (1), this
function returns an object from CodonGroup-class carrying the
DNA base/codon sequence and coordinates represented on the given Abelian
group."><meta property="og:description" content="Given a string denoting a codon or base from the DNA (or RNA)
alphabet and a genetic-code Abelian group as given in reference (1), this
function returns an object from CodonGroup-class carrying the
DNA base/codon sequence and coordinates represented on the given Abelian
group."><meta property="og:image" content="https://genomaths.github.io/genomautomorphism/logo.png"><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomAutomorphism</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.8.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../articles/GenomAutomorphism.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Package functions</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GenomAutomorphism.html">GenomAutomorphism</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree_II.html">A Short Introduction to Algebraic Taxonomy on Genes Regions II</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/Modular_Matrix_Operation_of_Mutational_Events.html">Modular Matrix Operations of Mutational Events</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>DNA base/codon sequence and coordinates represented on a given Abelian group.</h1>
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/R/get_coord.R" class="external-link"><code>R/get_coord.R</code></a></small>
      <div class="d-none name"><code>get_coord.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Given a string denoting a codon or base from the DNA (or RNA)
alphabet and a genetic-code Abelian group as given in reference (1), this
function returns an object from <code><a href="CodonGroup.html">CodonGroup-class</a></code> carrying the
DNA base/codon sequence and coordinates represented on the given Abelian
group.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_coord</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'BaseGroup_OR_CodonGroup'</span></span>
<span><span class="fu">get_coord</span><span class="op">(</span><span class="va">x</span>, output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"all"</span>, <span class="st">"matrix.list"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'DNAStringSet_OR_NULL'</span></span>
<span><span class="fu">get_coord</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"all"</span>, <span class="st">"matrix.list"</span><span class="op">)</span>,</span>
<span>  base_seq <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  filepath <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cube <span class="op">=</span> <span class="st">"ACGT"</span>,</span>
<span>  group <span class="op">=</span> <span class="st">"Z4"</span>,</span>
<span>  start <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  end <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  chr <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  strand <span class="op">=</span> <span class="st">"+"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>An object from a <code><a href="BaseGroup.html">BaseGroup-class</a></code>,
<code><a href="CodonGroup.html">CodonGroup-class</a></code>,
<code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></code> or
<code><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">DNAMultipleAlignment</a></code> class carrying the DNA
pairwise alignment of two sequences. Objects from
<code><a href="BaseGroup.html">BaseGroup-class</a></code> and <code><a href="CodonGroup.html">CodonGroup-class</a></code> are
generated with functions: <code><a href="base_methods.html">base_coord</a></code> and
<code><a href="codon_coord.html">codon_coord</a></code>, respectively.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Not in use.</p></dd>


<dt id="arg-output">output<a class="anchor" aria-label="anchor" href="#arg-output"></a></dt>
<dd><p>See 'Value' section.</p></dd>


<dt id="arg-base-seq">base_seq<a class="anchor" aria-label="anchor" href="#arg-base-seq"></a></dt>
<dd><p>Logical. Whether to return the base or codon coordinates on
the selected Abelian group. If codon coordinates are requested, then the
number of the DNA bases in the given sequences must be multiple of 3.</p></dd>


<dt id="arg-filepath">filepath<a class="anchor" aria-label="anchor" href="#arg-filepath"></a></dt>
<dd><p>A character vector containing the path to a file in
<em><strong>fasta</strong></em> format to be read. This argument must be given if
<em>codon &amp; base</em> arguments are not provided.</p></dd>


<dt id="arg-cube">cube<a class="anchor" aria-label="anchor" href="#arg-cube"></a></dt>
<dd><p>A character string denoting one of the 24 Genetic-code cubes,
as given in references (2 2 3).</p></dd>


<dt id="arg-group">group<a class="anchor" aria-label="anchor" href="#arg-group"></a></dt>
<dd><p>A character string denoting the group representation for the
given base or codon as shown in reference (1).</p></dd>


<dt id="arg-start-end-chr-strand">start, end, chr, strand<a class="anchor" aria-label="anchor" href="#arg-start-end-chr-strand"></a></dt>
<dd><p>Optional parameters required to build a
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code>. If not provided the default
values given for the function definition will be used.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>An object from <code><a href="CodonGroup.html">CodonGroup-class</a></code> class is returned
when <em>output = 'all'</em>. This has two slots, the first one carrying a
list of matrices and the second one carrying the codon/base sequence
information. That is, if <em>x</em> is an object from
<code><a href="CodonGroup.html">CodonGroup-class</a></code> class, then a list of matrices of codon
coordinate can be retrieved as x@CoordList and the information on the
codon sequence as x@SeqRanges.</p>
<p>if <em>output = 'matrix.list'</em>, then an object from
<code><a href="MatrixList.html">MatrixList</a></code> class is returned.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Symbols '-' and 'N' usually found in DNA sequence alignments to
denote gaps and missing/unknown bases are represented by the number: '-1'
on Z4 and '0' in Z5. In Z64 the symbol 'NA' will be returned for codons
including symbols '-' and 'N'.</p>
<p>Although the <code><a href="CodonGroup.html">CodonGroup-class</a></code> object returned by
functions <code><a href="codon_coord.html">codon_coord</a></code> and <code><a href="base_methods.html">base_coord</a></code> are useful
to store genomic information, the base and codon coordinates are not given
on them as numeric magnitudes. Function <code>get_coord</code> provides
the way to get the coordinates in a numeric object in object from and still
to preserve the base/codon sequence information.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Load a pairwise alignment</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"aln"</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">aln</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet object of length 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]    51 <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span>---<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2]    51 <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span>------<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## DNA base representation in the Abelian group Z5</span></span></span>
<span class="r-in"><span><span class="va">coord</span> <span class="op">&lt;-</span> <span class="fu">get_coord</span><span class="op">(</span></span></span>
<span class="r-in"><span>    x <span class="op">=</span> <span class="va">aln</span>,</span></span>
<span class="r-in"><span>    cube <span class="op">=</span> <span class="st">"ACGT"</span>,</span></span>
<span class="r-in"><span>    group <span class="op">=</span> <span class="st">"Z5"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">coord</span> <span class="co">## A list of vectors</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CodonSeq object of length: 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> names(2): coord1 coord2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Vector of length: 51 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] 1 2 2 4 1 4 3 4 4 3 3 4 1 4 4 0 0 0 3 2 3 2 4 2 2 1 1 2 4 2 2 4 4 3 3 2 4 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [39] 4 1 3 2 4 2 1 2 4 1 2 1 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;1 more numeric element(s)&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Two slots: 'CoordList' &amp; 'SeqRanges'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Extract the coordinate list</span></span></span>
<span class="r-in"><span><span class="fu"><a href="CodonSeq.html">coordList</a></span><span class="op">(</span><span class="va">coord</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coord1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] 1 2 2 4 1 4 3 4 4 3 3 4 1 4 4 0 0 0 3 2 3 2 4 2 2 1 1 2 4 2 2 4 4 3 3 2 4 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [39] 4 1 3 2 4 2 1 2 4 1 2 1 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coord2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] 1 4 2 4 1 4 3 4 4 3 3 4 1 4 4 1 2 3 1 2 3 2 4 2 2 1 1 4 4 2 2 4 4 3 3 3 4 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [39] 2 0 0 0 0 0 0 2 4 2 2 4 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Extract the sequence list</span></span></span>
<span class="r-in"><span><span class="fu"><a href="CodonSeq.html">seqRanges</a></span><span class="op">(</span><span class="va">coord</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 51 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        seqnames    ranges strand |        seq1        seq2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]        1         1      + |           A           A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]        1         2      + |           C           T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]        1         3      + |           C           C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]        1         4      + |           T           T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]        1         5      + |           A           A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...      ...       ...    ... .         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [47]        1        47      + |           T           T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [48]        1        48      + |           A           C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [49]        1        49      + |           C           C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [50]        1        50      + |           A           T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [51]        1        51      + |           T           T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## DNA codon representation in the Abelian group Z64</span></span></span>
<span class="r-in"><span><span class="va">coord</span> <span class="op">&lt;-</span> <span class="fu">get_coord</span><span class="op">(</span></span></span>
<span class="r-in"><span>    x <span class="op">=</span> <span class="va">aln</span>,</span></span>
<span class="r-in"><span>    base_seq <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>    cube <span class="op">=</span> <span class="st">"ACGT"</span>,</span></span>
<span class="r-in"><span>    group <span class="op">=</span> <span class="st">"Z64"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">coord</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CodonSeq object of length: 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> names(2): coord1 coord2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Vector of length: 17 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] 17 15 59 43 51 NA 26 53  4 53 55 41 31 33 28 52  7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;1 more numeric element(s)&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Two slots: 'CoordList' &amp; 'SeqRanges'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Extract the coordinate list</span></span></span>
<span class="r-in"><span><span class="fu"><a href="CodonSeq.html">coordList</a></span><span class="op">(</span><span class="va">coord</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coord1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] 17 15 59 43 51 NA 26 53  4 53 55 41 31 33 28 52  7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coord2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] 49 15 59 43 51 18 18 53  4 61 55 42 29 NA NA 53 55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Extract the sequence list</span></span></span>
<span class="r-in"><span><span class="fu"><a href="CodonSeq.html">seqRanges</a></span><span class="op">(</span><span class="va">coord</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 17 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        seqnames    ranges strand |        seq1        seq2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]        1         1      + |         ACC         ATC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]        1         2      + |         TAT         TAT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]        1         3      + |         GTT         GTT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]        1         4      + |         GGT         GGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]        1         5      + |         ATT         ATT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...      ...       ...    ... .         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [13]        1        13      + |         TCT         TCC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [14]        1        14      + |         AGC         ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [15]        1        15      + |         TCA         ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [16]        1        16      + |         CTA         CTC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [17]        1        17      + |         CAT         CTT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

