<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>DNA base/codon sequence and coordinates represented on a given Abelian
group. — get_coord • GenomAutomorphism</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="DNA base/codon sequence and coordinates represented on a given Abelian
group. — get_coord" />
<meta property="og:description" content="Given a string denoting a codon or base from the DNA (or RNA)
alphabet and a genetic-code Abelian group as given in reference (1), this
function returns an object from CodonGroup-class carrying the
DNA base/codon sequence and coordinates represented on the given Abelian
group." />
<meta property="og:image" content="https://genomaths.github.io/GenomAutomorphism/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomAutomorphism</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/GenomAutomorphism.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/GenomAutomorphism.html">GenomAutomorphism</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/GenomAutomorphism/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>DNA base/codon sequence and coordinates represented on a given Abelian
group.</h1>
    <small class="dont-index">Source: <a href='https://github.com/genomaths/GenomAutomorphism/blob/master/R/get_coord.R'><code>R/get_coord.R</code></a></small>
    <div class="hidden name"><code>get_coord.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given a string denoting a codon or base from the DNA (or RNA)
alphabet and a genetic-code Abelian group as given in reference (1), this
function returns an object from <code><a href='CodonGroup-class.html'>CodonGroup-class</a></code> carrying the
DNA base/codon sequence and coordinates represented on the given Abelian
group.</p>
    </div>

    <pre class="usage"><span class='fu'>get_coord</span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S4 method for BaseGroup_OR_CodonGroup</span>
<span class='fu'>get_coord</span><span class='op'>(</span><span class='va'>x</span>, output <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"all"</span>, <span class='st'>"matrix.list"</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># S4 method for DNAStringSet_OR_NULL</span>
<span class='fu'>get_coord</span><span class='op'>(</span>
  <span class='va'>x</span>,
  output <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"all"</span>, <span class='st'>"matrix.list"</span><span class='op'>)</span>,
  base_seq <span class='op'>=</span> <span class='cn'>TRUE</span>,
  filepath <span class='op'>=</span> <span class='cn'>NULL</span>,
  cube <span class='op'>=</span> <span class='st'>"ACGT"</span>,
  group <span class='op'>=</span> <span class='st'>"Z4"</span>,
  start <span class='op'>=</span> <span class='cn'>NA</span>,
  end <span class='op'>=</span> <span class='cn'>NA</span>,
  chr <span class='op'>=</span> <span class='fl'>1L</span>,
  strand <span class='op'>=</span> <span class='st'>"+"</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>An object from a <code><a href='BaseGroup-class.html'>BaseGroup-class</a></code>,
<code><a href='CodonGroup-class.html'>CodonGroup-class</a></code>,
<code><a href='https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html'>DNAStringSet</a></code> or
<code><a href='https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html'>DNAMultipleAlignment</a></code> class carrying the DNA
pairwise alignment of two sequences. Objects from
<code><a href='BaseGroup-class.html'>BaseGroup-class</a></code> and <code><a href='CodonGroup-class.html'>CodonGroup-class</a></code> are
generated with functions: <code><a href='base_coord.html'>base_coord</a></code> and
<code><a href='codon_coord.html'>codon_coord</a></code>, respectively.</p></td>
    </tr>
    <tr>
      <th>output</th>
      <td><p>See <a href='@return'>Value</a> section.</p></td>
    </tr>
    <tr>
      <th>base_seq</th>
      <td><p>Logical. Whether to return the base or codon coordinates on
the selected Abelian group. If codon coordinates are requested, then the
number of the DNA bases in the given sequences must be multiple of 3.</p></td>
    </tr>
    <tr>
      <th>filepath</th>
      <td><p>A character vector containing the path to a file in
<em><strong>fasta</strong></em> format to be read. This argument must be given if
<em>codon &amp; base</em> arguments are not provided.</p></td>
    </tr>
    <tr>
      <th>cube</th>
      <td><p>A character string denoting one of the 24 Genetic-code cubes,
as given in references (2 2 3).</p></td>
    </tr>
    <tr>
      <th>group</th>
      <td><p>A character string denoting the group representation for the
given base or codon as shown in reference (1).</p></td>
    </tr>
    <tr>
      <th>start, end, chr, strand</th>
      <td><p>Optional parameters required to build a
<code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges-class</a></code>. If not provided the default
values given for the function definition will be used.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object from <code><a href='CodonGroup-class.html'>CodonGroup-class</a></code> class is returned when
<em>output = "all"</em>. This has two slots, the first one carrying a list of
matrices and the second one carrying the codon/base sequence information.
That is, if <em>x</em> is an object from <code><a href='CodonGroup-class.html'>CodonGroup-class</a></code> class,
then a list of matrices of codon coordinate can be retrieved as x@CoordList
and the information on the codon sequence as x@SeqRanges.</p>
<p>if <em>output = "matrix.list"</em>, then an object from
<code><a href='MatrixList-class.html'>MatrixList</a></code> class is returned.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Symbols "-" and "N" usually found in DNA sequence alignments to
denote gaps and missing/unknown bases are represented by the number: '-1' on
Z4 and '0' in Z5. In Z64 the symbol 'NA' will be returned for codons
including symbols "-" and "N".</p>
<p>Although the <code><a href='CodonGroup-class.html'>CodonGroup-class</a></code> object returned by
functions <code><a href='codon_coord.html'>codon_coord</a></code> and <code><a href='base_coord.html'>base_coord</a></code> are useful
to store genomic information, the base and codon coordinates are not given on
them as numeric magnitudes. Function <code>get_coord</code> provides the way
to get the coordinates in a numeric object in object from and still to
preserve the base/codon sequence information.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Load a pairwise alignment </span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>aln</span><span class='op'>)</span>
<span class='va'>aln</span>
</div><div class='output co'>#&gt; DNAStringSet object of length 2:
#&gt;     width seq
#&gt; [1]    51 <span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFFF;'>GG</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span>---<span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFAFAF;'>AA</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFFF;'>GG</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>T</span>
#&gt; [2]    51 <span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFFF;'>GG</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFAFAF;'>AA</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFFF;'>GGG</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span>------<span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span></div><div class='input'>
<span class='co'>## DNA base representation in the Abelian group Z5</span>
<span class='va'>coord</span> <span class='op'>&lt;-</span> <span class='fu'>get_coord</span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='va'>aln</span>, 
    cube <span class='op'>=</span> <span class='st'>"ACGT"</span>,
    group <span class='op'>=</span> <span class='st'>"Z5"</span><span class='op'>)</span>
<span class='va'>coord</span> <span class='co'>## A list of vectors</span>
</div><div class='output co'>#&gt; CodonSeq object of length: 2
#&gt; names(2): coord1 coord2 
#&gt; ------- 
#&gt; Vector of length: 51 
#&gt;   1: 2: 3: 4: 5: ... 47: 48: 49: 50: 51:
#&gt; 1  1  2  2  4  1 ...   4   1   2   1   4
#&gt; ...
#&gt; &lt;1 more numeric element(s)&gt;
#&gt; Two slots: 'CoordList' &amp; 'SeqRanges'
#&gt; ------- </div><div class='input'>
<span class='co'>## Extract the coordinate list</span>
<span class='va'>coord</span><span class='op'>@</span><span class='va'>CoordList</span>
</div><div class='output co'>#&gt; $coord1
#&gt;  [1] 1 2 2 4 1 4 3 4 4 3 3 4 1 4 4 0 0 0 3 2 3 2 4 2 2 1 1 2 4 2 2 4 4 3 3 2 4 2
#&gt; [39] 4 1 3 2 4 2 1 2 4 1 2 1 4
#&gt; 
#&gt; $coord2
#&gt;  [1] 1 4 2 4 1 4 3 4 4 3 3 4 1 4 4 1 2 3 1 2 3 2 4 2 2 1 1 4 4 2 2 4 4 3 3 3 4 2
#&gt; [39] 2 0 0 0 0 0 0 2 4 2 2 4 4
#&gt; </div><div class='input'>
<span class='co'>## Extract the sequence list</span>
<span class='va'>coord</span><span class='op'>@</span><span class='va'>SeqRanges</span>
</div><div class='output co'>#&gt; GRanges object with 51 ranges and 2 metadata columns:
#&gt;        seqnames    ranges strand |        seq1        seq2
#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
#&gt;    [1]        1         1      + |           A           A
#&gt;    [2]        1         2      + |           C           T
#&gt;    [3]        1         3      + |           C           C
#&gt;    [4]        1         4      + |           T           T
#&gt;    [5]        1         5      + |           A           A
#&gt;    ...      ...       ...    ... .         ...         ...
#&gt;   [47]        1        47      + |           T           T
#&gt;   [48]        1        48      + |           A           C
#&gt;   [49]        1        49      + |           C           C
#&gt;   [50]        1        50      + |           A           T
#&gt;   [51]        1        51      + |           T           T
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>## DNA codon representation in the Abelian group Z64</span>
<span class='va'>coord</span> <span class='op'>&lt;-</span> <span class='fu'>get_coord</span><span class='op'>(</span>
            x <span class='op'>=</span> <span class='va'>aln</span>, 
            base_seq <span class='op'>=</span> <span class='cn'>FALSE</span>,
            cube <span class='op'>=</span> <span class='st'>"ACGT"</span>,
            group <span class='op'>=</span> <span class='st'>"Z64"</span><span class='op'>)</span>
<span class='va'>coord</span>
</div><div class='output co'>#&gt; CodonSeq object of length: 2
#&gt; names(2): coord1 coord2 
#&gt; ------- 
#&gt; Vector of length: 17 
#&gt;   1: 2: 3: 4: 5: ... 13: 14: 15: 16: 17:
#&gt; 1 17 15 59 43 51 ...  31  33  28  52   7
#&gt; ...
#&gt; &lt;1 more numeric element(s)&gt;
#&gt; Two slots: 'CoordList' &amp; 'SeqRanges'
#&gt; ------- </div><div class='input'>
<span class='co'>## Extract the coordinate list</span>
<span class='va'>coord</span><span class='op'>@</span><span class='va'>CoordList</span>
</div><div class='output co'>#&gt; $coord1
#&gt;  [1] 17 15 59 43 51 NA 26 53  4 53 55 41 31 33 28 52  7
#&gt; 
#&gt; $coord2
#&gt;  [1] 49 15 59 43 51 18 18 53  4 61 55 42 29 NA NA 53 55
#&gt; </div><div class='input'>
<span class='co'>## Extract the sequence list</span>
<span class='va'>coord</span><span class='op'>@</span><span class='va'>SeqRanges</span>
</div><div class='output co'>#&gt; GRanges object with 17 ranges and 2 metadata columns:
#&gt;        seqnames    ranges strand |        seq1        seq2
#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
#&gt;    [1]        1         1      + |         ACC         ATC
#&gt;    [2]        1         2      + |         TAT         TAT
#&gt;    [3]        1         3      + |         GTT         GTT
#&gt;    [4]        1         4      + |         GGT         GGT
#&gt;    [5]        1         5      + |         ATT         ATT
#&gt;    ...      ...       ...    ... .         ...         ...
#&gt;   [13]        1        13      + |         TCT         TCC
#&gt;   [14]        1        14      + |         AGC         ---
#&gt;   [15]        1        15      + |         TCA         ---
#&gt;   [16]        1        16      + |         CTA         CTC
#&gt;   [17]        1        17      + |         CAT         CTT
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://genomaths.com'>Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


