<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get Mutation Score from an AAindex or a Mutation/Distance Matrix — get_mutscore • GenomAutomorphism</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Get Mutation Score from an AAindex or a Mutation/Distance Matrix — get_mutscore"><meta name="description" content="This function is applied to get the mutation or contact
potential scores representing the similarity/distance between amino acids
corresponding to substitution mutations. The scores are retrieved from a
mutation matrix or a statistical protein contact potentials matrix from
AAindex (ver.9.2).
Alternatively, the mutation scores can be estimated based on an user
mutation matrix, for example, see aminoacid_dist and codon_dist_matrix."><meta property="og:description" content="This function is applied to get the mutation or contact
potential scores representing the similarity/distance between amino acids
corresponding to substitution mutations. The scores are retrieved from a
mutation matrix or a statistical protein contact potentials matrix from
AAindex (ver.9.2).
Alternatively, the mutation scores can be estimated based on an user
mutation matrix, for example, see aminoacid_dist and codon_dist_matrix."><meta property="og:image" content="https://genomaths.github.io/genomautomorphism/logo.png"><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomAutomorphism</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.8.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../articles/GenomAutomorphism.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Package functions</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GenomAutomorphism.html">GenomAutomorphism</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree_II.html">A Short Introduction to Algebraic Taxonomy on Genes Regions II</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/Modular_Matrix_Operation_of_Mutational_Events.html">Modular Matrix Operations of Mutational Events</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Get Mutation Score from an AAindex or a Mutation/Distance Matrix</h1>
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/R/get_mutscore.R" class="external-link"><code>R/get_mutscore.R</code></a></small>
      <div class="d-none name"><code>get_mutscore.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function is applied to get the mutation or contact
potential scores representing the similarity/distance between amino acids
corresponding to substitution mutations. The scores are retrieved from a
mutation matrix or a statistical protein contact potentials matrix from
<a href="https://www.genome.jp/aaindex/" class="external-link">AAindex</a> (ver.9.2).</p>
<p>Alternatively, the mutation scores can be estimated based on an user
mutation matrix, for example, see <a href="aminoacid_dist.html">aminoacid_dist</a> and <a href="codon_dist_matrix.html">codon_dist_matrix</a>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_mutscore</span><span class="op">(</span><span class="va">aa1</span>, <span class="va">aa2</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'character,character'</span></span>
<span><span class="fu">get_mutscore</span><span class="op">(</span></span>
<span>  <span class="va">aa1</span>,</span>
<span>  <span class="va">aa2</span>,</span>
<span>  acc <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  aaindex <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  mutmat <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  alphabet <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AA"</span>, <span class="st">"DNA"</span><span class="op">)</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'BaseSeq,missing'</span></span>
<span><span class="fu">get_mutscore</span><span class="op">(</span></span>
<span>  <span class="va">aa1</span>,</span>
<span>  <span class="va">aa2</span>,</span>
<span>  acc <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  aaindex <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  mutmat <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  alphabet <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AA"</span>, <span class="st">"DNA"</span><span class="op">)</span>,</span>
<span>  stat <span class="op">=</span> <span class="va">mean</span>,</span>
<span>  numcores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"dist"</span>, <span class="st">"matrix"</span>, <span class="st">"vector"</span><span class="op">)</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'DNAStringSet,missing'</span></span>
<span><span class="fu">get_mutscore</span><span class="op">(</span></span>
<span>  <span class="va">aa1</span>,</span>
<span>  <span class="va">aa2</span>,</span>
<span>  acc <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  aaindex <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  mutmat <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  alphabet <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AA"</span>, <span class="st">"DNA"</span><span class="op">)</span>,</span>
<span>  stat <span class="op">=</span> <span class="va">mean</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"dist"</span>, <span class="st">"matrix"</span>, <span class="st">"vector"</span><span class="op">)</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'DNAMultipleAlignment,missing'</span></span>
<span><span class="fu">get_mutscore</span><span class="op">(</span></span>
<span>  <span class="va">aa1</span>,</span>
<span>  <span class="va">aa2</span>,</span>
<span>  acc <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  aaindex <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  mutmat <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  alphabet <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AA"</span>, <span class="st">"DNA"</span><span class="op">)</span>,</span>
<span>  stat <span class="op">=</span> <span class="va">mean</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"dist"</span>, <span class="st">"matrix"</span>, <span class="st">"vector"</span><span class="op">)</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-aa-aa-">aa1, aa2<a class="anchor" aria-label="anchor" href="#arg-aa-aa-"></a></dt>
<dd><p>A simple character representing an amino acids or a
character string of letter from the amino acid alphabet or base-triplets
from the DNA/RNA alphabet. If <strong><em>aa1</em></strong> is an object from any
of the classes: <a href="BaseSeq.html">BaseSeq</a>, <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></code>, or
<code><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">DNAMultipleAlignment</a></code>, then argument
<strong><em>aa2</em></strong> is not required.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Not in use.</p></dd>


<dt id="arg-acc">acc<a class="anchor" aria-label="anchor" href="#arg-acc"></a></dt>
<dd><p>Accession id for a specified mutation or contact potential
matrix.</p></dd>


<dt id="arg-aaindex">aaindex<a class="anchor" aria-label="anchor" href="#arg-aaindex"></a></dt>
<dd><p>Database where the requested accession id is locate. The
possible values are:  "aaindex2" or "aaindex3".</p></dd>


<dt id="arg-mutmat">mutmat<a class="anchor" aria-label="anchor" href="#arg-mutmat"></a></dt>
<dd><p>A mutation or any score matrix provided by the user.</p></dd>


<dt id="arg-alphabet">alphabet<a class="anchor" aria-label="anchor" href="#arg-alphabet"></a></dt>
<dd><p>Whether the alphabet is from the 20 amino acid (AA) or
four (DNA)/RNA base alphabet. This would prevent mistakes, i.e.,
the strings "ACG" would be a base-triplet on the DNA alphabet or simply
the amino acid sequence of alanine, cysteine, and glutamic acid.</p></dd>


<dt id="arg-num-cores-tasks">num.cores, tasks<a class="anchor" aria-label="anchor" href="#arg-num-cores-tasks"></a></dt>
<dd><p>Parameters for parallel computation using package
<code><a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallel-package.html" class="external-link">BiocParallel-package</a></code>: the number of cores to
use, i.e. at most how many child processes will be run simultaneously (see
<code><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></code> and the number of tasks per job (only
for Linux OS).</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Optional. Only if num.cores &gt; 1. If TRUE, prints the
function log to stdout.</p></dd>


<dt id="arg-stat">stat<a class="anchor" aria-label="anchor" href="#arg-stat"></a></dt>
<dd><p>Statistic that will be used to summarize the scores of the
DNA sequences provided. Only if <strong><em>aa1</em></strong> is an object from any of
the classes: <a href="BaseSeq.html">BaseSeq</a>, <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></code>, or
<code><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">DNAMultipleAlignment</a></code>.</p></dd>


<dt id="arg-numcores">numcores<a class="anchor" aria-label="anchor" href="#arg-numcores"></a></dt>
<dd><p>An integer to setup the number of parallel workers via
<code><a href="https://rdrr.io/r/parallel/makeCluster.html" class="external-link">makeCluster</a></code>.</p></dd>


<dt id="arg-output">output<a class="anchor" aria-label="anchor" href="#arg-output"></a></dt>
<dd><p>Optional. Class of the returned object. Only if
<strong><em>aa1</em></strong> is an object from any of the classes: <a href="BaseSeq.html">BaseSeq</a>,
<code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></code>, or
<code><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">DNAMultipleAlignment</a></code>.</p></dd>


<dt id="arg-na-rm">na.rm<a class="anchor" aria-label="anchor" href="#arg-na-rm"></a></dt>
<dd><p>a logical evaluating to TRUE or FALSE indicating whether NA
values should be stripped before the computation proceeds.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A single numeric score or a numerical vector, or if
<strong><em>aa1</em></strong> is an object from any of the classes: <a href="BaseSeq.html">BaseSeq</a>,
<code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></code>, or
<code><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">DNAMultipleAlignment</a></code>, then depending on the
user selection the returned object will be:</p><ol><li><p>A lower diagonal numerical vector of the sequence pairwise scores.</p></li>
<li><p>A <code><a href="https://rdrr.io/r/stats/dist.html" class="external-link">dist</a></code>-class object.</p></li>
<li><p>A whole score matrix.</p></li>
</ol></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>If a score matrix is provided by the user, then it must be a
symmetric matrix 20x20.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><a href="aa_phychem_index.html">aa_mutmat</a>, <a href="aaindex2.html">aaindex2</a> and <a href="aaindex3.html">aaindex3</a>.</p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Robersy Sanchez <a href="https://genomaths.com" class="external-link">https://genomaths.com</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## A single amino acids substitution mutation</span></span></span>
<span class="r-in"><span><span class="fu">get_mutscore</span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"C"</span>, acc <span class="op">=</span> <span class="st">"MIYS930101"</span>, aaindex <span class="op">=</span> <span class="st">"aaindex2"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] -0.51</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## A tri-peptide mutation</span></span></span>
<span class="r-in"><span><span class="fu">get_mutscore</span><span class="op">(</span>aa1 <span class="op">=</span> <span class="st">"ACG"</span>, aa2 <span class="op">=</span> <span class="st">"ATG"</span>, acc <span class="op">=</span> <span class="st">"MIYS930101"</span>,</span></span>
<span class="r-in"><span>            aaindex <span class="op">=</span> <span class="st">"aaindex2"</span>, alphabet <span class="op">=</span> <span class="st">"AA"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]  0.34 -0.62  0.43</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## A single base-triple mutation, i.e., a single amino acid substitution </span></span></span>
<span class="r-in"><span><span class="co">## mutation</span></span></span>
<span class="r-in"><span><span class="fu">get_mutscore</span><span class="op">(</span>aa1 <span class="op">=</span> <span class="st">"ACG"</span>, aa2 <span class="op">=</span> <span class="st">"CTA"</span>, acc <span class="op">=</span> <span class="st">"MIYS930101"</span>,</span></span>
<span class="r-in"><span>            aaindex <span class="op">=</span> <span class="st">"aaindex2"</span>, alphabet <span class="op">=</span> <span class="st">"DNA"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] -0.77</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Peptides can be also written as:</span></span></span>
<span class="r-in"><span><span class="fu">get_mutscore</span><span class="op">(</span>aa1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="st">"C"</span>,<span class="st">"G"</span><span class="op">)</span>, aa2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"C"</span>,<span class="st">"T"</span>,<span class="st">"A"</span><span class="op">)</span>, </span></span>
<span class="r-in"><span>            acc <span class="op">=</span> <span class="st">"MIYS930101"</span>, aaindex <span class="op">=</span> <span class="st">"aaindex2"</span>, alphabet <span class="op">=</span> <span class="st">"AA"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] -0.51 -0.62  0.19</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

