<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Definition of MatrixSeq-class — MatrixSeq-class • GenomAutomorphism</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Definition of MatrixSeq-class — MatrixSeq-class"><meta name="description" content="This is a very simple flexible class to store DNA and
aminoacid aligned sequences together with their physicochemical properties.
That is, a place where each aminoacid or codon from the sequence is
represented by numerical value from a physicochemical index."><meta property="og:description" content="This is a very simple flexible class to store DNA and
aminoacid aligned sequences together with their physicochemical properties.
That is, a place where each aminoacid or codon from the sequence is
represented by numerical value from a physicochemical index."><meta property="og:image" content="https://genomaths.github.io/genomautomorphism/logo.png"><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomAutomorphism</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.8.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../articles/GenomAutomorphism.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Package functions</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GenomAutomorphism.html">GenomAutomorphism</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree_II.html">A Short Introduction to Algebraic Taxonomy on Genes Regions II</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/Modular_Matrix_Operation_of_Mutational_Events.html">Modular Matrix Operations of Mutational Events</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Definition of MatrixSeq-class</h1>
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/R/allClasses.R" class="external-link"><code>R/allClasses.R</code></a>, <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/R/show-methods.R" class="external-link"><code>R/show-methods.R</code></a></small>
      <div class="d-none name"><code>MatrixSeq.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This is a very simple flexible class to store DNA and
aminoacid aligned sequences together with their physicochemical properties.
That is, a place where each aminoacid or codon from the sequence is
represented by numerical value from a physicochemical index.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">MatrixSeq</span><span class="op">(</span><span class="va">seqs</span>, <span class="va">matrix</span>, <span class="va">names</span>, <span class="va">aaindex</span>, <span class="va">phychem</span>, <span class="va">accession</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'MatrixSeq'</span></span>
<span><span class="fu">show</span><span class="op">(</span><span class="va">object</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seqs-matrix-names-aaindex-phychem-accession">seqs, matrix, names, aaindex, phychem, accession<a class="anchor" aria-label="anchor" href="#arg-seqs-matrix-names-aaindex-phychem-accession"></a></dt>
<dd><p>See detail section</p></dd>


<dt id="arg-object">object<a class="anchor" aria-label="anchor" href="#arg-object"></a></dt>
<dd><p>An object from 'MatrixSeq' class</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Given the slot values, it defines a MatrixSeq-class.</p>
<p>A MatrixSeq-class object</p>
<p>Print/show of a MatrixSeq-class object.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>

<dl><dt><strong>seqs</strong>: </dt>
<dd><p>A string character vector of DNA or aminoacid
sequences.</p></dd>

<dt><strong>matrix</strong>: </dt>
<dd><p>A numerical matrix or a numerical vector
(in the constructor) carrying the specified aminoacid physicochemical
indices for aminoacid in the DNA or aminoacid sequence(s).</p></dd>

<dt><strong>names</strong>: </dt>
<dd><p>Alias/names/IDs DNA or aminoacid sequences.</p></dd>

<dt><strong>aaindex</strong>: </dt>
<dd><p>Aminoacid index database where the
physicochemical index can be found.</p></dd>

<dt><strong>phychem</strong>: </dt>
<dd><p>Description of the physicochemical index applied
to represent the DNA or aminoacid sequences.</p></dd>

<dt><strong>accession</strong>: </dt>
<dd><p>Accession number or ID of the applied
physicochemical index in the database.</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Robersy Sanchez <a href="https://genomaths.com" class="external-link">https://genomaths.com</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">aln</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>S1 <span class="op">=</span> <span class="st">"ATGCGGATTAGA"</span>, S2 <span class="op">=</span> <span class="st">"ATGACGATCACA"</span>, S3 <span class="op">=</span> <span class="st">"ATGAGATCACAG"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">cd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">DNAMultipleAlignment</a></span><span class="op">(</span><span class="va">aln</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">r1</span> <span class="op">&lt;-</span> <span class="fu"><a href="peptide_phychem_index.html">peptide_phychem_index</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/MaskedXString-class.html" class="external-link">unmasked</a></span><span class="op">(</span><span class="va">cd</span><span class="op">)</span>, acc <span class="op">=</span> <span class="st">"EISD840101"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">r1</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MatrixSeq with 3 rows (sequences) and 4 columns (aminoacids/codons):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      A1    A2    A3    A4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S1 0.26 -1.76  0.73 -1.76</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S2 0.26 -0.18  0.73 -0.18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S3 0.26 -1.76 -0.26 -0.69</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slots: 'seqs', 'matrix', 'names', 'aaindex', 'phychem', 'accession</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Extract the second aminoacid sequence</span></span></span>
<span class="r-in"><span><span class="va">r1</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MatrixSeq with 4 rows (aminoacids/codons) and 1 column (sequence):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       S2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A1  0.26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A2 -0.18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A3  0.73</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A4 -0.18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slots: 'seqs', 'matrix', 'names', 'aaindex', 'phychem', 'accession</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Using the sequence given name </span></span></span>
<span class="r-in"><span><span class="va">r1</span><span class="op">$</span><span class="va">S1</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MatrixSeq with 4 rows (aminoacids/codons) and 1 column (sequence):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       S1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A1  0.26</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A2 -1.76</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A3  0.73</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A4 -1.76</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slots: 'seqs', 'matrix', 'names', 'aaindex', 'phychem', 'accession</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Extract the second aminoacid value from the first sequence</span></span></span>
<span class="r-in"><span><span class="va">r1</span><span class="op">[</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       A2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S1 -1.76</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slots: 'seqs', 'matrix', 'names', 'aaindex', 'phychem', 'accession</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Change the name the second sequence</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">r1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'S1'</span>, <span class="st">'Seq1'</span>, <span class="st">'S1'</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">r1</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MatrixSeq with 3 rows (sequences) and 4 columns (aminoacids/codons):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        A1    A2    A3    A4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S1   0.26 -1.76  0.73 -1.76</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Seq1 0.26 -0.18  0.73 -0.18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> S1   0.26 -1.76 -0.26 -0.69</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slots: 'seqs', 'matrix', 'names', 'aaindex', 'phychem', 'accession</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Extract the amino acid sequences</span></span></span>
<span class="r-in"><span><span class="fu">slot</span><span class="op">(</span><span class="va">r1</span>, <span class="st">'seqs'</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     S1   Seq1     S1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "MRIR" "MTIT" "MRSQ" </span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

