<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Extract the gene ranges and coordinates from a pairwise
alignment of codon/base sequences represented on a given Abelian group."><title>Get DNA sequence Ranges and Coordinates representation on a given
Abelian Group — seqranges • GenomAutomorphism</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Get DNA sequence Ranges and Coordinates representation on a given
Abelian Group — seqranges"><meta property="og:description" content="Extract the gene ranges and coordinates from a pairwise
alignment of codon/base sequences represented on a given Abelian group."><meta property="og:image" content="https://genomaths.github.io/GenomAutomorphism/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GenomAutomorphism</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/GenomAutomorphism.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Package functions</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/GenomAutomorphism.html">GenomAutomorphism</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Get DNA sequence Ranges and Coordinates representation on a given
Abelian Group</h1>
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/R/seqranges.R" class="external-link"><code>R/seqranges.R</code></a></small>
      <div class="d-none name"><code>seqranges.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Extract the gene ranges and coordinates from a pairwise
alignment of codon/base sequences represented on a given Abelian group.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">seqranges</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S4 method for CodonSeq</span>
<span class="fu">seqranges</span><span class="op">(</span><span class="va">x</span>, granges <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># S4 method for DNAStringSet_OR_NULL</span>
<span class="fu">seqranges</span><span class="op">(</span>
  <span class="va">x</span>,
  granges <span class="op">=</span> <span class="cn">TRUE</span>,
  base_seq <span class="op">=</span> <span class="cn">TRUE</span>,
  filepath <span class="op">=</span> <span class="cn">NULL</span>,
  start <span class="op">=</span> <span class="cn">NA</span>,
  end <span class="op">=</span> <span class="cn">NA</span>,
  chr <span class="op">=</span> <span class="fl">1L</span>,
  strand <span class="op">=</span> <span class="st">"+"</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>x</dt>
<dd><p>An object from a <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></code> or
<code><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">DNAMultipleAlignment</a></code> class carrying the DNA
pairwise alignment of two sequences.</p></dd>
<dt>granges</dt>
<dd><p>Logical. Whether to return a
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> object or a
<code><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></code>.</p></dd>
<dt>base_seq</dt>
<dd><p>Logical. Whether to return the base or codon coordinates on
the selected Abelian group. If codon coordinates are requested, then the
number of the DNA bases in the given sequences must be multiple of 3.</p></dd>
<dt>filepath</dt>
<dd><p>A character vector containing the path to a file in
<em><strong>fasta</strong></em> format to be read. This argument must be given if
<em>codon &amp; base</em> arguments are not provided.</p></dd>
<dt>start, end, chr, strand</dt>
<dd><p>Optional parameters required to build a
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code>. If not provided the default
values given for the function definition will be used.</p></dd>
<dt>cube</dt>
<dd><p>A character string denoting one of the 24 Genetic-code cubes,
as given in references (2-3).</p></dd>
<dt>group</dt>
<dd><p>A character string denoting the group representation for the
given base or codon as shown in reference (1).</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function provide an alternative way to get the codon coordinate
and the information on the codon sequence from a <code><a href="CodonSeq.html">CodonSeq</a></code> class
objects. The function can either take the output from functions
<code><a href="codon_coord.html">codon_coord</a></code> or to operate directly on a
<code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></code> or to retrieve the a DNA sequence
alignment from a file.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    
<ol><li><p>Robersy Sanchez, Jesús Barreto (2021) Genomic Abelian Finite
Groups.
<a href="https://doi.org/10.1101/2021.06.01.446543" class="external-link">doi: 10.1101/2021.06.01.446543</a>.</p></li>
<li><p>M. V José, E.R. Morgado, R. Sánchez, T. Govezensky, The 24 possible
algebraic representations of the standard genetic code in six or in three
dimensions, Adv. Stud. Biol. 4 (2012) 119–152.<a href="https://is.gd/na9eap" class="external-link">PDF</a>.</p></li>
<li><p>R. Sanchez. Symmetric Group of the Genetic–Code Cubes. Effect of the
Genetic–Code Architecture on the Evolutionary Process MATCH Commun. Math.
Comput. Chem. 79 (2018) 527-560.</p></li>
</ol></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="matrices.html">matrices</a></code>, <code><a href="codon_coord.html">codon_coord</a></code>, and
<code><a href="base_coord.html">base_coord</a></code>.</p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Robersy Sanchez <a href="https://genomaths.com" class="external-link">https://genomaths.com</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co">## Load a pairwise alignment</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">aln</span><span class="op">)</span></span>
<span class="r-in"><span class="va">aln</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet object of length 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]    51 <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span>---<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2]    51 <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span>------<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## A GRanges object carrying the aligned DNA sequence.</span></span>
<span class="r-in"><span class="fu">seqranges</span><span class="op">(</span></span>
<span class="r-in">    x <span class="op">=</span> <span class="va">aln</span>,</span>
<span class="r-in">    base_seq <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span class="r-in">    filepath <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 51 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        seqnames    ranges strand |        seq1        seq2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]        1         1      + |           A           A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]        1         2      + |           C           T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]        1         3      + |           C           C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]        1         4      + |           T           T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]        1         5      + |           A           A</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...      ...       ...    ... .         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [47]        1        47      + |           T           T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [48]        1        48      + |           A           C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [49]        1        49      + |           C           C</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [50]        1        50      + |           A           T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [51]        1        51      + |           T           T</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## A GRanges object carrying the aligned codon sequence.</span></span>
<span class="r-in"><span class="fu">seqranges</span><span class="op">(</span></span>
<span class="r-in">    x <span class="op">=</span> <span class="va">aln</span>,</span>
<span class="r-in">    base_seq <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span class="r-in">    filepath <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 17 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        seqnames    ranges strand |        seq1        seq2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]        1         1      + |         ACC         ATC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]        1         2      + |         TAT         TAT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]        1         3      + |         GTT         GTT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]        1         4      + |         GGT         GGT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]        1         5      + |         ATT         ATT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...      ...       ...    ... .         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [13]        1        13      + |         TCT         TCC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [14]        1        14      + |         AGC         ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [15]        1        15      + |         TCA         ---</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [16]        1        16      + |         CTA         CTC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [17]        1        17      + |         CAT         CTT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in">        </span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer></div>

  

  

  </body></html>

