<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Get DNA sequence Ranges and Coordinates representation on a given
Abelian Group — seqranges • GenomAutomorphism</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Get DNA sequence Ranges and Coordinates representation on a given
Abelian Group — seqranges" />
<meta property="og:description" content="Extract the gene ranges and coordinates from a pairwise
alignment of codon/base sequences represented on a given Abelian group." />
<meta property="og:image" content="https://genomaths.github.io/GenomAutomorphism/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomAutomorphism</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/GenomAutomorphism.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/GenomAutomorphism.html">GenomAutomorphism</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/GenomAutomorphism/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get DNA sequence Ranges and Coordinates representation on a given
Abelian Group</h1>
    <small class="dont-index">Source: <a href='https://github.com/genomaths/GenomAutomorphism/blob/master/R/seqranges.R'><code>R/seqranges.R</code></a></small>
    <div class="hidden name"><code>seqranges.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Extract the gene ranges and coordinates from a pairwise
alignment of codon/base sequences represented on a given Abelian group.</p>
    </div>

    <pre class="usage"><span class='fu'>seqranges</span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S4 method for CodonSeq</span>
<span class='fu'>seqranges</span><span class='op'>(</span><span class='va'>x</span>, granges <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>

<span class='co'># S4 method for DNAStringSet_OR_NULL</span>
<span class='fu'>seqranges</span><span class='op'>(</span>
  <span class='va'>x</span>,
  granges <span class='op'>=</span> <span class='cn'>TRUE</span>,
  base_seq <span class='op'>=</span> <span class='cn'>TRUE</span>,
  filepath <span class='op'>=</span> <span class='cn'>NULL</span>,
  start <span class='op'>=</span> <span class='cn'>NA</span>,
  end <span class='op'>=</span> <span class='cn'>NA</span>,
  chr <span class='op'>=</span> <span class='fl'>1L</span>,
  strand <span class='op'>=</span> <span class='st'>"+"</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>An object from a <code><a href='https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html'>DNAStringSet</a></code> or
<code><a href='https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html'>DNAMultipleAlignment</a></code> class carrying the DNA
pairwise alignment of two sequences.</p></td>
    </tr>
    <tr>
      <th>granges</th>
      <td><p>Logical. Whether to return a
<code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges-class</a></code> object or a
<code><a href='https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html'>DataFrame</a></code>.</p></td>
    </tr>
    <tr>
      <th>base_seq</th>
      <td><p>Logical. Whether to return the base or codon coordinates on
the selected Abelian group. If codon coordinates are requested, then the
number of the DNA bases in the given sequences must be multiple of 3.</p></td>
    </tr>
    <tr>
      <th>filepath</th>
      <td><p>A character vector containing the path to a file in
<em><strong>fasta</strong></em> format to be read. This argument must be given if
<em>codon &amp; base</em> arguments are not provided.</p></td>
    </tr>
    <tr>
      <th>start, end, chr, strand</th>
      <td><p>Optional parameters required to build a
<code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges-class</a></code>. If not provided the default
values given for the function definition will be used.</p></td>
    </tr>
    <tr>
      <th>cube</th>
      <td><p>A character string denoting one of the 24 Genetic-code cubes,
as given in references (2-3).</p></td>
    </tr>
    <tr>
      <th>group</th>
      <td><p>A character string denoting the group representation for the
given base or codon as shown in reference (1).</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function provide an alternative way to get the codon coordinate
and the information on the codon sequence from a <code><a href='CodonSeq-class.html'>CodonSeq</a></code> class
objects. The function can either take the output from functions
<code><a href='codon_coord.html'>codon_coord</a></code> or to operate directly on a
<code><a href='https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html'>DNAStringSet</a></code> or to retrieve the a DNA sequence
alignment from a file.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    
<ol>
<li><p>Robersy Sanchez, Jesús Barreto (2021) Genomic Abelian Finite
Groups.
<a href='https://doi.org/10.1101/2021.06.01.446543'>doi: 10.1101/2021.06.01.446543</a>.</p></li>
<li><p>M. V José, E.R. Morgado, R. Sánchez, T. Govezensky, The 24 possible
algebraic representations of the standard genetic code in six or in three
dimensions, Adv. Stud. Biol. 4 (2012) 119–152.<a href='https://is.gd/na9eap'>PDF</a>.</p></li>
<li><p>R. Sanchez. Symmetric Group of the Genetic–Code Cubes. Effect of the
Genetic–Code Architecture on the Evolutionary Process MATCH Commun. Math.
Comput. Chem. 79 (2018) 527-560.</p></li>
</ol>

    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='matrices.html'>matrices</a></code>, <code><a href='codon_coord.html'>codon_coord</a></code>, and
<code><a href='base_coord.html'>base_coord</a></code>.</p></div>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Robersy Sanchez <a href='https://genomaths.com'>https://genomaths.com</a></p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Load a pairwise alignment</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>aln</span><span class='op'>)</span>
<span class='va'>aln</span>
</div><div class='output co'>#&gt; DNAStringSet object of length 2:
#&gt;     width seq
#&gt; [1]    51 <span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFFF;'>GG</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span>---<span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFAFAF;'>AA</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFFF;'>GG</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>T</span>
#&gt; [2]    51 <span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFFF;'>GG</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #FFAFAF;'>A</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #AFFFFF;'>G</span><span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFAFAF;'>AA</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span><span style='color: #000000; background-color: #AFFFFF;'>GGG</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span>------<span style='color: #000000; background-color: #AFFFAF;'>C</span><span style='color: #000000; background-color: #FFD7AF;'>T</span><span style='color: #000000; background-color: #AFFFAF;'>CC</span><span style='color: #000000; background-color: #FFD7AF;'>TT</span></div><div class='input'>
<span class='co'>## A GRanges object carrying the aligned DNA sequence.</span>
<span class='fu'>seqranges</span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='va'>aln</span>,
    base_seq <span class='op'>=</span> <span class='cn'>TRUE</span>,
    filepath <span class='op'>=</span> <span class='cn'>NULL</span>,
<span class='op'>)</span>
</div><div class='output co'>#&gt; GRanges object with 51 ranges and 2 metadata columns:
#&gt;        seqnames    ranges strand |        seq1        seq2
#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
#&gt;    [1]        1         1      + |           A           A
#&gt;    [2]        1         2      + |           C           T
#&gt;    [3]        1         3      + |           C           C
#&gt;    [4]        1         4      + |           T           T
#&gt;    [5]        1         5      + |           A           A
#&gt;    ...      ...       ...    ... .         ...         ...
#&gt;   [47]        1        47      + |           T           T
#&gt;   [48]        1        48      + |           A           C
#&gt;   [49]        1        49      + |           C           C
#&gt;   [50]        1        50      + |           A           T
#&gt;   [51]        1        51      + |           T           T
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>## A GRanges object carrying the aligned codon sequence.</span>
<span class='fu'>seqranges</span><span class='op'>(</span>
    x <span class='op'>=</span> <span class='va'>aln</span>,
    base_seq <span class='op'>=</span> <span class='cn'>FALSE</span>,
    filepath <span class='op'>=</span> <span class='cn'>NULL</span>,
<span class='op'>)</span>
</div><div class='output co'>#&gt; GRanges object with 17 ranges and 2 metadata columns:
#&gt;        seqnames    ranges strand |        seq1        seq2
#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
#&gt;    [1]        1         1      + |         ACC         ATC
#&gt;    [2]        1         2      + |         TAT         TAT
#&gt;    [3]        1         3      + |         GTT         GTT
#&gt;    [4]        1         4      + |         GGT         GGT
#&gt;    [5]        1         5      + |         ATT         ATT
#&gt;    ...      ...       ...    ... .         ...         ...
#&gt;   [13]        1        13      + |         TCT         TCC
#&gt;   [14]        1        14      + |         AGC         ---
#&gt;   [15]        1        15      + |         TCA         ---
#&gt;   [16]        1        16      + |         CTA         CTC
#&gt;   [17]        1        17      + |         CAT         CTT
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>        
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://genomaths.com'>Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


