<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Example of Methylome Analysis with MethylIT using Cancer Datasets • MethylIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Example of Methylome Analysis with MethylIT using Cancer Datasets">
<meta property="og:description" content="MethylIT">
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/MethylIT/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="cancer_example_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Example of Methylome Analysis with MethylIT using Cancer Datasets</h1>
            <h3 class="subtitle"><h1>
A <a href="https://github.com/genomaths/%20MethylIT">Methyl-IT</a> user guide. A guide for users</h1></h3>
                        <h4 class="author"><div class="line-block">Robersy Sanchez<span class="math inline">\(^1\)</span><br><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a><br><br>
Sally Mackenzie<span class="math inline">\(^{1,2}\)</span><br><a href="mailto:sam795@psu.edu" class="email">sam795@psu.edu</a><br><br><span class="math inline">\(^1\)</span>Department of Biology.<br>
Pennsylvania State University, University Park, PA 16802<br><br><span class="math inline">\(^2\)</span>Department of Plant Science.<br>
Pennsylvania State University, University Park, PA 16802</div></h4>
            
            <h4 class="date">14 July 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/master/vignettes/cancer_example.Rmd"><code>vignettes/cancer_example.Rmd</code></a></small>
      <div class="hidden name"><code>cancer_example.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>An example of methylation analysis with Methyl-IT, a novel methylome analysis procedure based on a signal-detection and machine- learning approach, is provided here. Methylation analysis involves a signal detection problem, and the method was designed to discriminate methylation regulatory signal from background noise induced by thermal fluctuations. Methyl-IT enhances the resolution of genome methylation behavior to reveal network-associated responses, offering resolution of gene pathway influences not attainable with previous methods. Herein, an example of MethylIT application to the analysis of breast cancer methylomes is presented.</p>
    </div>
    
<style type="text/css">

.list-group-item.active, .list-group-item.active:focus,
.list-group-item.active:hover {
    z-index: 2;
    color: #fff;
    background-color: #337ab7;
    border-color: #337ab7;
}

.tocify-subheader .tocify-item {
    font-size: 0.80em;
    padding-left: 25px;
    text-indent: 0;
}

.tocify-header .tocify-item {
    font-size: 0.85em;
    padding-left: 20px;
    text-indent: 0;
}

body{ /* Normal */
    font-size: 18px;
    font-family: "Times New Roman", Times, serif;
    text-align: justify
}
td { /* Table */
    font-size: 8px;
}

h1.title {
    font-size: 38px;
    font-family: "Times New Roman", Times, serif;
    color: DarkRed;
    .subTitle {
        font-size: 24px;
        font-family: "Times New Roman", Times, serif;
        color: DarkRed;
    }
}

h1 { /* Header 1 */
    font-size: 28px;
    font-family: "Times New Roman", Times, serif;
    color: DarkBlue;
}
h2 { /* Header 2 */
    font-size: 22px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
h3 { /* Header 3 */
    font-size: 18px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
code.r{ /* Code block */
    font-size: 12px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>
<div id="methyl-it" class="section level1">
<h1 class="hasAnchor">
<a href="#methyl-it" class="anchor"></a><a href="https://github.com/genomaths/MethylIT">Methyl-IT</a>
</h1>
<p>MethylIT is an R package for methylome analysis based on a signal-detection machine learning approach (SD-ML). This approach is postulated to provide greater sensitivity for resolving true signal from the methylation background within the methylome <span class="citation">(1, 2)</span>. Because the biological signal created within the dynamic methylome environment characteristic of plants is not free from background noise, the approach, designated Methyl-IT, includes the application of signal detection theory <span class="citation">(3–6)</span>.</p>
<p>A basic requirement for the application of signal detection is the knowledge of the probability distribution of the background noise, which is used as null hypothesis in the detection of the methylation signal. This probability distribution is, in general, a member of generalized gamma distribution family, and it can be deduced on a statistical mechanical/thermodynamics basis from DNA methylation induced by thermal fluctuations <span class="citation">(1)</span>. Herein, we provide an example of Methyl-IT application to the analysis of breast cancer methylomes. Due to the size of human methylome the current example only covers the analysis of chromosome 13. A full description of Methyl-IT application of methylome analysis in plants is given in the manuscript <span class="citation">(2)</span>.</p>
<p>The R packages required in this example are:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
        <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/MethylIT">MethylIT</a></span><span class="op">)</span>
        <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span> <span class="co"># graphic</span>
        <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hadley/reshape">reshape2</a></span><span class="op">)</span> <span class="co"># To reshape the data frame</span>
        <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">grid</span><span class="op">)</span> <span class="co"># For multiple plots</span>
        <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span> <span class="co"># For multiple plots</span>
        <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">VennDiagram</span><span class="op">)</span>
        <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span> <span class="co"># To import gene annotation</span>
    <span class="op">}</span>
<span class="op">)</span></code></pre></div>
<div id="available-datasets-and-reading" class="section level2">
<h2 class="hasAnchor">
<a href="#available-datasets-and-reading" class="anchor"></a>Available datasets and reading</h2>
<p>Methylome datasets of whole-genome bisulfite sequencing (WGBS) are available at Gene Expression Omnibus (GEO DataSets). For the current example, datasets from breast tissues (normal and cancer) and embryonic stem cells will be downloaded from GEO. The data set are downloaded providing the GEO accession numbers for each data set to the function ‘getGEOSuppFiles’ (see the function <a href="https://is.gd/JodM5G">getGEOSuppFiles</a> help). Some times the ftp site is not available and the information must be manually downloaded from GEO database.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Embryonic stem cells datasets</span>
<span class="va">esc.files</span> <span class="op">=</span> <span class="fu"><a href="../reference/getGEOSuppFiles.html">getGEOSuppFiles</a></span><span class="op">(</span>GEO <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GSM2041690"</span>, <span class="st">"GSM2041691"</span>, <span class="st">"GSM2041692"</span><span class="op">)</span>,
                            verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co"># Breast tissues (normal, cancer, metastasis)</span>
<span class="va">cancer.files</span> <span class="op">=</span> <span class="fu"><a href="../reference/getGEOSuppFiles.html">getGEOSuppFiles</a></span><span class="op">(</span>GEO <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GSM1279517"</span>,<span class="st">"GSM1279514"</span>,
                                        <span class="st">"GSM1279513"</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>The file path and name of each downloaded dataset is found in the output variables ‘esc.files’ and ‘cancer.files.’</p>
</div>
<div id="reading-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#reading-datasets" class="anchor"></a><em>Reading datasets</em>
</h2>
<p>Datasets for our example can be read with function <a href="https://is.gd/5pyk1N">readCounts2GRangesList</a>. To specify the reading of only chromosome 13, we can specify the parameter ‘chromosomes = “Chr13”.’ The symbol chromosome 13, in this case “Chr13,” must be consistent with the annotation provided in the given GEO dataset. Each file is wholly read with the setting ‘chromosomes = “Chr13”’ and then the GRanges are built only with chromosome 13, which could be time consuming. However, users working on Linux OS can specify the reading of specific lines from each file by using regular expressions. For example, if only chromosomes 1 and 3 are required, then we can set chromosomes = NULL (default) and ‘chromosome.pattern = “^Chr[1,3]”.’ This will read all the lines in the downloaded files starting with the words “Chr1” or “Chr3.” If we are interested in chromosomes 1 and 2, then we can set ‘chromosome.pattern = “^Chr[1-2]”.’ If all the chromosomes are required, then set chromosomes = NULL and chromosome.pattern = NULL (default). Obviously, before read the files, user must check them to see which annotation was used by the experimenters to denote the chromosomes, e.g., “chr13” or just “13,” etc.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Embryonic stem cells datasets</span>
<span class="va">ref</span> <span class="op">=</span> <span class="fu"><a href="../reference/readCounts2GRangesList.html">readCounts2GRangesList</a></span><span class="op">(</span>filenames <span class="op">=</span> <span class="va">esc.files</span>,
                            sample.id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ESC1"</span>,<span class="st">"ESC2"</span>, <span class="st">"ESC3"</span><span class="op">)</span>,
                            columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fl">1</span>, start <span class="op">=</span> <span class="fl">2</span>,
                                        mC <span class="op">=</span> <span class="fl">4</span>, uC <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"^chr13"</span>,
                            remove <span class="op">=</span> <span class="cn">TRUE</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co"># Breast tissues (normal, cancer, metastasis)</span>
<span class="va">LR</span> <span class="op">=</span> <span class="fu"><a href="../reference/readCounts2GRangesList.html">readCounts2GRangesList</a></span><span class="op">(</span>filenames <span class="op">=</span> <span class="va">cancer.files</span>,
                            sample.id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>, <span class="st">"Breast_cancer"</span>,
                                            <span class="st">"Breast_metastasis"</span><span class="op">)</span>,
                            columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fl">1</span>, start <span class="op">=</span> <span class="fl">2</span>,
                                        mC <span class="op">=</span> <span class="fl">3</span>, uC <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>,
                            remove <span class="op">=</span> <span class="cn">TRUE</span>, pattern <span class="op">=</span> <span class="st">"^13"</span>,
                            chromosome.names <span class="op">=</span> <span class="st">"chr13"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>In real data analysis, a bigger sample size must be used. For the purpose of this example, to illustrate the application of methylation analysis with Methyl-IT, we will just add an artificial normal sample.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">LR</span><span class="op">$</span><span class="va">Breast_normal1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/poolFromGRlist.html">poolFromGRlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">ref</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="va">LR</span><span class="op">$</span><span class="va">Breast_normal</span>,
                                        <span class="va">LR</span><span class="op">$</span><span class="va">Breast_normal</span>, <span class="va">LR</span><span class="op">$</span><span class="va">Breast_normal</span>,
                                        <span class="va">LR</span><span class="op">$</span><span class="va">Breast_normal</span>, <span class="va">LR</span><span class="op">$</span><span class="va">Breast_normal</span><span class="op">)</span>,
                                    stat <span class="op">=</span> <span class="st">"mean"</span>, num.cores <span class="op">=</span> <span class="fl">6L</span>,
                                    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>In the metacolumn of the last GRanges object, mC and uC stand for the methylated and unmethylated read counts, respectively. Notice that option ‘remove = TRUE’ remove the decompressed files (default: FALSE, see ? readCounts2GRangesList for more details about this function).</p>
</div>
<div id="the-reference-individual" class="section level2">
<h2 class="hasAnchor">
<a href="#the-reference-individual" class="anchor"></a>The reference individual</h2>
<p>Any two objects can be compared based on some measured variable if, and only if, the measurements where taken in the same metric space, the same coordinate system, in respect to the same origin of coordinates. Usually, in our daily 3D experience, our brain automatically sets up the origin of coordinates equal to zero. The differences found in the comparison depend on the reference used to perform the measurements and from the metric system. The space where the objects are located (or the set of objects) together with the metric is called metric space.</p>
<p>Unfortunately, this is not the case for the methylation process. Each individual from the same population and even each chromosome follows an independent stochastic methylation process. This is not a surprise, since cytosine methylation affect the mechanical properties of DNA molecule with strong dependence of the DNA sequence context (until, at least. 6 bases at both sides of each cytosine site) <span class="citation">(7, 8)</span>.</p>
<p>To evaluate the methylation differences between individuals from control and treatment we introduce a metric in the bidimensional space of methylation levels <span class="math inline">\(P_i = (p_i, 1-p_i)\)</span>. Vectors <span class="math inline">\(P_i\)</span> provide a measurement of the uncertainty of methylation levels <span class="citation">(2)</span>. However, to perform the comparison between the uncertainty of methylation levels from each group of individuals, control <span class="math inline">\((c)\)</span> and treatment <span class="math inline">\((t)\)</span>, we should estimate the uncertainty variation with respect to the same individual reference on the mentioned metric space. The reason to measure the uncertainty variation with respect to the same reference resides in that even sibling individuals follow an independent ontogenetic development and, consequently, their developments follow independent stochastic processes. This a consequence of the “omnipresent” action of the second law of thermodynamics in living organisms. In the current example, we will create the reference individual by pooling the methylation counts from the embryonic stem cells.</p>
<p>It should be noticed that the results are sensitive to the reference used. The statistics mean, median, or sum of the read counts at each cytosine site of some control samples can be used to create a virtual reference sample. A virtual reference individual can be built with function <a href="https://is.gd/qR79kW">poolFromGRlist</a>. It is up to the user whether to apply the ‘row sum,’ ‘row mean’ or ‘row median’ of methylated and unmethylated read counts at each cytosine site across individuals. Notice that when “<em>mean</em>” is selected the virtual reference individual is the group centroid, which play a fundamental role in multivariate statistic.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Ref</span> <span class="op">=</span> <span class="fu"><a href="../reference/poolFromGRlist.html">poolFromGRlist</a></span><span class="op">(</span><span class="va">ref</span>, stat <span class="op">=</span> <span class="st">"mean"</span>, num.cores <span class="op">=</span> <span class="fl">6L</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="va">Ref</span></code></pre></div>
<pre><code>## GRanges object with 1560637 ranges and 2 metadata columns:
##             seqnames    ranges strand |        mC        uC
##                &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;
##         [1]    chr13  19020631      * |         1         1
##         [2]    chr13  19020633      * |         2         2
##         [3]    chr13  19020642      * |         1         1
##         [4]    chr13  19020643      * |         2         2
##         [5]    chr13  19020679      * |         1         1
##         ...      ...       ...    ... .       ...       ...
##   [1560633]    chr13 115108993      * |         1         3
##   [1560634]    chr13 115109022      * |         1         1
##   [1560635]    chr13 115109023      * |         3         4
##   [1560636]    chr13 115109523      * |         2         2
##   [1560637]    chr13 115109524      * |         1         1
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<p>Only direct lab experiments can reveal whether differences detected with distinct references outside the experimental conditions for control and treatment groups are real. The best reference would be estimated using a subset of individuals from control group. Such a reference will contribute to remove the intragroup variation, in control and in treatment groups, induced by environmental changes external to or not controlled by the experimental conditions.</p>
<p>Methylation analysis for each cytosine position is frequently performed in the bidimensional space of <span class="math inline">\((methylated, unmethylated)\)</span> read counts. Frequently, Fisher’s exact test is applied to a single cytosine position, under the null hypothesis that the proportions <span class="math inline">\(p_{ct}=methylated_{ct}/(methylated_{ct}+unmethylated_{ct})\)</span> and <span class="math inline">\(p_{tt}=methylated_{tt}/(methylated_{tt}+unmethylated_{tt})\)</span> are the same for control and treatment, respectively. In this case, the implicit reference point for the counts at every cytosine positions is <span class="math inline">\((methylated = 0,unmethylated = 0)\)</span>, which corresponds to the point <span class="math inline">\(P_i = (0, 1)\)</span>.</p>
<p>In our case, the Hellinger divergence (one of the metric used, here) of each individual in respect to the reference is the variable to test in place of <span class="math inline">\((methylated, unmethylated)\)</span> read counts or the methylation levels <span class="math inline">\(P_i = (p_i, 1-p_i)\)</span>.</p>
<p>The use of references is restricted by the thermodynamics basis of the theory. The current information-thermodynamics based approach is supported on the following postulate: <em>“High changes of Hellinger divergences are less frequent than low changes, provided that the divergence is proportional to the amount of energy required to process the change of information in the methylation system”</em>.</p>
<p>The last postulate acknowledges the action of the second law of thermodynamics on the biomolecular methylation system. For the methylation system, it implies that the frequencies of the information divergences between methylation levels must be proportional to a Boltzmann factor (see supplementary information from reference <span class="citation">(1)</span>). In other words, the frequencies of information divergences values should follow a trend proportional to an exponential decay.</p>
<p>If we do not observe such a behavior, then either the reference is too far from experimental condition or we are dealing with an extreme situation where the methylation machinery in the cell is dysfunctional. The last situation is found, for example, in the silencing mutation at the gene of cytosine-DNA-methyltransferase in Arabidopsis <em>thaliana</em>. Methylation of 5-methylcytosine at CpG dinucleotides is maintained by MET1 in plants.</p>
</div>
<div id="hellinger-divergence-estimation" class="section level2">
<h2 class="hasAnchor">
<a href="#hellinger-divergence-estimation" class="anchor"></a>Hellinger divergence estimation</h2>
<p>To perform the comparison between the uncertainty of methylation levels from each group of individuals, control <span class="math inline">\((c)\)</span> and treatment $(t) $, the divergence between the methylation levels of each individual is estimated with respect to the same reference on the metric space formed by the vector set <span class="math inline">\(P_i = (p_i, 1-p_i)\)</span> and the Hellinger divergence <span class="math inline">\(H\)</span>. Basically, the information divergence between the methylation levels of an individual <span class="math inline">\(j\)</span> and reference sample <span class="math inline">\(r\)</span> is estimated according to the Hellinger divergence given by the formula:</p>
<p><span class="math display">\[H(\hat p_{ij},\hat p_{ir}) = w_i[(\sqrt{\hat p_{ij}} - \sqrt{\hat
p_{ir}})^2+(\sqrt{1-\hat p_{ij}} - \sqrt{1-\hat p_{ir}})^2]\]</span> where $w_i = 2</p>
<p>$, <span class="math inline">\(m_{ij} = {n_i}^{mC_j}+{n_i}^{uC_j}+1\)</span>, <span class="math inline">\(m_{ir} = {n_i}^{mC_r}+ {n_i}^{uC_r}+1\)</span> and <span class="math inline">\(j \in {\{c,t}\}\)</span>.</p>
<p>This equation for Hellinger divergence is given in reference <span class="citation">(9)</span>, but other information theoretical divergences can be used as well. Next, the information divergence for control (Breast_normal) and treatment (Breast_cancer and Breast_metastasis) samples are estimated with respect to the reference virtual individual. A Bayesian correction of counts can be selected or not. In a Bayesian framework, methylated read counts are modeled by a beta-binomial distribution, which accounts for both the biological and sampling variations <span class="citation">(10–12)</span>.</p>
<p>In our case we adopted the Bayesian approach suggested in reference <span class="citation">(13)</span> (Chapter 3). In a Bayesian framework with uniform priors, the methylation level can be defined as: <span class="math inline">\(p = (mC + 1)/ (mC + uC + 2)\)</span>. However, the most natural statistical model for replicated BS-seq DNA methylation measurements is beta-binomial (the beta distribution is a prior conjugate of binomial distribution). We consider the parameter <span class="math inline">\(p\)</span> (methylation level) in the binomial distribution as randomly drawn from a beta distribution. The hyper-parameters <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span> from the beta-binomial distribution are interpreted as pseudo-counts. The information divergence is estimated here using the function <a href="https://is.gd/0ViBO9">estimateDivergence</a>:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">HD</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateDivergence.html">estimateDivergence</a></span><span class="op">(</span>ref <span class="op">=</span> <span class="va">Ref</span>, indiv <span class="op">=</span> <span class="va">LR</span>, Bayesian <span class="op">=</span> <span class="cn">TRUE</span>,
                        min.coverage <span class="op">=</span> <span class="fl">5</span>, high.coverage <span class="op">=</span> <span class="fl">300</span>,
                        percentile <span class="op">=</span> <span class="fl">0.999</span>, num.cores <span class="op">=</span> <span class="fl">6L</span>, tasks <span class="op">=</span> <span class="fl">0L</span>,
                        verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="va">HD</span><span class="op">$</span><span class="va">Breast_cancer</span></code></pre></div>
<pre><code>## GRanges object with 584538 ranges and 9 metadata columns:
##            seqnames    ranges strand |        c1        t1        c2        t2
##               &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##        [1]    chr13  19020631      * |         1         1        14        24
##        [2]    chr13  19020633      * |         2         2        14        25
##        [3]    chr13  19020643      * |         2         2         7        38
##        [4]    chr13  19020782      * |         1         1         6        56
##        [5]    chr13  19020786      * |         1         1        10        53
##        ...      ...       ...    ... .       ...       ...       ...       ...
##   [584534]    chr13 115108776      * |         1         1        52        20
##   [584535]    chr13 115108789      * |         2         2        27        43
##   [584536]    chr13 115108993      * |         1         3        72         5
##   [584537]    chr13 115109023      * |         3         4        56        36
##   [584538]    chr13 115109524      * |         1         1        31         9
##                   p1        p2        TV     bay.TV      hdiv
##            &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;
##        [1]  0.847247  0.386143 -0.131579  -0.461103   1.35726
##        [2]  0.783075  0.376688 -0.141026  -0.406386   1.58528
##        [3]  0.783075  0.178991 -0.344444  -0.604083   3.67278
##        [4]  0.847247  0.115664 -0.403226  -0.731583   3.65879
##        [5]  0.847247  0.175570 -0.341270  -0.671677   2.97393
##        ...       ...       ...       ...        ...       ...
##   [584534]  0.847247  0.722968  0.222222 -0.1242787 0.1342042
##   [584535]  0.783075  0.395100 -0.114286 -0.3879750 1.5233014
##   [584536]  0.690673  0.930796  0.685065  0.2401223 0.9753622
##   [584537]  0.685463  0.611508  0.180124 -0.0739546 0.0885556
##   [584538]  0.847247  0.773995  0.275000 -0.0732517 0.0492369
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<p>Function ‘estimateDivergence’ returns a list of GRanges objects with the four columns of counts, the information divergence, and additional columns:</p>
<pre><code>1. The original matrix of methylated ($c_i$) and unmethylated
    ($t_i$) read counts from control ($i=1$) and treatment ($i=2$) samples.
2. "p1" and "p2": methylation levels for control and treatment,
    respectively.
3. "bay.TV": total variation TV = p2 - p1.
4. "TV": total variation based on simple counts: $TV=c1/(c1+t1)-
    c2/(c2+t2)$.
5. "hdiv": Hellinger divergence.</code></pre>
<p>If Bayesian = TRUE, results are based on the posterior estimations of methylation levels <span class="math inline">\(p1\)</span> and <span class="math inline">\(p2\)</span>. Filtering by coverage is provided at this step which would be used unless previous filtering by coverage had been applied. This is a pairwise filtering. Cytosine sites with ‘coverage’ &gt; ‘min.coverage’ and ‘coverage’ &lt; ‘percentile’ (e.g., 99.9 coverage percentile) in at least one of the samples are preserved. The coverage percentile used is the maximum estimated from both samples: reference and individual.</p>
<p>For some GEO datasets only the methylation levels for each cytosine site are provided. In this case, Hellinger divergence can be estimated as given in reference <span class="citation">(1)</span>:</p>
<p><span class="math display">\[H(\hat p_{ij},\hat p_{ir}) = 2 \Big[(\sqrt{\hat p_{ij}} - \sqrt{\hat
p_{ir}})^2+(\sqrt{1-\hat p_{ij}} - \sqrt{1-\hat p_{ir}})^2\Big]\]</span></p>
</div>
<div id="descriptive-statistical-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#descriptive-statistical-analysis" class="anchor"></a><em>Descriptive statistical analysis</em>
</h2>
<p>A descriptive analysis on the distribution of Hellinger divergences of methylation levels is recommended. Like in any other statistical analysis, descriptive statistical analysis help us to detect potential issues in the raw data. It is the user responsibility to perform quality check of his/her dataset before start to apply Methyl-IT. Nevertheless, the quality checking of the raw data is not perfect. So, it is healthy to sees for potential issues.</p>
<div id="histogram-and-boxplots-of-divergences-estimated-in-each-sample" class="section level3">
<h3 class="hasAnchor">
<a href="#histogram-and-boxplots-of-divergences-estimated-in-each-sample" class="anchor"></a><em>Histogram and boxplots of divergences estimated in each sample</em>
</h3>
<p>First, the data of interest (Hellinger divergences, “hdiv”) are selected from the GRanges objects:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">normal</span> <span class="op">=</span> <span class="va">HD</span><span class="op">$</span><span class="va">Breast_normal</span><span class="op">[</span>, <span class="st">"hdiv"</span><span class="op">]</span>
<span class="va">normal1</span> <span class="op">=</span> <span class="va">HD</span><span class="op">$</span><span class="va">Breast_normal1</span><span class="op">[</span>, <span class="st">"hdiv"</span><span class="op">]</span>
<span class="va">cancer</span> <span class="op">=</span> <span class="va">HD</span><span class="op">$</span><span class="va">Breast_cancer</span><span class="op">[</span>, <span class="st">"hdiv"</span><span class="op">]</span>
<span class="va">metastasis</span> <span class="op">=</span> <span class="va">HD</span><span class="op">$</span><span class="va">Breast_metastasis</span><span class="op">[</span>, <span class="st">"hdiv"</span><span class="op">]</span></code></pre></div>
<p>Next, a single GRanges object is built from the above set of GRanges objects using the function <a href="https://is.gd/S6e4br">uniqueGRanges</a>. Notice that the number of cores to use for parallel computation can be specified.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hd</span> <span class="op">=</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">normal</span>, <span class="va">normal1</span>, <span class="va">cancer</span>, <span class="va">metastasis</span><span class="op">)</span>,
                    missing <span class="op">=</span> <span class="cn">NA</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, num.cores <span class="op">=</span> <span class="fl">6L</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">hd</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"normal"</span>, <span class="st">"normal1"</span>, <span class="st">"cancer"</span>, <span class="st">"metastasis"</span><span class="op">)</span>
<span class="va">hd</span></code></pre></div>
<pre><code>## GRanges object with 793551 ranges and 4 metadata columns:
##            seqnames    ranges strand |     normal    normal1    cancer
##               &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;
##        [1]    chr13  19020631      * | 0.23426425  0.2202754   1.35726
##        [2]    chr13  19020633      * | 0.00504582  0.0218855   1.58528
##        [3]    chr13  19020643      * | 0.08442261  0.0707124   3.67278
##        [4]    chr13  19020680      * | 0.47116162  0.4551107        NA
##        [5]    chr13  19020687      * | 1.15699629  1.0412108        NA
##        ...      ...       ...    ... .        ...        ...       ...
##   [793547]    chr13 115108776      * | 0.08604288 0.08290688 0.1342042
##   [793548]    chr13 115108789      * | 0.14205651 0.15221981 1.5233014
##   [793549]    chr13 115108993      * | 0.71277914 0.58702523 0.9753622
##   [793550]    chr13 115109023      * | 0.71153929 0.71910910 0.0885556
##   [793551]    chr13 115109524      * | 0.00320206 0.00321577 0.0492369
##             metastasis
##              &lt;numeric&gt;
##        [1]    0.351124
##        [2]    0.939944
##        [3]    0.914811
##        [4]          NA
##        [5]          NA
##        ...         ...
##   [793547]          NA
##   [793548]          NA
##   [793549] 1.495097458
##   [793550] 0.000630095
##   [793551]          NA
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<p>Now, the Hellinger divergences estimated for each sample are in a single matrix on the metacolumn of the GRanges object and we can proceed to build the histogram and boxplot graphics for these data. Finally, the data included in the graphic are:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Define an auxiliary function</span>
<span class="va">fun_length</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="va">x</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span>, 
                      label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"n = "</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>normal <span class="op">=</span> <span class="va">hd</span><span class="op">$</span><span class="va">normal</span>, normal1 <span class="op">=</span> <span class="va">hd</span><span class="op">$</span><span class="va">normal1</span>,
                    cancer <span class="op">=</span> <span class="va">hd</span><span class="op">$</span><span class="va">cancer</span>, metastasis <span class="op">=</span> <span class="va">hd</span><span class="op">$</span><span class="va">metastasis</span><span class="op">)</span>
<span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast.tissue"</span>, <span class="st">"HD"</span><span class="op">)</span>
<span class="va">data</span> <span class="op">=</span> <span class="va">data</span><span class="op">[</span><span class="va">data</span><span class="op">$</span><span class="va">HD</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="op">]</span>
<span class="fu">DataFrame</span><span class="op">(</span><span class="va">data</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 3174204 rows and 2 columns
##           Breast.tissue          HD
##                &lt;factor&gt;   &lt;numeric&gt;
## 1                normal  0.23426425
## 2                normal  0.00504582
## 3                normal  0.08442261
## 4                normal  0.47116162
## 5                normal  1.15699629
## ...                 ...         ...
## NA.617427    NA                  NA
## NA.617428    NA                  NA
## 3174202      metastasis 1.495097458
## 3174203      metastasis 0.000630095
## NA.617429    NA                  NA</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">HD</span>, fill <span class="op">=</span> <span class="va">Breast.tissue</span>, colour <span class="op">=</span> <span class="va">Breast.tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html">geom_histogram</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span>, binwidth <span class="op">=</span> <span class="fl">0.5</span>, position <span class="op">=</span> <span class="st">"identity"</span>,
                na.rm <span class="op">=</span> <span class="cn">TRUE</span>, size <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.4</span>,<span class="fl">20</span><span class="op">)</span> <span class="op">+</span>
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,
                axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, 
                                        color <span class="op">=</span>  <span class="st">"black"</span>, hjust <span class="op">=</span> <span class="fl">0.5</span>, 
                                        vjust <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span>,
        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>,color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,
        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span> <span class="st">"Counts"</span> <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Histogram of Hellinger Divergence"</span><span class="op">)</span>

<span class="co"># For visualization purposes HD is limited to the interval 0 to 50</span>
<span class="va">dt</span> <span class="op">=</span> <span class="va">data</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">HD</span> <span class="op">&lt;</span> <span class="fl">50</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">dt</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Breast.tissue</span>, y <span class="op">=</span> <span class="va">HD</span> , fill <span class="op">=</span> <span class="va">Breast.tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span>na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">ylim</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.1</span>, <span class="fl">20</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_summary.html">stat_summary</a></span><span class="op">(</span>fun.data <span class="op">=</span> <span class="va">fun_length</span>, geom <span class="op">=</span> <span class="st">"text"</span>,
                position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html">position_dodge</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="fl">1</span>,
                size <span class="op">=</span> <span class="fl">6</span>, fontface <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,
        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span>,
                                    hjust <span class="op">=</span> <span class="fl">0.5</span>, vjust <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span>,
        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>,color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        legend.position <span class="op">=</span> <span class="st">"none"</span> <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Boxplot of Hellinger Divergence"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: Removed 11 rows containing non-finite values (stat_summary).</code></pre>
<p><img src="cancer_example_files/figure-html/fig1-1.png" width="1920"></p>
<p>Except for the tail, most of the methylation changes occurred under the area covered by the density curve corresponding to the normal breast tissue. This is theoretically expected. This area is explainable in statistical physical terms and, theoretically, it should fit a Weibull distribution. The tails regions cover the methylation changes that, with high probability, are not induced by thermal fluctuation and are not addressed to stabilize the DNA molecule. These changes are methylation signal. Professor David J. Miller (Department of Electrical Engineering, Penn State) proposed modeling the distribution as a mixed Weibull distribution to simultaneously describe the background methylation noise and the methylation signal (personal communication, January, 2018). This model approach seems to be supported by the above histogram, but it must be studied before being incorporated in a future version of Methyl-IT.</p>
</div>
<div id="checking-cytosine-read-counts" class="section level3">
<h3 class="hasAnchor">
<a href="#checking-cytosine-read-counts" class="anchor"></a><em>Checking cytosine read counts</em>
</h3>
<p>According to the parameter values set for the function <a href="https://is.gd/0ViBO9">estimateDivergence</a>, for each cytosine site, at least one individual must have a 10 reads or more, and the maximum coverage must be 500 or less.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Maximum coverage at each cytosine site</span>
<span class="va">covr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">HD</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">cov1</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">c1</span> <span class="op">+</span> <span class="va">x</span><span class="op">$</span><span class="va">t1</span>
    <span class="va">cov2</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">c2</span> <span class="op">+</span> <span class="va">x</span><span class="op">$</span><span class="va">t2</span>
    <span class="va">cov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">cov1</span>, <span class="va">cov2</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">max</span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">cov</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># Quantiles</span>
<span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">covr</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">q60</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.6</span><span class="op">)</span>
    <span class="va">q9999</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.9999</span><span class="op">)</span>
    <span class="va">idx1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="va">q60</span><span class="op">)</span>
    <span class="va">idx2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span>
    <span class="va">q95</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.95</span><span class="op">)</span>

    <span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span><span class="op">(</span><span class="va">idx1</span>, <span class="va">idx2</span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span>, <span class="va">q60</span>, 
            <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.95</span>, <span class="fl">0.99</span>, <span class="fl">0.999</span>, <span class="fl">0.9999</span><span class="op">)</span><span class="op">)</span>,
            num.siteGreater_8 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="fl">8</span><span class="op">)</span>,
            q60_to_500 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="va">q60</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">(</span><span class="va">x</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span>,
            num.siteGreater_500 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<pre><code>##                   Min. 1st Qu. Median Mean 3rd Qu. Max. 60% 95% 99% 99.9%
## Breast_normal        5      19     30   31      41  291  34  57  69    81
## Breast_cancer        5      18     31   37      51  266  38  88 112   137
## Breast_metastasis    5      14     27   31      43  281  32  70  91   115
## Breast_normal1       5      17     26   27      35  279  29  48  58    68
##                     99.99% num.siteGreater_8 q60_to_500 num.siteGreater_500
## Breast_normal      94.0000            747005     319227                   0
## Breast_cancer     165.0000            569474     234247                   0
## Breast_metastasis 158.8469            435153     187884                   0
## Breast_normal1     80.0000            738787     318786                   0</code></pre>
<p>The quantiles indicates acceptable values for the coverages. Descriptive statistics for methylated reads can be checked as well.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Maximum coverage at each cytosine site</span>
<span class="va">methc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">HD</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">c1</span>, <span class="va">x</span><span class="op">$</span><span class="va">c2</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">max</span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">r</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># Quantiles</span>
<span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">methc</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">q10</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.1</span><span class="op">)</span>
    <span class="va">q60</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.6</span><span class="op">)</span>
    <span class="va">q9999</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.9999</span><span class="op">)</span>
    <span class="va">idx1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="va">q60</span><span class="op">)</span>
    <span class="va">idx2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span>
    <span class="va">q95</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.95</span><span class="op">)</span>

    <span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span><span class="op">(</span><span class="va">idx1</span>, <span class="va">idx2</span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span>, <span class="va">q10</span>, <span class="va">q60</span>,
            <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.95</span>, <span class="fl">0.99</span>, <span class="fl">0.999</span>, <span class="fl">0.9999</span><span class="op">)</span><span class="op">)</span>,
            num.siteGreater_8 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="fl">8</span><span class="op">)</span>,
            q60_to_500 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="va">q60</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">(</span><span class="va">x</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span>,
            num.siteGreater_500 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<pre><code>##                   Min. 1st Qu. Median Mean 3rd Qu. Max. 10% 60% 95% 99% 99.9%
## Breast_normal        4      14     24   25      34  265   7  28  49  60    72
## Breast_cancer        4      10     21   27      38  252   5  27  74  98   123
## Breast_metastasis    4       7     16   21      30  234   4  21  56  75    98
## Breast_normal1       4      12     20   21      29  258   6  24  41  50    60
##                     99.99% num.siteGreater_8 q60_to_500 num.siteGreater_500
## Breast_normal      84.0000            677552     309300                   0
## Breast_cancer     149.5463            472596     237092                   0
## Breast_metastasis 121.0000            328544     184064                   0
## Breast_normal1     70.0000            664530     304195                   0</code></pre>
<p>The critical values from empirical cumulative probability distributions will be used in the downstream analysis.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">critical.val</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">HD</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">hd.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">hdiv</span>, <span class="fl">0.95</span><span class="op">)</span>
    <span class="va">tv.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">bay.TV</span><span class="op">)</span>, <span class="fl">0.95</span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>tv <span class="op">=</span> <span class="va">tv.95</span>, hd <span class="op">=</span> <span class="va">hd.95</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span><span class="op">)</span>

<span class="va">critical.val</span></code></pre></div>
<pre><code>##                      tv.95%   hd.95%
## Breast_normal     0.3093663 1.448279
## Breast_cancer     0.5564614 2.967515
## Breast_metastasis 0.5806465 3.488804
## Breast_normal1    0.2932203 1.178657</code></pre>
</div>
</div>
<div id="nonlinear-fit-of-probability-distribution-models" class="section level2">
<h2 class="hasAnchor">
<a href="#nonlinear-fit-of-probability-distribution-models" class="anchor"></a>Nonlinear fit of probability distribution models</h2>
<p>A basic requirement for the application of signal detection is the knowledge of the probability distribution of the background noise. Probability distribution, as a Weibull distribution model, can be deduced on a statistical mechanical/thermodynamics basis for DNA methylation induced by thermal fluctuations <span class="citation">(1)</span>.</p>
<p>Assuming that this background methylation variation is consistent with a Poisson process, it can be distinguished from variation associated with methylation regulatory machinery, which is non-independent for all genomic regions <span class="citation">(1)</span>. An information-theoretic divergence to express the variation in methylation induced by background thermal fluctuations will follow a Weibull distribution model, provided that it is proportional to the minimum energy dissipated per bit of information associated with the methylation change.</p>
<p>The nonlinear fit to a Weibull distribution model is performed by the function <a href="https://is.gd/0BLLkT">nonlinearFitDist</a>. In general, an information divergence of methylation levels will follow a probability distribution from the family of generalized gamma distribution. Next, the best fitted model will be selected from one of the following distributions: two- and three-parameter Weibull distributions, gamma with two- and three-parameters (“Gamma2P” or “Gamma3P”), and generalized gamma with three-parameter (“GGamma3P”).</p>
<p>The best model is selected on the basis of Akaike’s information criterion and the correlation coefficient of cross-validations for the nonlinear regressions (R.Cross.val). These criteria evaluate different information inferred from the models. AIC deals with the trade-off between the goodness of fit (GOF) of the model and the complexity of the model, while R.Cross.val provides information on the prediction power/performance of the model when confronted with external dataset. Cross-validations for the nonlinear regressions (R.Cross.val) were performed as described in reference <span class="citation">(14)</span>. In addition, Stein’s formula for adjusted R squared (<span class="math inline">\(\rho\)</span>) was used as an estimator of the average cross-validation predictive power <span class="citation">(14)</span>.</p>
<p>In general, the best fitted model yields the best DMP classification performance (see below). However, on special dataset, as the used in the current example. Function <a href="https://is.gd/W7dlP0">gofReport</a> search for the best fitted model between the set of models requested by the user.</p>
<p>The best model including “GGamma3P” model:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Weibull2P"</span>, <span class="st">"Weibull3P"</span>, <span class="st">"Gamma2P"</span>, <span class="st">"Gamma3P"</span>, <span class="st">"GGamma3P"</span><span class="op">)</span>
<span class="va">gof_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gofReport.html">gofReport</a></span><span class="op">(</span>HD <span class="op">=</span> <span class="va">HD</span>, column <span class="op">=</span> <span class="fl">9L</span>, model <span class="op">=</span> <span class="va">d</span>, num.cores <span class="op">=</span> <span class="fl">6L</span>,
                    output <span class="op">=</span> <span class="st">"all"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |======================================================================| 100%
## 
##  *** Creating report ...</code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gof_1</span><span class="op">$</span><span class="va">bestModel</span></code></pre></div>
<pre><code>##     Breast_normal     Breast_cancer Breast_metastasis    Breast_normal1 
##       "Weibull2P"        "GGamma3P"        "GGamma3P"        "GGamma3P"</code></pre>
<p>The best model excluding “GGamma3P” model:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Weibull2P"</span>, <span class="st">"Weibull3P"</span>, <span class="st">"Gamma2P"</span>, <span class="st">"Gamma3P"</span><span class="op">)</span>
<span class="va">gof_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gofReport.html">gofReport</a></span><span class="op">(</span>HD <span class="op">=</span> <span class="va">HD</span>, column <span class="op">=</span> <span class="fl">9L</span>, model <span class="op">=</span> <span class="va">d</span>, num.cores <span class="op">=</span> <span class="fl">6L</span>, 
                    output <span class="op">=</span> <span class="st">"all"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%
## 
##  *** Creating report ...</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gof_2</span><span class="op">$</span><span class="va">bestModel</span></code></pre></div>
<pre><code>##     Breast_normal     Breast_cancer Breast_metastasis    Breast_normal1 
##       "Weibull2P"       "Weibull2P"       "Weibull2P"         "Gamma2P"</code></pre>
<p>When “GGamma3P” model is included, the goodness-of-fit indicators used here suggest that it is best fitted model. However, a final decision must take into account the DMP classification performance as well (see below), which is accomplished after a signal detection step. The reader must keep in mind that we are dealing here with numerical algorithms, which are not perfect. Depending on dataset, the numerical computation will confront different challengers. The difference between g3p_R.Cross.val and gg3p_R.Cross.val suggests that model predictions powers for “GGamma3P” and “Gamma3P” are close.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gof_1</span><span class="op">$</span><span class="va">stats</span></code></pre></div>
<pre><code>##                    w2p_AIC w2p_R.Cross.val w3p_AIC w3p_R.Cross.val  g2p_AIC
## Breast_normal     -5402841       0.9997066      NA              NA -4642723
## Breast_cancer     -3364404       0.9988986      NA              NA -3067557
## Breast_metastasis -2275378       0.9977765      NA              NA -2099156
## Breast_normal1    -5163449       0.9996714      NA              NA -5712541
##                   g2p_R.Cross.val  g3p_AIC g3p_R.Cross.val gg3p_AIC
## Breast_normal           0.9992363 -4675350       0.9992639 -5341937
## Breast_cancer           0.9983518 -3067261       0.9983491 -3414837
## Breast_metastasis       0.9969408 -2100775       0.9969462 -2371248
## Breast_normal1          0.9998179      Inf       0.0000000 -6174567
##                   gg3p_R.Cross.val bestModel
## Breast_normal            0.9996828       w2p
## Breast_cancer            0.9990857      gg3p
## Breast_metastasis        0.9984283      gg3p
## Breast_normal1           0.9999009      gg3p</code></pre>
<p>It is worthy to observe that although in the current example we successfully performed the nonlinear fit for GGamma3P model, this is not the general case when the number of cytosine sites goes close or above <span class="math inline">\(10^6\)</span>, for which the computational cost is very high. The nonlinear fit of GGamma3P or GGamma4P is, in general, difficult since two different set of estimated parameters could produce very close distribution curves. Also, it is important to check the value of the scaling parameter from each distribution. The numerical algorithms could find the best fitted model with scaling parameter values close to zero, say e.g. <span class="math inline">\(10^{-8}\)</span>, and with the highest GOFs. Situations like this are biologically meaningless and the fitted models must be discarded <span class="citation">(1)</span>.</p>
</div>
<div id="signal-detection" class="section level2">
<h2 class="hasAnchor">
<a href="#signal-detection" class="anchor"></a>Signal detection</h2>
<p>The information thermodynamics-based approach is postulated to provide greater sensitivity for resolving true signal from the thermodynamic background within the methylome <span class="citation">(1)</span>. Since the biological signal created within the dynamic methylome environment characteristic of living organisms is not free from background noise, the approach, designated Methyl-IT, includes the application of signal detection theory <span class="citation">(3–6)</span>. Signal detection is a critical step to increase sensitivity and resolution of methylation signal by reducing the signal-to-noise ratio and objectively controlling the false positive rate and prediction accuracy/risk.</p>
</div>
<div id="potential-methylation-signal" class="section level2">
<h2 class="hasAnchor">
<a href="#potential-methylation-signal" class="anchor"></a><em>Potential methylation signal</em>
</h2>
<p>The first estimation in our signal detection step is the identification of the cytosine sites carrying potential methylation signal <span class="math inline">\(PS\)</span>. The methylation regulatory signal does not hold the theoretical distribution and, consequently, for a given level of significance <span class="math inline">\(\alpha\)</span> (Type I error probability, e.g. <span class="math inline">\(\alpha = 0.05\)</span>), cytosine positions <span class="math inline">\(k\)</span> with information divergence <span class="math inline">\(H_k &gt;= H_{\alpha = 0.05}\)</span> can be selected as sites carrying potential signals <span class="math inline">\(PS\)</span>. The value of <span class="math inline">\(\alpha\)</span> can be specified.</p>
<p>For example, potential signals with <span class="math inline">\(H_k &gt; H_{\alpha = 0.01}\)</span> can be selected. For each sample, cytosine sites are selected based on the corresponding fitted theoretical distribution model estimated in the previous step. Additionally, since cytosine with <span class="math inline">\(|TV_{d_k}| &lt; 0.1\)</span> are the most abundant sites, depending on the sample (experiment), cytosine positions <span class="math inline">\(k\)</span> with <span class="math inline">\(H_k &gt;= H_{\alpha = 0.05}\)</span> and <span class="math inline">\(|TV_{d_k}| &lt; 0.1\)</span> can be observed. To prevent the last situation we can select the <span class="math inline">\(PS\)</span> with the additional constraint <span class="math inline">\(|TV_{d_k}| &gt; TV_0\)</span>, where <span class="math inline">\(TV_0\)</span> (‘tv.cut’) is a user specified value. The <span class="math inline">\(PS\)</span> is detected with the function <a href="https://is.gd/yGORtg">getPotentialDIMP</a>:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">PS1</span> <span class="op">=</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">HD</span>, dist.name <span class="op">=</span> <span class="va">gof_1</span><span class="op">$</span><span class="va">bestModel</span>, nlms <span class="op">=</span> <span class="va">gof_1</span><span class="op">$</span><span class="va">nlms</span>,
                        div.col <span class="op">=</span> <span class="fl">9</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, tv.col <span class="op">=</span> <span class="fl">8</span>, tv.cut <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="va">PS2</span> <span class="op">=</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">HD</span>, dist.name <span class="op">=</span> <span class="va">gof_2</span><span class="op">$</span><span class="va">bestModel</span>, nlms <span class="op">=</span> <span class="va">gof_2</span><span class="op">$</span><span class="va">nlms</span>,
                        div.col <span class="op">=</span> <span class="fl">9</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, tv.col <span class="op">=</span> <span class="fl">8</span>, tv.cut <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="va">PS1</span><span class="op">$</span><span class="va">Breast_normal</span></code></pre></div>
<pre><code>## GRanges object with 7261 ranges and 10 metadata columns:
##          seqnames    ranges strand |        c1        t1        c2        t2
##             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##      [1]    chr13  19022281      * |         1         2         7        32
##      [2]    chr13  19029464      * |         4         5         1        22
##      [3]    chr13  19035351      * |         1         3         5        37
##      [4]    chr13  19076973      * |         1         2         4        21
##      [5]    chr13  19085115      * |         0         1         8        43
##      ...      ...       ...    ... .       ...       ...       ...       ...
##   [7257]    chr13 115046280      * |         1         2         6        34
##   [7258]    chr13 115046470      * |         1         3         7        51
##   [7259]    chr13 115046475      * |         1         3         5        49
##   [7260]    chr13 115077626      * |         2         2        13        42
##   [7261]    chr13 115105490      * |         2         2         4        22
##                 p1        p2        TV    bay.TV      hdiv      wprob
##          &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;
##      [1]  0.777611  0.238760 -0.153846 -0.538850   2.29329 0.02321319
##      [2]  0.678080  0.157892 -0.400966 -0.520187   4.29543 0.00389135
##      [3]  0.711347  0.179653 -0.130952 -0.531694   2.79317 0.01424325
##      [4]  0.777611  0.250594 -0.173333 -0.527017   2.08451 0.02880588
##      [5]  0.841173  0.204821  0.156863 -0.636351   1.76831 0.04060771
##      ...       ...       ...       ...       ...       ...        ...
##   [7257]  0.777611  0.210578 -0.183333 -0.567032   2.57071 0.01762194
##   [7258]  0.711347  0.165539 -0.129310 -0.545808   3.06173 0.01110891
##   [7259]  0.711347  0.142455 -0.157407 -0.568892   3.37593 0.00839203
##   [7260]  0.796561  0.275735 -0.263636 -0.520827   2.70800 0.01544046
##   [7261]  0.796561  0.242228 -0.346154 -0.554333   2.83681 0.01367128
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">PS2</span><span class="op">$</span><span class="va">Breast_normal</span></code></pre></div>
<pre><code>## GRanges object with 7261 ranges and 10 metadata columns:
##          seqnames    ranges strand |        c1        t1        c2        t2
##             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##      [1]    chr13  19022281      * |         1         2         7        32
##      [2]    chr13  19029464      * |         4         5         1        22
##      [3]    chr13  19035351      * |         1         3         5        37
##      [4]    chr13  19076973      * |         1         2         4        21
##      [5]    chr13  19085115      * |         0         1         8        43
##      ...      ...       ...    ... .       ...       ...       ...       ...
##   [7257]    chr13 115046280      * |         1         2         6        34
##   [7258]    chr13 115046470      * |         1         3         7        51
##   [7259]    chr13 115046475      * |         1         3         5        49
##   [7260]    chr13 115077626      * |         2         2        13        42
##   [7261]    chr13 115105490      * |         2         2         4        22
##                 p1        p2        TV    bay.TV      hdiv      wprob
##          &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;
##      [1]  0.777611  0.238760 -0.153846 -0.538850   2.29329 0.02321319
##      [2]  0.678080  0.157892 -0.400966 -0.520187   4.29543 0.00389135
##      [3]  0.711347  0.179653 -0.130952 -0.531694   2.79317 0.01424325
##      [4]  0.777611  0.250594 -0.173333 -0.527017   2.08451 0.02880588
##      [5]  0.841173  0.204821  0.156863 -0.636351   1.76831 0.04060771
##      ...       ...       ...       ...       ...       ...        ...
##   [7257]  0.777611  0.210578 -0.183333 -0.567032   2.57071 0.01762194
##   [7258]  0.711347  0.165539 -0.129310 -0.545808   3.06173 0.01110891
##   [7259]  0.711347  0.142455 -0.157407 -0.568892   3.37593 0.00839203
##   [7260]  0.796561  0.275735 -0.263636 -0.520827   2.70800 0.01544046
##   [7261]  0.796561  0.242228 -0.346154 -0.554333   2.83681 0.01367128
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<p>Model “GGamma3P” reported 20263 - 11314 = 8949 more DMPs in the normal tissue than model “Gamma3P.”</p>
<p>Notice that the total variation distance <span class="math inline">\(TV_d=|TV|\)</span> is an information divergence as well and that it can be used in place of Hellinger divergence <span class="citation">(1)</span>. The set of vectors <span class="math inline">\(P_i = (p_i, 1-p_i)\)</span> and distance function <span class="math inline">\(TV_d\)</span> integrate a metric space <span class="citation">(2)</span>. In particular, after takes the Manhattan distance between healthy individuals and patients methylation levels we have:</p>
<p><span class="math display">\[M_d(p_{ij}, p_{ir})=\frac{1}{2}(|\hat p_{ij} - \hat p_{ir}| + |
(1-\hat p_{ij}) - (1-\hat p_{ir})|)=|\hat p_{ij} - \hat p_{ir}|=TV_d(p_{ij},
p_{ir})\]</span>.</p>
<p>That is, the absolute difference of methylation levels is a particular case of Manhattan distance, which is named total variation distance <span class="math inline">\(TV_d\)</span>. The quantitative effect of the vector components <span class="math inline">\(1-\hat p_{ij}\)</span> and <span class="math inline">\(1-\hat p_{ir}\)</span> (in our case, the effect of unmethylated read counts) is not present in <span class="math inline">\(TV_d\)</span> as in <span class="math inline">\(H(\hat p_{ij},\hat p_{ir})\)</span>.</p>
</div>
<div id="histogram-and-boxplots-of-methylation-potential-signals" class="section level2">
<h2 class="hasAnchor">
<a href="#histogram-and-boxplots-of-methylation-potential-signals" class="anchor"></a><em>Histogram and boxplots of methylation potential signals</em>
</h2>
<p>As before, a single GRanges object is built from the above set GRanges objects using the function <a href="https://is.gd/S6e4br">uniqueGRanges</a>, and the Hellinger divergences of the cytosine sites carrying <span class="math inline">\(PS\)</span> (for each sample) are located in a single matrix on the metacolumn of the GRanges object.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Define an auxiliar function</span>
<span class="va">fun_length</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1.5</span>, label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"n = "</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>

<span class="co"># The data to be used in the boxplot</span>
<span class="va">ps1</span> <span class="op">=</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">PS1</span>, missing <span class="op">=</span> <span class="cn">NA</span>, columns <span class="op">=</span> <span class="fl">9</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>,
                    num.cores <span class="op">=</span> <span class="fl">6L</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">ps1</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"normal"</span>, <span class="st">"cancer"</span>, <span class="st">"metastasis"</span>, <span class="st">"normal1"</span><span class="op">)</span>
<span class="va">dat1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>normal <span class="op">=</span> <span class="va">ps1</span><span class="op">$</span><span class="va">normal</span>, normal1 <span class="op">=</span> <span class="va">ps1</span><span class="op">$</span><span class="va">normal1</span>,
                cancer <span class="op">=</span> <span class="va">ps1</span><span class="op">$</span><span class="va">cancer</span>, metastasis <span class="op">=</span> <span class="va">ps1</span><span class="op">$</span><span class="va">metastasis</span><span class="op">)</span>
<span class="va">dat1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast.tissue"</span>, <span class="st">"HD"</span><span class="op">)</span>
<span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">$</span><span class="va">HD</span><span class="op">)</span><span class="op">)</span>
<span class="va">dat1</span> <span class="op">&lt;-</span> <span class="va">dat1</span><span class="op">[</span><span class="op">-</span><span class="va">idx</span>, <span class="op">]</span> <span class="co"># To remove missing data</span>

<span class="va">ps2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">PS2</span>, missing <span class="op">=</span> <span class="cn">NA</span>, columns <span class="op">=</span> <span class="fl">9</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>,
                    num.cores <span class="op">=</span> <span class="fl">6L</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">ps2</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"normal"</span>, <span class="st">"cancer"</span>, <span class="st">"metastasis"</span>, <span class="st">"normal1"</span><span class="op">)</span>
<span class="va">dat2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>normal <span class="op">=</span> <span class="va">ps2</span><span class="op">$</span><span class="va">normal</span>, normal1 <span class="op">=</span> <span class="va">ps2</span><span class="op">$</span><span class="va">normal1</span>,
        cancer <span class="op">=</span> <span class="va">ps2</span><span class="op">$</span><span class="va">cancer</span>, metastasis <span class="op">=</span> <span class="va">ps2</span><span class="op">$</span><span class="va">metastasis</span><span class="op">)</span>
<span class="va">dat2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast.tissue"</span>, <span class="st">"HD"</span><span class="op">)</span>
<span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">$</span><span class="va">HD</span><span class="op">)</span><span class="op">)</span>
<span class="va">dat2</span> <span class="op">&lt;-</span> <span class="va">dat2</span><span class="op">[</span><span class="op">-</span><span class="va">idx</span>, <span class="op">]</span> <span class="co"># To remove missing data</span>

<span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">dat1</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">HD</span>, fill <span class="op">=</span> <span class="va">Breast.tissue</span>, colour <span class="op">=</span> <span class="va">Breast.tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html">geom_histogram</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span>, bins <span class="op">=</span> <span class="fl">50</span>, position <span class="op">=</span> <span class="st">"identity"</span>,
        na.rm <span class="op">=</span> <span class="cn">TRUE</span>, size <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Counts"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">30</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,
        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span>,
                                hjust <span class="op">=</span> <span class="fl">0.5</span>, vjust <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span>,
        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>,color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,
        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Histogram for Potential methylation signal. GGamma3P"</span><span class="op">)</span>

<span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">dat1</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Breast.tissue</span>, y <span class="op">=</span> <span class="va">HD</span> , fill <span class="op">=</span> <span class="va">Breast.tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span>na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">ylim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">15</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_summary.html">stat_summary</a></span><span class="op">(</span>fun.data <span class="op">=</span> <span class="va">fun_length</span>, geom <span class="op">=</span> <span class="st">"text"</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span>,
                position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html">position_dodge</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="fl">1</span>,
                size <span class="op">=</span> <span class="fl">6</span>, fontface <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">6</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,
        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span>,
                                    hjust <span class="op">=</span> <span class="fl">0.5</span>, vjust <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span>,
        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>,color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        legend.position <span class="op">=</span> <span class="st">"none"</span> <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Boxplot for Potential methylation signal. GGamma3P"</span><span class="op">)</span>


<span class="va">p3</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">dat2</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">HD</span>, fill <span class="op">=</span> <span class="va">Breast.tissue</span>, colour <span class="op">=</span> <span class="va">Breast.tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html">geom_histogram</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span>, bins <span class="op">=</span> <span class="fl">50</span>, position <span class="op">=</span> <span class="st">"identity"</span>,
        na.rm <span class="op">=</span> <span class="cn">TRUE</span>, size <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Counts"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">30</span><span class="op">)</span> <span class="op">+</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,
        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span>,
                                    hjust <span class="op">=</span> <span class="fl">0.5</span>, vjust <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span>,
        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>,color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,
        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Histogram for Potential methylation signal. Gamma3P"</span><span class="op">)</span>

<span class="va">p4</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">dat2</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Breast.tissue</span>, y <span class="op">=</span> <span class="va">HD</span> , fill <span class="op">=</span> <span class="va">Breast.tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span>na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">ylim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">15</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_summary.html">stat_summary</a></span><span class="op">(</span>fun.data <span class="op">=</span> <span class="va">fun_length</span>, geom <span class="op">=</span> <span class="st">"text"</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span>,
                position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html">position_dodge</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="fl">1</span>,
                size <span class="op">=</span> <span class="fl">6</span>, fontface <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">6</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,
        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span>,
                                    hjust <span class="op">=</span> <span class="fl">0.5</span>, vjust <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span>,
        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>,color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,
        legend.position <span class="op">=</span> <span class="st">"none"</span> <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Boxplot for Potential methylation signal. Gamma3P"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, <span class="va">p4</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="cancer_example_files/figure-html/fig2-1.png" width="1920"></p>
<p>The graphics show that “Gamma3P” provides more information on the discrimination of treatment potential DMPs from the control.</p>
</div>
<div id="cutpoint-estimation-laws-of-statistical-physics-can-account-for-background" class="section level2">
<h2 class="hasAnchor">
<a href="#cutpoint-estimation-laws-of-statistical-physics-can-account-for-background" class="anchor"></a>Cutpoint estimation Laws of statistical physics can account for background</h2>
<p>methylation, a response to thermal fluctuations that presumably functions in DNA stability <span class="citation">(1, 2)</span>. True signal is detected based on the optimal cutpoint <span class="citation">(15)</span>, which can be estimated from the area under the curve (AUC) of a receiver operating characteristic (ROC) curve built from a logistic regression performed with the potential signals from controls and treatments. The ROC AUC is equivalent to the probability that a randomly- chosen positive instance is ranked more highly than a randomly-chosen negative instance <span class="citation">(16)</span>. In the current context, the AUC is equivalent to the probability to distinguish a randomly-chosen methylation regulatory signal induced by the treatment from a randomly-chosen signal in the control.</p>
<p>The need for the application of (what is now known as) signal detection in cancer research was pointed out by Youden in the midst of the last century <span class="citation">(17)</span>. In the next example, the simple cutpoint estimation available in Methyl-IT is based on the application of <a href="https://is.gd/t8UKvK">Youden index</a> <span class="citation">(17)</span>. Although cutpoints are estimated for a single variable, the classification performance can be evaluated for several variables and applying different model classifiers. A optimal cutpoint distinguishes disease stages from healthy individual. The performance of this classification is given in the output of function <a href="https://is.gd/RR9iWY">estimateCutPoint</a>.</p>
<p>A model classifier can be requested for further predictions and its classification performance is also provided. Below, the selected model classifier is a logistic model applied on the principal components derived from a previous principal component analysis (PCA, <em>classifier1</em> = “pca.logistic”", <em>clas.perf</em> = TRUE). Four predictor variables are available: the Hellinger divergence of methylation levels (<em>hdiv</em>), total variation distance (with Bayesian correction for the methylation levels, <em>bay.TV</em>, absolute difference of methylation levels), relative position of cytosine site in the chromosome (<em>pos</em>), and the logarithm base two of the probability to observe a Hellinger divergence value <span class="math inline">\(H\)</span> greater than the critical value <span class="math inline">\(H_{\alpha = 0.05}\)</span> (values given as probabilities in object PS, <em>wprob</em>).</p>
<p>Notice that the cutpoint can be estimated for any of the two currently available information divergences: <em>Hellinger divergence</em> (<em>div.col</em> = 9) or the <em>total</em> <em>variation distance</em> (with Bayesian correction, <em>div.col</em> = 8).</p>
<div id="cutpoint-based-on-youden-index" class="section level3">
<h3 class="hasAnchor">
<a href="#cutpoint-based-on-youden-index" class="anchor"></a><em>Cutpoint based on <em>Youden index</em></em>
</h3>
<p>Null distribution “GGamma3P”:</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutpoint1</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">PS1</span>,
                    control.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>,<span class="st">"Breast_normal1"</span><span class="op">)</span>,
                    treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast_cancer"</span>, <span class="st">"Breast_metastasis"</span><span class="op">)</span>,
                    simple <span class="op">=</span> <span class="cn">TRUE</span>,
                    classifier1 <span class="op">=</span> <span class="st">"pca.logistic"</span>,
                    column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>, 
                                wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,
                    n.pc <span class="op">=</span> <span class="fl">4</span> , center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,
                    div.col <span class="op">=</span> <span class="fl">9</span>, clas.perf <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">cutpoint1</span><span class="op">$</span><span class="va">cutpoint</span></code></pre></div>
<pre><code>## [1] 3.039526</code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutpoint1</span><span class="op">$</span><span class="va">testSetPerformance</span></code></pre></div>
<pre><code>## Confusion Matrix and Statistics
## 
##           Reference
## Prediction    CT    TT
##         CT  1124     0
##         TT     0 17732
##                                      
##                Accuracy : 1          
##                  95% CI : (0.9998, 1)
##     No Information Rate : 0.9404     
##     P-Value [Acc &gt; NIR] : &lt; 2.2e-16  
##                                      
##                   Kappa : 1          
##                                      
##  Mcnemar's Test P-Value : NA         
##                                      
##             Sensitivity : 1.0000     
##             Specificity : 1.0000     
##          Pos Pred Value : 1.0000     
##          Neg Pred Value : 1.0000     
##              Prevalence : 0.9404     
##          Detection Rate : 0.9404     
##    Detection Prevalence : 0.9404     
##       Balanced Accuracy : 1.0000     
##                                      
##        'Positive' Class : TT         
## </code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutpoint1</span><span class="op">$</span><span class="va">testSetModel.FDR</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>Null distribution “Gamma3P”:</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutpoint2</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">PS2</span>,
                    control.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>,<span class="st">"Breast_normal1"</span><span class="op">)</span>,
                    treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast_cancer"</span>, <span class="st">"Breast_metastasis"</span><span class="op">)</span>,
                    simple <span class="op">=</span> <span class="cn">TRUE</span>,
                    classifier1 <span class="op">=</span> <span class="st">"pca.logistic"</span>,
                    column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>,
                                wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,
                    n.pc <span class="op">=</span> <span class="fl">4</span> , center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,
                    div.col <span class="op">=</span> <span class="fl">9</span>, clas.perf <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">cutpoint2</span><span class="op">$</span><span class="va">cutpoint</span></code></pre></div>
<pre><code>## [1] 2.813603</code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutpoint2</span><span class="op">$</span><span class="va">testSetPerformance</span></code></pre></div>
<pre><code>## Confusion Matrix and Statistics
## 
##           Reference
## Prediction    CT    TT
##         CT  1352     0
##         TT     0 19129
##                                      
##                Accuracy : 1          
##                  95% CI : (0.9998, 1)
##     No Information Rate : 0.934      
##     P-Value [Acc &gt; NIR] : &lt; 2.2e-16  
##                                      
##                   Kappa : 1          
##                                      
##  Mcnemar's Test P-Value : NA         
##                                      
##             Sensitivity : 1.000      
##             Specificity : 1.000      
##          Pos Pred Value : 1.000      
##          Neg Pred Value : 1.000      
##              Prevalence : 0.934      
##          Detection Rate : 0.934      
##    Detection Prevalence : 0.934      
##       Balanced Accuracy : 1.000      
##                                      
##        'Positive' Class : TT         
## </code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutpoint2</span><span class="op">$</span><span class="va">testSetModel.FDR</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>The cutpoint estimated based “GGamma3P” model is lower than the value based on “Gamma3P” model. Often, this is a good detail, since it permits the identification of more sites with high probability to carry methylation signal. However, the evaluation of the classification performance has the last word on model selection. In the current case, the DMP classification performance is much better for “Gamma3P” model than for “GGamma3P.”</p>
<p>In particular, the probabilities derived from each distribution model provide fundamental information to the model classifier. Hence, the probability that a potential DMP could carry a methylation signal estimated using “Gamma3P” model is more informative (for the model classifier) than the probability estimated using “GGamma3P” model.</p>
</div>
<div id="cutpoint-estimated-using-a-model-classifier" class="section level3">
<h3 class="hasAnchor">
<a href="#cutpoint-estimated-using-a-model-classifier" class="anchor"></a><em>Cutpoint estimated using a model classifier</em>
</h3>
<p>In the current case, the DMP classification performance for the signal derived with null distribution “Gamma3P” can be improved searching for an optimal cutpoint with a model classifier and using the total variation distance ( with Bayesian correction, bay.TV).</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutpoint2</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">PS2</span>,
                    control.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>,<span class="st">"Breast_normal1"</span><span class="op">)</span>,
                    treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast_cancer"</span>, <span class="st">"Breast_metastasis"</span><span class="op">)</span>,
                    simple <span class="op">=</span> <span class="cn">FALSE</span>,
                    classifier1 <span class="op">=</span> <span class="st">"pca.logistic"</span>,
                    classifier2 <span class="op">=</span> <span class="st">"pca.qda"</span>,
                    column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>,
                                wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,
                    n.pc <span class="op">=</span> <span class="fl">4</span> , center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,
                    div.col <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>
<span class="va">cutpoint2</span><span class="op">$</span><span class="va">cutpoint</span></code></pre></div>
<pre><code>## [1] 0.5000021</code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutpoint2</span><span class="op">$</span><span class="va">testSetPerformance</span></code></pre></div>
<pre><code>## Confusion Matrix and Statistics
## 
##           Reference
## Prediction    CT    TT
##         CT  2688     3
##         TT     0 19329
##                                      
##                Accuracy : 0.9999     
##                  95% CI : (0.9996, 1)
##     No Information Rate : 0.8779     
##     P-Value [Acc &gt; NIR] : &lt;2e-16     
##                                      
##                   Kappa : 0.9994     
##                                      
##  Mcnemar's Test P-Value : 0.2482     
##                                      
##             Sensitivity : 0.9998     
##             Specificity : 1.0000     
##          Pos Pred Value : 1.0000     
##          Neg Pred Value : 0.9989     
##              Prevalence : 0.8779     
##          Detection Rate : 0.8778     
##    Detection Prevalence : 0.8778     
##       Balanced Accuracy : 0.9999     
##                                      
##        'Positive' Class : TT         
## </code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutpoint2</span><span class="op">$</span><span class="va">testSetModel.FDR</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>The good DMP classification performance was kept at lower cutpoint value.</p>
</div>
</div>
<div id="dmps" class="section level2">
<h2 class="hasAnchor">
<a href="#dmps" class="anchor"></a>DMPs</h2>
<p>Cytosine sites carrying a methylation signal are designated differentially informative methylated positions (DMPs). The probability that a DMP is not induced by the treatment is given by the probability of false alarm (P_FA, false positive, Sanchez et al. 2019). That is, the biological signal is naturally present in the control as well as in the treatment. According to the classical view in the epigenomic field, a DMP is a cytosine position carrying a significant methylation signal, which might be identified as differentially methylated cytosine site by Fisher’s exact test (or other current tests).</p>
<p>In Methyl-IT, a DMP is a DNA cytosine position with high probability to be differentially hyper-methylated or hypo-methylated in the treatment with respect to a given control (reference individual)<em><em>. Notice that Methyl- IT definition of DMP is not deterministic in an ordinary sense, but stochastic- deterministic in physico-mathematical terms. In other words, if a given cytosine position is identified as a hyper-methylated DMP in the treatment, then with high probability such a site will be found more frequently hyper- methylated than hypo-methylated in the </em></em>treatment population*_. Observe that it does not mean that, for all arbitrary subset of samples, the assumed cytosine position will be hypermethylated.</p>
<p>DMPs are selected with the function <a href="https://is.gd/%207cVEEd">selectDIMP</a></p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">DMPs</span> <span class="op">=</span> <span class="fu"><a href="../reference/selectDIMP.html">selectDIMP</a></span><span class="op">(</span><span class="va">PS2</span>, div.col <span class="op">=</span> <span class="fl">8</span>, cutpoint <span class="op">=</span> <span class="va">cutpoint2</span><span class="op">$</span><span class="va">cutpoint</span><span class="op">)</span></code></pre></div>
<div id="the-descriptive-statistic-of-the-read-counts-at-dmps" class="section level3">
<h3 class="hasAnchor">
<a href="#the-descriptive-statistic-of-the-read-counts-at-dmps" class="anchor"></a>The descriptive statistic of the read counts at DMPs</h3>
<p>After the signal detection step the read counts at DMPs is notable greater than at the general methylated cytosine background:</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Maximum coverage at each cytosine site</span>
<span class="va">methc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">DMPs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
                        <span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">c1</span>, <span class="va">x</span><span class="op">$</span><span class="va">c2</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">max</span><span class="op">)</span>
                        <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">r</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># Quantiles</span>
<span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">methc</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">q10</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.1</span><span class="op">)</span>
    <span class="va">q60</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.6</span><span class="op">)</span>
    <span class="va">q9999</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.9999</span><span class="op">)</span>
    <span class="va">idx1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="va">q60</span><span class="op">)</span>
    <span class="va">idx2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span>
    <span class="va">q95</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.95</span><span class="op">)</span>

    <span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span><span class="op">(</span><span class="va">idx1</span>, <span class="va">idx2</span><span class="op">)</span>
    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span>, <span class="va">q10</span>, <span class="va">q60</span>,
            <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.95</span>, <span class="fl">0.99</span>, <span class="fl">0.999</span>, <span class="fl">0.9999</span><span class="op">)</span><span class="op">)</span>,
            num.siteGreater_8 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="fl">8</span><span class="op">)</span>,
            q60_to_500 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="va">q60</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">(</span><span class="va">x</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span>,
            num.siteGreater_500 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<pre><code>##                   Min. 1st Qu. Median Mean 3rd Qu. Max.  10%  60%  95%   99%
## Breast_normal       14      19     20   22      22   41 18.0 20.0 31.9 39.18
## Breast_cancer        7      18     39   39      47   93 13.0 44.2 78.5 90.14
## Breast_metastasis   13      34     55   51      66   87 26.6 57.4 75.6 85.52
## Breast_normal1      16      18     19   19      20   21 16.6 19.4 20.8 20.96
##                    99.9%  99.99% num.siteGreater_8 q60_to_500
## Breast_normal     40.818 40.9818                14          8
## Breast_cancer     92.714 92.9714                22          9
## Breast_metastasis 86.852 86.9852                75         30
## Breast_normal1    20.996 20.9996                 3          1
##                   num.siteGreater_500
## Breast_normal                       0
## Breast_cancer                       0
## Breast_metastasis                   0
## Breast_normal1                      0</code></pre>
<p>The minimum amount of methylated read counts is observed “Breast_metastasis” (15 reads) and for any the samples the tenth percent of DMPs is 26 or more reads.</p>
</div>
<div id="venn-diagram-of-dmps" class="section level3">
<h3 class="hasAnchor">
<a href="#venn-diagram-of-dmps" class="anchor"></a><em>Venn Diagram of DMPs</em>
</h3>
<p>The Venn diagram of DMPs reveals that the number cytosine site carrying methylation signal with a divergence level comparable to that observed in breast tissues with cancer and metastasis is relatively small (2797 DMPs). The number of DMPs decreased in the breast tissue with metastasis, but, as shown in the last boxplot, the intensity of the signal increased.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">n12</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/setops-methods.html">intersect</a></span><span class="op">(</span><span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_normal</span>, 
                                    <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_cancer</span><span class="op">)</span><span class="op">)</span>
<span class="va">n13</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/setops-methods.html">intersect</a></span><span class="op">(</span><span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_normal</span>, 
                                    <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_metastasis</span><span class="op">)</span><span class="op">)</span>
<span class="va">n23</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/setops-methods.html">intersect</a></span><span class="op">(</span><span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_cancer</span>,
                                    <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_metastasis</span><span class="op">)</span><span class="op">)</span>
<span class="va">n123</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/funprog.html">Reduce</a></span><span class="op">(</span><span class="fu">GenomicRanges</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/GenomicRanges/man/setops-methods.html">intersect</a></span>,
                    <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_normal</span>, <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_cancer</span>,
                        <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_metastasis</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/grid/grid.newpage.html">grid.newpage</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">v</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/VennDiagram/man/draw.triple.venn.html">draw.triple.venn</a></span><span class="op">(</span>area1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_normal</span><span class="op">)</span>,
                    area2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_cancer</span><span class="op">)</span>,
                    area3 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_metastasis</span><span class="op">)</span>,
                    n12 <span class="op">=</span> <span class="va">n12</span>, n23 <span class="op">=</span> <span class="va">n23</span>, n13 <span class="op">=</span> <span class="va">n13</span>, n123 <span class="op">=</span> <span class="va">n123</span>,
                    category <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>, <span class="st">"Breast_cancer"</span>,
                                <span class="st">"Breast_metastasis"</span><span class="op">)</span>,
                    lty <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"blank"</span>, <span class="fl">3</span><span class="op">)</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"yellow"</span>, 
                                                    <span class="st">"magenta"</span><span class="op">)</span>,
                    alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.2</span>, <span class="fl">0.3</span><span class="op">)</span>,
                    cat.pos <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">80</span>, <span class="fl">90</span>, <span class="fl">0</span><span class="op">)</span>,
                    cat.col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"darkorange"</span>, <span class="st">"red"</span><span class="op">)</span>,
                    cat.dist <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="op">-</span><span class="fl">0.1</span>, <span class="op">-</span><span class="fl">0.08</span>, <span class="op">-</span><span class="fl">0.26</span><span class="op">)</span>,
                    cex <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">1.7</span>, <span class="fl">7</span><span class="op">)</span>,
                    cat.cex <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="fl">1.5</span>, <span class="fl">1.5</span>, <span class="fl">1.5</span><span class="op">)</span>,
                    label.col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"blue"</span>, <span class="st">"darkorange"</span>, <span class="st">"darkorange"</span>,
                                <span class="st">"red"</span>, <span class="st">"white"</span>, <span class="st">"red"</span>, <span class="st">"red"</span><span class="op">)</span>,
                    scaled <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grid/grid.draw.html">grid.draw</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span></code></pre></div>
<p><img src="cancer_example_files/figure-html/fig4-1.png" width="672"></p>
<p>Notice that natural methylation regulatory signals (not induced by the treatment) are present in both groups, control and treatment. The signal detection step permits us to discriminate the “ordinary” signals observed in the control from those induced by the treatment (a disease in the current case). In addition, this diagram reflects a classification of DMPs only based on the cytosine positions. That is, this Venn diagram cannot tell us whether DMPs at the same position can be distinguishable or not. For example, DMPs at the same positions in control and treatment can happen with different probabilities estimated from their corresponding fitted Weibull distributions (see below).</p>
</div>
<div id="monte-carlo-evaluation-of-the-model-classification" class="section level3">
<h3 class="hasAnchor">
<a href="#monte-carlo-evaluation-of-the-model-classification" class="anchor"></a>Monte Carlo Evaluation of the Model Classification</h3>
<p>Performance The regulatory methylation signal is an output from a natural process that continuously takes place across the ontogenetic development of the organism. Therefore, we expect to see methylation signal in natural, ordinary conditions. Function ‘evaluateDIMPclass’ can be used to perform a classification of DMPs into two classes: DMPS from control and DMPs from treatment samples, as well as an evaluation of the classification performance (for more details see <a href="https://is.gd/nzy9j3">evaluateDIMPclass</a>).</p>
<p>As a matter of fact this function is called by function <a href="https://is.gd/RR9iWY">estimateCutPoint</a>. Additional feature not used before is the possibility to perform Monte Carlo evaluation of the model classifier performance. In above split of the sample into two subsets, training and test datasets, we would be just lucky getting the right proportion that yields the best classification performance. What about is the random split is repeat 300 times?</p>
<p>The performance of the “pca.qda” classifier is:</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">performance</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluateDIMPclass.html">evaluateDIMPclass</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">DMPs</span>,
                    control.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>, <span class="st">"Breast_normal1"</span><span class="op">)</span>,
                    treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast_cancer"</span>, <span class="st">"Breast_metastasis"</span><span class="op">)</span>,
                    column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>, 
                                wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,
                    classifier <span class="op">=</span> <span class="st">"pca.qda"</span>, prop <span class="op">=</span> <span class="fl">0.6</span>,
                    n.pc <span class="op">=</span> <span class="fl">4</span> , center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,
                    num.boot <span class="op">=</span> <span class="fl">300</span>, output <span class="op">=</span> <span class="st">"mc.val"</span>,
                    num.cores <span class="op">=</span> <span class="fl">6L</span>, tasks <span class="op">=</span> <span class="fl">2L</span><span class="op">)</span>
<span class="va">performance</span></code></pre></div>
<pre><code>##     Accuracy          Kappa        AccuracyLower    AccuracyUpper   
##  Min.   :0.8298   Min.   :0.2210   Min.   :0.6919   Min.   :0.9235  
##  1st Qu.:0.9362   1st Qu.:0.7860   1st Qu.:0.8246   1st Qu.:0.9866  
##  Median :0.9574   Median :0.8498   Median :0.8546   Median :0.9948  
##  Mean   :0.9586   Mean   :0.8464   Mean   :0.8590   Mean   :0.9918  
##  3rd Qu.:0.9787   3rd Qu.:0.9207   3rd Qu.:0.8871   3rd Qu.:0.9995  
##  Max.   :1.0000   Max.   :1.0000   Max.   :0.9245   Max.   :1.0000  
##                                                                     
##   AccuracyNull    AccuracyPValue      McnemarPValue      Sensitivity    
##  Min.   :0.8511   Min.   :0.0005108   Min.   :0.01333   Min.   :0.8000  
##  1st Qu.:0.8511   1st Qu.:0.0047118   1st Qu.:0.24821   1st Qu.:0.9500  
##  Median :0.8511   Median :0.0216210   Median :0.47950   Median :0.9750  
##  Mean   :0.8511   Mean   :0.0535310   Mean   :0.60025   Mean   :0.9643  
##  3rd Qu.:0.8511   3rd Qu.:0.0660075   3rd Qu.:1.00000   3rd Qu.:1.0000  
##  Max.   :0.8511   Max.   :0.7403189   Max.   :1.00000   Max.   :1.0000  
##                                       NA's   :48                        
##   Specificity     Pos Pred Value   Neg Pred Value     Precision     
##  Min.   :0.1429   Min.   :0.8696   Min.   :0.4667   Min.   :0.8696  
##  1st Qu.:0.8571   1st Qu.:0.9756   1st Qu.:0.7778   1st Qu.:0.9756  
##  Median :1.0000   Median :1.0000   Median :0.8750   Median :1.0000  
##  Mean   :0.9257   Mean   :0.9876   Mean   :0.8517   Mean   :0.9876  
##  3rd Qu.:1.0000   3rd Qu.:1.0000   3rd Qu.:1.0000   3rd Qu.:1.0000  
##  Max.   :1.0000   Max.   :1.0000   Max.   :1.0000   Max.   :1.0000  
##                                                                     
##      Recall             F1           Prevalence     Detection Rate  
##  Min.   :0.8000   Min.   :0.8889   Min.   :0.8511   Min.   :0.6809  
##  1st Qu.:0.9500   1st Qu.:0.9630   1st Qu.:0.8511   1st Qu.:0.8085  
##  Median :0.9750   Median :0.9750   Median :0.8511   Median :0.8298  
##  Mean   :0.9643   Mean   :0.9751   Mean   :0.8511   Mean   :0.8207  
##  3rd Qu.:1.0000   3rd Qu.:0.9873   3rd Qu.:0.8511   3rd Qu.:0.8511  
##  Max.   :1.0000   Max.   :1.0000   Max.   :0.8511   Max.   :0.8511  
##                                                                     
##  Detection Prevalence Balanced Accuracy
##  Min.   :0.6809       Min.   :0.5714   
##  1st Qu.:0.8085       1st Qu.:0.9286   
##  Median :0.8298       Median :0.9750   
##  Mean   :0.8318       Mean   :0.9450   
##  3rd Qu.:0.8511       3rd Qu.:0.9875   
##  Max.   :0.9787       Max.   :1.0000   
## </code></pre>
</div>
</div>
<div id="differentially-methylated-genomic-regions-dmrs" class="section level2">
<h2 class="hasAnchor">
<a href="#differentially-methylated-genomic-regions-dmrs" class="anchor"></a>Differentially methylated genomic regions (DMRs)</h2>
<p>In Methyl-IT, the concept of DMR is generalized and it is not limited to any particular genomic region found with specific clustering algorithm. It can be applied to any naturally or algorithmically defined genomic region. For example, an exon region identified statistically to be differentially methylated by using generalized linear regression model (GLM) is a DMR. Differentially methylated genes (DMGs) are estimated from group comparisons for the number of DMPs on gene-body regions between control and treatment. This permits the extension of the concept of DMR by considering a differentially methylated gene as a particular case of DMRs.</p>
<p>Our degree of confidence in whether DMP counts in both groups of samples, control and treatment, represent true biological signal was determined in the signal detection step. To estimate DMRs, we followed similar steps to those proposed in Bioconductor R package DESeq2 <span class="citation">(18)</span>, but our GLM test looks for statistical difference between the groups based on gene-body DMP counts overlapping a given genomic region rather than read counts. The regression analysis of the generalized linear model (GLM) with logarithmic link was applied to test the difference between group counts.</p>
<p>The fitting algorithmic approaches provided by ‘glm’ and ‘glm.nb’ functions from the R packages stat and MASS, respectively, were used for Poisson (PR), Quasi-Poisson (QPR) and Negative Binomial (NBR) linear regression analyses, respectively.</p>
<div id="differentially-methylated-genes-dmgs" class="section level3">
<h3 class="hasAnchor">
<a href="#differentially-methylated-genes-dmgs" class="anchor"></a><em>Differentially methylated genes (DMGs)</em>
</h3>
<p>We shall call DMGs those DIMRs restricted to gene-body regions. DMGs are detected using function <a href="https://is.gd/kltCXl">countTest2</a>. Gene annotation is taken from <a href="https://www.ensembl.org/">Ensembl</a></p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># To load human gene annotation</span>
<span class="co"># AG = import(con = paste0("ftp://ftp.ensembl.org/pub/release-91/gff3/",</span>
<span class="co"># "homo_sapiens/Homo_sapiens.GRCh38.91.gff3.gz"))</span>
<span class="co"># genes = AG[ AG$type == "gene", c( "gene_id", "biotype", "Name" ) ]</span>
<span class="co"># genes = genes[ genes$biotype == "protein_coding", "gene_id" ]</span>
<span class="co"># seqlevels(genes, "coarse") &lt;- "13" # To keep a consistent chromosome </span>
<span class="co"># annotation</span>
<span class="co"># seqlevels(genes) &lt;- "chr13"</span></code></pre></div>
<p>Here, for the sake of brevity, we just load the “RData” compressed file containing only the gene annotation:</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">con</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/connections.html">url</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"https://git.psu.edu/genomath/MethylIT_examples/raw/master/"</span>,
                <span class="st">"Homo_sapiens_GRCh38.91_genes.RData"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="va">con</span><span class="op">)</span>
<span class="va">genes</span> <span class="op">=</span> <span class="va">genes</span><span class="op">[</span> <span class="va">genes</span><span class="op">$</span><span class="va">biotype</span> <span class="op">==</span> <span class="st">"protein_coding"</span>, <span class="op">]</span>
<span class="fu">seqlevels</span><span class="op">(</span><span class="va">genes</span>, <span class="st">"coarse"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"13"</span> <span class="co"># To keep a consistent chromosome annotation</span>
<span class="fu">seqlevels</span><span class="op">(</span><span class="va">genes</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"chr13"</span>
<span class="va">genes</span></code></pre></div>
<pre><code>## GRanges object with 323 ranges and 3 metadata columns:
##         seqnames              ranges strand |         gene_id        biotype
##            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;    &lt;character&gt;
##     [1]    chr13   19173770-19181852      - | ENSG00000198033 protein_coding
##     [2]    chr13   19422877-19536762      - | ENSG00000132958 protein_coding
##     [3]    chr13   19633681-19673459      + | ENSG00000196199 protein_coding
##     [4]    chr13   19674752-19783019      - | ENSG00000121390 protein_coding
##     [5]    chr13   19823482-19863636      - | ENSG00000132950 protein_coding
##     ...      ...                 ...    ... .             ...            ...
##   [319]    chr13 113977783-114132611      - | ENSG00000185989 protein_coding
##   [320]    chr13 114179238-114223084      + | ENSG00000283361 protein_coding
##   [321]    chr13 114234887-114272723      + | ENSG00000130177 protein_coding
##   [322]    chr13 114281584-114305817      + | ENSG00000169062 protein_coding
##   [323]    chr13 114314513-114327328      + | ENSG00000198824 protein_coding
##                Name
##         &lt;character&gt;
##     [1]      TUBA3C
##     [2]       TPTE2
##     [3]    MPHOSPH8
##     [4]       PSPC1
##     [5]       ZMYM5
##     ...         ...
##   [319]       RASA3
##   [320]  AL160396.2
##   [321]       CDC16
##   [322]       UPF3A
##   [323]      CHAMP1
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<p>Function <a href="https://is.gd/WQzX5I">getDIMPatGenes</a> is used to count the number of DMPs at gene-body. Nevertheless, it can be used for any arbitrary specified genomic region as well. The operation of this function is based on the ‘<em>findOverlaps</em>’ function from the ‘<em>GenomicRanges</em>’ Bioconductor R package. The ‘<em>findOverlaps</em>’ function has several critical parameters like, for example, ‘<em>maxgap</em>,’ ‘<em>minoverlap</em>,’ and ‘<em>ignore.strand</em>.’ In our function <a href="https://is.gd/WQzX5I">getDIMPatGenes</a>, except for setting ignore.strand = TRUE and <em>type = “within”</em>, we preserve the rest of default ‘findOverlaps’ parameters.</p>
<p>In this case, these are important parameter settings because the local mechanical effect of methylation changes on a DNA region where a gene is located is independent of the strand where the gene is encoded. That is, methylation changes located in any of the two DNA strands inside the gene-body region will affect the flexibility of the DNA molecule <span class="citation">(7, 19)</span>.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">DMPsBN</span> <span class="op">=</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_normal</span>, GENES <span class="op">=</span> <span class="va">genes</span><span class="op">)</span>
<span class="va">DMPsBN1</span> <span class="op">=</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_normal1</span>, GENES <span class="op">=</span> <span class="va">genes</span><span class="op">)</span>
<span class="va">DMPsBC</span> <span class="op">=</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_cancer</span>, GENES <span class="op">=</span> <span class="va">genes</span><span class="op">)</span>
<span class="va">DMPsBM</span> <span class="op">=</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_metastasis</span>, GENES <span class="op">=</span>
        <span class="va">genes</span><span class="op">)</span></code></pre></div>
<p>The number of DMPs on the strand where a gene is encoded is obtained by setting ignore.strand = FALSE. However, for the current example results will be the same since the datasets downloaded from GEO do not have strand information. Next, the above GRanges objects carrying the DMP counts from each sample are grouped into a single GRanges object. Since we have only one control, to perform group comparison and to move forward with this example, we duplicated ‘Breast_normal’ sample. Obviously, the confidence on the results increases with the number of sample replications per group (in this case, it is only an illustrative example on how to perform the analysis, since a fair comparison requires for more than one replicate in the control group).</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Genes.DMPs</span> <span class="op">=</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">DMPsBN</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>, <span class="va">DMPsBN1</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>,
                                <span class="va">DMPsBC</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>, <span class="va">DMPsBM</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span>,
                            type <span class="op">=</span> <span class="st">"equal"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>,
                            ignore.strand <span class="op">=</span> <span class="cn">TRUE</span> <span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span> <span class="fu">mcols</span><span class="op">(</span><span class="va">Genes.DMPs</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>, <span class="st">"Breast_normal1"</span>,
                                    <span class="st">"Breast_cancer"</span>, <span class="st">"Breast_metastasis"</span><span class="op">)</span></code></pre></div>
<p>Next, the set of mapped genes are annotated</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">GeneID</span> <span class="op">=</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">genes</span>, <span class="va">Genes.DMPs</span>, type <span class="op">=</span> <span class="st">"equal"</span>,
                            ignore.strand <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span> <span class="va">Genes.DMPs</span> <span class="op">)</span> <span class="op">&lt;-</span> <span class="va">GeneID</span><span class="op">$</span><span class="va">gene_id</span>
<span class="va">Genes.DMPs</span></code></pre></div>
<pre><code>## GRanges object with 12 ranges and 4 metadata columns:
##                   seqnames              ranges strand | Breast_normal
##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;numeric&gt;
##   ENSG00000132953    chr13   20777329-20903048      * |             0
##   ENSG00000132964    chr13   26254104-26405238      * |             1
##   ENSG00000132938    chr13   28820348-29505947      * |             0
##   ENSG00000172915    chr13   34942287-35673022      * |             1
##   ENSG00000120699    chr13   36998816-37009613      * |             0
##               ...      ...                 ...    ... .           ...
##   ENSG00000175198    chr13 100089015-100530437      * |             0
##   ENSG00000125247    chr13 100603927-100675093      * |             0
##   ENSG00000134871    chr13 110305812-110513027      * |             0
##   ENSG00000068650    chr13 112690329-112887168      * |             0
##   ENSG00000185989    chr13 113977783-114132611      * |             0
##                   Breast_normal1 Breast_cancer Breast_metastasis
##                        &lt;numeric&gt;     &lt;numeric&gt;         &lt;numeric&gt;
##   ENSG00000132953              0             0                 1
##   ENSG00000132964              0             0                 1
##   ENSG00000132938              0             0                 7
##   ENSG00000172915              0             0                 0
##   ENSG00000120699              0             0                 2
##               ...            ...           ...               ...
##   ENSG00000175198              0             0                 3
##   ENSG00000125247              0             2                16
##   ENSG00000134871              0             0                 1
##   ENSG00000068650              0             1                 3
##   ENSG00000185989              0             1                 0
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<p>Now, we build a <a href="https://is.gd/VZaazz">glmDataSet</a> object.</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## An experiment design is set.</span>
<span class="va">colData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"BN"</span>,<span class="st">"BN"</span>,<span class="st">"BC"</span>,<span class="st">"BC"</span><span class="op">)</span>,
                                        levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"BN"</span>, <span class="st">"BC"</span><span class="op">)</span><span class="op">)</span>,
                        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>,<span class="st">"Breast_normal1"</span>,
                            <span class="st">"Breast_cancer"</span>,<span class="st">"Breast_metastasis"</span><span class="op">)</span>,
                        row.names <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="co">## A RangedGlmDataSet is created</span>
<span class="va">ds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">Genes.DMPs</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></code></pre></div>
<p>DMG analysis is performed with the function <a href="https://%20is.gd/kltCXl">countTest2</a></p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">DMGs</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds</span>, num.cores <span class="op">=</span> <span class="fl">4L</span>, minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,
                countFilter <span class="op">=</span> <span class="cn">TRUE</span>, maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,
                Minlog2FC <span class="op">=</span> <span class="fl">1</span>, pvalCutOff <span class="op">=</span> <span class="fl">0.05</span>,
                MVrate <span class="op">=</span> <span class="fl">.95</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in data.frame(group = groups, count = x): row names were found from a
## short variable and have been discarded</code></pre>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">DMGs</span></code></pre></div>
<pre><code>## GRanges object with 0 ranges and 12 metadata columns:
##    seqnames    ranges strand | Breast_normal Breast_normal1 Breast_cancer
##       &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;numeric&gt;      &lt;numeric&gt;     &lt;numeric&gt;
##    Breast_metastasis    log2FC scaled.deviance    pvalue     model  adj.pval
##            &lt;numeric&gt; &lt;logical&gt;       &lt;logical&gt; &lt;logical&gt; &lt;logical&gt; &lt;numeric&gt;
##    CT.SignalDensity TT.SignalDensity SignalDensityVariation
##           &lt;numeric&gt;        &lt;numeric&gt;              &lt;numeric&gt;
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<p>Here, we adds the gene alias:</p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hits</span> <span class="op">=</span> <span class="fu">findOverlaps</span><span class="op">(</span><span class="va">DMGs</span>, <span class="va">genes</span>, type <span class="op">=</span> <span class="st">"equal"</span><span class="op">)</span>
<span class="va">DMGs</span> <span class="op">=</span> <span class="va">DMGs</span><span class="op">[</span><span class="fu">queryHits</span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span>
<span class="va">DMGs</span><span class="op">$</span><span class="va">alias</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">$</span><span class="va">Name</span><span class="op">[</span><span class="fu">subjectHits</span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<p><em><strong>BRCA2, a breast cancer associated risk gene, is found between the DMG</strong>s</em></p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># DMGs</span>
<span class="va">DMGs</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span> <span class="st">"ENSG00000139618"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">DMGs</span><span class="op">)</span> <span class="op">)</span> <span class="op">]</span></code></pre></div>
<pre><code>## GRanges object with 0 ranges and 13 metadata columns:
##    seqnames    ranges strand | Breast_normal Breast_normal1 Breast_cancer
##       &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;numeric&gt;      &lt;numeric&gt;     &lt;numeric&gt;
##    Breast_metastasis    log2FC scaled.deviance    pvalue     model  adj.pval
##            &lt;numeric&gt; &lt;logical&gt;       &lt;logical&gt; &lt;logical&gt; &lt;logical&gt; &lt;numeric&gt;
##    CT.SignalDensity TT.SignalDensity SignalDensityVariation       alias
##           &lt;numeric&gt;        &lt;numeric&gt;              &lt;numeric&gt; &lt;character&gt;
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
<div id="concluding-remarks" class="section level2">
<h2 class="hasAnchor">
<a href="#concluding-remarks" class="anchor"></a>Concluding remarks</h2>
<p>Results fundamentally depend on the use of the proper reference samples. If there is not enough control samples to create an independent reference virtual individual, then the centroid from the control group can be used instead, which can be built with function <a href="https://is.gd/qR79kW">poolFromGRlist</a>.</p>
<p>The <em>reference</em> is used to put individuals from control and patients in the same <em>system of reference</em>. The information divergences of methylation levels with respect to the reference individual are estimated for each cytosine site for each individual, from control and treatment. As a result, the information on the natural spontaneous variation in the control and treatment populations is carried in the probability distributions of the information divergences, estimated for each individual. <em>It does not matter how statistically</em> <em>significant a DMP would be in the group of patients</em>.</p>
<p><em>What really matter is how big is the probability to observe the same methylation event in the control group in respect to the patient group</em>. Decisions are ultimately made on the basis of such classification probabilities at single cytosine positions. The signal detection based approach implemented in Methyl-IT is addressed to confront the mentioned issue and, in consequence, the level of resolution of Methyl-IT approach reach the single cytosine site.</p>
</div>
<div id="acknowledgments" class="section level2">
<h2 class="hasAnchor">
<a href="#acknowledgments" class="anchor"></a>Acknowledgments</h2>
<p>We thank Professor David J Miller for valuable conversations and suggestions on our mathematical modeling.</p>
</div>
<div id="funding-the-work-was-supported-by-funding-from-nsf-sbir" class="section level2">
<h2 class="hasAnchor">
<a href="#funding-the-work-was-supported-by-funding-from-nsf-sbir" class="anchor"></a>Funding The work was supported by funding from NSF-SBIR</h2>
<p>(2015-33610-23428-UNL) and the Bill and Melinda Gates Foundation (OPP1088661).</p>
</div>
<div id="references-" class="section level2">
<h2 class="hasAnchor">
<a href="#references-" class="anchor"></a>References.</h2>
<div id="refs" class="references csl-bib-body">
<div id="ref-Sanchez2016" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">R. Sanchez, S. A. Mackenzie, <span class="nocase">Information Thermodynamics of Cytosine DNA Methylation</span>. <em>PLOS ONE</em> <strong>11</strong>, e0150427 (2016).</div>
</div>
<div id="ref-Sanchez2019" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">R. Sanchez, X. Yang, T. Maher, S. Mackenzie, <span class="nocase">Discrimination of DNA Methylation Signal from Background Variation for Clinical Diagnostics</span>. <em>Int. J. Mol. Sci.</em> <strong>20</strong>, 5343 (2019).</div>
</div>
<div id="ref-Greiner2000" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">M. Greiner, D. Pfeiffer, R. D. Smith, <span class="nocase">Principles and practical application of the receiver-operating characteristic analysis for diagnostic tests</span>. <em>Preventive Veterinary Medicine</em> <strong>45</strong>, 23–41 (2000).</div>
</div>
<div id="ref-Carter2016" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">J. V. Carter, J. Pan, S. N. Rai, S. Galandiuk, <span class="nocase">ROC-ing along: Evaluation and interpretation of receiver operating characteristic curves</span>. <em>Surgery</em> <strong>159</strong>, 1638–1645 (2016).</div>
</div>
<div id="ref-Harpaz2013" class="csl-entry">
<div class="csl-left-margin">5. </div>
<div class="csl-right-inline">R. Harpaz, <em>et al.</em>, <span class="nocase">Performance of Pharmacovigilance Signal Detection Algorithms for the FDA Adverse Event Reporting System</span>. <em>Clin Pharmacol ther</em> <strong>93</strong>, 1–19 (2013).</div>
</div>
<div id="ref-Kruspe2017" class="csl-entry">
<div class="csl-left-margin">6. </div>
<div class="csl-right-inline">S. Kruspe, <em>et al.</em>, <span class="nocase">Rapid and Sensitive Detection of Breast Cancer Cells in Patient Blood with Nuclease-Activated Probe Technology</span>. <em>Molecular Therapy - Nucleic Acids</em> <strong>8</strong>, 542–557 (2017).</div>
</div>
<div id="ref-Severin2011" class="csl-entry">
<div class="csl-left-margin">7. </div>
<div class="csl-right-inline">P. M. D. Severin, X. Zou, H. E. Gaub, K. Schulten, <span class="nocase">Cytosine methylation alters DNA mechanical properties</span>. <em>Nucleic acids research</em> <strong>39</strong>, 8740–51 (2011).</div>
</div>
<div id="ref-Ngo2016" class="csl-entry">
<div class="csl-left-margin">8. </div>
<div class="csl-right-inline">T. T. M. Ngo, <em>et al.</em>, <span class="nocase">Effects of cytosine modifications on DNA flexibility and nucleosome mechanical stability</span>. <em>Nature Communications</em> <strong>7</strong>, 10813 (2016).</div>
</div>
<div id="ref-Basu2010" class="csl-entry">
<div class="csl-left-margin">9. </div>
<div class="csl-right-inline">A. Basu, A. Mandal, L. Pardo, <span class="nocase">Hypothesis testing for two discrete populations based on the Hellinger distance</span>. <em>Statistics <span>&amp;</span> Probability Letters</em> <strong>80</strong>, 206–214 (2010).</div>
</div>
<div id="ref-Hebestreit2013" class="csl-entry">
<div class="csl-left-margin">10. </div>
<div class="csl-right-inline">K. Hebestreit, M. Dugas, H.-U. Klein, <span class="nocase">Detection of significantly differentially methylated regions in targeted bisulfite sequencing data.</span> <em>Bioinformatics (Oxford, England)</em> <strong>29</strong>, 1647–53 (2013).</div>
</div>
<div id="ref-Robinson2014" class="csl-entry">
<div class="csl-left-margin">11. </div>
<div class="csl-right-inline">M. D. Robinson, <em>et al.</em>, <span class="nocase">Statistical methods for detecting differentially methylated loci and regions.</span> <em>Frontiers in genetics</em> <strong>5</strong>, 324 (2014).</div>
</div>
<div id="ref-Dolzhenko2014" class="csl-entry">
<div class="csl-left-margin">12. </div>
<div class="csl-right-inline">E. Dolzhenko, A. D. Smith, <span class="nocase">Using beta-binomial regression for high-precision differential methylation analysis in multifactor whole-genome bisulfite sequencing experiments</span>. <em>BMC bioinformatics</em> <strong>15</strong>, 215 (2014).</div>
</div>
<div id="ref-Baldi2001" class="csl-entry">
<div class="csl-left-margin">13. </div>
<div class="csl-right-inline">P. Baldi, S. Brunak, <em><span class="nocase">Bioinformatics: the machine learning approach</span></em>, Second (MIT Press, 2001).</div>
</div>
<div id="ref-Stevens2009" class="csl-entry">
<div class="csl-left-margin">14. </div>
<div class="csl-right-inline">J. P. Stevens, <em><span class="nocase">Applied Multivariate Statistics for the Social Sciences</span></em>, Fifth Edit (Routledge Academic, 2009).</div>
</div>
<div id="ref-lopez2014" class="csl-entry">
<div class="csl-left-margin">15. </div>
<div class="csl-right-inline">M. López-Ratón, M. X. Rodríguez-Álvarez, C. Cadarso-Suárez, F. Gude-Sampedro, Others, <span class="nocase">OptimalCutpoints: an R package for selecting optimal cutpoints in diagnostic tests</span>. <em>Journal of statistical software</em> <strong>61</strong>, 1–36 (2014).</div>
</div>
<div id="ref-Fawcett2005" class="csl-entry">
<div class="csl-left-margin">16. </div>
<div class="csl-right-inline">T. Fawcett, <span class="nocase">An introduction to ROC analysis</span> (2005) https:/doi.org/<a href="https://doi.org/10.1016/j.patrec.2005.10.010">10.1016/j.patrec.2005.10.010</a>.</div>
</div>
<div id="ref-Youden1950" class="csl-entry">
<div class="csl-left-margin">17. </div>
<div class="csl-right-inline">W. J. Youden, <span class="nocase">Index for rating diagnostic tests</span>. <em>Cancer</em> <strong>3</strong>, 32–35 (1950).</div>
</div>
<div id="ref-Love2014" class="csl-entry">
<div class="csl-left-margin">18. </div>
<div class="csl-right-inline">M. I. Love, W. Huber, S. Anders, <span class="nocase">Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</span>. <em>Genome Biology</em> <strong>15</strong>, 1–34 (2014).</div>
</div>
<div id="ref-Choy2010" class="csl-entry">
<div class="csl-left-margin">19. </div>
<div class="csl-right-inline">J. S. Choy, <em>et al.</em>, <span class="nocase">DNA methylation increases nucleosome compaction and rigidity</span>. <em>Journal of the American Chemical Society</em> <strong>132</strong>, 1782–1783 (2010).</div>
</div>
</div>
</div>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo()</a></code> on the system on which this document was compiled running pandoc 2.11.4:</p>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.3.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
##  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] VennDiagram_1.6.20   futile.logger_1.4.3  gridExtra_2.3       
##  [4] reshape2_1.4.4       ggplot2_3.3.5        MethylIT_0.3.2.2    
##  [7] rtracklayer_1.52.0   GenomicRanges_1.44.0 GenomeInfoDb_1.28.1 
## [10] IRanges_2.26.0       S4Vectors_0.30.0     BiocGenerics_0.38.0 
## [13] knitr_1.33          
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_2.0-2            rjson_0.2.20               
##   [3] ellipsis_0.3.2              class_7.3-19               
##   [5] rprojroot_2.0.2             XVector_0.32.0             
##   [7] fs_1.5.0                    proxy_0.4-26               
##   [9] farver_2.1.0                bit64_4.0.5                
##  [11] AnnotationDbi_1.54.1        prodlim_2019.11.13         
##  [13] fansi_0.5.0                 lubridate_1.7.10           
##  [15] codetools_0.2-18            splines_4.1.0              
##  [17] cachem_1.0.5                jsonlite_1.7.2             
##  [19] pROC_1.17.0.1               Rsamtools_2.8.0            
##  [21] caret_6.0-88                annotate_1.70.0            
##  [23] png_0.1-7                   compiler_4.1.0             
##  [25] httr_1.4.2                  assertthat_0.2.1           
##  [27] Matrix_1.3-4                fastmap_1.1.0              
##  [29] formatR_1.11                htmltools_0.5.1.1          
##  [31] tools_4.1.0                 gtable_0.3.0               
##  [33] glue_1.4.2                  GenomeInfoDbData_1.2.6     
##  [35] dplyr_1.0.7                 Rcpp_1.0.7                 
##  [37] Biobase_2.52.0              jquerylib_0.1.4            
##  [39] pkgdown_1.6.1               vctrs_0.3.8                
##  [41] Biostrings_2.60.1           nlme_3.1-152               
##  [43] iterators_1.0.13            timeDate_3043.102          
##  [45] gower_0.2.2                 xfun_0.24                  
##  [47] stringr_1.4.0               lifecycle_1.0.0            
##  [49] restfulr_0.0.13             XML_3.99-0.6               
##  [51] zlibbioc_1.38.0             MASS_7.3-54                
##  [53] scales_1.1.1                ipred_0.9-11               
##  [55] ragg_1.1.3                  MatrixGenerics_1.4.0       
##  [57] SummarizedExperiment_1.22.0 lambda.r_1.2.4             
##  [59] yaml_2.2.1                  memoise_2.0.0              
##  [61] sass_0.4.0                  rpart_4.1-15               
##  [63] stringi_1.6.2               RSQLite_2.2.7              
##  [65] highr_0.9                   genefilter_1.74.0          
##  [67] BiocIO_1.2.0                desc_1.3.0                 
##  [69] foreach_1.5.1               e1071_1.7-7                
##  [71] BiocParallel_1.26.1         lava_1.6.9                 
##  [73] rlang_0.4.11                pkgconfig_2.0.3            
##  [75] systemfonts_1.0.2           matrixStats_0.59.0         
##  [77] bitops_1.0-7                evaluate_0.14              
##  [79] lattice_0.20-44             purrr_0.3.4                
##  [81] labeling_0.4.2              GenomicAlignments_1.28.0   
##  [83] recipes_0.1.16              bit_4.0.4                  
##  [85] tidyselect_1.1.1            plyr_1.8.6                 
##  [87] magrittr_2.0.1              nls2_0.2                   
##  [89] R6_2.5.0                    generics_0.1.0             
##  [91] DelayedArray_0.18.0         DBI_1.1.1                  
##  [93] pillar_1.6.1                withr_2.4.2                
##  [95] survival_3.2-11             KEGGREST_1.32.0            
##  [97] RCurl_1.98-1.3              nnet_7.3-16                
##  [99] tibble_3.1.2                crayon_1.4.1               
## [101] futile.options_1.0.1        utf8_1.2.1                 
## [103] rmarkdown_2.9               minpack.lm_1.2-1           
## [105] data.table_1.14.0           blob_1.2.1                 
## [107] ModelMetrics_1.2.2.2        digest_0.6.27              
## [109] xtable_1.8-4                textshaping_0.3.5          
## [111] munsell_0.5.0               bslib_0.2.5.1</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
