<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Example of Methylome Analysis with MethylIT using Cancer Datasets • MethylIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Example of Methylome Analysis with MethylIT using Cancer Datasets">
<meta property="og:description" content="MethylIT">
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.6.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Example of Methylome Analysis with MethylIT
using Cancer Datasets</h1>
            <h3 data-toc-skip class="subtitle"><h1> ¶ A <a href="https://github.com/genomaths/%20MethylIT" class="external-link">Methyl-IT</a> user
guide. A guide for users</h1></h3>
                        <h4 data-toc-skip class="author"><div class="line-block">Robersy Sanchez<span class="math inline">\(^1\)</span><br><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a><br><br>
Sally Mackenzie<span class="math inline">\(^{1,2}\)</span><br><a href="mailto:sam795@psu.edu" class="email">sam795@psu.edu</a><br><br><span class="math inline">\(^1\)</span>Department of Biology.<br>
Pennsylvania State University, University Park, PA 16802<br><br><span class="math inline">\(^2\)</span>Department of Plant
Science.<br>
Pennsylvania State University, University Park, PA 16802</div></h4>
            
            <h4 data-toc-skip class="date">25 April 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/vignettes/cancer_example.Rmd" class="external-link"><code>vignettes/cancer_example.Rmd</code></a></small>
      <div class="hidden name"><code>cancer_example.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>An example of methylation analysis with Methyl-IT, a novel
      methylome analysis procedure based on a signal-detection and
      machine- learning approach, is provided here. Methylation analysis
      involves a signal detection problem, and the method was designed
      to discriminate methylation regulatory signal from background
      noise induced by thermal fluctuations. Methyl-IT enhances the
      resolution of genome methylation behavior to reveal
      network-associated responses, offering resolution of gene pathway
      influences not attainable with previous methods. Herein, an
      example of MethylIT application to the analysis of breast cancer
      methylomes is presented.</p>
    </div>
    
<style type="text/css">

.list-group-item.active, .list-group-item.active:focus,
.list-group-item.active:hover {
    z-index: 2;
    color: #fff;
    background-color: #337ab7;
    border-color: #337ab7;
}

.tocify-subheader .tocify-item {
    font-size: 0.80em;
    padding-left: 25px;
    text-indent: 0;
}

.tocify-header .tocify-item {
    font-size: 0.85em;
    padding-left: 20px;
    text-indent: 0;
}

body{ /* Normal */
    font-size: 18px;
    font-family: "Times New Roman", Times, serif;
    text-align: justify
}
td { /* Table */
    font-size: 8px;
}

h1.title {
    font-size: 38px;
    font-family: "Times New Roman", Times, serif;
    color: DarkRed;
    .subTitle {
        font-size: 24px;
        font-family: "Times New Roman", Times, serif;
        color: DarkRed;
    }
}

h1 { /* Header 1 */
    font-size: 28px;
    font-family: "Times New Roman", Times, serif;
    color: DarkBlue;
}
h2 { /* Header 2 */
    font-size: 22px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
h3 { /* Header 3 */
    font-size: 18px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
code.r{ /* Code block */
    font-size: 12px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>
<div class="section level2">
<h2 id="methyl-it">
<a href="https://github.com/genomaths/MethylIT" class="external-link">Methyl-IT</a><a class="anchor" aria-label="anchor" href="#methyl-it"></a>
</h2>
<p>MethylIT is an R package for methylome analysis based on a
signal-detection machine learning approach (SD-ML). This approach is
postulated to provide greater sensitivity for resolving true signal from
the methylation background within the methylome <span class="citation">(1, 2)</span>. Because the biological signal created
within the dynamic methylome environment characteristic of plants is not
free from background noise, the approach, designated Methyl-IT, includes
the application of signal detection theory <span class="citation">(3–6)</span>.</p>
<p>A basic requirement for the application of signal detection is the
knowledge of the probability distribution of the background noise, which
is used as null hypothesis in the detection of the methylation signal.
This probability distribution is, in general, a member of generalized
gamma distribution family, and it can be deduced on a statistical
mechanical/thermodynamics basis from DNA methylation induced by thermal
fluctuations <span class="citation">(1)</span>. Herein, we provide an
example of Methyl-IT application to the analysis of breast cancer
methylomes. Due to the size of human methylome the current example only
covers the analysis of chromosome 13. A full description of Methyl-IT
application of methylome analysis in plants is given in the manuscript
<span class="citation">(2)</span>.</p>
<p>The R packages required in this example are:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/MethylIT" class="external-link">MethylIT</a></span><span class="op">)</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Bioconductor/BiocParallel" class="external-link">BiocParallel</a></span><span class="op">)</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span> <span class="co"># graphic</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hadley/reshape" class="external-link">reshape2</a></span><span class="op">)</span> <span class="co"># To reshape the data frame</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">grid</span><span class="op">)</span> <span class="co"># For multiple plots</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span> <span class="co"># For multiple plots</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">VennDiagram</span><span class="op">)</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span> <span class="co"># To import gene annotation</span></span>
<span>    <span class="op">}</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="available-datasets-and-reading">Available datasets and reading<a class="anchor" aria-label="anchor" href="#available-datasets-and-reading"></a>
</h3>
<p>Methylome datasets of whole-genome bisulfite sequencing (WGBS) are
available at Gene Expression Omnibus (GEO DataSets). For the current
example, datasets from breast tissues (normal and cancer) and embryonic
stem cells will be downloaded from GEO. The data set are downloaded
providing the GEO accession numbers for each data set to the function
‘getGEOSuppFiles’ (see the function <a href="https://is.gd/JodM5G" class="external-link">getGEOSuppFiles</a> help). Some times the
ftp site is not available and the information must be manually
downloaded from GEO database.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Embryonic stem cells datasets</span></span>
<span><span class="va">esc.files</span> <span class="op">=</span> <span class="fu"><a href="../reference/getGEOSuppFiles.html">getGEOSuppFiles</a></span><span class="op">(</span>GEO <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSM2041690"</span>, <span class="st">"GSM2041691"</span>, <span class="st">"GSM2041692"</span><span class="op">)</span>,</span>
<span>                            verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># Breast tissues (normal, cancer, metastasis)</span></span>
<span><span class="va">cancer.files</span> <span class="op">=</span> <span class="fu"><a href="../reference/getGEOSuppFiles.html">getGEOSuppFiles</a></span><span class="op">(</span>GEO <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSM1279517"</span>,<span class="st">"GSM1279514"</span>,</span>
<span>                                        <span class="st">"GSM1279513"</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>The file path and name of each downloaded dataset is found in the
output variables ‘esc.files’ and ‘cancer.files’.</p>
<p>It important to keep in mind, however, that the file access at the <a href="https://www.ncbi.nlm.nih.gov/geo/" class="external-link">Gene Expression Omnibus -
NCBI</a> can be changed at the convenience of GEO database and, in this
case, getGEOSuppFile can fail.</p>
</div>
<div class="section level3">
<h3 id="reading-datasets">
<em>Reading datasets</em><a class="anchor" aria-label="anchor" href="#reading-datasets"></a>
</h3>
<p>Datasets for our example can be read with function <a href="https://is.gd/5pyk1N" class="external-link">readCounts2GRangesList</a>. To specify the
reading of only chromosome 13, we can specify the parameter ‘chromosomes
= “Chr13”’. The symbol chromosome 13, in this case “Chr13”, must be
consistent with the annotation provided in the given GEO dataset. Each
file is wholly read with the setting ‘chromosomes = “Chr13”’ and then
the GRanges are built only with chromosome 13, which could be time
consuming. However, users working on Linux OS can specify the reading of
specific lines from each file by using regular expressions. For example,
if only chromosomes 1 and 3 are required, then we can set chromosomes =
NULL (default) and ‘chromosome.pattern = “^Chr[1,3]”’. This will read
all the lines in the downloaded files starting with the words “Chr1” or
“Chr3”. If we are interested in chromosomes 1 and 2, then we can set
‘chromosome.pattern = “^Chr[1-2]”’. If all the chromosomes are required,
then set chromosomes = NULL and chromosome.pattern = NULL (default).
Obviously, before read the files, user must check them to see which
annotation was used by the experimenters to denote the chromosomes,
e.g., “chr13” or just “13”, etc.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Embryonic stem cells datasets</span></span>
<span><span class="va">ref</span> <span class="op">=</span> <span class="fu"><a href="../reference/readCounts2GRangesList.html">readCounts2GRangesList</a></span><span class="op">(</span>filenames <span class="op">=</span> <span class="va">esc.files</span>,</span>
<span>                            sample.id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ESC1"</span>,<span class="st">"ESC2"</span>, <span class="st">"ESC3"</span><span class="op">)</span>,</span>
<span>                            columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fl">1</span>, start <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                        mC <span class="op">=</span> <span class="fl">4</span>, uC <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"^chr13"</span>,</span>
<span>                            remove <span class="op">=</span> <span class="cn">TRUE</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># Breast tissues (normal, cancer, metastasis)</span></span>
<span><span class="va">LR</span> <span class="op">=</span> <span class="fu"><a href="../reference/readCounts2GRangesList.html">readCounts2GRangesList</a></span><span class="op">(</span>filenames <span class="op">=</span> <span class="va">cancer.files</span>,</span>
<span>                            sample.id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>, <span class="st">"Breast_cancer"</span>,</span>
<span>                                            <span class="st">"Breast_metastasis"</span><span class="op">)</span>,</span>
<span>                            columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fl">1</span>, start <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                        mC <span class="op">=</span> <span class="fl">3</span>, uC <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>,</span>
<span>                            remove <span class="op">=</span> <span class="cn">TRUE</span>, pattern <span class="op">=</span> <span class="st">"^13"</span>,</span>
<span>                            chromosome.names <span class="op">=</span> <span class="st">"chr13"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>In real data analysis, a bigger sample size must be used. For the
purpose of this example, to illustrate the application of methylation
analysis with Methyl-IT, we will just add an artificial normal
sample.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">LR</span><span class="op">$</span><span class="va">Breast_normal1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/poolFromGRlist.html">poolFromGRlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">ref</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="va">LR</span><span class="op">$</span><span class="va">Breast_normal</span>,</span>
<span>                                        <span class="va">LR</span><span class="op">$</span><span class="va">Breast_normal</span>, <span class="va">LR</span><span class="op">$</span><span class="va">Breast_normal</span>,</span>
<span>                                        <span class="va">LR</span><span class="op">$</span><span class="va">Breast_normal</span>, <span class="va">LR</span><span class="op">$</span><span class="va">Breast_normal</span><span class="op">)</span>,</span>
<span>                                    stat <span class="op">=</span> <span class="st">"mean"</span>, num.cores <span class="op">=</span> <span class="fl">6L</span>,</span>
<span>                                    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>In the metacolumn of the last GRanges object, mC and uC stand for the
methylated and unmethylated read counts, respectively. Notice that
option ‘remove = TRUE’ remove the decompressed files (default: FALSE,
see ? readCounts2GRangesList for more details about this function).</p>
</div>
<div class="section level3">
<h3 id="the-reference-individual">The reference individual<a class="anchor" aria-label="anchor" href="#the-reference-individual"></a>
</h3>
<p>Any two objects can be compared based on some measured variable if,
and only if, the measurements where taken in the same metric space, the
same coordinate system, in respect to the same origin of coordinates.
Usually, in our daily 3D experience, our brain automatically sets up the
origin of coordinates equal to zero. The differences found in the
comparison depend on the reference used to perform the measurements and
from the metric system. The space where the objects are located (or the
set of objects) together with the metric is called metric space.</p>
<p>Unfortunately, this is not the case for the methylation process. Each
individual from the same population and even each chromosome follows an
independent stochastic methylation process. This is not a surprise,
since cytosine methylation affect the mechanical properties of DNA
molecule with strong dependence of the DNA sequence context (until, at
least. 6 bases at both sides of each cytosine site) <span class="citation">(7, 8)</span>.</p>
<p>To evaluate the methylation differences between individuals from
control and treatment we introduce a metric in the bidimensional space
of methylation levels <span class="math inline">\(P_i = (p_i,
1-p_i)\)</span>. Vectors <span class="math inline">\(P_i\)</span>
provide a measurement of the uncertainty of methylation levels <span class="citation">(2)</span>. However, to perform the comparison between
the uncertainty of methylation levels from each group of individuals,
control <span class="math inline">\((c)\)</span> and treatment <span class="math inline">\((t)\)</span>, we should estimate the uncertainty
variation with respect to the same individual reference on the mentioned
metric space. The reason to measure the uncertainty variation with
respect to the same reference resides in that even sibling individuals
follow an independent ontogenetic development and, consequently, their
developments follow independent stochastic processes. This a consequence
of the “omnipresent” action of the second law of thermodynamics in
living organisms. In the current example, we will create the reference
individual by pooling the methylation counts from the embryonic stem
cells.</p>
<p>It should be noticed that the results are sensitive to the reference
used. The statistics mean, median, or sum of the read counts at each
cytosine site of some control samples can be used to create a virtual
reference sample. A virtual reference individual can be built with
function <a href="https://is.gd/qR79kW" class="external-link">poolFromGRlist</a>. It is up to
the user whether to apply the ‘row sum’, ‘row mean’ or ‘row median’ of
methylated and unmethylated read counts at each cytosine site across
individuals. Notice that when “<em>mean</em>” is selected the virtual
reference individual is the group centroid, which play a fundamental
role in multivariate statistic.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Ref</span> <span class="op">=</span> <span class="fu"><a href="../reference/poolFromGRlist.html">poolFromGRlist</a></span><span class="op">(</span><span class="va">ref</span>, stat <span class="op">=</span> <span class="st">"mean"</span>, num.cores <span class="op">=</span> <span class="fl">6L</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Ref</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 1560637 ranges and 2 metadata columns:</span></span>
<span><span class="co">##             seqnames    ranges strand |        mC        uC</span></span>
<span><span class="co">##                &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##         [1]    chr13  19020631      * |         1         1</span></span>
<span><span class="co">##         [2]    chr13  19020633      * |         2         2</span></span>
<span><span class="co">##         [3]    chr13  19020642      * |         1         1</span></span>
<span><span class="co">##         [4]    chr13  19020643      * |         2         2</span></span>
<span><span class="co">##         [5]    chr13  19020679      * |         1         1</span></span>
<span><span class="co">##         ...      ...       ...    ... .       ...       ...</span></span>
<span><span class="co">##   [1560633]    chr13 115108993      * |         1         3</span></span>
<span><span class="co">##   [1560634]    chr13 115109022      * |         1         1</span></span>
<span><span class="co">##   [1560635]    chr13 115109023      * |         3         4</span></span>
<span><span class="co">##   [1560636]    chr13 115109523      * |         2         2</span></span>
<span><span class="co">##   [1560637]    chr13 115109524      * |         1         1</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre>
<p>Only direct lab experiments can reveal whether differences detected
with distinct references outside the experimental conditions for control
and treatment groups are real. The best reference would be estimated
using a subset of individuals from control group. Such a reference will
contribute to remove the intragroup variation, in control and in
treatment groups, induced by environmental changes external to or not
controlled by the experimental conditions.</p>
<p>Methylation analysis for each cytosine position is frequently
performed in the bidimensional space of <span class="math inline">\((methylated, unmethylated)\)</span> read counts.
Frequently, Fisher’s exact test is applied to a single cytosine
position, under the null hypothesis that the proportions <span class="math inline">\(p_{ct}=methylated_{ct}/(methylated_{ct}+unmethylated_{ct})\)</span>
and <span class="math inline">\(p_{tt}=methylated_{tt}/(methylated_{tt}+unmethylated_{tt})\)</span>
are the same for control and treatment, respectively. In this case, the
implicit reference point for the counts at every cytosine positions is
<span class="math inline">\((methylated = 0,unmethylated = 0)\)</span>,
which corresponds to the point <span class="math inline">\(P_i = (0,
1)\)</span>.</p>
<p>In our case, the Hellinger divergence (one of the metric used, here)
of each individual in respect to the reference is the variable to test
in place of <span class="math inline">\((methylated,
unmethylated)\)</span> read counts or the methylation levels <span class="math inline">\(P_i = (p_i, 1-p_i)\)</span>.</p>
<p>The use of references is restricted by the thermodynamics basis of
the theory. The current information-thermodynamics based approach is
supported on the following postulate: <em>“High changes of Hellinger
divergences are less frequent than low changes, provided that the
divergence is proportional to the amount of energy required to process
the change of information in the methylation system”</em>.</p>
<p>The last postulate acknowledges the action of the second law of
thermodynamics on the biomolecular methylation system. For the
methylation system, it implies that the frequencies of the information
divergences between methylation levels must be proportional to a
Boltzmann factor (see supplementary information from reference <span class="citation">(1)</span>). In other words, the frequencies of
information divergences values should follow a trend proportional to an
exponential decay.</p>
<p>If we do not observe such a behavior, then either the reference is
too far from experimental condition or we are dealing with an extreme
situation where the methylation machinery in the cell is dysfunctional.
The last situation is found, for example, in the silencing mutation at
the gene of cytosine-DNA-methyltransferase in Arabidopsis
<em>thaliana</em>. Methylation of 5-methylcytosine at CpG dinucleotides
is maintained by MET1 in plants.</p>
</div>
<div class="section level3">
<h3 id="hellinger-divergence-estimation">Hellinger divergence estimation<a class="anchor" aria-label="anchor" href="#hellinger-divergence-estimation"></a>
</h3>
<p>To perform the comparison between the uncertainty of methylation
levels from each group of individuals, control <span class="math inline">\((c)\)</span> and treatment $(t) $, the divergence
between the methylation levels of each individual is estimated with
respect to the same reference on the metric space formed by the vector
set <span class="math inline">\(P_i = (p_i, 1-p_i)\)</span> and the
Hellinger divergence <span class="math inline">\(H\)</span>. Basically,
the information divergence between the methylation levels of an
individual <span class="math inline">\(j\)</span> and reference sample
<span class="math inline">\(r\)</span> is estimated according to the
Hellinger divergence given by the formula:</p>
<p><span class="math display">\[H(\hat p_{ij},\hat p_{ir}) =
w_i[(\sqrt{\hat p_{ij}} - \sqrt{\hat
p_{ir}})^2+(\sqrt{1-\hat p_{ij}} - \sqrt{1-\hat p_{ir}})^2]\]</span>
where $w_i = 2</p>
<p>$, <span class="math inline">\(m_{ij} =
{n_i}^{mC_j}+{n_i}^{uC_j}+1\)</span>, <span class="math inline">\(m_{ir}
= {n_i}^{mC_r}+ {n_i}^{uC_r}+1\)</span> and <span class="math inline">\(j \in {\{c,t}\}\)</span>.</p>
<p>This equation for Hellinger divergence is given in reference <span class="citation">(9)</span>, but other information theoretical
divergences can be used as well. Next, the information divergence for
control (Breast_normal) and treatment (Breast_cancer and
Breast_metastasis) samples are estimated with respect to the reference
virtual individual. A Bayesian correction of counts can be selected or
not. In a Bayesian framework, methylated read counts are modeled by a
beta-binomial distribution, which accounts for both the biological and
sampling variations <span class="citation">(10–12)</span>.</p>
<p>In our case we adopted the Bayesian approach suggested in reference
<span class="citation">(13)</span> (Chapter 3). In a Bayesian framework
with uniform priors, the methylation level can be defined as: <span class="math inline">\(p = (mC + 1)/ (mC + uC + 2)\)</span>. However, the
most natural statistical model for replicated BS-seq DNA methylation
measurements is beta-binomial (the beta distribution is a prior
conjugate of binomial distribution). We consider the parameter <span class="math inline">\(p\)</span> (methylation level) in the binomial
distribution as randomly drawn from a beta distribution. The
hyper-parameters <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span> from the beta-binomial distribution
are interpreted as pseudo-counts. The information divergence is
estimated here using the function <a href="https://is.gd/0ViBO9" class="external-link">estimateDivergence</a>:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/date.html" class="external-link">date</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Tue Apr 25 23:57:37 2023"</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ptm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/proc.time.html" class="external-link">proc.time</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">HD</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateDivergence.html">estimateDivergence</a></span><span class="op">(</span>ref <span class="op">=</span> <span class="va">Ref</span>, </span>
<span>                        indiv <span class="op">=</span> <span class="va">LR</span>, </span>
<span>                        Bayesian <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                        min.coverage <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                        and.min.cov <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                        high.coverage <span class="op">=</span> <span class="fl">300</span>,</span>
<span>                        percentile <span class="op">=</span> <span class="fl">0.999</span>, </span>
<span>                        verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/proc.time.html" class="external-link">proc.time</a></span><span class="op">(</span><span class="op">)</span> <span class="op">-</span> <span class="va">ptm</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">/</span><span class="fl">60</span>, <span class="st">"minutes."</span>, <span class="fu"><a href="https://rdrr.io/r/base/date.html" class="external-link">date</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="co"># in min</span></span></code></pre></div>
<pre><code><span><span class="co">## 0.742 minutes. Tue Apr 25 23:58:22 2023</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">HD</span><span class="op">$</span><span class="va">Breast_cancer</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 584538 ranges and 9 metadata columns:</span></span>
<span><span class="co">##            seqnames    ranges strand |        c1        t1        c2        t2</span></span>
<span><span class="co">##               &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##        [1]    chr13  19020631      * |         1         1        14        24</span></span>
<span><span class="co">##        [2]    chr13  19020633      * |         2         2        14        25</span></span>
<span><span class="co">##        [3]    chr13  19020643      * |         2         2         7        38</span></span>
<span><span class="co">##        [4]    chr13  19020782      * |         1         1         6        56</span></span>
<span><span class="co">##        [5]    chr13  19020786      * |         1         1        10        53</span></span>
<span><span class="co">##        ...      ...       ...    ... .       ...       ...       ...       ...</span></span>
<span><span class="co">##   [584534]    chr13 115108776      * |         1         1        52        20</span></span>
<span><span class="co">##   [584535]    chr13 115108789      * |         2         2        27        43</span></span>
<span><span class="co">##   [584536]    chr13 115108993      * |         1         3        72         5</span></span>
<span><span class="co">##   [584537]    chr13 115109023      * |         3         4        56        36</span></span>
<span><span class="co">##   [584538]    chr13 115109524      * |         1         1        31         9</span></span>
<span><span class="co">##                   p1        p2        TV     bay.TV      hdiv</span></span>
<span><span class="co">##            &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##        [1]  0.451770  0.372577 -0.131579 -0.0791929 0.0361197</span></span>
<span><span class="co">##        [2]  0.452016  0.364028 -0.141026 -0.0879881 0.0713908</span></span>
<span><span class="co">##        [3]  0.452016  0.178801 -0.344444 -0.2732151 0.8097173</span></span>
<span><span class="co">##        [4]  0.451770  0.118131 -0.403226 -0.3336386 0.8438179</span></span>
<span><span class="co">##        [5]  0.451770  0.175569 -0.341270 -0.2762003 0.5281366</span></span>
<span><span class="co">##        ...       ...       ...       ...        ...       ...</span></span>
<span><span class="co">##   [584534]  0.451770  0.703359  0.222222   0.251589 0.3807153</span></span>
<span><span class="co">##   [584535]  0.452016  0.387024 -0.114286  -0.064992 0.0405589</span></span>
<span><span class="co">##   [584536]  0.449459  0.905400  0.685065   0.455941 2.5160771</span></span>
<span><span class="co">##   [584537]  0.451113  0.599126  0.180124   0.148013 0.3253924</span></span>
<span><span class="co">##   [584538]  0.451770  0.737171  0.275000   0.285402 0.4843780</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre>
<p>Function ‘estimateDivergence’ returns a list of GRanges objects with
the four columns of counts, the information divergence, and additional
columns:</p>
<pre><code>1. The original matrix of methylated ($c_i$) and unmethylated
    ($t_i$) read counts from control ($i=1$) and treatment ($i=2$) samples.
2. "p1" and "p2": methylation levels for control and treatment,
    respectively.
3. "bay.TV": total variation TV = p2 - p1.
4. "TV": total variation based on simple counts: $TV=c1/(c1+t1)-
    c2/(c2+t2)$.
5. "hdiv": Hellinger divergence.</code></pre>
<p>If Bayesian = TRUE, results are based on the posterior estimations of
methylation levels <span class="math inline">\(p1\)</span> and <span class="math inline">\(p2\)</span>. Filtering by coverage is provided at
this step which would be used unless previous filtering by coverage had
been applied. This is a pairwise filtering. Cytosine sites with
‘coverage’ &gt; ‘min.coverage’ and ‘coverage’ &lt; ‘percentile’ (e.g.,
99.9 coverage percentile) in at least one of the samples are preserved.
The coverage percentile used is the maximum estimated from both samples:
reference and individual.</p>
<p>For some GEO datasets only the methylation levels for each cytosine
site are provided. In this case, Hellinger divergence can be estimated
as given in reference <span class="citation">(1)</span>:</p>
<p><span class="math display">\[H(\hat p_{ij},\hat p_{ir}) = 2
\Big[(\sqrt{\hat p_{ij}} - \sqrt{\hat
p_{ir}})^2+(\sqrt{1-\hat p_{ij}} - \sqrt{1-\hat
p_{ir}})^2\Big]\]</span></p>
</div>
<div class="section level3">
<h3 id="descriptive-statistical-analysis">
<em>Descriptive statistical analysis</em><a class="anchor" aria-label="anchor" href="#descriptive-statistical-analysis"></a>
</h3>
<p>A descriptive analysis on the distribution of Hellinger divergences
of methylation levels is recommended. Like in any other statistical
analysis, descriptive statistical analysis help us to detect potential
issues in the raw data. It is the user responsibility to perform quality
check of his/her dataset before start to apply Methyl-IT. Nevertheless,
the quality checking of the raw data is not perfect. So, it is healthy
to sees for potential issues.</p>
<div class="section level4">
<h4 id="histogram-and-boxplots-of-divergences-estimated-in-each-sample">
<em>Histogram and boxplots of divergences estimated in each
sample</em><a class="anchor" aria-label="anchor" href="#histogram-and-boxplots-of-divergences-estimated-in-each-sample"></a>
</h4>
<p>First, the data of interest (Hellinger divergences, “hdiv”) are
selected from the GRanges objects:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">normal</span> <span class="op">=</span> <span class="va">HD</span><span class="op">$</span><span class="va">Breast_normal</span><span class="op">[</span>, <span class="st">"hdiv"</span><span class="op">]</span></span>
<span><span class="va">normal1</span> <span class="op">=</span> <span class="va">HD</span><span class="op">$</span><span class="va">Breast_normal1</span><span class="op">[</span>, <span class="st">"hdiv"</span><span class="op">]</span></span>
<span><span class="va">cancer</span> <span class="op">=</span> <span class="va">HD</span><span class="op">$</span><span class="va">Breast_cancer</span><span class="op">[</span>, <span class="st">"hdiv"</span><span class="op">]</span></span>
<span><span class="va">metastasis</span> <span class="op">=</span> <span class="va">HD</span><span class="op">$</span><span class="va">Breast_metastasis</span><span class="op">[</span>, <span class="st">"hdiv"</span><span class="op">]</span></span></code></pre></div>
<p>Next, a single GRanges object is built from the above set of GRanges
objects using the function <a href="https://is.gd/S6e4br" class="external-link">uniqueGRanges</a>. Notice that the number of
cores to use for parallel computation can be specified.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hd</span> <span class="op">=</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">normal</span>, <span class="va">normal1</span>, <span class="va">cancer</span>, <span class="va">metastasis</span><span class="op">)</span>,</span>
<span>                    missing <span class="op">=</span> <span class="cn">NA</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">hd</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"normal"</span>, <span class="st">"normal1"</span>, <span class="st">"cancer"</span>, <span class="st">"metastasis"</span><span class="op">)</span></span>
<span><span class="va">hd</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 793551 ranges and 4 metadata columns:</span></span>
<span><span class="co">##            seqnames    ranges strand |     normal    normal1    cancer</span></span>
<span><span class="co">##               &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##        [1]    chr13  19020631      * | 0.16998196 0.15993886 0.0361197</span></span>
<span><span class="co">##        [2]    chr13  19020633      * | 0.55408069 0.43588832 0.0713908</span></span>
<span><span class="co">##        [3]    chr13  19020643      * | 0.35265356 0.34033618 0.8097173</span></span>
<span><span class="co">##        [4]    chr13  19020680      * | 0.00658000 0.00809711        NA</span></span>
<span><span class="co">##        [5]    chr13  19020687      * | 0.00676718 0.00155600        NA</span></span>
<span><span class="co">##        ...      ...       ...    ... .        ...        ...       ...</span></span>
<span><span class="co">##   [793547]    chr13 115108776      * |   0.472447   0.452470 0.3807153</span></span>
<span><span class="co">##   [793548]    chr13 115108789      * |   0.412965   0.374928 0.0405589</span></span>
<span><span class="co">##   [793549]    chr13 115108993      * |   2.033301   1.778906 2.5160771</span></span>
<span><span class="co">##   [793550]    chr13 115109023      * |   2.490385   2.362445 0.3253924</span></span>
<span><span class="co">##   [793551]    chr13 115109524      * |   0.756880   0.713464 0.4843780</span></span>
<span><span class="co">##            metastasis</span></span>
<span><span class="co">##             &lt;numeric&gt;</span></span>
<span><span class="co">##        [1]  1.1844907</span></span>
<span><span class="co">##        [2]  1.8145776</span></span>
<span><span class="co">##        [3]  0.0171404</span></span>
<span><span class="co">##        [4]         NA</span></span>
<span><span class="co">##        [5]         NA</span></span>
<span><span class="co">##        ...        ...</span></span>
<span><span class="co">##   [793547]         NA</span></span>
<span><span class="co">##   [793548]         NA</span></span>
<span><span class="co">##   [793549]   2.406915</span></span>
<span><span class="co">##   [793550]   0.523286</span></span>
<span><span class="co">##   [793551]         NA</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre>
<p>Now, the Hellinger divergences estimated for each sample are in a
single matrix on the metacolumn of the GRanges object and we can proceed
to build the histogram and boxplot graphics for these data. Finally, the
data included in the graphic are:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define an auxiliary function</span></span>
<span><span class="va">fun_length</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html" class="external-link">median</a></span><span class="op">(</span><span class="va">x</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span>, </span>
<span>                      label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"n = "</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>normal <span class="op">=</span> <span class="va">hd</span><span class="op">$</span><span class="va">normal</span>, normal1 <span class="op">=</span> <span class="va">hd</span><span class="op">$</span><span class="va">normal1</span>,</span>
<span>                    cancer <span class="op">=</span> <span class="va">hd</span><span class="op">$</span><span class="va">cancer</span>, metastasis <span class="op">=</span> <span class="va">hd</span><span class="op">$</span><span class="va">metastasis</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast.tissue"</span>, <span class="st">"HD"</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">=</span> <span class="va">data</span><span class="op">[</span><span class="va">data</span><span class="op">$</span><span class="va">HD</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 3174204 rows and 2 columns</span></span>
<span><span class="co">##           Breast.tissue         HD</span></span>
<span><span class="co">##                &lt;factor&gt;  &lt;numeric&gt;</span></span>
<span><span class="co">## 1                normal 0.16998196</span></span>
<span><span class="co">## 2                normal 0.55408069</span></span>
<span><span class="co">## 3                normal 0.35265356</span></span>
<span><span class="co">## 4                normal 0.00658000</span></span>
<span><span class="co">## 5                normal 0.00676718</span></span>
<span><span class="co">## ...                 ...        ...</span></span>
<span><span class="co">## NA.617427    NA                 NA</span></span>
<span><span class="co">## NA.617428    NA                 NA</span></span>
<span><span class="co">## 3174202      metastasis   2.406915</span></span>
<span><span class="co">## 3174203      metastasis   0.523286</span></span>
<span><span class="co">## NA.617429    NA                 NA</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">HD</span>, fill <span class="op">=</span> <span class="va">Breast.tissue</span>, colour <span class="op">=</span> <span class="va">Breast.tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span>, binwidth <span class="op">=</span> <span class="fl">0.5</span>, position <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>                na.rm <span class="op">=</span> <span class="cn">TRUE</span>, size <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">xlim</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.4</span>,<span class="fl">20</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,</span>
<span>                axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, </span>
<span>                                        color <span class="op">=</span>  <span class="st">"black"</span>, hjust <span class="op">=</span> <span class="fl">0.5</span>, </span>
<span>                                        vjust <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>,color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span> <span class="st">"Counts"</span> <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Histogram of Hellinger Divergence"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">## <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">## <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">## <span style="color: #555555;">generated.</span></span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># For visualization purposes HD is limited to the interval 0 to 50</span></span>
<span><span class="va">dt</span> <span class="op">=</span> <span class="va">data</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">HD</span> <span class="op">&lt;</span> <span class="fl">50</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">dt</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Breast.tissue</span>, y <span class="op">=</span> <span class="va">HD</span> , fill <span class="op">=</span> <span class="va">Breast.tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">ylim</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.1</span>, <span class="fl">20</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_summary.html" class="external-link">stat_summary</a></span><span class="op">(</span>fun.data <span class="op">=</span> <span class="va">fun_length</span>, geom <span class="op">=</span> <span class="st">"text"</span>,</span>
<span>                position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html" class="external-link">position_dodge</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                size <span class="op">=</span> <span class="fl">6</span>, fontface <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>                                    hjust <span class="op">=</span> <span class="fl">0.5</span>, vjust <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>,color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"none"</span> <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Boxplot of Hellinger Divergence"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html" class="external-link">grid.arrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 6 rows containing non-finite values (`stat_summary()`).</span></span></code></pre>
<p><img src="cancer_example_files/figure-html/fig1-1.png" width="1920"></p>
<p>Except for the tail, most of the methylation changes occurred under
the area covered by the density curve corresponding to the normal breast
tissue. This is theoretically expected. This area is explainable in
statistical physical terms and, theoretically, it should fit a Weibull
distribution. The tails regions cover the methylation changes that, with
high probability, are not induced by thermal fluctuation and are not
addressed to stabilize the DNA molecule. These changes are methylation
signal. Professor David J. Miller (Department of Electrical Engineering,
Penn State) proposed modeling the distribution as a mixed Weibull
distribution to simultaneously describe the background methylation noise
and the methylation signal (personal communication, January, 2018). This
model approach seems to be supported by the above histogram, but it must
be studied before being incorporated in a future version of
Methyl-IT.</p>
</div>
<div class="section level4">
<h4 id="checking-cytosine-read-counts">
<em>Checking cytosine read counts</em><a class="anchor" aria-label="anchor" href="#checking-cytosine-read-counts"></a>
</h4>
<p>According to the parameter values set for the function <a href="https://is.gd/0ViBO9" class="external-link">estimateDivergence</a>, for each cytosine
site, at least one individual must have a 10 reads or more, and the
maximum coverage must be 500 or less.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Maximum coverage at each cytosine site</span></span>
<span><span class="va">covr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">HD</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">cov1</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">c1</span> <span class="op">+</span> <span class="va">x</span><span class="op">$</span><span class="va">t1</span></span>
<span>    <span class="va">cov2</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">c2</span> <span class="op">+</span> <span class="va">x</span><span class="op">$</span><span class="va">t2</span></span>
<span>    <span class="va">cov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">cov1</span>, <span class="va">cov2</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">max</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">cov</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Quantiles</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">covr</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">q60</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.6</span><span class="op">)</span></span>
<span>    <span class="va">q9999</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.9999</span><span class="op">)</span></span>
<span>    <span class="va">idx1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="va">q60</span><span class="op">)</span></span>
<span>    <span class="va">idx2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span></span>
<span>    <span class="va">q95</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.95</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">idx1</span>, <span class="va">idx2</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span>, <span class="va">q60</span>, </span>
<span>            <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.95</span>, <span class="fl">0.99</span>, <span class="fl">0.999</span>, <span class="fl">0.9999</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            num.siteGreater_8 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="fl">8</span><span class="op">)</span>,</span>
<span>            q60_to_500 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="va">q60</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">(</span><span class="va">x</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            num.siteGreater_500 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                   Min. 1st Qu. Median Mean 3rd Qu. Max. 60% 95% 99% 99.9%</span></span>
<span><span class="co">## Breast_normal        5      19     30   31      41  291  34  57  69    81</span></span>
<span><span class="co">## Breast_cancer        5      18     31   37      51  266  38  88 112   137</span></span>
<span><span class="co">## Breast_metastasis    5      14     27   31      43  281  32  70  91   115</span></span>
<span><span class="co">## Breast_normal1       5      17     26   27      35  279  29  48  58    68</span></span>
<span><span class="co">##                     99.99% num.siteGreater_8 q60_to_500 num.siteGreater_500</span></span>
<span><span class="co">## Breast_normal      94.0000            747005     319227                   0</span></span>
<span><span class="co">## Breast_cancer     165.0000            569474     234247                   0</span></span>
<span><span class="co">## Breast_metastasis 158.8469            435153     187884                   0</span></span>
<span><span class="co">## Breast_normal1     80.0000            738787     318786                   0</span></span></code></pre>
<p>The quantiles indicates acceptable values for the coverages.
Descriptive statistics for methylated reads can be checked as well.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Maximum coverage at each cytosine site</span></span>
<span><span class="va">methc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">HD</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">c1</span>, <span class="va">x</span><span class="op">$</span><span class="va">c2</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">max</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">r</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Quantiles</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">methc</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">q10</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.1</span><span class="op">)</span></span>
<span>    <span class="va">q60</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.6</span><span class="op">)</span></span>
<span>    <span class="va">q9999</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.9999</span><span class="op">)</span></span>
<span>    <span class="va">idx1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="va">q60</span><span class="op">)</span></span>
<span>    <span class="va">idx2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span></span>
<span>    <span class="va">q95</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.95</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">idx1</span>, <span class="va">idx2</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span>, <span class="va">q10</span>, <span class="va">q60</span>,</span>
<span>            <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.95</span>, <span class="fl">0.99</span>, <span class="fl">0.999</span>, <span class="fl">0.9999</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            num.siteGreater_8 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="fl">8</span><span class="op">)</span>,</span>
<span>            q60_to_500 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="va">q60</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">(</span><span class="va">x</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            num.siteGreater_500 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                   Min. 1st Qu. Median Mean 3rd Qu. Max. 10% 60% 95% 99% 99.9%</span></span>
<span><span class="co">## Breast_normal        4      14     24   25      34  265   7  28  49  60    72</span></span>
<span><span class="co">## Breast_cancer        4      10     21   27      38  252   5  27  74  98   123</span></span>
<span><span class="co">## Breast_metastasis    4       7     16   21      30  234   4  21  56  75    98</span></span>
<span><span class="co">## Breast_normal1       4      12     20   21      29  258   6  24  41  50    60</span></span>
<span><span class="co">##                     99.99% num.siteGreater_8 q60_to_500 num.siteGreater_500</span></span>
<span><span class="co">## Breast_normal      84.0000            677552     309300                   0</span></span>
<span><span class="co">## Breast_cancer     149.5463            472596     237092                   0</span></span>
<span><span class="co">## Breast_metastasis 121.0000            328544     184064                   0</span></span>
<span><span class="co">## Breast_normal1     70.0000            664530     304195                   0</span></span></code></pre>
<p>The critical values from empirical cumulative probability
distributions will be used in the downstream analysis.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">critical.val</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">HD</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">hd.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">hdiv</span>, <span class="fl">0.95</span><span class="op">)</span></span>
<span>    <span class="va">tv.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">bay.TV</span><span class="op">)</span>, <span class="fl">0.95</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>tv <span class="op">=</span> <span class="va">tv.95</span>, hd <span class="op">=</span> <span class="va">hd.95</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">critical.val</span></span></code></pre></div>
<pre><code><span><span class="co">##                      tv.95%   hd.95%</span></span>
<span><span class="co">## Breast_normal     0.4625591 2.888832</span></span>
<span><span class="co">## Breast_cancer     0.4795708 3.413531</span></span>
<span><span class="co">## Breast_metastasis 0.4638647 3.104796</span></span>
<span><span class="co">## Breast_normal1    0.4442791 2.476178</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="nonlinear-fit-of-probability-distribution-models">Nonlinear fit of probability distribution models<a class="anchor" aria-label="anchor" href="#nonlinear-fit-of-probability-distribution-models"></a>
</h3>
<p>A basic requirement for the application of signal detection is the
knowledge of the probability distribution of the background noise.
Probability distribution, as a Weibull distribution model, can be
deduced on a statistical mechanical/thermodynamics basis for DNA
methylation induced by thermal fluctuations <span class="citation">(1)</span>.</p>
<p>Assuming that this background methylation variation is consistent
with a Poisson process, it can be distinguished from variation
associated with methylation regulatory machinery, which is
non-independent for all genomic regions <span class="citation">(1)</span>. An information-theoretic divergence to
express the variation in methylation induced by background thermal
fluctuations will follow a Weibull distribution model, provided that it
is proportional to the minimum energy dissipated per bit of information
associated with the methylation change.</p>
<p>The nonlinear fit to a Weibull distribution model is performed by the
function <a href="https://is.gd/0BLLkT" class="external-link">nonlinearFitDist</a>. In
general, an information divergence of methylation levels will follow a
probability distribution from the family of generalized gamma
distribution. Next, the best fitted model will be selected from one of
the following distributions: two- and three-parameter Weibull
distributions, gamma with two- and three-parameters (“Gamma2P” or
“Gamma3P”), and generalized gamma with three-parameter (“GGamma3P”).</p>
<p>The best model is selected on the basis of Akaike’s information
criterion and the correlation coefficient of cross-validations for the
nonlinear regressions (R.Cross.val). These criteria evaluate different
information inferred from the models. AIC deals with the trade-off
between the goodness of fit (GOF) of the model and the complexity of the
model, while R.Cross.val provides information on the prediction
power/performance of the model when confronted with external dataset.
Cross-validations for the nonlinear regressions (R.Cross.val) were
performed as described in reference <span class="citation">(14)</span>.
In addition, Stein’s formula for adjusted R squared (<span class="math inline">\(\rho\)</span>) was used as an estimator of the
average cross-validation predictive power <span class="citation">(14)</span>.</p>
<p>In general, the best fitted model yields the best DMP classification
performance (see below). However, on special dataset, as the used in the
current example. Function <a href="https://is.gd/W7dlP0" class="external-link">gofReport</a>
search for the best fitted model between the set of models requested by
the user.</p>
<p>The best model including “GGamma3P” model:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Weibull2P"</span>, <span class="st">"Weibull3P"</span>, <span class="st">"Gamma2P"</span>, <span class="st">"Gamma3P"</span>, <span class="st">"GGamma3P"</span><span class="op">)</span></span>
<span><span class="va">gof_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gofReport.html">gofReport</a></span><span class="op">(</span>HD <span class="op">=</span> <span class="va">HD</span>, column <span class="op">=</span> <span class="fl">9L</span>, model <span class="op">=</span> <span class="va">d</span>, </span>
<span>                   num.cores <span class="op">=</span> <span class="fl">6L</span>,</span>
<span>                   output <span class="op">=</span> <span class="st">"all"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |======================================================================| 100%
## 
##  *** Creating report ...</code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gof_1</span><span class="op">$</span><span class="va">bestModel</span></span></code></pre></div>
<pre><code><span><span class="co">##     Breast_normal     Breast_cancer Breast_metastasis    Breast_normal1 </span></span>
<span><span class="co">##        "GGamma3P"        "GGamma3P"        "GGamma3P"        "GGamma3P"</span></span></code></pre>
<p>The best model excluding “GGamma3P” model:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Weibull2P"</span>, <span class="st">"Weibull3P"</span>, <span class="st">"Gamma2P"</span>, <span class="st">"Gamma3P"</span><span class="op">)</span></span>
<span><span class="va">gof_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gofReport.html">gofReport</a></span><span class="op">(</span>HD <span class="op">=</span> <span class="va">HD</span>, </span>
<span>                   column <span class="op">=</span> <span class="fl">9L</span>, </span>
<span>                   model <span class="op">=</span> <span class="va">d</span>, </span>
<span>                   num.cores <span class="op">=</span> <span class="fl">6L</span>, </span>
<span>                   output <span class="op">=</span> <span class="st">"all"</span>, </span>
<span>                   verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%
## 
##  *** Creating report ...</code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gof_2</span><span class="op">$</span><span class="va">bestModel</span></span></code></pre></div>
<pre><code><span><span class="co">##     Breast_normal     Breast_cancer Breast_metastasis    Breast_normal1 </span></span>
<span><span class="co">##       "Weibull2P"         "Gamma2P"         "Gamma2P"       "Weibull2P"</span></span></code></pre>
<p>When “GGamma3P” model is included, the goodness-of-fit indicators
used here suggest that it is best fitted model. However, a final
decision must take into account the DMP classification performance as
well (see below), which is accomplished after a signal detection step.
The reader must keep in mind that we are dealing here with numerical
algorithms, which are not perfect. Depending on dataset, the numerical
computation will confront different challengers. The difference between
g3p_R.Cross.val and gg3p_R.Cross.val suggests that model predictions
powers for “GGamma3P” and “Gamma3P” are close.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gof_1</span><span class="op">$</span><span class="va">stats</span></span></code></pre></div>
<pre><code><span><span class="co">##                    w2p_AIC w2p_R.Cross.val w3p_AIC w3p_R.Cross.val  g2p_AIC</span></span>
<span><span class="co">## Breast_normal     -3194788       0.9955368     Inf               0 -3165495</span></span>
<span><span class="co">## Breast_cancer     -2001256       0.9906861     Inf               0 -2153574</span></span>
<span><span class="co">## Breast_metastasis -1635630       0.9910272     Inf               0 -1847487</span></span>
<span><span class="co">## Breast_normal1    -3371455       0.9964338     Inf               0 -3326676</span></span>
<span><span class="co">##                   g2p_R.Cross.val g3p_AIC g3p_R.Cross.val gg3p_AIC</span></span>
<span><span class="co">## Breast_normal           0.9945423     Inf               0 -4452069</span></span>
<span><span class="co">## Breast_cancer           0.9912956     Inf               0 -2813248</span></span>
<span><span class="co">## Breast_metastasis       0.9940890     Inf               0 -2324142</span></span>
<span><span class="co">## Breast_normal1          0.9957575     Inf               0 -5057365</span></span>
<span><span class="co">##                   gg3p_R.Cross.val bestModel</span></span>
<span><span class="co">## Breast_normal            0.9989777      gg3p</span></span>
<span><span class="co">## Breast_cancer            0.9971319      gg3p</span></span>
<span><span class="co">## Breast_metastasis        0.9979396      gg3p</span></span>
<span><span class="co">## Breast_normal1           0.9995615      gg3p</span></span></code></pre>
<p>It is worthy to observe that although in the current example we
successfully performed the nonlinear fit for GGamma3P model, this is not
the general case when the number of cytosine sites goes close or above
<span class="math inline">\(10^6\)</span>, for which the computational
cost is very high. The nonlinear fit of GGamma3P or GGamma4P is, in
general, difficult since two different set of estimated parameters could
produce very close distribution curves. Also, it is important to check
the value of the scaling parameter from each distribution. The numerical
algorithms could find the best fitted model with scaling parameter
values close to zero, say e.g. <span class="math inline">\(10^{-8}\)</span>, and with the highest GOFs.
Situations like this are biologically meaningless and the fitted models
must be discarded <span class="citation">(1)</span>.</p>
</div>
<div class="section level3">
<h3 id="signal-detection">Signal detection<a class="anchor" aria-label="anchor" href="#signal-detection"></a>
</h3>
<p>The information thermodynamics-based approach is postulated to
provide greater sensitivity for resolving true signal from the
thermodynamic background within the methylome <span class="citation">(1)</span>. Since the biological signal created within
the dynamic methylome environment characteristic of living organisms is
not free from background noise, the approach, designated Methyl-IT,
includes the application of signal detection theory <span class="citation">(3–6)</span>. Signal detection is a critical step to
increase sensitivity and resolution of methylation signal by reducing
the signal-to-noise ratio and objectively controlling the false positive
rate and prediction accuracy/risk.</p>
</div>
<div class="section level3">
<h3 id="potential-methylation-signal">
<em>Potential methylation signal</em><a class="anchor" aria-label="anchor" href="#potential-methylation-signal"></a>
</h3>
<p>The first estimation in our signal detection step is the
identification of the cytosine sites carrying potential methylation
signal <span class="math inline">\(PS\)</span>. The methylation
regulatory signal does not hold the theoretical distribution and,
consequently, for a given level of significance <span class="math inline">\(\alpha\)</span> (Type I error probability, e.g.
<span class="math inline">\(\alpha = 0.05\)</span>), cytosine positions
<span class="math inline">\(k\)</span> with information divergence <span class="math inline">\(H_k &gt;= H_{\alpha = 0.05}\)</span> can be
selected as sites carrying potential signals <span class="math inline">\(PS\)</span>. The value of <span class="math inline">\(\alpha\)</span> can be specified.</p>
<p>For example, potential signals with <span class="math inline">\(H_k
&gt; H_{\alpha = 0.01}\)</span> can be selected. For each sample,
cytosine sites are selected based on the corresponding fitted
theoretical distribution model estimated in the previous step.
Additionally, since cytosine with <span class="math inline">\(|TV_{d_k}|
&lt; 0.1\)</span> are the most abundant sites, depending on the sample
(experiment), cytosine positions <span class="math inline">\(k\)</span>
with <span class="math inline">\(H_k &gt;= H_{\alpha = 0.05}\)</span>
and <span class="math inline">\(|TV_{d_k}| &lt; 0.1\)</span> can be
observed. To prevent the last situation we can select the <span class="math inline">\(PS\)</span> with the additional constraint <span class="math inline">\(|TV_{d_k}| &gt; TV_0\)</span>, where <span class="math inline">\(TV_0\)</span> (‘tv.cut’) is a user specified
value. The <span class="math inline">\(PS\)</span> is detected with the
function <a href="https://is.gd/yGORtg" class="external-link">getPotentialDIMP</a>:</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PS1</span> <span class="op">=</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">HD</span>, dist.name <span class="op">=</span> <span class="va">gof_1</span><span class="op">$</span><span class="va">bestModel</span>, nlms <span class="op">=</span> <span class="va">gof_1</span><span class="op">$</span><span class="va">nlms</span>,</span>
<span>                        div.col <span class="op">=</span> <span class="fl">9</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, tv.col <span class="op">=</span> <span class="fl">8</span>, tv.cut <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="va">PS2</span> <span class="op">=</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">HD</span>, dist.name <span class="op">=</span> <span class="va">gof_2</span><span class="op">$</span><span class="va">bestModel</span>, nlms <span class="op">=</span> <span class="va">gof_2</span><span class="op">$</span><span class="va">nlms</span>,</span>
<span>                        div.col <span class="op">=</span> <span class="fl">9</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, tv.col <span class="op">=</span> <span class="fl">8</span>, tv.cut <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="va">PS1</span><span class="op">$</span><span class="va">Breast_normal</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 372 ranges and 10 metadata columns:</span></span>
<span><span class="co">##         seqnames    ranges strand |        c1        t1        c2        t2</span></span>
<span><span class="co">##            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##     [1]    chr13  20325510      * |         4         4        60         0</span></span>
<span><span class="co">##     [2]    chr13  20414617      * |         3         3        59         0</span></span>
<span><span class="co">##     [3]    chr13  20647291      * |         2         2        59         0</span></span>
<span><span class="co">##     [4]    chr13  20658445      * |         4         4        59         0</span></span>
<span><span class="co">##     [5]    chr13  20755264      * |         2         2        62         0</span></span>
<span><span class="co">##     ...      ...       ...    ... .       ...       ...       ...       ...</span></span>
<span><span class="co">##   [368]    chr13 114597079      * |         3         3        63         0</span></span>
<span><span class="co">##   [369]    chr13 114839635      * |         3         3        62         0</span></span>
<span><span class="co">##   [370]    chr13 114840738      * |         3         3        63         0</span></span>
<span><span class="co">##   [371]    chr13 115057919      * |         4         4        58         0</span></span>
<span><span class="co">##   [372]    chr13 115090233      * |         2         2        61         0</span></span>
<span><span class="co">##                p1        p2        TV    bay.TV      hdiv       wprob</span></span>
<span><span class="co">##         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">##     [1]  0.450574  0.952109       0.5  0.501535   5.73503 1.50139e-11</span></span>
<span><span class="co">##     [2]  0.450303  0.951394       0.5  0.501091   4.56377 1.27451e-05</span></span>
<span><span class="co">##     [3]  0.450029  0.951394       0.5  0.501365   3.36306 8.66788e-03</span></span>
<span><span class="co">##     [4]  0.450574  0.951394       0.5  0.500820   5.69272 2.94217e-11</span></span>
<span><span class="co">##     [5]  0.450029  0.953476       0.5  0.503447   3.42776 6.89877e-03</span></span>
<span><span class="co">##     ...       ...       ...       ...       ...       ...         ...</span></span>
<span><span class="co">##   [368]  0.450303  0.954131       0.5  0.503828   4.68786 4.56271e-06</span></span>
<span><span class="co">##   [369]  0.450303  0.953476       0.5  0.503173   4.65774 5.89952e-06</span></span>
<span><span class="co">##   [370]  0.450303  0.954131       0.5  0.503828   4.68786 4.56271e-06</span></span>
<span><span class="co">##   [371]  0.450574  0.950658       0.5  0.500084   5.64958 5.74997e-11</span></span>
<span><span class="co">##   [372]  0.450029  0.952802       0.5  0.502773   3.40662 7.44184e-03</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PS2</span><span class="op">$</span><span class="va">Breast_normal</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 237 ranges and 10 metadata columns:</span></span>
<span><span class="co">##         seqnames    ranges strand |        c1        t1        c2        t2</span></span>
<span><span class="co">##            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##     [1]    chr13  20325510      * |         4         4        60         0</span></span>
<span><span class="co">##     [2]    chr13  20414617      * |         3         3        59         0</span></span>
<span><span class="co">##     [3]    chr13  20658445      * |         4         4        59         0</span></span>
<span><span class="co">##     [4]    chr13  21002790      * |         4         4        62         0</span></span>
<span><span class="co">##     [5]    chr13  21289664      * |         2         2        69         0</span></span>
<span><span class="co">##     ...      ...       ...    ... .       ...       ...       ...       ...</span></span>
<span><span class="co">##   [233]    chr13 114497528      * |         3         3        60         0</span></span>
<span><span class="co">##   [234]    chr13 114597079      * |         3         3        63         0</span></span>
<span><span class="co">##   [235]    chr13 114839635      * |         3         3        62         0</span></span>
<span><span class="co">##   [236]    chr13 114840738      * |         3         3        63         0</span></span>
<span><span class="co">##   [237]    chr13 115057919      * |         4         4        58         0</span></span>
<span><span class="co">##                p1        p2        TV    bay.TV      hdiv      wprob</span></span>
<span><span class="co">##         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;</span></span>
<span><span class="co">##     [1]  0.450574  0.952109       0.5  0.501535   5.73503 0.00772371</span></span>
<span><span class="co">##     [2]  0.450303  0.951394       0.5  0.501091   4.56377 0.02095007</span></span>
<span><span class="co">##     [3]  0.450574  0.951394       0.5  0.500820   5.69272 0.00800731</span></span>
<span><span class="co">##     [4]  0.450574  0.953476       0.5  0.502902   5.81721 0.00720116</span></span>
<span><span class="co">##     [5]  0.450029  0.957704       0.5  0.507675   3.56496 0.04900990</span></span>
<span><span class="co">##     ...       ...       ...       ...       ...       ...        ...</span></span>
<span><span class="co">##   [233]  0.450303  0.952109       0.5  0.501806   4.59571  0.0203880</span></span>
<span><span class="co">##   [234]  0.450303  0.954131       0.5  0.503828   4.68786  0.0188494</span></span>
<span><span class="co">##   [235]  0.450303  0.953476       0.5  0.503173   4.65774  0.0193392</span></span>
<span><span class="co">##   [236]  0.450303  0.954131       0.5  0.503828   4.68786  0.0188494</span></span>
<span><span class="co">##   [237]  0.450574  0.950658       0.5  0.500084   5.64958  0.0083073</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre>
<p>Model “GGamma3P” reported 20263 - 11314 = 8949 more DMPs in the
normal tissue than model “Gamma3P”.</p>
<p>Notice that the total variation distance <span class="math inline">\(TV_d=|TV|\)</span> is an information divergence as
well and that it can be used in place of Hellinger divergence <span class="citation">(1)</span>. The set of vectors <span class="math inline">\(P_i = (p_i, 1-p_i)\)</span> and distance function
<span class="math inline">\(TV_d\)</span> integrate a metric space <span class="citation">(2)</span>. In particular, after takes the Manhattan
distance between healthy individuals and patients methylation levels we
have:</p>
<p><span class="math display">\[M_d(p_{ij}, p_{ir})=\frac{1}{2}(|\hat
p_{ij} - \hat p_{ir}| + |
(1-\hat p_{ij}) - (1-\hat p_{ir})|)=|\hat p_{ij} - \hat
p_{ir}|=TV_d(p_{ij},
p_{ir})\]</span>.</p>
<p>That is, the absolute difference of methylation levels is a
particular case of Manhattan distance, which is named total variation
distance <span class="math inline">\(TV_d\)</span>. The quantitative
effect of the vector components <span class="math inline">\(1-\hat
p_{ij}\)</span> and <span class="math inline">\(1-\hat p_{ir}\)</span>
(in our case, the effect of unmethylated read counts) is not present in
<span class="math inline">\(TV_d\)</span> as in <span class="math inline">\(H(\hat p_{ij},\hat p_{ir})\)</span>.</p>
</div>
<div class="section level3">
<h3 id="histogram-and-boxplots-of-methylation-potential-signals">
<em>Histogram and boxplots of methylation potential
signals</em><a class="anchor" aria-label="anchor" href="#histogram-and-boxplots-of-methylation-potential-signals"></a>
</h3>
<p>As before, a single GRanges object is built from the above set
GRanges objects using the function <a href="https://is.gd/S6e4br" class="external-link">uniqueGRanges</a>, and the Hellinger
divergences of the cytosine sites carrying <span class="math inline">\(PS\)</span> (for each sample) are located in a
single matrix on the metacolumn of the GRanges object.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define an auxiliar function</span></span>
<span><span class="va">fun_length</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html" class="external-link">median</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1.5</span>, label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"n = "</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># The data to be used in the boxplot</span></span>
<span><span class="va">ps1</span> <span class="op">=</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">PS1</span>, missing <span class="op">=</span> <span class="cn">NA</span>, columns <span class="op">=</span> <span class="fl">9</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                    num.cores <span class="op">=</span> <span class="fl">6L</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">ps1</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"normal"</span>, <span class="st">"cancer"</span>, <span class="st">"metastasis"</span>, <span class="st">"normal1"</span><span class="op">)</span></span>
<span><span class="va">dat1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>normal <span class="op">=</span> <span class="va">ps1</span><span class="op">$</span><span class="va">normal</span>, normal1 <span class="op">=</span> <span class="va">ps1</span><span class="op">$</span><span class="va">normal1</span>,</span>
<span>                cancer <span class="op">=</span> <span class="va">ps1</span><span class="op">$</span><span class="va">cancer</span>, metastasis <span class="op">=</span> <span class="va">ps1</span><span class="op">$</span><span class="va">metastasis</span><span class="op">)</span></span>
<span><span class="va">dat1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast.tissue"</span>, <span class="st">"HD"</span><span class="op">)</span></span>
<span><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">dat1</span><span class="op">$</span><span class="va">HD</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">dat1</span> <span class="op">&lt;-</span> <span class="va">dat1</span><span class="op">[</span><span class="op">-</span><span class="va">idx</span>, <span class="op">]</span> <span class="co"># To remove missing data</span></span>
<span></span>
<span><span class="va">ps2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">PS2</span>, missing <span class="op">=</span> <span class="cn">NA</span>, columns <span class="op">=</span> <span class="fl">9</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                    num.cores <span class="op">=</span> <span class="fl">6L</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">ps2</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"normal"</span>, <span class="st">"cancer"</span>, <span class="st">"metastasis"</span>, <span class="st">"normal1"</span><span class="op">)</span></span>
<span><span class="va">dat2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>normal <span class="op">=</span> <span class="va">ps2</span><span class="op">$</span><span class="va">normal</span>, normal1 <span class="op">=</span> <span class="va">ps2</span><span class="op">$</span><span class="va">normal1</span>,</span>
<span>        cancer <span class="op">=</span> <span class="va">ps2</span><span class="op">$</span><span class="va">cancer</span>, metastasis <span class="op">=</span> <span class="va">ps2</span><span class="op">$</span><span class="va">metastasis</span><span class="op">)</span></span>
<span><span class="va">dat2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast.tissue"</span>, <span class="st">"HD"</span><span class="op">)</span></span>
<span><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">dat2</span><span class="op">$</span><span class="va">HD</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">dat2</span> <span class="op">&lt;-</span> <span class="va">dat2</span><span class="op">[</span><span class="op">-</span><span class="va">idx</span>, <span class="op">]</span> <span class="co"># To remove missing data</span></span>
<span></span>
<span><span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">dat1</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">HD</span>, fill <span class="op">=</span> <span class="va">Breast.tissue</span>, colour <span class="op">=</span> <span class="va">Breast.tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span>, bins <span class="op">=</span> <span class="fl">50</span>, position <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>        na.rm <span class="op">=</span> <span class="cn">TRUE</span>, size <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Counts"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">xlim</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">30</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>                                hjust <span class="op">=</span> <span class="fl">0.5</span>, vjust <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>,color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Histogram for Potential methylation signal. GGamma3P"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">dat1</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Breast.tissue</span>, y <span class="op">=</span> <span class="va">HD</span> , fill <span class="op">=</span> <span class="va">Breast.tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">ylim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">15</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_summary.html" class="external-link">stat_summary</a></span><span class="op">(</span>fun.data <span class="op">=</span> <span class="va">fun_length</span>, geom <span class="op">=</span> <span class="st">"text"</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html" class="external-link">position_dodge</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                size <span class="op">=</span> <span class="fl">6</span>, fontface <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">6</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>                                    hjust <span class="op">=</span> <span class="fl">0.5</span>, vjust <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>,color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"none"</span> <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Boxplot for Potential methylation signal. GGamma3P"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">p3</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">dat2</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">HD</span>, fill <span class="op">=</span> <span class="va">Breast.tissue</span>, colour <span class="op">=</span> <span class="va">Breast.tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span>, bins <span class="op">=</span> <span class="fl">50</span>, position <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>        na.rm <span class="op">=</span> <span class="cn">TRUE</span>, size <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Counts"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">xlim</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">30</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>                                    hjust <span class="op">=</span> <span class="fl">0.5</span>, vjust <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>,color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span>,</span>
<span>        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Histogram for Potential methylation signal. Gamma3P"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p4</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">dat2</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Breast.tissue</span>, y <span class="op">=</span> <span class="va">HD</span> , fill <span class="op">=</span> <span class="va">Breast.tissue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">ylim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">15</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_summary.html" class="external-link">stat_summary</a></span><span class="op">(</span>fun.data <span class="op">=</span> <span class="va">fun_length</span>, geom <span class="op">=</span> <span class="st">"text"</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html" class="external-link">position_dodge</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                size <span class="op">=</span> <span class="fl">6</span>, fontface <span class="op">=</span> <span class="st">"bold"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">6</span>, linetype<span class="op">=</span><span class="st">"dashed"</span>, color <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>                                    hjust <span class="op">=</span> <span class="fl">0.5</span>, vjust <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">20</span>,color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"none"</span> <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Boxplot for Potential methylation signal. Gamma3P"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html" class="external-link">grid.arrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, <span class="va">p4</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="cancer_example_files/figure-html/fig2-1.png" width="1920"></p>
<p>The graphics show that “Gamma3P” provides more information on the
discrimination of treatment potential DMPs from the control.</p>
</div>
<div class="section level3">
<h3 id="cutpoint-estimation-laws-of-statistical-physics-can-account-for-background">Cutpoint estimation Laws of statistical physics can account for
background<a class="anchor" aria-label="anchor" href="#cutpoint-estimation-laws-of-statistical-physics-can-account-for-background"></a>
</h3>
<p>methylation, a response to thermal fluctuations that presumably
functions in DNA stability <span class="citation">(1, 2)</span>. True
signal is detected based on the optimal cutpoint <span class="citation">(15)</span>, which can be estimated from the area under
the curve (AUC) of a receiver operating characteristic (ROC) curve built
from a logistic regression performed with the potential signals from
controls and treatments. The ROC AUC is equivalent to the probability
that a randomly- chosen positive instance is ranked more highly than a
randomly-chosen negative instance <span class="citation">(16)</span>. In
the current context, the AUC is equivalent to the probability to
distinguish a randomly-chosen methylation regulatory signal induced by
the treatment from a randomly-chosen signal in the control.</p>
<p>The need for the application of (what is now known as) signal
detection in cancer research was pointed out by Youden in the midst of
the last century <span class="citation">(17)</span>. In the next
example, the simple cutpoint estimation available in Methyl-IT is based
on the application of <a href="https://is.gd/t8UKvK" class="external-link">Youden index</a>
<span class="citation">(17)</span>. Although cutpoints are estimated for
a single variable, the classification performance can be evaluated for
several variables and applying different model classifiers. A optimal
cutpoint distinguishes disease stages from healthy individual. The
performance of this classification is given in the output of function <a href="https://is.gd/RR9iWY" class="external-link">estimateCutPoint</a>.</p>
<p>A model classifier can be requested for further predictions and its
classification performance is also provided. Below, the selected model
classifier is a logistic model applied on the principal components
derived from a previous principal component analysis (PCA,
<em>classifier1</em> = “pca.logistic”“, <em>clas.perf</em> = TRUE). Four
predictor variables are available: the Hellinger divergence of
methylation levels (<em>hdiv</em>), total variation distance (with
Bayesian correction for the methylation levels, <em>bay.TV</em>,
absolute difference of methylation levels), relative position of
cytosine site in the chromosome (<em>pos</em>), and the logarithm base
two of the probability to observe a Hellinger divergence value <span class="math inline">\(H\)</span> greater than the critical value <span class="math inline">\(H_{\alpha = 0.05}\)</span> (values given as
probabilities in object PS, <em>wprob</em>).</p>
<p>Notice that the cutpoint can be estimated for any of the two
currently available information divergences: <em>Hellinger
divergence</em> (<em>div.col</em> = 9) or the <em>total</em>
<em>variation distance</em> (with Bayesian correction, <em>div.col</em>
= 8).</p>
<div class="section level4">
<h4 id="cutpoint-based-on-youden-index">
<em>Cutpoint based on <em>Youden index</em></em><a class="anchor" aria-label="anchor" href="#cutpoint-based-on-youden-index"></a>
</h4>
<p>Null distribution “GGamma3P”:</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint1</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">PS1</span>,</span>
<span>                    control.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>,<span class="st">"Breast_normal1"</span><span class="op">)</span>,</span>
<span>                    treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast_cancer"</span>, <span class="st">"Breast_metastasis"</span><span class="op">)</span>,</span>
<span>                    simple <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                    classifier1 <span class="op">=</span> <span class="st">"pca.logistic"</span>,</span>
<span>                    column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                    n.pc <span class="op">=</span> <span class="fl">4</span> , center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                    div.col <span class="op">=</span> <span class="fl">9</span>, clas.perf <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">cutpoint1</span><span class="op">$</span><span class="va">cutpoint</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 3.408457</span></span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint1</span><span class="op">$</span><span class="va">testSetPerformance</span></span></code></pre></div>
<pre><code><span><span class="co">## Confusion Matrix and Statistics</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##           Reference</span></span>
<span><span class="co">## Prediction   CT   TT</span></span>
<span><span class="co">##         CT    0    2</span></span>
<span><span class="co">##         TT  110 1538</span></span>
<span><span class="co">##                                           </span></span>
<span><span class="co">##                Accuracy : 0.9321          </span></span>
<span><span class="co">##                  95% CI : (0.9189, 0.9438)</span></span>
<span><span class="co">##     No Information Rate : 0.9333          </span></span>
<span><span class="co">##     P-Value [Acc &gt; NIR] : 0.6026          </span></span>
<span><span class="co">##                                           </span></span>
<span><span class="co">##                   Kappa : -0.0024         </span></span>
<span><span class="co">##                                           </span></span>
<span><span class="co">##  Mcnemar's Test P-Value : &lt;2e-16          </span></span>
<span><span class="co">##                                           </span></span>
<span><span class="co">##             Sensitivity : 0.9987          </span></span>
<span><span class="co">##             Specificity : 0.0000          </span></span>
<span><span class="co">##          Pos Pred Value : 0.9333          </span></span>
<span><span class="co">##          Neg Pred Value : 0.0000          </span></span>
<span><span class="co">##              Prevalence : 0.9333          </span></span>
<span><span class="co">##          Detection Rate : 0.9321          </span></span>
<span><span class="co">##    Detection Prevalence : 0.9988          </span></span>
<span><span class="co">##       Balanced Accuracy : 0.4994          </span></span>
<span><span class="co">##                                           </span></span>
<span><span class="co">##        'Positive' Class : TT              </span></span>
<span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint1</span><span class="op">$</span><span class="va">testSetModel.FDR</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.001298701</span></span></code></pre>
<p>Null distribution “Gamma3P”:</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint2</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">PS2</span>,</span>
<span>                    control.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>,<span class="st">"Breast_normal1"</span><span class="op">)</span>,</span>
<span>                    treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast_cancer"</span>, <span class="st">"Breast_metastasis"</span><span class="op">)</span>,</span>
<span>                    simple <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                    classifier1 <span class="op">=</span> <span class="st">"pca.logistic"</span>,</span>
<span>                    column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                    n.pc <span class="op">=</span> <span class="fl">4</span> , center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                    div.col <span class="op">=</span> <span class="fl">9</span>, clas.perf <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">cutpoint2</span><span class="op">$</span><span class="va">cutpoint</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 4.846669</span></span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint2</span><span class="op">$</span><span class="va">testSetPerformance</span></span></code></pre></div>
<pre><code><span><span class="co">## Confusion Matrix and Statistics</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##           Reference</span></span>
<span><span class="co">## Prediction  CT  TT</span></span>
<span><span class="co">##         CT  42   0</span></span>
<span><span class="co">##         TT   0 487</span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##                Accuracy : 1          </span></span>
<span><span class="co">##                  95% CI : (0.9931, 1)</span></span>
<span><span class="co">##     No Information Rate : 0.9206     </span></span>
<span><span class="co">##     P-Value [Acc &gt; NIR] : &lt; 2.2e-16  </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##                   Kappa : 1          </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##  Mcnemar's Test P-Value : NA         </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##             Sensitivity : 1.0000     </span></span>
<span><span class="co">##             Specificity : 1.0000     </span></span>
<span><span class="co">##          Pos Pred Value : 1.0000     </span></span>
<span><span class="co">##          Neg Pred Value : 1.0000     </span></span>
<span><span class="co">##              Prevalence : 0.9206     </span></span>
<span><span class="co">##          Detection Rate : 0.9206     </span></span>
<span><span class="co">##    Detection Prevalence : 0.9206     </span></span>
<span><span class="co">##       Balanced Accuracy : 1.0000     </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##        'Positive' Class : TT         </span></span>
<span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint2</span><span class="op">$</span><span class="va">testSetModel.FDR</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0</span></span></code></pre>
<p>The cutpoint estimated based “GGamma3P” model is lower than the value
based on “Gamma3P” model. Often, this is a good detail, since it permits
the identification of more sites with high probability to carry
methylation signal. However, the evaluation of the classification
performance has the last word on model selection. In the current case,
the DMP classification performance is much better for “Gamma3P” model
than for “GGamma3P”.</p>
<p>In particular, the probabilities derived from each distribution model
provide fundamental information to the model classifier. Hence, the
probability that a potential DMP could carry a methylation signal
estimated using “Gamma3P” model is more informative (for the model
classifier) than the probability estimated using “GGamma3P” model.</p>
</div>
<div class="section level4">
<h4 id="cutpoint-estimated-using-a-model-classifier">
<em>Cutpoint estimated using a model classifier</em><a class="anchor" aria-label="anchor" href="#cutpoint-estimated-using-a-model-classifier"></a>
</h4>
<p>In the current case, the DMP classification performance for the
signal derived with null distribution “Gamma3P” can be improved
searching for an optimal cutpoint with a model classifier and using the
total variation distance ( with Bayesian correction, bay.TV).</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint2</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">PS2</span>,</span>
<span>                    control.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>,<span class="st">"Breast_normal1"</span><span class="op">)</span>,</span>
<span>                    treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast_cancer"</span>, <span class="st">"Breast_metastasis"</span><span class="op">)</span>,</span>
<span>                    simple <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                    classifier1 <span class="op">=</span> <span class="st">"pca.logistic"</span>,</span>
<span>                    classifier2 <span class="op">=</span> <span class="st">"pca.qda"</span>,</span>
<span>                    column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                    n.pc <span class="op">=</span> <span class="fl">4</span> , center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                    div.col <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span>
<span><span class="va">cutpoint2</span><span class="op">$</span><span class="va">cutpoint</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.5000129</span></span></code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint2</span><span class="op">$</span><span class="va">testSetPerformance</span></span></code></pre></div>
<pre><code><span><span class="co">## Confusion Matrix and Statistics</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##           Reference</span></span>
<span><span class="co">## Prediction  CT  TT</span></span>
<span><span class="co">##         CT  97   0</span></span>
<span><span class="co">##         TT   0 514</span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##                Accuracy : 1         </span></span>
<span><span class="co">##                  95% CI : (0.994, 1)</span></span>
<span><span class="co">##     No Information Rate : 0.8412    </span></span>
<span><span class="co">##     P-Value [Acc &gt; NIR] : &lt; 2.2e-16 </span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##                   Kappa : 1         </span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##  Mcnemar's Test P-Value : NA        </span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##             Sensitivity : 1.0000    </span></span>
<span><span class="co">##             Specificity : 1.0000    </span></span>
<span><span class="co">##          Pos Pred Value : 1.0000    </span></span>
<span><span class="co">##          Neg Pred Value : 1.0000    </span></span>
<span><span class="co">##              Prevalence : 0.8412    </span></span>
<span><span class="co">##          Detection Rate : 0.8412    </span></span>
<span><span class="co">##    Detection Prevalence : 0.8412    </span></span>
<span><span class="co">##       Balanced Accuracy : 1.0000    </span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##        'Positive' Class : TT        </span></span>
<span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint2</span><span class="op">$</span><span class="va">testSetModel.FDR</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0</span></span></code></pre>
<p>The good DMP classification performance was kept at lower cutpoint
value.</p>
</div>
</div>
<div class="section level3">
<h3 id="dmps">DMPs<a class="anchor" aria-label="anchor" href="#dmps"></a>
</h3>
<p>Cytosine sites carrying a methylation signal are designated
differentially informative methylated positions (DMPs). The probability
that a DMP is not induced by the treatment is given by the probability
of false alarm (P_FA, false positive, Sanchez et al. 2019). That is, the
biological signal is naturally present in the control as well as in the
treatment. According to the classical view in the epigenomic field, a
DMP is a cytosine position carrying a significant methylation signal,
which might be identified as differentially methylated cytosine site by
Fisher’s exact test (or other current tests).</p>
<p>In Methyl-IT, a DMP is a DNA cytosine position with high probability
to be differentially hyper-methylated or hypo-methylated in the
treatment with respect to a given control (reference
individual)<em><em>. Notice that Methyl- IT definition of DMP is not
deterministic in an ordinary sense, but stochastic- deterministic in
physico-mathematical terms. In other words, if a given cytosine position
is identified as a hyper-methylated DMP in the treatment, then with high
probability such a site will be found more frequently hyper- methylated
than hypo-methylated in the </em></em>treatment population*_. Observe
that it does not mean that, for all arbitrary subset of samples, the
assumed cytosine position will be hypermethylated.</p>
<p>DMPs are selected with the function <a href="https://is.gd/%207cVEEd" class="external-link">selectDIMP</a></p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DMPs</span> <span class="op">=</span> <span class="fu"><a href="../reference/selectDIMP.html">selectDIMP</a></span><span class="op">(</span><span class="va">PS2</span>, div.col <span class="op">=</span> <span class="fl">8</span>, cutpoint <span class="op">=</span> <span class="va">cutpoint2</span><span class="op">$</span><span class="va">cutpoint</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="the-descriptive-statistic-of-the-read-counts-at-dmps">The descriptive statistic of the read counts at DMPs<a class="anchor" aria-label="anchor" href="#the-descriptive-statistic-of-the-read-counts-at-dmps"></a>
</h4>
<p>After the signal detection step the read counts at DMPs is notable
greater than at the general methylated cytosine background:</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Maximum coverage at each cytosine site</span></span>
<span><span class="va">methc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">DMPs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>                        <span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">c1</span>, <span class="va">x</span><span class="op">$</span><span class="va">c2</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">max</span><span class="op">)</span></span>
<span>                        <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">r</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Quantiles</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">methc</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">q10</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.1</span><span class="op">)</span></span>
<span>    <span class="va">q60</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.6</span><span class="op">)</span></span>
<span>    <span class="va">q9999</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.9999</span><span class="op">)</span></span>
<span>    <span class="va">idx1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="va">q60</span><span class="op">)</span></span>
<span>    <span class="va">idx2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span></span>
<span>    <span class="va">q95</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.95</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">idx1</span>, <span class="va">idx2</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span>, <span class="va">q10</span>, <span class="va">q60</span>,</span>
<span>            <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.95</span>, <span class="fl">0.99</span>, <span class="fl">0.999</span>, <span class="fl">0.9999</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            num.siteGreater_8 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="fl">8</span><span class="op">)</span>,</span>
<span>            q60_to_500 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="va">q60</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">(</span><span class="va">x</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            num.siteGreater_500 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                   Min. 1st Qu. Median Mean 3rd Qu. Max.  10%  60%    95%    99%</span></span>
<span><span class="co">## Breast_normal       27      59     62   63      67   91 57.0 63.0  76.00  84.92</span></span>
<span><span class="co">## Breast_cancer       43      70     83   84      99  160 57.6 89.0 118.00 130.00</span></span>
<span><span class="co">## Breast_metastasis   52      59     68   71      79  119 56.5 72.0  96.75 110.20</span></span>
<span><span class="co">## Breast_normal1      61      61     63   63      64   64 61.0 63.4  64.00  64.00</span></span>
<span><span class="co">##                     99.9%   99.99% num.siteGreater_8 q60_to_500</span></span>
<span><span class="co">## Breast_normal      89.820  90.8820               237        101</span></span>
<span><span class="co">## Breast_cancer     157.712 159.7928              1037        417</span></span>
<span><span class="co">## Breast_metastasis 117.775 118.8775               246         99</span></span>
<span><span class="co">## Breast_normal1     64.000  64.0000                 5          2</span></span>
<span><span class="co">##                   num.siteGreater_500</span></span>
<span><span class="co">## Breast_normal                       0</span></span>
<span><span class="co">## Breast_cancer                       0</span></span>
<span><span class="co">## Breast_metastasis                   0</span></span>
<span><span class="co">## Breast_normal1                      0</span></span></code></pre>
<p>The minimum amount of methylated read counts is observed
“Breast_metastasis” (15 reads) and for any the samples the tenth percent
of DMPs is 26 or more reads.</p>
</div>
<div class="section level4">
<h4 id="venn-diagram-of-dmps">
<em>Venn Diagram of DMPs</em><a class="anchor" aria-label="anchor" href="#venn-diagram-of-dmps"></a>
</h4>
<p>The Venn diagram of DMPs reveals that the number cytosine site
carrying methylation signal with a divergence level comparable to that
observed in breast tissues with cancer and metastasis is relatively
small (2797 DMPs). The number of DMPs decreased in the breast tissue
with metastasis, but, as shown in the last boxplot, the intensity of the
signal increased.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n12</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_normal</span>, </span>
<span>                                    <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_cancer</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n13</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_normal</span>, </span>
<span>                                    <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_metastasis</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n23</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_cancer</span>,</span>
<span>                                    <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_metastasis</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n123</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/funprog.html" class="external-link">Reduce</a></span><span class="op">(</span><span class="fu">GenomicRanges</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span>,</span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_normal</span>, <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_cancer</span>,</span>
<span>                        <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_metastasis</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grid/grid.newpage.html" class="external-link">grid.newpage</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">v</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/VennDiagram/man/draw.triple.venn.html" class="external-link">draw.triple.venn</a></span><span class="op">(</span>area1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_normal</span><span class="op">)</span>,</span>
<span>                    area2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_cancer</span><span class="op">)</span>,</span>
<span>                    area3 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_metastasis</span><span class="op">)</span>,</span>
<span>                    n12 <span class="op">=</span> <span class="va">n12</span>, n23 <span class="op">=</span> <span class="va">n23</span>, n13 <span class="op">=</span> <span class="va">n13</span>, n123 <span class="op">=</span> <span class="va">n123</span>,</span>
<span>                    category <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>, <span class="st">"Breast_cancer"</span>,</span>
<span>                                <span class="st">"Breast_metastasis"</span><span class="op">)</span>,</span>
<span>                    lty <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"blank"</span>, <span class="fl">3</span><span class="op">)</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"yellow"</span>, </span>
<span>                                                    <span class="st">"magenta"</span><span class="op">)</span>,</span>
<span>                    alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.2</span>, <span class="fl">0.3</span><span class="op">)</span>,</span>
<span>                    cat.pos <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">110</span>, <span class="fl">110</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>                    cat.col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"darkorange"</span>, <span class="st">"red"</span><span class="op">)</span>,</span>
<span>                    cat.dist <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span> <span class="op">-</span><span class="fl">0.1</span>, <span class="op">-</span><span class="fl">0.1</span>, <span class="op">-</span><span class="fl">0.26</span><span class="op">)</span>,</span>
<span>                    cex <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1.7</span>, <span class="fl">7</span><span class="op">)</span>,</span>
<span>                    cat.cex <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span> <span class="fl">1.5</span>, <span class="fl">1.5</span>, <span class="fl">1.5</span><span class="op">)</span>,</span>
<span>                    label.col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span> <span class="st">"blue"</span>, <span class="st">"darkorange"</span>, <span class="st">"darkorange"</span>,</span>
<span>                                <span class="st">"red"</span>, <span class="st">"white"</span>, <span class="st">"red"</span>, <span class="st">"red"</span><span class="op">)</span>,</span>
<span>                    scaled <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grid/grid.draw.html" class="external-link">grid.draw</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span></span></code></pre></div>
<p><img src="cancer_example_files/figure-html/fig4-1.png" width="672"></p>
<p>Notice that natural methylation regulatory signals (not induced by
the treatment) are present in both groups, control and treatment. The
signal detection step permits us to discriminate the “ordinary” signals
observed in the control from those induced by the treatment (a disease
in the current case). In addition, this diagram reflects a
classification of DMPs only based on the cytosine positions. That is,
this Venn diagram cannot tell us whether DMPs at the same position can
be distinguishable or not. For example, DMPs at the same positions in
control and treatment can happen with different probabilities estimated
from their corresponding fitted Weibull distributions (see below).</p>
</div>
<div class="section level4">
<h4 id="monte-carlo-evaluation-of-the-model-classification">Monte Carlo Evaluation of the Model Classification<a class="anchor" aria-label="anchor" href="#monte-carlo-evaluation-of-the-model-classification"></a>
</h4>
<p>Performance The regulatory methylation signal is an output from a
natural process that continuously takes place across the ontogenetic
development of the organism. Therefore, we expect to see methylation
signal in natural, ordinary conditions. Function ‘evaluateDIMPclass’ can
be used to perform a classification of DMPs into two classes: DMPS from
control and DMPs from treatment samples, as well as an evaluation of the
classification performance (for more details see <a href="https://is.gd/nzy9j3" class="external-link">evaluateDIMPclass</a>).</p>
<p>As a matter of fact this function is called by function <a href="https://is.gd/RR9iWY" class="external-link">estimateCutPoint</a>. Additional feature not
used before is the possibility to perform Monte Carlo evaluation of the
model classifier performance. In above split of the sample into two
subsets, training and test datasets, we would be just lucky getting the
right proportion that yields the best classification performance. What
about is the random split is repeat 300 times?</p>
<p>The performance of the “pca.qda” classifier is:</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">performance</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluateDIMPclass.html">evaluateDIMPclass</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">DMPs</span>,</span>
<span>                    control.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>, <span class="st">"Breast_normal1"</span><span class="op">)</span>,</span>
<span>                    treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast_cancer"</span>, <span class="st">"Breast_metastasis"</span><span class="op">)</span>,</span>
<span>                    column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                    classifier <span class="op">=</span> <span class="st">"pca.qda"</span>, prop <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>                    n.pc <span class="op">=</span> <span class="fl">4</span> , center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                    num.boot <span class="op">=</span> <span class="fl">300</span>, output <span class="op">=</span> <span class="st">"mc.val"</span>,</span>
<span>                    num.cores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html" class="external-link">multicoreWorkers</a></span><span class="op">(</span><span class="op">)</span>, tasks <span class="op">=</span> <span class="fl">2L</span><span class="op">)</span></span>
<span><span class="va">performance</span></span></code></pre></div>
<pre><code><span><span class="co">##     Accuracy          Kappa        AccuracyLower    AccuracyUpper   </span></span>
<span><span class="co">##  Min.   :0.9918   Min.   :0.9687   Min.   :0.9810   Min.   :0.9973  </span></span>
<span><span class="co">##  1st Qu.:1.0000   1st Qu.:1.0000   1st Qu.:0.9940   1st Qu.:1.0000  </span></span>
<span><span class="co">##  Median :1.0000   Median :1.0000   Median :0.9940   Median :1.0000  </span></span>
<span><span class="co">##  Mean   :0.9997   Mean   :0.9987   Mean   :0.9934   Mean   :0.9999  </span></span>
<span><span class="co">##  3rd Qu.:1.0000   3rd Qu.:1.0000   3rd Qu.:0.9940   3rd Qu.:1.0000  </span></span>
<span><span class="co">##  Max.   :1.0000   Max.   :1.0000   Max.   :0.9940   Max.   :1.0000  </span></span>
<span><span class="co">##                                                                     </span></span>
<span><span class="co">##   AccuracyNull    AccuracyPValue      McnemarPValue      Sensitivity</span></span>
<span><span class="co">##  Min.   :0.8412   Min.   :0.000e+00   Min.   :0.07364   Min.   :1   </span></span>
<span><span class="co">##  1st Qu.:0.8412   1st Qu.:0.000e+00   1st Qu.:0.24821   1st Qu.:1   </span></span>
<span><span class="co">##  Median :0.8412   Median :0.000e+00   Median :0.47950   Median :1   </span></span>
<span><span class="co">##  Mean   :0.8412   Mean   :1.603e-40   Mean   :0.35878   Mean   :1   </span></span>
<span><span class="co">##  3rd Qu.:0.8412   3rd Qu.:0.000e+00   3rd Qu.:0.47950   3rd Qu.:1   </span></span>
<span><span class="co">##  Max.   :0.8412   Max.   :2.341e-38   Max.   :0.47950   Max.   :1   </span></span>
<span><span class="co">##                                       NA's   :277                   </span></span>
<span><span class="co">##   Specificity     Pos Pred Value   Neg Pred Value   Precision          Recall </span></span>
<span><span class="co">##  Min.   :0.9485   Min.   :0.9904   Min.   :1      Min.   :0.9904   Min.   :1  </span></span>
<span><span class="co">##  1st Qu.:1.0000   1st Qu.:1.0000   1st Qu.:1      1st Qu.:1.0000   1st Qu.:1  </span></span>
<span><span class="co">##  Median :1.0000   Median :1.0000   Median :1      Median :1.0000   Median :1  </span></span>
<span><span class="co">##  Mean   :0.9979   Mean   :0.9996   Mean   :1      Mean   :0.9996   Mean   :1  </span></span>
<span><span class="co">##  3rd Qu.:1.0000   3rd Qu.:1.0000   3rd Qu.:1      3rd Qu.:1.0000   3rd Qu.:1  </span></span>
<span><span class="co">##  Max.   :1.0000   Max.   :1.0000   Max.   :1      Max.   :1.0000   Max.   :1  </span></span>
<span><span class="co">##                                                                               </span></span>
<span><span class="co">##        F1           Prevalence     Detection Rate   Detection Prevalence</span></span>
<span><span class="co">##  Min.   :0.9952   Min.   :0.8412   Min.   :0.8412   Min.   :0.8412      </span></span>
<span><span class="co">##  1st Qu.:1.0000   1st Qu.:0.8412   1st Qu.:0.8412   1st Qu.:0.8412      </span></span>
<span><span class="co">##  Median :1.0000   Median :0.8412   Median :0.8412   Median :0.8412      </span></span>
<span><span class="co">##  Mean   :0.9998   Mean   :0.8412   Mean   :0.8412   Mean   :0.8416      </span></span>
<span><span class="co">##  3rd Qu.:1.0000   3rd Qu.:0.8412   3rd Qu.:0.8412   3rd Qu.:0.8412      </span></span>
<span><span class="co">##  Max.   :1.0000   Max.   :0.8412   Max.   :0.8412   Max.   :0.8494      </span></span>
<span><span class="co">##                                                                         </span></span>
<span><span class="co">##  Balanced Accuracy</span></span>
<span><span class="co">##  Min.   :0.9742   </span></span>
<span><span class="co">##  1st Qu.:1.0000   </span></span>
<span><span class="co">##  Median :1.0000   </span></span>
<span><span class="co">##  Mean   :0.9990   </span></span>
<span><span class="co">##  3rd Qu.:1.0000   </span></span>
<span><span class="co">##  Max.   :1.0000   </span></span>
<span><span class="co">## </span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="differentially-methylated-genomic-regions-dmrs">Differentially methylated genomic regions (DMRs)<a class="anchor" aria-label="anchor" href="#differentially-methylated-genomic-regions-dmrs"></a>
</h3>
<p>In Methyl-IT, the concept of DMR is generalized and it is not limited
to any particular genomic region found with specific clustering
algorithm. It can be applied to any naturally or algorithmically defined
genomic region. For example, an exon region identified statistically to
be differentially methylated by using generalized linear regression
model (GLM) is a DMR. Differentially methylated genes (DMGs) are
estimated from group comparisons for the number of DMPs on gene-body
regions between control and treatment. This permits the extension of the
concept of DMR by considering a differentially methylated gene as a
particular case of DMRs.</p>
<p>Our degree of confidence in whether DMP counts in both groups of
samples, control and treatment, represent true biological signal was
determined in the signal detection step. To estimate DMRs, we followed
similar steps to those proposed in Bioconductor R package DESeq2 <span class="citation">(18)</span>, but our GLM test looks for statistical
difference between the groups based on gene-body DMP counts overlapping
a given genomic region rather than read counts. The regression analysis
of the generalized linear model (GLM) with logarithmic link was applied
to test the difference between group counts.</p>
<p>The fitting algorithmic approaches provided by ‘glm’ and ‘glm.nb’
functions from the R packages stat and MASS, respectively, were used for
Poisson (PR), Quasi-Poisson (QPR) and Negative Binomial (NBR) linear
regression analyses, respectively.</p>
<div class="section level4">
<h4 id="differentially-methylated-genes-dmgs">
<em>Differentially methylated genes (DMGs)</em><a class="anchor" aria-label="anchor" href="#differentially-methylated-genes-dmgs"></a>
</h4>
<p>We shall call DMGs those DIMRs restricted to gene-body regions. DMGs
are detected using function <a href="https://is.gd/kltCXl" class="external-link">countTest2</a>. Gene annotation is taken
from <a href="https://www.ensembl.org/" class="external-link">Ensembl</a></p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To load human gene annotation</span></span>
<span><span class="co"># AG = import(con = paste0("ftp://ftp.ensembl.org/pub/release-91/gff3/",</span></span>
<span><span class="co"># "homo_sapiens/Homo_sapiens.GRCh38.91.gff3.gz"))</span></span>
<span><span class="co"># genes = AG[ AG$type == "gene", c( "gene_id", "biotype", "Name" ) ]</span></span>
<span><span class="co"># genes = genes[ genes$biotype == "protein_coding", "gene_id" ]</span></span>
<span><span class="co"># seqlevels(genes, "coarse") &lt;- "13" # To keep a consistent chromosome </span></span>
<span><span class="co"># annotation</span></span>
<span><span class="co"># seqlevels(genes) &lt;- "chr13"</span></span></code></pre></div>
<p>Here, for the sake of brevity, we just load the “RData” compressed
file containing only the gene annotation:</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">con</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://git.psu.edu/genomath/MethylIT_examples/raw/master/"</span>,</span>
<span>                <span class="st">"Homo_sapiens_GRCh38.91_genes.RData"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">con</span><span class="op">)</span></span>
<span><span class="va">genes</span> <span class="op">=</span> <span class="va">genes</span><span class="op">[</span> <span class="va">genes</span><span class="op">$</span><span class="va">biotype</span> <span class="op">==</span> <span class="st">"protein_coding"</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">genes</span>, <span class="st">"coarse"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"13"</span> <span class="co"># To keep a consistent chromosome annotation</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"chr13"</span></span>
<span><span class="va">genes</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 323 ranges and 3 metadata columns:</span></span>
<span><span class="co">##         seqnames              ranges strand |         gene_id        biotype</span></span>
<span><span class="co">##            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;    &lt;character&gt;</span></span>
<span><span class="co">##     [1]    chr13   19173770-19181852      - | ENSG00000198033 protein_coding</span></span>
<span><span class="co">##     [2]    chr13   19422877-19536762      - | ENSG00000132958 protein_coding</span></span>
<span><span class="co">##     [3]    chr13   19633681-19673459      + | ENSG00000196199 protein_coding</span></span>
<span><span class="co">##     [4]    chr13   19674752-19783019      - | ENSG00000121390 protein_coding</span></span>
<span><span class="co">##     [5]    chr13   19823482-19863636      - | ENSG00000132950 protein_coding</span></span>
<span><span class="co">##     ...      ...                 ...    ... .             ...            ...</span></span>
<span><span class="co">##   [319]    chr13 113977783-114132611      - | ENSG00000185989 protein_coding</span></span>
<span><span class="co">##   [320]    chr13 114179238-114223084      + | ENSG00000283361 protein_coding</span></span>
<span><span class="co">##   [321]    chr13 114234887-114272723      + | ENSG00000130177 protein_coding</span></span>
<span><span class="co">##   [322]    chr13 114281584-114305817      + | ENSG00000169062 protein_coding</span></span>
<span><span class="co">##   [323]    chr13 114314513-114327328      + | ENSG00000198824 protein_coding</span></span>
<span><span class="co">##                Name</span></span>
<span><span class="co">##         &lt;character&gt;</span></span>
<span><span class="co">##     [1]      TUBA3C</span></span>
<span><span class="co">##     [2]       TPTE2</span></span>
<span><span class="co">##     [3]    MPHOSPH8</span></span>
<span><span class="co">##     [4]       PSPC1</span></span>
<span><span class="co">##     [5]       ZMYM5</span></span>
<span><span class="co">##     ...         ...</span></span>
<span><span class="co">##   [319]       RASA3</span></span>
<span><span class="co">##   [320]  AL160396.2</span></span>
<span><span class="co">##   [321]       CDC16</span></span>
<span><span class="co">##   [322]       UPF3A</span></span>
<span><span class="co">##   [323]      CHAMP1</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre>
<p>Function <a href="https://is.gd/WQzX5I" class="external-link">getDIMPatGenes</a> is used to
count the number of DMPs at gene-body. Nevertheless, it can be used for
any arbitrary specified genomic region as well. The operation of this
function is based on the ‘<em>findOverlaps</em>’ function from the
‘<em>GenomicRanges</em>’ Bioconductor R package. The
‘<em>findOverlaps</em>’ function has several critical parameters like,
for example, ‘<em>maxgap</em>’, ‘<em>minoverlap</em>’, and
‘<em>ignore.strand</em>’. In our function <a href="https://is.gd/WQzX5I" class="external-link">getDIMPatGenes</a>, except for setting
ignore.strand = TRUE and <em>type = “within”</em>, we preserve the rest
of default ‘findOverlaps’ parameters.</p>
<p>In this case, these are important parameter settings because the
local mechanical effect of methylation changes on a DNA region where a
gene is located is independent of the strand where the gene is encoded.
That is, methylation changes located in any of the two DNA strands
inside the gene-body region will affect the flexibility of the DNA
molecule <span class="citation">(7, 19)</span>.</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DMPsBN</span> <span class="op">=</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_normal</span>, GENES <span class="op">=</span> <span class="va">genes</span><span class="op">)</span></span>
<span><span class="va">DMPsBN1</span> <span class="op">=</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_normal1</span>, GENES <span class="op">=</span> <span class="va">genes</span><span class="op">)</span></span>
<span><span class="va">DMPsBC</span> <span class="op">=</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_cancer</span>, GENES <span class="op">=</span> <span class="va">genes</span><span class="op">)</span></span>
<span><span class="va">DMPsBM</span> <span class="op">=</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">DMPs</span><span class="op">$</span><span class="va">Breast_metastasis</span>, GENES <span class="op">=</span></span>
<span>        <span class="va">genes</span><span class="op">)</span></span></code></pre></div>
<p>The number of DMPs on the strand where a gene is encoded is obtained
by setting ignore.strand = FALSE. However, for the current example
results will be the same since the datasets downloaded from GEO do not
have strand information. Next, the above GRanges objects carrying the
DMP counts from each sample are grouped into a single GRanges object.
Since we have only one control, to perform group comparison and to move
forward with this example, we duplicated ‘Breast_normal’ sample.
Obviously, the confidence on the results increases with the number of
sample replications per group (in this case, it is only an illustrative
example on how to perform the analysis, since a fair comparison requires
for more than one replicate in the control group).</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Genes.DMPs</span> <span class="op">=</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">DMPsBN</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>, <span class="va">DMPsBN1</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>,</span>
<span>                                <span class="va">DMPsBC</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>, <span class="va">DMPsBM</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                            type <span class="op">=</span> <span class="st">"equal"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                            ignore.strand <span class="op">=</span> <span class="cn">TRUE</span> <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">Genes.DMPs</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>, <span class="st">"Breast_normal1"</span>,</span>
<span>                                    <span class="st">"Breast_cancer"</span>, <span class="st">"Breast_metastasis"</span><span class="op">)</span></span></code></pre></div>
<p>Next, the set of mapped genes are annotated</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GeneID</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">genes</span>, <span class="va">Genes.DMPs</span>, type <span class="op">=</span> <span class="st">"equal"</span>,</span>
<span>                            ignore.strand <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span> <span class="va">Genes.DMPs</span> <span class="op">)</span> <span class="op">&lt;-</span> <span class="va">GeneID</span><span class="op">$</span><span class="va">gene_id</span></span>
<span><span class="va">Genes.DMPs</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 127 ranges and 4 metadata columns:</span></span>
<span><span class="co">##                   seqnames              ranges strand | Breast_normal</span></span>
<span><span class="co">##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;numeric&gt;</span></span>
<span><span class="co">##   ENSG00000121390    chr13   19674752-19783019      * |             0</span></span>
<span><span class="co">##   ENSG00000165475    chr13   20403667-20525857      * |             1</span></span>
<span><span class="co">##   ENSG00000032742    chr13   20567069-20691437      * |             1</span></span>
<span><span class="co">##   ENSG00000150456    chr13   20728731-20773958      * |             0</span></span>
<span><span class="co">##   ENSG00000150457    chr13   20973032-21061547      * |             1</span></span>
<span><span class="co">##               ...      ...                 ...    ... .           ...</span></span>
<span><span class="co">##   ENSG00000283199    chr13 113953705-113973997      * |             0</span></span>
<span><span class="co">##   ENSG00000185989    chr13 113977783-114132611      * |             1</span></span>
<span><span class="co">##   ENSG00000283361    chr13 114179238-114223084      * |             2</span></span>
<span><span class="co">##   ENSG00000130177    chr13 114234887-114272723      * |             2</span></span>
<span><span class="co">##   ENSG00000169062    chr13 114281584-114305817      * |             0</span></span>
<span><span class="co">##                   Breast_normal1 Breast_cancer Breast_metastasis</span></span>
<span><span class="co">##                        &lt;numeric&gt;     &lt;numeric&gt;         &lt;numeric&gt;</span></span>
<span><span class="co">##   ENSG00000121390              0             0                 1</span></span>
<span><span class="co">##   ENSG00000165475              0             0                 1</span></span>
<span><span class="co">##   ENSG00000032742              0             2                 1</span></span>
<span><span class="co">##   ENSG00000150456              0             0                 2</span></span>
<span><span class="co">##   ENSG00000150457              0             0                 3</span></span>
<span><span class="co">##               ...            ...           ...               ...</span></span>
<span><span class="co">##   ENSG00000283199              0             2                 0</span></span>
<span><span class="co">##   ENSG00000185989              0            32                 5</span></span>
<span><span class="co">##   ENSG00000283361              0            20                 6</span></span>
<span><span class="co">##   ENSG00000130177              0            11                 3</span></span>
<span><span class="co">##   ENSG00000169062              0             6                 1</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre>
<p>Now, we build a <a href="https://is.gd/VZaazz" class="external-link">glmDataSet</a>
object.</p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## An experiment design is set.</span></span>
<span><span class="va">colData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BN"</span>,<span class="st">"BN"</span>,<span class="st">"BC"</span>,<span class="st">"BC"</span><span class="op">)</span>,</span>
<span>                                        levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BN"</span>, <span class="st">"BC"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                        <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Breast_normal"</span>,<span class="st">"Breast_normal1"</span>,</span>
<span>                            <span class="st">"Breast_cancer"</span>,<span class="st">"Breast_metastasis"</span><span class="op">)</span>,</span>
<span>                        row.names <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">## A RangedGlmDataSet is created</span></span>
<span><span class="va">ds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">Genes.DMPs</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></span></code></pre></div>
<p>DMG analysis is performed with the function <a href="https://%20is.gd/kltCXl" class="external-link">countTest2</a></p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DMGs</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds</span>, num.cores <span class="op">=</span> <span class="fl">4L</span>, minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,</span>
<span>                countFilter <span class="op">=</span> <span class="cn">TRUE</span>, maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>                Minlog2FC <span class="op">=</span> <span class="fl">1</span>, pvalCutOff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                MVrate <span class="op">=</span> <span class="fl">.95</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">DMGs</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 3 ranges and 12 metadata columns:</span></span>
<span><span class="co">##                   seqnames              ranges strand | Breast_normal</span></span>
<span><span class="co">##                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;numeric&gt;</span></span>
<span><span class="co">##   ENSG00000088387    chr13   98793429-99086625      * |             2</span></span>
<span><span class="co">##   ENSG00000185989    chr13 113977783-114132611      * |             1</span></span>
<span><span class="co">##   ENSG00000283361    chr13 114179238-114223084      * |             2</span></span>
<span><span class="co">##                   Breast_normal1 Breast_cancer Breast_metastasis    log2FC</span></span>
<span><span class="co">##                        &lt;numeric&gt;     &lt;numeric&gt;         &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##   ENSG00000088387              0            33                 7   2.35138</span></span>
<span><span class="co">##   ENSG00000185989              0            32                 5   2.56495</span></span>
<span><span class="co">##   ENSG00000283361              0            20                 6   1.94591</span></span>
<span><span class="co">##                   scaled.deviance     pvalue          model   adj.pval</span></span>
<span><span class="co">##                         &lt;numeric&gt;  &lt;numeric&gt;    &lt;character&gt;  &lt;numeric&gt;</span></span>
<span><span class="co">##   ENSG00000088387         10.2110 0.00192563   Neg.Binomial 0.00332571</span></span>
<span><span class="co">##   ENSG00000185989          9.4506 0.00258248 Neg.Binomial.W 0.00332571</span></span>
<span><span class="co">##   ENSG00000283361         10.0206 0.00332571   Neg.Binomial 0.00332571</span></span>
<span><span class="co">##                   CT.SignalDensity TT.SignalDensity SignalDensityVariation</span></span>
<span><span class="co">##                          &lt;numeric&gt;        &lt;numeric&gt;              &lt;numeric&gt;</span></span>
<span><span class="co">##   ENSG00000088387      3.41068e-06      6.82135e-05            6.48028e-05</span></span>
<span><span class="co">##   ENSG00000185989      3.22937e-06      1.19487e-04            1.16257e-04</span></span>
<span><span class="co">##   ENSG00000283361      2.28066e-05      2.96486e-04            2.73679e-04</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre>
<p>Here, we adds the gene alias:</p>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hits</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a></span><span class="op">(</span><span class="va">DMGs</span>, <span class="va">genes</span>, type <span class="op">=</span> <span class="st">"equal"</span><span class="op">)</span></span>
<span><span class="va">DMGs</span> <span class="op">=</span> <span class="va">DMGs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">queryHits</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">DMGs</span><span class="op">$</span><span class="va">alias</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">$</span><span class="va">Name</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">subjectHits</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<p><em><strong>BRCA2, a breast cancer associated risk gene, is found
between the DMG</strong>s</em></p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># DMGs</span></span>
<span><span class="va">DMGs</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span> <span class="st">"ENSG00000139618"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">DMGs</span><span class="op">)</span> <span class="op">)</span> <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 0 ranges and 13 metadata columns:</span></span>
<span><span class="co">##    seqnames    ranges strand | Breast_normal Breast_normal1 Breast_cancer</span></span>
<span><span class="co">##       &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;numeric&gt;      &lt;numeric&gt;     &lt;numeric&gt;</span></span>
<span><span class="co">##    Breast_metastasis    log2FC scaled.deviance    pvalue       model  adj.pval</span></span>
<span><span class="co">##            &lt;numeric&gt; &lt;numeric&gt;       &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##    CT.SignalDensity TT.SignalDensity SignalDensityVariation       alias</span></span>
<span><span class="co">##           &lt;numeric&gt;        &lt;numeric&gt;              &lt;numeric&gt; &lt;character&gt;</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="concluding-remarks">Concluding remarks<a class="anchor" aria-label="anchor" href="#concluding-remarks"></a>
</h3>
<p>Results fundamentally depend on the use of the proper reference
samples. If there is not enough control samples to create an independent
reference virtual individual, then the centroid from the control group
can be used instead, which can be built with function <a href="https://is.gd/qR79kW" class="external-link">poolFromGRlist</a>.</p>
<p>The <em>reference</em> is used to put individuals from control and
patients in the same <em>system of reference</em>. The information
divergences of methylation levels with respect to the reference
individual are estimated for each cytosine site for each individual,
from control and treatment. As a result, the information on the natural
spontaneous variation in the control and treatment populations is
carried in the probability distributions of the information divergences,
estimated for each individual. <em>It does not matter how
statistically</em> <em>significant a DMP would be in the group of
patients</em>.</p>
<p><em>What really matter is how big is the probability to observe the
same methylation event in the control group in respect to the patient
group</em>. Decisions are ultimately made on the basis of such
classification probabilities at single cytosine positions. The signal
detection based approach implemented in Methyl-IT is addressed to
confront the mentioned issue and, in consequence, the level of
resolution of Methyl-IT approach reach the single cytosine site.</p>
</div>
<div class="section level3">
<h3 id="acknowledgments">Acknowledgments<a class="anchor" aria-label="anchor" href="#acknowledgments"></a>
</h3>
<p>We thank Professor David J Miller for valuable conversations and
suggestions on our mathematical modeling.</p>
</div>
<div class="section level3">
<h3 id="funding-the-work-was-supported-by-funding-from-nsf-sbir">Funding The work was supported by funding from NSF-SBIR<a class="anchor" aria-label="anchor" href="#funding-the-work-was-supported-by-funding-from-nsf-sbir"></a>
</h3>
<p>(2015-33610-23428-UNL) and the Bill and Melinda Gates Foundation
(OPP1088661).</p>
</div>
<div class="section level3">
<h3 id="references-">References.<a class="anchor" aria-label="anchor" href="#references-"></a>
</h3>
<div id="refs" class="references csl-bib-body">
<div id="ref-Sanchez2016" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">R.
Sanchez, S. A. Mackenzie, <a href="https://doi.org/10.1371/journal.pone.0150427" class="external-link"><span class="nocase">Information Thermodynamics of Cytosine DNA
Methylation</span></a>. <em>PLOS ONE</em> <strong>11</strong>, e0150427
(2016).</div>
</div>
<div id="ref-Sanchez2019" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">R.
Sanchez, X. Yang, T. Maher, S. Mackenzie, <a href="https://doi.org/10.3390/ijms20215343" class="external-link"><span class="nocase">Discrimination of DNA Methylation Signal from Background
Variation for Clinical Diagnostics</span></a>. <em>Int. J. Mol.
Sci.</em> <strong>20</strong>, 5343 (2019).</div>
</div>
<div id="ref-Greiner2000" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">M.
Greiner, D. Pfeiffer, R. D. Smith, <a href="https://doi.org/10.1016/S0167-5877(00)00115-X" class="external-link"><span class="nocase">Principles and practical application of the
receiver-operating characteristic analysis for diagnostic
tests</span></a>. <em>Preventive Veterinary Medicine</em>
<strong>45</strong>, 23–41 (2000).</div>
</div>
<div id="ref-Carter2016" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">J.
V. Carter, J. Pan, S. N. Rai, S. Galandiuk, <a href="https://doi.org/10.1016/j.surg.2015.12.029" class="external-link"><span class="nocase">ROC-ing along: Evaluation and interpretation of receiver
operating characteristic curves</span></a>. <em>Surgery</em>
<strong>159</strong>, 1638–1645 (2016).</div>
</div>
<div id="ref-Harpaz2013" class="csl-entry">
<div class="csl-left-margin">5. </div>
<div class="csl-right-inline">R.
Harpaz, <em>et al.</em>, <a href="https://doi.org/10.1038/clpt.2013.24.Performance" class="external-link"><span class="nocase">Performance of Pharmacovigilance Signal Detection
Algorithms for the FDA Adverse Event Reporting System</span></a>.
<em>Clin Pharmacol ther</em> <strong>93</strong>, 1–19 (2013).</div>
</div>
<div id="ref-Kruspe2017" class="csl-entry">
<div class="csl-left-margin">6. </div>
<div class="csl-right-inline">S.
Kruspe, <em>et al.</em>, <a href="https://doi.org/10.1016/j.omtn.2017.08.004" class="external-link"><span class="nocase">Rapid and Sensitive Detection of Breast Cancer Cells in
Patient Blood with Nuclease-Activated Probe Technology</span></a>.
<em>Molecular Therapy - Nucleic Acids</em> <strong>8</strong>, 542–557
(2017).</div>
</div>
<div id="ref-Severin2011" class="csl-entry">
<div class="csl-left-margin">7. </div>
<div class="csl-right-inline">P.
M. D. Severin, X. Zou, H. E. Gaub, K. Schulten, <a href="https://doi.org/10.1093/nar/gkr578" class="external-link"><span class="nocase">Cytosine
methylation alters DNA mechanical properties</span></a>. <em>Nucleic
acids research</em> <strong>39</strong>, 8740–51 (2011).</div>
</div>
<div id="ref-Ngo2016" class="csl-entry">
<div class="csl-left-margin">8. </div>
<div class="csl-right-inline">T.
T. M. Ngo, <em>et al.</em>, <a href="https://doi.org/10.1038/ncomms10813" class="external-link"><span class="nocase">Effects
of cytosine modifications on DNA flexibility and nucleosome mechanical
stability</span></a>. <em>Nature Communications</em> <strong>7</strong>,
10813 (2016).</div>
</div>
<div id="ref-Basu2010" class="csl-entry">
<div class="csl-left-margin">9. </div>
<div class="csl-right-inline">A.
Basu, A. Mandal, L. Pardo, <a href="https://doi.org/10.1016/j.spl.2009.10.008" class="external-link"><span class="nocase">Hypothesis testing for two discrete populations based on
the Hellinger distance</span></a>. <em>Statistics <span>&amp;</span>
Probability Letters</em> <strong>80</strong>, 206–214 (2010).</div>
</div>
<div id="ref-Hebestreit2013" class="csl-entry">
<div class="csl-left-margin">10. </div>
<div class="csl-right-inline">K.
Hebestreit, M. Dugas, H.-U. Klein, <a href="https://doi.org/10.1093/bioinformatics/btt263" class="external-link"><span class="nocase">Detection of significantly differentially methylated
regions in targeted bisulfite sequencing data.</span></a>
<em>Bioinformatics (Oxford, England)</em> <strong>29</strong>, 1647–53
(2013).</div>
</div>
<div id="ref-Robinson2014" class="csl-entry">
<div class="csl-left-margin">11. </div>
<div class="csl-right-inline">M.
D. Robinson, <em>et al.</em>, <a href="https://doi.org/10.3389/fgene.2014.00324" class="external-link"><span class="nocase">Statistical methods for detecting differentially
methylated loci and regions.</span></a> <em>Frontiers in genetics</em>
<strong>5</strong>, 324 (2014).</div>
</div>
<div id="ref-Dolzhenko2014" class="csl-entry">
<div class="csl-left-margin">12. </div>
<div class="csl-right-inline">E.
Dolzhenko, A. D. Smith, <a href="https://doi.org/10.1186/1471-2105-15-215" class="external-link"><span class="nocase">Using beta-binomial regression for high-precision
differential methylation analysis in multifactor whole-genome bisulfite
sequencing experiments</span></a>. <em>BMC bioinformatics</em>
<strong>15</strong>, 215 (2014).</div>
</div>
<div id="ref-Baldi2001" class="csl-entry">
<div class="csl-left-margin">13. </div>
<div class="csl-right-inline">P.
Baldi, S. Brunak, <em><span class="nocase">Bioinformatics: the machine
learning approach</span></em>, Second (MIT Press, 2001).</div>
</div>
<div id="ref-Stevens2009" class="csl-entry">
<div class="csl-left-margin">14. </div>
<div class="csl-right-inline">J.
P. Stevens, <em><span class="nocase">Applied Multivariate Statistics for
the Social Sciences</span></em>, Fifth Edit (Routledge Academic,
2009).</div>
</div>
<div id="ref-lopez2014" class="csl-entry">
<div class="csl-left-margin">15. </div>
<div class="csl-right-inline">M.
López-Ratón, M. X. Rodríguez-Álvarez, C. Cadarso-Suárez, F.
Gude-Sampedro, Others, <a href="https://www.jstatsoft.org/article/view/v061i08" class="external-link"><span class="nocase">OptimalCutpoints: an R package for selecting optimal
cutpoints in diagnostic tests</span></a>. <em>Journal of statistical
software</em> <strong>61</strong>, 1–36 (2014).</div>
</div>
<div id="ref-Fawcett2005" class="csl-entry">
<div class="csl-left-margin">16. </div>
<div class="csl-right-inline">T.
Fawcett, <span class="nocase">An introduction to ROC analysis</span>
(2005) https:/doi.org/<a href="https://doi.org/10.1016/j.patrec.2005.10.010" class="external-link">10.1016/j.patrec.2005.10.010</a>.</div>
</div>
<div id="ref-Youden1950" class="csl-entry">
<div class="csl-left-margin">17. </div>
<div class="csl-right-inline">W.
J. Youden, <a href="https://doi.org/10.1002/1097-0142(1950)3:1&lt;32::AID-CNCR2820030106&gt;3.0.CO;2-3" class="external-link"><span class="nocase">Index for rating diagnostic tests</span></a>.
<em>Cancer</em> <strong>3</strong>, 32–35 (1950).</div>
</div>
<div id="ref-Love2014" class="csl-entry">
<div class="csl-left-margin">18. </div>
<div class="csl-right-inline">M.
I. Love, W. Huber, S. Anders, <a href="https://doi.org/Artn%20550%5CrDoi%2010.1186/S13059-014-0550-8" class="external-link"><span class="nocase">Moderated estimation of fold change and dispersion for
RNA-seq data with DESeq2</span></a>. <em>Genome Biology</em>
<strong>15</strong>, 1–34 (2014).</div>
</div>
<div id="ref-Choy2010" class="csl-entry">
<div class="csl-left-margin">19. </div>
<div class="csl-right-inline">J.
S. Choy, <em>et al.</em>, <a href="https://doi.org/10.1021/ja910264z" class="external-link"><span class="nocase">DNA
methylation increases nucleosome compaction and rigidity</span></a>.
<em>Journal of the American Chemical Society</em> <strong>132</strong>,
1782–1783 (2010).</div>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code> on the system on
which this document was compiled running pandoc 2.19.2:</p>
<pre><code><span><span class="co">## R version 4.2.3 (2023-03-15)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] grid      stats4    stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [8] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] VennDiagram_1.7.3    futile.logger_1.4.3  gridExtra_2.3       </span></span>
<span><span class="co">##  [4] reshape2_1.4.4       ggplot2_3.4.1        BiocParallel_1.32.5 </span></span>
<span><span class="co">##  [7] MethylIT_0.3.2.6.3   rtracklayer_1.58.0   GenomicRanges_1.50.2</span></span>
<span><span class="co">## [10] GenomeInfoDb_1.34.9  IRanges_2.32.0       S4Vectors_0.36.1    </span></span>
<span><span class="co">## [13] BiocGenerics_0.44.0  knitr_1.42          </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] colorspace_2.1-0            rjson_0.2.21               </span></span>
<span><span class="co">##   [3] class_7.3-21                rprojroot_2.0.3            </span></span>
<span><span class="co">##   [5] XVector_0.38.0              fs_1.6.1                   </span></span>
<span><span class="co">##   [7] proxy_0.4-27                rstudioapi_0.14            </span></span>
<span><span class="co">##   [9] farver_2.1.1                listenv_0.9.0              </span></span>
<span><span class="co">##  [11] bit64_4.0.5                 AnnotationDbi_1.60.0       </span></span>
<span><span class="co">##  [13] prodlim_2019.11.13          fansi_1.0.4                </span></span>
<span><span class="co">##  [15] lubridate_1.9.2             codetools_0.2-19           </span></span>
<span><span class="co">##  [17] splines_4.2.3               cachem_1.0.7               </span></span>
<span><span class="co">##  [19] jsonlite_1.8.4              pROC_1.18.0                </span></span>
<span><span class="co">##  [21] Rsamtools_2.14.0            caret_6.0-93               </span></span>
<span><span class="co">##  [23] annotate_1.76.0             png_0.1-8                  </span></span>
<span><span class="co">##  [25] compiler_4.2.3              httr_1.4.5                 </span></span>
<span><span class="co">##  [27] Matrix_1.5-3                fastmap_1.1.1              </span></span>
<span><span class="co">##  [29] cli_3.6.0                   formatR_1.14               </span></span>
<span><span class="co">##  [31] htmltools_0.5.4             tools_4.2.3                </span></span>
<span><span class="co">##  [33] gtable_0.3.2                glue_1.6.2                 </span></span>
<span><span class="co">##  [35] GenomeInfoDbData_1.2.9      dplyr_1.1.0                </span></span>
<span><span class="co">##  [37] Rcpp_1.0.10                 Biobase_2.58.0             </span></span>
<span><span class="co">##  [39] jquerylib_0.1.4             pkgdown_2.0.7              </span></span>
<span><span class="co">##  [41] vctrs_0.6.0                 Biostrings_2.66.0          </span></span>
<span><span class="co">##  [43] nlme_3.1-162                iterators_1.0.14           </span></span>
<span><span class="co">##  [45] timeDate_4022.108           gower_1.0.1                </span></span>
<span><span class="co">##  [47] xfun_0.37                   stringr_1.5.0              </span></span>
<span><span class="co">##  [49] proto_1.0.0                 globals_0.16.2             </span></span>
<span><span class="co">##  [51] timechange_0.2.0            lifecycle_1.0.3            </span></span>
<span><span class="co">##  [53] restfulr_0.0.15             XML_3.99-0.13              </span></span>
<span><span class="co">##  [55] future_1.31.0               zlibbioc_1.44.0            </span></span>
<span><span class="co">##  [57] MASS_7.3-58.2               scales_1.2.1               </span></span>
<span><span class="co">##  [59] ipred_0.9-13                ragg_1.2.5                 </span></span>
<span><span class="co">##  [61] MatrixGenerics_1.10.0       parallel_4.2.3             </span></span>
<span><span class="co">##  [63] SummarizedExperiment_1.28.0 lambda.r_1.2.4             </span></span>
<span><span class="co">##  [65] yaml_2.3.7                  memoise_2.0.1              </span></span>
<span><span class="co">##  [67] sass_0.4.5                  rpart_4.1.19               </span></span>
<span><span class="co">##  [69] stringi_1.7.12              RSQLite_2.2.20             </span></span>
<span><span class="co">##  [71] highr_0.10                  genefilter_1.80.3          </span></span>
<span><span class="co">##  [73] BiocIO_1.8.0                desc_1.4.2                 </span></span>
<span><span class="co">##  [75] randomForest_4.7-1.1        foreach_1.5.2              </span></span>
<span><span class="co">##  [77] e1071_1.7-13                hardhat_1.2.0              </span></span>
<span><span class="co">##  [79] lava_1.7.1                  rlang_1.1.0                </span></span>
<span><span class="co">##  [81] pkgconfig_2.0.3             systemfonts_1.0.4          </span></span>
<span><span class="co">##  [83] matrixStats_0.63.0          bitops_1.0-7               </span></span>
<span><span class="co">##  [85] evaluate_0.20               lattice_0.20-45            </span></span>
<span><span class="co">##  [87] purrr_1.0.1                 labeling_0.4.2             </span></span>
<span><span class="co">##  [89] GenomicAlignments_1.34.0    recipes_1.0.4              </span></span>
<span><span class="co">##  [91] bit_4.0.5                   tidyselect_1.2.0           </span></span>
<span><span class="co">##  [93] parallelly_1.34.0           nls2_0.3-3                 </span></span>
<span><span class="co">##  [95] plyr_1.8.8                  magrittr_2.0.3             </span></span>
<span><span class="co">##  [97] R6_2.5.1                    generics_0.1.3             </span></span>
<span><span class="co">##  [99] DelayedArray_0.24.0         DBI_1.1.3                  </span></span>
<span><span class="co">## [101] pillar_1.8.1                withr_2.5.0                </span></span>
<span><span class="co">## [103] KEGGREST_1.38.0             survival_3.5-5             </span></span>
<span><span class="co">## [105] RCurl_1.98-1.10             nnet_7.3-18                </span></span>
<span><span class="co">## [107] tibble_3.2.0                future.apply_1.10.0        </span></span>
<span><span class="co">## [109] crayon_1.5.2                futile.options_1.0.1       </span></span>
<span><span class="co">## [111] utf8_1.2.3                  rmarkdown_2.20             </span></span>
<span><span class="co">## [113] minpack.lm_1.2-3            data.table_1.14.6          </span></span>
<span><span class="co">## [115] blob_1.2.4                  ModelMetrics_1.2.2.2       </span></span>
<span><span class="co">## [117] digest_0.6.31               xtable_1.8-4               </span></span>
<span><span class="co">## [119] textshaping_0.3.6           munsell_0.5.0              </span></span>
<span><span class="co">## [121] bslib_0.4.2</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
