<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Optimal cutpoint for the methylation signal • MethylIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Optimal cutpoint for the methylation signal">
<meta property="og:description" content="MethylIT">
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Optimal cutpoint for the methylation signal</h1>
            <h3 data-toc-skip class="subtitle">Detection of the
methylation signal with Methyl-IT</h3>
                        <h4 data-toc-skip class="author"><div class="line-block">Robersy Sanchez<br><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a><br>
Mackenzie’s lab<br><br>
Department of Biology.<br>
Pennsylvania State University, University Park, PA 16802</div></h4>
            
            <h4 data-toc-skip class="date">08 February 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/vignettes/Cutpoint_estimation_with_Methyl-IT.Rmd" class="external-link"><code>vignettes/Cutpoint_estimation_with_Methyl-IT.Rmd</code></a></small>
      <div class="hidden name"><code>Cutpoint_estimation_with_Methyl-IT.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The discrimination of the methylation signal from the
      stochastic methylation background resultant from the standard
      (non-stressful) biological processes is a critical step for the
      genome-wide methylation analysis. Such a discrimination requires
      for the knowledge of the probability distribution of the
      information divergence of methylation levels and a proper
      evaluation of the classification performance of differentially
      methylated positions (DMPs) into two classes: DMPs from control
      and DMPs from treatment.</p>
    </div>
    
<style type="text/css">

.list-group-item.active, .list-group-item.active:focus, 
.list-group-item.active:hover {
    z-index: 2;
    color: #fff;
    background-color: #337ab7;
    border-color: #337ab7;
}

.tocify-subheader .tocify-item {
    /* font-size: 0.80em; */ 
    padding-left: 25px;
    text-indent: 0;
}

.tocify-header .tocify-item {
    /* font-size: 0.85em; */ 
    padding-left: 20px;
    text-indent: 0;
}

body{ /* Normal  */
    /* font-size: 18px; */
    font-family: "Times New Roman", Times, serif;
    text-align: justify
}
td {  /* Table  */
    /* font-size: 8px; */
}

h1.title {
    /* font-size: 38px; */
    font-family: "Times New Roman", Times, serif;
    color: DarkRed;
    .subTitle {
        /* font-size: 24px; */
        font-family: "Times New Roman", Times, serif;
        color: DarkRed;
    }
}

h1 { /* Header 1 */
    /* font-size: 28px; */
    font-family: "Times New Roman", Times, serif;
    color: DarkBlue;
}
h2 { /* Header 2 */
    /* font-size: 22px; */
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
h3 { /* Header 3 */
    /* font-size: 18px; */
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
code.r{ /* Code block */
    font-size: 14px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>
<div class="section level3">
<h3 id="motivation">Motivation<a class="anchor" aria-label="anchor" href="#motivation"></a>
</h3>
<p>The probability of extreme methylation changes occurring
spontaneously in a control population by the stochastic fluctuations
inherent to biochemical processes and DNA maintenance <span class="citation">(1)</span>, requires the discrimination of this
background variation from a biological treatment signal. The
stochasticity of the the methylation process derives from the
stochasticity inherent to biochemical processes <span class="citation">(2, 3)</span>. There are fundamental physical reasons
to acknowledge that biochemical processes are subject to noise and
fluctuations <span class="citation">(4, 5)</span>. So, regardless
constant environment, statistically significant methylation changes can
be found in control population with probability greater than zero and
proportional to a Boltzmann factor <span class="citation">(6)</span>.</p>
<p>Natural signals and those generated by human technology are not free
of noise and, as mentioned above, the methylation signal is no
exception. Only signal detection based approaches are designed to filter
out the signal from the noise , in natural and in human generated
signals.</p>
<p>The stochasticity of methylation regulatory machinery effects is
presumed to reflect system heterogeneity; cells from the same tissue are
not necessarily in the same state, and therefore, corresponding cytosine
sites differ in their methylation status. Consequently, overall
organismal response is conveyed as a statistical outcome that
distinguishes the regulatory methylation signal from statistical
background “noise”. Estimation of optimal cutoff value for the signal is
an additional step to remove any remaining potential methylation
background noise with probability <span class="math inline">\(0 ≤ \alpha
≤ 0.05\)</span>. We define as a methylation signal (DMP) each cytosine
site with Hellinger Divergence values above the cutpoint (shown <span class="citation">(7)</span>.</p>
<p>As a result, a differentially methylated position (DMP) as a cytosine
position with high probability to be altered in methylation due to a
treatment effect, distinct from spontaneous variation detected in the
control population.</p>
<p>Note: This example was made with the MethylIT version 0.3.2 available
at <a href="https://github.com/genomaths/MethylIT" class="external-link uri">https://github.com/genomaths/MethylIT</a>.</p>
</div>
<div class="section level3">
<h3 id="data-generation">Data generation<a class="anchor" aria-label="anchor" href="#data-generation"></a>
</h3>
<p>For the current example on methylation analysis with Methyl-IT we
will use simulated data. Read-count matrices of methylated and
unmethylated cytosine are generated with MethylIT.utils function
<em>simulateCounts</em>. A standard analysis of this dataset is given in
the web page: <a href="https://genomaths.github.io/Methylation_analysis_with_Methyl-IT.html" class="external-link">Methylation
analysis with Methyl-IT</a></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/MethylIT" class="external-link">MethylIT</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MethylIT.utils</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Bioconductor/BiocParallel" class="external-link">BiocParallel</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">bmean</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">alpha</span>, <span class="va">beta</span><span class="op">)</span> <span class="va">alpha</span><span class="op">/</span><span class="op">(</span><span class="va">alpha</span> <span class="op">+</span> <span class="va">beta</span><span class="op">)</span></span>
<span><span class="va">alpha.ct</span> <span class="op">&lt;-</span> <span class="fl">0.09</span></span>
<span><span class="va">alpha.tt</span> <span class="op">&lt;-</span> <span class="fl">0.2</span></span>
<span></span>
<span><span class="co"># The number of cytosine sites to generate</span></span>
<span><span class="va">sites</span> <span class="op">=</span> <span class="fl">50000</span> </span>
<span><span class="co"># Set a seed for pseudo-random number generation</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">124</span><span class="op">)</span></span>
<span><span class="va">control.nam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"C1"</span>, <span class="st">"C2"</span>, <span class="st">"C3"</span><span class="op">)</span></span>
<span><span class="va">treatment.nam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T1"</span>, <span class="st">"T2"</span>, <span class="st">"T3"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Reference group </span></span>
<span><span class="va">ref0</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html" class="external-link">simulateCounts</a></span><span class="op">(</span>num.samples <span class="op">=</span> <span class="fl">4</span>, sites <span class="op">=</span> <span class="va">sites</span>, alpha <span class="op">=</span> <span class="va">alpha.ct</span>,</span>
<span>                    beta <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">50</span>, theta <span class="op">=</span> <span class="fl">4.5</span>, </span>
<span>                    sample.ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"R1"</span>, <span class="st">"R2"</span>, <span class="st">"R3"</span>, <span class="st">"R4"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Control group</span></span>
<span><span class="va">ctrl</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html" class="external-link">simulateCounts</a></span><span class="op">(</span>num.samples <span class="op">=</span> <span class="fl">3</span>, sites <span class="op">=</span> <span class="va">sites</span>, alpha <span class="op">=</span> <span class="va">alpha.ct</span>,</span>
<span>                    beta <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">50</span>, theta <span class="op">=</span> <span class="fl">4.5</span>, </span>
<span>                    sample.ids <span class="op">=</span> <span class="va">control.nam</span><span class="op">)</span></span>
<span><span class="co"># Treatment group</span></span>
<span><span class="va">treat</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html" class="external-link">simulateCounts</a></span><span class="op">(</span>num.samples <span class="op">=</span> <span class="fl">3</span>, sites <span class="op">=</span> <span class="va">sites</span>, alpha <span class="op">=</span> <span class="va">alpha.tt</span>,</span>
<span>                    beta <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">50</span>, theta <span class="op">=</span> <span class="fl">4.5</span>, </span>
<span>                    sample.ids <span class="op">=</span> <span class="va">treatment.nam</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Reference sample</span></span>
<span><span class="va">ref</span> <span class="op">=</span> <span class="fu"><a href="../reference/poolFromGRlist.html">poolFromGRlist</a></span><span class="op">(</span><span class="va">ref0</span>, stat <span class="op">=</span> <span class="st">"mean"</span>, num.cores <span class="op">=</span> <span class="fl">4L</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="methylation-level-divergences">Methylation level divergences<a class="anchor" aria-label="anchor" href="#methylation-level-divergences"></a>
</h3>
<p>Total variation distance and Hellinger divergence are computed with
<a href="https://is.gd/0ViBO9" class="external-link">estimateDivergence</a> function. The
setting and.min.cov = TRUE in <a href="https://is.gd/0ViBO9" class="external-link">estimateDivergence</a> will accomplish
pairwise deletion of missing cytosine sites or sites no holding the
filtering condition.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">divs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/estimateDivergence.html">estimateDivergence</a></span><span class="op">(</span>ref <span class="op">=</span> <span class="va">ref</span>, </span>
<span>                           indiv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">ctrl</span>, <span class="va">treat</span><span class="op">)</span>, </span>
<span>                           Bayesian <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                           and.min.cov <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                           num.cores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html" class="external-link">multicoreWorkers</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>                           percentile <span class="op">=</span> <span class="fl">1</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>To get some statistical description about the sample is useful. Here,
empirical critical values for the probability distribution of <span class="math inline">\(H\)</span> and <span class="math inline">\(TV\)</span> can is obtained using
<em>quantile</em> function from the R package <em>stats</em>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">critical.val</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">divs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">x</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="va">x</span><span class="op">$</span><span class="va">hdiv</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span></span>
<span>    <span class="va">hd.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">hdiv</span>, <span class="fl">0.95</span><span class="op">)</span></span>
<span>    <span class="va">tv.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">TV</span><span class="op">)</span>, <span class="fl">0.95</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>tv <span class="op">=</span> <span class="va">tv.95</span>, hd <span class="op">=</span> <span class="va">hd.95</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">critical.val</span></span></code></pre></div>
<pre><code><span><span class="co">##       tv.95%    hd.95%</span></span>
<span><span class="co">## C1 0.6842105  66.76003</span></span>
<span><span class="co">## C2 0.6800000  66.71971</span></span>
<span><span class="co">## C3 0.6777456  65.98486</span></span>
<span><span class="co">## T1 0.9397681 138.68200</span></span>
<span><span class="co">## T2 0.9478351 141.72617</span></span>
<span><span class="co">## T3 0.9466565 141.77236</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="estimation-of-potential-dmps-with-methyl-it">Estimation of potential DMPs with Methyl-IT<a class="anchor" aria-label="anchor" href="#estimation-of-potential-dmps-with-methyl-it"></a>
</h3>
<p>In Methyl-IT, DMP estimation requires for the knowledge of the
probability distribution of the noise (plus signal), which is used as
null hypothesis.</p>
<p>The best fitted distribution model can be estimated with function <a href="https://genomaths.github.io/methylit/reference/gofReport.html">gofReport</a>.
Here, for illustration purposes, we will use specific estimations based
on 2- and 3-parameter gamma distribution models directly using function
<a href="https://is.gd/0BLLkT" class="external-link">nonlinearFitDist</a>.</p>
<div class="section level4">
<h4 id="potential-dmps-estimated-with-2-parameter-gamma-distribution-model">Potential DMPs estimated with 2-parameter gamma distribution
model<a class="anchor" aria-label="anchor" href="#potential-dmps-estimated-with-2-parameter-gamma-distribution-model"></a>
</h4>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nlms.g2p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/nonlinearFitDist.html">nonlinearFitDist</a></span><span class="op">(</span><span class="va">divs</span>, column <span class="op">=</span> <span class="fl">9L</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                            num.cores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html" class="external-link">multicoreWorkers</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                            dist.name <span class="op">=</span> <span class="st">"Gamma2P"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Potential DMPs from 'Gamma2P' model</span></span>
<span><span class="va">pDMPs.g2p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">divs</span>, nlms <span class="op">=</span> <span class="va">nlms.g2p</span>,  div.col <span class="op">=</span> <span class="fl">9L</span>, </span>
<span>                            tv.cut <span class="op">=</span> <span class="fl">0.68</span>, tv.col <span class="op">=</span> <span class="fl">7</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, </span>
<span>                            dist.name <span class="op">=</span> <span class="st">"Gamma2P"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="potential-dmps-estimated-with-3-parameter-gamma-distribution-model">Potential DMPs estimated with 3-parameter gamma distribution
model<a class="anchor" aria-label="anchor" href="#potential-dmps-estimated-with-3-parameter-gamma-distribution-model"></a>
</h4>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nlms.g3p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/nonlinearFitDist.html">nonlinearFitDist</a></span><span class="op">(</span><span class="va">divs</span>, column <span class="op">=</span> <span class="fl">9L</span>, </span>
<span>                            verbose <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                            num.cores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html" class="external-link">multicoreWorkers</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                            dist.name <span class="op">=</span> <span class="st">"Gamma3P"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Potential DMPs from 'Gamma2P' model</span></span>
<span><span class="va">pDMPs.g3p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">divs</span>, nlms <span class="op">=</span> <span class="va">nlms.g3p</span>,  div.col <span class="op">=</span> <span class="fl">9L</span>, </span>
<span>                            tv.cut <span class="op">=</span> <span class="fl">0.68</span>, tv.col <span class="op">=</span> <span class="fl">7</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, </span>
<span>                            dist.name <span class="op">=</span> <span class="st">"Gamma3P"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="cutpoint-estimation">Cutpoint estimation<a class="anchor" aria-label="anchor" href="#cutpoint-estimation"></a>
</h3>
<p>As a result of the natural spontaneous variation, naturally occurring
DMPs can be identified in samples from the control and treatment
populations. Machine-learning algorithms implemented in Methyl-IT are
applied to discriminate treatment-induced DMPs from those naturally
generated.</p>
<p>The simple cutpoint estimation available in Methyl-IT is based on the
application of Youden index <span class="citation">(8)</span>. Although
cutpoints are estimated for a single variable, the classification
performance can be evaluated for several variables and applying
different model classifiers.</p>
<p>In the current example, the column carrying <em>TV</em>
(<em>div.col</em> = 7L) will be used to estimate the cutpoint. The
column values will be expressed in terms of <span class="math inline">\(TV_d=|p_{tt}-p_{ct}|\)</span>. A minimum cutpoint
value for <em>TV</em> derived from the minimum 95% quantile
(<em>tv.cut</em> = 0.92) found in the treatment group will be applied
(see <a href="https://is.gd/gHrnCp" class="external-link">Methylation analysis with
Methyl-IT</a>).</p>
<p>Next, a logistic model classifier will be fitted with the 60%
(<em>prop</em> = 0.6) of the raw data (training set) and then the
resting 40% of individual samples will be used to evaluate the model
performance. The predictor variable included in the model are specified
with function parameter <em>column</em> (for more detail see <a href="https://is.gd/RR9iWY" class="external-link"><em>estimateCutPoint</em></a> or type
?estimateCutPoint in R console).</p>
<div class="section level4">
<h4 id="simple-cutpoint-estimation-for-gamma2p-model">Simple cutpoint estimation for Gamma2P model<a class="anchor" aria-label="anchor" href="#simple-cutpoint-estimation-for-gamma2p-model"></a>
</h4>
<p>Here, we use the results of modeling the distribution of the
Hellinger divergence (<em>HD</em>) of methylation levels through a
2-parameter gamma probability distribution model. The critical values
for <span class="math inline">\(HD_{\alpha = 0.05}^{CT_{G2P}}\)</span>
used to get potential DMPs were:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">nlms.g2p</span><span class="op">)</span></span>
<span><span class="va">crit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">nlms.g2p</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/stats/GammaDist.html" class="external-link">qgamma</a></span><span class="op">(</span><span class="fl">0.95</span>, shape <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Estimate</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>                                                    scale <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Estimate</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">crit</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">nams</span></span>
<span><span class="va">crit</span></span></code></pre></div>
<pre><code><span><span class="co">##        C1        C2        C3        T1        T2        T3 </span></span>
<span><span class="co">##  58.59151  57.99945  57.80987 112.39982 113.92337 114.48812</span></span></code></pre>
<p>As before the cutpoint is estimated based on ‘Youden Index’ <span class="citation">(8)</span>. A PCA+LDA model classifier
(<em>classifier</em> = “pca.lda”) is applied. That is, a principal
component analysis (PCA) is applied on the original raw matrix of data
and the four possible component (<em>n.pc</em> = 4) derived from the
analysis are used in a further linear discriminant analysis (LDA). A
scaling step is applied to the raw matrix of data before the application
of the mentioned procedure (<em>center</em> = TRUE, <em>scale</em> =
TRUE). Here, PCA will yield new orthogonal (non-correlated) variables,
the principal components, which prevent any potential bias effect
originated by any correlation or association of the original
variables.</p>
<p>By using function <a href="https://is.gd/RR9iWY" class="external-link">estimateCutPoint</a>, we can estimate the
cutpoint, based on <em>HD</em> (<em>div.col</em> = 9L) or on <span class="math inline">\(TV_d\)</span> (<em>div.col</em> = 7L):</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Cutpoint estimation for the FT approach using the ECDF critical value</span></span>
<span><span class="va">cut.g2p</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">pDMPs.g2p</span>, </span>
<span>                            simple <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                        wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                            classifier1 <span class="op">=</span> <span class="st">"pca.lda"</span>, n.pc <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                            control.names <span class="op">=</span> <span class="va">control.nam</span>,</span>
<span>                            treatment.names <span class="op">=</span> <span class="va">treatment.nam</span>,</span>
<span>                            center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            clas.perf <span class="op">=</span> <span class="cn">TRUE</span>, prop <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>                            div.col <span class="op">=</span> <span class="fl">9L</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Found more than one class "pDMP" in cache; using the first, from namespace 'MethylIT'</span></span></code></pre>
<pre><code><span><span class="co">## Also defined by 'MethylIT.utils'</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cut.g2p</span></span></code></pre></div>
<pre><code><span><span class="co">## Cutpoint estimation with 'Youden Index' </span></span>
<span><span class="co">## Simple cutpoint estimation </span></span>
<span><span class="co">## Cutpoint = 114.22 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Cytosine sites from treatment have divergence values &gt;= 114.22 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## The accessible objects in the output list are: </span></span>
<span><span class="co">##                     Length Class           Mode     </span></span>
<span><span class="co">## cutpoint            1      -none-          numeric  </span></span>
<span><span class="co">## testSetPerformance  6      confusionMatrix list     </span></span>
<span><span class="co">## testSetModel.FDR    1      -none-          numeric  </span></span>
<span><span class="co">## model               5      pcaLDA          list     </span></span>
<span><span class="co">## modelConfMatrix     6      confusionMatrix list     </span></span>
<span><span class="co">## initModel           1      -none-          character</span></span>
<span><span class="co">## postProbCut         1      -none-          logical  </span></span>
<span><span class="co">## postCut             1      -none-          logical  </span></span>
<span><span class="co">## classifier          1      -none-          character</span></span>
<span><span class="co">## statistic           1      -none-          logical  </span></span>
<span><span class="co">## optStatVal          1      -none-          logical  </span></span>
<span><span class="co">## cutpData            1      -none-          logical  </span></span>
<span><span class="co">## initModelConfMatrix 6      confusionMatrix list</span></span></code></pre>
<p>As indicated above, the model classifier performance and its
corresponding false discovery rate can be retrieved as:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cut.g2p</span><span class="op">$</span><span class="va">testSetPerformance</span></span></code></pre></div>
<pre><code><span><span class="co">## Confusion Matrix and Statistics</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##           Reference</span></span>
<span><span class="co">## Prediction   CT   TT</span></span>
<span><span class="co">##         CT  319    0</span></span>
<span><span class="co">##         TT    0 3544</span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##                Accuracy : 1         </span></span>
<span><span class="co">##                  95% CI : (0.999, 1)</span></span>
<span><span class="co">##     No Information Rate : 0.9174    </span></span>
<span><span class="co">##     P-Value [Acc &gt; NIR] : &lt; 2.2e-16 </span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##                   Kappa : 1         </span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##  Mcnemar's Test P-Value : NA        </span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##             Sensitivity : 1.0000    </span></span>
<span><span class="co">##             Specificity : 1.0000    </span></span>
<span><span class="co">##          Pos Pred Value : 1.0000    </span></span>
<span><span class="co">##          Neg Pred Value : 1.0000    </span></span>
<span><span class="co">##              Prevalence : 0.9174    </span></span>
<span><span class="co">##          Detection Rate : 0.9174    </span></span>
<span><span class="co">##    Detection Prevalence : 0.9174    </span></span>
<span><span class="co">##       Balanced Accuracy : 1.0000    </span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##        'Positive' Class : TT        </span></span>
<span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cut.g2p</span><span class="op">$</span><span class="va">testSetModel.FDR</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0</span></span></code></pre>
<p>Here, DMP classification is modeled with PCA+QDA classifier
(<em>classifier</em> = “pca.lda”). That is, principal component analysis
(PCA) is applied on the original raw matrix of data and the four
possible component (<em>n.pc</em> = 4) are used in a further linear
discriminant analysis (LDA). A scaling step is applied to the raw matrix
of data before the application of the mentioned procedure
(<em>center</em> = TRUE, <em>scale</em> = TRUE). Next, a different model
classifier can be applied to model the classification derived from the
previous cutpoint estimation.</p>
</div>
<div class="section level4">
<h4 id="simple-cutpoint-estimation-for-gamma3p-model">Simple cutpoint estimation for Gamma3P model<a class="anchor" aria-label="anchor" href="#simple-cutpoint-estimation-for-gamma3p-model"></a>
</h4>
<p>The same analyses for the cutpoint estimation can be performed for 3P
gamma model</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Cutpoint estimation for the FT approach using the ECDF critical value</span></span>
<span><span class="va">cut.g3p</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">pDMPs.g3p</span>, simple <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                        wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                            classifier1 <span class="op">=</span> <span class="st">"pca.lda"</span>, n.pc <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                            control.names <span class="op">=</span> <span class="va">control.nam</span>,</span>
<span>                            treatment.names <span class="op">=</span> <span class="va">treatment.nam</span>,</span>
<span>                            center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            clas.perf <span class="op">=</span> <span class="cn">TRUE</span>, prop <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>                            div.col <span class="op">=</span> <span class="fl">9L</span><span class="op">)</span></span>
<span><span class="va">cut.g3p</span></span></code></pre></div>
<pre><code><span><span class="co">## Cutpoint estimation with 'Youden Index' </span></span>
<span><span class="co">## Simple cutpoint estimation </span></span>
<span><span class="co">## Cutpoint = 115.24 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Cytosine sites from treatment have divergence values &gt;= 115.24 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## The accessible objects in the output list are: </span></span>
<span><span class="co">##                     Length Class           Mode     </span></span>
<span><span class="co">## cutpoint            1      -none-          numeric  </span></span>
<span><span class="co">## testSetPerformance  6      confusionMatrix list     </span></span>
<span><span class="co">## testSetModel.FDR    1      -none-          numeric  </span></span>
<span><span class="co">## model               5      pcaLDA          list     </span></span>
<span><span class="co">## modelConfMatrix     6      confusionMatrix list     </span></span>
<span><span class="co">## initModel           1      -none-          character</span></span>
<span><span class="co">## postProbCut         1      -none-          logical  </span></span>
<span><span class="co">## postCut             1      -none-          logical  </span></span>
<span><span class="co">## classifier          1      -none-          character</span></span>
<span><span class="co">## statistic           1      -none-          logical  </span></span>
<span><span class="co">## optStatVal          1      -none-          logical  </span></span>
<span><span class="co">## cutpData            1      -none-          logical  </span></span>
<span><span class="co">## initModelConfMatrix 6      confusionMatrix list</span></span></code></pre>
<p>As indicated above, the model classifier performance and its
corresponding false discovery rate can be retrieved as:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cut.g3p</span><span class="op">$</span><span class="va">testSetPerformance</span></span></code></pre></div>
<pre><code><span><span class="co">## Confusion Matrix and Statistics</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##           Reference</span></span>
<span><span class="co">## Prediction   CT   TT</span></span>
<span><span class="co">##         CT  309    0</span></span>
<span><span class="co">##         TT    0 3481</span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##                Accuracy : 1         </span></span>
<span><span class="co">##                  95% CI : (0.999, 1)</span></span>
<span><span class="co">##     No Information Rate : 0.9185    </span></span>
<span><span class="co">##     P-Value [Acc &gt; NIR] : &lt; 2.2e-16 </span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##                   Kappa : 1         </span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##  Mcnemar's Test P-Value : NA        </span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##             Sensitivity : 1.0000    </span></span>
<span><span class="co">##             Specificity : 1.0000    </span></span>
<span><span class="co">##          Pos Pred Value : 1.0000    </span></span>
<span><span class="co">##          Neg Pred Value : 1.0000    </span></span>
<span><span class="co">##              Prevalence : 0.9185    </span></span>
<span><span class="co">##          Detection Rate : 0.9185    </span></span>
<span><span class="co">##    Detection Prevalence : 0.9185    </span></span>
<span><span class="co">##       Balanced Accuracy : 1.0000    </span></span>
<span><span class="co">##                                     </span></span>
<span><span class="co">##        'Positive' Class : TT        </span></span>
<span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cut.g3p</span><span class="op">$</span><span class="va">testSetModel.FDR</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="dmp-prediction-based-on-classification-models">DMP prediction based on classification models<a class="anchor" aria-label="anchor" href="#dmp-prediction-based-on-classification-models"></a>
</h4>
<p>The model obtained in the previous step can be used for further
prediction with function <a href="https://genomaths.github.io/methylit/reference/pcaLDA.html">predict</a>.
For example, we would take a random sample and run:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">randsampl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/unlist.html">unlist</a></span><span class="op">(</span><span class="va">pDMPs.g3p</span><span class="op">)</span></span>
<span><span class="va">randsampl</span> <span class="op">&lt;-</span> <span class="va">randsampl</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample.int</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">randsampl</span><span class="op">)</span>, <span class="fl">10</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="va">pred</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predict.html">predict</a></span><span class="op">(</span><span class="va">cut.g3p</span><span class="op">$</span><span class="va">model</span>, newdata <span class="op">=</span> <span class="va">randsampl</span><span class="op">)</span></span>
<span><span class="va">pred</span></span></code></pre></div>
<pre><code><span><span class="co">## $class</span></span>
<span><span class="co">##  [1] CT TT TT TT TT TT TT TT TT TT</span></span>
<span><span class="co">## Levels: CT TT</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $posterior</span></span>
<span><span class="co">##                 CT           TT</span></span>
<span><span class="co">##  [1,] 1.000000e+00 9.949362e-09</span></span>
<span><span class="co">##  [2,] 1.323743e-46 1.000000e+00</span></span>
<span><span class="co">##  [3,] 1.435707e-45 1.000000e+00</span></span>
<span><span class="co">##  [4,] 2.648143e-46 1.000000e+00</span></span>
<span><span class="co">##  [5,] 7.965328e-48 1.000000e+00</span></span>
<span><span class="co">##  [6,] 1.108810e-46 1.000000e+00</span></span>
<span><span class="co">##  [7,] 1.260233e-46 1.000000e+00</span></span>
<span><span class="co">##  [8,] 8.773698e-47 1.000000e+00</span></span>
<span><span class="co">##  [9,] 4.528686e-47 1.000000e+00</span></span>
<span><span class="co">## [10,] 3.130172e-54 1.000000e+00</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $x</span></span>
<span><span class="co">##              LD1</span></span>
<span><span class="co">##  [1,] -7.4843076</span></span>
<span><span class="co">##  [2,]  1.1181324</span></span>
<span><span class="co">##  [3,]  0.9528444</span></span>
<span><span class="co">##  [4,]  1.0700533</span></span>
<span><span class="co">##  [5,]  1.3130110</span></span>
<span><span class="co">##  [6,]  1.1304175</span></span>
<span><span class="co">##  [7,]  1.1215415</span></span>
<span><span class="co">##  [8,]  1.1466507</span></span>
<span><span class="co">##  [9,]  1.1925061</span></span>
<span><span class="co">## [10,]  2.3357223</span></span></code></pre>
<p>The variable <em>pred$posterior</em> provides the posterior
classification probabilities that a DMP could belong to control
(<em>CT</em>) or to treatment (<em>TT</em>) group. The variable
‘<em>x</em>’ provides the cytosine methylation changes in terms of its
values in the linear discriminant function LD1. Notice that, for each
row, the sum of posterior probabilities is equal 1. By default,
individuals with <em>TT</em> posterior probabilities greater than 0.5
are predicted to belong to the treatment class. For example:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">classfiction</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"CT"</span>, <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">classfiction</span><span class="op">[</span><span class="va">pred</span><span class="op">$</span><span class="va">posterior</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"TT"</span></span>
<span><span class="va">classfiction</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "CT" "TT" "TT" "TT" "TT" "TT" "TT" "TT" "TT" "TT"</span></span></code></pre>
<p>We can be more strict increasing the posterior classification
probability cutoff</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">classfiction</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"CT"</span>, <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">classfiction</span><span class="op">[</span><span class="va">pred</span><span class="op">$</span><span class="va">posterior</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span> <span class="op">&gt;</span> <span class="fl">0.7</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"TT"</span></span>
<span><span class="va">classfiction</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "CT" "TT" "TT" "TT" "TT" "TT" "TT" "TT" "TT" "TT"</span></span></code></pre>
<p>The posterior classification probability cutoff can be controlled
with parameter <em>post.cut</em> from <a href="https://is.gd/RR9iWY" class="external-link">estimateCutPoint</a> function (default:
<span class="math inline">\(post.cut=0.5\)</span>).</p>
</div>
</div>
<div class="section level3">
<h3 id="machine-learning-ml-based-approach-to-search-for-an-optimal-cutpoint">Machine-learning (ML) based approach to search for an optimal
cutpoint<a class="anchor" aria-label="anchor" href="#machine-learning-ml-based-approach-to-search-for-an-optimal-cutpoint"></a>
</h3>
<p>In the next example the cutpoint estimation for the Hellinger
divergence of methylation levels (<em>div.col</em> = 9L) is
accomplished. Function <em>estimateCutPoint</em> can be used to search
for a cutpoint as well. Two model classifiers can be used.
<em>classifiers1</em> will be used to estimate the posterior
classification probabilities of DMP into those from control and those
from treatment. These probabilities are then used to estimate the
cutpoint in the range of values from, say, 0.5 to 0.8. Next, a
<em>classifier2</em> will be used to evaluate the classification
performance. In this case, the search for an optimal cutpoint is
accomplished maximizing the accuracy (<em>stat</em> = 0) of
<em>classifier2</em>.</p>
<div class="section level4">
<h4 id="ml-cutpoint-estimation-for-potential-dmps-based-on-gamma2p-model">ML cutpoint estimation for potential DMPs based on Gamma2P
model<a class="anchor" aria-label="anchor" href="#ml-cutpoint-estimation-for-potential-dmps-based-on-gamma2p-model"></a>
</h4>
<p>The ML search for an optimal cutpoint is accomplished in the set of
potential DMPs, which were identified using a Gamma2P probability
distribution model as null hypothesis.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cut.g2p</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">pDMPs.g2p</span>, simple <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                            column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                        wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                            classifier1 <span class="op">=</span> <span class="st">"pca.lda"</span>, </span>
<span>                            classifier2 <span class="op">=</span> <span class="st">"pca.qda"</span>, stat <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                            control.names <span class="op">=</span> <span class="va">control.nam</span>, </span>
<span>                            treatment.names <span class="op">=</span> <span class="va">treatment.nam</span>, </span>
<span>                            cut.values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">45</span>, <span class="fl">114</span>, <span class="fl">1</span><span class="op">)</span>, post.cut <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                            prop <span class="op">=</span> <span class="fl">0.6</span>, center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            n.pc <span class="op">=</span> <span class="fl">4</span>, div.col <span class="op">=</span> <span class="fl">9L</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Found more than one class "pDMP" in cache; using the first, from namespace 'MethylIT'</span></span></code></pre>
<pre><code><span><span class="co">## Also defined by 'MethylIT.utils'</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cut.g2p</span></span></code></pre></div>
<pre><code><span><span class="co">## Cutpoint estimation with 'pca.lda' classifier </span></span>
<span><span class="co">## Cutpoint search performed using model posterior probabilities </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Posterior probability used to get the cutpoint = 0.5 </span></span>
<span><span class="co">## Cytosine sites with treatment PostProbCut &gt;= 0.5 have a </span></span>
<span><span class="co">## divergence value &gt;= 112.4245 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Optimized statistic: Accuracy = 1 </span></span>
<span><span class="co">## Cutpoint = 112.42 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Model classifier 'pca.qda' </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## The accessible objects in the output list are: </span></span>
<span><span class="co">##                    Length Class           Mode     </span></span>
<span><span class="co">## cutpoint           1      -none-          numeric  </span></span>
<span><span class="co">## testSetPerformance 6      confusionMatrix list     </span></span>
<span><span class="co">## testSetModel.FDR   1      -none-          numeric  </span></span>
<span><span class="co">## model              5      pcaQDA          list     </span></span>
<span><span class="co">## modelConfMatrix    6      confusionMatrix list     </span></span>
<span><span class="co">## initModel          1      -none-          character</span></span>
<span><span class="co">## postProbCut        1      -none-          numeric  </span></span>
<span><span class="co">## postCut            1      -none-          numeric  </span></span>
<span><span class="co">## classifier         1      -none-          character</span></span>
<span><span class="co">## statistic          1      -none-          character</span></span>
<span><span class="co">## optStatVal         1      -none-          numeric  </span></span>
<span><span class="co">## cutpData           1      -none-          logical</span></span></code></pre>
<p>Model performance in the test dataset is:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cut.g2p</span><span class="op">$</span><span class="va">testSetPerformance</span></span></code></pre></div>
<pre><code><span><span class="co">## Confusion Matrix and Statistics</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##           Reference</span></span>
<span><span class="co">## Prediction   CT   TT</span></span>
<span><span class="co">##         CT 1274    0</span></span>
<span><span class="co">##         TT    0 3580</span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##                Accuracy : 1          </span></span>
<span><span class="co">##                  95% CI : (0.9992, 1)</span></span>
<span><span class="co">##     No Information Rate : 0.7375     </span></span>
<span><span class="co">##     P-Value [Acc &gt; NIR] : &lt; 2.2e-16  </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##                   Kappa : 1          </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##  Mcnemar's Test P-Value : NA         </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##             Sensitivity : 1.0000     </span></span>
<span><span class="co">##             Specificity : 1.0000     </span></span>
<span><span class="co">##          Pos Pred Value : 1.0000     </span></span>
<span><span class="co">##          Neg Pred Value : 1.0000     </span></span>
<span><span class="co">##              Prevalence : 0.7375     </span></span>
<span><span class="co">##          Detection Rate : 0.7375     </span></span>
<span><span class="co">##    Detection Prevalence : 0.7375     </span></span>
<span><span class="co">##       Balanced Accuracy : 1.0000     </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##        'Positive' Class : TT         </span></span>
<span><span class="co">## </span></span></code></pre>
<p>Model performance in in the whole dataset is:</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cut.g2p</span><span class="op">$</span><span class="va">modelConfMatrix</span></span></code></pre></div>
<pre><code><span><span class="co">## Confusion Matrix and Statistics</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##           Reference</span></span>
<span><span class="co">## Prediction   CT   TT</span></span>
<span><span class="co">##         CT 3184    0</span></span>
<span><span class="co">##         TT    0 8948</span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##                Accuracy : 1          </span></span>
<span><span class="co">##                  95% CI : (0.9997, 1)</span></span>
<span><span class="co">##     No Information Rate : 0.7376     </span></span>
<span><span class="co">##     P-Value [Acc &gt; NIR] : &lt; 2.2e-16  </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##                   Kappa : 1          </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##  Mcnemar's Test P-Value : NA         </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##             Sensitivity : 1.0000     </span></span>
<span><span class="co">##             Specificity : 1.0000     </span></span>
<span><span class="co">##          Pos Pred Value : 1.0000     </span></span>
<span><span class="co">##          Neg Pred Value : 1.0000     </span></span>
<span><span class="co">##              Prevalence : 0.7376     </span></span>
<span><span class="co">##          Detection Rate : 0.7376     </span></span>
<span><span class="co">##    Detection Prevalence : 0.7376     </span></span>
<span><span class="co">##       Balanced Accuracy : 1.0000     </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##        'Positive' Class : TT         </span></span>
<span><span class="co">## </span></span></code></pre>
<p>The False discovery rate is:</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cut.g2p</span><span class="op">$</span><span class="va">testSetModel.FDR</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0</span></span></code></pre>
<p>The model classifier <em>PCA+LDA</em> has enough discriminatory power
to discriminate control DMP from those induced by the treatment for
<em>HD</em> values <span class="math inline">\(112.4247 \le
HD\)</span>.</p>
<p>The probabilities <span class="math inline">\(P(HD \le
122.43)\)</span> to observe a cytosine site with <span class="math inline">\(HD \le 112.4247\)</span> on each individual
is:</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">nlms.g2p</span><span class="op">)</span></span>
<span><span class="va">crit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">nlms.g2p</span>, </span>
<span>                    <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> </span>
<span>                        <span class="fu"><a href="https://rdrr.io/r/stats/GammaDist.html" class="external-link">pgamma</a></span><span class="op">(</span><span class="va">cut.g2p</span><span class="op">$</span><span class="va">cutpoint</span>, shape <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Estimate</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>                                scale <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">Estimate</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">crit</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">nams</span></span>
<span><span class="va">crit</span></span></code></pre></div>
<pre><code><span><span class="co">##        C1        C2        C3        T1        T2        T3 </span></span>
<span><span class="co">## 0.9964705 0.9966560 0.9967314 0.9500280 0.9483024 0.9476607</span></span></code></pre>
<p>In other words, most of the methylation changes are not (and should
not be) considered DMPs. Notice that although the same <em>HD</em> value
could be found in the same differentially methylated cytosine site in
control and treatment, if the probabilities distributions of the
information divergences (null hypotheses) from control and treatment are
different, then these DMPs can be distinguished.</p>
</div>
<div class="section level4">
<h4 id="ml-cutpoint-estimation-for-potential-dmps-based-on-gamma3p-model">ML cutpoint estimation for potential DMPs based on Gamma3P
model<a class="anchor" aria-label="anchor" href="#ml-cutpoint-estimation-for-potential-dmps-based-on-gamma3p-model"></a>
</h4>
<p>Likewise, for the 3-parameter gamma model we can search for an
optimal cutpoint:</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cut.g3p</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">pDMPs.g3p</span>, simple <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                            column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, TV <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                        wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                            classifier1 <span class="op">=</span> <span class="st">"pca.lda"</span>, </span>
<span>                            classifier2 <span class="op">=</span> <span class="st">"pca.qda"</span>, stat <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                            control.names <span class="op">=</span> <span class="va">control.nam</span>, </span>
<span>                            treatment.names <span class="op">=</span> <span class="va">treatment.nam</span>, </span>
<span>                            cut.values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">45</span>, <span class="fl">114</span>, <span class="fl">1</span><span class="op">)</span>, post.cut <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                            clas.perf <span class="op">=</span> <span class="cn">TRUE</span>, prop <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>                            center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            n.pc <span class="op">=</span> <span class="fl">4</span>, div.col <span class="op">=</span> <span class="fl">9L</span><span class="op">)</span></span>
<span><span class="va">cut.g3p</span></span></code></pre></div>
<pre><code><span><span class="co">## Cutpoint estimation with 'pca.lda' classifier </span></span>
<span><span class="co">## Cutpoint search performed using model posterior probabilities </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Posterior probability used to get the cutpoint = 0.5 </span></span>
<span><span class="co">## Cytosine sites with treatment PostProbCut &gt;= 0.5 have a </span></span>
<span><span class="co">## divergence value &gt;= 113.5828 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Optimized statistic: Accuracy = 1 </span></span>
<span><span class="co">## Cutpoint = 113.58 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Model classifier 'pca.qda' </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## The accessible objects in the output list are: </span></span>
<span><span class="co">##                    Length Class           Mode     </span></span>
<span><span class="co">## cutpoint           1      -none-          numeric  </span></span>
<span><span class="co">## testSetPerformance 6      confusionMatrix list     </span></span>
<span><span class="co">## testSetModel.FDR   1      -none-          numeric  </span></span>
<span><span class="co">## model              5      pcaQDA          list     </span></span>
<span><span class="co">## modelConfMatrix    6      confusionMatrix list     </span></span>
<span><span class="co">## initModel          1      -none-          character</span></span>
<span><span class="co">## postProbCut        1      -none-          numeric  </span></span>
<span><span class="co">## postCut            1      -none-          numeric  </span></span>
<span><span class="co">## classifier         1      -none-          character</span></span>
<span><span class="co">## statistic          1      -none-          character</span></span>
<span><span class="co">## optStatVal         1      -none-          numeric  </span></span>
<span><span class="co">## cutpData           1      -none-          logical</span></span></code></pre>
<p>DMPs are identified with function <a href="https://is.gd/7cVEEd" class="external-link">selectDIMP</a></p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">g2p.dmps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectDIMP.html">selectDIMP</a></span><span class="op">(</span><span class="va">pDMPs.g2p</span>, div.col <span class="op">=</span> <span class="fl">9L</span>, cutpoint <span class="op">=</span> <span class="va">cut.g2p</span><span class="op">$</span><span class="va">cutpoint</span><span class="op">)</span></span>
<span><span class="va">g3p.dmps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectDIMP.html">selectDIMP</a></span><span class="op">(</span><span class="va">pDMPs.g3p</span>, div.col <span class="op">=</span> <span class="fl">9L</span>, cutpoint <span class="op">=</span> <span class="va">cut.g3p</span><span class="op">$</span><span class="va">cutpoint</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="dmps-estimated-with-fishers-exact-test-ft">DMPs estimated with Fisher’s exact Test (FT)<a class="anchor" aria-label="anchor" href="#dmps-estimated-with-fishers-exact-test-ft"></a>
</h3>
<p>For comparison purposes DMPs are estimated with Fisher’s exact test
as well.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ft.</span> <span class="op">=</span> <span class="fu"><a href="../reference/FisherTest.html">FisherTest</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">divs</span>, tv.cut <span class="op">=</span> <span class="fl">0.68</span>, </span>
<span>                 pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>                 pvalCutOff <span class="op">=</span> <span class="fl">0.05</span>, </span>
<span>                 num.cores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html" class="external-link">multicoreWorkers</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                 verbose <span class="op">=</span> <span class="cn">FALSE</span>, saveAll <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">ft.dmps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">ft.</span>, div.col <span class="op">=</span> <span class="fl">9L</span>, dist.name <span class="op">=</span> <span class="st">"None"</span>,</span>
<span>                            tv.cut <span class="op">=</span> <span class="fl">0.68</span>, tv.col <span class="op">=</span> <span class="fl">7</span>, alpha <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="classification-performance-of-dmp-classification">Classification performance of DMP classification<a class="anchor" aria-label="anchor" href="#classification-performance-of-dmp-classification"></a>
</h3>
<p>After the cutpoint application to select DMPs, a Monte Carlo
(bootstrap) analysis reporting several classifier performance indicators
can be accomplished by using function <a href="https://is.gd/nzy9j3" class="external-link"><em>evaluateDIMPclass</em></a> and its
settings <em>output = “mc.val”</em> and <em>num.boot</em>.</p>
<div class="section level4">
<h4 id="classification-performance-for-dmps-based-on-gamma2p-model">Classification performance for DMPs based on Gamma2P model<a class="anchor" aria-label="anchor" href="#classification-performance-for-dmps-based-on-gamma2p-model"></a>
</h4>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">g2p.class</span> <span class="op">=</span> <span class="fu"><a href="../reference/evaluateDIMPclass.html">evaluateDIMPclass</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">g2p.dmps</span>, control.names <span class="op">=</span> <span class="va">control.nam</span>,</span>
<span>                            treatment.names <span class="op">=</span> <span class="va">treatment.nam</span>,</span>
<span>                            column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, TV <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                    wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                            classifier <span class="op">=</span> <span class="st">"pca.qda"</span>, n.pc <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                            center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>, num.boot <span class="op">=</span> <span class="fl">300</span>,</span>
<span>                            output <span class="op">=</span> <span class="st">"all"</span>, prop <span class="op">=</span> <span class="fl">0.6</span></span>
<span><span class="op">)</span></span>
<span><span class="va">g2p.class</span><span class="op">$</span><span class="va">mc.val</span></span></code></pre></div>
<pre><code><span><span class="co">##     Accuracy     Kappa   AccuracyLower    AccuracyUpper  AccuracyNull   </span></span>
<span><span class="co">##  Min.   :1   Min.   :1   Min.   :0.9991   Min.   :1     Min.   :0.9139  </span></span>
<span><span class="co">##  1st Qu.:1   1st Qu.:1   1st Qu.:0.9991   1st Qu.:1     1st Qu.:0.9139  </span></span>
<span><span class="co">##  Median :1   Median :1   Median :0.9991   Median :1     Median :0.9139  </span></span>
<span><span class="co">##  Mean   :1   Mean   :1   Mean   :0.9991   Mean   :1     Mean   :0.9139  </span></span>
<span><span class="co">##  3rd Qu.:1   3rd Qu.:1   3rd Qu.:0.9991   3rd Qu.:1     3rd Qu.:0.9139  </span></span>
<span><span class="co">##  Max.   :1   Max.   :1   Max.   :0.9991   Max.   :1     Max.   :0.9139  </span></span>
<span><span class="co">##                                                                         </span></span>
<span><span class="co">##  AccuracyPValue       McnemarPValue  Sensitivity  Specificity Pos Pred Value</span></span>
<span><span class="co">##  Min.   :9.096e-154   Min.   : NA   Min.   :1    Min.   :1    Min.   :1     </span></span>
<span><span class="co">##  1st Qu.:9.096e-154   1st Qu.: NA   1st Qu.:1    1st Qu.:1    1st Qu.:1     </span></span>
<span><span class="co">##  Median :9.096e-154   Median : NA   Median :1    Median :1    Median :1     </span></span>
<span><span class="co">##  Mean   :9.096e-154   Mean   :NaN   Mean   :1    Mean   :1    Mean   :1     </span></span>
<span><span class="co">##  3rd Qu.:9.096e-154   3rd Qu.: NA   3rd Qu.:1    3rd Qu.:1    3rd Qu.:1     </span></span>
<span><span class="co">##  Max.   :9.096e-154   Max.   : NA   Max.   :1    Max.   :1    Max.   :1     </span></span>
<span><span class="co">##                       NA's   :300                                           </span></span>
<span><span class="co">##  Neg Pred Value   Precision     Recall        F1      Prevalence    </span></span>
<span><span class="co">##  Min.   :1      Min.   :1   Min.   :1   Min.   :1   Min.   :0.9139  </span></span>
<span><span class="co">##  1st Qu.:1      1st Qu.:1   1st Qu.:1   1st Qu.:1   1st Qu.:0.9139  </span></span>
<span><span class="co">##  Median :1      Median :1   Median :1   Median :1   Median :0.9139  </span></span>
<span><span class="co">##  Mean   :1      Mean   :1   Mean   :1   Mean   :1   Mean   :0.9139  </span></span>
<span><span class="co">##  3rd Qu.:1      3rd Qu.:1   3rd Qu.:1   3rd Qu.:1   3rd Qu.:0.9139  </span></span>
<span><span class="co">##  Max.   :1      Max.   :1   Max.   :1   Max.   :1   Max.   :0.9139  </span></span>
<span><span class="co">##                                                                     </span></span>
<span><span class="co">##  Detection Rate   Detection Prevalence Balanced Accuracy</span></span>
<span><span class="co">##  Min.   :0.9139   Min.   :0.9139       Min.   :1        </span></span>
<span><span class="co">##  1st Qu.:0.9139   1st Qu.:0.9139       1st Qu.:1        </span></span>
<span><span class="co">##  Median :0.9139   Median :0.9139       Median :1        </span></span>
<span><span class="co">##  Mean   :0.9139   Mean   :0.9139       Mean   :1        </span></span>
<span><span class="co">##  3rd Qu.:0.9139   3rd Qu.:0.9139       3rd Qu.:1        </span></span>
<span><span class="co">##  Max.   :0.9139   Max.   :0.9139       Max.   :1        </span></span>
<span><span class="co">## </span></span></code></pre>
</div>
<div class="section level4">
<h4 id="classification-performance-for-dmps-based-on-gamma3p-model">Classification performance for DMPs based on Gamma3P model<a class="anchor" aria-label="anchor" href="#classification-performance-for-dmps-based-on-gamma3p-model"></a>
</h4>
<p>Likewise the evaluation of the DMP classification performance can be
accomplished for DMPs estimated based on the <span class="math inline">\('Gamma3P'\)</span> model:</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">g3p.class</span> <span class="op">=</span> <span class="fu"><a href="../reference/evaluateDIMPclass.html">evaluateDIMPclass</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">g3p.dmps</span>, control.names <span class="op">=</span> <span class="va">control.nam</span>,</span>
<span>                            treatment.names <span class="op">=</span> <span class="va">treatment.nam</span>,</span>
<span>                            column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, TV <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                    wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                            classifier <span class="op">=</span> <span class="st">"pca.qda"</span>, n.pc <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                            center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>, num.boot <span class="op">=</span> <span class="fl">300</span>,</span>
<span>                            output <span class="op">=</span> <span class="st">"all"</span>, prop <span class="op">=</span> <span class="fl">0.6</span></span>
<span><span class="op">)</span></span>
<span><span class="va">g3p.class</span><span class="op">$</span><span class="va">mc.val</span></span></code></pre></div>
<pre><code><span><span class="co">##     Accuracy     Kappa   AccuracyLower   AccuracyUpper  AccuracyNull  </span></span>
<span><span class="co">##  Min.   :1   Min.   :1   Min.   :0.999   Min.   :1     Min.   :0.914  </span></span>
<span><span class="co">##  1st Qu.:1   1st Qu.:1   1st Qu.:0.999   1st Qu.:1     1st Qu.:0.914  </span></span>
<span><span class="co">##  Median :1   Median :1   Median :0.999   Median :1     Median :0.914  </span></span>
<span><span class="co">##  Mean   :1   Mean   :1   Mean   :0.999   Mean   :1     Mean   :0.914  </span></span>
<span><span class="co">##  3rd Qu.:1   3rd Qu.:1   3rd Qu.:0.999   3rd Qu.:1     3rd Qu.:0.914  </span></span>
<span><span class="co">##  Max.   :1   Max.   :1   Max.   :0.999   Max.   :1     Max.   :0.914  </span></span>
<span><span class="co">##                                                                       </span></span>
<span><span class="co">##  AccuracyPValue       McnemarPValue  Sensitivity  Specificity Pos Pred Value</span></span>
<span><span class="co">##  Min.   :1.387e-150   Min.   : NA   Min.   :1    Min.   :1    Min.   :1     </span></span>
<span><span class="co">##  1st Qu.:1.387e-150   1st Qu.: NA   1st Qu.:1    1st Qu.:1    1st Qu.:1     </span></span>
<span><span class="co">##  Median :1.387e-150   Median : NA   Median :1    Median :1    Median :1     </span></span>
<span><span class="co">##  Mean   :1.387e-150   Mean   :NaN   Mean   :1    Mean   :1    Mean   :1     </span></span>
<span><span class="co">##  3rd Qu.:1.387e-150   3rd Qu.: NA   3rd Qu.:1    3rd Qu.:1    3rd Qu.:1     </span></span>
<span><span class="co">##  Max.   :1.387e-150   Max.   : NA   Max.   :1    Max.   :1    Max.   :1     </span></span>
<span><span class="co">##                       NA's   :300                                           </span></span>
<span><span class="co">##  Neg Pred Value   Precision     Recall        F1      Prevalence   </span></span>
<span><span class="co">##  Min.   :1      Min.   :1   Min.   :1   Min.   :1   Min.   :0.914  </span></span>
<span><span class="co">##  1st Qu.:1      1st Qu.:1   1st Qu.:1   1st Qu.:1   1st Qu.:0.914  </span></span>
<span><span class="co">##  Median :1      Median :1   Median :1   Median :1   Median :0.914  </span></span>
<span><span class="co">##  Mean   :1      Mean   :1   Mean   :1   Mean   :1   Mean   :0.914  </span></span>
<span><span class="co">##  3rd Qu.:1      3rd Qu.:1   3rd Qu.:1   3rd Qu.:1   3rd Qu.:0.914  </span></span>
<span><span class="co">##  Max.   :1      Max.   :1   Max.   :1   Max.   :1   Max.   :0.914  </span></span>
<span><span class="co">##                                                                    </span></span>
<span><span class="co">##  Detection Rate  Detection Prevalence Balanced Accuracy</span></span>
<span><span class="co">##  Min.   :0.914   Min.   :0.914        Min.   :1        </span></span>
<span><span class="co">##  1st Qu.:0.914   1st Qu.:0.914        1st Qu.:1        </span></span>
<span><span class="co">##  Median :0.914   Median :0.914        Median :1        </span></span>
<span><span class="co">##  Mean   :0.914   Mean   :0.914        Mean   :1        </span></span>
<span><span class="co">##  3rd Qu.:0.914   3rd Qu.:0.914        3rd Qu.:1        </span></span>
<span><span class="co">##  Max.   :0.914   Max.   :0.914        Max.   :1        </span></span>
<span><span class="co">## </span></span></code></pre>
<p>We do not have evidence to support statistical differences between
the classification performances estimated for <em>‘Gamma2P’</em> and
<em>‘Gamma3P’</em> probability distribution models. Hence, in this case
we select the model that yield the lowest cutpoint</p>
</div>
<div class="section level4">
<h4 id="classification-performance-for-dmps-based-on-fishers-exact-test">Classification performance for DMPs based on Fisher’s exact
test<a class="anchor" aria-label="anchor" href="#classification-performance-for-dmps-based-on-fishers-exact-test"></a>
</h4>
<p>Classification performance results obtained with Monte Carlos
sampling for the <span class="math inline">\(Gamma2P\)</span> and <span class="math inline">\(Gamma3P\)</span> models are quite different from
those obtained with FT:</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ft.class</span> <span class="op">=</span> <span class="fu"><a href="../reference/evaluateDIMPclass.html">evaluateDIMPclass</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">ft.dmps</span>, control.names <span class="op">=</span> <span class="va">control.nam</span>,</span>
<span>                            treatment.names <span class="op">=</span> <span class="va">treatment.nam</span>, </span>
<span>                            column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, TV <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                    wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                            classifier <span class="op">=</span> <span class="st">"pca.lda"</span>, n.pc <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                            center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>, num.boot <span class="op">=</span> <span class="fl">300</span>,</span>
<span>                            output <span class="op">=</span> <span class="st">"all"</span>, prop <span class="op">=</span> <span class="fl">0.6</span></span>
<span><span class="op">)</span></span>
<span><span class="va">ft.class</span><span class="op">$</span><span class="va">mc.val</span></span></code></pre></div>
<pre><code><span><span class="co">##     Accuracy          Kappa           AccuracyLower    AccuracyUpper   </span></span>
<span><span class="co">##  Min.   :0.8084   Min.   :-0.001334   Min.   :0.8010   Min.   :0.8156  </span></span>
<span><span class="co">##  1st Qu.:0.8104   1st Qu.: 0.005460   1st Qu.:0.8030   1st Qu.:0.8177  </span></span>
<span><span class="co">##  Median :0.8111   Median : 0.007802   Median :0.8037   Median :0.8183  </span></span>
<span><span class="co">##  Mean   :0.8110   Mean   : 0.007737   Mean   :0.8036   Mean   :0.8182  </span></span>
<span><span class="co">##  3rd Qu.:0.8115   3rd Qu.: 0.010172   3rd Qu.:0.8041   3rd Qu.:0.8187  </span></span>
<span><span class="co">##  Max.   :0.8130   Max.   : 0.020023   Max.   :0.8057   Max.   :0.8202  </span></span>
<span><span class="co">##   AccuracyNull    AccuracyPValue   McnemarPValue  Sensitivity    </span></span>
<span><span class="co">##  Min.   :0.8188   Min.   :0.9456   Min.   :0     Min.   :0.9836  </span></span>
<span><span class="co">##  1st Qu.:0.8188   1st Qu.:0.9781   1st Qu.:0     1st Qu.:0.9857  </span></span>
<span><span class="co">##  Median :0.8188   Median :0.9837   Median :0     Median :0.9864  </span></span>
<span><span class="co">##  Mean   :0.8188   Mean   :0.9821   Mean   :0     Mean   :0.9863  </span></span>
<span><span class="co">##  3rd Qu.:0.8188   3rd Qu.:0.9895   3rd Qu.:0     3rd Qu.:0.9869  </span></span>
<span><span class="co">##  Max.   :0.8188   Max.   :0.9979   Max.   :0     Max.   :0.9889  </span></span>
<span><span class="co">##   Specificity      Pos Pred Value   Neg Pred Value     Precision     </span></span>
<span><span class="co">##  Min.   :0.01331   Min.   :0.8187   Min.   :0.1732   Min.   :0.8187  </span></span>
<span><span class="co">##  1st Qu.:0.01726   1st Qu.:0.8194   1st Qu.:0.2159   1st Qu.:0.8194  </span></span>
<span><span class="co">##  Median :0.01874   Median :0.8196   Median :0.2327   Median :0.8196  </span></span>
<span><span class="co">##  Mean   :0.01865   Mean   :0.8196   Mean   :0.2317   Mean   :0.8196  </span></span>
<span><span class="co">##  3rd Qu.:0.02022   3rd Qu.:0.8198   3rd Qu.:0.2484   3rd Qu.:0.8198  </span></span>
<span><span class="co">##  Max.   :0.02564   Max.   :0.8208   Max.   :0.3077   Max.   :0.8208  </span></span>
<span><span class="co">##      Recall             F1           Prevalence     Detection Rate  </span></span>
<span><span class="co">##  Min.   :0.9836   Min.   :0.8937   Min.   :0.8188   Min.   :0.8054  </span></span>
<span><span class="co">##  1st Qu.:0.9857   1st Qu.:0.8949   1st Qu.:0.8188   1st Qu.:0.8071  </span></span>
<span><span class="co">##  Median :0.9864   Median :0.8953   Median :0.8188   Median :0.8077  </span></span>
<span><span class="co">##  Mean   :0.9863   Mean   :0.8953   Mean   :0.8188   Mean   :0.8076  </span></span>
<span><span class="co">##  3rd Qu.:0.9869   3rd Qu.:0.8956   3rd Qu.:0.8188   3rd Qu.:0.8081  </span></span>
<span><span class="co">##  Max.   :0.9889   Max.   :0.8965   Max.   :0.8188   Max.   :0.8097  </span></span>
<span><span class="co">##  Detection Prevalence Balanced Accuracy</span></span>
<span><span class="co">##  Min.   :0.9830       Min.   :0.4996   </span></span>
<span><span class="co">##  1st Qu.:0.9848       1st Qu.:0.5018   </span></span>
<span><span class="co">##  Median :0.9854       Median :0.5025   </span></span>
<span><span class="co">##  Mean   :0.9854       Mean   :0.5025   </span></span>
<span><span class="co">##  3rd Qu.:0.9859       3rd Qu.:0.5033   </span></span>
<span><span class="co">##  Max.   :0.9878       Max.   :0.5064</span></span></code></pre>
<p>A quite different story is found when information on the probability
distribution of noise (null hypothesis) is added to the classifier:</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ft.g2p.dmps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">ft.</span>, nlms <span class="op">=</span> <span class="va">nlms.g2p</span>,  div.col <span class="op">=</span> <span class="fl">9L</span>, </span>
<span>                                tv.cut <span class="op">=</span> <span class="fl">0.68</span>, tv.col <span class="op">=</span> <span class="fl">7</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, </span>
<span>                                dist.name <span class="op">=</span> <span class="st">"Gamma2P"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ft.g2p.class</span> <span class="op">=</span> <span class="fu"><a href="../reference/evaluateDIMPclass.html">evaluateDIMPclass</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">ft.g2p.dmps</span>, control.names <span class="op">=</span> <span class="va">control.nam</span>,</span>
<span>                                treatment.names <span class="op">=</span> <span class="va">treatment.nam</span>,</span>
<span>                                column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, TV <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                            wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                                classifier <span class="op">=</span> <span class="st">"pca.lda"</span>, n.pc <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                                center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                num.boot <span class="op">=</span> <span class="fl">300</span>,</span>
<span>                                num.cores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html" class="external-link">multicoreWorkers</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                                output <span class="op">=</span> <span class="st">"all"</span>, </span>
<span>                                prop <span class="op">=</span> <span class="fl">0.6</span></span>
<span><span class="op">)</span></span>
<span><span class="va">ft.g2p.class</span><span class="op">$</span><span class="va">mc.val</span></span></code></pre></div>
<pre><code><span><span class="co">##     Accuracy     Kappa   AccuracyLower    AccuracyUpper  AccuracyNull   </span></span>
<span><span class="co">##  Min.   :1   Min.   :1   Min.   :0.9992   Min.   :1     Min.   :0.7375  </span></span>
<span><span class="co">##  1st Qu.:1   1st Qu.:1   1st Qu.:0.9992   1st Qu.:1     1st Qu.:0.7375  </span></span>
<span><span class="co">##  Median :1   Median :1   Median :0.9992   Median :1     Median :0.7375  </span></span>
<span><span class="co">##  Mean   :1   Mean   :1   Mean   :0.9992   Mean   :1     Mean   :0.7375  </span></span>
<span><span class="co">##  3rd Qu.:1   3rd Qu.:1   3rd Qu.:0.9992   3rd Qu.:1     3rd Qu.:0.7375  </span></span>
<span><span class="co">##  Max.   :1   Max.   :1   Max.   :0.9992   Max.   :1     Max.   :0.7375  </span></span>
<span><span class="co">##                                                                         </span></span>
<span><span class="co">##  AccuracyPValue McnemarPValue  Sensitivity  Specificity Pos Pred Value</span></span>
<span><span class="co">##  Min.   :0      Min.   : NA   Min.   :1    Min.   :1    Min.   :1     </span></span>
<span><span class="co">##  1st Qu.:0      1st Qu.: NA   1st Qu.:1    1st Qu.:1    1st Qu.:1     </span></span>
<span><span class="co">##  Median :0      Median : NA   Median :1    Median :1    Median :1     </span></span>
<span><span class="co">##  Mean   :0      Mean   :NaN   Mean   :1    Mean   :1    Mean   :1     </span></span>
<span><span class="co">##  3rd Qu.:0      3rd Qu.: NA   3rd Qu.:1    3rd Qu.:1    3rd Qu.:1     </span></span>
<span><span class="co">##  Max.   :0      Max.   : NA   Max.   :1    Max.   :1    Max.   :1     </span></span>
<span><span class="co">##                 NA's   :300                                           </span></span>
<span><span class="co">##  Neg Pred Value   Precision     Recall        F1      Prevalence    </span></span>
<span><span class="co">##  Min.   :1      Min.   :1   Min.   :1   Min.   :1   Min.   :0.7375  </span></span>
<span><span class="co">##  1st Qu.:1      1st Qu.:1   1st Qu.:1   1st Qu.:1   1st Qu.:0.7375  </span></span>
<span><span class="co">##  Median :1      Median :1   Median :1   Median :1   Median :0.7375  </span></span>
<span><span class="co">##  Mean   :1      Mean   :1   Mean   :1   Mean   :1   Mean   :0.7375  </span></span>
<span><span class="co">##  3rd Qu.:1      3rd Qu.:1   3rd Qu.:1   3rd Qu.:1   3rd Qu.:0.7375  </span></span>
<span><span class="co">##  Max.   :1      Max.   :1   Max.   :1   Max.   :1   Max.   :0.7375  </span></span>
<span><span class="co">##                                                                     </span></span>
<span><span class="co">##  Detection Rate   Detection Prevalence Balanced Accuracy</span></span>
<span><span class="co">##  Min.   :0.7375   Min.   :0.7375       Min.   :1        </span></span>
<span><span class="co">##  1st Qu.:0.7375   1st Qu.:0.7375       1st Qu.:1        </span></span>
<span><span class="co">##  Median :0.7375   Median :0.7375       Median :1        </span></span>
<span><span class="co">##  Mean   :0.7375   Mean   :0.7375       Mean   :1        </span></span>
<span><span class="co">##  3rd Qu.:0.7375   3rd Qu.:0.7375       3rd Qu.:1        </span></span>
<span><span class="co">##  Max.   :0.7375   Max.   :0.7375       Max.   :1        </span></span>
<span><span class="co">## </span></span></code></pre>
<p>Now, we add additional information about the optimal cutpoint</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ft.g2p_cutp.dmps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectDIMP.html">selectDIMP</a></span><span class="op">(</span><span class="va">ft.g2p.dmps</span>, div.col <span class="op">=</span> <span class="fl">9L</span>, </span>
<span>                                cutpoint <span class="op">=</span> <span class="va">cut.g2p</span><span class="op">$</span><span class="va">cutpoint</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ft.g2p_cut.class</span> <span class="op">=</span> <span class="fu"><a href="../reference/evaluateDIMPclass.html">evaluateDIMPclass</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">ft.g2p_cutp.dmps</span>, </span>
<span>                                control.names <span class="op">=</span> <span class="va">control.nam</span>,</span>
<span>                                treatment.names <span class="op">=</span> <span class="va">treatment.nam</span>,</span>
<span>                                column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, TV <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                            wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                                classifier <span class="op">=</span> <span class="st">"pca.lda"</span>, n.pc <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                                center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>, num.boot <span class="op">=</span> <span class="fl">300</span>,</span>
<span>                                output <span class="op">=</span> <span class="st">"all"</span>, prop <span class="op">=</span> <span class="fl">0.6</span></span>
<span><span class="op">)</span></span>
<span><span class="va">ft.g2p_cut.class</span><span class="op">$</span><span class="va">mc.val</span></span></code></pre></div>
<pre><code><span><span class="co">##     Accuracy     Kappa   AccuracyLower    AccuracyUpper  AccuracyNull   </span></span>
<span><span class="co">##  Min.   :1   Min.   :1   Min.   :0.9991   Min.   :1     Min.   :0.9139  </span></span>
<span><span class="co">##  1st Qu.:1   1st Qu.:1   1st Qu.:0.9991   1st Qu.:1     1st Qu.:0.9139  </span></span>
<span><span class="co">##  Median :1   Median :1   Median :0.9991   Median :1     Median :0.9139  </span></span>
<span><span class="co">##  Mean   :1   Mean   :1   Mean   :0.9991   Mean   :1     Mean   :0.9139  </span></span>
<span><span class="co">##  3rd Qu.:1   3rd Qu.:1   3rd Qu.:0.9991   3rd Qu.:1     3rd Qu.:0.9139  </span></span>
<span><span class="co">##  Max.   :1   Max.   :1   Max.   :0.9991   Max.   :1     Max.   :0.9139  </span></span>
<span><span class="co">##                                                                         </span></span>
<span><span class="co">##  AccuracyPValue       McnemarPValue  Sensitivity  Specificity Pos Pred Value</span></span>
<span><span class="co">##  Min.   :9.096e-154   Min.   : NA   Min.   :1    Min.   :1    Min.   :1     </span></span>
<span><span class="co">##  1st Qu.:9.096e-154   1st Qu.: NA   1st Qu.:1    1st Qu.:1    1st Qu.:1     </span></span>
<span><span class="co">##  Median :9.096e-154   Median : NA   Median :1    Median :1    Median :1     </span></span>
<span><span class="co">##  Mean   :9.096e-154   Mean   :NaN   Mean   :1    Mean   :1    Mean   :1     </span></span>
<span><span class="co">##  3rd Qu.:9.096e-154   3rd Qu.: NA   3rd Qu.:1    3rd Qu.:1    3rd Qu.:1     </span></span>
<span><span class="co">##  Max.   :9.096e-154   Max.   : NA   Max.   :1    Max.   :1    Max.   :1     </span></span>
<span><span class="co">##                       NA's   :300                                           </span></span>
<span><span class="co">##  Neg Pred Value   Precision     Recall        F1      Prevalence    </span></span>
<span><span class="co">##  Min.   :1      Min.   :1   Min.   :1   Min.   :1   Min.   :0.9139  </span></span>
<span><span class="co">##  1st Qu.:1      1st Qu.:1   1st Qu.:1   1st Qu.:1   1st Qu.:0.9139  </span></span>
<span><span class="co">##  Median :1      Median :1   Median :1   Median :1   Median :0.9139  </span></span>
<span><span class="co">##  Mean   :1      Mean   :1   Mean   :1   Mean   :1   Mean   :0.9139  </span></span>
<span><span class="co">##  3rd Qu.:1      3rd Qu.:1   3rd Qu.:1   3rd Qu.:1   3rd Qu.:0.9139  </span></span>
<span><span class="co">##  Max.   :1      Max.   :1   Max.   :1   Max.   :1   Max.   :0.9139  </span></span>
<span><span class="co">##                                                                     </span></span>
<span><span class="co">##  Detection Rate   Detection Prevalence Balanced Accuracy</span></span>
<span><span class="co">##  Min.   :0.9139   Min.   :0.9139       Min.   :1        </span></span>
<span><span class="co">##  1st Qu.:0.9139   1st Qu.:0.9139       1st Qu.:1        </span></span>
<span><span class="co">##  Median :0.9139   Median :0.9139       Median :1        </span></span>
<span><span class="co">##  Mean   :0.9139   Mean   :0.9139       Mean   :1        </span></span>
<span><span class="co">##  3rd Qu.:0.9139   3rd Qu.:0.9139       3rd Qu.:1        </span></span>
<span><span class="co">##  Max.   :0.9139   Max.   :0.9139       Max.   :1        </span></span>
<span><span class="co">## </span></span></code></pre>
<p>In other words, information on the probability distributions of the
natural spontaneous methylation variation in the control and treatment
population are essential to discriminate the background noise from the
treatment induced signal.</p>
</div>
</div>
<div class="section level3">
<h3 id="graphics-of-dmp-classification-performance">Graphics of DMP classification performance<a class="anchor" aria-label="anchor" href="#graphics-of-dmp-classification-performance"></a>
</h3>
<p>DMP count data:</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span>G2P <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">g2p.dmps</span>, <span class="va">length</span><span class="op">)</span>,</span>
<span>            G3P <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">g3p.dmps</span>, <span class="va">length</span><span class="op">)</span>,</span>
<span>            FT <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">ft.dmps</span>, <span class="va">length</span><span class="op">)</span>,</span>
<span>            FT.G2P <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">ft.g2p.dmps</span>, <span class="va">length</span><span class="op">)</span>,</span>
<span>            FT.SD <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">ft.g2p_cutp.dmps</span>, <span class="va">length</span><span class="op">)</span></span>
<span>    <span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Found more than one class "pDMP" in cache; using the first, from namespace 'MethylIT'</span></span></code></pre>
<pre><code><span><span class="co">## Also defined by 'MethylIT.utils'</span></span></code></pre>
<pre><code><span><span class="co">## Found more than one class "pDMP" in cache; using the first, from namespace 'MethylIT'</span></span></code></pre>
<pre><code><span><span class="co">## Also defined by 'MethylIT.utils'</span></span></code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dt</span></span></code></pre></div>
<pre><code><span><span class="co">##     G2P  G3P   FT FT.G2P FT.SD</span></span>
<span><span class="co">## C1  255  251 1730   1055   255</span></span>
<span><span class="co">## C2  306  298 1682   1070   306</span></span>
<span><span class="co">## C3  281  276 1657   1059   281</span></span>
<span><span class="co">## T1 2925 2870 7589   2926  2925</span></span>
<span><span class="co">## T2 3032 2961 7646   3032  3032</span></span>
<span><span class="co">## T3 2990 2935 7679   2990  2990</span></span></code></pre>
<p>The comparison between the approaches FT.G2P and FT.SD (full signal
detection on FT output) tells us that only 255, 306, and 281 cytosine
sites detected with FT in the control samples C1, C2, and C3,
respectively, carry methylation signals comparable (in magnitude) to
those signals induced by the treatment.</p>
<p>Classification performance data:</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FT"</span>, <span class="st">"FT.G2P"</span>, <span class="st">"FT.SD"</span>, <span class="st">"G2P"</span>, <span class="st">"G3P"</span><span class="op">)</span>, </span>
<span>                <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colMeans</a></span><span class="op">(</span><span class="va">ft.class</span><span class="op">$</span><span class="va">boots</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">8</span><span class="op">:</span><span class="fl">11</span>, <span class="fl">18</span><span class="op">)</span><span class="op">]</span>, </span>
<span>                        FDR <span class="op">=</span> <span class="va">ft.class</span><span class="op">$</span><span class="va">con.mat</span><span class="op">$</span><span class="va">FDR</span><span class="op">)</span>,</span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colMeans</a></span><span class="op">(</span><span class="va">ft.g2p.class</span><span class="op">$</span><span class="va">boots</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">8</span><span class="op">:</span><span class="fl">11</span>, <span class="fl">18</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                        FDR <span class="op">=</span> <span class="va">ft.g2p.class</span><span class="op">$</span><span class="va">con.mat</span><span class="op">$</span><span class="va">FDR</span><span class="op">)</span>,</span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colMeans</a></span><span class="op">(</span><span class="va">ft.g2p_cut.class</span><span class="op">$</span><span class="va">boots</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">8</span><span class="op">:</span><span class="fl">11</span>, <span class="fl">18</span><span class="op">)</span><span class="op">]</span>, </span>
<span>                        FDR <span class="op">=</span> <span class="va">ft.g2p_cut.class</span><span class="op">$</span><span class="va">con.mat</span><span class="op">$</span><span class="va">FDR</span><span class="op">)</span>,</span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colMeans</a></span><span class="op">(</span><span class="va">g2p.class</span><span class="op">$</span><span class="va">boots</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">8</span><span class="op">:</span><span class="fl">11</span>, <span class="fl">18</span><span class="op">)</span><span class="op">]</span>, </span>
<span>                        FDR <span class="op">=</span> <span class="va">g2p.class</span><span class="op">$</span><span class="va">con.mat</span><span class="op">$</span><span class="va">FDR</span><span class="op">)</span>,</span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colMeans</a></span><span class="op">(</span><span class="va">g3p.class</span><span class="op">$</span><span class="va">boots</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">8</span><span class="op">:</span><span class="fl">11</span>, <span class="fl">18</span><span class="op">)</span><span class="op">]</span>, </span>
<span>                        FDR <span class="op">=</span> <span class="va">g3p.class</span><span class="op">$</span><span class="va">con.mat</span><span class="op">$</span><span class="va">FDR</span><span class="op">)</span></span>
<span>                <span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Graphics:</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">color</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"darkgreen"</span>, <span class="st">"#147714FF"</span>, <span class="st">"#3D9F3DFF"</span>, <span class="st">"#66C666FF"</span>,  <span class="st">"#90EE90FF"</span><span class="op">)</span></span>
<span><span class="va">dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">dt</span>, sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">g2p.dmps</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## ------------------------- DMP count graphic ---------------------------------</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, lwd <span class="op">=</span> <span class="fl">0.1</span>, cex <span class="op">=</span> <span class="fl">1</span>, mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">5</span>,<span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span>, mfcol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">FT</span>, <span class="va">FT.G2P</span>, <span class="va">FT.SD</span>, <span class="va">G2P</span>, <span class="va">G3P</span><span class="op">)</span> <span class="op">~</span> <span class="va">sample</span>, </span>
<span>        panel.first<span class="op">=</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, pch<span class="op">=</span><span class="fl">16</span>, cex<span class="op">=</span><span class="fl">1e6</span>, col<span class="op">=</span><span class="st">"grey95"</span><span class="op">)</span></span>
<span>            <span class="fu"><a href="https://rdrr.io/r/graphics/grid.html" class="external-link">grid</a></span><span class="op">(</span>col<span class="op">=</span><span class="st">"white"</span>, lty <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">}</span>,</span>
<span>        data <span class="op">=</span> <span class="va">dt</span>, beside <span class="op">=</span> <span class="cn">TRUE</span>,  legend.text <span class="op">=</span> <span class="cn">TRUE</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>        yaxt <span class="op">=</span> <span class="st">"n"</span>, cex.names <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span>, xlab <span class="op">=</span> <span class="st">""</span>, col <span class="op">=</span> <span class="va">color</span>,</span>
<span>        args.legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>x<span class="op">=</span><span class="fl">10</span>, y<span class="op">=</span><span class="fl">6000</span>, text.font <span class="op">=</span> <span class="fl">2</span>, box.lwd <span class="op">=</span> <span class="fl">0</span>, </span>
<span>                        horiz <span class="op">=</span> <span class="cn">FALSE</span>, adj <span class="op">=</span> <span class="fl">0</span>, xjust <span class="op">=</span> <span class="fl">0.65</span>, yjust <span class="op">=</span> <span class="fl">0.8</span>, </span>
<span>                        bty <span class="op">=</span> <span class="st">"n"</span>, cex <span class="op">=</span> <span class="fl">1.2</span>, x.intersp <span class="op">=</span> <span class="fl">0.2</span>, inset <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, </span>
<span>                        ncol <span class="op">=</span> <span class="fl">1</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></span><span class="op">(</span><span class="fl">2</span>, hadj <span class="op">=</span> <span class="fl">0.9</span>, las <span class="op">=</span> <span class="fl">2</span>, lwd <span class="op">=</span> <span class="fl">0.4</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.02</span>, cex.axis <span class="op">=</span> <span class="fl">1.2</span>, font <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    line <span class="op">=</span> <span class="op">-</span><span class="fl">0.2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html" class="external-link">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>, text <span class="op">=</span> <span class="st">"DMP count"</span>, line <span class="op">=</span> <span class="fl">3</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## ------------------ DMP classifiction performance graphic -------------------</span></span>
<span><span class="va">color</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mediumblue"</span>, <span class="st">"#0000FFFF"</span>, <span class="st">"#3949F6FF"</span>, <span class="st">"#566CF2FF"</span>, <span class="st">"#7390EEFF"</span>,</span>
<span>            <span class="st">"#90B3EAFF"</span>, <span class="st">"#ADD8E6FF"</span><span class="op">)</span></span>
<span><span class="va">labs</span> <span class="op">&lt;-</span> <span class="va">df</span><span class="op">$</span><span class="va">method</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, lwd <span class="op">=</span> <span class="fl">0.1</span>, cex <span class="op">=</span> <span class="fl">1</span>, mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">Accuracy</span>, <span class="va">Sensitivity</span>, <span class="va">Specificity</span>, <span class="va">Pos.Pred.Value</span>, </span>
<span>                    <span class="va">Neg.Pred.Value</span>, <span class="va">Balanced.Accuracy</span>, <span class="va">FDR</span><span class="op">)</span> <span class="op">~</span> <span class="va">method</span>, </span>
<span>            panel.first <span class="op">=</span> <span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, pch<span class="op">=</span><span class="fl">16</span>, cex<span class="op">=</span><span class="fl">1e6</span>, col<span class="op">=</span><span class="st">"grey95"</span><span class="op">)</span></span>
<span>                            <span class="fu"><a href="https://rdrr.io/r/graphics/grid.html" class="external-link">grid</a></span><span class="op">(</span>col<span class="op">=</span><span class="st">"white"</span>, lty <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">}</span>, </span>
<span>            data <span class="op">=</span> <span class="va">df</span>, beside <span class="op">=</span> <span class="cn">TRUE</span>,  legend.text <span class="op">=</span> <span class="cn">TRUE</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>            yaxt <span class="op">=</span> <span class="st">"n"</span>, cex.names <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span>, xlab <span class="op">=</span> <span class="st">""</span>, col <span class="op">=</span> <span class="va">color</span>, </span>
<span>            ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span><span class="op">)</span>, </span>
<span>            args.legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fl">52</span>, y <span class="op">=</span> <span class="fl">1.</span>, text.font <span class="op">=</span> <span class="fl">2</span>, box.lwd <span class="op">=</span> <span class="fl">0</span>, </span>
<span>                            horiz <span class="op">=</span> <span class="cn">FALSE</span>, adj <span class="op">=</span> <span class="fl">0</span>, xjust <span class="op">=</span> <span class="fl">0.65</span>, yjust <span class="op">=</span> <span class="fl">0.8</span>, </span>
<span>                            bty <span class="op">=</span> <span class="st">"n"</span>, cex <span class="op">=</span> <span class="fl">1.2</span>, x.intersp <span class="op">=</span> <span class="fl">0.2</span>, inset <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, </span>
<span>                            ncol <span class="op">=</span> <span class="fl">1</span>, fill <span class="op">=</span> <span class="va">color</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></span><span class="op">(</span><span class="fl">2</span>, hadj <span class="op">=</span> <span class="fl">0.8</span>, las <span class="op">=</span> <span class="fl">2</span>, lwd <span class="op">=</span> <span class="fl">0.4</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.02</span>, cex.axis <span class="op">=</span> <span class="fl">1.2</span>, font <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    line <span class="op">=</span> <span class="op">-</span><span class="fl">0.4</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html" class="external-link">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>, text <span class="op">=</span> <span class="st">"Performance value"</span>, line <span class="op">=</span> <span class="fl">2</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="Cutpoint_estimation_with_Methyl-IT_files/figure-html/dmp_graph-1.png" width="1344"></p>
<p>FT.G2P and FT.SD approaches lead to excellent classification
performances on this data set. At this point, we can appeal the
parsimony principle, follows from <a href="https://www.britannica.com/topic/Occams-razor" class="external-link">Occam’s razor</a>
that states “among competing hypotheses, the hypothesis with the fewest
assumptions should be selected. In other words, results indicates that
the signal-detection and machine-learning approach is sufficient <span class="citation">(7, 9)</span>.</p>
</div>
<div class="section level2">
<h2 id="conclusions">Conclusions<a class="anchor" aria-label="anchor" href="#conclusions"></a>
</h2>
<ol style="list-style-type: decimal">
<li><p>A proper discrimination of the methylation signal from the
stochastic methylation background requires for the knowledge of
probability distributions of the methylation signal from control and
treatment population. Such a knowledge permits a suitable estimation of
the cutoff value to discriminate the methylation signal induced by the
treatment from the stochastic methylation background detected in the
control group.</p></li>
<li><p>It does not matter how significant a differentially methylation
event for a given cytosine position would be (after the application of
some statistical test), but on how big the probability to be observed in
the control group is. In simple words, if for a given DMP the
probability of to be observed in the control is <em>big enough</em>,
then such a DMP did not result from a treatment effect.</p></li>
<li><p>A suitable evaluation on how much the mentioned probability can
be <em>big enough</em> derives by estimating an optimal cutpoint. But a
classification into two groups results from the cutpoint estimation and
the problem on the estimation of such a cutpoint is equivalent to find a
discriminatory function (as set by Fisher, <span class="citation">(10,
11)</span>). Cases with function values below some cutoff are classified
in one group, while values above the cutoff are put in the other
group.</p></li>
<li><p>MethylIT function <a href="https://is.gd/RR9iWY" class="external-link">estimateCutPoint</a> permits the estimation
and search for an optimal cutpoint by confronting the problem as in the
spirit of the classical signal detection and as a classification
problem. The best model classifier will depend on the dataset under
study.So, uses must check for which is the model classifier with the
best classification performance for his/her dataset.</p></li>
</ol>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body">
<div id="ref-Ngo2016" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">T.
T. M. Ngo, <em>et al.</em>, <a href="https://doi.org/10.1038/ncomms10813" class="external-link"><span class="nocase">Effects
of cytosine modifications on DNA flexibility and nucleosome mechanical
stability</span></a>. <em>Nature Communications</em> <strong>7</strong>,
10813 (2016).</div>
</div>
<div id="ref-Min2005" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">W.
Min, <em>et al.</em>, <a href="https://doi.org/10.1021/nl0521773" class="external-link"><span class="nocase">Nonequilibrium steady state of a nanometric biochemical
system: Determining the thermodynamic driving force from single enzyme
turnover time traces</span></a>. <em>Nano Letters</em>
<strong>5</strong>, 2373–2378 (2005).</div>
</div>
<div id="ref-Koslover2012" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">E.
F. Koslover, A. J. Spakowitz, <a href="https://www.ncbi.nlm.nih.gov/pubmed/23005451" class="external-link"><span class="nocase">Force fluctuations impact kinetics of biomolecular
systems</span></a>. <em>Physical review. E, Statistical, nonlinear, and
soft matter physics</em> <strong>86</strong>, 011906 (2012).</div>
</div>
<div id="ref-Samoilov2006" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">M.
S. Samoilov, G. Price, A. P. Arkin, <a href="https://doi.org/10.1126/stke.3662006re17" class="external-link"><span class="nocase">From fluctuations to phenotypes: the physiology of
noise.</span></a> <em>Science’s STKE : signal transduction knowledge
environment</em> <strong>2006</strong> (2006).</div>
</div>
<div id="ref-Eldar2010" class="csl-entry">
<div class="csl-left-margin">5. </div>
<div class="csl-right-inline">A.
Eldar, M. B. Elowitz, <a href="https://doi.org/10.1038/nature09326" class="external-link"><span class="nocase">Functional roles for noise in genetic
circuits</span></a>. <strong>467</strong>, 167–173 (2010).</div>
</div>
<div id="ref-Sanchez2016" class="csl-entry">
<div class="csl-left-margin">6. </div>
<div class="csl-right-inline">R.
Sanchez, S. A. Mackenzie, <a href="https://doi.org/10.1371/journal.pone.0150427" class="external-link"><span class="nocase">Information Thermodynamics of Cytosine DNA
Methylation</span></a>. <em>PLOS ONE</em> <strong>11</strong>, e0150427
(2016).</div>
</div>
<div id="ref-Sanchez2019" class="csl-entry">
<div class="csl-left-margin">7. </div>
<div class="csl-right-inline">R.
Sanchez, X. Yang, T. Maher, S. Mackenzie, <a href="https://doi.org/10.3390/ijms20215343" class="external-link"><span class="nocase">Discrimination of DNA Methylation Signal from Background
Variation for Clinical Diagnostics</span></a>. <em>Int. J. Mol.
Sci.</em> <strong>20</strong>, 5343 (2019).</div>
</div>
<div id="ref-Youden1950" class="csl-entry">
<div class="csl-left-margin">8. </div>
<div class="csl-right-inline">W.
J. Youden, <a href="https://doi.org/10.1002/1097-0142(1950)3:1&lt;32::AID-CNCR2820030106&gt;3.0.CO;2-3" class="external-link"><span class="nocase">Index for rating diagnostic tests</span></a>.
<em>Cancer</em> <strong>3</strong>, 32–35 (1950).</div>
</div>
<div id="ref-ElNaqa2018" class="csl-entry">
<div class="csl-left-margin">9. </div>
<div class="csl-right-inline">I.
El Naqa, <em>et al.</em>, <a href="https://doi.org/10.1002/mp.12811" class="external-link"><span class="nocase">Machine
learning and modeling: Data, validation, communication
challenges</span></a>. <em>Medical Physics</em> <strong>45</strong>,
e834–e840 (2018).</div>
</div>
<div id="ref-Fisher1938" class="csl-entry">
<div class="csl-left-margin">10. </div>
<div class="csl-right-inline">R.
A. Fisher, <span class="nocase">The statistical utilization of multiple
measurents</span>. <em>Annals of Eugenics</em> <strong>8</strong>,
376–386 (1938).</div>
</div>
<div id="ref-Green1979" class="csl-entry">
<div class="csl-left-margin">11. </div>
<div class="csl-right-inline">B.
F. Green, <span class="nocase">The Two Kinds of Linear Discriminant
Functions and Their Relationship</span>. <em>Journal of Educational
Statistics</em> <strong>4</strong>, 247–263 (1979).</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
