<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>DMR detection with Methyl-IT • MethylIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="DMR detection with Methyl-IT">
<meta property="og:description" content="MethylIT">
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/MethylIT/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="dmr_at_memory_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>DMR detection with Methyl-IT</h1>
            <h3 class="subtitle">Analysis for Arabidopsis MSH1-memory versus non-memory first generation</h3>
                        <h4 class="author"><div class="line-block">Robersy Sanchez<br><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a><br>
Mackenzie’s lab<br><br>
Department of Biology and Plant Science.<br>
Pennsylvania State University, University Park, PA 16802<br>
Maintainer: Robersy Sanchez</div></h4>
            
            <h4 class="date">04 November 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/master/vignettes/dmr_at_memory.Rmd"><code>vignettes/dmr_at_memory.Rmd</code></a></small>
      <div class="hidden name"><code>dmr_at_memory.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Herein, an example of Methyl-IT application to the detection of differential methylated regions (DMRs) on first generation MSH1 RNAi transgene segregants data set is presented.</p>
    </div>
    
<style type="text/css">

.list-group-item.active, .list-group-item.active:focus,
.list-group-item.active:hover {
    z-index: 2;
    color: #fff;
    background-color: #337ab7;
    border-color: #337ab7;
}

.tocify-subheader .tocify-item {
    font-size: 0.80em;
    padding-left: 25px;
    text-indent: 0;
}

.tocify-header .tocify-item {
    font-size: 0.85em;
    padding-left: 20px;
    text-indent: 0;
}

body{ /* Normal */
    font-size: 18px;
    font-family: "Times New Roman", Times, serif;
    text-align: justify
}
td { /* Table */
    font-size: 8px;
}

h1.title {
    font-size: 38px;
    font-family: "Times New Roman", Times, serif;
    color: DarkRed;
    .subTitle {
        font-size: 24px;
        font-family: "Times New Roman", Times, serif;
        color: DarkRed;
    }
}

h1 { /* Header 1 */
    font-size: 28px;
    font-family: "Times New Roman", Times, serif;
    color: DarkBlue;
}
h2 { /* Header 2 */
    font-size: 22px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
h3 { /* Header 3 */
    font-size: 18px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
code.r{ /* Code block */
    font-size: 12px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>MethylIT is an R package for methylome analysis based on a signal-detection machine learning approach (SD-ML). This approach is postulated to provide greater sensitivity for resolving true signal from the methylation background within the methylome <span class="citation">(1, 2)</span>. Because the biological signal created within the dynamic methylome environment characteristic of plants is not free from background noise, the approach, designated Methyl-IT, includes the application of signal detection theory <span class="citation">(3–6)</span>.</p>
<p>A basic requirement for the application of signal detection is the knowledge of the probability distribution of the background noise, which is used as null hypothesis in the detection of the methylation signal. This probability distribution is, in general, a member of generalized gamma distribution family, and it can be deduced on a statistical mechanical/thermodynamics basis from DNA methylation induced by thermal fluctuations <span class="citation">(1)</span>.</p>
<p>Herein, we provide an example of Methyl-IT application to DMR estimation on patients with PALL <span class="citation">(7)</span>. Due to the size of human methylome the current example only covers the analysis of chromosome 9. A full description of Methyl-IT application of methylome analysis for Clinical Diagnostics is given in the manuscript <span class="citation">(2)</span>. A wider analysis of PALL data set is given in <span class="citation">(8)</span>.</p>
<p>The first step of DMR estimation follows the general step of Methyl-IT pipeline for the estimation of DMPs. In Methyl-IT, a DMP is a cytosine positions that, with high probability, carries a statistically significant methylation change in respect to the same position in a control population. Once DMPs are identified, a simple algorithm can be applied to identify clusters of DMPs using <a href="https://genomaths.github.io/methylit/reference/dmpClusters.html">dmpClusters</a> function. Next, <a href="https://is.gd/kltCXl">countTest2</a> is applied to test whether a given cluster of DMPs is a DMR.</p>
<p>It is worthy to notices that any clustering algorithm will impose non-natural constraints on the methylation patterning, which not necessarily will be associated to the local DNA physico-chemical properties and to local molecular evolutionary features of every DMR detected. Every cytosine methylation has a concrete physical effect of the DNA, altering the DNA persistence length, i.e., altering the local DNA flexibility <span class="citation">(9–11)</span>. DNA flexibility was shown to affect the binding of proteins to methylated DNA and DNA-containing lesion <span class="citation">(12, 13)</span> and nucleosome mechanical stability <span class="citation">(14)</span>. In other words, the sizes of naturally occurrying methylation motifs are not arbitrarily set, but very well constrained under molecular evolutionary pressure, probably, particular for each species.</p>
<p>The R packages required in this example are:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R">        <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/MethylIT">MethylIT</a></span><span class="op">)</span>
        <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">MethylIT.utils</span><span class="op">)</span>
        <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span> <span class="co"># graphic</span>
        <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">grid</span><span class="op">)</span> <span class="co"># For multiple plots</span>
        <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span> <span class="co"># For multiple plots</span>
        <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span> <span class="co"># To import gene annotation</span></code></pre></div>
</div>
<div id="estimation-of-differentially-methylated-regions-dmrs" class="section level1">
<h1 class="hasAnchor">
<a href="#estimation-of-differentially-methylated-regions-dmrs" class="anchor"></a>Estimation of differentially methylated regions (DMRs)</h1>
<p>DMP data set is available for download.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## ===== Download methylation data from Github ========</span>
<span class="va">url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"https://github.com/genomaths/genomaths.github.io/raw/master/AT_memory/"</span>, 
              <span class="st">"dmps_mm-vs-nm_all-context_gen1_nmRef_min-meth-3_cov-4_04-16-20.RData"</span><span class="op">)</span>
<span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".RData"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="va">url</span>, destfile <span class="op">=</span> <span class="va">temp</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.remove</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">temp</span>, <span class="va">url</span><span class="op">)</span></code></pre></div>
<p>The gene annotation</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## ===== Download methylation data from GitHub ========</span>
<span class="va">url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"https://github.com/genomaths/genomaths.github.io/raw/master/"</span>, 
              <span class="st">"Annotation/Arabidopsis_thaliana_genes_TAIR10.38.gtf.RData"</span><span class="op">)</span>
<span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".RData"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="va">url</span>, destfile <span class="op">=</span> <span class="va">temp</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.remove</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">temp</span>, <span class="va">url</span><span class="op">)</span>;

<span class="va">GENES</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">type</span> <span class="op">==</span> <span class="st">"gene"</span><span class="op">]</span>

<span class="co">## To get exons and introns</span>
<span class="va">exons</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">]</span>
<span class="va">introns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/setops-methods.html">setdiff</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">genes</span>, y <span class="op">=</span> <span class="va">exons</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co">## To add the corresponding gene names to introns</span>
<span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html">findOverlaps</a></span><span class="op">(</span><span class="va">introns</span>, <span class="va">genes</span>, type <span class="op">=</span> <span class="st">"within"</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">introns</span> <span class="op">&lt;-</span> <span class="va">introns</span><span class="op">[</span><span class="fu">queryHits</span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span>
<span class="fu">mcols</span><span class="op">(</span><span class="va">introns</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">mcols</span><span class="op">(</span><span class="va">genes</span><span class="op">[</span><span class="fu">subjectHits</span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<div id="cluster-of-dmps" class="section level2">
<h2 class="hasAnchor">
<a href="#cluster-of-dmps" class="anchor"></a>Cluster of DMPs</h2>
<p>Function <a href="https://is.gd/K2cRgZ">dmpClusters</a> search for cluster of DMPs found from all the samples. The basic idea reside in that if a given cluster of DMP is a DMR, then it will be consistently differentially methylated in all treatment samples, and we can identify them applying function <a href="https://%20is.gd/kltCXl">countTest2</a>.</p>
<p>There is not a unique approach to build DMP clusters, since several clustering criteria can be applied. Herein, we use the two approach implemented in <a href="https://is.gd/K2cRgZ">dmpClusters</a> function (follow the link or type ?dmpClusters to see the help for the function).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">clust_50_100</span> <span class="op">=</span> <span class="fu"><a href="../reference/dmpClusters.html">dmpClusters</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, maxDist <span class="op">=</span> <span class="fl">50</span>, minNumDMPs <span class="op">=</span> <span class="fl">10</span>, 
                           maxClustDist <span class="op">=</span> <span class="fl">100</span>, method <span class="op">=</span> <span class="st">"fixed.int"</span>,
                           num.cores <span class="op">=</span> <span class="fl">1L</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="va">clust_50_100</span></code></pre></div>
<pre><code>## GRanges object with 18035 ranges and 1 metadata column:
##           seqnames            ranges strand |      dmps
##              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##       [1]        1           183-330      * |        27
##       [2]        1           647-809      * |        10
##       [3]        1       18052-18224      * |        14
##       [4]        1       18643-18746      * |        12
##       [5]        1       42279-42423      * |        40
##       ...      ...               ...    ... .       ...
##   [18031]        5 26670460-26670651      * |        15
##   [18032]        5 26788467-26788575      * |        11
##   [18033]        5 26945319-26945446      * |        11
##   [18034]        5 26974071-26974175      * |        10
##   [18035]        5 26974820-26975362      * |        39
##   -------
##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
<p>DMPs clusterization strongly depends on the maximum distance permitted between two consecutive DMPs that belong to the same cluster (parameter: <code>minNumDMPs</code>) and on the minimum number of DMPs that must be present on a detected agglomeration of DMPs to be considered a cluster (parameter: <code>maxDist</code>).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">.clust_50_100</span> <span class="op">=</span> <span class="fu"><a href="../reference/dmpClusters.html">dmpClusters</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, maxDist <span class="op">=</span> <span class="fl">50</span>, minNumDMPs <span class="op">=</span> <span class="fl">10</span>, 
                           maxClustDist <span class="op">=</span> <span class="fl">100</span>, method <span class="op">=</span> <span class="st">"relaxed"</span>,
                           num.cores <span class="op">=</span> <span class="fl">1L</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">.clust_50_100</span></code></pre></div>
<pre><code>## GRanges object with 17528 ranges and 1 metadata column:
##           seqnames            ranges strand |      dmps
##              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##       [1]        1            94-330      * |        29
##       [2]        1           647-880      * |        13
##       [3]        1       17833-18512      * |        35
##       [4]        1       18643-18746      * |        12
##       [5]        1       42279-42423      * |        40
##       ...      ...               ...    ... .       ...
##   [17524]        5 26788399-26788575      * |        12
##   [17525]        5 26835058-26835302      * |        13
##   [17526]        5 26945319-26945446      * |        11
##   [17527]        5 26974071-26974329      * |        14
##   [17528]        5 26974728-26975362      * |        40
##   -------
##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
<p>Alternatively, we can use function <em>dmpClustering</em> from MethylIT.utils R package. Notice that although <em>‘method’</em> parameter are named the same, on each function these stand for different approaches.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">clust.50.100</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/dmpClustering.html">dmpClustering</a></span><span class="op">(</span>dmps <span class="op">=</span> <span class="va">dmps</span>, minNumDMPs <span class="op">=</span> <span class="fl">10</span>, win.size <span class="op">=</span> <span class="fl">50</span>,
                            step.size <span class="op">=</span> <span class="fl">50</span>, method <span class="op">=</span> <span class="st">"fixed.int"</span>, 
                            maxClustDist <span class="op">=</span> <span class="fl">100</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">clust.50.100</span></code></pre></div>
<pre><code>## GRanges object with 15521 ranges and 1 metadata column:
##           seqnames            ranges strand |      dmps
##              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##       [1]        1            94-330      * |        29
##       [2]        1           647-880      * |        13
##       [3]        1       17833-18746      * |        47
##       [4]        1       42279-42423      * |        40
##       [5]        1       44603-44848      * |        19
##       ...      ...               ...    ... .       ...
##   [15517]        5 26835058-26835302      * |        13
##   [15518]        5 26891628-26892012      * |        10
##   [15519]        5 26945319-26945446      * |        11
##   [15520]        5 26974071-26974329      * |        14
##   [15521]        5 26974728-26975362      * |        40
##   -------
##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">.clust.50.100</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/dmpClustering.html">dmpClustering</a></span><span class="op">(</span>dmps <span class="op">=</span> <span class="va">dmps</span>, minNumDMPs <span class="op">=</span> <span class="fl">10</span>, method <span class="op">=</span> <span class="st">"relaxed"</span>, 
                            maxClustDist <span class="op">=</span> <span class="fl">100</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">.clust.50.100</span></code></pre></div>
<pre><code>## GRanges object with 17647 ranges and 1 metadata column:
##           seqnames            ranges strand |      dmps
##              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
##       [1]        1            94-330      * |        29
##       [2]        1           647-880      * |        13
##       [3]        1       17833-18512      * |        35
##       [4]        1       18643-18746      * |        12
##       [5]        1       42279-42423      * |        40
##       ...      ...               ...    ... .       ...
##   [17643]        5 26788399-26788575      * |        12
##   [17644]        5 26835058-26835302      * |        13
##   [17645]        5 26945319-26945446      * |        11
##   [17646]        5 26974071-26974329      * |        14
##   [17647]        5 26974728-26975362      * |        40
##   -------
##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="dmr-detection" class="section level2">
<h2 class="hasAnchor">
<a href="#dmr-detection" class="anchor"></a>DMR detection</h2>
<p>The count of the number of DMPs on each region will be accomplished with <a href="https://genomaths.github.io/methylit/reference/getDMPatRegions.html">getDMPatRegions</a> function. This function operates based on <a href="https://is.gd/zJ7U9l"><em>findOverlaps</em></a> function from the ‘<em>GenomicRanges</em>’ Bioconductor R package. The <a href="https://is.gd/zJ7U9l"><em>findOverlaps</em></a> function has several critical parameters like, for example, ‘<em>maxgap</em>,’ ‘<em>minoverlap</em>,’ and ‘<em>ignore.strand</em>.’ In our <a href="https://genomaths.github.io/methylit/reference/getDMPatRegions.html">getDMPatRegions</a> function, except for setting ignore.strand = TRUE and <em>type = “within”</em>, we preserve the rest of default ‘findOverlaps’ parameters.</p>
<div id="experiment-design-for-each-clustering-approach" class="section level3">
<h3 class="hasAnchor">
<a href="#experiment-design-for-each-clustering-approach" class="anchor"></a>Experiment design for each clustering approach</h3>
<p>Design for method = “fixed.int” from dmpClusters function</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span>
<span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span>
<span class="va">dmps_at_clust_50_100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDMPatRegions.html">getDMPatRegions</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, regions <span class="op">=</span> <span class="va">clust_50_100</span>,
                                       ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">dmps_at_clust_50_100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">dmps_at_clust_50_100</span>, columns <span class="op">=</span> <span class="fl">2L</span>,
                                      ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>, type <span class="op">=</span> <span class="st">'equal'</span>,
                                      verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dmps_at_clust_50_100</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">nams</span>

<span class="va">colData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"MM"</span>, <span class="st">"MM"</span>,<span class="st">"MM"</span>,<span class="st">"MM"</span>, <span class="st">"MM"</span>, 
                                           <span class="st">"NM"</span>,<span class="st">"NM"</span>,<span class="st">"NM"</span>,<span class="st">"NM"</span><span class="op">)</span>,
                                         levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NM"</span>, <span class="st">"MM"</span><span class="op">)</span><span class="op">)</span>,
                      <span class="va">nams</span>,
                      row.names <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>

<span class="co">## Build a RangedGlmDataSet object</span>
<span class="va">ds_50_100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps_at_clust_50_100</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></code></pre></div>
<p>The GRanges object with DMPs on it</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmps_at_clust_50_100</span></code></pre></div>
<pre><code>## GRanges object with 18035 ranges and 9 metadata columns:
##           seqnames            ranges strand |   M_1_105   M_1_168   M_1_179
##              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##       [1]        1           183-330      * |         0        22         2
##       [2]        1           647-809      * |         0         3         4
##       [3]        1       18052-18224      * |         2         2         1
##       [4]        1       18643-18746      * |         2         1         0
##       [5]        1       42279-42423      * |        15        16         0
##       ...      ...               ...    ... .       ...       ...       ...
##   [18031]        5 26670460-26670651      * |         5         1         8
##   [18032]        5 26788467-26788575      * |         7         1         1
##   [18033]        5 26945319-26945446      * |         3         2         0
##   [18034]        5 26974071-26974175      * |         3         4         0
##   [18035]        5 26974820-26975362      * |         4        11         8
##              M_1_27     M_1_3   NM_1_12   NM_1_45   NM_1_70   NM_1_86
##           &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##       [1]         3         1         0         0         2         0
##       [2]         2         1         0         0         0         0
##       [3]         3         7         0         0         0         0
##       [4]         3         0         0         2         1         3
##       [5]        13        27         0         0         0         0
##       ...       ...       ...       ...       ...       ...       ...
##   [18031]         3         0         0         0         0         0
##   [18032]         1         0         1         0         0         0
##   [18033]         2         2         2         1         0         1
##   [18034]         3         0         1         4         2         3
##   [18035]         3        11         2         3         0         0
##   -------
##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
<p>The <a href="https://is.gd/VZaazz">glmDataSet</a> object.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ds_50_100</span></code></pre></div>
<pre><code>## RangedGlmDataSet with 18035 regions and 9 columns (individuals) with factor levels 'NM' and 'MM' 
## The accessible objects in the dataset are: 
##             Length Class      Mode     
## GR           18035 GRanges    S4       
## counts      162315 -none-     numeric  
## colData          1 data.frame list     
## sampleNames      9 -none-     character
## levels           2 -none-     character
## optionData       0 -none-     NULL</code></pre>
<p>Design for method = “relaxed” from dmpClusters function</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span>
<span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span>
<span class="va">.dmps_at_clust_50_100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDMPatRegions.html">getDMPatRegions</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, regions <span class="op">=</span> <span class="va">.clust_50_100</span>,
                                       ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">.dmps_at_clust_50_100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">.dmps_at_clust_50_100</span>, columns <span class="op">=</span> <span class="fl">2L</span>,
                                      ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>, type <span class="op">=</span> <span class="st">'equal'</span>,
                                      verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">.dmps_at_clust_50_100</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">nams</span>

<span class="co">## Build a RangedGlmDataSet object</span>
<span class="va">.ds_50_100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">.dmps_at_clust_50_100</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></code></pre></div>
<p>Design for method = “fixed.int” from dmpClustering function</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span>
<span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span>
<span class="va">dmps_at_clust.50.100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDMPatRegions.html">getDMPatRegions</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, regions <span class="op">=</span> <span class="va">clust.50.100</span>,
                                       ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">dmps_at_clust.50.100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">dmps_at_clust.50.100</span>, columns <span class="op">=</span> <span class="fl">2L</span>,
                                      ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>, type <span class="op">=</span> <span class="st">'equal'</span>,
                                      verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dmps_at_clust.50.100</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">nams</span>

<span class="co">## Build a RangedGlmDataSet object</span>
<span class="va">ds.50.100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps_at_clust.50.100</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></code></pre></div>
<p>Design for method = “relaxed” from dmpClustering function</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span>
<span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span>
<span class="va">dmps_at.clust.50.100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDMPatRegions.html">getDMPatRegions</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, regions <span class="op">=</span> <span class="va">.clust.50.100</span>,
                                       ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">dmps_at.clust.50.100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">dmps_at.clust.50.100</span>, columns <span class="op">=</span> <span class="fl">2L</span>,
                                      ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>, type <span class="op">=</span> <span class="st">'equal'</span>,
                                      verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dmps_at.clust.50.100</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">nams</span>

<span class="co">## Build a RangedGlmDataSet object</span>
<span class="va">ds_50.100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps_at.clust.50.100</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></code></pre></div>
</div>
<div id="dmr-detection-1" class="section level3">
<h3 class="hasAnchor">
<a href="#dmr-detection-1" class="anchor"></a>DMR detection</h3>
<p>Herein, we are requesting a minimum density DMP of 1/100 bp = 0.01/bp (CountPerBp = 0.01), minimum of logarithm of fold changes in the number of DMPs, treatment versus control samples, equal to 1 (Minlog2FC = 1). Also, it is requested a coefficient of variation, on each sample set, not greater than 1 (maxGrpCV = c(1, 1)).</p>
<p>Notice that in the case of species with big genes, for example, human genes, we might want to use lower densities, e.g., CountPerBp = 0.001.</p>
<p>DMRs derived method = “fixed.int” from dmpClusters function</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmrs_fixed.int</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds_50_100</span>, 
                            minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,
                            maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,
                            Minlog2FC <span class="op">=</span> <span class="fl">1</span>,
                            CountPerBp <span class="op">=</span> <span class="fl">0.01</span>, 
                            test <span class="op">=</span> <span class="st">'LRT'</span>,
                            num.cores <span class="op">=</span> <span class="fl">4L</span>, 
                            verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="va">dmrs_fixed.int</span></code></pre></div>
<p>The DMR associated genes at a <span class="math inline">\(distance \leq 1000\)</span>:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/nearest-methods.html">distanceToNearest</a></span><span class="op">(</span><span class="va">dmrs_fixed.int</span>, <span class="va">GENES</span>, ignore.strand<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>

<span class="co">## DMR associated genes </span>
<span class="va">hits</span> <span class="op">&lt;-</span> <span class="va">hits</span><span class="op">[</span><span class="fu">mcols</span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;=</span> <span class="fl">1000</span><span class="op">]</span>
<span class="va">GENES</span><span class="op">[</span><span class="fu">subjectHits</span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code>## GRanges object with 975 ranges and 4 metadata columns:
##         seqnames            ranges strand |     gene_id     type   gene_biotype
##            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;factor&gt;    &lt;character&gt;
##     [1]        1       43087-43295      - |   AT1G04003     gene         lncRNA
##     [2]        1     239841-242703      - |   AT1G01660     gene protein_coding
##     [3]        1     258717-258963      - |   AT1G04037     gene         lncRNA
##     [4]        1     428650-430720      - |   AT1G02220     gene protein_coding
##     [5]        1   1194187-1196889      + |   AT1G04425     gene          ncRNA
##     ...      ...               ...    ... .         ...      ...            ...
##   [971]        5 23775891-23779855      + |   AT5G58880     gene protein_coding
##   [972]        5 25934568-25935612      + |   AT5G64890     gene protein_coding
##   [973]        5 25967824-25968638      + |   AT5G65005     gene protein_coding
##   [974]        5 26068112-26069652      + |   AT5G65230     gene protein_coding
##   [975]        5 26411796-26414609      - |   AT5G66050     gene protein_coding
##           gene_name
##         &lt;character&gt;
##     [1]        &lt;NA&gt;
##     [2]        &lt;NA&gt;
##     [3]        &lt;NA&gt;
##     [4]      NAC003
##     [5]        &lt;NA&gt;
##     ...         ...
##   [971]        &lt;NA&gt;
##   [972]        PEP2
##   [973]        &lt;NA&gt;
##   [974]     AtMYB53
##   [975]        &lt;NA&gt;
##   -------
##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
<p>DMRs derived method = “relaxed” from dmpClusters function</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">.dmrs_relaxed</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">.ds_50_100</span>, 
                           minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,
                           maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,
                           Minlog2FC <span class="op">=</span> <span class="fl">1</span>,
                           CountPerBp <span class="op">=</span> <span class="fl">0.01</span>, 
                           test <span class="op">=</span> <span class="st">'LRT'</span>,
                           num.cores <span class="op">=</span> <span class="fl">4L</span>, 
                           verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">.dmrs_relaxed</span></code></pre></div>
<p>The DMR assciated genes at a <span class="math inline">\(distance \leq 1000\)</span>:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/nearest-methods.html">distanceToNearest</a></span><span class="op">(</span><span class="va">.dmrs_relaxed</span>, <span class="va">GENES</span>, ignore.strand<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>

<span class="co">## DMR associated genes </span>
<span class="va">hits</span> <span class="op">&lt;-</span> <span class="va">hits</span><span class="op">[</span><span class="fu">mcols</span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;=</span> <span class="fl">1000</span><span class="op">]</span>
<span class="va">GENES</span><span class="op">[</span><span class="fu">subjectHits</span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code>## GRanges object with 1215 ranges and 4 metadata columns:
##          seqnames            ranges strand |     gene_id     type
##             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;factor&gt;
##      [1]        1       43087-43295      - |   AT1G04003     gene
##      [2]        1     239841-242703      - |   AT1G01660     gene
##      [3]        1     258717-258963      - |   AT1G04037     gene
##      [4]        1     428650-430720      - |   AT1G02220     gene
##      [5]        1   1194187-1196889      + |   AT1G04425     gene
##      ...      ...               ...    ... .         ...      ...
##   [1211]        5 25934568-25935612      + |   AT5G64890     gene
##   [1212]        5 25967824-25968638      + |   AT5G65005     gene
##   [1213]        5 26068112-26069652      + |   AT5G65230     gene
##   [1214]        5 26161624-26169191      - |   AT5G65460     gene
##   [1215]        5 26411796-26414609      - |   AT5G66050     gene
##            gene_biotype   gene_name
##             &lt;character&gt; &lt;character&gt;
##      [1]         lncRNA        &lt;NA&gt;
##      [2] protein_coding        &lt;NA&gt;
##      [3]         lncRNA        &lt;NA&gt;
##      [4] protein_coding      NAC003
##      [5]          ncRNA        &lt;NA&gt;
##      ...            ...         ...
##   [1211] protein_coding        PEP2
##   [1212] protein_coding        &lt;NA&gt;
##   [1213] protein_coding     AtMYB53
##   [1214] protein_coding        KCA2
##   [1215] protein_coding        &lt;NA&gt;
##   -------
##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
<p>DMRs derived method = “fixed.int” from dmpClustering function</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmrs.fixed.int</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds.50.100</span>,
                            minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,
                            maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,
                            Minlog2FC <span class="op">=</span> <span class="fl">1</span>,
                            CountPerBp <span class="op">=</span> <span class="fl">0.01</span>,
                            test <span class="op">=</span> <span class="st">'LRT'</span>,
                            num.cores <span class="op">=</span> <span class="fl">4L</span>,
                            verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">dmrs.fixed.int</span></code></pre></div>
<p>The DMR assciated genes at a <span class="math inline">\(distance \leq 1000\)</span>:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/nearest-methods.html">distanceToNearest</a></span><span class="op">(</span><span class="va">dmrs.fixed.int</span>, <span class="va">GENES</span>, ignore.strand<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>

<span class="co">## DMR associated genes </span>
<span class="va">hits</span> <span class="op">&lt;-</span> <span class="va">hits</span><span class="op">[</span><span class="fu">mcols</span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;=</span> <span class="fl">1000</span><span class="op">]</span>
<span class="va">GENES</span><span class="op">[</span><span class="fu">subjectHits</span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code>## GRanges object with 1151 ranges and 4 metadata columns:
##          seqnames            ranges strand |     gene_id     type
##             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;factor&gt;
##      [1]        1       43087-43295      - |   AT1G04003     gene
##      [2]        1     258717-258963      - |   AT1G04037     gene
##      [3]        1     428650-430720      - |   AT1G02220     gene
##      [4]        1   1194187-1196889      + |   AT1G04425     gene
##      [5]        1   1431177-1433010      - |   AT1G05010     gene
##      ...      ...               ...    ... .         ...      ...
##   [1147]        5 25934568-25935612      + |   AT5G64890     gene
##   [1148]        5 25967824-25968638      + |   AT5G65005     gene
##   [1149]        5 26068112-26069652      + |   AT5G65230     gene
##   [1150]        5 26161624-26169191      - |   AT5G65460     gene
##   [1151]        5 26411796-26414609      - |   AT5G66050     gene
##            gene_biotype   gene_name
##             &lt;character&gt; &lt;character&gt;
##      [1]         lncRNA        &lt;NA&gt;
##      [2]         lncRNA        &lt;NA&gt;
##      [3] protein_coding      NAC003
##      [4]          ncRNA        &lt;NA&gt;
##      [5] protein_coding        ACO4
##      ...            ...         ...
##   [1147] protein_coding        PEP2
##   [1148] protein_coding        &lt;NA&gt;
##   [1149] protein_coding     AtMYB53
##   [1150] protein_coding        KCA2
##   [1151] protein_coding        &lt;NA&gt;
##   -------
##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
<p>DMRs derived method = “relaxed” from dmpClustering function</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmrs.relaxed</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds_50.100</span>,
                          minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,
                          maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, 
                          Minlog2FC <span class="op">=</span> <span class="fl">1</span>,
                          CountPerBp <span class="op">=</span> <span class="fl">0.01</span>,
                          test <span class="op">=</span> <span class="st">'LRT'</span>,
                          num.cores <span class="op">=</span> <span class="fl">4L</span>,
                          verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">dmrs.relaxed</span></code></pre></div>
<p>The DMR assciated genes at a <span class="math inline">\(distance \leq 1000\)</span>:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/nearest-methods.html">distanceToNearest</a></span><span class="op">(</span><span class="va">dmrs.relaxed</span>, <span class="va">GENES</span>, ignore.strand<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>

<span class="co">## DMR associated genes </span>
<span class="va">hits</span> <span class="op">&lt;-</span> <span class="va">hits</span><span class="op">[</span><span class="fu">mcols</span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;=</span> <span class="fl">1000</span><span class="op">]</span>
<span class="va">GENES</span><span class="op">[</span><span class="fu">subjectHits</span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code>## GRanges object with 1214 ranges and 4 metadata columns:
##          seqnames            ranges strand |     gene_id     type
##             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;factor&gt;
##      [1]        1       43087-43295      - |   AT1G04003     gene
##      [2]        1     239841-242703      - |   AT1G01660     gene
##      [3]        1     258717-258963      - |   AT1G04037     gene
##      [4]        1     428650-430720      - |   AT1G02220     gene
##      [5]        1   1194187-1196889      + |   AT1G04425     gene
##      ...      ...               ...    ... .         ...      ...
##   [1210]        5 25934568-25935612      + |   AT5G64890     gene
##   [1211]        5 25967824-25968638      + |   AT5G65005     gene
##   [1212]        5 26068112-26069652      + |   AT5G65230     gene
##   [1213]        5 26161624-26169191      - |   AT5G65460     gene
##   [1214]        5 26411796-26414609      - |   AT5G66050     gene
##            gene_biotype   gene_name
##             &lt;character&gt; &lt;character&gt;
##      [1]         lncRNA        &lt;NA&gt;
##      [2] protein_coding        &lt;NA&gt;
##      [3]         lncRNA        &lt;NA&gt;
##      [4] protein_coding      NAC003
##      [5]          ncRNA        &lt;NA&gt;
##      ...            ...         ...
##   [1210] protein_coding        PEP2
##   [1211] protein_coding        &lt;NA&gt;
##   [1212] protein_coding     AtMYB53
##   [1213] protein_coding        KCA2
##   [1214] protein_coding        &lt;NA&gt;
##   -------
##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
</div>
<div id="naturally-ocurring-regions" class="section level1">
<h1 class="hasAnchor">
<a href="#naturally-ocurring-regions" class="anchor"></a>Naturally ocurring regions</h1>
<p>As was mentioned in the introduction algorithmically identified DMRs do not necessarily will be associated to the local DNA physico-chemical properties and to local molecular evolutionary features of every DMR detected.</p>
<p>All the weight of the evolutionary pressure lie down on naturally ocurrying methylation patternings inside, for example (but not limited to), exons and introns.</p>
<p>For introns we have:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span>
<span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span>
<span class="va">dmps_at_exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, GENES <span class="op">=</span> <span class="va">exons</span>,
                                ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,
                                output <span class="op">=</span> <span class="st">"GRanges"</span><span class="op">)</span>

<span class="co">## Build a RangedGlmDataSet object</span>
<span class="va">ds_exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps_at_exons</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></code></pre></div>
<p>Differentially methylated exons</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmrs_exons</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds_exons</span>,
                        minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,
                        maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, 
                        Minlog2FC <span class="op">=</span> <span class="fl">1</span>,
                        CountPerBp <span class="op">=</span> <span class="fl">0.01</span>, 
                        test <span class="op">=</span> <span class="st">'LRT'</span>,
                        num.cores <span class="op">=</span> <span class="fl">4L</span>,
                        verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## *** Number of GR after filtering counts 194</code></pre>
<pre><code>## *** GLM...</code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmrs_exons</span></code></pre></div>
<pre><code>## GRanges object with 140 ranges and 17 metadata columns:
##             seqnames            ranges strand |   M_1_105   M_1_168   M_1_179
##                &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##   AT1G10745        1   3572879-3573173      * |         5         7        26
##   AT1G17150        1   5866658-5867176      * |         5         6        16
##   AT1G33030        1 11964776-11965351      * |         6        18        21
##   AT1G33720        1 12222407-12224108      * |        36        12         0
##   AT1G33720        1 12222407-12224108      * |        36        12         0
##         ...      ...               ...    ... .       ...       ...       ...
##   AT5G49010        5 19867969-19868877      * |        24        14        20
##   AT5G49010        5 19867969-19868877      * |        24        14        20
##   AT5G51930        5 21103008-21103568      * |         6        20         0
##   AT5G52090        5 21167432-21169462      * |        46        30        19
##   AT5G54700        5 22222886-22224526      * |        25        17        22
##                M_1_27     M_1_3   NM_1_12   NM_1_45   NM_1_70   NM_1_86
##             &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##   AT1G10745         7         5         1         0         0         0
##   AT1G17150         8         6         0         0         1         0
##   AT1G33030         6        12         0         3         0         3
##   AT1G33720        20        24         4         8         0         0
##   AT1G33720        20        24         4         8         0         0
##         ...       ...       ...       ...       ...       ...       ...
##   AT5G49010         6        20         2         0         0         0
##   AT5G49010         6        20         2         0         0         0
##   AT5G51930         8        34         2         0         0         2
##   AT5G52090        27        27         8         4         6         9
##   AT5G54700        14        12         1         6         2         2
##                log2FC scaled.deviance      pvalue        model    adj.pval
##             &lt;numeric&gt;       &lt;numeric&gt;   &lt;numeric&gt;  &lt;character&gt;   &lt;numeric&gt;
##   AT1G10745   2.17475        17.49778 2.87643e-05 Neg.Binomial 4.57614e-05
##   AT1G17150   1.99606        24.10357 9.12911e-07 Neg.Binomial 1.85228e-06
##   AT1G33030   1.69378        19.38609 1.06782e-05 Neg.Binomial 1.82311e-05
##   AT1G33720   1.57898         6.53006 1.06066e-02 Neg.Binomial 1.06066e-02
##   AT1G33720   1.57898         6.53006 1.06066e-02 Neg.Binomial 1.06066e-02
##         ...       ...             ...         ...          ...         ...
##   AT5G49010   2.47373        42.93430 5.66094e-11 Neg.Binomial 1.72289e-10
##   AT5G49010   2.47373        42.93430 5.66094e-11 Neg.Binomial 1.72289e-10
##   AT5G51930   1.98787         9.76251 1.78107e-03 Neg.Binomial 2.13119e-03
##   AT5G52090   1.48497        41.38165 1.25228e-10 Neg.Binomial 3.65247e-10
##   AT5G54700   1.87877        45.37607 1.62609e-11 Neg.Binomial 5.69132e-11
##             CT.SignalDensity TT.SignalDensity SignalDensityVariation
##                    &lt;numeric&gt;        &lt;numeric&gt;              &lt;numeric&gt;
##   AT1G10745      0.000847458        0.0338983             0.03305085
##   AT1G17150      0.000481696        0.0157996             0.01531792
##   AT1G33030      0.002604167        0.0218750             0.01927083
##   AT1G33720      0.001762632        0.0108108             0.00904818
##   AT1G33720      0.001762632        0.0108108             0.00904818
##         ...              ...              ...                    ...
##   AT5G49010      0.000550055        0.0184818             0.01793179
##   AT5G49010      0.000550055        0.0184818             0.01793179
##   AT5G51930      0.001782531        0.0242424             0.02245989
##   AT5G52090      0.003323486        0.0146726             0.01134909
##   AT5G54700      0.001675807        0.0109689             0.00929311
##   -------
##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
<p>For introns we have:</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span>
<span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span>
<span class="va">dmps_at_introns</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, GENES <span class="op">=</span> <span class="va">introns</span>,
                                 ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,
                                 output <span class="op">=</span> <span class="st">"GRanges"</span><span class="op">)</span>

<span class="co">## Build a RangedGlmDataSet object</span>
<span class="va">ds_introns</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps_at_introns</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></code></pre></div>
<p>Differentially methylated introns</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmrs_introns</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds_introns</span>,
                          minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,
                          maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, 
                          Minlog2FC <span class="op">=</span> <span class="fl">1</span>,
                          CountPerBp <span class="op">=</span> <span class="fl">0.01</span>,
                          test <span class="op">=</span> <span class="st">'LRT'</span>,
                          num.cores <span class="op">=</span> <span class="fl">4L</span>,
                          verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## *** Number of GR after filtering counts 2</code></pre>
<pre><code>## *** GLM...</code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmrs_introns</span></code></pre></div>
<pre><code>## GRanges object with 2 ranges and 17 metadata columns:
##             seqnames            ranges strand |   M_1_105   M_1_168   M_1_179
##                &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##   AT2G14000        2   5885110-5885454      * |         4        12        10
##   AT5G52115        5 21174335-21175487      * |        12        19        13
##                M_1_27     M_1_3   NM_1_12   NM_1_45   NM_1_70   NM_1_86
##             &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##   AT2G14000         5         9         5         1         1         1
##   AT5G52115        10         9         0         2         0         3
##                log2FC scaled.deviance      pvalue        model    adj.pval
##             &lt;numeric&gt;       &lt;numeric&gt;   &lt;numeric&gt;  &lt;character&gt;   &lt;numeric&gt;
##   AT2G14000   1.38629         14.2099 1.63508e-04 Neg.Binomial 1.63508e-04
##   AT5G52115   1.79914         38.9928 4.25375e-10 Neg.Binomial 8.50750e-10
##             CT.SignalDensity TT.SignalDensity SignalDensityVariation
##                    &lt;numeric&gt;        &lt;numeric&gt;              &lt;numeric&gt;
##   AT2G14000       0.00579710        0.0231884             0.01739130
##   AT5G52115       0.00108413        0.0109280             0.00984389
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<div id="differentially-methylated-genes-dmgs" class="section level3">
<h3 class="hasAnchor">
<a href="#differentially-methylated-genes-dmgs" class="anchor"></a>Differentially methylated genes DMGs</h3>
<p>The last results for introns and exons contrasts with the result for the whole gene-body.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span>
<span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span>
<span class="va">dmps_at_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, GENES <span class="op">=</span> <span class="va">genes</span>,
                                ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,
                                output <span class="op">=</span> <span class="st">"GRanges"</span><span class="op">)</span>

<span class="co">## Build a RangedGlmDataSet object</span>
<span class="va">ds_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps_at_genes</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></code></pre></div>
<p>Differentically methylated genes</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmrs_genes</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds_genes</span>, 
                        minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,
                        maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,
                        Minlog2FC <span class="op">=</span> <span class="fl">1</span>,
                        CountPerBp <span class="op">=</span> <span class="fl">0.01</span>,
                        test <span class="op">=</span> <span class="st">'LRT'</span>,
                        num.cores <span class="op">=</span> <span class="fl">4L</span>, 
                        verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## *** Number of GR after filtering counts 970</code></pre>
<pre><code>## *** GLM...</code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmrs_genes</span></code></pre></div>
<pre><code>## GRanges object with 893 ranges and 17 metadata columns:
##             seqnames            ranges strand |   M_1_105   M_1_168   M_1_179
##                &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##   AT1G05035        1   1444983-1446116      * |        11        15        12
##   AT1G07025        1   2157658-2158158      * |         8        10        18
##   AT1G07025        1   2157658-2158158      * |         8        10        18
##   AT1G09260        1   2993381-2993820      * |        10         6        24
##   AT1G09260        1   2993381-2993820      * |        10         6        24
##         ...      ...               ...    ... .       ...       ...       ...
##   AT5G52115        5 21174321-21175857      * |        18        27        17
##   AT5G54700        5 22222886-22225616      * |        42        28        36
##   AT5G56920        5 23024896-23025915      * |        19        38        17
##   AT5G60945        5 24526245-24526636      * |         4         8        14
##   AT5G60945        5 24526245-24526636      * |         4         8        14
##                M_1_27     M_1_3   NM_1_12   NM_1_45   NM_1_70   NM_1_86
##             &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##   AT1G05035        22         9         7         2         2         7
##   AT1G07025        16         8         4         2         0         2
##   AT1G07025        16         8         4         2         0         2
##   AT1G09260         2         2         2         0         0         6
##   AT1G09260         2         2         2         0         0         6
##         ...       ...       ...       ...       ...       ...       ...
##   AT5G52115        12        23         0         2         0         3
##   AT5G54700        23        20         2        11         3         3
##   AT5G56920        19        33         2         4         6        14
##   AT5G60945         4        10         2         0         0         0
##   AT5G60945         4        10         2         0         0         0
##                log2FC scaled.deviance      pvalue        model    adj.pval
##             &lt;numeric&gt;       &lt;numeric&gt;   &lt;numeric&gt;  &lt;character&gt;   &lt;numeric&gt;
##   AT1G05035   1.12059        16.07033 6.10329e-05 Neg.Binomial 8.03869e-05
##   AT1G07025   1.46634        25.55214 4.30588e-07 Neg.Binomial 8.48819e-07
##   AT1G07025   1.46634        25.55214 4.30588e-07 Neg.Binomial 8.48819e-07
##   AT1G09260   1.18377         4.41686 3.55854e-02 Neg.Binomial 3.55854e-02
##   AT1G09260   1.18377         4.41686 3.55854e-02 Neg.Binomial 3.55854e-02
##         ...       ...             ...         ...          ...         ...
##   AT5G52115   2.20460         63.4528 1.64254e-15 Neg.Binomial 1.16411e-14
##   AT5G54700   1.83636         42.8798 5.82085e-11 Neg.Binomial 2.24053e-10
##   AT5G56920   1.35504         18.3625 1.82615e-05 Neg.Binomial 2.73616e-05
##   AT5G60945   1.79176         20.6233 5.59125e-06 Neg.Binomial 9.21215e-06
##   AT5G60945   1.79176         20.6233 5.59125e-06 Neg.Binomial 9.21215e-06
##             CT.SignalDensity TT.SignalDensity SignalDensityVariation
##                    &lt;numeric&gt;        &lt;numeric&gt;              &lt;numeric&gt;
##   AT1G05035       0.00396825        0.0121693             0.00820106
##   AT1G07025       0.00399202        0.0239521             0.01996008
##   AT1G07025       0.00399202        0.0239521             0.01996008
##   AT1G09260       0.00454545        0.0200000             0.01545455
##   AT1G09260       0.00454545        0.0200000             0.01545455
##         ...              ...              ...                    ...
##   AT5G52115      0.000813273        0.0126220             0.01180872
##   AT5G54700      0.001739290        0.0109118             0.00917246
##   AT5G56920      0.006372549        0.0247059             0.01833333
##   AT5G60945      0.001275510        0.0204082             0.01913265
##   AT5G60945      0.001275510        0.0204082             0.01913265
##   -------
##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
<p>It seems to be that methylation patterning inside of gene-body forms a sort continuous regions simultaneously covering portions of exon and introns. In particular, this is true for genes experiencing alternative splicing, where some regions inside introns works as exons as well.</p>
</div>
</div>
<div id="references-" class="section level1">
<h1 class="hasAnchor">
<a href="#references-" class="anchor"></a>References.</h1>
<div id="refs" class="references csl-bib-body">
<div id="ref-Sanchez2016" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">R. Sanchez, S. A. Mackenzie, <span class="nocase">Information Thermodynamics of Cytosine DNA Methylation</span>. <em>PLOS ONE</em> <strong>11</strong>, e0150427 (2016).</div>
</div>
<div id="ref-Sanchez2019" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">R. Sanchez, X. Yang, T. Maher, S. Mackenzie, <span class="nocase">Discrimination of DNA Methylation Signal from Background Variation for Clinical Diagnostics</span>. <em>Int. J. Mol. Sci.</em> <strong>20</strong>, 5343 (2019).</div>
</div>
<div id="ref-Greiner2000" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">M. Greiner, D. Pfeiffer, R. D. Smith, <span class="nocase">Principles and practical application of the receiver-operating characteristic analysis for diagnostic tests</span>. <em>Preventive Veterinary Medicine</em> <strong>45</strong>, 23–41 (2000).</div>
</div>
<div id="ref-Carter2016" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">J. V. Carter, J. Pan, S. N. Rai, S. Galandiuk, <span class="nocase">ROC-ing along: Evaluation and interpretation of receiver operating characteristic curves</span>. <em>Surgery</em> <strong>159</strong>, 1638–1645 (2016).</div>
</div>
<div id="ref-Harpaz2013" class="csl-entry">
<div class="csl-left-margin">5. </div>
<div class="csl-right-inline">R. Harpaz, <em>et al.</em>, <span class="nocase">Performance of Pharmacovigilance Signal Detection Algorithms for the FDA Adverse Event Reporting System</span>. <em>Clin Pharmacol ther</em> <strong>93</strong>, 1–19 (2013).</div>
</div>
<div id="ref-Kruspe2017" class="csl-entry">
<div class="csl-left-margin">6. </div>
<div class="csl-right-inline">S. Kruspe, <em>et al.</em>, <span class="nocase">Rapid and Sensitive Detection of Breast Cancer Cells in Patient Blood with Nuclease-Activated Probe Technology</span>. <em>Molecular Therapy - Nucleic Acids</em> <strong>8</strong>, 542–557 (2017).</div>
</div>
<div id="ref-Wahlberg2016" class="csl-entry">
<div class="csl-left-margin">7. </div>
<div class="csl-right-inline">P. Wahlberg, <em>et al.</em>, <span class="nocase">DNA methylome analysis of acute lymphoblastic leukemia cells reveals stochastic de novo DNA methylation in CpG islands</span>. <em>Epigenomics</em> <strong>8</strong>, 1367–1387 (2016).</div>
</div>
<div id="ref-Sanchez2020" class="csl-entry">
<div class="csl-left-margin">8. </div>
<div class="csl-right-inline">R. Sanchez, S. A. Mackenzie, <span class="nocase">Integrative Network Analysis of Differentially Methylated and Expressed Genes for Biomarker Identification in Leukemia</span>. <em>Scientific Reports</em> <strong>10</strong>, 2123 (2020).</div>
</div>
<div id="ref-Severin2011" class="csl-entry">
<div class="csl-left-margin">9. </div>
<div class="csl-right-inline">P. M. D. Severin, X. Zou, H. E. Gaub, K. Schulten, <span class="nocase">Cytosine methylation alters DNA mechanical properties</span>. <em>Nucleic acids research</em> <strong>39</strong>, 8740–51 (2011).</div>
</div>
<div id="ref-Perez2012" class="csl-entry">
<div class="csl-left-margin">10. </div>
<div class="csl-right-inline">A. Pérez, <em>et al.</em>, <span class="nocase">Impact of methylation on the physical properties of DNA</span>. <em>Biophysical Journal</em> <strong>102</strong>, 2140–2148 (2012).</div>
</div>
<div id="ref-Yusufaly2013" class="csl-entry">
<div class="csl-left-margin">11. </div>
<div class="csl-right-inline">T. I. Yusufaly, Y. Li, W. K. Olson, <span class="nocase">5-Methylation of cytosine in CG:CG base-pair steps: a physicochemical mechanism for the epigenetic control of DNA nanomechanics.</span> <em>The journal of physical chemistry. B</em> <strong>117</strong>, 16436–42 (2013).</div>
</div>
<div id="ref-Teng2018" class="csl-entry">
<div class="csl-left-margin">12. </div>
<div class="csl-right-inline">X. Teng, W. Hwang, <span class="nocase">Effect of Methylation on Local Mechanics and Hydration Structure of DNA</span>. <em>Biophysical Journal</em> <strong>114</strong>, 1791–1803 (2018).</div>
</div>
<div id="ref-Hodges2020" class="csl-entry">
<div class="csl-left-margin">13. </div>
<div class="csl-right-inline">A. J. Hodges, N. O. Hudson, B. A. Buck-Koehntop, <span class="nocase">Cys2His2 Zinc Finger Methyl-CpG Binding Proteins: Getting a Handle on Methylated DNA</span>. <em>Journal of Molecular Biology</em> <strong>432</strong>, 1640–1660 (2020).</div>
</div>
<div id="ref-Ngo2016" class="csl-entry">
<div class="csl-left-margin">14. </div>
<div class="csl-right-inline">T. T. M. Ngo, <em>et al.</em>, <span class="nocase">Effects of cytosine modifications on DNA flexibility and nucleosome mechanical stability</span>. <em>Nature Communications</em> <strong>7</strong>, 10813 (2016).</div>
</div>
</div>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo()</a></code> on the system on which this document was compiled running pandoc 2.11.4:</p>
<pre><code>## R version 4.1.1 (2021-08-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.3.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
##  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] gridExtra_2.3        ggplot2_3.3.5        MethylIT.utils_0.1  
##  [4] MethylIT_0.3.2.3     rtracklayer_1.52.1   GenomicRanges_1.44.0
##  [7] GenomeInfoDb_1.28.4  IRanges_2.26.0       S4Vectors_0.30.1    
## [10] BiocGenerics_0.38.0  knitr_1.33          
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_2.0-2            rjson_0.2.20               
##   [3] modeltools_0.2-23           ellipsis_0.3.2             
##   [5] class_7.3-19                rprojroot_2.0.2            
##   [7] XVector_0.32.0              fs_1.5.0                   
##   [9] flexmix_2.3-17              bit64_4.0.5                
##  [11] AnnotationDbi_1.54.1        prodlim_2019.11.13         
##  [13] fansi_0.5.0                 lubridate_1.7.10           
##  [15] codetools_0.2-18            splines_4.1.1              
##  [17] cachem_1.0.5                Formula_1.2-4              
##  [19] jsonlite_1.7.2              pROC_1.17.0.1              
##  [21] Rsamtools_2.8.0             caret_6.0-88               
##  [23] annotate_1.70.0             png_0.1-7                  
##  [25] compiler_4.1.1              httr_1.4.2                 
##  [27] assertthat_0.2.1            Matrix_1.3-4               
##  [29] fastmap_1.1.0               htmltools_0.5.1.1          
##  [31] tools_4.1.1                 gtable_0.3.0               
##  [33] glue_1.4.2                  GenomeInfoDbData_1.2.6     
##  [35] reshape2_1.4.4              dplyr_1.0.7                
##  [37] Rcpp_1.0.7                  Biobase_2.52.0             
##  [39] jquerylib_0.1.4             pkgdown_1.6.1              
##  [41] vctrs_0.3.8                 Biostrings_2.60.2          
##  [43] nlme_3.1-153                betareg_3.1-4              
##  [45] iterators_1.0.13            lmtest_0.9-38              
##  [47] timeDate_3043.102           gower_0.2.2                
##  [49] xfun_0.24                   stringr_1.4.0              
##  [51] lifecycle_1.0.0             restfulr_0.0.13            
##  [53] XML_3.99-0.6                zoo_1.8-9                  
##  [55] zlibbioc_1.38.0             MASS_7.3-54                
##  [57] scales_1.1.1                ipred_0.9-11               
##  [59] ragg_1.1.3                  MatrixGenerics_1.4.3       
##  [61] sandwich_3.0-1              SummarizedExperiment_1.22.0
##  [63] yaml_2.2.1                  memoise_2.0.0              
##  [65] sass_0.4.0                  rpart_4.1-15               
##  [67] stringi_1.6.2               RSQLite_2.2.8              
##  [69] genefilter_1.74.0           BiocIO_1.2.0               
##  [71] desc_1.3.0                  foreach_1.5.1              
##  [73] BiocParallel_1.26.2         lava_1.6.9                 
##  [75] rlang_0.4.11                pkgconfig_2.0.3            
##  [77] systemfonts_1.0.2           matrixStats_0.59.0         
##  [79] bitops_1.0-7                evaluate_0.14              
##  [81] lattice_0.20-45             purrr_0.3.4                
##  [83] GenomicAlignments_1.28.0    recipes_0.1.16             
##  [85] bit_4.0.4                   tidyselect_1.1.1           
##  [87] plyr_1.8.6                  magrittr_2.0.1             
##  [89] nls2_0.2                    R6_2.5.0                   
##  [91] generics_0.1.0              DelayedArray_0.18.0        
##  [93] DBI_1.1.1                   pillar_1.6.1               
##  [95] withr_2.4.2                 survival_3.2-13            
##  [97] KEGGREST_1.32.0             RCurl_1.98-1.3             
##  [99] nnet_7.3-16                 tibble_3.1.2               
## [101] crayon_1.4.1                utf8_1.2.1                 
## [103] rmarkdown_2.10              minpack.lm_1.2-1           
## [105] data.table_1.14.0           blob_1.2.2                 
## [107] ModelMetrics_1.2.2.2        digest_0.6.27              
## [109] xtable_1.8-4                textshaping_0.3.5          
## [111] munsell_0.5.0               bslib_0.2.5.1</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
