<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>DMR detection with Methyl-IT • MethylIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="DMR detection with Methyl-IT">
<meta property="og:description" content="MethylIT">
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>DMR detection with Methyl-IT</h1>
            <h3 data-toc-skip class="subtitle">Analysis for Arabidopsis
MSH1-memory versus non-memory first generation</h3>
                        <h4 data-toc-skip class="author"><div class="line-block">Robersy Sanchez<br><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a><br>
Mackenzie’s lab<br><br>
Department of Biology and Plant Science.<br>
Pennsylvania State University, University Park, PA 16802<br>
Maintainer: Robersy Sanchez</div></h4>
            
            <h4 data-toc-skip class="date">24 June 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/vignettes/dmr_at_memory.Rmd" class="external-link"><code>vignettes/dmr_at_memory.Rmd</code></a></small>
      <div class="hidden name"><code>dmr_at_memory.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Herein, an example of Methyl-IT application to the detection of
      differential methylated regions (DMRs) on first generation MSH1
      RNAi transgene segregants data set is presented.</p>
    </div>
    
<style type="text/css">

.list-group-item.active, .list-group-item.active:focus,
.list-group-item.active:hover {
    z-index: 2;
    color: #fff;
    background-color: #337ab7;
    border-color: #337ab7;
}

.tocify-subheader .tocify-item {
    font-size: 0.80em;
    padding-left: 25px;
    text-indent: 0;
}

.tocify-header .tocify-item {
    font-size: 0.85em;
    padding-left: 20px;
    text-indent: 0;
}

body{ /* Normal */
    font-size: 18px;
    font-family: "Times New Roman", Times, serif;
    text-align: justify
}
td { /* Table */
    font-size: 8px;
}

h1.title {
    font-size: 38px;
    font-family: "Times New Roman", Times, serif;
    color: DarkRed;
    .subTitle {
        font-size: 24px;
        font-family: "Times New Roman", Times, serif;
        color: DarkRed;
    }
}

h1 { /* Header 1 */
    font-size: 28px;
    font-family: "Times New Roman", Times, serif;
    color: DarkBlue;
}
h2 { /* Header 2 */
    font-size: 22px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
h3 { /* Header 3 */
    font-size: 18px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
code.r{ /* Code block */
    font-size: 12px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>MethylIT is an R package for methylome analysis based on a
signal-detection machine learning approach (SD-ML). This approach is
postulated to provide greater sensitivity for resolving true signal from
the methylation background within the methylome <span class="citation">(1, 2)</span>. Because the biological signal created
within the dynamic methylome environment characteristic of plants is not
free from background noise, the approach, designated Methyl-IT, includes
the application of signal detection theory <span class="citation">(3–6)</span>.</p>
<p>A basic requirement for the application of signal detection is the
knowledge of the probability distribution of the background noise, which
is used as null hypothesis in the detection of the methylation signal.
This probability distribution is, in general, a member of generalized
gamma distribution family, and it can be deduced on a statistical
mechanical/thermodynamics basis from DNA methylation induced by thermal
fluctuations <span class="citation">(1)</span>.</p>
<p>Herein, we provide an example of Methyl-IT application to DMR
estimation on patients with PALL <span class="citation">(7)</span>. Due
to the size of human methylome the current example only covers the
analysis of chromosome 9. A full description of Methyl-IT application of
methylome analysis for Clinical Diagnostics is given in the manuscript
<span class="citation">(2)</span>. A wider analysis of PALL data set is
given in <span class="citation">(8)</span>.</p>
<p>The first step of DMR estimation follows the general step of
Methyl-IT pipeline for the estimation of DMPs. In Methyl-IT, a DMP is a
cytosine positions that, with high probability, carries a statistically
significant methylation change in respect to the same position in a
control population. Once DMPs are identified, a simple algorithm can be
applied to identify clusters of DMPs using <a href="https://genomaths.github.io/methylit/reference/dmpClusters.html">dmpClusters</a>
function. Next, <a href="https://is.gd/kltCXl" class="external-link">countTest2</a> is applied
to test whether a given cluster of DMPs is a DMR.</p>
<p>It is worthy to notices that any clustering algorithm will impose
non-natural constraints on the methylation patterning, which not
necessarily will be associated to the local DNA physico-chemical
properties and to local molecular evolutionary features of every DMR
detected. Every cytosine methylation has a concrete physical effect of
the DNA, altering the DNA persistence length, i.e., altering the local
DNA flexibility <span class="citation">(9–11)</span>. DNA flexibility
was shown to affect the binding of proteins to methylated DNA and
DNA-containing lesion <span class="citation">(12, 13)</span> and
nucleosome mechanical stability <span class="citation">(14)</span>. In
other words, the sizes of naturally occurrying methylation motifs are
not arbitrarily set, but very well constrained under molecular
evolutionary pressure, probably, particular for each species.</p>
<p>The R packages required in this example are:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>        <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/MethylIT" class="external-link">MethylIT</a></span><span class="op">)</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MethylIT.utils</span><span class="op">)</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span> <span class="co"># graphic</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">grid</span><span class="op">)</span> <span class="co"># For multiple plots</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span> <span class="co"># For multiple plots</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span> <span class="co"># To import gene annotation</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="estimation-of-differentially-methylated-regions-dmrs">Estimation of differentially methylated regions (DMRs)<a class="anchor" aria-label="anchor" href="#estimation-of-differentially-methylated-regions-dmrs"></a>
</h2>
<p>DMP data set is available for download.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## ===== Download methylation data from Github ========</span></span>
<span><span class="va">url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://github.com/genomaths/genomaths.github.io/raw/master/AT_memory/"</span>, </span>
<span>              <span class="st">"dmps_mm-vs-nm_all-context_gen1_nmRef_min-meth-3_cov-4_04-16-20.RData"</span><span class="op">)</span></span>
<span><span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".RData"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="va">url</span>, destfile <span class="op">=</span> <span class="va">temp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">temp</span>, <span class="va">url</span><span class="op">)</span></span></code></pre></div>
<p>The gene annotation</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## ===== Download methylation data from GitHub ========</span></span>
<span><span class="va">url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://github.com/genomaths/genomaths.github.io/raw/master/"</span>, </span>
<span>              <span class="st">"Annotation/Arabidopsis_thaliana_genes_TAIR10.38.gtf.RData"</span><span class="op">)</span></span>
<span><span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".RData"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="va">url</span>, destfile <span class="op">=</span> <span class="va">temp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">temp</span>, <span class="va">url</span><span class="op">)</span>;</span>
<span></span>
<span><span class="va">GENES</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">type</span> <span class="op">==</span> <span class="st">"gene"</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## To get exons and introns</span></span>
<span><span class="va">exons</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">type</span> <span class="op">==</span> <span class="st">"exon"</span><span class="op">]</span></span>
<span><span class="va">introns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">setdiff</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">genes</span>, y <span class="op">=</span> <span class="va">exons</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## To add the corresponding gene names to introns</span></span>
<span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a></span><span class="op">(</span><span class="va">introns</span>, <span class="va">genes</span>, type <span class="op">=</span> <span class="st">"within"</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">introns</span> <span class="op">&lt;-</span> <span class="va">introns</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">queryHits</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">introns</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">genes</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">subjectHits</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="cluster-of-dmps">Cluster of DMPs<a class="anchor" aria-label="anchor" href="#cluster-of-dmps"></a>
</h3>
<p>Function <a href="https://is.gd/K2cRgZ" class="external-link">dmpClusters</a> search for
cluster of DMPs found from all the samples. The basic idea reside in
that if a given cluster of DMP is a DMR, then it will be consistently
differentially methylated in all treatment samples, and we can identify
them applying function <a href="https://%20is.gd/kltCXl" class="external-link">countTest2</a>.</p>
<p>There is not a unique approach to build DMP clusters, since several
clustering criteria can be applied. Herein, we use the two approach
implemented in <a href="https://is.gd/K2cRgZ" class="external-link">dmpClusters</a> function
(follow the link or type ?dmpClusters to see the help for the
function).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clust_50_100</span> <span class="op">=</span> <span class="fu"><a href="../reference/dmpClusters.html">dmpClusters</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, maxDist <span class="op">=</span> <span class="fl">50</span>, minNumDMPs <span class="op">=</span> <span class="fl">10</span>, </span>
<span>                           maxClustDist <span class="op">=</span> <span class="fl">100</span>, method <span class="op">=</span> <span class="st">"fixed.int"</span>,</span>
<span>                           num.cores <span class="op">=</span> <span class="fl">1L</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">clust_50_100</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 18035 ranges and 1 metadata column:</span></span>
<span><span class="co">##           seqnames            ranges strand |      dmps</span></span>
<span><span class="co">##              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span></span>
<span><span class="co">##       [1]        1           183-330      * |        27</span></span>
<span><span class="co">##       [2]        1           647-809      * |        10</span></span>
<span><span class="co">##       [3]        1       18052-18224      * |        14</span></span>
<span><span class="co">##       [4]        1       18643-18746      * |        12</span></span>
<span><span class="co">##       [5]        1       42279-42423      * |        40</span></span>
<span><span class="co">##       ...      ...               ...    ... .       ...</span></span>
<span><span class="co">##   [18031]        5 26670460-26670651      * |        15</span></span>
<span><span class="co">##   [18032]        5 26788467-26788575      * |        11</span></span>
<span><span class="co">##   [18033]        5 26945319-26945446      * |        11</span></span>
<span><span class="co">##   [18034]        5 26974071-26974175      * |        10</span></span>
<span><span class="co">##   [18035]        5 26974820-26975362      * |        39</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>DMPs clusterization strongly depends on the maximum distance
permitted between two consecutive DMPs that belong to the same cluster
(parameter: <code>minNumDMPs</code>) and on the minimum number of DMPs
that must be present on a detected agglomeration of DMPs to be
considered a cluster (parameter: <code>maxDist</code>).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">.clust_50_100</span> <span class="op">=</span> <span class="fu"><a href="../reference/dmpClusters.html">dmpClusters</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, maxDist <span class="op">=</span> <span class="fl">50</span>, minNumDMPs <span class="op">=</span> <span class="fl">10</span>, </span>
<span>                           maxClustDist <span class="op">=</span> <span class="fl">100</span>, method <span class="op">=</span> <span class="st">"relaxed"</span>,</span>
<span>                           num.cores <span class="op">=</span> <span class="fl">1L</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">.clust_50_100</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 17528 ranges and 1 metadata column:</span></span>
<span><span class="co">##           seqnames            ranges strand |      dmps</span></span>
<span><span class="co">##              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span></span>
<span><span class="co">##       [1]        1            94-330      * |        29</span></span>
<span><span class="co">##       [2]        1           647-880      * |        13</span></span>
<span><span class="co">##       [3]        1       17833-18512      * |        35</span></span>
<span><span class="co">##       [4]        1       18643-18746      * |        12</span></span>
<span><span class="co">##       [5]        1       42279-42423      * |        40</span></span>
<span><span class="co">##       ...      ...               ...    ... .       ...</span></span>
<span><span class="co">##   [17524]        5 26788399-26788575      * |        12</span></span>
<span><span class="co">##   [17525]        5 26835058-26835302      * |        13</span></span>
<span><span class="co">##   [17526]        5 26945319-26945446      * |        11</span></span>
<span><span class="co">##   [17527]        5 26974071-26974329      * |        14</span></span>
<span><span class="co">##   [17528]        5 26974728-26975362      * |        40</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>Alternatively, we can use function <em>dmpClustering</em> from
MethylIT.utils R package. Notice that although <em>‘method’</em>
parameter are named the same, on each function these stand for different
approaches.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clust.50.100</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/dmpClustering.html" class="external-link">dmpClustering</a></span><span class="op">(</span>dmps <span class="op">=</span> <span class="va">dmps</span>, minNumDMPs <span class="op">=</span> <span class="fl">10</span>, win.size <span class="op">=</span> <span class="fl">50</span>,</span>
<span>                            step.size <span class="op">=</span> <span class="fl">50</span>, method <span class="op">=</span> <span class="st">"fixed.int"</span>, </span>
<span>                            maxClustDist <span class="op">=</span> <span class="fl">100</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">clust.50.100</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 15521 ranges and 1 metadata column:</span></span>
<span><span class="co">##           seqnames            ranges strand |      dmps</span></span>
<span><span class="co">##              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span></span>
<span><span class="co">##       [1]        1            94-330      * |        29</span></span>
<span><span class="co">##       [2]        1           647-880      * |        13</span></span>
<span><span class="co">##       [3]        1       17833-18746      * |        47</span></span>
<span><span class="co">##       [4]        1       42279-42423      * |        40</span></span>
<span><span class="co">##       [5]        1       44603-44848      * |        19</span></span>
<span><span class="co">##       ...      ...               ...    ... .       ...</span></span>
<span><span class="co">##   [15517]        5 26835058-26835302      * |        13</span></span>
<span><span class="co">##   [15518]        5 26891628-26892012      * |        10</span></span>
<span><span class="co">##   [15519]        5 26945319-26945446      * |        11</span></span>
<span><span class="co">##   [15520]        5 26974071-26974329      * |        14</span></span>
<span><span class="co">##   [15521]        5 26974728-26975362      * |        40</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">.clust.50.100</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/dmpClustering.html" class="external-link">dmpClustering</a></span><span class="op">(</span>dmps <span class="op">=</span> <span class="va">dmps</span>, minNumDMPs <span class="op">=</span> <span class="fl">10</span>, method <span class="op">=</span> <span class="st">"relaxed"</span>, </span>
<span>                            maxClustDist <span class="op">=</span> <span class="fl">100</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">.clust.50.100</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 17647 ranges and 1 metadata column:</span></span>
<span><span class="co">##           seqnames            ranges strand |      dmps</span></span>
<span><span class="co">##              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span></span>
<span><span class="co">##       [1]        1            94-330      * |        29</span></span>
<span><span class="co">##       [2]        1           647-880      * |        13</span></span>
<span><span class="co">##       [3]        1       17833-18512      * |        35</span></span>
<span><span class="co">##       [4]        1       18643-18746      * |        12</span></span>
<span><span class="co">##       [5]        1       42279-42423      * |        40</span></span>
<span><span class="co">##       ...      ...               ...    ... .       ...</span></span>
<span><span class="co">##   [17643]        5 26788399-26788575      * |        12</span></span>
<span><span class="co">##   [17644]        5 26835058-26835302      * |        13</span></span>
<span><span class="co">##   [17645]        5 26945319-26945446      * |        11</span></span>
<span><span class="co">##   [17646]        5 26974071-26974329      * |        14</span></span>
<span><span class="co">##   [17647]        5 26974728-26975362      * |        40</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="dmr-detection">DMR detection<a class="anchor" aria-label="anchor" href="#dmr-detection"></a>
</h3>
<p>The count of the number of DMPs on each region will be accomplished
with <a href="https://genomaths.github.io/methylit/reference/getDMPatRegions.html">getDMPatRegions</a>
function. This function operates based on <a href="https://is.gd/zJ7U9l" class="external-link"><em>findOverlaps</em></a> function from the
‘<em>GenomicRanges</em>’ Bioconductor R package. The <a href="https://is.gd/zJ7U9l" class="external-link"><em>findOverlaps</em></a> function has
several critical parameters like, for example, ‘<em>maxgap</em>’,
‘<em>minoverlap</em>’, and ‘<em>ignore.strand</em>’. In our <a href="https://genomaths.github.io/methylit/reference/getDMPatRegions.html">getDMPatRegions</a>
function, except for setting ignore.strand = TRUE and <em>type =
“within”</em>, we preserve the rest of default ‘findOverlaps’
parameters.</p>
<div class="section level4">
<h4 id="experiment-design-for-each-clustering-approach">Experiment design for each clustering approach<a class="anchor" aria-label="anchor" href="#experiment-design-for-each-clustering-approach"></a>
</h4>
<p>Design for method = “fixed.int” from dmpClusters function</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span></span>
<span><span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span></span>
<span><span class="va">dmps_at_clust_50_100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDMPatRegions.html">getDMPatRegions</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, regions <span class="op">=</span> <span class="va">clust_50_100</span>,</span>
<span>                                       ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">dmps_at_clust_50_100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">dmps_at_clust_50_100</span>, columns <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>                                      ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>, type <span class="op">=</span> <span class="st">'equal'</span>,</span>
<span>                                      verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">dmps_at_clust_50_100</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">nams</span></span>
<span></span>
<span><span class="va">colData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MM"</span>, <span class="st">"MM"</span>,<span class="st">"MM"</span>,<span class="st">"MM"</span>, <span class="st">"MM"</span>, </span>
<span>                                           <span class="st">"NM"</span>,<span class="st">"NM"</span>,<span class="st">"NM"</span>,<span class="st">"NM"</span><span class="op">)</span>,</span>
<span>                                         levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NM"</span>, <span class="st">"MM"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                      <span class="va">nams</span>,</span>
<span>                      row.names <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Build a RangedGlmDataSet object</span></span>
<span><span class="va">ds_50_100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps_at_clust_50_100</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></span></code></pre></div>
<p>The GRanges object with DMPs on it</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dmps_at_clust_50_100</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 18035 ranges and 9 metadata columns:</span></span>
<span><span class="co">##           seqnames            ranges strand |   M_1_105   M_1_168   M_1_179</span></span>
<span><span class="co">##              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##       [1]        1           183-330      * |         0        22         2</span></span>
<span><span class="co">##       [2]        1           647-809      * |         0         3         4</span></span>
<span><span class="co">##       [3]        1       18052-18224      * |         2         2         1</span></span>
<span><span class="co">##       [4]        1       18643-18746      * |         2         1         0</span></span>
<span><span class="co">##       [5]        1       42279-42423      * |        15        16         0</span></span>
<span><span class="co">##       ...      ...               ...    ... .       ...       ...       ...</span></span>
<span><span class="co">##   [18031]        5 26670460-26670651      * |         5         1         8</span></span>
<span><span class="co">##   [18032]        5 26788467-26788575      * |         7         1         1</span></span>
<span><span class="co">##   [18033]        5 26945319-26945446      * |         3         2         0</span></span>
<span><span class="co">##   [18034]        5 26974071-26974175      * |         3         4         0</span></span>
<span><span class="co">##   [18035]        5 26974820-26975362      * |         4        11         8</span></span>
<span><span class="co">##              M_1_27     M_1_3   NM_1_12   NM_1_45   NM_1_70   NM_1_86</span></span>
<span><span class="co">##           &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##       [1]         3         1         0         0         2         0</span></span>
<span><span class="co">##       [2]         2         1         0         0         0         0</span></span>
<span><span class="co">##       [3]         3         7         0         0         0         0</span></span>
<span><span class="co">##       [4]         3         0         0         2         1         3</span></span>
<span><span class="co">##       [5]        13        27         0         0         0         0</span></span>
<span><span class="co">##       ...       ...       ...       ...       ...       ...       ...</span></span>
<span><span class="co">##   [18031]         3         0         0         0         0         0</span></span>
<span><span class="co">##   [18032]         1         0         1         0         0         0</span></span>
<span><span class="co">##   [18033]         2         2         2         1         0         1</span></span>
<span><span class="co">##   [18034]         3         0         1         4         2         3</span></span>
<span><span class="co">##   [18035]         3        11         2         3         0         0</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>The <a href="https://is.gd/VZaazz" class="external-link">glmDataSet</a> object.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ds_50_100</span></span></code></pre></div>
<pre><code><span><span class="co">## RangedGlmDataSet with 18035 regions and 9 columns (individuals) with factor levels 'NM' and 'MM' </span></span>
<span><span class="co">## The accessible objects in the dataset are: </span></span>
<span><span class="co">##             Length Class      Mode     </span></span>
<span><span class="co">## GR           18035 GRanges    S4       </span></span>
<span><span class="co">## counts      162315 -none-     numeric  </span></span>
<span><span class="co">## colData          1 data.frame list     </span></span>
<span><span class="co">## sampleNames      9 -none-     character</span></span>
<span><span class="co">## levels           2 -none-     character</span></span>
<span><span class="co">## optionData       0 -none-     NULL</span></span></code></pre>
<p>Design for method = “relaxed” from dmpClusters function</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span></span>
<span><span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span></span>
<span><span class="va">.dmps_at_clust_50_100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDMPatRegions.html">getDMPatRegions</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, regions <span class="op">=</span> <span class="va">.clust_50_100</span>,</span>
<span>                                       ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">.dmps_at_clust_50_100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">.dmps_at_clust_50_100</span>, columns <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>                                      ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>, type <span class="op">=</span> <span class="st">'equal'</span>,</span>
<span>                                      verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">.dmps_at_clust_50_100</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">nams</span></span>
<span></span>
<span><span class="co">## Build a RangedGlmDataSet object</span></span>
<span><span class="va">.ds_50_100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">.dmps_at_clust_50_100</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></span></code></pre></div>
<p>Design for method = “fixed.int” from dmpClustering function</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span></span>
<span><span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span></span>
<span><span class="va">dmps_at_clust.50.100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDMPatRegions.html">getDMPatRegions</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, regions <span class="op">=</span> <span class="va">clust.50.100</span>,</span>
<span>                                       ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">dmps_at_clust.50.100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">dmps_at_clust.50.100</span>, columns <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>                                      ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>, type <span class="op">=</span> <span class="st">'equal'</span>,</span>
<span>                                      verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">dmps_at_clust.50.100</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">nams</span></span>
<span></span>
<span><span class="co">## Build a RangedGlmDataSet object</span></span>
<span><span class="va">ds.50.100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps_at_clust.50.100</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></span></code></pre></div>
<p>Design for method = “relaxed” from dmpClustering function</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span></span>
<span><span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span></span>
<span><span class="va">dmps_at.clust.50.100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDMPatRegions.html">getDMPatRegions</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, regions <span class="op">=</span> <span class="va">.clust.50.100</span>,</span>
<span>                                       ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">dmps_at.clust.50.100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">dmps_at.clust.50.100</span>, columns <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>                                      ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>, type <span class="op">=</span> <span class="st">'equal'</span>,</span>
<span>                                      verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">dmps_at.clust.50.100</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">nams</span></span>
<span></span>
<span><span class="co">## Build a RangedGlmDataSet object</span></span>
<span><span class="va">ds_50.100</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps_at.clust.50.100</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="dmr-detection-1">DMR detection<a class="anchor" aria-label="anchor" href="#dmr-detection-1"></a>
</h4>
<p>Herein, we are requesting a minimum density DMP of 1/100 bp = 0.01/bp
(CountPerBp = 0.01), minimum of logarithm of fold changes in the number
of DMPs, treatment versus control samples, equal to 1 (Minlog2FC = 1).
Also, it is requested a coefficient of variation, on each sample set,
not greater than 1 (maxGrpCV = c(1, 1)).</p>
<p>Notice that in the case of species with big genes, for example, human
genes, we might want to use lower densities, e.g., CountPerBp =
0.001.</p>
<p>DMRs derived method = “fixed.int” from dmpClusters function</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dmrs_fixed.int</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds_50_100</span>, </span>
<span>                            minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,</span>
<span>                            maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>                            Minlog2FC <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                            CountPerBp <span class="op">=</span> <span class="fl">0.01</span>, </span>
<span>                            test <span class="op">=</span> <span class="st">'LRT'</span>,</span>
<span>                            num.cores <span class="op">=</span> <span class="fl">4L</span>, </span>
<span>                            verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dmrs_fixed.int</span></span></code></pre></div>
<p>The DMR associated genes at a <span class="math inline">\(distance
\leq 1000\)</span>:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/nearest-methods.html" class="external-link">distanceToNearest</a></span><span class="op">(</span><span class="va">dmrs_fixed.int</span>, <span class="va">GENES</span>, ignore.strand<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## DMR associated genes </span></span>
<span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="va">hits</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;=</span> <span class="fl">1000</span><span class="op">]</span></span>
<span><span class="va">GENES</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">subjectHits</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 975 ranges and 4 metadata columns:</span></span>
<span><span class="co">##         seqnames            ranges strand |     gene_id     type   gene_biotype</span></span>
<span><span class="co">##            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;factor&gt;    &lt;character&gt;</span></span>
<span><span class="co">##     [1]        1       43087-43295      - |   AT1G04003     gene         lncRNA</span></span>
<span><span class="co">##     [2]        1     239841-242703      - |   AT1G01660     gene protein_coding</span></span>
<span><span class="co">##     [3]        1     258717-258963      - |   AT1G04037     gene         lncRNA</span></span>
<span><span class="co">##     [4]        1     428650-430720      - |   AT1G02220     gene protein_coding</span></span>
<span><span class="co">##     [5]        1   1194187-1196889      + |   AT1G04425     gene          ncRNA</span></span>
<span><span class="co">##     ...      ...               ...    ... .         ...      ...            ...</span></span>
<span><span class="co">##   [971]        5 23775891-23779855      + |   AT5G58880     gene protein_coding</span></span>
<span><span class="co">##   [972]        5 25934568-25935612      + |   AT5G64890     gene protein_coding</span></span>
<span><span class="co">##   [973]        5 25967824-25968638      + |   AT5G65005     gene protein_coding</span></span>
<span><span class="co">##   [974]        5 26068112-26069652      + |   AT5G65230     gene protein_coding</span></span>
<span><span class="co">##   [975]        5 26411796-26414609      - |   AT5G66050     gene protein_coding</span></span>
<span><span class="co">##           gene_name</span></span>
<span><span class="co">##         &lt;character&gt;</span></span>
<span><span class="co">##     [1]        &lt;NA&gt;</span></span>
<span><span class="co">##     [2]        &lt;NA&gt;</span></span>
<span><span class="co">##     [3]        &lt;NA&gt;</span></span>
<span><span class="co">##     [4]      NAC003</span></span>
<span><span class="co">##     [5]        &lt;NA&gt;</span></span>
<span><span class="co">##     ...         ...</span></span>
<span><span class="co">##   [971]        &lt;NA&gt;</span></span>
<span><span class="co">##   [972]        PEP2</span></span>
<span><span class="co">##   [973]        &lt;NA&gt;</span></span>
<span><span class="co">##   [974]     AtMYB53</span></span>
<span><span class="co">##   [975]        &lt;NA&gt;</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>DMRs derived method = “relaxed” from dmpClusters function</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">.dmrs_relaxed</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">.ds_50_100</span>, </span>
<span>                           minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,</span>
<span>                           maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>                           Minlog2FC <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                           CountPerBp <span class="op">=</span> <span class="fl">0.01</span>, </span>
<span>                           test <span class="op">=</span> <span class="st">'LRT'</span>,</span>
<span>                           num.cores <span class="op">=</span> <span class="fl">4L</span>, </span>
<span>                           verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">.dmrs_relaxed</span></span></code></pre></div>
<p>The DMR assciated genes at a <span class="math inline">\(distance
\leq 1000\)</span>:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/nearest-methods.html" class="external-link">distanceToNearest</a></span><span class="op">(</span><span class="va">.dmrs_relaxed</span>, <span class="va">GENES</span>, ignore.strand<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## DMR associated genes </span></span>
<span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="va">hits</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;=</span> <span class="fl">1000</span><span class="op">]</span></span>
<span><span class="va">GENES</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">subjectHits</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 1215 ranges and 4 metadata columns:</span></span>
<span><span class="co">##          seqnames            ranges strand |     gene_id     type</span></span>
<span><span class="co">##             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;factor&gt;</span></span>
<span><span class="co">##      [1]        1       43087-43295      - |   AT1G04003     gene</span></span>
<span><span class="co">##      [2]        1     239841-242703      - |   AT1G01660     gene</span></span>
<span><span class="co">##      [3]        1     258717-258963      - |   AT1G04037     gene</span></span>
<span><span class="co">##      [4]        1     428650-430720      - |   AT1G02220     gene</span></span>
<span><span class="co">##      [5]        1   1194187-1196889      + |   AT1G04425     gene</span></span>
<span><span class="co">##      ...      ...               ...    ... .         ...      ...</span></span>
<span><span class="co">##   [1211]        5 25934568-25935612      + |   AT5G64890     gene</span></span>
<span><span class="co">##   [1212]        5 25967824-25968638      + |   AT5G65005     gene</span></span>
<span><span class="co">##   [1213]        5 26068112-26069652      + |   AT5G65230     gene</span></span>
<span><span class="co">##   [1214]        5 26161624-26169191      - |   AT5G65460     gene</span></span>
<span><span class="co">##   [1215]        5 26411796-26414609      - |   AT5G66050     gene</span></span>
<span><span class="co">##            gene_biotype   gene_name</span></span>
<span><span class="co">##             &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">##      [1]         lncRNA        &lt;NA&gt;</span></span>
<span><span class="co">##      [2] protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">##      [3]         lncRNA        &lt;NA&gt;</span></span>
<span><span class="co">##      [4] protein_coding      NAC003</span></span>
<span><span class="co">##      [5]          ncRNA        &lt;NA&gt;</span></span>
<span><span class="co">##      ...            ...         ...</span></span>
<span><span class="co">##   [1211] protein_coding        PEP2</span></span>
<span><span class="co">##   [1212] protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">##   [1213] protein_coding     AtMYB53</span></span>
<span><span class="co">##   [1214] protein_coding        KCA2</span></span>
<span><span class="co">##   [1215] protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>DMRs derived method = “fixed.int” from dmpClustering function</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dmrs.fixed.int</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds.50.100</span>,</span>
<span>                            minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,</span>
<span>                            maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>                            Minlog2FC <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                            CountPerBp <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                            test <span class="op">=</span> <span class="st">'LRT'</span>,</span>
<span>                            num.cores <span class="op">=</span> <span class="fl">4L</span>,</span>
<span>                            verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">dmrs.fixed.int</span></span></code></pre></div>
<p>The DMR assciated genes at a <span class="math inline">\(distance
\leq 1000\)</span>:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/nearest-methods.html" class="external-link">distanceToNearest</a></span><span class="op">(</span><span class="va">dmrs.fixed.int</span>, <span class="va">GENES</span>, ignore.strand<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## DMR associated genes </span></span>
<span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="va">hits</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;=</span> <span class="fl">1000</span><span class="op">]</span></span>
<span><span class="va">GENES</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">subjectHits</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 1151 ranges and 4 metadata columns:</span></span>
<span><span class="co">##          seqnames            ranges strand |     gene_id     type</span></span>
<span><span class="co">##             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;factor&gt;</span></span>
<span><span class="co">##      [1]        1       43087-43295      - |   AT1G04003     gene</span></span>
<span><span class="co">##      [2]        1     258717-258963      - |   AT1G04037     gene</span></span>
<span><span class="co">##      [3]        1     428650-430720      - |   AT1G02220     gene</span></span>
<span><span class="co">##      [4]        1   1194187-1196889      + |   AT1G04425     gene</span></span>
<span><span class="co">##      [5]        1   1431177-1433010      - |   AT1G05010     gene</span></span>
<span><span class="co">##      ...      ...               ...    ... .         ...      ...</span></span>
<span><span class="co">##   [1147]        5 25934568-25935612      + |   AT5G64890     gene</span></span>
<span><span class="co">##   [1148]        5 25967824-25968638      + |   AT5G65005     gene</span></span>
<span><span class="co">##   [1149]        5 26068112-26069652      + |   AT5G65230     gene</span></span>
<span><span class="co">##   [1150]        5 26161624-26169191      - |   AT5G65460     gene</span></span>
<span><span class="co">##   [1151]        5 26411796-26414609      - |   AT5G66050     gene</span></span>
<span><span class="co">##            gene_biotype   gene_name</span></span>
<span><span class="co">##             &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">##      [1]         lncRNA        &lt;NA&gt;</span></span>
<span><span class="co">##      [2]         lncRNA        &lt;NA&gt;</span></span>
<span><span class="co">##      [3] protein_coding      NAC003</span></span>
<span><span class="co">##      [4]          ncRNA        &lt;NA&gt;</span></span>
<span><span class="co">##      [5] protein_coding        ACO4</span></span>
<span><span class="co">##      ...            ...         ...</span></span>
<span><span class="co">##   [1147] protein_coding        PEP2</span></span>
<span><span class="co">##   [1148] protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">##   [1149] protein_coding     AtMYB53</span></span>
<span><span class="co">##   [1150] protein_coding        KCA2</span></span>
<span><span class="co">##   [1151] protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>DMRs derived method = “relaxed” from dmpClustering function</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dmrs.relaxed</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds_50.100</span>,</span>
<span>                          minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,</span>
<span>                          maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, </span>
<span>                          Minlog2FC <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                          CountPerBp <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                          test <span class="op">=</span> <span class="st">'LRT'</span>,</span>
<span>                          num.cores <span class="op">=</span> <span class="fl">4L</span>,</span>
<span>                          verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">dmrs.relaxed</span></span></code></pre></div>
<p>The DMR assciated genes at a <span class="math inline">\(distance
\leq 1000\)</span>:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/nearest-methods.html" class="external-link">distanceToNearest</a></span><span class="op">(</span><span class="va">dmrs.relaxed</span>, <span class="va">GENES</span>, ignore.strand<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## DMR associated genes </span></span>
<span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="va">hits</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">&lt;=</span> <span class="fl">1000</span><span class="op">]</span></span>
<span><span class="va">GENES</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Hits-class.html" class="external-link">subjectHits</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 1214 ranges and 4 metadata columns:</span></span>
<span><span class="co">##          seqnames            ranges strand |     gene_id     type</span></span>
<span><span class="co">##             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;factor&gt;</span></span>
<span><span class="co">##      [1]        1       43087-43295      - |   AT1G04003     gene</span></span>
<span><span class="co">##      [2]        1     239841-242703      - |   AT1G01660     gene</span></span>
<span><span class="co">##      [3]        1     258717-258963      - |   AT1G04037     gene</span></span>
<span><span class="co">##      [4]        1     428650-430720      - |   AT1G02220     gene</span></span>
<span><span class="co">##      [5]        1   1194187-1196889      + |   AT1G04425     gene</span></span>
<span><span class="co">##      ...      ...               ...    ... .         ...      ...</span></span>
<span><span class="co">##   [1210]        5 25934568-25935612      + |   AT5G64890     gene</span></span>
<span><span class="co">##   [1211]        5 25967824-25968638      + |   AT5G65005     gene</span></span>
<span><span class="co">##   [1212]        5 26068112-26069652      + |   AT5G65230     gene</span></span>
<span><span class="co">##   [1213]        5 26161624-26169191      - |   AT5G65460     gene</span></span>
<span><span class="co">##   [1214]        5 26411796-26414609      - |   AT5G66050     gene</span></span>
<span><span class="co">##            gene_biotype   gene_name</span></span>
<span><span class="co">##             &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">##      [1]         lncRNA        &lt;NA&gt;</span></span>
<span><span class="co">##      [2] protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">##      [3]         lncRNA        &lt;NA&gt;</span></span>
<span><span class="co">##      [4] protein_coding      NAC003</span></span>
<span><span class="co">##      [5]          ncRNA        &lt;NA&gt;</span></span>
<span><span class="co">##      ...            ...         ...</span></span>
<span><span class="co">##   [1210] protein_coding        PEP2</span></span>
<span><span class="co">##   [1211] protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">##   [1212] protein_coding     AtMYB53</span></span>
<span><span class="co">##   [1213] protein_coding        KCA2</span></span>
<span><span class="co">##   [1214] protein_coding        &lt;NA&gt;</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
</div>
</div>
</div>
<div class="section level2">
<h2 id="naturally-ocurring-regions">Naturally ocurring regions<a class="anchor" aria-label="anchor" href="#naturally-ocurring-regions"></a>
</h2>
<p>As was mentioned in the introduction algorithmically identified DMRs
do not necessarily will be associated to the local DNA physico-chemical
properties and to local molecular evolutionary features of every DMR
detected.</p>
<p>All the weight of the evolutionary pressure lie down on naturally
ocurrying methylation patternings inside, for example (but not limited
to), exons and introns.</p>
<p>For introns we have:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span></span>
<span><span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span></span>
<span><span class="va">dmps_at_exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, GENES <span class="op">=</span> <span class="va">exons</span>,</span>
<span>                                ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                output <span class="op">=</span> <span class="st">"GRanges"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Build a RangedGlmDataSet object</span></span>
<span><span class="va">ds_exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps_at_exons</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></span></code></pre></div>
<p>Differentially methylated exons</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dmrs_exons</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds_exons</span>,</span>
<span>                        minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,</span>
<span>                        maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, </span>
<span>                        Minlog2FC <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                        CountPerBp <span class="op">=</span> <span class="fl">0.01</span>, </span>
<span>                        test <span class="op">=</span> <span class="st">'LRT'</span>,</span>
<span>                        num.cores <span class="op">=</span> <span class="fl">4L</span>,</span>
<span>                        verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## *** Number of GR after filtering counts 179</span></span></code></pre>
<pre><code><span><span class="co">## *** GLM...</span></span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dmrs_exons</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 149 ranges and 17 metadata columns:</span></span>
<span><span class="co">##         seqnames            ranges strand |   M_1_105   M_1_168   M_1_179</span></span>
<span><span class="co">##            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##     [1]        1     627482-628469      * |         0         0        30</span></span>
<span><span class="co">##     [2]        1   1506117-1509152      * |        42         0        44</span></span>
<span><span class="co">##     [3]        1   3572879-3573173      * |         5         7        26</span></span>
<span><span class="co">##     [4]        1   3892580-3894677      * |        24        17        23</span></span>
<span><span class="co">##     [5]        1   3979823-3981079      * |        25         9        13</span></span>
<span><span class="co">##     ...      ...               ...    ... .       ...       ...       ...</span></span>
<span><span class="co">##   [145]        5 19867969-19868877      * |        24        14        20</span></span>
<span><span class="co">##   [146]        5 21103008-21103568      * |         6        20         0</span></span>
<span><span class="co">##   [147]        5 21167432-21169462      * |        46        30        19</span></span>
<span><span class="co">##   [148]        5 22222886-22224526      * |        25        17        22</span></span>
<span><span class="co">##   [149]        5 22478712-22480498      * |        27         0        36</span></span>
<span><span class="co">##            M_1_27     M_1_3   NM_1_12   NM_1_45   NM_1_70   NM_1_86    log2FC</span></span>
<span><span class="co">##         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##     [1]        25        20         0         0         0         0   2.72579</span></span>
<span><span class="co">##     [2]        35        58         0         0         0         0   3.60550</span></span>
<span><span class="co">##     [3]         7         5         1         0         0         0   2.17475</span></span>
<span><span class="co">##     [4]        22        23         0         7         5         0   1.74047</span></span>
<span><span class="co">##     [5]        20        15         7         0         4         0   1.53471</span></span>
<span><span class="co">##     ...       ...       ...       ...       ...       ...       ...       ...</span></span>
<span><span class="co">##   [145]         6        20         2         0         0         0   2.47373</span></span>
<span><span class="co">##   [146]         8        34         2         0         0         2   1.98787</span></span>
<span><span class="co">##   [147]        27        27         8         4         6         9   1.48497</span></span>
<span><span class="co">##   [148]        14        12         1         6         2         2   1.87877</span></span>
<span><span class="co">##   [149]        18        27         0         0         0         0   3.11795</span></span>
<span><span class="co">##         scaled.deviance      pvalue          model    adj.pval CT.SignalDensity</span></span>
<span><span class="co">##               &lt;numeric&gt;   &lt;numeric&gt;    &lt;character&gt;   &lt;numeric&gt;        &lt;numeric&gt;</span></span>
<span><span class="co">##     [1]         86.6123 1.32022e-20 Neg.Binomial.W 7.86853e-20      0.000000000</span></span>
<span><span class="co">##     [2]        184.0775 6.24017e-42   Neg.Binomial 2.32446e-40      0.000000000</span></span>
<span><span class="co">##     [3]         17.4978 2.87643e-05   Neg.Binomial 3.86115e-05      0.000847458</span></span>
<span><span class="co">##     [4]         62.9822 2.08580e-15   Neg.Binomial 8.39957e-15      0.001429933</span></span>
<span><span class="co">##     [5]         23.1115 1.52876e-06   Neg.Binomial 2.42325e-06      0.002187749</span></span>
<span><span class="co">##     ...             ...         ...            ...         ...              ...</span></span>
<span><span class="co">##   [145]        42.93430 5.66094e-11   Neg.Binomial 1.36045e-10      0.000550055</span></span>
<span><span class="co">##   [146]         9.76251 1.78107e-03   Neg.Binomial 1.90920e-03      0.001782531</span></span>
<span><span class="co">##   [147]        41.38165 1.25228e-10   Neg.Binomial 2.87060e-10      0.003323486</span></span>
<span><span class="co">##   [148]        45.37607 1.62609e-11   Neg.Binomial 4.32657e-11      0.001675807</span></span>
<span><span class="co">##   [149]       104.45853 1.60518e-24   Neg.Binomial 1.13892e-23      0.000000000</span></span>
<span><span class="co">##         TT.SignalDensity SignalDensityVariation</span></span>
<span><span class="co">##                &lt;numeric&gt;              &lt;numeric&gt;</span></span>
<span><span class="co">##     [1]        0.0151822             0.01518219</span></span>
<span><span class="co">##     [2]        0.0117918             0.01179183</span></span>
<span><span class="co">##     [3]        0.0338983             0.03305085</span></span>
<span><span class="co">##     [4]        0.0103908             0.00896092</span></span>
<span><span class="co">##     [5]        0.0130469             0.01085919</span></span>
<span><span class="co">##     ...              ...                    ...</span></span>
<span><span class="co">##   [145]        0.0184818             0.01793179</span></span>
<span><span class="co">##   [146]        0.0242424             0.02245989</span></span>
<span><span class="co">##   [147]        0.0146726             0.01134909</span></span>
<span><span class="co">##   [148]        0.0109689             0.00929311</span></span>
<span><span class="co">##   [149]        0.0120873             0.01208730</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>For introns we have:</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span></span>
<span><span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span></span>
<span><span class="va">dmps_at_introns</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, GENES <span class="op">=</span> <span class="va">introns</span>,</span>
<span>                                 ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                 output <span class="op">=</span> <span class="st">"GRanges"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Build a RangedGlmDataSet object</span></span>
<span><span class="va">ds_introns</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps_at_introns</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></span></code></pre></div>
<p>Differentially methylated introns</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dmrs_introns</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds_introns</span>,</span>
<span>                          minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,</span>
<span>                          maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, </span>
<span>                          Minlog2FC <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                          CountPerBp <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                          test <span class="op">=</span> <span class="st">'LRT'</span>,</span>
<span>                          num.cores <span class="op">=</span> <span class="fl">4L</span>,</span>
<span>                          verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## *** Number of GR after filtering counts 5</span></span></code></pre>
<pre><code><span><span class="co">## *** GLM...</span></span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dmrs_introns</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 5 ranges and 17 metadata columns:</span></span>
<span><span class="co">##       seqnames            ranges strand |   M_1_105   M_1_168   M_1_179</span></span>
<span><span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##   [1]        1 13148737-13149481      * |        14        14         6</span></span>
<span><span class="co">##   [2]        1 17534249-17534923      * |         6        14         7</span></span>
<span><span class="co">##   [3]        2   5885110-5885454      * |         4        12        10</span></span>
<span><span class="co">##   [4]        5 21161354-21162074      * |         0         0        10</span></span>
<span><span class="co">##   [5]        5 21174335-21175487      * |        12        19        13</span></span>
<span><span class="co">##          M_1_27     M_1_3   NM_1_12   NM_1_45   NM_1_70   NM_1_86    log2FC</span></span>
<span><span class="co">##       &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##   [1]        10        10         0         0         0         0   2.47751</span></span>
<span><span class="co">##   [2]        11         7         2         0         0         2   1.60944</span></span>
<span><span class="co">##   [3]         5         9         5         1         1         1   1.38629</span></span>
<span><span class="co">##   [4]        23        10         0         0         2         0   1.85630</span></span>
<span><span class="co">##   [5]        10         9         0         2         0         3   1.79914</span></span>
<span><span class="co">##       scaled.deviance      pvalue          model    adj.pval CT.SignalDensity</span></span>
<span><span class="co">##             &lt;numeric&gt;   &lt;numeric&gt;    &lt;character&gt;   &lt;numeric&gt;        &lt;numeric&gt;</span></span>
<span><span class="co">##   [1]        85.35744 2.49023e-20 Neg.Binomial.W 1.24512e-19      0.000000000</span></span>
<span><span class="co">##   [2]        25.21553 5.12678e-07   Neg.Binomial 8.54464e-07      0.001481481</span></span>
<span><span class="co">##   [3]        14.20990 1.63508e-04   Neg.Binomial 2.04385e-04      0.005797101</span></span>
<span><span class="co">##   [4]         6.92139 8.51706e-03   Neg.Binomial 8.51706e-03      0.000693481</span></span>
<span><span class="co">##   [5]        38.99278 4.25375e-10   Neg.Binomial 1.06344e-09      0.001084128</span></span>
<span><span class="co">##       TT.SignalDensity SignalDensityVariation</span></span>
<span><span class="co">##              &lt;numeric&gt;              &lt;numeric&gt;</span></span>
<span><span class="co">##   [1]        0.0144966             0.01449664</span></span>
<span><span class="co">##   [2]        0.0133333             0.01185185</span></span>
<span><span class="co">##   [3]        0.0231884             0.01739130</span></span>
<span><span class="co">##   [4]        0.0119279             0.01123440</span></span>
<span><span class="co">##   [5]        0.0109280             0.00984389</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<div class="section level4">
<h4 id="differentially-methylated-genes-dmgs">Differentially methylated genes DMGs<a class="anchor" aria-label="anchor" href="#differentially-methylated-genes-dmgs"></a>
</h4>
<p>The last results for introns and exons contrasts with the result for
the whole gene-body.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span></span>
<span><span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span></span>
<span><span class="va">dmps_at_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, GENES <span class="op">=</span> <span class="va">genes</span>,</span>
<span>                                ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                output <span class="op">=</span> <span class="st">"GRanges"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Build a RangedGlmDataSet object</span></span>
<span><span class="va">ds_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps_at_genes</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></span></code></pre></div>
<p>Differentically methylated genes</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dmrs_genes</span> <span class="op">=</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds_genes</span>, </span>
<span>                        minCountPerIndv <span class="op">=</span> <span class="fl">8</span>,</span>
<span>                        maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>                        Minlog2FC <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                        CountPerBp <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                        test <span class="op">=</span> <span class="st">'LRT'</span>,</span>
<span>                        num.cores <span class="op">=</span> <span class="fl">4L</span>, </span>
<span>                        verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## *** Number of GR after filtering counts 704</span></span></code></pre>
<pre><code><span><span class="co">## *** GLM...</span></span></code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dmrs_genes</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 646 ranges and 17 metadata columns:</span></span>
<span><span class="co">##         seqnames            ranges strand |   M_1_105   M_1_168   M_1_179</span></span>
<span><span class="co">##            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##     [1]        1   1444983-1446116      * |        11        15        12</span></span>
<span><span class="co">##     [2]        1   2157658-2158158      * |         8        10        18</span></span>
<span><span class="co">##     [3]        1   2993381-2993820      * |        10         6        24</span></span>
<span><span class="co">##     [4]        1   3572674-3573173      * |        10        14        52</span></span>
<span><span class="co">##     [5]        1   3892580-3894677      * |        49        37        48</span></span>
<span><span class="co">##     ...      ...               ...    ... .       ...       ...       ...</span></span>
<span><span class="co">##   [642]        5 21167432-21169462      * |        92        60        38</span></span>
<span><span class="co">##   [643]        5 21174321-21175857      * |        18        27        17</span></span>
<span><span class="co">##   [644]        5 22222886-22225616      * |        42        28        36</span></span>
<span><span class="co">##   [645]        5 23024896-23025915      * |        19        38        17</span></span>
<span><span class="co">##   [646]        5 24526245-24526636      * |         4         8        14</span></span>
<span><span class="co">##            M_1_27     M_1_3   NM_1_12   NM_1_45   NM_1_70   NM_1_86    log2FC</span></span>
<span><span class="co">##         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##     [1]        22         9         7         2         2         7   1.12059</span></span>
<span><span class="co">##     [2]        16         8         4         2         0         2   1.46634</span></span>
<span><span class="co">##     [3]         2         2         2         0         0         6   1.18377</span></span>
<span><span class="co">##     [4]        14        10         2         0         0         0   2.63906</span></span>
<span><span class="co">##     [5]        45        47         0        16        11         8   1.55571</span></span>
<span><span class="co">##     ...       ...       ...       ...       ...       ...       ...       ...</span></span>
<span><span class="co">##   [642]        54        54        16         8        12        18   1.48497</span></span>
<span><span class="co">##   [643]        12        23         0         2         0         3   2.20460</span></span>
<span><span class="co">##   [644]        23        20         2        11         3         3   1.83636</span></span>
<span><span class="co">##   [645]        19        33         2         4         6        14   1.35504</span></span>
<span><span class="co">##   [646]         4        10         2         0         0         0   1.79176</span></span>
<span><span class="co">##         scaled.deviance      pvalue        model    adj.pval CT.SignalDensity</span></span>
<span><span class="co">##               &lt;numeric&gt;   &lt;numeric&gt;  &lt;character&gt;   &lt;numeric&gt;        &lt;numeric&gt;</span></span>
<span><span class="co">##     [1]        16.07033 6.10329e-05 Neg.Binomial 7.76127e-05       0.00396825</span></span>
<span><span class="co">##     [2]        25.55214 4.30588e-07 Neg.Binomial 7.81348e-07       0.00399202</span></span>
<span><span class="co">##     [3]         4.41686 3.55854e-02 Neg.Binomial 3.55854e-02       0.00454545</span></span>
<span><span class="co">##     [4]        24.79981 6.36036e-07 Neg.Binomial 1.11956e-06       0.00100000</span></span>
<span><span class="co">##     [5]       111.81692 3.91877e-26 Neg.Binomial 7.67128e-25       0.00417064</span></span>
<span><span class="co">##     ...             ...         ...          ...         ...              ...</span></span>
<span><span class="co">##   [642]         48.4620 3.36757e-12 Neg.Binomial 1.36821e-11      0.006646972</span></span>
<span><span class="co">##   [643]         63.4528 1.64254e-15 Neg.Binomial 9.55926e-15      0.000813273</span></span>
<span><span class="co">##   [644]         42.8798 5.82085e-11 Neg.Binomial 1.94833e-10      0.001739290</span></span>
<span><span class="co">##   [645]         18.3625 1.82615e-05 Neg.Binomial 2.58138e-05      0.006372549</span></span>
<span><span class="co">##   [646]         20.6233 5.59125e-06 Neg.Binomial 8.62040e-06      0.001275510</span></span>
<span><span class="co">##         TT.SignalDensity SignalDensityVariation</span></span>
<span><span class="co">##                &lt;numeric&gt;              &lt;numeric&gt;</span></span>
<span><span class="co">##     [1]        0.0121693             0.00820106</span></span>
<span><span class="co">##     [2]        0.0239521             0.01996008</span></span>
<span><span class="co">##     [3]        0.0200000             0.01545455</span></span>
<span><span class="co">##     [4]        0.0400000             0.03900000</span></span>
<span><span class="co">##     [5]        0.0215443             0.01737369</span></span>
<span><span class="co">##     ...              ...                    ...</span></span>
<span><span class="co">##   [642]        0.0293452             0.02269818</span></span>
<span><span class="co">##   [643]        0.0126220             0.01180872</span></span>
<span><span class="co">##   [644]        0.0109118             0.00917246</span></span>
<span><span class="co">##   [645]        0.0247059             0.01833333</span></span>
<span><span class="co">##   [646]        0.0204082             0.01913265</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>It seems to be that methylation patterning inside of gene-body forms
a sort continuous regions simultaneously covering portions of exon and
introns. In particular, this is true for genes experiencing alternative
splicing, where some regions inside introns works as exons as well.</p>
</div>
</div>
<div class="section level2">
<h2 id="dmr-finder-function">DMR finder function<a class="anchor" aria-label="anchor" href="#dmr-finder-function"></a>
</h2>
<p>Following MethylIT pipeline, DMRs can be identified in control and in
treatment groups. The concept of DMR is subjective in some extend. That
is, while MethylIT pipeline identify cytosine positions with high
probability to be differential methylated in the treatment group with
respect to the control group, it will correspond to the user to define
whether to say, e.g., a region 100 bp long carrying 3 DMPs is a DMR.
There is not theoretical ground to decide whether such a DMR would be
biologically relevant or not for given species. Only the experimentation
or further downstream analyses including, for example, the network
enrichment analysis of DMR-associated genes would gives us an
answer.</p>
<p>Herein, we propose third function to search for DMRs: <a href="https://genomaths.github.io/methylit/reference/dmrfinder.html">dmrfinder</a></p>
<p>Let’s search for hypo-methylated DMRs with at least 20 DMPs on 200
bp. The graphics of samples 1 to 9 on chromosome “1” are returned.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hyp_dmrs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dmrfinder.html">dmrfinder</a></span><span class="op">(</span></span>
<span>    obj <span class="op">=</span> <span class="va">dmps</span>,</span>
<span>    hypo <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    min.signal <span class="op">=</span> <span class="fl">10L</span>,</span>
<span>    win.size <span class="op">=</span> <span class="fl">200</span>,</span>
<span>    step.size <span class="op">=</span> <span class="fl">50</span>,</span>
<span>    stat <span class="op">=</span> <span class="st">"count"</span>,</span>
<span>    tv.colum <span class="op">=</span> <span class="fl">8L</span>,</span>
<span>    naming <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    plots <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    mfcol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>    num.plot <span class="op">=</span> <span class="fl">9</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="dmr_at_memory_files/figure-html/dmr3-1.png" width="1728"></p>
<p>The hyper-DMRs on chromosome “1”</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hyp_dmrs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dmrfinder.html">dmrfinder</a></span><span class="op">(</span></span>
<span>    obj <span class="op">=</span> <span class="va">dmps</span>,</span>
<span>    hyper <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    min.signal <span class="op">=</span> <span class="fl">10L</span>,</span>
<span>    win.size <span class="op">=</span> <span class="fl">200</span>,</span>
<span>    step.size <span class="op">=</span> <span class="fl">50</span>,</span>
<span>    stat <span class="op">=</span> <span class="st">"count"</span>,</span>
<span>    tv.colum <span class="op">=</span> <span class="fl">8L</span>,</span>
<span>    naming <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    plots <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    mfcol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>    num.plot <span class="op">=</span> <span class="fl">9</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="dmr_at_memory_files/figure-html/dmr2-1.png" width="1728"></p>
<p>The hypo-DMRs on chromosome “2”</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hyp_dmrs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dmrfinder.html">dmrfinder</a></span><span class="op">(</span></span>
<span>    obj <span class="op">=</span> <span class="va">dmps</span>,</span>
<span>    hyper <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    min.signal <span class="op">=</span> <span class="fl">10L</span>,</span>
<span>    win.size <span class="op">=</span> <span class="fl">200</span>,</span>
<span>    step.size <span class="op">=</span> <span class="fl">50</span>,</span>
<span>    stat <span class="op">=</span> <span class="st">"count"</span>,</span>
<span>    tv.colum <span class="op">=</span> <span class="fl">8L</span>,</span>
<span>    naming <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    plots <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    mfcol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>    plot.chr <span class="op">=</span> <span class="st">"2"</span>,</span>
<span>    num.plot <span class="op">=</span> <span class="fl">9</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="dmr_at_memory_files/figure-html/dmr4-1.png" width="1728"></p>
<p>Observe that the same DMR can be present both groups. control and
treatment, where some DMPs can share and some not, or simply the DMR can
be mostly hypo-methylated in control group and mostly hyper-methylated
in the treatment group.</p>
<p>It is the user responsibility to evaluate whether a DMR is specific
for one of groups. That is, a further downstream analysis must identify
the DMRs that carry discriminatory power to distinguish control
individuals from the treatment group, which can be done, e.g., applying
a machine learning approach.</p>
</div>
<div class="section level2">
<h2 id="references-">References.<a class="anchor" aria-label="anchor" href="#references-"></a>
</h2>
<div id="refs" class="references csl-bib-body">
<div id="ref-Sanchez2016" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">R.
Sanchez, S. A. Mackenzie, <a href="https://doi.org/10.1371/journal.pone.0150427" class="external-link"><span class="nocase">Information Thermodynamics of Cytosine DNA
Methylation</span></a>. <em>PLOS ONE</em> <strong>11</strong>, e0150427
(2016).</div>
</div>
<div id="ref-Sanchez2019" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">R.
Sanchez, X. Yang, T. Maher, S. Mackenzie, <a href="https://doi.org/10.3390/ijms20215343" class="external-link"><span class="nocase">Discrimination of DNA Methylation Signal from Background
Variation for Clinical Diagnostics</span></a>. <em>Int. J. Mol.
Sci.</em> <strong>20</strong>, 5343 (2019).</div>
</div>
<div id="ref-Greiner2000" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">M.
Greiner, D. Pfeiffer, R. D. Smith, <a href="https://doi.org/10.1016/S0167-5877(00)00115-X" class="external-link"><span class="nocase">Principles and practical application of the
receiver-operating characteristic analysis for diagnostic
tests</span></a>. <em>Preventive Veterinary Medicine</em>
<strong>45</strong>, 23–41 (2000).</div>
</div>
<div id="ref-Carter2016" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">J.
V. Carter, J. Pan, S. N. Rai, S. Galandiuk, <a href="https://doi.org/10.1016/j.surg.2015.12.029" class="external-link"><span class="nocase">ROC-ing along: Evaluation and interpretation of receiver
operating characteristic curves</span></a>. <em>Surgery</em>
<strong>159</strong>, 1638–1645 (2016).</div>
</div>
<div id="ref-Harpaz2013" class="csl-entry">
<div class="csl-left-margin">5. </div>
<div class="csl-right-inline">R.
Harpaz, <em>et al.</em>, <a href="https://doi.org/10.1038/clpt.2013.24.Performance" class="external-link"><span class="nocase">Performance of Pharmacovigilance Signal Detection
Algorithms for the FDA Adverse Event Reporting System</span></a>.
<em>Clin Pharmacol ther</em> <strong>93</strong>, 1–19 (2013).</div>
</div>
<div id="ref-Kruspe2017" class="csl-entry">
<div class="csl-left-margin">6. </div>
<div class="csl-right-inline">S.
Kruspe, <em>et al.</em>, <a href="https://doi.org/10.1016/j.omtn.2017.08.004" class="external-link"><span class="nocase">Rapid and Sensitive Detection of Breast Cancer Cells in
Patient Blood with Nuclease-Activated Probe Technology</span></a>.
<em>Molecular Therapy - Nucleic Acids</em> <strong>8</strong>, 542–557
(2017).</div>
</div>
<div id="ref-Wahlberg2016" class="csl-entry">
<div class="csl-left-margin">7. </div>
<div class="csl-right-inline">P.
Wahlberg, <em>et al.</em>, <a href="https://doi.org/10.2217/epi-2016-0052" class="external-link"><span class="nocase">DNA
methylome analysis of acute lymphoblastic leukemia cells reveals
stochastic de novo DNA methylation in CpG islands</span></a>.
<em>Epigenomics</em> <strong>8</strong>, 1367–1387 (2016).</div>
</div>
<div id="ref-Sanchez2020" class="csl-entry">
<div class="csl-left-margin">8. </div>
<div class="csl-right-inline">R.
Sanchez, S. A. Mackenzie, <a href="https://doi.org/10.1038/s41598-020-58123-2" class="external-link"><span class="nocase">Integrative Network Analysis of Differentially Methylated
and Expressed Genes for Biomarker Identification in Leukemia</span></a>.
<em>Scientific Reports</em> <strong>10</strong>, 2123 (2020).</div>
</div>
<div id="ref-Severin2011" class="csl-entry">
<div class="csl-left-margin">9. </div>
<div class="csl-right-inline">P.
M. D. Severin, X. Zou, H. E. Gaub, K. Schulten, <a href="https://doi.org/10.1093/nar/gkr578" class="external-link"><span class="nocase">Cytosine
methylation alters DNA mechanical properties</span></a>. <em>Nucleic
acids research</em> <strong>39</strong>, 8740–51 (2011).</div>
</div>
<div id="ref-Perez2012" class="csl-entry">
<div class="csl-left-margin">10. </div>
<div class="csl-right-inline">A.
Pérez, <em>et al.</em>, <a href="https://doi.org/10.1016/j.bpj.2012.03.056" class="external-link"><span class="nocase">Impact of methylation on the physical properties of
DNA</span></a>. <em>Biophysical Journal</em> <strong>102</strong>,
2140–2148 (2012).</div>
</div>
<div id="ref-Yusufaly2013" class="csl-entry">
<div class="csl-left-margin">11. </div>
<div class="csl-right-inline">T.
I. Yusufaly, Y. Li, W. K. Olson, <a href="https://doi.org/10.1021/jp409887t" class="external-link"><span class="nocase">5-Methylation of cytosine in CG:CG base-pair steps: a
physicochemical mechanism for the epigenetic control of DNA
nanomechanics.</span></a> <em>The journal of physical chemistry. B</em>
<strong>117</strong>, 16436–42 (2013).</div>
</div>
<div id="ref-Teng2018" class="csl-entry">
<div class="csl-left-margin">12. </div>
<div class="csl-right-inline">X.
Teng, W. Hwang, <a href="https://doi.org/10.1016/j.bpj.2018.03.022" class="external-link"><span class="nocase">Effect of Methylation on Local Mechanics and Hydration
Structure of DNA</span></a>. <em>Biophysical Journal</em>
<strong>114</strong>, 1791–1803 (2018).</div>
</div>
<div id="ref-Hodges2020" class="csl-entry">
<div class="csl-left-margin">13. </div>
<div class="csl-right-inline">A.
J. Hodges, N. O. Hudson, B. A. Buck-Koehntop, <a href="https://doi.org/10.1016/j.jmb.2019.09.012" class="external-link"><span class="nocase">Cys2His2 Zinc Finger Methyl-CpG Binding Proteins: Getting
a Handle on Methylated DNA</span></a>. <em>Journal of Molecular
Biology</em> <strong>432</strong>, 1640–1660 (2020).</div>
</div>
<div id="ref-Ngo2016" class="csl-entry">
<div class="csl-left-margin">14. </div>
<div class="csl-right-inline">T.
T. M. Ngo, <em>et al.</em>, <a href="https://doi.org/10.1038/ncomms10813" class="external-link"><span class="nocase">Effects
of cytosine modifications on DNA flexibility and nucleosome mechanical
stability</span></a>. <em>Nature Communications</em> <strong>7</strong>,
10813 (2016).</div>
</div>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code> on the system on
which this document was compiled running pandoc 2.19.2:</p>
<pre><code><span><span class="co">## R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/New_York</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] grid      stats4    stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [8] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] gridExtra_2.3        ggplot2_3.4.2        MethylIT.utils_0.1.1</span></span>
<span><span class="co">##  [4] MethylIT_0.3.2.7     rtracklayer_1.60.0   GenomicRanges_1.52.0</span></span>
<span><span class="co">##  [7] GenomeInfoDb_1.36.1  IRanges_2.34.1       S4Vectors_0.38.1    </span></span>
<span><span class="co">## [10] BiocGenerics_0.46.0  knitr_1.43          </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] rstudioapi_0.14             jsonlite_1.8.5             </span></span>
<span><span class="co">##   [3] magrittr_2.0.3              modeltools_0.2-23          </span></span>
<span><span class="co">##   [5] rmarkdown_2.22              fs_1.6.2                   </span></span>
<span><span class="co">##   [7] BiocIO_1.10.0               zlibbioc_1.46.0            </span></span>
<span><span class="co">##   [9] ragg_1.2.5                  vctrs_0.6.3                </span></span>
<span><span class="co">##  [11] memoise_2.0.1               Rsamtools_2.16.0           </span></span>
<span><span class="co">##  [13] RCurl_1.98-1.12             htmltools_0.5.5            </span></span>
<span><span class="co">##  [15] S4Arrays_1.0.4              Formula_1.2-5              </span></span>
<span><span class="co">##  [17] pROC_1.18.2                 caret_6.0-94               </span></span>
<span><span class="co">##  [19] sass_0.4.6                  parallelly_1.36.0          </span></span>
<span><span class="co">##  [21] bslib_0.5.0                 desc_1.4.2                 </span></span>
<span><span class="co">##  [23] sandwich_3.0-2              plyr_1.8.8                 </span></span>
<span><span class="co">##  [25] zoo_1.8-12                  lubridate_1.9.2            </span></span>
<span><span class="co">##  [27] cachem_1.0.8                GenomicAlignments_1.36.0   </span></span>
<span><span class="co">##  [29] lifecycle_1.0.3             minpack.lm_1.2-3           </span></span>
<span><span class="co">##  [31] iterators_1.0.14            pkgconfig_2.0.3            </span></span>
<span><span class="co">##  [33] Matrix_1.5-4.1              R6_2.5.1                   </span></span>
<span><span class="co">##  [35] fastmap_1.1.1               GenomeInfoDbData_1.2.10    </span></span>
<span><span class="co">##  [37] MatrixGenerics_1.12.2       future_1.32.0              </span></span>
<span><span class="co">##  [39] digest_0.6.31               colorspace_2.1-0           </span></span>
<span><span class="co">##  [41] AnnotationDbi_1.62.1        rprojroot_2.0.3            </span></span>
<span><span class="co">##  [43] textshaping_0.3.6           RSQLite_2.3.1              </span></span>
<span><span class="co">##  [45] randomForest_4.7-1.1        fansi_1.0.4                </span></span>
<span><span class="co">##  [47] timechange_0.2.0            httr_1.4.6                 </span></span>
<span><span class="co">##  [49] compiler_4.3.0              bit64_4.0.5                </span></span>
<span><span class="co">##  [51] withr_2.5.0                 BiocParallel_1.34.2        </span></span>
<span><span class="co">##  [53] betareg_3.1-4               DBI_1.1.3                  </span></span>
<span><span class="co">##  [55] highr_0.10                  MASS_7.3-60                </span></span>
<span><span class="co">##  [57] lava_1.7.2.1                DelayedArray_0.26.3        </span></span>
<span><span class="co">##  [59] rjson_0.2.21                ModelMetrics_1.2.2.2       </span></span>
<span><span class="co">##  [61] tools_4.3.0                 lmtest_0.9-40              </span></span>
<span><span class="co">##  [63] future.apply_1.11.0         nnet_7.3-19                </span></span>
<span><span class="co">##  [65] glue_1.6.2                  restfulr_0.0.15            </span></span>
<span><span class="co">##  [67] nlme_3.1-162                reshape2_1.4.4             </span></span>
<span><span class="co">##  [69] generics_0.1.3              recipes_1.0.6              </span></span>
<span><span class="co">##  [71] gtable_0.3.3                class_7.3-22               </span></span>
<span><span class="co">##  [73] data.table_1.14.8           flexmix_2.3-19             </span></span>
<span><span class="co">##  [75] utf8_1.2.3                  XVector_0.40.0             </span></span>
<span><span class="co">##  [77] foreach_1.5.2               pillar_1.9.0               </span></span>
<span><span class="co">##  [79] stringr_1.5.0               genefilter_1.82.1          </span></span>
<span><span class="co">##  [81] splines_4.3.0               dplyr_1.1.2                </span></span>
<span><span class="co">##  [83] lattice_0.21-8              survival_3.5-5             </span></span>
<span><span class="co">##  [85] bit_4.0.5                   annotate_1.78.0            </span></span>
<span><span class="co">##  [87] tidyselect_1.2.0            Biostrings_2.68.1          </span></span>
<span><span class="co">##  [89] SummarizedExperiment_1.30.2 xfun_0.39                  </span></span>
<span><span class="co">##  [91] Biobase_2.60.0              hardhat_1.3.0              </span></span>
<span><span class="co">##  [93] timeDate_4022.108           matrixStats_1.0.0          </span></span>
<span><span class="co">##  [95] proto_1.0.0                 stringi_1.7.12             </span></span>
<span><span class="co">##  [97] yaml_2.3.7                  evaluate_0.21              </span></span>
<span><span class="co">##  [99] codetools_0.2-19            tibble_3.2.1               </span></span>
<span><span class="co">## [101] cli_3.6.1                   rpart_4.1.19               </span></span>
<span><span class="co">## [103] nls2_0.3-3                  xtable_1.8-4               </span></span>
<span><span class="co">## [105] systemfonts_1.0.4           munsell_0.5.0              </span></span>
<span><span class="co">## [107] jquerylib_0.1.4             Rcpp_1.0.10                </span></span>
<span><span class="co">## [109] globals_0.16.2              png_0.1-8                  </span></span>
<span><span class="co">## [111] XML_3.99-0.14               parallel_4.3.0             </span></span>
<span><span class="co">## [113] pkgdown_2.0.7               gower_1.0.1                </span></span>
<span><span class="co">## [115] blob_1.2.4                  bitops_1.0-7               </span></span>
<span><span class="co">## [117] listenv_0.9.0               ipred_0.9-14               </span></span>
<span><span class="co">## [119] scales_1.2.1                prodlim_2023.03.31         </span></span>
<span><span class="co">## [121] purrr_1.0.1                 crayon_1.5.2               </span></span>
<span><span class="co">## [123] rlang_1.1.1                 KEGGREST_1.40.0</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
