<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Association Between Gene Expression and Cytosine DNA Methylation at gene-body • MethylIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Association Between Gene Expression and Cytosine DNA Methylation at gene-body">
<meta property="og:description" content="MethylIT">
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/MethylIT/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="GenExp_Methylation_association_hyper_down-regulated_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Association Between Gene Expression and Cytosine DNA Methylation at gene-body</h1>
            
            <h4 class="date">04 November 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/master/vignettes/GenExp_Methylation_association_hyper_down-regulated.Rmd"><code>vignettes/GenExp_Methylation_association_hyper_down-regulated.Rmd</code></a></small>
      <div class="hidden name"><code>GenExp_Methylation_association_hyper_down-regulated.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This document is Supplementary Information for the manuscript: <em>“Segregation of an MSH1 RNAi transgene produces heritable non-genetic memory in association with methylome reprogramming”</em> <span class="citation">(1)</span>. Statistical analysis supporting the association between gene expression and cytosine DNA methylation at gene-body is provided.</p>
    </div>
    
<style type="text/css">

.list-group-item.active, .list-group-item.active:focus, 
.list-group-item.active:hover {
    z-index: 2;
    color: #fff;
    background-color: #337ab7;
    border-color: #337ab7;
}

.tocify-subheader .tocify-item {
  font-size: 0.80em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify-header .tocify-item {
  font-size: 0.85em;
  padding-left: 20px;
  text-indent: 0;
}

body{ /* Normal  */
      font-size: 18px;
      font-family: "Times New Roman", Times, serif;
      text-align: justify
  }
td {  /* Table  */
  font-size: 8px;
}

h1.title {
  font-size: 38px;
  font-family: "Times New Roman", Times, serif;
  color: DarkRed;
  .subTitle {
  font-size: 24px;
  font-family: "Times New Roman", Times, serif;
  color: DarkRed;
  }
}

h1 { /* Header 1 */
  font-size: 28px;
  font-family: "Times New Roman", Times, serif;
  color: DarkBlue;
}
h2 { /* Header 2 */
    font-size: 22px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
h3 { /* Header 3 */
   font-size: 18px;
   color: DarkBlue;
   font-family: "Times New Roman", Times, serif;
}
code.r{ /* Code block */
    font-size: 12px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>
<div id="background" class="section level1">
<h1 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h1>
<p>The classical statistical approach to analyze the association between two variables is based in correlation tests. However, small or zero correlation coefficient does not necessarily implies independence of the tested variables/processes and, consequently, does not implies the lack of association at all. There various steps involved in investigating the dependence between two random variables. The state of the art include the application of copula distributions. Herein, we provide study of the statistical dependence structure between gene expression and methylation in memory line. To uncover the association between gene expression and cytosine DNA methylation, first, the probability distributions of gene expression (in terms of logarithm of fold changes: logFC) and methylation divergences are estimated.</p>
<p>Next, the joint probability distribution is estimated, expressed in term of copula distribution. All the analyses are performed in R and can be reproduced as presented here. The association is investigated with the application of <a href="https://en.wikipedia.org/wiki/Copula_(probability_theory)">copula distributions</a>.</p>
</div>
<div id="required-r-packages" class="section level1">
<h1 class="hasAnchor">
<a href="#required-r-packages" class="anchor"></a>Required R packages</h1>
<p>With the exception of <em>usefr</em> R package, all the packages are available in Bioconductor and in CRAN repository. Package <em>usefr</em> is available in GitHub at <a href="https://github.com/genomaths/usefr" class="uri">https://github.com/genomaths/usefr</a> You can install usefr package from GitHub typing the following command in the R console:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_git</a></span><span class="op">(</span><span class="st">"https://github.com/genomaths/usefr.git"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span> <span class="va"><a href="https://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></span> <span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span> <span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span> <span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span> <span class="va"><a href="http://bioinf.wehi.edu.au/edgeR">edgeR</a></span> <span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span> <span class="va">usefr</span> <span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span> <span class="va">plot3D</span> <span class="op">)</span></code></pre></div>
<pre><code>## Warning in fun(libname, pkgname): couldn't connect to display ":0"</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span> <span class="va"><a href="http://copula.r-forge.r-project.org/">copula</a></span> <span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span> <span class="va"><a href="http://coin.r-forge.r-project.org">coin</a></span> <span class="op">)</span></code></pre></div>
</div>
<div id="analysis-for-down-regulated-genes" class="section level1">
<h1 class="hasAnchor">
<a href="#analysis-for-down-regulated-genes" class="anchor"></a>Analysis for down-regulated genes</h1>
<div id="data-sets" class="section level2">
<h2 class="hasAnchor">
<a href="#data-sets" class="anchor"></a>Data sets</h2>
<p>Gene expression data sets were derived using the R package edgeR. All the genes with the minimal quality of gene expression differences are included in the analyses. That is, to evaluate the existence of an association between gene expression and cytosine methylation all the genes with meaningful expression and methylation values must be included, regardless they would be differentially methylated and/or differentially expressed or not.</p>
<p>Gene expression data for Arabidopsis Memory first generation (G1)</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## ===== Download methylation data from Github ========</span>
<span class="va">url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"github.com/genomaths/genomaths.github.io/raw/master/supp-inf/"</span>, 
              <span class="st">"mm_nm_wt_transgen_meth_density_and_gene-exp_at_genes_06-01-19."</span>,
              <span class="st">"RData"</span><span class="op">)</span>
<span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".RData"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="va">url</span>, destfile <span class="op">=</span> <span class="va">temp</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.remove</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">temp</span>, <span class="va">url</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Gene expresion data set</span>
<span class="va">deg_mm1_wt1</span></code></pre></div>
<pre><code>## An object of class "DGEExact"
## $table
##                 logFC    logCPM       PValue     adj.pval
## AT1G01010 -0.60092985  3.066880 9.377730e-05 0.0004180724
## AT1G01020 -0.60729200  2.913263 2.803733e-05 0.0001409788
## AT1G01030 -0.25438232 -1.478816 5.857299e-01 0.6890000078
## AT1G01040  0.03597925  5.632805 7.833824e-01 0.8505857888
## AT1G01046 -1.61271935 -2.996378 2.235591e-02 0.0498548032
## 20284 more rows ...
## 
## $comparison
## [1] "WT1" "MM1"
## 
## $genes
##           genes.seqnames genes.start genes.end genes.strand genes.type
## AT1G01010              1        3631      5899            +       gene
## AT1G01020              1        6788      9130            -       gene
## AT1G01030              1       11649     13714            -       gene
## AT1G01040              1       23121     31227            +       gene
## AT1G01046              1       28500     28706            +       gene
##           genes.gene_biotype genes.gene_id genes.gene_name length
## AT1G01010     protein_coding     AT1G01010          NAC001   2269
## AT1G01020     protein_coding     AT1G01020            ARV1   2343
## AT1G01030     protein_coding     AT1G01030            NGA3   2066
## AT1G01040     protein_coding     AT1G01040            DCL1   8107
## AT1G01046              miRNA     AT1G01046         MIR838A    207
## 20284 more rows ...</code></pre>
<p>Methylation dataset contains the densities of methylation levels from control (Col-0) and Memory lines samples at DMP on gene-body regions. Densities were computed using function <a href="https://genomaths.github.io/MethylIT_utils_HTML_Manual/getGRegionsStat-methods.html">getGRegionsStat</a> from the R package <a href="https://github.com/genomaths/MethylIT.utils">MethylIT.utils</a>.</p>
<p>Methylation data for a memory G1 sample</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmp_den_genes</span><span class="op">[</span>, <span class="st">"M_1_105"</span><span class="op">]</span></code></pre></div>
<pre><code>## GRanges object with 33164 ranges and 1 metadata column:
##             seqnames            ranges strand |   M_1_105
##                &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   AT1G01010        1         3631-5899      * |  0.841607
##   AT1G01020        1         6788-9130      * |  0.389668
##   AT1G03987        1       11101-11372      * |  0.000000
##   AT1G01030        1       11649-13714      * |  0.000000
##   AT1G01040        1       23121-31227      * |  6.776141
##         ...      ...               ...    ... .       ...
##   AT5G09945        5 26966885-26967079      * |   0.00000
##   AT5G67630        5 26967378-26969400      * |   1.28801
##   AT5G67640        5 26969516-26970668      * |   0.00000
##   AT5G09955        5 26971389-26971689      * |   1.05172
##   AT5G09965        5 26972177-26972644      * |   0.00000
##   -------
##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
<div id="dataset-for-the-first-generation" class="section level2">
<h2 class="hasAnchor">
<a href="#dataset-for-the-first-generation" class="anchor"></a>Dataset for the first generation</h2>
<p>The analysis will be performed with Memory line (M1) and wild type (WT1) control, both of them, MM1 and WT1, first generation G1.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html">na.omit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">deg_mm1_wt1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">dmp_den_genes</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">dmp_den_m1</span> <span class="op">&lt;-</span> <span class="va">dmp_den_genes</span><span class="op">[</span><span class="va">idx</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"M_1_105"</span>, <span class="st">"M_1_168"</span>, <span class="st">"M_1_179"</span>, <span class="st">"M_1_27"</span>, <span class="st">"M_1_3"</span><span class="op">)</span><span class="op">]</span>
<span class="va">dmp_den_wt1</span> <span class="op">&lt;-</span> <span class="va">dmp_den_genes</span><span class="op">[</span><span class="va">idx</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"W_1_1"</span>, <span class="st">"W_1_3"</span>, <span class="st">"W_1_4"</span>, <span class="st">"W_1_5"</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<p>The means of methylation levels densities are computed for each group</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">mcols</span><span class="op">(</span><span class="va">dmp_den_wt1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dmp_den_wt1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu">mcols</span><span class="op">(</span><span class="va">dmp_den_m1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowMeans</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dmp_den_m1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>The absolute difference of the group density means of methylation levels between control (Col-0) and Memory lines at gene-body regions (G1).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmp_den_wt1_mm1</span> <span class="op">&lt;-</span> <span class="va">dmp_den_m1</span>
<span class="fu">mcols</span><span class="op">(</span><span class="va">dmp_den_wt1_mm1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dmp_den_m1</span><span class="op">)</span><span class="op">)</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dmp_den_wt1</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dmp_den_wt1_mm1</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"dmp_den"</span>
<span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">dmp_den_wt1_mm1</span><span class="op">$</span><span class="va">dmp_den</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>
<span class="va">dmp_den_wt1_mm1</span> <span class="op">&lt;-</span> <span class="va">dmp_den_wt1_mm1</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span>
<span class="va">dmp_den_wt1_mm1</span></code></pre></div>
<pre><code>## GRanges object with 20022 ranges and 1 metadata column:
##             seqnames            ranges strand |   dmp_den
##                &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
##   AT1G01010        1         3631-5899      * |  0.734204
##   AT1G01020        1         6788-9130      * |  0.304498
##   AT1G01030        1       11649-13714      * |  0.180618
##   AT1G01040        1       23121-31227      * |  6.571474
##   AT1G01046        1       28500-28706      * |  2.597185
##         ...      ...               ...    ... .       ...
##   AT4G40060        4 18571239-18573084      * |  0.166230
##   AT4G40065        4 18574503-18575585      * |  0.121346
##   AT4G40070        4 18576216-18577774      * |  0.151446
##   AT4G40085        4 18579194-18580702      * |  0.161783
##   AT4G40080        4 18579314-18580717      * |  0.141613
##   -------
##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
<p>Final variables: |<em>group means density differences</em>| and <span class="math inline">\(|logFC|\)</span>. For the sake of brevity and clarity, we shall call the first variable: <em>divergence of methylation densities</em> or simply <em>methylation divergence</em> (<span class="math inline">\(MD\)</span>).</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">.deg_mm1_wt1</span> <span class="op">&lt;-</span> <span class="va">deg_mm1_wt1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">dmp_den_wt1_mm1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">deg_mm1_wt1</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span>

<span class="va">MD</span> <span class="op">&lt;-</span> <span class="va">dmp_den_wt1_mm1</span><span class="op">$</span><span class="va">dmp_den</span> <span class="co"># Methylation (|group mean density differences|)</span>
<span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">.deg_mm1_wt1</span><span class="op">$</span><span class="va">table</span><span class="op">$</span><span class="va">logFC</span> <span class="op">&lt;</span> <span class="fl">0</span> <span class="op">)</span> <span class="co"># Down-regulated</span>
<span class="va">logFC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">.deg_mm1_wt1</span><span class="op">$</span><span class="va">table</span><span class="op">$</span><span class="va">logFC</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span><span class="op">)</span> <span class="co"># Gene Expression (|logFC|)</span>
<span class="va">MD</span> <span class="op">&lt;-</span> <span class="va">MD</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span>
<span class="fu">DataFrame</span><span class="op">(</span><span class="va">MD</span>,<span class="va">logFC</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 13962 rows and 2 columns
##              MD     logFC
##       &lt;numeric&gt; &lt;numeric&gt;
## 1      0.734204  0.600930
## 2      0.304498  0.607292
## 3      0.180618  0.254382
## 4      2.597185  1.612719
## 5      0.823158  0.515993
## ...         ...       ...
## 13958  0.383503  0.371768
## 13959  0.532277  0.136153
## 13960  0.166230  0.201576
## 13961  0.161783  0.115828
## 13962  0.141613  1.628195</code></pre>
<p>There are 13962 genes to be included in the downstream analysis.</p>
</div>
<div id="distribution-of-the-divergence-of-methylation-level-density-at-gene-body--mm1-vs-wt1" class="section level2">
<h2 class="hasAnchor">
<a href="#distribution-of-the-divergence-of-methylation-level-density-at-gene-body--mm1-vs-wt1" class="anchor"></a>Distribution of the divergence of methylation level density at gene-body. MM1 vs WT1</h2>
<p>Searching for the best fitted distribution functions by using R functions <a href="https://www.rdocumentation.org/packages/MASS/versions/7.3-51.4/topics/fitdistr">fitdistr</a> from <a href="https://www.rdocumentation.org/packages/MASS/versions/7.3-50">MASS</a> R package and <a href="https://genomaths.github.io/usefr_manual/fitCDF.html">fitCDF</a> from <a href="https://github.com/genomaths/usefr">usefr</a> R package did not yield a suitable results. The best fitted model was found for a mixture distribution model obtained with the function <a href="https://genomaths.github.io/usefr_manual/fitMixDist.html">fitMixDist</a> from <a href="https://github.com/genomaths/usefr">usefr</a> R package. The best fitted model for <span class="math inline">\(MD\)</span>, from comparison MM1 <em>vs</em> WT1, was a mixture of Normal and Weibull distributions.</p>
<p>We have left the algorithm to estimate the initial value</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">MD</span><span class="op">[</span> <span class="va">MD</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span> <span class="co"># Weibull distribution is defineed for MD &gt; 0.</span>

<span class="va">meth_FIT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/fitMixDist.html">fitMixDist</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">x</span>, args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>norm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>mean <span class="op">=</span> <span class="cn">NA</span>, sd <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
                                          weibull <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>shape <span class="op">=</span> <span class="cn">NA</span>, scale <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span><span class="op">)</span>,
                  dens <span class="op">=</span> <span class="cn">TRUE</span>, npoints <span class="op">=</span> <span class="fl">900</span><span class="op">)</span></code></pre></div>
<pre><code>## fitting ...
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |======================================================================| 100%
## *** Performing nonlinear regression model  crossvalidation...</code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">meth_FIT</span><span class="op">$</span><span class="va">fit</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Parameters:
##       Estimate Std. Error t value Pr(&gt;|t|)    
## mean  0.280637   0.001605   174.8   &lt;2e-16 ***
## sd    0.173838   0.001339   129.9   &lt;2e-16 ***
## shape 2.110394   0.019824   106.5   &lt;2e-16 ***
## scale 1.386566   0.008017   173.0   &lt;2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 0.02186 on 1022 degrees of freedom
## Number of iterations to termination: 14 
## Reason for termination: Relative error in the sum of squares is at most `ftol'.</code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\n Mixture proportions (phi):"</span>, <span class="va">meth_FIT</span><span class="op">$</span><span class="va">phi</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##  Mixture proportions (phi): 0.4116523 0.5883477</code></pre>
<p>The graphic for histogram and density distribution curves:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">4</span>, <span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span>, font.lab<span class="op">=</span><span class="fl">2</span>, font<span class="op">=</span><span class="fl">2</span>,font.axis<span class="op">=</span><span class="fl">2</span>,family<span class="op">=</span><span class="st">"serif"</span> , lwd <span class="op">=</span> <span class="fl">1.2</span>, cex <span class="op">=</span> <span class="fl">1.4</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">240</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">1</span>, <span class="co">#ylim = c(0, 0.021), cex = 1,</span>
     panel.first<span class="op">=</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, pch<span class="op">=</span><span class="fl">16</span>, cex<span class="op">=</span><span class="fl">1e6</span>, col<span class="op">=</span><span class="st">"grey95"</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/graphics/grid.html">grid</a></span><span class="op">(</span>col<span class="op">=</span><span class="st">"white"</span>, lty <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">}</span>, main <span class="op">=</span> <span class="va">text</span>,
     cex.main <span class="op">=</span> <span class="fl">0.9</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span><span class="op">)</span>, family <span class="op">=</span> <span class="st">"serif"</span>,
     col <span class="op">=</span> <span class="st">"cyan1"</span>, border <span class="op">=</span> <span class="st">"deepskyblue"</span>, xaxt <span class="op">=</span><span class="st">"n"</span>, yaxt <span class="op">=</span> <span class="st">"n"</span>, ann <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">1</span>, padj <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.8</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.02</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">2</span>, hadj <span class="op">=</span> <span class="fl">0.7</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.8</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.02</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>, text <span class="op">=</span> <span class="st">"Divergence of methylation level density at gene-body"</span>,
      line <span class="op">=</span> <span class="fl">1.6</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>, text <span class="op">=</span> <span class="st">"Density distribution"</span>, line <span class="op">=</span> <span class="fl">2.1</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span><span class="st">"(Normal &amp; Weibull Mixture Dists.)"</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">10</span>, by <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span><span class="op">(</span><span class="va">x1</span>, <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/mixtdistr.html">dmixtdistr</a></span><span class="op">(</span><span class="va">x1</span>, phi <span class="op">=</span> <span class="va">meth_FIT</span><span class="op">$</span><span class="va">phi</span>, arg <span class="op">=</span> <span class="va">meth_FIT</span><span class="op">$</span><span class="va">args</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span></code></pre></div>
<p><img src="GenExp_Methylation_association_hyper_down-regulated_files/figure-html/graphic_meth-1.png" width="700"></p>
</div>
<div id="distribution-of-the-logfc--mm1-vs-wt1" class="section level2">
<h2 class="hasAnchor">
<a href="#distribution-of-the-logfc--mm1-vs-wt1" class="anchor"></a>Distribution of the <span class="math inline">\(|logFC|\)</span>. MM1 vs WT1</h2>
<p>The best fitted model for the <span class="math inline">\(|logFC|\)</span>, from comparison MM1 <em>vs</em> WT1, was a mixture of Gamma and Weibull distributions. Function <a href="https://genomaths.github.io/usefr_manual/fitMixDist.html">fitMixDist</a> calls the mixture function models set up in function <a href="https://genomaths.github.io/usefr_manual/mixtdistr.html">mixtdistr</a></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">logFC</span><span class="op">[</span> <span class="va">logFC</span> <span class="op">&gt;</span> <span class="fl">0</span> <span class="op">]</span>

<span class="va">gexp_FIT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/fitMixDist.html">fitMixDist</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">x</span>, args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>weibull <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>shape <span class="op">=</span> <span class="cn">NA</span>, scale <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
                                          norm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>mean <span class="op">=</span> <span class="cn">NA</span>, sd <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span><span class="op">)</span>,
                       dens <span class="op">=</span> <span class="cn">TRUE</span>, npoints <span class="op">=</span> <span class="fl">900</span><span class="op">)</span></code></pre></div>
<pre><code>## fitting ...
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |======================================================================| 100%
## *** Performing nonlinear regression model  crossvalidation...</code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">gexp_FIT</span><span class="op">$</span><span class="va">fit</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Parameters:
##       Estimate Std. Error t value Pr(&gt;|t|)    
## shape 1.320692   0.005010  263.60   &lt;2e-16 ***
## scale 0.375421   0.001982  189.41   &lt;2e-16 ***
## mean  0.651172   0.037848   17.20   &lt;2e-16 ***
## sd    0.710641   0.022864   31.08   &lt;2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 0.03132 on 937 degrees of freedom
## Number of iterations to termination: 20 
## Reason for termination: Relative error in the sum of squares is at most `ftol'.</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\n Mixture proportions (phi):"</span>, <span class="va">gexp_FIT</span><span class="op">$</span><span class="va">phi</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##  Mixture proportions (phi): 0.7850078 0.2149922</code></pre>
<p>The graphic for histogram and density distribution curves:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span>, font.lab<span class="op">=</span><span class="fl">2</span>, font<span class="op">=</span><span class="fl">2</span>,font.axis<span class="op">=</span><span class="fl">2</span>,family<span class="op">=</span><span class="st">"serif"</span>, lwd <span class="op">=</span> <span class="fl">1.2</span>, cex <span class="op">=</span> <span class="fl">1.4</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">200</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">1</span>, <span class="co">#ylim = c(0, 0.021), cex = 1,</span>
     panel.first<span class="op">=</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, pch<span class="op">=</span><span class="fl">16</span>, cex<span class="op">=</span><span class="fl">1e6</span>, col<span class="op">=</span><span class="st">"grey95"</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/graphics/grid.html">grid</a></span><span class="op">(</span>col<span class="op">=</span><span class="st">"white"</span>, lty <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">}</span>, main <span class="op">=</span> <span class="va">text</span>, 
     cex.main <span class="op">=</span> <span class="fl">0.9</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">4</span><span class="op">)</span>, family <span class="op">=</span> <span class="st">"serif"</span>,
     col <span class="op">=</span> <span class="st">"cyan1"</span>, border <span class="op">=</span> <span class="st">"deepskyblue"</span>, xaxt <span class="op">=</span><span class="st">"n"</span>, yaxt <span class="op">=</span> <span class="st">"n"</span>, ann <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">1</span>, padj <span class="op">=</span> <span class="op">-</span><span class="fl">1.5</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.5</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.02</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">2</span>, hadj <span class="op">=</span> <span class="fl">0.4</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.5</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.02</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>, text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">logFC</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
      line <span class="op">=</span> <span class="fl">1.3</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>, text <span class="op">=</span> <span class="st">"Density distribution"</span>, line <span class="op">=</span> <span class="fl">1.6</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span><span class="st">"(Weibull &amp; Normal Mixture Dists.)"</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">10</span>, by <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span><span class="op">(</span><span class="va">x1</span>, <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/mixtdistr.html">dmixtdistr</a></span><span class="op">(</span><span class="va">x1</span>, phi <span class="op">=</span> <span class="va">gexp_FIT</span><span class="op">$</span><span class="va">phi</span>, arg <span class="op">=</span> <span class="va">gexp_FIT</span><span class="op">$</span><span class="va">args</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span></code></pre></div>
<p><img src="GenExp_Methylation_association_hyper_down-regulated_files/figure-html/logFC_grh-1.png" width="700"></p>
</div>
<div id="joint-distribution-of-md-and-logfc-expressed-in-terms-of-copula-distribution-" class="section level2">
<h2 class="hasAnchor">
<a href="#joint-distribution-of-md-and-logfc-expressed-in-terms-of-copula-distribution-" class="anchor"></a>Joint distribution of <span class="math inline">\(MD\)</span> and <span class="math inline">\(|logFC|\)</span> expressed in terms of Copula distribution.</h2>
<p>There various steps involved in investigating the dependence between two random variables. The state of the art in this subject include the application of copula distributions.</p>
<p>Sklar’s theorem (<span class="citation">(2, 3)</span>) named after Abe Sklar, provides the theoretical foundation for the application of copulas. Sklar’s theorem states that every multivariate cumulative distribution function (CDF):</p>
<p><span class="math display">\[H(x_1, ..., x_d)=Pr(X_1 \leq x_1, ..., X_d \leq x_d)\]</span></p>
<p>of any <span class="math inline">\(d-tuple\)</span> of continuous random variables <span class="math inline">\((X_1, ..., X_d)\)</span> can be expressed in terms of its marginals CDFs: <span class="math inline">\(F_i(x_i)=Pr(X_i \leq x_i)\)</span> a copula <span class="math inline">\(C\)</span>:</p>
<p><span class="math display">\[H(x_1, ..., x_d)=C(F_1(x_1), ..., F_d(x_d))\]</span> While Sklar showed that <span class="math inline">\(C\)</span> and <span class="math inline">\(F_i\)</span> are uniquely determined when H is known, a valid model for <span class="math inline">\((X_1, ..., X_d)\)</span> arises from the last equation whenever the “ingredients” are chosen from given parametric families of distribution (<span class="citation">(4)</span>): <span class="math display">\[F \in (F_\delta), C \in (C_\alpha)\]</span></p>
<p>Current literature on copula theory provides abundant information on models and copula families. Here, for the particular case of bi-dimensional copula defined for <span class="math inline">\(MD\)</span> and <span class="math inline">\(|logFC|\)</span>, we have chosen Farlie–Gumbel–Morgenstern family of copulas, which are defined for, each <span class="math inline">\(\theta \in [-1, 1]\)</span> by: <span class="math display">\[C_\alpha(u,
v)=uv+\alpha uv(1-u)(1-v)\]</span> <span class="math inline">\(u,v \in [0,1]\)</span>. The correlation coefficient <span class="math inline">\(\rho\)</span> is linked to <span class="math inline">\(\alpha\)</span> by the relationship: <span class="math inline">\(\rho=\frac{\alpha}{3}\)</span>, which ranges from <span class="math inline">\(-\frac{1}{3}\)</span> to <span class="math inline">\(\frac{1}{3}\)</span>, where <span class="math inline">\(u=F(x)\)</span>, <span class="math inline">\(v=G(y)\)</span>, with marginal CDFs: <span class="math inline">\(F(x)\)</span> and <span class="math inline">\(G(y)\)</span>.</p>
<div id="fitting-farliegumbelmorgenstern-copulas-with-md-and-logfc-marginal-cdfs" class="section level3">
<h3 class="hasAnchor">
<a href="#fitting-farliegumbelmorgenstern-copulas-with-md-and-logfc-marginal-cdfs" class="anchor"></a>Fitting Farlie–Gumbel–Morgenstern copulas with <span class="math inline">\(MD\)</span> and <span class="math inline">\(|logFC|\)</span> marginal CDFs</h3>
<p>The first step requires for provide the parameters values, previously estimated, of the marginal distributions, which were given throughout the function <a href="https://genomaths.github.io/usefr_manual/mixtdistr.html">mixtdistr</a>. The the specific parameter values are retrieved from the previous nonlinear fit obtained for the marginal CDFs:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">margins</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"mixtdistr"</span>, <span class="st">"mixtdistr"</span><span class="op">)</span>
<span class="va">parMargins</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>phi <span class="op">=</span> <span class="va">meth_FIT</span><span class="op">$</span><span class="va">phi</span>, arg <span class="op">=</span> <span class="va">meth_FIT</span><span class="op">$</span><span class="va">args</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>phi <span class="op">=</span> <span class="va">gexp_FIT</span><span class="op">$</span><span class="va">phi</span>, arg <span class="op">=</span> <span class="va">gexp_FIT</span><span class="op">$</span><span class="va">args</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>The parameter estimation and goodness-of-fit (GOF) test take advantages of the tools available in the R package <a href="https://www.rdocumentation.org/packages/copula/versions/0.999-19.1">copula</a>. Both, parameter estimation and GOF, can be accomplished with function <a href="https://genomaths.github.io/usefr_manual/bicopulaGOF.html">bicopulaGOF</a> from the R package <a href="https://github.com/genomaths/usefr">usefr</a>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fgm_cop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/bicopulaGOF.html">bicopulaGOF</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">MD</span>, y <span class="op">=</span> <span class="va">logFC</span>, copula <span class="op">=</span> <span class="st">"fgmCopula"</span>, sample.size <span class="op">=</span> <span class="fl">13000</span>, 
                       margins <span class="op">=</span> <span class="va">margins</span>, paramMargins <span class="op">=</span> <span class="va">parMargins</span>, nboots <span class="op">=</span> <span class="fl">10</span>, 
                       approach <span class="op">=</span> <span class="st">"chisq"</span>, breaks <span class="op">=</span> <span class="fl">30</span>, num.cores <span class="op">=</span> <span class="fl">6L</span>, seed <span class="op">=</span> <span class="fl">123</span>,
                       verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>The null hypothesis that the data follows FGM copula distribution is not rejected.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fgm_cop</span><span class="op">$</span><span class="va">gof</span></code></pre></div>
<pre><code>##   Chisq.stat   mc_p.value  sample.size    num.sampl 
## 1.117630e+03 1.818182e-01 1.300000e+04 1.000000e+01</code></pre>
<p>All the information needed about the fitted FGM copula is found in the object <em>fgm_cop$copula</em>, which is a <a href="https://www.rdocumentation.org/packages/copula/versions/0.999-19.1/topics/mvdc-class">mvdc</a> object from <a href="https://www.rdocumentation.org/packages/copula/versions/0.999-19.1">copula</a> package with several slots. In particular, the specific information about FGM copula is found in:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fgm_cop</span><span class="op">$</span><span class="va">copula</span><span class="op">@</span><span class="va">copula</span></code></pre></div>
<pre><code>## Farlie-Gumbel-Morgenstern (FGM) copula, dim. d = 2
##  param.: -0.509 
## Dimension:  2 
## Parameters:
##   alpha1.2   = -0.509199</code></pre>
</div>
<div id="p-p-plot--graphical-diagnostics-of-gof" class="section level3">
<h3 class="hasAnchor">
<a href="#p-p-plot--graphical-diagnostics-of-gof" class="anchor"></a>P-P plot. Graphical Diagnostics of GOF</h3>
<p>The P-P plot is build with function <a href="https://genomaths.github.io/usefr_manual/ppCplot.html">ppCplot</a> from the R package <a href="https://github.com/genomaths/usefr">usefr</a>. If the GOF is good enough, then it is expected that points in the graphics are closed to the diagonal (red) straight line. Details on how the computation is performed are provided in the help for function. <a href="https://genomaths.github.io/usefr_manual/ppCplot.html">ppCplot</a>.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/ppCplot.html">ppCplot</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">MD</span>, Y <span class="op">=</span> <span class="va">logFC</span>, copula <span class="op">=</span> <span class="va">fgm_cop</span><span class="op">$</span><span class="va">copula</span>, npoints <span class="op">=</span> <span class="fl">500</span>, col <span class="op">=</span> <span class="st">"blue"</span>, cex <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></code></pre></div>
<p><img src="GenExp_Methylation_association_hyper_down-regulated_files/figure-html/ppPlot-1.png" width="700"></p>
<p>Information on the data and copula used to yield the graphic are stored in the output variable <em>“g”</em>. For example, the empirical and theoretical probabilities used to built the graphic can be retrieved from the data frame: <em>“g$data”</em>:</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">DataFrame</span><span class="op">(</span><span class="va">g</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 500 rows and 2 columns
##        emprob    thprob
##     &lt;numeric&gt; &lt;numeric&gt;
## 1   0.2497493 0.2433976
## 2   0.0267870 0.0248706
## 3   0.2102134 0.2038670
## 4   0.0447644 0.0472481
## 5   0.1990403 0.1922045
## ...       ...       ...
## 496 0.2213150 0.2224191
## 497 0.0421143 0.0447297
## 498 0.2534737 0.2648909
## 499 0.4080361 0.4120571
## 500 0.1593611 0.1660077</code></pre>
</div>
<div id="three-dimensional-copula-density-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#three-dimensional-copula-density-plot" class="anchor"></a>Three-dimensional copula density plot</h3>
<p>The domain region <span class="math inline">\((X, Y)\)</span> is defined and function <a href="https://www.rdocumentation.org/packages/copula/versions/0.999-19.1/topics/Mvdc">dMvdc</a> is used to compute the density <span class="math inline">\(z\)</span> in the grid <span class="math inline">\([X,Y] \times [X,Y]\)</span> using the previous FMG fitted copula.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">X</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1.5</span>, length <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>
<span class="va">Y</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1.5</span>, length <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>
<span class="va">sp_rho</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/copula/man/assocMeasures.html">rho</a></span><span class="op">(</span><span class="va">fgm_cop</span><span class="op">$</span><span class="va">copula</span><span class="op">@</span><span class="va">copula</span><span class="op">)</span>,<span class="fl">3</span><span class="op">)</span> <span class="co"># Spearman's rho: -0.171</span>


<span class="va">z</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/outer.html">outer</a></span><span class="op">(</span><span class="va">X</span>, <span class="va">Y</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/copula/man/Mvdc.html">dMvdc</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cbind.html">cbind</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span>,<span class="va">fgm_cop</span><span class="op">$</span><span class="va">copula</span><span class="op">)</span><span class="op">)</span> <span class="op">*</span> <span class="fl">4</span>
<span class="va">phi</span> <span class="op">=</span> <span class="fl">20</span>; <span class="va">theta</span> <span class="op">=</span> <span class="fl">45</span>

<span class="va">zf</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/copula/man/Mvdc.html">dMvdc</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cbind.html">cbind</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span>,<span class="va">fgm_cop</span><span class="op">$</span><span class="va">copula</span><span class="op">)</span> <span class="op">*</span> <span class="fl">3.2</span>

<span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"Spearman's rho = "</span>, <span class="va">sp_rho</span><span class="op">)</span></code></pre></div>
<pre><code>## Spearman's rho =  -0.17</code></pre>
<p>An auxiliary function to draw the panels</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">panelfirst</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">pmat</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">7</span>,<span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, font<span class="op">=</span><span class="fl">2</span>,lwd <span class="op">=</span> <span class="fl">0.01</span>, cex <span class="op">=</span><span class="fl">1</span>, family <span class="op">=</span> <span class="st">"serif"</span><span class="op">)</span>
  <span class="va">YZ</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/plot3D/man/trans3D.html">trans3D</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span><span class="op">)</span>, <span class="va">Y</span>, <span class="va">z</span><span class="op">[</span><span class="fl">18</span>,<span class="op">]</span>, pmat <span class="op">=</span> <span class="va">pmat</span><span class="op">)</span>

  <span class="fu"><a href="https://rdrr.io/pkg/plot3D/man/scatter.html">scatter2D</a></span><span class="op">(</span><span class="va">YZ</span><span class="op">$</span><span class="va">x</span>, <span class="va">YZ</span><span class="op">$</span><span class="va">y</span>, colvar <span class="op">=</span> <span class="va">z</span><span class="op">[</span><span class="fl">18</span>,<span class="op">]</span>, lwd <span class="op">=</span> <span class="fl">1</span>, 
            type <span class="op">=</span> <span class="st">"l"</span>, add <span class="op">=</span> <span class="cn">TRUE</span>, colkey <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

  <span class="va">XZ</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/plot3D/man/trans3D.html">trans3D</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">X</span>, y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span><span class="op">)</span>,
                z <span class="op">=</span> <span class="va">z</span><span class="op">[</span>, <span class="fl">12</span><span class="op">]</span>, pmat <span class="op">=</span> <span class="va">pmat</span><span class="op">)</span>

  <span class="fu"><a href="https://rdrr.io/pkg/plot3D/man/scatter.html">scatter2D</a></span><span class="op">(</span><span class="va">XZ</span><span class="op">$</span><span class="va">x</span>, <span class="va">XZ</span><span class="op">$</span><span class="va">y</span>, colvar <span class="op">=</span> <span class="va">z</span><span class="op">[</span>, <span class="fl">12</span><span class="op">]</span>, lwd <span class="op">=</span> <span class="fl">1</span>, 
            type <span class="op">=</span> <span class="st">"l"</span>, add <span class="op">=</span> <span class="cn">TRUE</span>, colkey <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
  
<span class="op">}</span></code></pre></div>
<p>The graphical settings depend on the final printing destiny and on the dataset.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">7</span>,<span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, font<span class="op">=</span><span class="fl">2</span>,lwd <span class="op">=</span> <span class="fl">0.1</span>, cex <span class="op">=</span><span class="fl">1</span>, family <span class="op">=</span> <span class="st">"serif"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/plot3D/man/persp3D.html">persp3D</a></span><span class="op">(</span><span class="va">X</span>, <span class="va">Y</span>, <span class="va">z</span>, lwd <span class="op">=</span> <span class="fl">0.01</span>, image <span class="op">=</span> <span class="cn">TRUE</span>,
        xlab <span class="op">=</span> <span class="st">""</span>, ylab <span class="op">=</span> <span class="st">""</span>, zlab <span class="op">=</span> <span class="st">""</span>, family <span class="op">=</span> <span class="st">"serif"</span>,
        box <span class="op">=</span> <span class="cn">T</span>, panel.first <span class="op">=</span> <span class="va">panelfirst</span>, bty <span class="op">=</span> <span class="st">"u"</span>,
        col.panel <span class="op">=</span> <span class="st">"steelblue"</span>, col.grid <span class="op">=</span> <span class="st">"black"</span>, lwd.grid <span class="op">=</span> <span class="fl">0.5</span>,
        contour <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>col <span class="op">=</span> <span class="st">"black"</span>, side <span class="op">=</span> <span class="st">"zmin"</span>, lwd <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span>,
        ticktype <span class="op">=</span> <span class="st">"detailed"</span>, clab <span class="op">=</span> <span class="st">""</span>, expand <span class="op">=</span> <span class="fl">0.7</span>, d <span class="op">=</span> <span class="fl">2</span>,
        phi <span class="op">=</span> <span class="va">phi</span>, theta <span class="op">=</span> <span class="va">theta</span>, resfac <span class="op">=</span> <span class="fl">2.5</span>, cex.lab <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.005</span>,
        xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1.5</span><span class="op">)</span>, zlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">5</span>, <span class="fl">6</span><span class="op">)</span>, font<span class="op">=</span><span class="fl">2</span>, cex.axis <span class="op">=</span> <span class="fl">1.2</span>,
        colkey <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>length <span class="op">=</span> <span class="fl">0.7</span>, width <span class="op">=</span> <span class="fl">0.6</span>, col.ticks <span class="op">=</span> <span class="st">"black"</span>, lwd <span class="op">=</span> <span class="fl">0.01</span>,
                      font<span class="op">=</span><span class="fl">2</span>, shift <span class="op">=</span> <span class="fl">0</span>, dist <span class="op">=</span> <span class="fl">0</span>, lwd.ticks <span class="op">=</span> <span class="fl">0.01</span>,  cex.clab <span class="op">=</span> <span class="fl">1.2</span>,
                      cex.axis <span class="op">=</span> <span class="fl">1</span>, line <span class="op">=</span> <span class="op">-</span><span class="fl">0.8</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="fl">0.38</span>,<span class="op">-</span><span class="fl">0.48</span>, srt <span class="op">=</span> <span class="fl">34</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"|"</span>, <span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="va">log</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="va">FC</span><span class="op">)</span>, <span class="st">"|"</span><span class="op">)</span><span class="op">)</span>,
     xpd <span class="op">=</span> <span class="cn">TRUE</span>,font <span class="op">=</span> <span class="fl">2</span>, cex <span class="op">=</span> <span class="fl">1.2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.3</span>,<span class="op">-</span><span class="fl">0.44</span>, srt <span class="op">=</span> <span class="op">-</span><span class="fl">28</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="st">"MD"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
     xpd <span class="op">=</span> <span class="cn">TRUE</span>, font<span class="op">=</span><span class="fl">2</span>, cex <span class="op">=</span> <span class="fl">1.2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.3</span>,<span class="op">-</span><span class="fl">0.6</span>, srt <span class="op">=</span> <span class="fl">0</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Spearman' s "</span>, <span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="va">rho</span><span class="op">)</span>, <span class="st">": "</span>, <span class="op">-</span><span class="fl">0.17</span><span class="op">)</span><span class="op">)</span>,
     xpd <span class="op">=</span> <span class="cn">TRUE</span>, font <span class="op">=</span> <span class="fl">1</span>, cex <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="GenExp_Methylation_association_hyper_down-regulated_files/figure-html/3d_cop-1.png" width="700"></p>
<p>The marginal distribtuions are drawn in the <span class="math inline">\(XZ\)</span> and <span class="math inline">\(YZ\)</span> vertical planes. Consistent with the Spearman’ s <span class="math inline">\(\rho=-0.17\)</span>, for a fixed value of <span class="math inline">\(|log_2FC|\)</span>, (a line parallel to <span class="math inline">\(MD\)</span> axis) the joint probability density distribution function <span class="math inline">\(f(X \leq MD, Y \leq |log_2FC|)\)</span> decreases with the increment of <span class="math inline">\(MD\)</span>. That is, the probability to observe DEGs with values, say between 0.1 and 0.5,<br><span class="math inline">\(0.1 \leq |log_2FC| \leq 0.5\)</span>, is greater for <span class="math inline">\(0.1 \eq MD \leq 0.5\)</span> than for <span class="math inline">\(0.5 &lt; MD \leq 1\)</span>.</p>
</div>
<div id="two-dimensional-copula-density-plot-contour-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#two-dimensional-copula-density-plot-contour-plot" class="anchor"></a>Two-dimensional copula density plot, Contour plot</h3>
<p>The 2D-dimensional plot permits to see in more detail the projection of the density surface on the <span class="math inline">\(XY\)</span> plane and contour lines.</p>
<p><img src="GenExp_Methylation_association_hyper_down-regulated_files/figure-html/contour-1.png" width="700"></p>
<p>It is not difficult to see that for a fixed value of <span class="math inline">\(MD\)</span> (a line parallel to <span class="math inline">\(|log_2FC|\)</span> axis) the joint probability density distribution function <span class="math inline">\(f(X \leq MD, Y \leq |log_2FC|)\)</span> (color gradient from red to blue) decreases with the increment of <span class="math inline">\(|log_2FC|\)</span>. That is, the probability to observe <span class="math inline">\(MD\)</span> values, say between 0.1 and 0.5 (see the squared region below) <span class="math inline">\(0.1 \leq MD \leq 0.5\)</span>, is greater for <span class="math inline">\(0.1 \leq |log_2FC| \leq 0.5\)</span> than for <span class="math inline">\(0.5 &lt; |log_2FC| \leq 1\)</span>.</p>
<p>In summary, gene expression and methylation processes (2kb upstream gene TSS regions) are not independent in memory line first generation. There exist an stochastic deterministic dependence between <span class="math inline">\(|log_2FC|\)</span> and <span class="math inline">\(MD\)</span> that can be expressed in terms joint probability distribution, which leads to an inverse relationship between the conditioned expected values <span class="math inline">\(|log_2FC|\)</span> given <span class="math inline">\(MD\)</span>, and vice versa. This association is, however, weaker than the association found for gene expression and methylation at gene body.</p>
</div>
</div>
<div id="asymptotic-linear-by-linear-association-test" class="section level2">
<h2 class="hasAnchor">
<a href="#asymptotic-linear-by-linear-association-test" class="anchor"></a>Asymptotic linear-by-linear association test</h2>
<p>The above 3D and 2D density plots and results suggest that most of the genes have low <span class="math inline">\(|logFC|\)</span> and low <span class="math inline">\(MD\)</span>, and the number of genes decrease for middle levels of <span class="math inline">\(|logFC|\)</span> and <span class="math inline">\(MD\)</span>. This observation leads to the application of a linear-by-linear association test.</p>
<p>The linear-by-linear test (LLT) is used to test a linear association among variables in a contingency table with ordered categories (<span class="citation">(5)</span>). The null hypothesis for the linear-by-linear test is that there is no association among the variables in the table.</p>
<p>It is obvious from the 2D density plots that a relationship for <span class="math inline">\(MD\)</span> &lt; 0.08 is not meaningful. In particular, both variables, <span class="math inline">\(|logFC|\)</span> and <span class="math inline">\(MD\)</span>, are at the level of technical noise. The percentage of the genes at noise level are:</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fl">100</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">MD</span> <span class="op">&lt;</span> <span class="fl">0.08</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">MD</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 1.78</code></pre>
<p>Hence, the genes considered in the downstream analysis are:</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">MD</span> <span class="op">&gt;</span> <span class="fl">0.08</span><span class="op">)</span>
<span class="va">md</span>  <span class="op">&lt;-</span> <span class="va">MD</span><span class="op">[</span> <span class="va">idx</span><span class="op">]</span>
<span class="va">logfc</span>  <span class="op">&lt;-</span> <span class="va">logFC</span><span class="op">[</span> <span class="va">idx</span> <span class="op">]</span>
<span class="va">dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>md <span class="op">=</span> <span class="va">md</span> , logfc <span class="op">=</span> <span class="va">logfc</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">dt</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 13714</code></pre>
<div id="discretization-of-md-and-logfc" class="section level3">
<h3 class="hasAnchor">
<a href="#discretization-of-md-and-logfc" class="anchor"></a>Discretization of <span class="math inline">\(MD\)</span> and <span class="math inline">\(|logFC|\)</span>
</h3>
<p>To apply the LLT the values of <span class="math inline">\(|logFC|\)</span> and <span class="math inline">\(MD\)</span> must be discretized into ordinal values. A K-means discretization approach is applied with following auxiliary function</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># ------------------- Auxiliary function for discretization ------------------ #</span>
<span class="va">Kmean.discretization</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">categories</span>, <span class="va">algorithm</span> <span class="op">=</span> <span class="st">"Hartigan-Wong"</span>,
                                 <span class="va">iter.max</span> <span class="op">=</span> <span class="fl">10</span><span class="op">^</span><span class="fl">6</span>, <span class="va">column</span> <span class="op">=</span> <span class="fl">1L</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">centers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/kmeans.html">kmeans</a></span><span class="op">(</span><span class="va">x</span>, centers <span class="op">=</span> <span class="va">categories</span>, algorithm <span class="op">=</span> <span class="va">algorithm</span>,
                    iter.max <span class="op">=</span> <span class="va">iter.max</span><span class="op">)</span><span class="op">$</span><span class="va">centers</span>
  <span class="co"># order the centers</span>
  <span class="va">centers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="va">centers</span><span class="op">)</span>
  <span class="va">cl</span> <span class="op">&lt;-</span> <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/kmeans.html">kmeans</a></span><span class="op">(</span><span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html">na.omit</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, centers <span class="op">=</span> <span class="va">centers</span>,
                      algorithm <span class="op">=</span> <span class="va">algorithm</span>, iter.max <span class="op">=</span> <span class="va">iter.max</span> <span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">x</span>, cluster <span class="op">=</span> <span class="va">cl</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span> <span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div id="discretization-for-md" class="section level3">
<h3 class="hasAnchor">
<a href="#discretization-for-md" class="anchor"></a>Discretization for <span class="math inline">\(MD\)</span>
</h3>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cl</span> <span class="op">=</span> <span class="fu">Kmean.discretization</span><span class="op">(</span><span class="va">md</span>, categories <span class="op">=</span> <span class="fl">3</span>, algorithm <span class="op">=</span> <span class="st">"MacQueen"</span> <span class="op">)</span>
<span class="va">cl_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="kw">function</span><span class="op">(</span><span class="va">k</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">cl</span><span class="op">$</span><span class="va">x</span><span class="op">[</span><span class="va">cl</span><span class="op">$</span><span class="va">cluster</span> <span class="op">==</span> <span class="va">k</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">cl_summary</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"low"</span>, <span class="st">"middle"</span>, <span class="st">"high"</span><span class="op">)</span>
<span class="va">cl_summary</span></code></pre></div>
<pre><code>##                low    middle      high
## Min.    0.08010906 0.9747216  2.342326
## 1st Qu. 0.25324833 1.1864777  2.553813
## Median  0.39765342 1.4393783  2.849022
## Mean    0.44924279 1.5001513  3.182966
## 3rd Qu. 0.62903372 1.7719768  3.410265
## Max.    0.97438946 2.3393763 20.519587</code></pre>
<p>Cutpoint are now applied:</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutpoint1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">cl_summary</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="va">cutpoint2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">cl_summary</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span><span class="op">)</span>

<span class="va">dt</span><span class="op">$</span><span class="va">md_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"High"</span><span class="op">)</span>
<span class="va">dt</span><span class="op">$</span><span class="va">md_level</span><span class="op">[</span> <span class="va">dt</span><span class="op">$</span><span class="va">md</span> <span class="op">&lt;</span> <span class="va">cutpoint2</span> <span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Middle"</span>
<span class="va">dt</span><span class="op">$</span><span class="va">md_level</span><span class="op">[</span> <span class="va">dt</span><span class="op">$</span><span class="va">md</span> <span class="op">&lt;</span> <span class="va">cutpoint1</span>   <span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Low"</span>
<span class="va">dt</span><span class="op">$</span><span class="va">md_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">dt</span><span class="op">$</span><span class="va">md_level</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Low"</span>, <span class="st">"Middle"</span>, <span class="st">"High"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="discretization-for-logfc" class="section level3">
<h3 class="hasAnchor">
<a href="#discretization-for-logfc" class="anchor"></a>Discretization for <span class="math inline">\(|logFC|\)</span>
</h3>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cl</span> <span class="op">=</span> <span class="fu">Kmean.discretization</span><span class="op">(</span><span class="va">logfc</span>, categories <span class="op">=</span> <span class="fl">3</span>, algorithm <span class="op">=</span> <span class="st">"MacQueen"</span> <span class="op">)</span>
<span class="va">cl_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="kw">function</span><span class="op">(</span><span class="va">k</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">cl</span><span class="op">$</span><span class="va">x</span><span class="op">[</span><span class="va">cl</span><span class="op">$</span><span class="va">cluster</span> <span class="op">==</span> <span class="va">k</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">cl_summary</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"eLow"</span>, <span class="st">"eMiddle"</span>, <span class="st">"eHigh"</span><span class="op">)</span>
<span class="va">cl_summary</span></code></pre></div>
<pre><code>##                 eLow   eMiddle    eHigh
## Min.    2.270881e-05 0.6366017 1.835555
## 1st Qu. 1.345836e-01 0.7410084 2.070626
## Median  2.526246e-01 0.9079608 2.399485
## Mean    2.729755e-01 0.9997545 2.671355
## 3rd Qu. 4.015427e-01 1.1817728 2.905358
## Max.    6.360073e-01 1.8328136 9.407983</code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutpoint1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">cl_summary</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="va">cutpoint2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">cl_summary</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span><span class="op">)</span>

<span class="va">dt</span><span class="op">$</span><span class="va">logfc_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"eHigh"</span><span class="op">)</span>
<span class="va">dt</span><span class="op">$</span><span class="va">logfc_level</span><span class="op">[</span> <span class="va">dt</span><span class="op">$</span><span class="va">logfc</span> <span class="op">&lt;</span> <span class="va">cutpoint2</span> <span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"eMiddle"</span>
<span class="va">dt</span><span class="op">$</span><span class="va">logfc_level</span><span class="op">[</span> <span class="va">dt</span><span class="op">$</span><span class="va">logfc</span> <span class="op">&lt;</span> <span class="va">cutpoint1</span>   <span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"eLow"</span>

<span class="va">dt</span><span class="op">$</span><span class="va">logfc_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">dt</span><span class="op">$</span><span class="va">logfc_level</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"eLow"</span>, <span class="st">"eMiddle"</span>, <span class="st">"eHigh"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="the-contingency-table-and-its-graphic-visualization" class="section level3">
<h3 class="hasAnchor">
<a href="#the-contingency-table-and-its-graphic-visualization" class="anchor"></a>The contingency table and its graphic visualization</h3>
<p>In the current case, the contingency table suggests an obvious relationship between <span class="math inline">\(|logFC|\)</span> and <span class="math inline">\(MD\)</span>.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span> <span class="va">dt</span><span class="op">$</span><span class="va">logfc_level</span>, <span class="va">dt</span><span class="op">$</span><span class="va">md_level</span><span class="op">)</span> <span class="op">)</span>
<span class="va">m</span></code></pre></div>
<pre><code>##          
##            Low Middle High
##   eLow    5807   3894  765
##   eMiddle 1995    630  179
##   eHigh    354     69   21</code></pre>
<p>The <a href="https://www.rdocumentation.org/packages/graphics/versions/3.6.0/topics/spineplot">spineplot</a> of this contingency table (table(x = logfc_level, y = md_level)) help to visualize the existent relationship between <span class="math inline">\(|logFC|\)</span> and <span class="math inline">\(MD\)</span>.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span> mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">7</span>, <span class="fl">4</span>, <span class="fl">4</span><span class="op">)</span>, family <span class="op">=</span> <span class="st">"Serif"</span>, font <span class="op">=</span> <span class="fl">2</span>, cex <span class="op">=</span> <span class="fl">0.8</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.2</span> <span class="op">)</span>
<span class="va">tb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/spineplot.html">spineplot</a></span><span class="op">(</span><span class="va">m</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">2</span>,<span class="fl">1</span><span class="op">)</span><span class="op">]</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"blue"</span>, <span class="st">"dodgerblue"</span>,<span class="st">"deepskyblue"</span> <span class="op">)</span>,
                xlab <span class="op">=</span> <span class="st">""</span>, ylab <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>, text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">logFC</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, 
      line <span class="op">=</span> <span class="fl">2.5</span>, cex <span class="op">=</span> <span class="fl">1</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>, text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="st">"MD"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
      line <span class="op">=</span> <span class="fl">4</span>, cex <span class="op">=</span> <span class="fl">1</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>a <span class="op">=</span> <span class="fl">0.0</span>, b <span class="op">=</span> <span class="fl">0.72</span>, col <span class="op">=</span> <span class="st">"white"</span>, lty <span class="op">=</span> <span class="st">"dashed"</span>, lwd <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="GenExp_Methylation_association_hyper_down-regulated_files/figure-html/spineplot-1.png" width="768"></p>
<p>A dashed line was added to emphasize the direction of linear trend. From <em>“Low”</em> to <em>“high”</em> there seems to be a linear relationship between pairwise proportion of <span class="math inline">\(|logFC|\)</span> and <span class="math inline">\(MD\)</span>.</p>
<p>Conceptually, the <em>spineplot</em> is a plot of the conditional probability <span class="math inline">\(P(y|x)\)</span> against <span class="math inline">\(P(x)\)</span>. Since the joint probability distribution of these variable is described by a FGM copula, a linear relationship between <span class="math inline">\(P(y|x)\)</span> against <span class="math inline">\(P(x)\)</span> must be expected, which is a property of FGM copula <span class="citation">(6)</span>.</p>
</div>
<div id="testing-association-between-logfc-and-md" class="section level3">
<h3 class="hasAnchor">
<a href="#testing-association-between-logfc-and-md" class="anchor"></a>Testing association between <span class="math inline">\(|logFC|\)</span> and <span class="math inline">\(MD\)</span>
</h3>
<p>Here, we use the LLT test implemented in function <a href="https://www.rdocumentation.org/packages/coin/versions/1.3-0/topics/ContingencyTests">lbl_test</a> from the R package <em>coin</em>. Two test variants are available: 1) permutation test and asymptotic test.</p>
<p>The asymptotic LLT test:</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/coin/man/ContingencyTests.html">lbl_test</a></span><span class="op">(</span> <span class="va">m</span> <span class="op">)</span></code></pre></div>
<pre><code>## 
##  Asymptotic Linear-by-Linear Association Test
## 
## data:  Var2 (ordered) by Var1 (eLow &lt; eMiddle &lt; eHigh)
## Z = -14.599, p-value &lt; 2.2e-16
## alternative hypothesis: two.sided</code></pre>
<p>The permutation LLT test:</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/coin/man/ContingencyTests.html">lbl_test</a></span><span class="op">(</span> <span class="va">m</span>, distribution <span class="op">=</span> <span class="st">"approximate"</span>, B <span class="op">=</span> <span class="fl">10000</span>, alternative <span class="op">=</span> <span class="st">"less"</span> <span class="op">)</span></code></pre></div>
<pre><code>## 
##  Approximative Linear-by-Linear Association Test
## 
## data:  Var2 (ordered) by Var1 (eLow &lt; eMiddle &lt; eHigh)
## Z = -14.599, p-value &lt; 1e-04
## alternative hypothesis: less</code></pre>
<p>The null hypothesis of no association among the variables in the table is rejected. In the current case the p-value &lt; 1e-04 and, hence, we do not have enough evidence to support the null hypothesis.</p>
</div>
</div>
<div id="generalized-linear-model-glm" class="section level2">
<h2 class="hasAnchor">
<a href="#generalized-linear-model-glm" class="anchor"></a>Generalized linear model (GLM)</h2>
<p>The variable <span class="math inline">\(MD\)</span> was split into three levels, “0”: “Low” (<span class="math inline">\(0 &lt; MD \leq 0.5\)</span>), “1”: “Middle” (<span class="math inline">\(0.5 &lt; MD \leq 1\)</span>), and “2”: “High” ($1 &lt; MD $)</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meth_cut</span> <span class="op">&lt;-</span> <span class="fl">1</span>

<span class="va">xy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cbind.html">cbind</a></span><span class="op">(</span><span class="va">MD</span>, <span class="va">logFC</span><span class="op">)</span>
<span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">logFC</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="va">xy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">xy</span><span class="op">[</span><span class="va">idx</span>, <span class="op">]</span><span class="op">)</span>
<span class="va">xy</span><span class="op">$</span><span class="va">md</span> <span class="op">&lt;-</span> <span class="fl">0</span>
<span class="va">xy</span><span class="op">$</span><span class="va">md</span><span class="op">[</span><span class="va">xy</span><span class="op">$</span><span class="va">MD</span> <span class="op">&gt;</span> <span class="va">meth_cut</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span></code></pre></div>
<p>The glm support the decrease of <span class="math inline">\(-logFC\)</span> with th increment of <span class="math inline">\(MD\)</span>. The model coefficients are highly statistically significant.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="va">logFC</span> <span class="op">~</span> <span class="va">md</span>, data<span class="op">=</span><span class="va">xy</span>, family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/family.html">Gamma</a></span><span class="op">(</span>link <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span> <span class="va">gm</span> <span class="op">)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = logFC ~ md, family = Gamma(link = "identity"), 
##     data = xy)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -0.6916  -0.4656  -0.2364   0.1463   3.3044  
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)  1.09233    0.01227  89.030   &lt;2e-16 ***
## md          -0.17546    0.02040  -8.602   &lt;2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## (Dispersion parameter for Gamma family taken to be 0.4203748)
## 
##     Null deviance: 1283.8  on 4662  degrees of freedom
## Residual deviance: 1255.4  on 4661  degrees of freedom
## AIC: 6431
## 
## Number of Fisher Scoring iterations: 3</code></pre>
<p>The ANOVA test confirm the negative trend given by the coefficient of MD (highly statistically significant).</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">gm</span>, test <span class="op">=</span> <span class="st">"LRT"</span><span class="op">)</span></code></pre></div>
<pre><code>## Analysis of Deviance Table
## 
## Model: Gamma, link: identity
## 
## Response: logFC
## 
## Terms added sequentially (first to last)
## 
## 
##      Df Deviance Resid. Df Resid. Dev  Pr(&gt;Chi)    
## NULL                  4662     1283.8              
## md    1   28.416      4661     1255.4 &lt; 2.2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
<div id="boxplot" class="section level3">
<h3 class="hasAnchor">
<a href="#boxplot" class="anchor"></a>Boxplot</h3>
<p>The boxplot suggest a clear trend of <span class="math inline">\(-logFC\)</span> decreasing with increment of MD levels. Notice that the <em>notchs</em> from the boxes do not overlap, which implies that the differences between medians from three groups (“Low,” “Middle,” and “High”) are statistically significant. So, the GLM results can be visualized.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>bty <span class="op">=</span> <span class="st">"n"</span>, family <span class="op">=</span> <span class="st">"serif"</span>, font <span class="op">=</span> <span class="fl">2</span>, cex <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.1</span>, mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">2</span>, <span class="fl">1.1</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span><span class="op">(</span><span class="va">logFC</span> <span class="op">~</span> <span class="va">md</span>, data <span class="op">=</span> <span class="va">xy</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.4</span>, <span class="fl">2.5</span><span class="op">)</span>, xaxt <span class="op">=</span><span class="st">"n"</span>,  yaxt <span class="op">=</span> <span class="st">"n"</span>, ann <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/rect.html">rect</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.4</span>, <span class="fl">3.5</span>, <span class="fl">5</span>, col <span class="op">=</span> <span class="st">"grey94"</span>, lty <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/grid.html">grid</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span>, col <span class="op">=</span> <span class="st">"white"</span>, lwd <span class="op">=</span> <span class="fl">2</span>, lty <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span><span class="op">(</span><span class="va">logFC</span> <span class="op">~</span> <span class="va">md</span>, data <span class="op">=</span> <span class="va">xy</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.4</span>, <span class="fl">2.5</span><span class="op">)</span>, notch <span class="op">=</span> <span class="cn">TRUE</span>, xaxt <span class="op">=</span><span class="st">"n"</span>,
        yaxt <span class="op">=</span> <span class="st">"n"</span>, ann <span class="op">=</span> <span class="cn">FALSE</span>, pch <span class="op">=</span> <span class="fl">20</span>, medcol <span class="op">=</span> <span class="st">"darkblue"</span>, whisklwd <span class="op">=</span> <span class="fl">1</span>, staplelwd <span class="op">=</span> <span class="fl">1</span>,
        col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"deepskyblue"</span>, <span class="st">"dodgerblue"</span><span class="op">)</span>, outcol <span class="op">=</span> <span class="st">"red"</span>, add  <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">1</span>, padj <span class="op">=</span> <span class="fl">0.1</span>, at <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, lwd <span class="op">=</span> <span class="fl">0.5</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.02</span>, line <span class="op">=</span> <span class="op">-</span><span class="fl">0.6</span>, cex.axis <span class="op">=</span> <span class="fl">1.3</span>, 
     labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Low"</span>, <span class="st">"High"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">2</span>, hadj <span class="op">=</span> <span class="fl">1</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.5</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.02</span>, line <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, cex.axis <span class="op">=</span> <span class="fl">1.4</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>, text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="st">"MD"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
      line <span class="op">=</span> <span class="fl">2.2</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>, text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="op">-</span><span class="va">logFC</span><span class="op">)</span><span class="op">)</span>, line <span class="op">=</span> <span class="fl">2.1</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fl">1.5</span>, y <span class="op">=</span> <span class="fl">2.2</span>, <span class="st">"***"</span>, pos <span class="op">=</span> <span class="fl">3</span>, cex <span class="op">=</span> <span class="fl">1.2</span> <span class="op">)</span></code></pre></div>
<p><img src="GenExp_Methylation_association_hyper_down-regulated_files/figure-html/boxplot_2-1.png" width="864"></p>
</div>
</div>
</div>
<div id="analysis-for-up-regulated-genes" class="section level1">
<h1 class="hasAnchor">
<a href="#analysis-for-up-regulated-genes" class="anchor"></a>Analysis for up-regulated genes</h1>
<div id="data-set-ii" class="section level2">
<h2 class="hasAnchor">
<a href="#data-set-ii" class="anchor"></a>Data set (II)</h2>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">.deg_mm1_wt1</span> <span class="op">&lt;-</span> <span class="va">deg_mm1_wt1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">dmp_den_wt1_mm1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">deg_mm1_wt1</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span>

<span class="va">MD</span> <span class="op">&lt;-</span> <span class="va">dmp_den_wt1_mm1</span><span class="op">$</span><span class="va">dmp_den</span> <span class="co"># Methylation (|group mean density differences|)</span>
<span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">.deg_mm1_wt1</span><span class="op">$</span><span class="va">table</span><span class="op">$</span><span class="va">logFC</span> <span class="op">&gt;</span> <span class="fl">0</span> <span class="op">)</span> <span class="co"># Up-regulated</span>
<span class="va">logFC</span> <span class="op">&lt;-</span> <span class="va">.deg_mm1_wt1</span><span class="op">$</span><span class="va">table</span><span class="op">$</span><span class="va">logFC</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span> <span class="co"># Gene Expression (|logFC|)</span>
<span class="va">MD</span> <span class="op">&lt;-</span> <span class="va">MD</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span>
<span class="fu">DataFrame</span><span class="op">(</span><span class="va">MD</span>,<span class="va">logFC</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 6060 rows and 2 columns
##             MD     logFC
##      &lt;numeric&gt; &lt;numeric&gt;
## 1     6.571474 0.0359792
## 2     1.199583 0.5287217
## 3     0.762753 0.1245334
## 4     0.274631 0.0142820
## 5     0.623658 0.4497949
## ...        ...       ...
## 6056  0.223080  0.317756
## 6057  0.634709  0.394906
## 6058  0.296653  0.943018
## 6059  0.121346  1.680553
## 6060  0.151446  1.921001</code></pre>
<p>There are 6060 genes with <span class="math inline">\(MD &gt; 0\)</span> and <span class="math inline">\(logFC &gt; 0\)</span> to be included in the downstream analysis.</p>
</div>
<div id="distribution-of-the-divergence-of-methylation-level-density-at-gene-body--mm1-vs-wt1-ii" class="section level2">
<h2 class="hasAnchor">
<a href="#distribution-of-the-divergence-of-methylation-level-density-at-gene-body--mm1-vs-wt1-ii" class="anchor"></a>Distribution of the divergence of methylation level density at gene-body. MM1 vs WT1 (II)</h2>
<p>The best fitted model for <span class="math inline">\(MD\)</span>, from comparison MM1 <em>vs</em> WT1, was a mixture of Log-Normal and Weibull distributions.</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">MD</span><span class="op">[</span> <span class="va">MD</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span>
<span class="va">meth_FIT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/fitMixDist.html">fitMixDist</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">x</span>, args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>norm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>mean <span class="op">=</span> <span class="cn">NA</span>, sd <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
                                     weibull <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>shape <span class="op">=</span> <span class="cn">NA</span>, scale <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span><span class="op">)</span>,
                  dens <span class="op">=</span> <span class="cn">TRUE</span>, npoints <span class="op">=</span> <span class="fl">900</span><span class="op">)</span></code></pre></div>
<pre><code>## fitting ...
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |======================================================================| 100%
## *** Performing nonlinear regression model  crossvalidation...</code></pre>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">meth_FIT</span><span class="op">$</span><span class="va">fit</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Parameters:
##       Estimate Std. Error t value Pr(&gt;|t|)    
## mean  0.285902   0.002125  134.57   &lt;2e-16 ***
## sd    0.163987   0.001777   92.27   &lt;2e-16 ***
## shape 2.094322   0.029282   71.52   &lt;2e-16 ***
## scale 1.360858   0.011847  114.86   &lt;2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 0.04602 on 1044 degrees of freedom
## Number of iterations to termination: 12 
## Reason for termination: Relative error in the sum of squares is at most `ftol'.</code></pre>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\n Mixture proportions (phi):"</span>, <span class="va">meth_FIT</span><span class="op">$</span><span class="va">phi</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##  Mixture proportions (phi): 0.4203008 0.5796992</code></pre>
<p>The graphic for histogram and density distribution curves:</p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">4</span>, <span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span>, font.lab<span class="op">=</span><span class="fl">2</span>, font<span class="op">=</span><span class="fl">2</span>,font.axis<span class="op">=</span><span class="fl">2</span>,family<span class="op">=</span><span class="st">"serif"</span> , lwd <span class="op">=</span> <span class="fl">1.2</span>, cex <span class="op">=</span> <span class="fl">1.4</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">MD</span>, <span class="fl">90</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">1</span>, <span class="co">#ylim = c(0, 0.021), cex = 1,</span>
     panel.first<span class="op">=</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, pch<span class="op">=</span><span class="fl">16</span>, cex<span class="op">=</span><span class="fl">1e6</span>, col<span class="op">=</span><span class="st">"grey95"</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/graphics/grid.html">grid</a></span><span class="op">(</span>col<span class="op">=</span><span class="st">"white"</span>, lty <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">}</span>, main <span class="op">=</span> <span class="va">text</span>,
     cex.main <span class="op">=</span> <span class="fl">0.9</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span><span class="op">)</span>, family <span class="op">=</span> <span class="st">"serif"</span>,
     col <span class="op">=</span> <span class="st">"cyan1"</span>, border <span class="op">=</span> <span class="st">"deepskyblue"</span>, xaxt <span class="op">=</span><span class="st">"n"</span>, yaxt <span class="op">=</span> <span class="st">"n"</span>, ann <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">1</span>, padj <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.8</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.02</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">2</span>, hadj <span class="op">=</span> <span class="fl">0.7</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.8</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.02</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>, text <span class="op">=</span> <span class="st">"Divergence of methylation level density at gene-body"</span>,
      line <span class="op">=</span> <span class="fl">1.6</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>, text <span class="op">=</span> <span class="st">"Density distribution"</span>, line <span class="op">=</span> <span class="fl">2.1</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span><span class="st">"(Normal &amp; Weibull Mixture Dists.)"</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">10</span>, by <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span><span class="op">(</span><span class="va">x1</span>, <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/mixtdistr.html">dmixtdistr</a></span><span class="op">(</span><span class="va">x1</span>, phi <span class="op">=</span> <span class="va">meth_FIT</span><span class="op">$</span><span class="va">phi</span>, arg <span class="op">=</span> <span class="va">meth_FIT</span><span class="op">$</span><span class="va">args</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span></code></pre></div>
<p><img src="GenExp_Methylation_association_hyper_down-regulated_files/figure-html/graphic_meth_2-1.png" width="768"></p>
</div>
<div id="distribution-of-the-logfc--mm1-vs-wt1-ii" class="section level2">
<h2 class="hasAnchor">
<a href="#distribution-of-the-logfc--mm1-vs-wt1-ii" class="anchor"></a>Distribution of the <span class="math inline">\(|logFC|\)</span>. MM1 vs WT1 (II)</h2>
<p>The best fitted model for the <span class="math inline">\(|logFC|\)</span>, from comparison MM1 <em>vs</em> WT1, was a mixture of Gamma and Weibull distributions. Function <a href="https://genomaths.github.io/usefr_manual/fitMixDist.html">fitMixDist</a> calls the mixture function models set up in function <a href="https://genomaths.github.io/usefr_manual/mixtdistr.html">mixtdistr</a></p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gexp_FIT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/fitMixDist.html">fitMixDist</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">logFC</span>, args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>weibull <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>shape <span class="op">=</span> <span class="cn">NA</span>, scale <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span>,
                                              norm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>mean <span class="op">=</span> <span class="cn">NA</span>, sd <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span><span class="op">)</span>,
                       dens <span class="op">=</span> <span class="cn">TRUE</span>, npoints <span class="op">=</span> <span class="fl">900</span><span class="op">)</span></code></pre></div>
<pre><code>## fitting ...
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |======================================================================| 100%
## *** Performing nonlinear regression model  crossvalidation...</code></pre>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">gexp_FIT</span><span class="op">$</span><span class="va">fit</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Parameters:
##       Estimate Std. Error t value Pr(&gt;|t|)    
## shape 1.052405   0.005327  197.56   &lt;2e-16 ***
## scale 0.267858   0.002407  111.31   &lt;2e-16 ***
## mean  0.578792   0.039787   14.55   &lt;2e-16 ***
## sd    0.616634   0.024157   25.53   &lt;2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 0.04727 on 791 degrees of freedom
## Number of iterations to termination: 16 
## Reason for termination: Relative error in the sum of squares is at most `ftol'.</code></pre>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\n Mixture proportions (phi):"</span>, <span class="va">gexp_FIT</span><span class="op">$</span><span class="va">phi</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##  Mixture proportions (phi): 0.7545028 0.2454972</code></pre>
<p>The graphic for histogram and density distribution curves:</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span>, font.lab<span class="op">=</span><span class="fl">2</span>, font<span class="op">=</span><span class="fl">2</span>,font.axis<span class="op">=</span><span class="fl">2</span>,family<span class="op">=</span><span class="st">"serif"</span>, lwd <span class="op">=</span> <span class="fl">1.2</span>, cex <span class="op">=</span> <span class="fl">1.4</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">logFC</span>, <span class="fl">120</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">1</span>, <span class="co">#ylim = c(0, 0.021), cex = 1,</span>
     panel.first<span class="op">=</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, pch<span class="op">=</span><span class="fl">16</span>, cex<span class="op">=</span><span class="fl">1e6</span>, col<span class="op">=</span><span class="st">"grey95"</span><span class="op">)</span>
       <span class="fu"><a href="https://rdrr.io/r/graphics/grid.html">grid</a></span><span class="op">(</span>col<span class="op">=</span><span class="st">"white"</span>, lty <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">}</span>, main <span class="op">=</span> <span class="va">text</span>, 
     cex.main <span class="op">=</span> <span class="fl">0.9</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">4</span><span class="op">)</span>, family <span class="op">=</span> <span class="st">"serif"</span>,
     col <span class="op">=</span> <span class="st">"cyan1"</span>, border <span class="op">=</span> <span class="st">"deepskyblue"</span>, xaxt <span class="op">=</span><span class="st">"n"</span>, yaxt <span class="op">=</span> <span class="st">"n"</span>, ann <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">1</span>, padj <span class="op">=</span> <span class="op">-</span><span class="fl">1.5</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.5</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.02</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">2</span>, hadj <span class="op">=</span> <span class="fl">0.4</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.5</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.02</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>, text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">logFC</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
      line <span class="op">=</span> <span class="fl">1.3</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>, text <span class="op">=</span> <span class="st">"Density distribution"</span>, line <span class="op">=</span> <span class="fl">1.6</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span><span class="st">"(Weibull &amp; Normal Mixture Dists.)"</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">10</span>, by <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span><span class="op">(</span><span class="va">x1</span>, <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/mixtdistr.html">dmixtdistr</a></span><span class="op">(</span><span class="va">x1</span>, phi <span class="op">=</span> <span class="va">gexp_FIT</span><span class="op">$</span><span class="va">phi</span>, arg <span class="op">=</span> <span class="va">gexp_FIT</span><span class="op">$</span><span class="va">args</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"blue"</span><span class="op">)</span></code></pre></div>
<p><img src="GenExp_Methylation_association_hyper_down-regulated_files/figure-html/logFC_grh_2-1.png" width="768"></p>
</div>
<div id="joint-distribution-of-md-and-logfc-expressed-in-terms-of-copula-distribution-ii" class="section level2">
<h2 class="hasAnchor">
<a href="#joint-distribution-of-md-and-logfc-expressed-in-terms-of-copula-distribution-ii" class="anchor"></a>Joint distribution of <span class="math inline">\(MD\)</span> and <span class="math inline">\(|logFC|\)</span> expressed in terms of Copula distribution (II)</h2>
<div id="fitting-farliegumbelmorgenstern-copulas-with-md-and-logfc-marginal-cdfs-ii" class="section level3">
<h3 class="hasAnchor">
<a href="#fitting-farliegumbelmorgenstern-copulas-with-md-and-logfc-marginal-cdfs-ii" class="anchor"></a>Fitting Farlie–Gumbel–Morgenstern copulas with <span class="math inline">\(MD\)</span> and <span class="math inline">\(|logFC|\)</span> marginal CDFs (II)</h3>
<p>The first step requires for provide the parameters values, previously estimated, of the marginal distributions, which were given throughout the function <a href="https://genomaths.github.io/usefr_manual/mixtdistr.html">mixtdistr</a>. The specific parameter values are retrieved from the previous nonlinear fit obtained for the marginal CDFs:</p>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">margins</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"mixtdistr"</span>, <span class="st">"mixtdistr"</span><span class="op">)</span>
<span class="va">parMargins</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>phi <span class="op">=</span> <span class="va">meth_FIT</span><span class="op">$</span><span class="va">phi</span>, arg <span class="op">=</span> <span class="va">meth_FIT</span><span class="op">$</span><span class="va">args</span><span class="op">)</span>,
  <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>phi <span class="op">=</span> <span class="va">gexp_FIT</span><span class="op">$</span><span class="va">phi</span>, arg <span class="op">=</span> <span class="va">gexp_FIT</span><span class="op">$</span><span class="va">args</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>The parameter estimation and goodness-of-fit (GOF) test take advantage of the tools available in the R package <a href="https://www.rdocumentation.org/packages/copula/versions/0.999-19.1">copula</a>. Both, parameter estimation and GOF, can be accomplished with function <a href="https://genomaths.github.io/usefr_manual/bicopulaGOF.html">bicopulaGOF</a> from the R package <a href="https://github.com/genomaths/usefr">usefr</a>.</p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fgm_cop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/bicopulaGOF.html">bicopulaGOF</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">MD</span>, y <span class="op">=</span> <span class="va">logFC</span>, copula <span class="op">=</span> <span class="st">"fgmCopula"</span>, sample.size <span class="op">=</span> <span class="fl">5999</span>, 
                       margins <span class="op">=</span> <span class="va">margins</span>, paramMargins <span class="op">=</span> <span class="va">parMargins</span>, nboots <span class="op">=</span> <span class="fl">10</span>, 
                       approach <span class="op">=</span> <span class="st">"chisq"</span>, breaks <span class="op">=</span> <span class="fl">30</span>, num.cores <span class="op">=</span> <span class="fl">6L</span>, seed <span class="op">=</span> <span class="fl">123</span>,
                       verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>The null hypothesis that the data follows FGM copula distribution is not rejected.</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fgm_cop</span><span class="op">$</span><span class="va">gof</span></code></pre></div>
<pre><code>##   Chisq.stat   mc_p.value  sample.size    num.sampl 
##  924.4173554    0.1818182 5999.0000000   10.0000000</code></pre>
<p>All the information needed about the fitted FGM copula is found in the object <em>fgm_cop$copula</em>, which is a <a href="https://www.rdocumentation.org/packages/copula/versions/0.999-19.1/topics/mvdc-class">mvdc</a> object from <a href="https://www.rdocumentation.org/packages/copula/versions/0.999-19.1">copula</a> package with several slots. In particular, the specific information about FGM copula is found in:</p>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fgm_cop</span><span class="op">$</span><span class="va">copula</span><span class="op">@</span><span class="va">copula</span></code></pre></div>
<pre><code>## Farlie-Gumbel-Morgenstern (FGM) copula, dim. d = 2
##  param.: -0.492 
## Dimension:  2 
## Parameters:
##   alpha1.2   = -0.491595</code></pre>
</div>
<div id="p-p-plot--graphical-diagnostics-of-gof-ii" class="section level3">
<h3 class="hasAnchor">
<a href="#p-p-plot--graphical-diagnostics-of-gof-ii" class="anchor"></a>P-P plot. Graphical Diagnostics of GOF (II)</h3>
<p>The P-P plot is build with function <a href="https://genomaths.github.io/usefr_manual/ppCplot.html">ppCplot</a> from the R package <a href="https://github.com/genomaths/usefr">usefr</a>. If the GOF is good enough, then it is expected that points in the graphics are closed to the diagonal (red) straight line. Details on how the computation is performed are provided in the help for function. <a href="https://genomaths.github.io/usefr_manual/ppCplot.html">ppCplot</a>.</p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/ppCplot.html">ppCplot</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">MD</span>, Y <span class="op">=</span> <span class="va">logFC</span>, copula <span class="op">=</span> <span class="va">fgm_cop</span><span class="op">$</span><span class="va">copula</span>, npoints <span class="op">=</span> <span class="fl">500</span>, col <span class="op">=</span> <span class="st">"blue"</span>, cex <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></code></pre></div>
<p><img src="GenExp_Methylation_association_hyper_down-regulated_files/figure-html/ppPlot_2-1.png" width="768"></p>
<p>Information on the data and copula used to yield the graphic are stored in the output variable <em>“g”</em>. For example, the empirical and theoretical probabilities used to built the graphic can be retrieved from the data frame: <em>“g$data”</em>:</p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">DataFrame</span><span class="op">(</span><span class="va">g</span><span class="op">$</span><span class="va">data</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 500 rows and 2 columns
##         emprob     thprob
##      &lt;numeric&gt;  &lt;numeric&gt;
## 1    0.3437294   0.344225
## 2    0.1165017   0.113694
## 3    0.3369637   0.337510
## 4    0.0292079   0.026738
## 5    0.6600660   0.662105
## ...        ...        ...
## 496 0.34570957 0.35502986
## 497 0.09917492 0.10102626
## 498 0.00907591 0.00668829
## 499 0.16105611 0.16210131
## 500 0.17854785 0.18345803</code></pre>
</div>
<div id="three-dimentional-copula-density-plot-ii" class="section level3">
<h3 class="hasAnchor">
<a href="#three-dimentional-copula-density-plot-ii" class="anchor"></a>Three-dimentional copula density plot (II)</h3>
<p>The domain region <span class="math inline">\((X, Y)\)</span> is defined and function <a href="https://www.rdocumentation.org/packages/copula/versions/0.999-19.1/topics/Mvdc">dMvdc</a> is used to compute the density <span class="math inline">\(z\)</span> in the grid <span class="math inline">\([X,Y] \times [X,Y]\)</span> using the previous FMG fitted copula.</p>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">X</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1.5</span>, length <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>
<span class="va">Y</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1.8</span>, length <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>
<span class="va">sp_rho</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/copula/man/assocMeasures.html">rho</a></span><span class="op">(</span><span class="va">fgm_cop</span><span class="op">$</span><span class="va">copula</span><span class="op">@</span><span class="va">copula</span><span class="op">)</span>,<span class="fl">3</span><span class="op">)</span> <span class="co"># Spearman' s rho: -0.171</span>

<span class="va">z</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/outer.html">outer</a></span><span class="op">(</span><span class="va">X</span>, <span class="va">Y</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/copula/man/Mvdc.html">dMvdc</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cbind.html">cbind</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span>,<span class="va">fgm_cop</span><span class="op">$</span><span class="va">copula</span><span class="op">)</span><span class="op">)</span> <span class="op">*</span> <span class="fl">4</span>
<span class="va">phi</span> <span class="op">=</span> <span class="fl">15</span>; <span class="va">theta</span> <span class="op">=</span> <span class="fl">45</span>
<span class="va">sp_rho</span></code></pre></div>
<pre><code>## [1] -0.164</code></pre>
<p>An auxiliary function to draw the panels</p>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">panelfirst</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">pmat</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">7</span>,<span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, font<span class="op">=</span><span class="fl">2</span>,lwd <span class="op">=</span> <span class="fl">0.01</span>, cex <span class="op">=</span><span class="fl">1</span>, family <span class="op">=</span> <span class="st">"serif"</span><span class="op">)</span>
  <span class="va">YZ</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/plot3D/man/trans3D.html">trans3D</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span><span class="op">)</span>, <span class="va">Y</span>, <span class="va">z</span><span class="op">[</span><span class="fl">18</span>,<span class="op">]</span>, pmat <span class="op">=</span> <span class="va">pmat</span><span class="op">)</span>

  <span class="fu"><a href="https://rdrr.io/pkg/plot3D/man/scatter.html">scatter2D</a></span><span class="op">(</span><span class="va">YZ</span><span class="op">$</span><span class="va">x</span>, <span class="va">YZ</span><span class="op">$</span><span class="va">y</span>, colvar <span class="op">=</span> <span class="va">z</span><span class="op">[</span><span class="fl">18</span>,<span class="op">]</span>, lwd <span class="op">=</span> <span class="fl">1</span>, 
            type <span class="op">=</span> <span class="st">"l"</span>, add <span class="op">=</span> <span class="cn">TRUE</span>, colkey <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

  <span class="va">XZ</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/plot3D/man/trans3D.html">trans3D</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">X</span>, y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">Y</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span><span class="op">)</span>,
                z <span class="op">=</span> <span class="va">z</span><span class="op">[</span>, <span class="fl">12</span><span class="op">]</span>, pmat <span class="op">=</span> <span class="va">pmat</span><span class="op">)</span>

  <span class="fu"><a href="https://rdrr.io/pkg/plot3D/man/scatter.html">scatter2D</a></span><span class="op">(</span><span class="va">XZ</span><span class="op">$</span><span class="va">x</span>, <span class="va">XZ</span><span class="op">$</span><span class="va">y</span>, colvar <span class="op">=</span> <span class="va">z</span><span class="op">[</span>, <span class="fl">12</span><span class="op">]</span>, lwd <span class="op">=</span> <span class="fl">1</span>, 
            type <span class="op">=</span> <span class="st">"l"</span>, add <span class="op">=</span> <span class="cn">TRUE</span>, colkey <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
  
<span class="op">}</span></code></pre></div>
<p>The graphical settings depend on the final printing destiny and on the dataset.</p>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">7</span>,<span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, font<span class="op">=</span><span class="fl">2</span>,lwd <span class="op">=</span> <span class="fl">0.1</span>, cex <span class="op">=</span><span class="fl">1</span>, family <span class="op">=</span> <span class="st">"serif"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/plot3D/man/persp3D.html">persp3D</a></span><span class="op">(</span><span class="va">X</span>, <span class="va">Y</span>, <span class="va">z</span>, lwd <span class="op">=</span> <span class="fl">0.01</span>, image <span class="op">=</span> <span class="cn">TRUE</span>,
        xlab <span class="op">=</span> <span class="st">""</span>, ylab <span class="op">=</span> <span class="st">""</span>, zlab <span class="op">=</span> <span class="st">""</span>, family <span class="op">=</span> <span class="st">"serif"</span>,
        box <span class="op">=</span> <span class="cn">T</span>, panel.first <span class="op">=</span> <span class="va">panelfirst</span>, bty <span class="op">=</span> <span class="st">"u"</span>,
        col.panel <span class="op">=</span> <span class="st">"steelblue"</span>, col.grid <span class="op">=</span> <span class="st">"black"</span>, lwd.grid <span class="op">=</span> <span class="fl">0.5</span>,
        contour <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>col <span class="op">=</span> <span class="st">"black"</span>, side <span class="op">=</span> <span class="st">"zmin"</span>, lwd <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span>,
        ticktype <span class="op">=</span> <span class="st">"detailed"</span>, clab <span class="op">=</span> <span class="st">""</span>, expand <span class="op">=</span> <span class="fl">0.7</span>, d <span class="op">=</span> <span class="fl">2</span>,
        phi <span class="op">=</span> <span class="va">phi</span>, theta <span class="op">=</span> <span class="va">theta</span>, resfac <span class="op">=</span> <span class="fl">2.5</span>, cex.lab <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.005</span>,
        xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1.5</span><span class="op">)</span>, zlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">5</span>, <span class="fl">9</span><span class="op">)</span>, font<span class="op">=</span><span class="fl">2</span>, cex.axis <span class="op">=</span> <span class="fl">1.2</span>,
        colkey <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>length <span class="op">=</span> <span class="fl">0.5</span>, width <span class="op">=</span> <span class="fl">0.6</span>, col.ticks <span class="op">=</span> <span class="st">"black"</span>, lwd <span class="op">=</span> <span class="fl">0.01</span>,
                      font<span class="op">=</span><span class="fl">2</span>, shift <span class="op">=</span> <span class="fl">0</span>, dist <span class="op">=</span> <span class="fl">0</span>, lwd.ticks <span class="op">=</span> <span class="fl">0.01</span>,  cex.clab <span class="op">=</span> <span class="fl">1.2</span>,
                      cex.axis <span class="op">=</span> <span class="fl">1</span>, line <span class="op">=</span> <span class="op">-</span><span class="fl">0.8</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="fl">0.38</span>,<span class="op">-</span><span class="fl">0.4</span>, srt <span class="op">=</span> <span class="fl">38</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"|"</span>, <span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="va">log</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="va">FC</span><span class="op">)</span>, <span class="st">"|"</span><span class="op">)</span><span class="op">)</span>,
     xpd <span class="op">=</span> <span class="cn">TRUE</span>,font <span class="op">=</span> <span class="fl">2</span>, cex <span class="op">=</span> <span class="fl">1.2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.41</span>,<span class="op">-</span><span class="fl">0.39</span>, srt <span class="op">=</span> <span class="op">-</span><span class="fl">20</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="st">"MD"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
     xpd <span class="op">=</span> <span class="cn">TRUE</span>, font<span class="op">=</span><span class="fl">2</span>, cex <span class="op">=</span> <span class="fl">1.2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.36</span>,<span class="op">-</span><span class="fl">0.5</span>, srt <span class="op">=</span> <span class="fl">0</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Spearman' s "</span>, <span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="va">rho</span><span class="op">)</span>, <span class="st">": "</span>, <span class="op">-</span><span class="fl">0.164</span><span class="op">)</span><span class="op">)</span>,
     xpd <span class="op">=</span> <span class="cn">TRUE</span>, font <span class="op">=</span> <span class="fl">1</span>, cex <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="GenExp_Methylation_association_hyper_down-regulated_files/figure-html/3d_cop_2-1.png" width="768"></p>
<p>The marginal distributions are drawn in the <span class="math inline">\(XZ\)</span> and <span class="math inline">\(YZ\)</span> vertical planes. Consistent with the Spearman’ s <span class="math inline">\(\rho=-0.164\)</span>, for a fixed value of <span class="math inline">\(|log_2FC|\)</span>, (a line parallel to <span class="math inline">\(MD\)</span> axis) the joint probability density distribution function <span class="math inline">\(f(X \leq MD, Y \leq log_2FC)\)</span> decreasses with the increment of <span class="math inline">\(MD\)</span>. That is, the probability to observe DEGs with values, say between 0.1 and 0.5,<br><span class="math inline">\(0.1 \leq log_2FC \leq 0.5\)</span>, is greater for <span class="math inline">\(0.1 \leq MD \leq 0.5\)</span> than for <span class="math inline">\(0.5 &lt; MD \leq 1\)</span>.</p>
</div>
<div id="two-dimensional-copula-density-plot-contour-plot-ii" class="section level3">
<h3 class="hasAnchor">
<a href="#two-dimensional-copula-density-plot-contour-plot-ii" class="anchor"></a>Two-dimensional copula density plot, Contour plot (II)</h3>
<p>The 2D-dimensional plot permits to see in more detail the projection of the density surface on the <span class="math inline">\(XY\)</span> plane and contour lines.</p>
<p>It is not difficult to see that for a fixed value of <span class="math inline">\(MD\)</span> (a line parallel to <span class="math inline">\(|log_2FC|\)</span> axis) the joint probability density distribution function <span class="math inline">\(f(X \leq MD, Y \leq |log_2FC|)\)</span> (color gradient from red to blue) decreasses with the increment of <span class="math inline">\(|log_2FC|\)</span>. That is, the probability to observe <span class="math inline">\(MD\)</span> values, say between 0.1 and 0.5 (see the squared region below) <span class="math inline">\(0.1 \leq MD \leq 0.5\)</span>, is greater for <span class="math inline">\(0.1 \leq |log_2FC| \leq 0.5\)</span> than for <span class="math inline">\(0.5 &lt; |log_2FC| \leq 1\)</span>.</p>
<p><img src="GenExp_Methylation_association_hyper_down-regulated_files/figure-html/contour2_2-1.png" width="768"></p>
<p>In summary, gene expression and methylation processes are not independent in memory line first generation. There exist an stochastic deterministic dependence between <span class="math inline">\(|log_2FC|\)</span> and <span class="math inline">\(MD\)</span> that can be expressed in terms joint probability distribution, which leads to an inverse relationship between the conditioned expected values <span class="math inline">\(|log_2FC|\)</span> given <span class="math inline">\(MD\)</span>, and vice versa.</p>
</div>
</div>
<div id="asymptotic-linear-by-linear-association-test-ii" class="section level2">
<h2 class="hasAnchor">
<a href="#asymptotic-linear-by-linear-association-test-ii" class="anchor"></a>Asymptotic linear-by-linear association test (II)</h2>
<p>As in the previous case (above), tt is obvious from the 2D density plots that for <span class="math inline">\(MD\)</span> &lt; 0.08 the any relationship is not meaningful. In particular, both variables, <span class="math inline">\(logFC\)</span> and <span class="math inline">\(MD\)</span>, are at the level of technical noise. The percentage of the genes at noise level are:</p>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fl">100</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">MD</span> <span class="op">&lt;</span> <span class="fl">0.08</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">MD</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 1.42</code></pre>
<p>The genes for the downstream analysis are:</p>
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">MD</span> <span class="op">&gt;</span> <span class="fl">0.08</span><span class="op">)</span>
<span class="va">md</span>  <span class="op">&lt;-</span> <span class="va">MD</span><span class="op">[</span> <span class="va">idx</span><span class="op">]</span>
<span class="va">logfc</span>  <span class="op">&lt;-</span> <span class="va">logFC</span><span class="op">[</span> <span class="va">idx</span> <span class="op">]</span>
<span class="va">dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>md <span class="op">=</span> <span class="va">md</span> , logfc <span class="op">=</span> <span class="va">logfc</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">dt</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 5974</code></pre>
<div id="discretization-of-md-and-logfc-ii" class="section level3">
<h3 class="hasAnchor">
<a href="#discretization-of-md-and-logfc-ii" class="anchor"></a>Discretization of <span class="math inline">\(MD\)</span> and <span class="math inline">\(logFC\)</span> (II)</h3>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># ------------------- Auxiliary function for discretization ------------------ #</span>
<span class="va">Kmean.discretization</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">categories</span>, <span class="va">algorithm</span> <span class="op">=</span> <span class="st">"Hartigan-Wong"</span>,
                                 <span class="va">iter.max</span> <span class="op">=</span> <span class="fl">10</span><span class="op">^</span><span class="fl">6</span>, <span class="va">column</span> <span class="op">=</span> <span class="fl">1L</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">centers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/kmeans.html">kmeans</a></span><span class="op">(</span><span class="va">x</span>, centers <span class="op">=</span> <span class="va">categories</span>, algorithm <span class="op">=</span> <span class="va">algorithm</span>,
                    iter.max <span class="op">=</span> <span class="va">iter.max</span><span class="op">)</span><span class="op">$</span><span class="va">centers</span>
  <span class="co"># order the centers</span>
  <span class="va">centers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="va">centers</span><span class="op">)</span>
  <span class="va">cl</span> <span class="op">&lt;-</span> <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/kmeans.html">kmeans</a></span><span class="op">(</span><span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html">na.omit</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, centers <span class="op">=</span> <span class="va">centers</span>,
                      algorithm <span class="op">=</span> <span class="va">algorithm</span>, iter.max <span class="op">=</span> <span class="va">iter.max</span> <span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">x</span>, cluster <span class="op">=</span> <span class="va">cl</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span> <span class="op">)</span>
<span class="op">}</span></code></pre></div>
</div>
<div id="discretization-for-md-ii" class="section level3">
<h3 class="hasAnchor">
<a href="#discretization-for-md-ii" class="anchor"></a>Discretization for <span class="math inline">\(MD\)</span> (II)</h3>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cl</span> <span class="op">=</span> <span class="fu">Kmean.discretization</span><span class="op">(</span><span class="va">md</span>, categories <span class="op">=</span> <span class="fl">3</span>, algorithm <span class="op">=</span> <span class="st">"MacQueen"</span> <span class="op">)</span>
<span class="va">cl_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="kw">function</span><span class="op">(</span><span class="va">k</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">cl</span><span class="op">$</span><span class="va">x</span><span class="op">[</span><span class="va">cl</span><span class="op">$</span><span class="va">cluster</span> <span class="op">==</span> <span class="va">k</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">cl_summary</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"low"</span>, <span class="st">"middle"</span>, <span class="st">"high"</span><span class="op">)</span>
<span class="va">cl_summary</span></code></pre></div>
<pre><code>##               low    middle      high
## Min.    0.0809488 0.9703714  2.348914
## 1st Qu. 0.2543436 1.1740022  2.546431
## Median  0.3925193 1.4335135  2.922496
## Mean    0.4465135 1.4933123  3.204277
## 3rd Qu. 0.6216258 1.7647503  3.509952
## Max.    0.9683489 2.3473622 10.496433</code></pre>
<p>Cutpoint are now applied:</p>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutpoint1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">cl_summary</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="va">cutpoint2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">cl_summary</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span><span class="op">)</span>

<span class="va">dt</span><span class="op">$</span><span class="va">md_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"High"</span><span class="op">)</span>
<span class="va">dt</span><span class="op">$</span><span class="va">md_level</span><span class="op">[</span> <span class="va">dt</span><span class="op">$</span><span class="va">md</span> <span class="op">&lt;</span> <span class="va">cutpoint2</span> <span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Middle"</span>
<span class="va">dt</span><span class="op">$</span><span class="va">md_level</span><span class="op">[</span> <span class="va">dt</span><span class="op">$</span><span class="va">md</span> <span class="op">&lt;</span> <span class="va">cutpoint1</span>   <span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Low"</span>
<span class="va">dt</span><span class="op">$</span><span class="va">md_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">dt</span><span class="op">$</span><span class="va">md_level</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Low"</span>, <span class="st">"Middle"</span>, <span class="st">"High"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="discretization-for-logfc-ii" class="section level3">
<h3 class="hasAnchor">
<a href="#discretization-for-logfc-ii" class="anchor"></a>Discretization for <span class="math inline">\(logFC\)</span> (II)</h3>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cl</span> <span class="op">=</span> <span class="fu">Kmean.discretization</span><span class="op">(</span><span class="va">logfc</span>, categories <span class="op">=</span> <span class="fl">3</span>, algorithm <span class="op">=</span> <span class="st">"MacQueen"</span> <span class="op">)</span>
<span class="va">cl_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="kw">function</span><span class="op">(</span><span class="va">k</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">cl</span><span class="op">$</span><span class="va">x</span><span class="op">[</span><span class="va">cl</span><span class="op">$</span><span class="va">cluster</span> <span class="op">==</span> <span class="va">k</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">cl_summary</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"eLow"</span>, <span class="st">"eMiddle"</span>, <span class="st">"eHigh"</span><span class="op">)</span>
<span class="va">cl_summary</span></code></pre></div>
<pre><code>##                 eLow   eMiddle    eHigh
## Min.    3.772575e-05 0.6448911 2.052246
## 1st Qu. 8.178209e-02 0.7713312 2.324534
## Median  1.865436e-01 0.9716668 2.575225
## Mean    2.257154e-01 1.0636271 3.027001
## 3rd Qu. 3.430369e-01 1.2671241 3.394137
## Max.    6.443115e-01 2.0306128 7.945112</code></pre>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutpoint1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">cl_summary</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="va">cutpoint2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">cl_summary</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span><span class="op">)</span>

<span class="va">dt</span><span class="op">$</span><span class="va">logfc_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"eHigh"</span><span class="op">)</span>
<span class="va">dt</span><span class="op">$</span><span class="va">logfc_level</span><span class="op">[</span> <span class="va">dt</span><span class="op">$</span><span class="va">logfc</span> <span class="op">&lt;</span> <span class="va">cutpoint2</span> <span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"eMiddle"</span>
<span class="va">dt</span><span class="op">$</span><span class="va">logfc_level</span><span class="op">[</span> <span class="va">dt</span><span class="op">$</span><span class="va">logfc</span> <span class="op">&lt;</span> <span class="va">cutpoint1</span>   <span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"eLow"</span>

<span class="va">dt</span><span class="op">$</span><span class="va">logfc_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">dt</span><span class="op">$</span><span class="va">logfc_level</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"eLow"</span>, <span class="st">"eMiddle"</span>, <span class="st">"eHigh"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="the-contingency-table-and-its-graphic-visualization-ii" class="section level3">
<h3 class="hasAnchor">
<a href="#the-contingency-table-and-its-graphic-visualization-ii" class="anchor"></a>The contingency table and its graphic visualization (II)</h3>
<p>In the current case, the contingency table suggests an obvious relationship between <span class="math inline">\(|logFC|\)</span> and <span class="math inline">\(MD\)</span>.</p>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span> <span class="va">dt</span><span class="op">$</span><span class="va">logfc_level</span>, <span class="va">dt</span><span class="op">$</span><span class="va">md_level</span><span class="op">)</span> <span class="op">)</span>
<span class="va">m</span></code></pre></div>
<pre><code>##          
##            Low Middle High
##   eLow    2721   1676  355
##   eMiddle  795    210   59
##   eHigh    128     24    6</code></pre>
<p>The <a href="https://www.rdocumentation.org/packages/graphics/versions/3.6.0/topics/spineplot">spineplot</a> of this contingency table (table(x = logfc_level, y = md_level)) help to visualize the existent relationship between <span class="math inline">\(|logFC|\)</span> and <span class="math inline">\(MD\)</span>.</p>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span> mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">7</span>, <span class="fl">4</span>, <span class="fl">4</span><span class="op">)</span>, family <span class="op">=</span> <span class="st">"Serif"</span>, font <span class="op">=</span> <span class="fl">2</span>, cex <span class="op">=</span> <span class="fl">0.8</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.2</span> <span class="op">)</span>
<span class="va">tb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/spineplot.html">spineplot</a></span><span class="op">(</span><span class="va">m</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">2</span>,<span class="fl">1</span><span class="op">)</span><span class="op">]</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="st">"blue"</span>, <span class="st">"dodgerblue"</span>,<span class="st">"deepskyblue"</span> <span class="op">)</span>,
                xlab <span class="op">=</span> <span class="st">""</span>, ylab <span class="op">=</span> <span class="st">""</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>, text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="va">logFC</span><span class="op">)</span><span class="op">)</span>, 
      line <span class="op">=</span> <span class="fl">2.5</span>, cex <span class="op">=</span> <span class="fl">1</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>, text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="st">"MD"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
      line <span class="op">=</span> <span class="fl">4</span>, cex <span class="op">=</span> <span class="fl">1</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>a <span class="op">=</span> <span class="fl">0.0</span>, b <span class="op">=</span> <span class="fl">0.72</span>, col <span class="op">=</span> <span class="st">"white"</span>, lty <span class="op">=</span> <span class="st">"dashed"</span>, lwd <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="GenExp_Methylation_association_hyper_down-regulated_files/figure-html/spineplot_2-1.png" width="700"></p>
<p>A dashed line was added to emphasize the direction of linear trend. From <em>“Low”</em> to <em>“high”</em> there seems to be a linear relationship between pairwise proportion of <span class="math inline">\(|logFC|\)</span> and <span class="math inline">\(MD\)</span>.</p>
<p>Conceptually, the <em>spineplot</em> is a plot of the conditional probability <span class="math inline">\(P(y|x)\)</span> against <span class="math inline">\(P(x)\)</span>. Since the joint probability distribution of these variable is described by a FGM copula, a linear relationship between <span class="math inline">\(P(y|x)\)</span> against <span class="math inline">\(P(x)\)</span> must be expected, which is a property of FGM copula <span class="citation">(6)</span>.</p>
</div>
<div id="testing-association-between-logfc-and-md-ii" class="section level3">
<h3 class="hasAnchor">
<a href="#testing-association-between-logfc-and-md-ii" class="anchor"></a>Testing association between <span class="math inline">\(logFC\)</span> and <span class="math inline">\(MD\)</span> (II)</h3>
<p>Here, we use the LLT test implemented in function <a href="https://www.rdocumentation.org/packages/coin/versions/1.3-0/topics/ContingencyTests">lbl_test</a> from the R package <em>coin</em>. Two test variants are available: 1) permutation test and asymptotic test.</p>
<p>The asymptotic LLT test:</p>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/coin/man/ContingencyTests.html">lbl_test</a></span><span class="op">(</span> <span class="va">m</span> <span class="op">)</span></code></pre></div>
<pre><code>## 
##  Asymptotic Linear-by-Linear Association Test
## 
## data:  Var2 (ordered) by Var1 (eLow &lt; eMiddle &lt; eHigh)
## Z = -10.163, p-value &lt; 2.2e-16
## alternative hypothesis: two.sided</code></pre>
<p>The permutation LLT test:</p>
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/coin/man/ContingencyTests.html">lbl_test</a></span><span class="op">(</span> <span class="va">m</span>, distribution <span class="op">=</span> <span class="st">"approximate"</span>, B <span class="op">=</span> <span class="fl">10000</span>, alternative <span class="op">=</span> <span class="st">"less"</span> <span class="op">)</span></code></pre></div>
<pre><code>## 
##  Approximative Linear-by-Linear Association Test
## 
## data:  Var2 (ordered) by Var1 (eLow &lt; eMiddle &lt; eHigh)
## Z = -10.163, p-value &lt; 1e-04
## alternative hypothesis: less</code></pre>
<p>The null hypothesis of no association among the variables in the table is rejected. In the current case the <span class="math inline">\(p-value &lt; 1e-04\)</span> and, hence, we do not have enough evidence to support the null hypothesis.</p>
</div>
</div>
<div id="generalized-linear-model-glm-ii" class="section level2">
<h2 class="hasAnchor">
<a href="#generalized-linear-model-glm-ii" class="anchor"></a>Generalized linear model (GLM) (II)</h2>
<p>The variable <span class="math inline">\(MD\)</span> is split int three levels, 0:“Low” (<span class="math inline">\(0 &lt; MD \leq 0.5\)</span>), 1:“Middle” (<span class="math inline">\(0.5 &lt; MD \leq 1\)</span>), and 2:“High” ($1 &lt; MD $)</p>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meth_cut1</span> <span class="op">&lt;-</span> <span class="fl">0.5</span>
<span class="va">meth_cut2</span> <span class="op">&lt;-</span> <span class="fl">1.2</span>

<span class="va">xy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cbind.html">cbind</a></span><span class="op">(</span><span class="va">MD</span>, <span class="va">logFC</span><span class="op">)</span>
<span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">logFC</span> <span class="op">&gt;</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="va">xy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">xy</span><span class="op">[</span><span class="va">idx</span>, <span class="op">]</span><span class="op">)</span>
<span class="va">xy</span><span class="op">$</span><span class="va">md</span> <span class="op">&lt;-</span> <span class="fl">0</span>
<span class="va">xy</span><span class="op">$</span><span class="va">md</span><span class="op">[</span><span class="va">xy</span><span class="op">$</span><span class="va">MD</span> <span class="op">&gt;</span> <span class="va">meth_cut1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span>
<span class="va">xy</span><span class="op">$</span><span class="va">md</span><span class="op">[</span><span class="va">xy</span><span class="op">$</span><span class="va">MD</span> <span class="op">&gt;</span> <span class="va">meth_cut2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">2</span></code></pre></div>
<p>The glm support the decrease of <span class="math inline">\(-logFC\)</span> with the increment of <span class="math inline">\(MD\)</span>. The model coefficients are higly statistically significant.</p>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="va">logFC</span> <span class="op">~</span> <span class="va">md</span>, data<span class="op">=</span><span class="va">xy</span>, family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/family.html">Gamma</a></span><span class="op">(</span>link <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span> <span class="va">gm</span> <span class="op">)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = logFC ~ md, family = Gamma(link = "identity"), 
##     data = xy)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -0.7559  -0.5068  -0.2627   0.1476   2.9485  
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)  1.18737    0.02644  44.915  &lt; 2e-16 ***
## md          -0.10495    0.02264  -4.635 3.84e-06 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## (Dispersion parameter for Gamma family taken to be 0.490523)
## 
##     Null deviance: 532.96  on 1702  degrees of freedom
## Residual deviance: 522.93  on 1701  degrees of freedom
## AIC: 2751.1
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<p>The ANOVA test confirm the negative trend given by the coefficient of MD (highly statistically significant).</p>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">gm</span>, test <span class="op">=</span> <span class="st">"LRT"</span><span class="op">)</span></code></pre></div>
<pre><code>## Analysis of Deviance Table
## 
## Model: Gamma, link: identity
## 
## Response: logFC
## 
## Terms added sequentially (first to last)
## 
## 
##      Df Deviance Resid. Df Resid. Dev  Pr(&gt;Chi)    
## NULL                  1702     532.96              
## md    1   10.034      1701     522.93 6.106e-06 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
<div id="glm-group-i-low-md-versus-group-ii-middle-md-ii" class="section level3">
<h3 class="hasAnchor">
<a href="#glm-group-i-low-md-versus-group-ii-middle-md-ii" class="anchor"></a>GLM group I (“Low” <span class="math inline">\(MD\)</span>) versus group II (“Middle” <span class="math inline">\(MD\)</span>) (II)</h3>
<p>Only genes with <span class="math inline">\(MD &lt; 1\)</span> are analyzed. That is, only two groups are considered: 0:“Middle” (<span class="math inline">\(0.5 &lt; MD \leq 1\)</span>), and 1:“High” ($1 &lt; MD $)</p>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dt</span> <span class="op">&lt;-</span> <span class="va">xy</span><span class="op">[</span><span class="va">xy</span><span class="op">$</span><span class="va">MD</span> <span class="op">&lt;</span> <span class="va">meth_cut2</span>, <span class="op">]</span>
<span class="va">dt</span><span class="op">$</span><span class="va">md</span> <span class="op">&lt;-</span> <span class="fl">0</span>
<span class="va">dt</span><span class="op">$</span><span class="va">md</span><span class="op">[</span> <span class="va">dt</span><span class="op">$</span><span class="va">MD</span> <span class="op">&gt;</span> <span class="va">meth_cut1</span>   <span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span>

<span class="va">gm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="va">logFC</span> <span class="op">~</span> <span class="va">md</span>, data <span class="op">=</span> <span class="va">dt</span>, family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/family.html">Gamma</a></span><span class="op">(</span>link <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span> <span class="va">gm</span> <span class="op">)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = logFC ~ md, family = Gamma(link = "identity"), 
##     data = dt)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -0.7587  -0.5299  -0.2535   0.1411   2.9756  
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)  1.19166    0.02834  42.049  &lt; 2e-16 ***
## md          -0.12276    0.04503  -2.726  0.00649 ** 
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## (Dispersion parameter for Gamma family taken to be 0.4971405)
## 
##     Null deviance: 430.00  on 1342  degrees of freedom
## Residual deviance: 426.45  on 1341  degrees of freedom
## AIC: 2291.1
## 
## Number of Fisher Scoring iterations: 3</code></pre>
<div class="sourceCode" id="cb121"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\n =============== The ANOVA test  =============== \n\n"</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##  =============== The ANOVA test  ===============</code></pre>
<div class="sourceCode" id="cb123"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">gm</span>, test <span class="op">=</span> <span class="st">"LRT"</span><span class="op">)</span></code></pre></div>
<pre><code>## Analysis of Deviance Table
## 
## Model: Gamma, link: identity
## 
## Response: logFC
## 
## Terms added sequentially (first to last)
## 
## 
##      Df Deviance Resid. Df Resid. Dev Pr(&gt;Chi)   
## NULL                  1342     430.00            
## md    1   3.5479      1341     426.45 0.007552 **
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div id="glm-group-ii-middle-md-versus-group-iii-high-md-ii" class="section level3">
<h3 class="hasAnchor">
<a href="#glm-group-ii-middle-md-versus-group-iii-high-md-ii" class="anchor"></a>GLM group II (“Middle” <span class="math inline">\(MD\)</span>) versus group III (“High” <span class="math inline">\(MD\)</span>) (II)</h3>
<p>Only genes with <span class="math inline">\(MD &lt; 1\)</span> are analyzed. That is, only two groups are considered: 0:“Low” (<span class="math inline">\(0 &lt; MD \leq 0.5\)</span>), 1:“Middle” (<span class="math inline">\(0.5 &lt; MD \leq 1\)</span>).</p>
<div class="sourceCode" id="cb125"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dt</span> <span class="op">&lt;-</span> <span class="va">xy</span><span class="op">[</span><span class="va">xy</span><span class="op">$</span><span class="va">MD</span> <span class="op">&gt;</span> <span class="va">meth_cut1</span>, <span class="op">]</span>
<span class="va">dt</span><span class="op">$</span><span class="va">md</span> <span class="op">&lt;-</span> <span class="fl">0</span>
<span class="va">dt</span><span class="op">$</span><span class="va">md</span><span class="op">[</span> <span class="va">dt</span><span class="op">$</span><span class="va">MD</span> <span class="op">&gt;</span> <span class="va">meth_cut2</span>   <span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span>

<span class="va">gm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html">glm</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="va">logFC</span> <span class="op">~</span> <span class="va">md</span>, data <span class="op">=</span> <span class="va">dt</span>, family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/family.html">Gamma</a></span><span class="op">(</span>link <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span> <span class="va">gm</span> <span class="op">)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = logFC ~ md, family = Gamma(link = "identity"), 
##     data = dt)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -0.6729  -0.4812  -0.2722   0.1356   2.9756  
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)  1.06890    0.03514  30.419   &lt;2e-16 ***
## md          -0.08433    0.05084  -1.659   0.0976 .  
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## (Dispersion parameter for Gamma family taken to be 0.5014626)
## 
##     Null deviance: 238.85  on 823  degrees of freedom
## Residual deviance: 237.48  on 822  degrees of freedom
## AIC: 1171.9
## 
## Number of Fisher Scoring iterations: 3</code></pre>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\n =============== The ANOVA test  =============== \n\n"</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##  =============== The ANOVA test  ===============</code></pre>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">gm</span>, test <span class="op">=</span> <span class="st">"LRT"</span><span class="op">)</span></code></pre></div>
<pre><code>## Analysis of Deviance Table
## 
## Model: Gamma, link: identity
## 
## Response: logFC
## 
## Terms added sequentially (first to last)
## 
## 
##      Df Deviance Resid. Df Resid. Dev Pr(&gt;Chi)  
## NULL                   823     238.85           
## md    1   1.3641       822     237.48  0.09908 .
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div id="boxplot-ii" class="section level3">
<h3 class="hasAnchor">
<a href="#boxplot-ii" class="anchor"></a>Boxplot (II)</h3>
<p>The boxplot suggest a clear trend of <span class="math inline">\(-logFC\)</span> decreasing with increment of MD levels. Notice that the <em>notchs</em> from the boxes do not overlap, which implies that the differences between medians from three groups (“Low,” “Middle,” and “High”) are statistically significant. So, the GLM results can be visualized.</p>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>bty <span class="op">=</span> <span class="st">"n"</span>, family <span class="op">=</span> <span class="st">"serif"</span>, font <span class="op">=</span> <span class="fl">2</span>, cex <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span><span class="op">(</span><span class="va">logFC</span> <span class="op">~</span> <span class="va">md</span>, data <span class="op">=</span> <span class="va">xy</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.4</span>, <span class="fl">2.5</span><span class="op">)</span>, xaxt <span class="op">=</span><span class="st">"n"</span>,  yaxt <span class="op">=</span> <span class="st">"n"</span>, ann <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/rect.html">rect</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.4</span>, <span class="fl">3.5</span>, <span class="fl">5</span>, col <span class="op">=</span> <span class="st">"grey94"</span>, lty <span class="op">=</span> <span class="fl">0</span>, lwd <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/grid.html">grid</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span>, col <span class="op">=</span> <span class="st">"white"</span>, lwd <span class="op">=</span> <span class="fl">2</span>, lty <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span><span class="op">(</span><span class="va">logFC</span> <span class="op">~</span> <span class="va">md</span>, data <span class="op">=</span> <span class="va">xy</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.4</span>, <span class="fl">2.5</span><span class="op">)</span>, notch <span class="op">=</span> <span class="cn">TRUE</span>, xaxt <span class="op">=</span><span class="st">"n"</span>,
        yaxt <span class="op">=</span> <span class="st">"n"</span>, ann <span class="op">=</span> <span class="cn">FALSE</span>, pch <span class="op">=</span> <span class="fl">20</span>, medcol <span class="op">=</span> <span class="st">"darkblue"</span>, whisklwd <span class="op">=</span> <span class="fl">1</span>, staplelwd <span class="op">=</span> <span class="fl">1</span>,
        col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"deepskyblue"</span>, <span class="st">"dodgerblue"</span>, <span class="st">"dodgerblue3"</span> <span class="op">)</span>, outcol <span class="op">=</span> <span class="st">"red"</span>, add  <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">1</span>, padj <span class="op">=</span> <span class="op">-</span><span class="fl">0.6</span>, at <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, lwd <span class="op">=</span> <span class="fl">0.5</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.02</span>, line <span class="op">=</span> <span class="op">-</span><span class="fl">0.6</span>, cex.axis <span class="op">=</span> <span class="fl">1.3</span>, 
     labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Low"</span>, <span class="st">"Middle"</span>, <span class="st">"High"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">2</span>, hadj <span class="op">=</span> <span class="fl">0.6</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.5</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.02</span>, line <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, cex.axis <span class="op">=</span> <span class="fl">1.3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>, text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="st">"MD"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
      line <span class="op">=</span> <span class="fl">1.3</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>, text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="op">-</span><span class="va">logFC</span><span class="op">)</span><span class="op">)</span>, line <span class="op">=</span> <span class="fl">1</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, font <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fl">1.5</span>, y <span class="op">=</span> <span class="fl">2.2</span>, <span class="st">"**"</span>, pos <span class="op">=</span> <span class="fl">3</span>, cex <span class="op">=</span> <span class="fl">1.2</span> <span class="op">)</span></code></pre></div>
<p><img src="GenExp_Methylation_association_hyper_down-regulated_files/figure-html/boxplot_3-1.png" width="864"></p>
</div>
</div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body">
<div id="ref-Yang2020" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">X. Yang, <em>et al.</em>, <span class="nocase">Segregation of an MSH1 RNAi transgene produces heritable non-genetic memory in association with methylome reprogramming</span>. <em>Nature Communications</em> <strong>11</strong>, 2214 (2020).</div>
</div>
<div id="ref-Sklar1959" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">A. Sklar, <span class="nocase">Fonctions de répartition à n dimensions et leurs marges</span>. <em>Publ. Inst. Statist. Univ. Paris</em> <strong>8</strong>, 229–231 (1959).</div>
</div>
<div id="ref-Sklar1973" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">A. Sklar, <span class="nocase">Random Variables, Joint Distribution Functions, and Copulas</span>. <em>Kybernetika</em> <strong>9</strong>, 449–460 (1973).</div>
</div>
<div id="ref-Christian2007" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">G. Christian, A. Favre, <span class="nocase">Everything You Always Wanted to Know about Copula Modeling but Were Afraid to Ask</span>. <em>J HYDROL ENG</em> <strong>12</strong>, 347–368 (2007).</div>
</div>
<div id="ref-Agresti2007" class="csl-entry">
<div class="csl-left-margin">5. </div>
<div class="csl-right-inline">A. Agresti, <em><span class="nocase">An Introduction to Categorical Data Analysis</span></em> (John Wiley <span>&amp;</span> Sons, Inc., 2007).</div>
</div>
<div id="ref-Balakrishna2009" class="csl-entry">
<div class="csl-left-margin">6. </div>
<div class="csl-right-inline">N. Balakrishna, C. D. Lai, <em><span class="nocase">Distributions Expressed as Copulas</span></em> (Springer New York, 2009).</div>
</div>
</div>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo()</a></code> on the system on which this document was compiled running pandoc 2.11.4:</p>
<pre><code>## R version 4.1.1 (2021-08-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.3.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] coin_1.4-1           survival_3.2-13      copula_1.0-1        
##  [4] plot3D_1.4           usefr_0.1.0          edgeR_3.34.1        
##  [7] limma_3.48.3         ggplot2_3.3.5        GenomicRanges_1.44.0
## [10] GenomeInfoDb_1.28.4  IRanges_2.26.0       S4Vectors_0.30.1    
## [13] BiocGenerics_0.38.0 
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-7           matrixStats_0.59.0     fs_1.5.0              
##  [4] rprojroot_2.0.2        numDeriv_2016.8-1.1    tools_4.1.1           
##  [7] bslib_0.2.5.1          utf8_1.2.1             R6_2.5.0              
## [10] DBI_1.1.1              colorspace_2.0-2       withr_2.4.2           
## [13] tidyselect_1.1.1       compiler_4.1.1         pspline_1.0-18        
## [16] textshaping_0.3.5      sandwich_3.0-1         desc_1.3.0            
## [19] sass_0.4.0             scales_1.1.1           mvtnorm_1.1-2         
## [22] pkgdown_1.6.1          systemfonts_1.0.2      stringr_1.4.0         
## [25] digest_0.6.27          rmarkdown_2.10         XVector_0.32.0        
## [28] pkgconfig_2.0.3        htmltools_0.5.1.1      highr_0.9             
## [31] fastmap_1.1.0          stabledist_0.7-1       ADGofTest_0.3         
## [34] rlang_0.4.11           jquerylib_0.1.4        generics_0.1.0        
## [37] zoo_1.8-9              jsonlite_1.7.2         mixdist_0.5-5         
## [40] mclust_5.4.7           BiocParallel_1.26.2    dplyr_1.0.7           
## [43] RCurl_1.98-1.3         magrittr_2.0.1         nls2_0.2              
## [46] modeltools_0.2-23      GenomeInfoDbData_1.2.6 Matrix_1.3-4          
## [49] Rcpp_1.0.7             munsell_0.5.0          fansi_0.5.0           
## [52] lifecycle_1.0.0        stringi_1.6.2          multcomp_1.4-17       
## [55] yaml_2.2.1             MASS_7.3-54            zlibbioc_1.38.0       
## [58] grid_4.1.1             misc3d_0.9-0           crayon_1.4.1          
## [61] lattice_0.20-45        splines_4.1.1          locfit_1.5-9.4        
## [64] knitr_1.33             pillar_1.6.1           tcltk_4.1.1           
## [67] cubature_2.0.4.2       codetools_0.2-18       glue_1.4.2            
## [70] evaluate_0.14          vctrs_0.3.8            gtable_0.3.0          
## [73] purrr_0.3.4            assertthat_0.2.1       cachem_1.0.5          
## [76] xfun_0.24              libcoin_1.0-8          ragg_1.1.3            
## [79] pcaPP_1.9-74           gsl_2.1-7              minpack.lm_1.2-1      
## [82] tibble_3.1.2           memoise_2.0.0          TH.data_1.0-10        
## [85] ellipsis_0.3.2</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
