<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Methylation analysis with Methyl-IT • MethylIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Methylation analysis with Methyl-IT">
<meta property="og:description" content="MethylIT">
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Methylation analysis with Methyl-IT</h1>
            <h3 data-toc-skip class="subtitle"><h1> ¶ An example with
simulated datasets. A guide for users</h1></h3>
                        <h4 data-toc-skip class="author"><div class="line-block">Robersy Sanchez<br><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a><br>
Mackenzie’s lab<br><br>
Department of Biology and Plant Science.<br>
Pennsylvania State University, University Park, PA 16802</div></h4>
            
            <h4 data-toc-skip class="date">08 February 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/vignettes/Methylation_analysis_with_Methyl-IT.Rmd" class="external-link"><code>vignettes/Methylation_analysis_with_Methyl-IT.Rmd</code></a></small>
      <div class="hidden name"><code>Methylation_analysis_with_Methyl-IT.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Methylation analysis with <a href="https://github.com/genomaths/MethylIT" class="external-link">Methyl-IT</a> is
      illustrated on simulated datasets of methylated and unmethylated
      read counts with relatively high average of methylation levels:
      0.15 and 0.286 for control and treatment groups, respectively. The
      main Methyl-IT downstream analysis is presented alongside the
      application of Fisher’s exact test. The importance of a signal
      detection step is shown.</p>
    </div>
    
<style type="text/css">

.list-group-item.active, .list-group-item.active:focus, 
.list-group-item.active:hover {
    z-index: 2;
    color: #fff;
    background-color: #337ab7;
    border-color: #337ab7;
}

.tocify-subheader .tocify-item {
    font-size: 0.80em;
    padding-left: 25px;
    text-indent: 0;
}

.tocify-header .tocify-item {
    font-size: 0.85em;
    padding-left: 20px;
    text-indent: 0;
}

body{ /* Normal  */
    font-size: 18px;
    font-family: "Times New Roman", Times, serif;
    text-align: justify
}
td {  /* Table  */
    font-size: 8px;
}

h1.title {
    font-size: 38px;
    font-family: "Times New Roman", Times, serif;
    color: DarkRed;
    .subTitle {
        font-size: 24px;
        font-family: "Times New Roman", Times, serif;
        color: DarkRed;
    }
}

h1 { /* Header 1 */
    font-size: 28px;
    font-family: "Times New Roman", Times, serif;
    color: DarkBlue;
}
h2 { /* Header 2 */
    font-size: 22px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
h3 { /* Header 3 */
    font-size: 18px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
code.r{ /* Code block */
    font-size: 12px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>
<div class="section level2">
<h2 id="background">Background<a class="anchor" aria-label="anchor" href="#background"></a>
</h2>
<p>Methyl-IT R package offers a methylome analysis approach based on
information thermodynamics (IT) and signal detection. Methyl-IT approach
confront detection of differentially methylated cytosine as a signal
detection problem. This approach was designed to discriminate
methylation regulatory signal from background noise induced by molecular
stochastic fluctuations. Methyl-IT R package is not limited to the IT
approach but also includes Fisher’s exact test (FT), Root-mean-square
statistic (RMST) or Hellinger divergence (HDT) tests. Herein, we will
show that a signal detection step is also required for FT, RMST, and HDT
as well. It is worthy to notice that, as for any standard statistical
analysis, any serious robust methylation analysis requires for a
descriptive statistical analysis at the beginning or at different
downstream steps. This is an ABC principle taught in any undergraduate
course on statistics. Methylation analysis is not the exception of the
rule. This detail is included in this example.</p>
<p>Note: This example was made with the MethylIT version at <a href="https://github.com/genomaths/MethylIT" class="external-link uri">https://github.com/genomaths/MethylIT</a>.</p>
</div>
<div class="section level2">
<h2 id="available-datasets-and-reading">Available datasets and reading<a class="anchor" aria-label="anchor" href="#available-datasets-and-reading"></a>
</h2>
<p>Methylome datasets of whole-genome bisulfite sequencing (WGBS) are
available at Gene Expression Omnibus (GEO DataSets). The data set are
downloaded providing the GEO accession numbers for each data set to the
function <em>getGEOSuppFiles</em> (for details type ?getGEOSuppFiles in
the R console). Then, datasets can be read using function
<em>readCounts2GRangesList</em>. An example on how to load datasets of
read-counts of methylated and unmethylated cytosine into Methyl-IT is
given in the Cancer example at <a href="https://git.psu.edu/genomath/MethylIT" class="external-link uri">https://git.psu.edu/genomath/MethylIT</a></p>
</div>
<div class="section level2">
<h2 id="data-generation">Data generation<a class="anchor" aria-label="anchor" href="#data-generation"></a>
</h2>
<p>For the current example on methylation analysis with Methyl-IT we
will use simulated data. Read-count matrices of methylated and
unmethylated cytosine are generated with Methyl-IT function <a href="https://is.gd/gg0yTp" class="external-link">simulateCounts</a>.</p>
<p>Function <em>simulateCounts</em> randomly generates prior methylation
levels using Beta distribution function. The expected mean of
methylation levels that we would like to have can be estimated using the
auxiliary function:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bmean</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">alpha</span>, <span class="va">beta</span><span class="op">)</span> <span class="va">alpha</span><span class="op">/</span><span class="op">(</span><span class="va">alpha</span> <span class="op">+</span> <span class="va">beta</span><span class="op">)</span></span>
<span><span class="va">alpha.ct</span> <span class="op">&lt;-</span> <span class="fl">0.09</span></span>
<span><span class="va">alpha.tt</span> <span class="op">&lt;-</span> <span class="fl">0.2</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>control.group <span class="op">=</span> <span class="fu">bmean</span><span class="op">(</span><span class="va">alpha.ct</span>, <span class="fl">0.5</span><span class="op">)</span>, treatment.group <span class="op">=</span> <span class="fu">bmean</span><span class="op">(</span><span class="va">alpha.tt</span>, <span class="fl">0.5</span><span class="op">)</span>, </span>
<span>    mean.diff <span class="op">=</span> <span class="fu">bmean</span><span class="op">(</span><span class="va">alpha.tt</span>, <span class="fl">0.5</span><span class="op">)</span> <span class="op">-</span> <span class="fu">bmean</span><span class="op">(</span><span class="va">alpha.ct</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">##   control.group treatment.group       mean.diff </span></span>
<span><span class="co">##       0.1525424       0.2857143       0.1331719</span></span></code></pre>
<p>This simple function uses the <span class="math inline">\(\alpha\)</span> (<em>shape1</em>) and <span class="math inline">\(\beta\)</span> (<em>shape2</em>) parameters from
the Beta distribution function to compute the expected value of
methylation levels. In the current case, we expect to have a difference
of methylation levels about 0.133 between the control and the
treatment.</p>
<div class="section level3">
<h3 id="simulation">Simulation<a class="anchor" aria-label="anchor" href="#simulation"></a>
</h3>
<p>Function <em>simulateCounts</em> from <a href="https://github.com/genomaths/MethylIT.utils" class="external-link">MethylIT.utils</a> R
package will be used to generate the datasets, which will include three
group of samples: reference, control, and treatment.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/MethylIT" class="external-link">MethylIT</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MethylIT.utils</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Bioconductor/BiocParallel" class="external-link">BiocParallel</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The number of cytosine sites to generate</span></span>
<span><span class="va">sites</span> <span class="op">=</span> <span class="fl">50000</span> </span>
<span><span class="co"># Set a seed for pseudo-random number generation</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">124</span><span class="op">)</span></span>
<span><span class="va">control.nam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"C1"</span>, <span class="st">"C2"</span>, <span class="st">"C3"</span><span class="op">)</span></span>
<span><span class="va">treatment.nam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T1"</span>, <span class="st">"T2"</span>, <span class="st">"T3"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Reference group </span></span>
<span><span class="va">ref0</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html" class="external-link">simulateCounts</a></span><span class="op">(</span>num.samples <span class="op">=</span> <span class="fl">4</span>, sites <span class="op">=</span> <span class="va">sites</span>, alpha <span class="op">=</span> <span class="va">alpha.ct</span>, </span>
<span>                    beta <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">50</span>, theta <span class="op">=</span> <span class="fl">4.5</span>, </span>
<span>                    sample.ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"R1"</span>, <span class="st">"R2"</span>, <span class="st">"R3"</span>, <span class="st">"R4"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Control group</span></span>
<span><span class="va">ctrl</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html" class="external-link">simulateCounts</a></span><span class="op">(</span>num.samples <span class="op">=</span> <span class="fl">3</span>, sites <span class="op">=</span> <span class="va">sites</span>, alpha <span class="op">=</span> <span class="va">alpha.ct</span>, </span>
<span>                    beta <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">50</span>, theta <span class="op">=</span> <span class="fl">4.5</span>,</span>
<span>                    sample.ids <span class="op">=</span> <span class="va">control.nam</span><span class="op">)</span></span>
<span><span class="co"># Treatment group</span></span>
<span><span class="va">treat</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html" class="external-link">simulateCounts</a></span><span class="op">(</span>num.samples <span class="op">=</span> <span class="fl">3</span>, sites <span class="op">=</span> <span class="va">sites</span>, alpha <span class="op">=</span> <span class="va">alpha.tt</span>, </span>
<span>                    beta <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">50</span>, theta <span class="op">=</span> <span class="fl">4.5</span>, </span>
<span>                    sample.ids <span class="op">=</span> <span class="va">treatment.nam</span><span class="op">)</span></span></code></pre></div>
<p>Notice that reference and control groups of samples are not identical
but belong to the same population.</p>
</div>
<div class="section level3">
<h3 id="divergences-of-methylation-levels">Divergences of methylation levels<a class="anchor" aria-label="anchor" href="#divergences-of-methylation-levels"></a>
</h3>
<p>The estimation of the divergences of methylation levels is required
to proceed with the application of signal detection basic approach. The
information divergence is estimated here using the function
<em>estimateDivergence</em>. For each cytosine site, methylation levels
are estimated according to the formulas: <span class="math inline">\(p_i={n_i}^{mC_j}/({n_i}^{mC_j}+{n_i}^{C_j})\)</span>,
where <span class="math inline">\({n_i}^{mC_j}\)</span> and <span class="math inline">\({n_i}^{C_j}\)</span> are the number of methylated
and unmethylated cytosines at site <span class="math inline">\(i\)</span>.</p>
<p>If a Bayesian correction of counts is selected in function
<em>estimateDivergence</em>, then methylated read counts are modeled by
a beta-binomial distribution in a Bayesian framework, which accounts for
the biological and sampling variations <span class="citation">(1–3)</span>. In our case we adopted the Bayesian
approach suggested in reference <span class="citation">(4)</span>
(Chapter 3).</p>
<p>Two types of information divergences are estimated: <em>TV</em>,
total variation (<em>TV</em>, absolute value of methylation levels) and
Hellinger divergence (<em>H</em>). <em>TV</em> is computed according to
the formula: <span class="math inline">\(TV_d=|p_{tt}-p_{ct}|\)</span>
and <em>H</em>:</p>
<p><span class="math display">\[H(\hat p_{ij},\hat p_{ir}) =
w_i\Big[(\sqrt{\hat p_{ij}} - \sqrt{\hat
p_{ir}})^2+(\sqrt{1-\hat p_{ij}} - \sqrt{1-\hat p_{ir}})^2\Big]\]</span>
where <span class="math inline">\(w_i = 2 \frac{m_{ij} m_{ir}}{m_{ij} +
m_{ir}}\)</span>, <span class="math inline">\(m_{ij} =
{n_i}^{mC_j}+{n_i}^{uC_j}+1\)</span>, <span class="math inline">\(m_{ir}
= {n_i}^{mC_r}+{n_i}^{uC_r}+1\)</span> and <span class="math inline">\(j
\in {\{c,t}\}\)</span></p>
<p>The equation for Hellinger divergence is given in reference <span class="citation">(5)</span>, but any other information theoretical
divergences could be used as well.</p>
<p>Divergences are estimated for control and treatment groups in respect
to a virtual sample, which is created applying function
<em>poolFromGRlist</em> on the reference group.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Reference sample</span></span>
<span><span class="va">ref</span> <span class="op">=</span> <span class="fu"><a href="../reference/poolFromGRlist.html">poolFromGRlist</a></span><span class="op">(</span><span class="va">ref0</span>, stat <span class="op">=</span> <span class="st">"mean"</span>, num.cores <span class="op">=</span> <span class="fl">4L</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Methylation level divergences</span></span>
<span><span class="va">DIVs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/estimateDivergence.html">estimateDivergence</a></span><span class="op">(</span></span>
<span>                        ref <span class="op">=</span> <span class="va">ref</span>,</span>
<span>                        indiv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">ctrl</span>, <span class="va">treat</span><span class="op">)</span>, </span>
<span>                        Bayesian <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                        and.min.cov <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                        num.cores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html" class="external-link">multicoreWorkers</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>                        percentile <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                        verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>The setting and.min.cov = TRUE in <a href="https://is.gd/0ViBO9" class="external-link">estimateDivergence</a> will accomplish
pairwise deletion of missing cytosine sites or sites no holding the
filtering condition. The mean of methylation levels differences is:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">DIVs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span>, <span class="fl">7</span><span class="op">]</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##           C1           C2           C3           T1           T2           T3 </span></span>
<span><span class="co">## 2.809709e-03 2.642337e-03 7.127907e-05 1.615112e-01 1.631192e-01 1.642946e-01</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="methylation-signal">Methylation signal<a class="anchor" aria-label="anchor" href="#methylation-signal"></a>
</h2>
<p>Likewise for any other signal in nature, the analysis of methylation
signal requires for the knowledge of its probability distribution. In
the current case , the signal is represented in terms of the Hellinger
divergence of methylation levels (<em>H</em>).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To remove hd == 0 to estimate. The methylation signal only is given for  </span></span>
<span><span class="va">divs</span> <span class="op">=</span> <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">DIVs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">div</span><span class="op">)</span> <span class="va">div</span><span class="op">[</span> <span class="va">div</span><span class="op">$</span><span class="va">hdiv</span> <span class="op">&gt;</span> <span class="fl">0</span> <span class="op">]</span>, keep.attr <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">divs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">DIVs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Data frame with the Hellinger divergences from both groups of samples samples </span></span>
<span><span class="va">l</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span>; <span class="kw">for</span> <span class="op">(</span><span class="va">k</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">divs</span><span class="op">)</span><span class="op">)</span> <span class="va">l</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">l</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">divs</span><span class="op">[[</span><span class="va">k</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">data_div</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>H <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">divs</span><span class="op">$</span><span class="va">C1</span><span class="op">$</span><span class="va">hdiv</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">divs</span><span class="op">$</span><span class="va">C2</span><span class="op">$</span><span class="va">hdiv</span><span class="op">)</span>, </span>
<span>                            <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">divs</span><span class="op">$</span><span class="va">C3</span><span class="op">$</span><span class="va">hdiv</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">divs</span><span class="op">$</span><span class="va">T1</span><span class="op">$</span><span class="va">hdiv</span><span class="op">)</span>, </span>
<span>                            <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">divs</span><span class="op">$</span><span class="va">T2</span><span class="op">$</span><span class="va">hdiv</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">divs</span><span class="op">$</span><span class="va">T3</span><span class="op">$</span><span class="va">hdiv</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                    sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"C1"</span>, <span class="va">l</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"C2"</span>,<span class="va">l</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"C3"</span>,<span class="va">l</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                                <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"T1"</span>,<span class="va">l</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"T2"</span>,<span class="va">l</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"T3"</span>,<span class="va">l</span><span class="op">[</span><span class="fl">6</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Empirical critical values for the probability distribution of <span class="math inline">\(H\)</span> and <span class="math inline">\(TV\)</span> can be obtained using
<em>quantile</em> function from the R package <em>stats</em>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">critical.val</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">divs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">hd.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">hdiv</span>, <span class="fl">0.95</span><span class="op">)</span></span>
<span>    <span class="va">tv.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">bay.TV</span><span class="op">)</span>, <span class="fl">0.95</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>tv <span class="op">=</span> <span class="va">tv.95</span>, hd <span class="op">=</span> <span class="va">hd.95</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">critical.val</span></span></code></pre></div>
<pre><code><span><span class="co">##       tv.95%    hd.95%</span></span>
<span><span class="co">## C1 0.6732098  66.76003</span></span>
<span><span class="co">## C2 0.6707588  66.71971</span></span>
<span><span class="co">## C3 0.6669981  65.98486</span></span>
<span><span class="co">## T1 0.9232609 138.68200</span></span>
<span><span class="co">## T2 0.9318191 141.72617</span></span>
<span><span class="co">## T3 0.9301631 141.77236</span></span></code></pre>
<p>The kernel density estimation yields the empirical density shown in
the graphics:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Some information for graphic</span></span>
<span><span class="va">crit.val.ct</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">critical.val</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"C1"</span>, <span class="st">"C2"</span>, <span class="st">"C3"</span><span class="op">)</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="co"># 67</span></span>
<span><span class="va">crit.val.tt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">critical.val</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T1"</span>, <span class="st">"T2"</span>, <span class="st">"T3"</span><span class="op">)</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="co"># 139</span></span>
<span></span>
<span><span class="co"># Density plot with ggplot</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">data_div</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">H</span>, colour <span class="op">=</span> <span class="va">sample</span>, fill <span class="op">=</span> <span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html" class="external-link">geom_density</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.05</span>, bw <span class="op">=</span> <span class="fl">0.2</span>, position <span class="op">=</span> <span class="st">"identity"</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            linewidth <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span>   </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/expression.html" class="external-link">expression</a></span><span class="op">(</span><span class="fu">bolditalic</span><span class="op">(</span><span class="st">"Hellinger divergence (H)"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/expression.html" class="external-link">expression</a></span><span class="op">(</span><span class="fu">bolditalic</span><span class="op">(</span><span class="st">"Density"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Density distribution for control and treatment"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="va">crit.val.ct</span>, color <span class="op">=</span> <span class="st">"red"</span>, </span>
<span>                linetype <span class="op">=</span> <span class="st">"dashed"</span>, linewidth <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotate.html" class="external-link">annotate</a></span><span class="op">(</span>geom <span class="op">=</span> <span class="st">"text"</span>, x <span class="op">=</span> <span class="va">crit.val.ct</span><span class="op">-</span><span class="fl">2</span>, y <span class="op">=</span> <span class="fl">0.08</span>, size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>            label <span class="op">=</span> <span class="st">'bolditalic(H[alpha == 0.05]^CT == 67)'</span>,</span>
<span>            family <span class="op">=</span> <span class="st">"serif"</span>, color <span class="op">=</span> <span class="st">"red"</span>, parse <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="va">crit.val.tt</span>, color <span class="op">=</span> <span class="st">"blue"</span>, </span>
<span>                linetype <span class="op">=</span> <span class="st">"dashed"</span>, linewidth <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotate.html" class="external-link">annotate</a></span><span class="op">(</span>geom <span class="op">=</span> <span class="st">"text"</span>, x <span class="op">=</span> <span class="va">crit.val.tt</span> <span class="op">-</span><span class="fl">2</span>, y <span class="op">=</span> <span class="fl">0.1</span>, size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>            label <span class="op">=</span> <span class="st">'bolditalic(H[alpha == 0.05]^TT == 139)'</span>,</span>
<span>            family <span class="op">=</span> <span class="st">"serif"</span>, color <span class="op">=</span> <span class="st">"blue"</span>, parse <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        axis.text.x  <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span> face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">12</span>, color<span class="op">=</span><span class="st">"black"</span>,</span>
<span>                                margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">margin</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0</span>, unit <span class="op">=</span> <span class="st">"pt"</span> <span class="op">)</span><span class="op">)</span>,</span>
<span>        axis.text.y  <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span> face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">12</span>, color<span class="op">=</span><span class="st">"black"</span>, </span>
<span>                                margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">margin</a></span><span class="op">(</span> <span class="fl">0</span>,<span class="fl">0.1</span>,<span class="fl">0</span>,<span class="fl">0</span>, unit <span class="op">=</span> <span class="st">"mm"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">13</span>,</span>
<span>                                color<span class="op">=</span><span class="st">"black"</span>, vjust <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">13</span>,</span>
<span>                                color<span class="op">=</span><span class="st">"black"</span>, vjust <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span>,</span>
<span>        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">margin</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.3</span>, <span class="fl">0.3</span>, <span class="fl">0.3</span><span class="op">)</span>, unit <span class="op">=</span> <span class="st">'mm'</span><span class="op">)</span>,</span>
<span>        legend.box.spacing <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="st">"lines"</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">12</span>, family <span class="op">=</span> <span class="st">"serif"</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="Methylation_analysis_with_Methyl-IT_files/figure-html/dens.graph-1.png" width="700"></p>
<p>The graphic above shows that with high probability the methylation
signal induced by the treatment has <em>H</em> values <span class="math inline">\(H^{TT}_{\alpha=0.05}\geq139\)</span>. According to
the critical value estimated for the differences of methylation levels,
the methylation signal holds <span class="math inline">\(TV^{TT}_{\alpha=0.05}\geq0.67\)</span>.</p>
<p>Notice that most of the methylation changes are not signal but
methylation background noise (found to the left of the critical values).
This situation is typical for all the natural and technologically
generated signals <span class="citation">(6, 7)</span>.</p>
<div class="section level3">
<h3 id="signal-detection">Signal detection<a class="anchor" aria-label="anchor" href="#signal-detection"></a>
</h3>
<p>Signal detection is a critical step to increase sensitivity and
resolution of methylation signal by reducing the signal-to-noise ratio
and objectively controlling the false positive rate and prediction
accuracy/risk.</p>
<p>The first estimation in our signal detection step is the
identification of the cytosine sites carrying potential methylation
signal <span class="math inline">\(PS\)</span>. To identify cytosine
sites carrying <span class="math inline">\(PS\)</span>, we must find the
best fitted probability distribution model.</p>
<div class="section level4">
<h4 id="estimation-of-the-best-fitted-probability-distribution-model">
<em>Estimation of the best fitted probability distribution
model</em><a class="anchor" aria-label="anchor" href="#estimation-of-the-best-fitted-probability-distribution-model"></a>
</h4>
<p>Potential DMPs can be estimated using the critical values derived
from the best fitted probability distribution model, which is obtained
after the non-linear fit of the theoretical model on the genome-wide
<span class="math inline">\(H\)</span> values for each individual sample
using Methyl-IT function <a href="https://is.gd/0BLLkT" class="external-link">nonlinearFitDist</a> <span class="citation">(6)</span>. As before, only cytosine sites with
critical values <span class="math inline">\(TV_d&gt;0.926\)</span> are
considered DMPs. Notice that, it is always possible to use any other
values of <span class="math inline">\(H\)</span> and <span class="math inline">\(TV_d\)</span> as critical values, but whatever
could be the value it will affect the final accuracy of the
classification performance of DMPs into two groups, DMPs from control
and DMPs from treatment (see below). So, it is important to do an good
choices of the critical values.</p>
<p>Function <a href="https://is.gd/W7dlP0" class="external-link">gofReport</a> search for the
best fitted model between the set of models requested by the user.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Weibull2P"</span>, <span class="st">"Weibull3P"</span>, <span class="st">"Gamma2P"</span>, <span class="st">"Gamma3P"</span>, <span class="st">"GGamma3P"</span><span class="op">)</span></span>
<span><span class="va">gof</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gofReport.html">gofReport</a></span><span class="op">(</span>HD <span class="op">=</span> <span class="va">DIVs</span>, column <span class="op">=</span> <span class="fl">9L</span>, model <span class="op">=</span> <span class="va">d</span>, num.cores <span class="op">=</span> <span class="fl">4L</span>,</span>
<span>            tasks <span class="op">=</span> <span class="fl">2L</span>, output <span class="op">=</span> <span class="st">"all"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">gof</span><span class="op">$</span><span class="va">bestModel</span></span></code></pre></div>
<p>There is not a one-to-one mapping between <span class="math inline">\(TV_d\)</span> and <span class="math inline">\(H\)</span>. However, at each cytosine site <span class="math inline">\(i\)</span>, these information divergences hold the
inequality: <span class="math display">\[TV_d(p^{tt}_i,p^{ct}_i)\leq
\sqrt{2}H_d(p^{tt}_i,p^{ct}_i)\]</span></p>
<p>where <span class="math inline">\(H_d(p^{tt}_i,p^{ct}_i)=\sqrt{\frac{H(p^{tt}_i,p^{ct}_i)}w}\)</span>
is the Hellinger distance and <span class="math inline">\(H(p^{tt}_i,p^{ct}_i)\)</span> is the Hellinger
divergence equation given above.</p>
</div>
<div class="section level4">
<h4 id="potential-methylation-signal">
<em>Potential methylation signal</em><a class="anchor" aria-label="anchor" href="#potential-methylation-signal"></a>
</h4>
<p>Methylation regulatory signals do not hold the theoretical
distribution and, consequently, for a given level of significance <span class="math inline">\(\alpha\)</span> (Type I error probability,
e.g. <span class="math inline">\(\alpha = 0.05\)</span>), cytosine
positions <span class="math inline">\(k\)</span> with information
divergence <span class="math inline">\(H_k &gt;= H_{\alpha =
0.05}\)</span> can be selected as sites carrying potential signals <span class="math inline">\(PS\)</span>. The <span class="math inline">\(PS\)</span> is detected with the function <a href="https://is.gd/yGORtg" class="external-link">getPotentialDIMP</a>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PS</span> <span class="op">=</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">DIVs</span>, dist.name <span class="op">=</span> <span class="va">gof</span><span class="op">$</span><span class="va">bestModel</span>, nlms <span class="op">=</span> <span class="va">gof</span><span class="op">$</span><span class="va">nlms</span>,</span>
<span>                    div.col <span class="op">=</span> <span class="fl">9</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, tv.col <span class="op">=</span> <span class="fl">8</span>, tv.cut <span class="op">=</span>  <span class="fl">0.674</span><span class="op">)</span></span></code></pre></div>
<p>Observe that we have used the <span class="math inline">\(|TV|\)</span> values based on the Bayesian
estimation of methylation levels (bay.TV, tv.col = 8).</p>
</div>
<div class="section level4">
<h4 id="potential-dmps-detected-with-fishers-exact-test-ft">
<em>Potential DMPs detected with Fisher’s exact test (FT)</em><a class="anchor" aria-label="anchor" href="#potential-dmps-detected-with-fishers-exact-test-ft"></a>
</h4>
<p>Fisher’s exact test (FT) is run here as standard alternative
approach. A full discussion on the performance of FT and other
statistical approaches in comparison with Methyl-IT pipeline can be read
in <span class="citation">(7)</span>. Notice that the standard
application of FT does not include a signal detection step (see more at
<span class="citation">(7)</span>).</p>
<p>In Methyl-IT, FT is implemented in function <em>FisherTest</em>. In
the current case, a pairwise group application of FT to each cytosine
site is performed. The differences between the group means of read
counts of methylated and unmethylated cytosines at each site are used
for testing (<span class="math inline">\(pooling.stat
="mean"\)</span>). Notice that only cytosine sites with
critical values <span class="math inline">\(TV_d&gt;0.67\)</span> are
tested (<em>tv.cut = 0.67</em>).</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ft</span> <span class="op">=</span> <span class="fu"><a href="../reference/FisherTest.html">FisherTest</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">divs</span>, tv.cut <span class="op">=</span> <span class="fl">0.67</span>,</span>
<span>                    pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,  pooling.stat <span class="op">=</span> <span class="st">"mean"</span>, </span>
<span>                    pvalCutOff <span class="op">=</span> <span class="fl">0.05</span>, num.cores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html" class="external-link">multicoreWorkers</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                    verbose <span class="op">=</span> <span class="cn">FALSE</span>, saveAll <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># TV column is the standard TV value (without Bayesian correction of methylation</span></span>
<span><span class="co"># levels)</span></span>
<span><span class="va">ft.tv</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">ft</span>, div.col <span class="op">=</span> <span class="fl">9L</span>, dist.name <span class="op">=</span> <span class="st">"None"</span>,</span>
<span>                        tv.cut <span class="op">=</span> <span class="fl">0.67</span>, tv.col <span class="op">=</span> <span class="fl">7</span>, alpha <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
<p>The above setting would impose additional constrains on the output of
DMPs resulting from Fisher’s exact test depending on how strong is the
methylation background noise in the control population. Basically,
methylation variations that can spontaneously occur in the control
population with relatively high frequencies are disregarded. The
decisions are based on the empirical cumulative distribution function
(ECDF).</p>
<p>Potential DMPs detected with FT can be constrained with the critical
values derived for the best fitted probability distribution models</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Potential DMPs from Fisher's exact test </span></span>
<span><span class="va">ft.hd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">ft</span>, div.col <span class="op">=</span> <span class="fl">9L</span>, nlms <span class="op">=</span> <span class="va">gof</span><span class="op">$</span><span class="va">nlms</span>, </span>
<span>                        tv.cut <span class="op">=</span> <span class="fl">0.674</span>, tv.col <span class="op">=</span> <span class="fl">8</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, </span>
<span>                        dist.name <span class="op">=</span> <span class="va">gof</span><span class="op">$</span><span class="va">bestModel</span><span class="op">)</span></span>
<span><span class="va">ft.hd</span><span class="op">$</span><span class="va">T1</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 2226 ranges and 12 metadata columns:</span></span>
<span><span class="co">##          seqnames    ranges strand |        c1        t1        c2        t2</span></span>
<span><span class="co">##             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##      [1]        1        29      * |         6       110       112         4</span></span>
<span><span class="co">##      [2]        1        42      * |         0       112       112         0</span></span>
<span><span class="co">##      [3]        1        70      * |         0        84        84         0</span></span>
<span><span class="co">##      [4]        1       103      * |         0        94        91         3</span></span>
<span><span class="co">##      [5]        1       105      * |         0       168       118        50</span></span>
<span><span class="co">##      ...      ...       ...    ... .       ...       ...       ...       ...</span></span>
<span><span class="co">##   [2222]        1     49928      * |        10       148       158         0</span></span>
<span><span class="co">##   [2223]        1     49974      * |        19       130       147         2</span></span>
<span><span class="co">##   [2224]        1     49980      * |        45       139       184         0</span></span>
<span><span class="co">##   [2225]        1     49981      * |        34       143       177         0</span></span>
<span><span class="co">##   [2226]        1     49996      * |        20       196       216         0</span></span>
<span><span class="co">##                  p1        p2        TV    bay.TV      hdiv       pvalue</span></span>
<span><span class="co">##           &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;    &lt;numeric&gt;</span></span>
<span><span class="co">##      [1] 0.05779995  0.961168  0.913793  0.903368   134.086  1.22001e-52</span></span>
<span><span class="co">##      [2] 0.00845858  0.995244  1.000000  0.986785   189.744  1.39309e-66</span></span>
<span><span class="co">##      [3] 0.01112222  0.993674  1.000000  0.982551   138.682  8.69664e-50</span></span>
<span><span class="co">##      [4] 0.00999781  0.962704  0.968085  0.952706   134.850  1.29337e-50</span></span>
<span><span class="co">##      [5] 0.00571921  0.700659  0.702381  0.694940   132.207  2.02668e-50</span></span>
<span><span class="co">##      ...        ...       ...       ...       ...       ...          ...</span></span>
<span><span class="co">##   [2222]  0.0674516  0.996621  0.936709  0.929170   217.700  1.25471e-78</span></span>
<span><span class="co">##   [2223]  0.1298691  0.983069  0.859060  0.853200   156.395  3.15691e-60</span></span>
<span><span class="co">##   [2224]  0.2434358  0.997096  0.755435  0.753661   170.369  1.01280e-61</span></span>
<span><span class="co">##   [2225]  0.1923356  0.996982  0.807910  0.804647   182.532  2.77686e-66</span></span>
<span><span class="co">##   [2226]  0.0950087  0.997525  0.907407  0.902516   279.852 2.42409e-100</span></span>
<span><span class="co">##             adj.pval      wprob</span></span>
<span><span class="co">##            &lt;numeric&gt;  &lt;numeric&gt;</span></span>
<span><span class="co">##      [1] 4.86816e-52  0.0480895</span></span>
<span><span class="co">##      [2] 1.07813e-65  0.0215268</span></span>
<span><span class="co">##      [3] 3.03410e-49  0.0448150</span></span>
<span><span class="co">##      [4] 4.68846e-50  0.0475260</span></span>
<span><span class="co">##      [5] 7.31951e-50  0.0495093</span></span>
<span><span class="co">##      ...         ...        ...</span></span>
<span><span class="co">##   [2222] 1.67316e-77 0.01487102</span></span>
<span><span class="co">##   [2223] 1.80286e-59 0.03441153</span></span>
<span><span class="co">##   [2224] 6.21624e-61 0.02815467</span></span>
<span><span class="co">##   [2225] 2.11753e-65 0.02375878</span></span>
<span><span class="co">##   [2226] 9.74758e-99 0.00693179</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre>
<p>The above result shows column ‘adj.pval’ with the FT’s p.values
adjusted for multiple comparisons and column ‘wprob’ with the predicted
probabilities: <span class="math inline">\(1 - P(H_k &gt;= H_{\alpha =
0.05})\)</span> from the best fitted model.</p>
<p>Summary table:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>ft <span class="op">=</span> <span class="fu"><a href="../reference/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">ft.tv</span>, <span class="va">length</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            ft.hd <span class="op">=</span> <span class="fu"><a href="../reference/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">ft.hd</span>, <span class="va">length</span><span class="op">)</span><span class="op">)</span>,</span>
<span>BestModel <span class="op">=</span> <span class="fu"><a href="../reference/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">PS</span>, <span class="va">length</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      ft ft.hd BestModel</span></span>
<span><span class="co">## C1 1865  1043      1043</span></span>
<span><span class="co">## C2 1824  1075      1075</span></span>
<span><span class="co">## C3 1765  1039      1039</span></span>
<span><span class="co">## T1 7800  2226      2226</span></span>
<span><span class="co">## T2 7847  2167      2167</span></span>
<span><span class="co">## T3 7879  2189      2189</span></span></code></pre>
<p>Notice that there are significant differences between the numbers of
PSs detected by FT and other approaches. There at least about 7800 -
2226 = 5574 or more PSs in the treatment samples that are not induced by
the treatment, which can naturally occurs in both groups, control and
treatment. In other words, the best probabilistic distribution models
suggest that those methylation events are resultant of the normal
(non-stressful) biological processes. Any information divergence of
methylation level below the critical value <span class="math inline">\(TV_{\alpha=0.05}\)</span> or <span class="math inline">\(H_{\alpha=0.05}\)</span> is a methylation event
that can occur in normal conditions. <em>It does not matter how much
significant a test like FT could be, since it is not about the
signification of the test, but about how big is the probability to
observe that methylation event in the control population as
well</em>.</p>
</div>
<div class="section level4">
<h4 id="critical-values-based-on-fitted-models">Critical values based on fitted models<a class="anchor" aria-label="anchor" href="#critical-values-based-on-fitted-models"></a>
</h4>
<p>Critical values <span class="math inline">\(H_{\alpha=0.05}\)</span>
can be estimated from each fitted model using <a href="https://is.gd/CCqNYS" class="external-link">predict</a> function:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ==== 95% quantiles </span></span>
<span><span class="va">q95</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>q95 <span class="op">=</span> <span class="fu"><a href="../reference/predict.html">predict</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">gof</span><span class="op">$</span><span class="va">nlms</span>, pred <span class="op">=</span> <span class="st">"quant"</span>, q <span class="op">=</span> <span class="fl">0.95</span>,</span>
<span>                                dist.name <span class="op">=</span> <span class="va">gof</span><span class="op">$</span><span class="va">bestModel</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">q95</span></span></code></pre></div>
<pre><code><span><span class="co">##          q95</span></span>
<span><span class="co">## C1  59.08915</span></span>
<span><span class="co">## C2  58.32001</span></span>
<span><span class="co">## C3  58.32643</span></span>
<span><span class="co">## T1 131.57161</span></span>
<span><span class="co">## T2 137.55786</span></span>
<span><span class="co">## T3 136.17154</span></span></code></pre>
</div>
</div>
</div>
<div class="section level2">
<h2 id="cutpoint-for-the-spontaneous-variability-in-the-control-group">Cutpoint for the spontaneous variability in the control group<a class="anchor" aria-label="anchor" href="#cutpoint-for-the-spontaneous-variability-in-the-control-group"></a>
</h2>
<p>Normally, there is a spontaneous variability in the control group.
This is a consequence of the random fluctuations, or noise, inherent to
the methylation process. The stochasticity of the the methylation
process is derives from the stochasticity inherent in biochemical
processes. There are fundamental physical reasons to acknowledge that
biochemical processes are subject to noise and fluctuations <span class="citation">(8, 9)</span>. So, regardless constant environment,
statistically significant methylation changes can be found in control
population with probability greater than zero and proportional to a
Boltzmann factor <span class="citation">(6)</span>.</p>
<p>Natural signals and those generated by human technology are not free
of noise and, as mentioned above, the methylation signal is no
exception. Only signal detection based approaches are designed to filter
out the signal from the noise, in natural and in human generated
signals.</p>
<p>The need for the application of (what is now known as) signal
detection in cancer research was pointed out by Youden in the midst of
the last century <span class="citation">(10)</span>. Here, the
application of signal detection approach was performed according with
the standard practice in current implementations of clinical diagnostic
test <span class="citation">(11–13)</span>. That is, optimal cutoff
values of the methylation signal were estimated on the receiver
operating characteristic curves (ROCs) and applied to identify DMPs. The
decision of whether a DMP detected by Fisher’s exact test (or any other
statistical test implemented in Methyl-IT) is taken based on the optimal
cutoff value.</p>
<p>In the current example, the column carrying <em>TV</em>
(<em>div.col</em> = 7L) will be used to estimate the cutpoint. The
column values will be expressed in terms of <span class="math inline">\(TV_d=|p_{tt}-p_{ct}|\)</span> (<em>absolute</em> =
TRUE):</p>
<div class="section level4">
<h4 id="cutpoint-for-ps-estimated-based-on-the-best-probability-distribution-model">Cutpoint for PS estimated based on the best probability distribution
model<a class="anchor" aria-label="anchor" href="#cutpoint-for-ps-estimated-based-on-the-best-probability-distribution-model"></a>
</h4>
<p>Next, the search for an optimal cutpoint is accomplished with a model
classifier and using the total variation distance (with Bayesian
correction, bay.TV)</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">PS</span>, </span>
<span>                            control.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"C1"</span>,<span class="st">"C2"</span>, <span class="st">"C3"</span><span class="op">)</span>, </span>
<span>                            treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T1"</span>, <span class="st">"T2"</span>, <span class="st">"T3"</span><span class="op">)</span>,</span>
<span>                            simple <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                            classifier1 <span class="op">=</span> <span class="st">"pca.logistic"</span>, </span>
<span>                            classifier2 <span class="op">=</span> <span class="st">"pca.qda"</span>, </span>
<span>                            column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                        wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                            n.pc <span class="op">=</span> <span class="fl">4</span> , center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            div.col <span class="op">=</span> <span class="fl">9</span>, clas.perf <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Found more than one class "pDMP" in cache; using the first, from namespace 'MethylIT'</span></span></code></pre>
<pre><code><span><span class="co">## Also defined by 'MethylIT.utils'</span></span></code></pre>
<pre><code><span><span class="co">## Found more than one class "pDMP" in cache; using the first, from namespace 'MethylIT'</span></span></code></pre>
<pre><code><span><span class="co">## Also defined by 'MethylIT.utils'</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint</span><span class="op">$</span><span class="va">cutpoint</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 131.5809</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint</span><span class="op">$</span><span class="va">testSetPerformance</span></span></code></pre></div>
<pre><code><span><span class="co">## Confusion Matrix and Statistics</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##           Reference</span></span>
<span><span class="co">## Prediction   CT   TT</span></span>
<span><span class="co">##         CT  198    0</span></span>
<span><span class="co">##         TT    0 2633</span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##                Accuracy : 1          </span></span>
<span><span class="co">##                  95% CI : (0.9987, 1)</span></span>
<span><span class="co">##     No Information Rate : 0.9301     </span></span>
<span><span class="co">##     P-Value [Acc &gt; NIR] : &lt; 2.2e-16  </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##                   Kappa : 1          </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##  Mcnemar's Test P-Value : NA         </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##             Sensitivity : 1.0000     </span></span>
<span><span class="co">##             Specificity : 1.0000     </span></span>
<span><span class="co">##          Pos Pred Value : 1.0000     </span></span>
<span><span class="co">##          Neg Pred Value : 1.0000     </span></span>
<span><span class="co">##              Prevalence : 0.9301     </span></span>
<span><span class="co">##          Detection Rate : 0.9301     </span></span>
<span><span class="co">##    Detection Prevalence : 0.9301     </span></span>
<span><span class="co">##       Balanced Accuracy : 1.0000     </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##        'Positive' Class : TT         </span></span>
<span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint</span><span class="op">$</span><span class="va">testSetModel.FDR</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="cutpoint-for-ps-estimated-based-on-ft">Cutpoint for PS estimated based on FT<a class="anchor" aria-label="anchor" href="#cutpoint-for-ps-estimated-based-on-ft"></a>
</h4>
<p>The best classification performance based PS estimated based on FT
are obtained with the application of <a href="https://is.gd/t8UKvK" class="external-link">Youden index</a> <span class="citation">(10)</span>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint2</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">ft.tv</span>, </span>
<span>                            control.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"C1"</span>,<span class="st">"C2"</span>, <span class="st">"C3"</span><span class="op">)</span>, </span>
<span>                            treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T1"</span>, <span class="st">"T2"</span>, <span class="st">"T3"</span><span class="op">)</span>,</span>
<span>                            simple <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            classifier1 <span class="op">=</span> <span class="st">"pca.qda"</span>, </span>
<span>                            column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                        wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                            n.pc <span class="op">=</span> <span class="fl">4</span> , center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            div.col <span class="op">=</span> <span class="fl">9</span>, clas.perf <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            post.cut <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span></span>
<span><span class="va">cutpoint2</span><span class="op">$</span><span class="va">cutpoint</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 99.94609</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint2</span><span class="op">$</span><span class="va">testSetPerformance</span></span></code></pre></div>
<pre><code><span><span class="co">## Confusion Matrix and Statistics</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##           Reference</span></span>
<span><span class="co">## Prediction   CT   TT</span></span>
<span><span class="co">##         CT   11   38</span></span>
<span><span class="co">##         TT  474 4332</span></span>
<span><span class="co">##                                          </span></span>
<span><span class="co">##                Accuracy : 0.8945         </span></span>
<span><span class="co">##                  95% CI : (0.8856, 0.903)</span></span>
<span><span class="co">##     No Information Rate : 0.9001         </span></span>
<span><span class="co">##     P-Value [Acc &gt; NIR] : 0.9052         </span></span>
<span><span class="co">##                                          </span></span>
<span><span class="co">##                   Kappa : 0.0233         </span></span>
<span><span class="co">##                                          </span></span>
<span><span class="co">##  Mcnemar's Test P-Value : &lt;2e-16         </span></span>
<span><span class="co">##                                          </span></span>
<span><span class="co">##             Sensitivity : 0.99130        </span></span>
<span><span class="co">##             Specificity : 0.02268        </span></span>
<span><span class="co">##          Pos Pred Value : 0.90137        </span></span>
<span><span class="co">##          Neg Pred Value : 0.22449        </span></span>
<span><span class="co">##              Prevalence : 0.90010        </span></span>
<span><span class="co">##          Detection Rate : 0.89228        </span></span>
<span><span class="co">##    Detection Prevalence : 0.98991        </span></span>
<span><span class="co">##       Balanced Accuracy : 0.50699        </span></span>
<span><span class="co">##                                          </span></span>
<span><span class="co">##        'Positive' Class : TT             </span></span>
<span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint2</span><span class="op">$</span><span class="va">testSetModel.FDR</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.008695652</span></span></code></pre>
<p>Addition of information on the probability distribution of the
methylation signal to potential signal identified with FT will emulate
the results obtained with the Methyl-IT pipeline.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint3</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">ft.hd</span>, </span>
<span>                            control.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"C1"</span>,<span class="st">"C2"</span>, <span class="st">"C3"</span><span class="op">)</span>, </span>
<span>                            treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T1"</span>, <span class="st">"T2"</span>, <span class="st">"T3"</span><span class="op">)</span>,</span>
<span>                            simple <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                            classifier1 <span class="op">=</span> <span class="st">"pca.logistic"</span>, </span>
<span>                            classifier2 <span class="op">=</span> <span class="st">"pca.qda"</span>, </span>
<span>                            column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                        wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                            n.pc <span class="op">=</span> <span class="fl">4</span> , center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            div.col <span class="op">=</span> <span class="fl">9</span>, clas.perf <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">cutpoint3</span><span class="op">$</span><span class="va">cutpoint</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 131.5809</span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint3</span><span class="op">$</span><span class="va">testSetPerformance</span></span></code></pre></div>
<pre><code><span><span class="co">## Confusion Matrix and Statistics</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##           Reference</span></span>
<span><span class="co">## Prediction   CT   TT</span></span>
<span><span class="co">##         CT  198    0</span></span>
<span><span class="co">##         TT    0 2633</span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##                Accuracy : 1          </span></span>
<span><span class="co">##                  95% CI : (0.9987, 1)</span></span>
<span><span class="co">##     No Information Rate : 0.9301     </span></span>
<span><span class="co">##     P-Value [Acc &gt; NIR] : &lt; 2.2e-16  </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##                   Kappa : 1          </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##  Mcnemar's Test P-Value : NA         </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##             Sensitivity : 1.0000     </span></span>
<span><span class="co">##             Specificity : 1.0000     </span></span>
<span><span class="co">##          Pos Pred Value : 1.0000     </span></span>
<span><span class="co">##          Neg Pred Value : 1.0000     </span></span>
<span><span class="co">##              Prevalence : 0.9301     </span></span>
<span><span class="co">##          Detection Rate : 0.9301     </span></span>
<span><span class="co">##    Detection Prevalence : 0.9301     </span></span>
<span><span class="co">##       Balanced Accuracy : 1.0000     </span></span>
<span><span class="co">##                                      </span></span>
<span><span class="co">##        'Positive' Class : TT         </span></span>
<span><span class="co">## </span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cutpoint3</span><span class="op">$</span><span class="va">testSetModel.FDR</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0</span></span></code></pre>
<p>Therefore, invoking the parsimony principle, we assume that signal
detection and machine-learning classifiers are sufficient <span class="citation">(14)</span>.</p>
</div>
<div class="section level3">
<h3 id="dmps">DMPs<a class="anchor" aria-label="anchor" href="#dmps"></a>
</h3>
<p>Cytosine sites carrying a methylation signal are designated
differentially informative methylated positions (DMPs). With high
probability true DMPs can be selected with Methyl-IT function <a href="https://is.gd/7cVEEd" class="external-link">selectDIMP</a>.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DMPs</span> <span class="op">=</span> <span class="fu"><a href="../reference/selectDIMP.html">selectDIMP</a></span><span class="op">(</span><span class="va">PS</span>, div.col <span class="op">=</span> <span class="fl">9</span>, cutpoint <span class="op">=</span> <span class="va">cutpoint</span><span class="op">$</span><span class="va">cutpoint</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="evaluation-of-dmp-classification">Evaluation of DMP classification<a class="anchor" aria-label="anchor" href="#evaluation-of-dmp-classification"></a>
</h4>
<p>As shown above, DMPs are found in the control population as well.
Hence, it is important to know whether a DMP is the resulting effect of
the treatment or just spontaneously occurred in the control sample as
well. In particular, the confrontation of this issue is extremely
important when methylation analysis is intended to be used as part of a
diagnostic clinical test and a decision making in biotechnology
industry.</p>
<p>Methyl-IT function <a href="https://is.gd/nzy9j3" class="external-link">evaluateDIMPclass</a> is used here to
evaluate the classification of DMPs into one of the two classes, control
and treatment. Several classifiers are available to be used with this
function (see the help/manual for this function or type
?evaluateDIMPclass in command line).</p>
<p>A first evaluation of DMP classification performance was already done
after setting “clas.perf = TRUE” in function <a href="https://is.gd/RR9iWY" class="external-link">estimateCutPoint</a>. However, this
performance derives just from randomly split the set of sample into two
subsets: training (60%) and testing (40%). A good classification
performance report would be just the result from a lucky split of the
set of samples! Hence, what would happen if the random split and further
analysis are repeated 300 times?</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">class.perf</span> <span class="op">=</span> <span class="fu"><a href="../reference/evaluateDIMPclass.html">evaluateDIMPclass</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">DMPs</span>, control.names <span class="op">=</span> <span class="va">control.nam</span>,</span>
<span>                            treatment.names <span class="op">=</span> <span class="va">treatment.nam</span>,</span>
<span>                            column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, TV <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                        wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                            classifier <span class="op">=</span> <span class="st">"pca.qda"</span>, n.pc <span class="op">=</span> <span class="fl">4</span>, pval.col <span class="op">=</span> <span class="fl">10L</span>, </span>
<span>                            center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>, num.boot <span class="op">=</span> <span class="fl">300</span>,</span>
<span>                            output <span class="op">=</span> <span class="st">"mc.val"</span>, prop <span class="op">=</span> <span class="fl">0.6</span>, num.cores <span class="op">=</span> <span class="fl">4L</span>,</span>
<span>                            tasks <span class="op">=</span> <span class="fl">2L</span></span>
<span><span class="op">)</span></span>
<span><span class="va">class.perf</span></span></code></pre></div>
<pre><code><span><span class="co">##     Accuracy     Kappa   AccuracyLower    AccuracyUpper  AccuracyNull   </span></span>
<span><span class="co">##  Min.   :1   Min.   :1   Min.   :0.9987   Min.   :1     Min.   :0.9301  </span></span>
<span><span class="co">##  1st Qu.:1   1st Qu.:1   1st Qu.:0.9987   1st Qu.:1     1st Qu.:0.9301  </span></span>
<span><span class="co">##  Median :1   Median :1   Median :0.9987   Median :1     Median :0.9301  </span></span>
<span><span class="co">##  Mean   :1   Mean   :1   Mean   :0.9987   Mean   :1     Mean   :0.9301  </span></span>
<span><span class="co">##  3rd Qu.:1   3rd Qu.:1   3rd Qu.:0.9987   3rd Qu.:1     3rd Qu.:0.9301  </span></span>
<span><span class="co">##  Max.   :1   Max.   :1   Max.   :0.9987   Max.   :1     Max.   :0.9301  </span></span>
<span><span class="co">##                                                                         </span></span>
<span><span class="co">##  AccuracyPValue      McnemarPValue  Sensitivity  Specificity Pos Pred Value</span></span>
<span><span class="co">##  Min.   :7.155e-90   Min.   : NA   Min.   :1    Min.   :1    Min.   :1     </span></span>
<span><span class="co">##  1st Qu.:7.155e-90   1st Qu.: NA   1st Qu.:1    1st Qu.:1    1st Qu.:1     </span></span>
<span><span class="co">##  Median :7.155e-90   Median : NA   Median :1    Median :1    Median :1     </span></span>
<span><span class="co">##  Mean   :7.155e-90   Mean   :NaN   Mean   :1    Mean   :1    Mean   :1     </span></span>
<span><span class="co">##  3rd Qu.:7.155e-90   3rd Qu.: NA   3rd Qu.:1    3rd Qu.:1    3rd Qu.:1     </span></span>
<span><span class="co">##  Max.   :7.155e-90   Max.   : NA   Max.   :1    Max.   :1    Max.   :1     </span></span>
<span><span class="co">##                      NA's   :300                                           </span></span>
<span><span class="co">##  Neg Pred Value   Precision     Recall        F1      Prevalence    </span></span>
<span><span class="co">##  Min.   :1      Min.   :1   Min.   :1   Min.   :1   Min.   :0.9301  </span></span>
<span><span class="co">##  1st Qu.:1      1st Qu.:1   1st Qu.:1   1st Qu.:1   1st Qu.:0.9301  </span></span>
<span><span class="co">##  Median :1      Median :1   Median :1   Median :1   Median :0.9301  </span></span>
<span><span class="co">##  Mean   :1      Mean   :1   Mean   :1   Mean   :1   Mean   :0.9301  </span></span>
<span><span class="co">##  3rd Qu.:1      3rd Qu.:1   3rd Qu.:1   3rd Qu.:1   3rd Qu.:0.9301  </span></span>
<span><span class="co">##  Max.   :1      Max.   :1   Max.   :1   Max.   :1   Max.   :0.9301  </span></span>
<span><span class="co">##                                                                     </span></span>
<span><span class="co">##  Detection Rate   Detection Prevalence Balanced Accuracy</span></span>
<span><span class="co">##  Min.   :0.9301   Min.   :0.9301       Min.   :1        </span></span>
<span><span class="co">##  1st Qu.:0.9301   1st Qu.:0.9301       1st Qu.:1        </span></span>
<span><span class="co">##  Median :0.9301   Median :0.9301       Median :1        </span></span>
<span><span class="co">##  Mean   :0.9301   Mean   :0.9301       Mean   :1        </span></span>
<span><span class="co">##  3rd Qu.:0.9301   3rd Qu.:0.9301       3rd Qu.:1        </span></span>
<span><span class="co">##  Max.   :0.9301   Max.   :0.9301       Max.   :1        </span></span>
<span><span class="co">## </span></span></code></pre>
</div>
</div>
</div>
<div class="section level2">
<h2 id="conclusions-summary">Conclusions Summary<a class="anchor" aria-label="anchor" href="#conclusions-summary"></a>
</h2>
<p>Herein, an illustrative example of methylation analysis with
Methyl-IT have been presented. Whatever could be the statistical
test/approach used to identify DMPs, the analysis with simulated
datasets, where the average of methylation levels in the control samples
is relatively high, indicates the need for the application of signal
detection based approaches.</p>
<p>For the current simulated dataset, the best classification
performance was obtained for the approach of DMP detection based on the
best fitted probability distribution model for the Hellinger divergence
of methylation levels. DMPs from treatment are distinguish from control
DMPs with very high accuracy.</p>
<div class="section level3">
<h3 id="concluding-remarks">Concluding remarks<a class="anchor" aria-label="anchor" href="#concluding-remarks"></a>
</h3>
<p>The simplest suggested steps to follow for a methylation analysis
with Methyl-IT are:</p>
<ol style="list-style-type: decimal">
<li>To estimate a reference virtual sample from a reference group by
using function <a href="https://is.gd/qR79kW" class="external-link">poolFromGRlist</a>. Notice
that several statistics are available to estimate the virtual samples,
i.e., <em>mean</em>, <em>median</em>, <em>sum</em>. For experiments
limited by the number of sample, at least, try the estimation of the
virtual sample from the control group. Alternatively, the whole
reference group can be used in pairwise comparisons with control and
treatment groups (computationally expensive).</li>
</ol>
<ol start="2" style="list-style-type: decimal">
<li>To estimate information divergence using function <a href="https://is.gd/0ViBO9" class="external-link">estimateDivergence</a>
</li>
</ol>
<ol start="3" style="list-style-type: decimal">
<li>To perform the estimation of the best fitted probability
distribution model using function <a href="https://is.gd/W7dlP0" class="external-link">gofReport</a>.</li>
</ol>
<ol start="4" style="list-style-type: decimal">
<li>To get the potential DMPs using function <a href="https://is.gd/yGORtg" class="external-link">getPotentialDIMP</a>.</li>
</ol>
<ol start="5" style="list-style-type: decimal">
<li>To estimate the optimal cutpoint using function <a href="https://is.gd/RR9iWY" class="external-link">estimateCutPoint</a>.</li>
</ol>
<ol start="6" style="list-style-type: decimal">
<li>To retrieve DMPs with function <a href="https://is.gd/7cVEEd" class="external-link">selectDIMP</a>.</li>
</ol>
<ol start="7" style="list-style-type: decimal">
<li>To evaluate the classification performance using function <a href="https://is.gd/nzy9j3" class="external-link">evaluateDIMPclass</a>.</li>
</ol>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body">
<div id="ref-Hebestreit2013" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">K.
Hebestreit, M. Dugas, H.-U. Klein, <a href="https://doi.org/10.1093/bioinformatics/btt263" class="external-link"><span class="nocase">Detection of significantly differentially methylated
regions in targeted bisulfite sequencing data.</span></a>
<em>Bioinformatics (Oxford, England)</em> <strong>29</strong>, 1647–53
(2013).</div>
</div>
<div id="ref-Robinson2014" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">M.
D. Robinson, <em>et al.</em>, <a href="https://doi.org/10.3389/fgene.2014.00324" class="external-link"><span class="nocase">Statistical methods for detecting differentially
methylated loci and regions.</span></a> <em>Frontiers in genetics</em>
<strong>5</strong>, 324 (2014).</div>
</div>
<div id="ref-Dolzhenko2014" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">E.
Dolzhenko, A. D. Smith, <a href="https://doi.org/10.1186/1471-2105-15-215" class="external-link"><span class="nocase">Using beta-binomial regression for high-precision
differential methylation analysis in multifactor whole-genome bisulfite
sequencing experiments</span></a>. <em>BMC bioinformatics</em>
<strong>15</strong>, 215 (2014).</div>
</div>
<div id="ref-Baldi2001" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">P.
Baldi, S. Brunak, <em><span class="nocase">Bioinformatics: the machine
learning approach</span></em>, Second (MIT Press, 2001).</div>
</div>
<div id="ref-Basu2010" class="csl-entry">
<div class="csl-left-margin">5. </div>
<div class="csl-right-inline">A.
Basu, A. Mandal, L. Pardo, <a href="https://doi.org/10.1016/j.spl.2009.10.008" class="external-link"><span class="nocase">Hypothesis testing for two discrete populations based on
the Hellinger distance</span></a>. <em>Statistics <span>&amp;</span>
Probability Letters</em> <strong>80</strong>, 206–214 (2010).</div>
</div>
<div id="ref-Sanchez2016" class="csl-entry">
<div class="csl-left-margin">6. </div>
<div class="csl-right-inline">R.
Sanchez, S. A. Mackenzie, <a href="https://doi.org/10.1371/journal.pone.0150427" class="external-link"><span class="nocase">Information Thermodynamics of Cytosine DNA
Methylation</span></a>. <em>PLOS ONE</em> <strong>11</strong>, e0150427
(2016).</div>
</div>
<div id="ref-Sanchez2019" class="csl-entry">
<div class="csl-left-margin">7. </div>
<div class="csl-right-inline">R.
Sanchez, X. Yang, T. Maher, S. Mackenzie, <a href="https://doi.org/10.3390/ijms20215343" class="external-link"><span class="nocase">Discrimination of DNA Methylation Signal from Background
Variation for Clinical Diagnostics</span></a>. <em>Int. J. Mol.
Sci.</em> <strong>20</strong>, 5343 (2019).</div>
</div>
<div id="ref-Samoilov2006" class="csl-entry">
<div class="csl-left-margin">8. </div>
<div class="csl-right-inline">M.
S. Samoilov, G. Price, A. P. Arkin, <a href="https://doi.org/10.1126/stke.3662006re17" class="external-link"><span class="nocase">From fluctuations to phenotypes: the physiology of
noise.</span></a> <em>Science’s STKE : signal transduction knowledge
environment</em> <strong>2006</strong> (2006).</div>
</div>
<div id="ref-Eldar2010" class="csl-entry">
<div class="csl-left-margin">9. </div>
<div class="csl-right-inline">A.
Eldar, M. B. Elowitz, <a href="https://doi.org/10.1038/nature09326" class="external-link"><span class="nocase">Functional roles for noise in genetic
circuits</span></a>. <strong>467</strong>, 167–173 (2010).</div>
</div>
<div id="ref-Youden1950" class="csl-entry">
<div class="csl-left-margin">10. </div>
<div class="csl-right-inline">W.
J. Youden, <a href="https://doi.org/10.1002/1097-0142(1950)3:1&lt;32::AID-CNCR2820030106&gt;3.0.CO;2-3" class="external-link"><span class="nocase">Index for rating diagnostic tests</span></a>.
<em>Cancer</em> <strong>3</strong>, 32–35 (1950).</div>
</div>
<div id="ref-Carter2016" class="csl-entry">
<div class="csl-left-margin">11. </div>
<div class="csl-right-inline">J.
V. Carter, J. Pan, S. N. Rai, S. Galandiuk, <a href="https://doi.org/10.1016/j.surg.2015.12.029" class="external-link"><span class="nocase">ROC-ing along: Evaluation and interpretation of receiver
operating characteristic curves</span></a>. <em>Surgery</em>
<strong>159</strong>, 1638–1645 (2016).</div>
</div>
<div id="ref-lopez2014" class="csl-entry">
<div class="csl-left-margin">12. </div>
<div class="csl-right-inline">M.
López-Ratón, M. X. Rodríguez-Álvarez, C. Cadarso-Suárez, F.
Gude-Sampedro, Others, <a href="https://www.jstatsoft.org/article/view/v061i08" class="external-link"><span class="nocase">OptimalCutpoints: an R package for selecting optimal
cutpoints in diagnostic tests</span></a>. <em>Journal of statistical
software</em> <strong>61</strong>, 1–36 (2014).</div>
</div>
<div id="ref-Hippenstiel2001" class="csl-entry">
<div class="csl-left-margin">13. </div>
<div class="csl-right-inline">R.
D. Hippenstiel, <em><span class="nocase">Detection theory: applications
and digital signal processing</span></em> (CRC Press, 2001).</div>
</div>
<div id="ref-ElNaqa2018" class="csl-entry">
<div class="csl-left-margin">14. </div>
<div class="csl-right-inline">I.
El Naqa, <em>et al.</em>, <a href="https://doi.org/10.1002/mp.12811" class="external-link"><span class="nocase">Machine
learning and modeling: Data, validation, communication
challenges</span></a>. <em>Medical Physics</em> <strong>45</strong>,
e834–e840 (2018).</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
