<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Methylation analysis with Methyl-IT • MethylIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Methylation analysis with Methyl-IT">
<meta property="og:description" content="MethylIT">
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/MethylIT/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Methylation analysis with Methyl-IT</h1>
            <h3 class="subtitle"><h1>
An example with simulated datasets. A guide for users</h1></h3>
                        <h4 class="author"><div class="line-block">Robersy Sanchez<br><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a><br>
Mackenzie’s lab<br><br>
Department of Biology and Plant Science.<br>
Pennsylvania State University, University Park, PA 16802</div></h4>
            
            <h4 class="date">09 June 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/master/vignettes/Methylation_analysis_with_Methyl-IT.Rmd"><code>vignettes/Methylation_analysis_with_Methyl-IT.Rmd</code></a></small>
      <div class="hidden name"><code>Methylation_analysis_with_Methyl-IT.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Methylation analysis with <a href="https://github.com/genomaths/MethylIT">Methyl-IT</a> is illustrated on simulated datasets of methylated and unmethylated read counts with relatively high average of methylation levels: 0.15 and 0.286 for control and treatment groups, respectively. The main Methyl-IT downstream analysis is presented alongside the application of Fisher’s exact test. The importance of a signal detection step is shown.</p>
    </div>
    
<style type="text/css">

.list-group-item.active, .list-group-item.active:focus, 
.list-group-item.active:hover {
    z-index: 2;
    color: #fff;
    background-color: #337ab7;
    border-color: #337ab7;
}

.tocify-subheader .tocify-item {
    font-size: 0.80em;
    padding-left: 25px;
    text-indent: 0;
}

.tocify-header .tocify-item {
    font-size: 0.85em;
    padding-left: 20px;
    text-indent: 0;
}

body{ /* Normal  */
    font-size: 18px;
    font-family: "Times New Roman", Times, serif;
    text-align: justify
}
td {  /* Table  */
    font-size: 8px;
}

h1.title {
    font-size: 38px;
    font-family: "Times New Roman", Times, serif;
    color: DarkRed;
    .subTitle {
        font-size: 24px;
        font-family: "Times New Roman", Times, serif;
        color: DarkRed;
    }
}

h1 { /* Header 1 */
    font-size: 28px;
    font-family: "Times New Roman", Times, serif;
    color: DarkBlue;
}
h2 { /* Header 2 */
    font-size: 22px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
h3 { /* Header 3 */
    font-size: 18px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
code.r{ /* Code block */
    font-size: 12px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>
<div id="background" class="section level1">
<h1 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h1>
<p>Methyl-IT R package offers a methylome analysis approach based on information thermodynamics (IT) and signal detection. Methyl-IT approach confront detection of differentially methylated cytosine as a signal detection problem. This approach was designed to discriminate methylation regulatory signal from background noise induced by molecular stochastic fluctuations. Methyl-IT R package is not limited to the IT approach but also includes Fisher’s exact test (FT), Root-mean-square statistic (RMST) or Hellinger divergence (HDT) tests. Herein, we will show that a signal detection step is also required for FT, RMST, and HDT as well. It is worthy to notice that, as for any standard statistical analysis, any serious robust methylation analysis requires for a descriptive statistical analysis at the beginning or at different downstream steps. This is an ABC principle taught in any undergraduate course on statistics. Methylation analysis is not the exception of the rule. This detail is included in this example.</p>
<p>Note: This example was made with the MethylIT version at <a href="https://github.com/genomaths/MethylIT" class="uri">https://github.com/genomaths/MethylIT</a>.</p>
</div>
<div id="available-datasets-and-reading" class="section level1">
<h1 class="hasAnchor">
<a href="#available-datasets-and-reading" class="anchor"></a>Available datasets and reading</h1>
<p>Methylome datasets of whole-genome bisulfite sequencing (WGBS) are available at Gene Expression Omnibus (GEO DataSets). The data set are downloaded providing the GEO accession numbers for each data set to the function <em>getGEOSuppFiles</em> (for details type ?getGEOSuppFiles in the R console). Then, datasets can be read using function <em>readCounts2GRangesList</em>. An example on how to load datasets of read-counts of methylated and unmethylated cytosine into Methyl-IT is given in the Cancer example at <a href="https://git.psu.edu/genomath/MethylIT" class="uri">https://git.psu.edu/genomath/MethylIT</a></p>
</div>
<div id="data-generation" class="section level1">
<h1 class="hasAnchor">
<a href="#data-generation" class="anchor"></a>Data generation</h1>
<p>For the current example on methylation analysis with Methyl-IT we will use simulated data. Read-count matrices of methylated and unmethylated cytosine are generated with Methyl-IT function <a href="https://is.gd/gg0yTp">simulateCounts</a>.</p>
<p>Function <em>simulateCounts</em> randomly generates prior methylation levels using Beta distribution function. The expected mean of methylation levels that we would like to have can be estimated using the auxiliary function:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="no">bmean</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">alpha</span>, <span class="no">beta</span>) <span class="no">alpha</span>/(<span class="no">alpha</span> + <span class="no">beta</span>)
<span class="no">alpha.ct</span> <span class="kw">&lt;-</span> <span class="fl">0.09</span>
<span class="no">alpha.tt</span> <span class="kw">&lt;-</span> <span class="fl">0.2</span>
<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">control.group</span> <span class="kw">=</span> <span class="fu">bmean</span>(<span class="no">alpha.ct</span>, <span class="fl">0.5</span>), <span class="kw">treatment.group</span> <span class="kw">=</span> <span class="fu">bmean</span>(<span class="no">alpha.tt</span>, <span class="fl">0.5</span>),
    <span class="kw">mean.diff</span> <span class="kw">=</span> <span class="fu">bmean</span>(<span class="no">alpha.tt</span>, <span class="fl">0.5</span>) - <span class="fu">bmean</span>(<span class="no">alpha.ct</span>, <span class="fl">0.5</span>))</pre></body></html></div>
<pre><code>##   control.group treatment.group       mean.diff 
##       0.1525424       0.2857143       0.1331719</code></pre>
<p>This simple function uses the <span class="math inline">\(\alpha\)</span> (<em>shape1</em>) and <span class="math inline">\(\beta\)</span> (<em>shape2</em>) parameters from the Beta distribution function to compute the expected value of methylation levels. In the current case, we expect to have a difference of methylation levels about 0.133 between the control and the treatment.</p>
<div id="simulation" class="section level2">
<h2 class="hasAnchor">
<a href="#simulation" class="anchor"></a>Simulation</h2>
<p>Function <em>simulateCounts</em> from <a href="https://github.com/genomaths/MethylIT.utils">MethylIT.utils</a> R package will be used to generate the datasets, which will include three group of samples: reference, control, and treatment.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>({
    <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">MethylIT</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">MethylIT.utils</span>)}
)

<span class="co"># The number of cytosine sites to generate</span>
<span class="no">sites</span> <span class="kw">=</span> <span class="fl">50000</span>
<span class="co"># Set a seed for pseudo-random number generation</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">124</span>)
<span class="no">control.nam</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"C1"</span>, <span class="st">"C2"</span>, <span class="st">"C3"</span>)
<span class="no">treatment.nam</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"T1"</span>, <span class="st">"T2"</span>, <span class="st">"T3"</span>)

<span class="co"># Reference group </span>
<span class="no">ref0</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html">simulateCounts</a></span>(<span class="kw">num.samples</span> <span class="kw">=</span> <span class="fl">4</span>, <span class="kw">sites</span> <span class="kw">=</span> <span class="no">sites</span>, <span class="kw">alpha</span> <span class="kw">=</span> <span class="no">alpha.ct</span>,
                    <span class="kw">beta</span> <span class="kw">=</span> <span class="fl">0.5</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">50</span>, <span class="kw">theta</span> <span class="kw">=</span> <span class="fl">4.5</span>,
                    <span class="kw">sample.ids</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"R1"</span>, <span class="st">"R2"</span>, <span class="st">"R3"</span>, <span class="st">"R4"</span>))
<span class="co"># Control group</span>
<span class="no">ctrl</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html">simulateCounts</a></span>(<span class="kw">num.samples</span> <span class="kw">=</span> <span class="fl">3</span>, <span class="kw">sites</span> <span class="kw">=</span> <span class="no">sites</span>, <span class="kw">alpha</span> <span class="kw">=</span> <span class="no">alpha.ct</span>,
                    <span class="kw">beta</span> <span class="kw">=</span> <span class="fl">0.5</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">50</span>, <span class="kw">theta</span> <span class="kw">=</span> <span class="fl">4.5</span>,
                    <span class="kw">sample.ids</span> <span class="kw">=</span> <span class="no">control.nam</span>)
<span class="co"># Treatment group</span>
<span class="no">treat</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html">simulateCounts</a></span>(<span class="kw">num.samples</span> <span class="kw">=</span> <span class="fl">3</span>, <span class="kw">sites</span> <span class="kw">=</span> <span class="no">sites</span>, <span class="kw">alpha</span> <span class="kw">=</span> <span class="no">alpha.tt</span>,
                    <span class="kw">beta</span> <span class="kw">=</span> <span class="fl">0.5</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">50</span>, <span class="kw">theta</span> <span class="kw">=</span> <span class="fl">4.5</span>,
                    <span class="kw">sample.ids</span> <span class="kw">=</span> <span class="no">treatment.nam</span>)</pre></body></html></div>
<p>Notice that reference and control groups of samples are not identical but belong to the same population.</p>
</div>
<div id="divergences-of-methylation-levels" class="section level2">
<h2 class="hasAnchor">
<a href="#divergences-of-methylation-levels" class="anchor"></a>Divergences of methylation levels</h2>
<p>The estimation of the divergences of methylation levels is required to proceed with the application of signal detection basic approach. The information divergence is estimated here using the function <em>estimateDivergence</em>. For each cytosine site, methylation levels are estimated according to the formulas: <span class="math inline">\(p_i={n_i}^{mC_j}/({n_i}^{mC_j}+{n_i}^{C_j})\)</span>, where <span class="math inline">\({n_i}^{mC_j}\)</span> and <span class="math inline">\({n_i}^{C_j}\)</span> are the number of methylated and unmethylated cytosines at site <span class="math inline">\(i\)</span>.</p>
<p>If a Bayesian correction of counts is selected in function <em>estimateDivergence</em>, then methylated read counts are modeled by a beta-binomial distribution in a Bayesian framework, which accounts for the biological and sampling variations <span class="citation">(Hebestreit, Dugas, and Klein 2013; Robinson et al. 2014; Dolzhenko and Smith 2014)</span>. In our case we adopted the Bayesian approach suggested in reference <span class="citation">(Baldi and Brunak 2001)</span> (Chapter 3).</p>
<p>Two types of information divergences are estimated: <em>TV</em>, total variation (<em>TV</em>, absolute value of methylation levels) and Hellinger divergence (<em>H</em>). <em>TV</em> is computed according to the formula: <span class="math inline">\(TV_d=|p_{tt}-p_{ct}|\)</span> and <em>H</em>:</p>
<p><span class="math display">\[H(\hat p_{ij},\hat p_{ir}) = w_i\Big[(\sqrt{\hat p_{ij}} - \sqrt{\hat
p_{ir}})^2+(\sqrt{1-\hat p_{ij}} - \sqrt{1-\hat p_{ir}})^2\Big]\]</span> where <span class="math inline">\(w_i = 2 \frac{m_{ij} m_{ir}}{m_{ij} + m_{ir}}\)</span>, <span class="math inline">\(m_{ij} = {n_i}^{mC_j}+{n_i}^{uC_j}+1\)</span>, <span class="math inline">\(m_{ir} = {n_i}^{mC_r}+{n_i}^{uC_r}+1\)</span> and <span class="math inline">\(j \in {\{c,t}\}\)</span></p>
<p>The equation for Hellinger divergence is given in reference <span class="citation">(Basu, Mandal, and Pardo 2010)</span>, but any other information theoretical divergences could be used as well.</p>
<p>Divergences are estimated for control and treatment groups in respect to a virtual sample, which is created applying function <em>poolFromGRlist</em> on the reference group.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="co"># Reference sample</span>
<span class="no">ref</span> <span class="kw">=</span> <span class="fu"><a href="../reference/poolFromGRlist.html">poolFromGRlist</a></span>(<span class="no">ref0</span>, <span class="kw">stat</span> <span class="kw">=</span> <span class="st">"mean"</span>, <span class="kw">num.cores</span> <span class="kw">=</span> <span class="fl">4L</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

<span class="co"># Methylation level divergences</span>
<span class="no">DIVs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/estimateDivergence.html">estimateDivergence</a></span>(<span class="kw">ref</span> <span class="kw">=</span> <span class="no">ref</span>, <span class="kw">indiv</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">ctrl</span>, <span class="no">treat</span>), <span class="kw">Bayesian</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                            <span class="kw">num.cores</span> <span class="kw">=</span> <span class="fl">6L</span>, <span class="kw">percentile</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
<p>The mean of methylation levels differences is:</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="../reference/unlist.html">unlist</a></span>(<span class="fu"><a href="../reference/lapply.html">lapply</a></span>(<span class="no">DIVs</span>, <span class="kw">function</span>(<span class="no">x</span>) <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(<span class="fu">mcols</span>(<span class="no">x</span>[, <span class="fl">7</span>])[,<span class="fl">1</span>])))</pre></body></html></div>
<pre><code>##           C1           C2           C3           T1           T2           T3 
## 2.809709e-03 2.642337e-03 7.127907e-05 1.615112e-01 1.631192e-01 1.642946e-01</code></pre>
</div>
</div>
<div id="methylation-signal" class="section level1">
<h1 class="hasAnchor">
<a href="#methylation-signal" class="anchor"></a>Methylation signal</h1>
<p>Likewise for any other signal in nature, the analysis of methylation signal requires for the knowledge of its probability distribution. In the current case , the signal is represented in terms of the Hellinger divergence of methylation levels (<em>H</em>).</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="co"># To remove hd == 0 to estimate. The methylation signal only is given for  </span>
<span class="no">divs</span> <span class="kw">=</span> <span class="fu"><a href="../reference/lapply.html">lapply</a></span>(<span class="no">DIVs</span>, <span class="kw">function</span>(<span class="no">div</span>) <span class="no">div</span>[ <span class="no">div</span>$<span class="no">hdiv</span> <span class="kw">&gt;</span> <span class="fl">0</span> ], <span class="kw">keep.attr</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">divs</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">DIVs</span>)

<span class="co"># Data frame with the Hellinger divergences from both groups of samples samples </span>
<span class="no">l</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(); <span class="kw">for</span> (<span class="no">k</span> <span class="kw">in</span> <span class="fl">1</span>:<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">divs</span>)) <span class="no">l</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">l</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">divs</span><span class="kw">[[</span><span class="no">k</span>]]))
<span class="no">data_div</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">H</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="no">divs</span>$<span class="no">C1</span>$<span class="no">hdiv</span>), <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="no">divs</span>$<span class="no">C2</span>$<span class="no">hdiv</span>),
                            <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="no">divs</span>$<span class="no">C3</span>$<span class="no">hdiv</span>), <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="no">divs</span>$<span class="no">T1</span>$<span class="no">hdiv</span>),
                            <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="no">divs</span>$<span class="no">T2</span>$<span class="no">hdiv</span>), <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="no">divs</span>$<span class="no">T3</span>$<span class="no">hdiv</span>)),
                    <span class="kw">sample</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"C1"</span>, <span class="no">l</span>[<span class="fl">1</span>]), <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"C2"</span>,<span class="no">l</span>[<span class="fl">2</span>]), <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"C3"</span>,<span class="no">l</span>[<span class="fl">3</span>]),
                                <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"T1"</span>,<span class="no">l</span>[<span class="fl">4</span>]), <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"T2"</span>,<span class="no">l</span>[<span class="fl">5</span>]), <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"T3"</span>,<span class="no">l</span>[<span class="fl">6</span>]))
)</pre></body></html></div>
<p>Empirical critical values for the probability distribution of <span class="math inline">\(H\)</span> and <span class="math inline">\(TV\)</span> can be obtained using <em>quantile</em> function from the R package <em>stats</em>.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="no">critical.val</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span>(<span class="no">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span>(<span class="no">divs</span>, <span class="kw">function</span>(<span class="no">x</span>) {
    <span class="no">hd.95</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span>(<span class="no">x</span>$<span class="no">hdiv</span>, <span class="fl">0.95</span>)
    <span class="no">tv.95</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="no">x</span>$<span class="no">bay.TV</span>), <span class="fl">0.95</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">tv</span> <span class="kw">=</span> <span class="no">tv.95</span>, <span class="kw">hd</span> <span class="kw">=</span> <span class="no">hd.95</span>))
}))

<span class="no">critical.val</span></pre></body></html></div>
<pre><code>##       tv.95%    hd.95%
## C1 0.6732107  66.76081
## C2 0.6707595  66.71995
## C3 0.6669996  65.98495
## T1 0.9232612 138.68237
## T2 0.9318194 141.72637
## T3 0.9301630 141.77241</code></pre>
<p>The kernel density estimation yields the empirical density shown in the graphics:</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>))

<span class="co"># Some information for graphic</span>
<span class="no">crit.val.ct</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span>(<span class="no">critical.val</span>[<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"C1"</span>, <span class="st">"C2"</span>, <span class="st">"C3"</span>), <span class="fl">2</span>])) <span class="co"># 67</span>
<span class="no">crit.val.tt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span>(<span class="no">critical.val</span>[<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"T1"</span>, <span class="st">"T2"</span>, <span class="st">"T3"</span>), <span class="fl">2</span>])) <span class="co"># 139</span>

<span class="co"># Density plot with ggplot</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">data_div</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">H</span>, <span class="kw">colour</span> <span class="kw">=</span> <span class="no">sample</span>, <span class="kw">fill</span> <span class="kw">=</span> <span class="no">sample</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span>(<span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">0.05</span>, <span class="kw">bw</span> <span class="kw">=</span> <span class="fl">0.2</span>, <span class="kw">position</span> <span class="kw">=</span> <span class="st">"identity"</span>, <span class="kw">na.rm</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
            <span class="kw">size</span> <span class="kw">=</span> <span class="fl">0.4</span>) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span>(<span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">bolditalic</a></span>(<span class="st">"Hellinger divergence (H)"</span>))) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span>(<span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">bolditalic</a></span>(<span class="st">"Density"</span>))) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"Density distribution for control and treatment"</span>) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span>(<span class="kw">xintercept</span> <span class="kw">=</span> <span class="no">crit.val.ct</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"red"</span>,
                <span class="kw">linetype</span> <span class="kw">=</span> <span class="st">"dashed"</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">0.4</span>) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotate.html">annotate</a></span>(<span class="kw">geom</span> <span class="kw">=</span> <span class="st">"text"</span>, <span class="kw">x</span> <span class="kw">=</span> <span class="no">crit.val.ct</span>-<span class="fl">2</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="fl">0.08</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">5</span>,
            <span class="kw">label</span> <span class="kw">=</span> <span class="st">'bolditalic(H[alpha == 0.05]^CT == 67)'</span>,
            <span class="kw">family</span> <span class="kw">=</span> <span class="st">"serif"</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"red"</span>, <span class="kw">parse</span> <span class="kw">=</span> <span class="fl">TRUE</span>) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span>(<span class="kw">xintercept</span> <span class="kw">=</span> <span class="no">crit.val.tt</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"blue"</span>,
                <span class="kw">linetype</span> <span class="kw">=</span> <span class="st">"dashed"</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">0.4</span>) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotate.html">annotate</a></span>(<span class="kw">geom</span> <span class="kw">=</span> <span class="st">"text"</span>, <span class="kw">x</span> <span class="kw">=</span> <span class="no">crit.val.tt</span> -<span class="fl">2</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="fl">0.1</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">5</span>,
            <span class="kw">label</span> <span class="kw">=</span> <span class="st">'bolditalic(H[alpha == 0.05]^TT == 139)'</span>,
            <span class="kw">family</span> <span class="kw">=</span> <span class="st">"serif"</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"blue"</span>, <span class="kw">parse</span> <span class="kw">=</span> <span class="fl">TRUE</span>) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(
        <span class="kw">axis.text.x</span>  <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>( <span class="kw">face</span> <span class="kw">=</span> <span class="st">"bold"</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">12</span>, <span class="kw">color</span><span class="kw">=</span><span class="st">"black"</span>,
                                <span class="kw">margin</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span>(<span class="fl">1</span>,<span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0</span>, <span class="kw">unit</span> <span class="kw">=</span> <span class="st">"pt"</span> )),
        <span class="kw">axis.text.y</span>  <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>( <span class="kw">face</span> <span class="kw">=</span> <span class="st">"bold"</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">12</span>, <span class="kw">color</span><span class="kw">=</span><span class="st">"black"</span>,
                                <span class="kw">margin</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span>( <span class="fl">0</span>,<span class="fl">0.1</span>,<span class="fl">0</span>,<span class="fl">0</span>, <span class="kw">unit</span> <span class="kw">=</span> <span class="st">"mm"</span>)),
        <span class="kw">axis.title.x</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="kw">face</span> <span class="kw">=</span> <span class="st">"bold"</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">13</span>,
                                <span class="kw">color</span><span class="kw">=</span><span class="st">"black"</span>, <span class="kw">vjust</span> <span class="kw">=</span> <span class="fl">0</span> ),
        <span class="kw">axis.title.y</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="kw">face</span> <span class="kw">=</span> <span class="st">"bold"</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">13</span>,
                                <span class="kw">color</span><span class="kw">=</span><span class="st">"black"</span>, <span class="kw">vjust</span> <span class="kw">=</span> <span class="fl">0</span> ),
        <span class="kw">legend.title</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span>(),
        <span class="kw">legend.margin</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.3</span>, <span class="fl">0.3</span>, <span class="fl">0.3</span>, <span class="fl">0.3</span>), <span class="kw">unit</span> <span class="kw">=</span> <span class="st">'mm'</span>),
        <span class="kw">legend.box.spacing</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/reexports.html">unit</a></span>(<span class="fl">0.5</span>, <span class="st">"lines"</span>),
        <span class="kw">legend.text</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="kw">face</span> <span class="kw">=</span> <span class="st">"bold"</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="fl">12</span>, <span class="kw">family</span> <span class="kw">=</span> <span class="st">"serif"</span>)
    )</pre></body></html></div>
<p><img src="Methylation_analysis_with_Methyl-IT_files/figure-html/dens.graph-1.png" width="700"></p>
<p>The graphic above shows that with high probability the methylation signal induced by the treatment has <em>H</em> values <span class="math inline">\(H^{TT}_{\alpha=0.05}\geq139\)</span>. According to the critical value estimated for the differences of methylation levels, the methylation signal holds <span class="math inline">\(TV^{TT}_{\alpha=0.05}\geq0.67\)</span>.</p>
<p>Notice that most of the methylation changes are not signal but methylation background noise (found to the left of the critical values). This situation is typical for all the natural and technologically generated signals <span class="citation">(Sanchez and Mackenzie 2016; Sanchez et al. 2019)</span>.</p>
<div id="signal-detection" class="section level2">
<h2 class="hasAnchor">
<a href="#signal-detection" class="anchor"></a>Signal detection</h2>
<p>Signal detection is a critical step to increase sensitivity and resolution of methylation signal by reducing the signal-to-noise ratio and objectively controlling the false positive rate and prediction accuracy/risk.</p>
<p>The first estimation in our signal detection step is the identification of the cytosine sites carrying potential methylation signal <span class="math inline">\(PS\)</span>. To identify cytosine sites carrying <span class="math inline">\(PS\)</span>, we must find the best fitted probability distribution model.</p>
<div id="estimation-of-the-best-fitted-probability-distribution-model" class="section level3">
<h3 class="hasAnchor">
<a href="#estimation-of-the-best-fitted-probability-distribution-model" class="anchor"></a><em>Estimation of the best fitted probability distribution model</em>
</h3>
<p>Potential DMPs can be estimated using the critical values derived from the best fitted probability distribution model, which is obtained after the non-linear fit of the theoretical model on the genome-wide <span class="math inline">\(H\)</span> values for each individual sample using Methyl-IT function <a href="https://is.gd/0BLLkT">nonlinearFitDist</a> <span class="citation">(Sanchez and Mackenzie 2016)</span>. As before, only cytosine sites with critical values <span class="math inline">\(TV_d&gt;0.926\)</span> are considered DMPs. Notice that, it is always possible to use any other values of <span class="math inline">\(H\)</span> and <span class="math inline">\(TV_d\)</span> as critical values, but whatever could be the value it will affect the final accuracy of the classification performance of DMPs into two groups, DMPs from control and DMPs from treatment (see below). So, it is important to do an good choices of the critical values.</p>
<p>Function <a href="https://is.gd/W7dlP0">gofReport</a> search for the best fitted model between the set of models requested by the user.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">d</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Weibull2P"</span>, <span class="st">"Weibull3P"</span>, <span class="st">"Gamma2P"</span>, <span class="st">"Gamma3P"</span>, <span class="st">"GGamma3P"</span>)
<span class="no">gof</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/gofReport.html">gofReport</a></span>(<span class="kw">HD</span> <span class="kw">=</span> <span class="no">DIVs</span>, <span class="kw">column</span> <span class="kw">=</span> <span class="fl">9L</span>, <span class="kw">model</span> <span class="kw">=</span> <span class="no">d</span>, <span class="kw">num.cores</span> <span class="kw">=</span> <span class="fl">4L</span>,
            <span class="kw">tasks</span> <span class="kw">=</span> <span class="fl">2L</span>, <span class="kw">output</span> <span class="kw">=</span> <span class="st">"all"</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="no">gof</span>$<span class="no">bestModel</span></pre></body></html></div>
<p>There is not a one-to-one mapping between <span class="math inline">\(TV_d\)</span> and <span class="math inline">\(H\)</span>. However, at each cytosine site <span class="math inline">\(i\)</span>, these information divergences hold the inequality: <span class="math display">\[TV_d(p^{tt}_i,p^{ct}_i)\leq \sqrt{2}H_d(p^{tt}_i,p^{ct}_i)\]</span></p>
<p>where <span class="math inline">\(H_d(p^{tt}_i,p^{ct}_i)=\sqrt{\frac{H(p^{tt}_i,p^{ct}_i)}w}\)</span> is the Hellinger distance and <span class="math inline">\(H(p^{tt}_i,p^{ct}_i)\)</span> is the Hellinger divergence equation given above.</p>
</div>
<div id="potential-methylation-signal" class="section level3">
<h3 class="hasAnchor">
<a href="#potential-methylation-signal" class="anchor"></a><em>Potential methylation signal</em>
</h3>
<p>Methylation regulatory signals do not hold the theoretical distribution and, consequently, for a given level of significance <span class="math inline">\(\alpha\)</span> (Type I error probability, e.g. <span class="math inline">\(\alpha = 0.05\)</span>), cytosine positions <span class="math inline">\(k\)</span> with information divergence <span class="math inline">\(H_k &gt;= H_{\alpha = 0.05}\)</span> can be selected as sites carrying potential signals <span class="math inline">\(PS\)</span>. The <span class="math inline">\(PS\)</span> is detected with the function <a href="https://is.gd/yGORtg">getPotentialDIMP</a>.</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="no">PS</span> <span class="kw">=</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span>(<span class="kw">LR</span> <span class="kw">=</span> <span class="no">DIVs</span>, <span class="kw">dist.name</span> <span class="kw">=</span> <span class="no">gof</span>$<span class="no">bestModel</span>, <span class="kw">nlms</span> <span class="kw">=</span> <span class="no">gof</span>$<span class="no">nlms</span>,
                    <span class="kw">div.col</span> <span class="kw">=</span> <span class="fl">9</span>, <span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">0.05</span>, <span class="kw">tv.col</span> <span class="kw">=</span> <span class="fl">8</span>, <span class="kw">tv.cut</span> <span class="kw">=</span>  <span class="fl">0.674</span>)</pre></body></html></div>
<p>Observe that we have used the <span class="math inline">\(|TV|\)</span> values based on the Bayesian estimation of methylation levels (bay.TV, tv.col = 8).</p>
</div>
<div id="potential-dmps-detected-with-fishers-exact-test-ft" class="section level3">
<h3 class="hasAnchor">
<a href="#potential-dmps-detected-with-fishers-exact-test-ft" class="anchor"></a><em>Potential DMPs detected with Fisher’s exact test (FT)</em>
</h3>
<p>Fisher’s exact test (FT) is run here as standard alternative approach. A full discussion on the performance of FT and other statistical approaches in comparison with Methyl-IT pipeline can be read in <span class="citation">(Sanchez et al. 2019)</span>. Notice that the standard application of FT does not include a signal detection step (see more at <span class="citation">(Sanchez et al. 2019)</span>).</p>
<p>In Methyl-IT, FT is implemented in function <em>FisherTest</em>. In the current case, a pairwise group application of FT to each cytosine site is performed. The differences between the group means of read counts of methylated and unmethylated cytosines at each site are used for testing (<span class="math inline">\(pooling.stat ="mean"\)</span>). Notice that only cytosine sites with critical values <span class="math inline">\(TV_d&gt;0.67\)</span> are tested (<em>tv.cut = 0.67</em>).</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="no">ft</span> <span class="kw">=</span> <span class="fu"><a href="../reference/FisherTest.html">FisherTest</a></span>(<span class="kw">LR</span> <span class="kw">=</span> <span class="no">divs</span>, <span class="kw">tv.cut</span> <span class="kw">=</span> <span class="fl">0.67</span>,
                    <span class="kw">pAdjustMethod</span> <span class="kw">=</span> <span class="st">"BH"</span>,  <span class="kw">pooling.stat</span> <span class="kw">=</span> <span class="st">"mean"</span>,
                    <span class="kw">pvalCutOff</span> <span class="kw">=</span> <span class="fl">0.05</span>, <span class="kw">num.cores</span> <span class="kw">=</span> <span class="fl">4L</span>,
                    <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">saveAll</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

<span class="co"># TV column is the standard TV value (without Bayesian correction of methylation</span>
<span class="co"># levels)</span>
<span class="no">ft.tv</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span>(<span class="kw">LR</span> <span class="kw">=</span> <span class="no">ft</span>, <span class="kw">div.col</span> <span class="kw">=</span> <span class="fl">9L</span>, <span class="kw">dist.name</span> <span class="kw">=</span> <span class="st">"None"</span>,
                        <span class="kw">tv.cut</span> <span class="kw">=</span> <span class="fl">0.67</span>, <span class="kw">tv.col</span> <span class="kw">=</span> <span class="fl">7</span>, <span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">0.05</span>)</pre></body></html></div>
<p>The above setting would impose additional constrains on the output of DMPs resulting from Fisher’s exact test depending on how strong is the methylation background noise in the control population. Basically, methylation variations that can spontaneously occur in the control population with relatively high frequencies are disregarded. The decisions are based on the empirical cumulative distribution function (ECDF).</p>
<p>Potential DMPs detected with FT can be constrained with the critical values derived for the best fitted probability distribution models</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="co"># Potential DMPs from Fisher's exact test </span>
<span class="no">ft.hd</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span>(<span class="kw">LR</span> <span class="kw">=</span> <span class="no">ft</span>, <span class="kw">div.col</span> <span class="kw">=</span> <span class="fl">9L</span>, <span class="kw">nlms</span> <span class="kw">=</span> <span class="no">gof</span>$<span class="no">nlms</span>,
                        <span class="kw">tv.cut</span> <span class="kw">=</span> <span class="fl">0.674</span>, <span class="kw">tv.col</span> <span class="kw">=</span> <span class="fl">8</span>, <span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">0.05</span>,
                        <span class="kw">dist.name</span> <span class="kw">=</span> <span class="no">gof</span>$<span class="no">bestModel</span>)
<span class="no">ft.hd</span>$<span class="no">T1</span></pre></body></html></div>
<pre><code>## GRanges object with 2226 ranges and 12 metadata columns:
##          seqnames    ranges strand |        c1        t1        c2        t2
##             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##      [1]        1        29      * |         6       110       112         4
##      [2]        1        42      * |         0       112       112         0
##      [3]        1        70      * |         0        84        84         0
##      [4]        1       103      * |         0        94        91         3
##      [5]        1       105      * |         0       168       118        50
##      ...      ...       ...    ... .       ...       ...       ...       ...
##   [2222]        1     49928      * |        10       148       158         0
##   [2223]        1     49974      * |        19       130       147         2
##   [2224]        1     49980      * |        45       139       184         0
##   [2225]        1     49981      * |        34       143       177         0
##   [2226]        1     49996      * |        20       196       216         0
##                  p1        p2        TV    bay.TV      hdiv       pvalue
##           &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;    &lt;numeric&gt;
##      [1] 0.05779987  0.961168  0.913793  0.903368   134.086  1.22001e-52
##      [2] 0.00845846  0.995244  1.000000  0.986785   189.744  1.39309e-66
##      [3] 0.01112206  0.993674  1.000000  0.982552   138.682  8.69664e-50
##      [4] 0.00999767  0.962704  0.968085  0.952706   134.851  1.29337e-50
##      [5] 0.00571913  0.700659  0.702381  0.694940   132.207  2.02668e-50
##      ...        ...       ...       ...       ...       ...          ...
##   [2222]  0.0674515  0.996621  0.936709  0.929170   217.700  1.25471e-78
##   [2223]  0.1298691  0.983070  0.859060  0.853200   156.395  3.15691e-60
##   [2224]  0.2434358  0.997097  0.755435  0.753661   170.369  1.01280e-61
##   [2225]  0.1923356  0.996982  0.807910  0.804647   182.533  2.77686e-66
##   [2226]  0.0950086  0.997525  0.907407  0.902516   279.852 2.42409e-100
##             adj.pval      wprob
##            &lt;numeric&gt;  &lt;numeric&gt;
##      [1] 4.86816e-52  0.0480895
##      [2] 1.07813e-65  0.0215267
##      [3] 3.03410e-49  0.0448148
##      [4] 4.68846e-50  0.0475259
##      [5] 7.31951e-50  0.0495093
##      ...         ...        ...
##   [2222] 1.67316e-77 0.01487098
##   [2223] 1.80286e-59 0.03441153
##   [2224] 6.21624e-61 0.02815461
##   [2225] 2.11753e-65 0.02375873
##   [2226] 9.74758e-99 0.00693177
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<p>The above result shows column ‘adj.pval’ with the FT’s p.values adjusted for multiple comparisons and column ‘wprob’ with the predicted probabilities: <span class="math inline">\(1 - P(H_k &gt;= H_{\alpha = 0.05})\)</span> from the best fitted model.</p>
<p>Summary table:</p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">ft</span> <span class="kw">=</span> <span class="fu"><a href="../reference/unlist.html">unlist</a></span>(<span class="fu"><a href="../reference/lapply.html">lapply</a></span>(<span class="no">ft.tv</span>, <span class="no">length</span>)),
            <span class="kw">ft.hd</span> <span class="kw">=</span> <span class="fu"><a href="../reference/unlist.html">unlist</a></span>(<span class="fu"><a href="../reference/lapply.html">lapply</a></span>(<span class="no">ft.hd</span>, <span class="no">length</span>)),
<span class="kw">BestModel</span> <span class="kw">=</span> <span class="fu"><a href="../reference/unlist.html">unlist</a></span>(<span class="fu"><a href="../reference/lapply.html">lapply</a></span>(<span class="no">PS</span>, <span class="no">length</span>)))</pre></body></html></div>
<pre><code>##      ft ft.hd BestModel
## C1 1865  1044      1044
## C2 1824  1075      1075
## C3 1765  1039      1039
## T1 7800  2226      2226
## T2 7847  2167      2167
## T3 7879  2189      2189</code></pre>
<p>Notice that there are significant differences between the numbers of PSs detected by FT and other approaches. There at least about 7800 - 2226 = 5574 or more PSs in the treatment samples that are not induced by the treatment, which can naturally occurs in both groups, control and treatment. In other words, the best probabilistic distribution models suggest that those methylation events are resultant of the normal (non-stressful) biological processes. Any information divergence of methylation level below the critical value <span class="math inline">\(TV_{\alpha=0.05}\)</span> or <span class="math inline">\(H_{\alpha=0.05}\)</span> is a methylation event that can occur in normal conditions. <em>It does not matter how much significant a test like FT could be, since it is not about the signification of the test, but about how big is the probability to observe that methylation event in the control population as well</em>.</p>
</div>
<div id="critical-values-based-on-fitted-models" class="section level3">
<h3 class="hasAnchor">
<a href="#critical-values-based-on-fitted-models" class="anchor"></a>Critical values based on fitted models</h3>
<p>Critical values <span class="math inline">\(H_{\alpha=0.05}\)</span> can be estimated from each fitted model using <a href="https://is.gd/CCqNYS">predict</a> function:</p>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="co"># ==== 95% quantiles </span>
<span class="no">q95</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">q95</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span>(<span class="kw">nlm</span> <span class="kw">=</span> <span class="no">gof</span>$<span class="no">nlms</span>, <span class="kw">pred</span> <span class="kw">=</span> <span class="st">"quant"</span>, <span class="kw">q</span> <span class="kw">=</span> <span class="fl">0.95</span>,
                                <span class="kw">dist.name</span> <span class="kw">=</span> <span class="no">gof</span>$<span class="no">bestModel</span>))
<span class="no">q95</span></pre></body></html></div>
<pre><code>##          q95
## C1  59.08940
## C2  58.32025
## C3  58.32671
## T1 131.57176
## T2 137.55804
## T3 136.17148</code></pre>
</div>
</div>
</div>
<div id="cutpoint-for-the-spontaneous-variability-in-the-control-group" class="section level1">
<h1 class="hasAnchor">
<a href="#cutpoint-for-the-spontaneous-variability-in-the-control-group" class="anchor"></a>Cutpoint for the spontaneous variability in the control group</h1>
<p>Normally, there is a spontaneous variability in the control group. This is a consequence of the random fluctuations, or noise, inherent to the methylation process. The stochasticity of the the methylation process is derives from the stochasticity inherent in biochemical processes. There are fundamental physical reasons to acknowledge that biochemical processes are subject to noise and fluctuations <span class="citation">(Samoilov, Price, and Arkin 2006; Eldar and Elowitz 2010)</span>. So, regardless constant environment, statistically significant methylation changes can be found in control population with probability greater than zero and proportional to a Boltzmann factor <span class="citation">(Sanchez and Mackenzie 2016)</span>.</p>
<p>Natural signals and those generated by human technology are not free of noise and, as mentioned above, the methylation signal is no exception. Only signal detection based approaches are designed to filter out the signal from the noise, in natural and in human generated signals.</p>
<p>The need for the application of (what is now known as) signal detection in cancer research was pointed out by Youden in the midst of the last century <span class="citation">(Youden 1950)</span>. Here, the application of signal detection approach was performed according with the standard practice in current implementations of clinical diagnostic test <span class="citation">(Carter et al. 2016; López-Ratón et al. 2014; Hippenstiel 2001)</span>. That is, optimal cutoff values of the methylation signal were estimated on the receiver operating characteristic curves (ROCs) and applied to identify DMPs. The decision of whether a DMP detected by Fisher’s exact test (or any other statistical test implemented in Methyl-IT) is taken based on the optimal cutoff value.</p>
<p>In the current example, the column carrying <em>TV</em> (<em>div.col</em> = 7L) will be used to estimate the cutpoint. The column values will be expressed in terms of <span class="math inline">\(TV_d=|p_{tt}-p_{ct}|\)</span> (<em>absolute</em> = TRUE):</p>
<div id="cutpoint-for-ps-estimated-based-on-the-best-probability-distribution-model" class="section level3">
<h3 class="hasAnchor">
<a href="#cutpoint-for-ps-estimated-based-on-the-best-probability-distribution-model" class="anchor"></a>Cutpoint for PS estimated based on the best probability distribution model</h3>
<p>Next, the search for an optimal cutpoint is accomplished with a model classifier and using the total variation distance (with Bayesian correction, bay.TV)</p>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="no">cutpoint</span> <span class="kw">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span>(<span class="kw">LR</span> <span class="kw">=</span> <span class="no">PS</span>,
                            <span class="kw">control.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"C1"</span>,<span class="st">"C2"</span>, <span class="st">"C3"</span>),
                            <span class="kw">treatment.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"T1"</span>, <span class="st">"T2"</span>, <span class="st">"T3"</span>),
                            <span class="kw">simple</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
                            <span class="kw">classifier1</span> <span class="kw">=</span> <span class="st">"pca.logistic"</span>,
                            <span class="kw">classifier2</span> <span class="kw">=</span> <span class="st">"pca.qda"</span>,
                            <span class="kw">column</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">hdiv</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">bay.TV</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                                        <span class="kw">wprob</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">pos</span> <span class="kw">=</span> <span class="fl">TRUE</span>),
                            <span class="kw">n.pc</span> <span class="kw">=</span> <span class="fl">4</span> , <span class="kw">center</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">scale</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                            <span class="kw">div.col</span> <span class="kw">=</span> <span class="fl">9</span>, <span class="kw">clas.perf</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="no">cutpoint</span>$<span class="no">cutpoint</span></pre></body></html></div>
<pre><code>## [1] 131.581</code></pre>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="no">cutpoint</span>$<span class="no">testSetPerformance</span></pre></body></html></div>
<pre><code>## Confusion Matrix and Statistics
## 
##           Reference
## Prediction   CT   TT
##         CT  198    0
##         TT    0 2633
##                                      
##                Accuracy : 1          
##                  95% CI : (0.9987, 1)
##     No Information Rate : 0.9301     
##     P-Value [Acc &gt; NIR] : &lt; 2.2e-16  
##                                      
##                   Kappa : 1          
##                                      
##  Mcnemar's Test P-Value : NA         
##                                      
##             Sensitivity : 1.0000     
##             Specificity : 1.0000     
##          Pos Pred Value : 1.0000     
##          Neg Pred Value : 1.0000     
##              Prevalence : 0.9301     
##          Detection Rate : 0.9301     
##    Detection Prevalence : 0.9301     
##       Balanced Accuracy : 1.0000     
##                                      
##        'Positive' Class : TT         
## </code></pre>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="no">cutpoint</span>$<span class="no">testSetModel.FDR</span></pre></body></html></div>
<pre><code>## [1] 0</code></pre>
</div>
<div id="cutpoint-for-ps-estimated-based-on-ft" class="section level3">
<h3 class="hasAnchor">
<a href="#cutpoint-for-ps-estimated-based-on-ft" class="anchor"></a>Cutpoint for PS estimated based on FT</h3>
<p>The best classification performance based PS estimated based on FT are obtained with the application of <a href="https://is.gd/t8UKvK">Youden index</a> <span class="citation">(Youden 1950)</span>.</p>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="no">cutpoint2</span> <span class="kw">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span>(<span class="kw">LR</span> <span class="kw">=</span> <span class="no">ft.tv</span>,
                            <span class="kw">control.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"C1"</span>,<span class="st">"C2"</span>, <span class="st">"C3"</span>),
                            <span class="kw">treatment.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"T1"</span>, <span class="st">"T2"</span>, <span class="st">"T3"</span>),
                            <span class="kw">simple</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                            <span class="kw">classifier1</span> <span class="kw">=</span> <span class="st">"pca.qda"</span>,
                            <span class="kw">column</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">hdiv</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">bay.TV</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                                        <span class="kw">wprob</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">pos</span> <span class="kw">=</span> <span class="fl">TRUE</span>),
                            <span class="kw">n.pc</span> <span class="kw">=</span> <span class="fl">4</span> , <span class="kw">center</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">scale</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                            <span class="kw">div.col</span> <span class="kw">=</span> <span class="fl">9</span>, <span class="kw">clas.perf</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                            <span class="kw">post.cut</span> <span class="kw">=</span> <span class="fl">0.9</span>)
<span class="no">cutpoint2</span>$<span class="no">cutpoint</span></pre></body></html></div>
<pre><code>## [1] 99.9462</code></pre>
<div class="sourceCode" id="cb28"><html><body><pre class="r"><span class="no">cutpoint2</span>$<span class="no">testSetPerformance</span></pre></body></html></div>
<pre><code>## Confusion Matrix and Statistics
## 
##           Reference
## Prediction   CT   TT
##         CT   11   38
##         TT  474 4332
##                                          
##                Accuracy : 0.8945         
##                  95% CI : (0.8856, 0.903)
##     No Information Rate : 0.9001         
##     P-Value [Acc &gt; NIR] : 0.9052         
##                                          
##                   Kappa : 0.0233         
##                                          
##  Mcnemar's Test P-Value : &lt;2e-16         
##                                          
##             Sensitivity : 0.99130        
##             Specificity : 0.02268        
##          Pos Pred Value : 0.90137        
##          Neg Pred Value : 0.22449        
##              Prevalence : 0.90010        
##          Detection Rate : 0.89228        
##    Detection Prevalence : 0.98991        
##       Balanced Accuracy : 0.50699        
##                                          
##        'Positive' Class : TT             
## </code></pre>
<div class="sourceCode" id="cb30"><html><body><pre class="r"><span class="no">cutpoint2</span>$<span class="no">testSetModel.FDR</span></pre></body></html></div>
<pre><code>## [1] 0.09862672</code></pre>
<p>Addition of information on the probability distribution of the methylation signal to potential signal identified with FT will emulate the results obtained with the Methyl-IT pipeline.</p>
<div class="sourceCode" id="cb32"><html><body><pre class="r"><span class="no">cutpoint3</span> <span class="kw">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span>(<span class="kw">LR</span> <span class="kw">=</span> <span class="no">ft.hd</span>,
                            <span class="kw">control.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"C1"</span>,<span class="st">"C2"</span>, <span class="st">"C3"</span>),
                            <span class="kw">treatment.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"T1"</span>, <span class="st">"T2"</span>, <span class="st">"T3"</span>),
                            <span class="kw">simple</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
                            <span class="kw">classifier1</span> <span class="kw">=</span> <span class="st">"pca.logistic"</span>,
                            <span class="kw">classifier2</span> <span class="kw">=</span> <span class="st">"pca.qda"</span>,
                            <span class="kw">column</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">hdiv</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">bay.TV</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                                        <span class="kw">wprob</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">pos</span> <span class="kw">=</span> <span class="fl">TRUE</span>),
                            <span class="kw">n.pc</span> <span class="kw">=</span> <span class="fl">4</span> , <span class="kw">center</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">scale</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                            <span class="kw">div.col</span> <span class="kw">=</span> <span class="fl">9</span>, <span class="kw">clas.perf</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="no">cutpoint3</span>$<span class="no">cutpoint</span></pre></body></html></div>
<pre><code>## [1] 131.581</code></pre>
<div class="sourceCode" id="cb34"><html><body><pre class="r"><span class="no">cutpoint3</span>$<span class="no">testSetPerformance</span></pre></body></html></div>
<pre><code>## Confusion Matrix and Statistics
## 
##           Reference
## Prediction   CT   TT
##         CT  198    0
##         TT    0 2633
##                                      
##                Accuracy : 1          
##                  95% CI : (0.9987, 1)
##     No Information Rate : 0.9301     
##     P-Value [Acc &gt; NIR] : &lt; 2.2e-16  
##                                      
##                   Kappa : 1          
##                                      
##  Mcnemar's Test P-Value : NA         
##                                      
##             Sensitivity : 1.0000     
##             Specificity : 1.0000     
##          Pos Pred Value : 1.0000     
##          Neg Pred Value : 1.0000     
##              Prevalence : 0.9301     
##          Detection Rate : 0.9301     
##    Detection Prevalence : 0.9301     
##       Balanced Accuracy : 1.0000     
##                                      
##        'Positive' Class : TT         
## </code></pre>
<div class="sourceCode" id="cb36"><html><body><pre class="r"><span class="no">cutpoint3</span>$<span class="no">testSetModel.FDR</span></pre></body></html></div>
<pre><code>## [1] 0</code></pre>
<p>Therefore, invoking the parsimony principle, we assume that signal detection and machine-learning classifiers are sufficient <span class="citation">(El Naqa et al. 2018)</span>.</p>
</div>
<div id="dmps" class="section level2">
<h2 class="hasAnchor">
<a href="#dmps" class="anchor"></a>DMPs</h2>
<p>Cytosine sites carrying a methylation signal are designated differentially informative methylated positions (DMPs). With high probability true DMPs can be selected with Methyl-IT function <a href="https://is.gd/7cVEEd">selectDIMP</a>.</p>
<div class="sourceCode" id="cb38"><html><body><pre class="r"><span class="no">DMPs</span> <span class="kw">=</span> <span class="fu"><a href="../reference/selectDIMP.html">selectDIMP</a></span>(<span class="no">PS</span>, <span class="kw">div.col</span> <span class="kw">=</span> <span class="fl">9</span>, <span class="kw">cutpoint</span> <span class="kw">=</span> <span class="no">cutpoint</span>$<span class="no">cutpoint</span>)</pre></body></html></div>
<div id="evaluation-of-dmp-classification" class="section level3">
<h3 class="hasAnchor">
<a href="#evaluation-of-dmp-classification" class="anchor"></a>Evaluation of DMP classification</h3>
<p>As shown above, DMPs are found in the control population as well. Hence, it is important to know whether a DMP is the resulting effect of the treatment or just spontaneously occurred in the control sample as well. In particular, the confrontation of this issue is extremely important when methylation analysis is intended to be used as part of a diagnostic clinical test and a decision making in biotechnology industry.</p>
<p>Methyl-IT function <a href="https://is.gd/nzy9j3">evaluateDIMPclass</a> is used here to evaluate the classification of DMPs into one of the two classes, control and treatment. Several classifiers are available to be used with this function (see the help/manual for this function or type ?evaluateDIMPclass in command line).</p>
<p>A first evaluation of DMP classification performance was already done after setting “clas.perf = TRUE” in function <a href="https://is.gd/RR9iWY">estimateCutPoint</a>. However, this performance derives just from randomly split the set of sample into two subsets: training (60%) and testing (40%). A good classification performance report would be just the result from a lucky split of the set of samples! Hence, what would happen if the random split and further analysis are repeated 300 times?</p>
<div class="sourceCode" id="cb39"><html><body><pre class="r"><span class="no">class.perf</span> <span class="kw">=</span> <span class="fu"><a href="../reference/evaluateDIMPclass.html">evaluateDIMPclass</a></span>(<span class="kw">LR</span> <span class="kw">=</span> <span class="no">DMPs</span>, <span class="kw">control.names</span> <span class="kw">=</span> <span class="no">control.nam</span>,
                            <span class="kw">treatment.names</span> <span class="kw">=</span> <span class="no">treatment.nam</span>,
                            <span class="kw">column</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">hdiv</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">TV</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                                        <span class="kw">wprob</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">pos</span> <span class="kw">=</span> <span class="fl">TRUE</span>),
                            <span class="kw">classifier</span> <span class="kw">=</span> <span class="st">"pca.qda"</span>, <span class="kw">n.pc</span> <span class="kw">=</span> <span class="fl">4</span>, <span class="kw">pval.col</span> <span class="kw">=</span> <span class="fl">10L</span>,
                            <span class="kw">center</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">scale</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">num.boot</span> <span class="kw">=</span> <span class="fl">300</span>,
                            <span class="kw">output</span> <span class="kw">=</span> <span class="st">"mc.val"</span>, <span class="kw">prop</span> <span class="kw">=</span> <span class="fl">0.6</span>, <span class="kw">num.cores</span> <span class="kw">=</span> <span class="fl">4L</span>,
                            <span class="kw">tasks</span> <span class="kw">=</span> <span class="fl">2L</span>
)
<span class="no">class.perf</span></pre></body></html></div>
<pre><code>##     Accuracy     Kappa   AccuracyLower    AccuracyUpper  AccuracyNull   
##  Min.   :1   Min.   :1   Min.   :0.9987   Min.   :1     Min.   :0.9301  
##  1st Qu.:1   1st Qu.:1   1st Qu.:0.9987   1st Qu.:1     1st Qu.:0.9301  
##  Median :1   Median :1   Median :0.9987   Median :1     Median :0.9301  
##  Mean   :1   Mean   :1   Mean   :0.9987   Mean   :1     Mean   :0.9301  
##  3rd Qu.:1   3rd Qu.:1   3rd Qu.:0.9987   3rd Qu.:1     3rd Qu.:0.9301  
##  Max.   :1   Max.   :1   Max.   :0.9987   Max.   :1     Max.   :0.9301  
##                                                                         
##  AccuracyPValue      McnemarPValue  Sensitivity  Specificity Pos Pred Value
##  Min.   :7.155e-90   Min.   : NA   Min.   :1    Min.   :1    Min.   :1     
##  1st Qu.:7.155e-90   1st Qu.: NA   1st Qu.:1    1st Qu.:1    1st Qu.:1     
##  Median :7.155e-90   Median : NA   Median :1    Median :1    Median :1     
##  Mean   :7.155e-90   Mean   :NaN   Mean   :1    Mean   :1    Mean   :1     
##  3rd Qu.:7.155e-90   3rd Qu.: NA   3rd Qu.:1    3rd Qu.:1    3rd Qu.:1     
##  Max.   :7.155e-90   Max.   : NA   Max.   :1    Max.   :1    Max.   :1     
##                      NA's   :300                                           
##  Neg Pred Value   Precision     Recall        F1      Prevalence    
##  Min.   :1      Min.   :1   Min.   :1   Min.   :1   Min.   :0.9301  
##  1st Qu.:1      1st Qu.:1   1st Qu.:1   1st Qu.:1   1st Qu.:0.9301  
##  Median :1      Median :1   Median :1   Median :1   Median :0.9301  
##  Mean   :1      Mean   :1   Mean   :1   Mean   :1   Mean   :0.9301  
##  3rd Qu.:1      3rd Qu.:1   3rd Qu.:1   3rd Qu.:1   3rd Qu.:0.9301  
##  Max.   :1      Max.   :1   Max.   :1   Max.   :1   Max.   :0.9301  
##                                                                     
##  Detection Rate   Detection Prevalence Balanced Accuracy
##  Min.   :0.9301   Min.   :0.9301       Min.   :1        
##  1st Qu.:0.9301   1st Qu.:0.9301       1st Qu.:1        
##  Median :0.9301   Median :0.9301       Median :1        
##  Mean   :0.9301   Mean   :0.9301       Mean   :1        
##  3rd Qu.:0.9301   3rd Qu.:0.9301       3rd Qu.:1        
##  Max.   :0.9301   Max.   :0.9301       Max.   :1        
## </code></pre>
</div>
</div>
</div>
<div id="conclusions-summary" class="section level1">
<h1 class="hasAnchor">
<a href="#conclusions-summary" class="anchor"></a>Conclusions Summary</h1>
<p>Herein, an illustrative example of methylation analysis with Methyl-IT have been presented. Whatever could be the statistical test/approach used to identify DMPs, the analysis with simulated datasets, where the average of methylation levels in the control samples is relatively high, indicates the need for the application of signal detection based approaches.</p>
<p>For the current simulated dataset, the best classification performance was obtained for the approach of DMP detection based on the best fitted probability distribution model for the Hellinger divergence of methylation levels. DMPs from treatment are distinguish from control DMPs with very high accuracy.</p>
<div id="concluding-remarks" class="section level2">
<h2 class="hasAnchor">
<a href="#concluding-remarks" class="anchor"></a>Concluding remarks</h2>
<p>The simplest suggested steps to follow for a methylation analysis with Methyl-IT are:</p>

<ol style="list-style-type: decimal">
<li>To estimate a reference virtual sample from a reference group by using function <a href="https://is.gd/qR79kW">poolFromGRlist</a>. Notice that several statistics are available to estimate the virtual samples, i.e., <em>mean</em>, <em>median</em>, <em>sum</em>. For experiments limited by the number of sample, at least, try the estimation of the virtual sample from the control group. Alternatively, the whole reference group can be used in pairwise comparisons with control and treatment groups (computationally expensive).</li>
</ol>
<ol start="2" style="list-style-type: decimal">
<li>To estimate information divergence using function <a href="https://is.gd/0ViBO9">estimateDivergence</a>
</li>
</ol>
<ol start="3" style="list-style-type: decimal">
<li>To perform the estimation of the best fitted probability distribution model using function <a href="https://is.gd/W7dlP0">gofReport</a>.</li>
</ol>
<ol start="4" style="list-style-type: decimal">
<li>To get the potential DMPs using function <a href="https://is.gd/yGORtg">getPotentialDIMP</a>.</li>
</ol>
<ol start="5" style="list-style-type: decimal">
<li>To estimate the optimal cutpoint using function <a href="https://is.gd/RR9iWY">estimateCutPoint</a>.</li>
</ol>
<ol start="6" style="list-style-type: decimal">
<li>To retrieve DMPs with function <a href="https://is.gd/7cVEEd">selectDIMP</a>.</li>
</ol>
<ol start="7" style="list-style-type: decimal">
<li>To evaluate the classification performance using function <a href="https://is.gd/nzy9j3">evaluateDIMPclass</a>.</li>
</ol>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Baldi2001">
<p>Baldi, Pierre, and Soren Brunak. 2001. <em>Bioinformatics: the machine learning approach</em>. Second. Cambridge: MIT Press.</p>
</div>
<div id="ref-Basu2010">
<p>Basu, A., A. Mandal, and L. Pardo. 2010. “Hypothesis testing for two discrete populations based on the Hellinger distance.” <em>Statistics &amp; Probability Letters</em> 80 (3-4): 206–14. <a href="https://doi.org/10.1016/j.spl.2009.10.008">https://doi.org/10.1016/j.spl.2009.10.008</a>.</p>
</div>
<div id="ref-Carter2016">
<p>Carter, Jane V., Jianmin Pan, Shesh N. Rai, and Susan Galandiuk. 2016. “ROC-ing along: Evaluation and interpretation of receiver operating characteristic curves.” <em>Surgery</em> 159 (6): 1638–45. <a href="https://doi.org/10.1016/j.surg.2015.12.029">https://doi.org/10.1016/j.surg.2015.12.029</a>.</p>
</div>
<div id="ref-Dolzhenko2014">
<p>Dolzhenko, Egor, and Andrew D Smith. 2014. “Using beta-binomial regression for high-precision differential methylation analysis in multifactor whole-genome bisulfite sequencing experiments.” <em>BMC Bioinformatics</em> 15 (1): 215. <a href="https://doi.org/10.1186/1471-2105-15-215">https://doi.org/10.1186/1471-2105-15-215</a>.</p>
</div>
<div id="ref-Eldar2010">
<p>Eldar, Avigdor, and Michael B. Elowitz. 2010. “Functional roles for noise in genetic circuits.” Nature Publishing Group. <a href="https://doi.org/10.1038/nature09326">https://doi.org/10.1038/nature09326</a>.</p>
</div>
<div id="ref-ElNaqa2018">
<p>El Naqa, Issam, Dan Ruan, Gilmer Valdes, Andre Dekker, Todd McNutt, Yaorong Ge, Q. Jackie Wu, et al. 2018. “Machine learning and modeling: Data, validation, communication challenges.” <em>Medical Physics</em> 45 (10): e834–e840. <a href="https://doi.org/10.1002/mp.12811">https://doi.org/10.1002/mp.12811</a>.</p>
</div>
<div id="ref-Hebestreit2013">
<p>Hebestreit, Katja, Martin Dugas, and Hans-Ulrich Klein. 2013. “Detection of significantly differentially methylated regions in targeted bisulfite sequencing data.” <em>Bioinformatics (Oxford, England)</em> 29 (13): 1647–53. <a href="https://doi.org/10.1093/bioinformatics/btt263">https://doi.org/10.1093/bioinformatics/btt263</a>.</p>
</div>
<div id="ref-Hippenstiel2001">
<p>Hippenstiel, Ralph D. 2001. <em>Detection theory: applications and digital signal processing</em>. CRC Press.</p>
</div>
<div id="ref-lopez2014">
<p>López-Ratón, Mónica, Mar<span class="math inline">\(\backslash\)</span>’<span class="math inline">\(\backslash\)</span>ia Xosé Rodríguez-Álvarez, Carmen Cadarso-Suárez, Francisco Gude-Sampedro, and Others. 2014. “OptimalCutpoints: an R package for selecting optimal cutpoints in diagnostic tests.” <em>Journal of Statistical Software</em> 61 (8): 1–36. <a href="https://www.jstatsoft.org/article/view/v061i08">https://www.jstatsoft.org/article/view/v061i08</a>.</p>
</div>
<div id="ref-Robinson2014">
<p>Robinson, Mark D., Abdullah Kahraman, Charity W. Law, Helen Lindsay, Malgorzata Nowicka, Lukas M. Weber, and Xiaobei Zhou. 2014. “Statistical methods for detecting differentially methylated loci and regions.” <em>Frontiers in Genetics</em> 5 (SEP): 324. <a href="https://doi.org/10.3389/fgene.2014.00324">https://doi.org/10.3389/fgene.2014.00324</a>.</p>
</div>
<div id="ref-Samoilov2006">
<p>Samoilov, Michael S., Gavin Price, and Adam P. Arkin. 2006. “From fluctuations to phenotypes: the physiology of noise.” <em>Science’s STKE : Signal Transduction Knowledge Environment</em> 2006 (366). <a href="https://doi.org/10.1126/stke.3662006re17">https://doi.org/10.1126/stke.3662006re17</a>.</p>
</div>
<div id="ref-Sanchez2016">
<p>Sanchez, Robersy, and Sally A. Mackenzie. 2016. “Information Thermodynamics of Cytosine DNA Methylation.” Edited by Barbara Bardoni. <em>PLOS ONE</em> 11 (3): e0150427. <a href="https://doi.org/10.1371/journal.pone.0150427">https://doi.org/10.1371/journal.pone.0150427</a>.</p>
</div>
<div id="ref-Sanchez2019">
<p>Sanchez, Robersy, Xiaodong Yang, Thomas Maher, and Sally Mackenzie. 2019. “Discrimination of DNA Methylation Signal from Background Variation for Clinical Diagnostics.” <em>Int. J. Mol. Sci.</em> 20 (21): 5343. <a href="https://doi.org/https://doi.org/10.3390/ijms20215343">https://doi.org/https://doi.org/10.3390/ijms20215343</a>.</p>
</div>
<div id="ref-Youden1950">
<p>Youden, W. J. 1950. “Index for rating diagnostic tests.” <em>Cancer</em> 3 (1): 32–35. <a href="https://doi.org/10.1002/1097-0142(1950)3:1%3C32::AID-CNCR2820030106%3E3.0.CO;2-3">https://doi.org/10.1002/1097-0142(1950)3:1&lt;32::AID-CNCR2820030106&gt;3.0.CO;2-3</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
