<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Get started-with Methyl-IT • MethylIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Get started-with Methyl-IT">
<meta property="og:description" content="MethylIT">
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Get started-with Methyl-IT</h1>
                        <h4 data-toc-skip class="author">Robersy
Sanchez</h4>
            <address class="author_afil">
      Department of Biology. Pennsylvania State University, University
Park, PA 16802<br><a class="author_email" href="mailto:#"></a><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">08 February 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/vignettes/MethylIT.Rmd" class="external-link"><code>vignettes/MethylIT.Rmd</code></a></small>
      <div class="hidden name"><code>MethylIT.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>A fast introduction into methylation analysis with Methyl-IT is
      provided here. The methylome analysis implemented in Methyl-IT
      based on a signal-detection and machine-learning approach.
      Vignettes with further examples are available at <a href="https://genomaths.github.io/methylit/" class="uri">https://genomaths.github.io/methylit/</a>.</p>
    </div>
    
<style type="text/css">

.sidebar {
font-size: 9px;
text-indent: 0;
}

</style>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/MethylIT" class="external-link">MethylIT</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="background">Background<a class="anchor" aria-label="anchor" href="#background"></a>
</h2>
<p>MethylIT is an R package for methylome analysis based on a
signal-detection machine learning approach (SD-ML). This approach is
postulated to provide greater sensitivity for resolving true signal from
the methylation background within the methylome <span class="citation">(Sanchez and Mackenzie 2016; Sanchez et al.
2019)</span>. Because the biological signal created within the dynamic
methylome environment characteristic of plants is not free from
background noise, the approach, designated Methyl-IT, includes the
application of signal detection theory <span class="citation">(Greiner,
Pfeiffer, and Smith 2000; Carter et al. 2016; Harpaz et al. 2013; Kruspe
et al. 2017)</span>. A basic requirement for the application of signal
detection is the knowledge of the probability distribution of the
background noise, which is used as null hypothesis in the detection of
the methylation signal. This probability distribution is, in general, a
member of generalized gamma distribution family, and it can be deduced
on a statistical mechanical/thermodynamics basis from DNA methylation
induced by thermal fluctuations <span class="citation">(Sanchez and
Mackenzie 2016)</span>.</p>
</div>
<div class="section level2">
<h2 id="data-sets">Data sets<a class="anchor" aria-label="anchor" href="#data-sets"></a>
</h2>
<p>Methyl-IT uses the read counts of methylation called derived from
whole-genome bisulfite sequencing (WGBS). Methylome datasets of WGBS are
available at Gene Expression Omnibus (GEO DataSets). For the current
example, we just load RData files containing the read-counts from
several samples. Please, notice that we are using here just a toy data
set, which permits us to move us fast on Methyl-IT pipeline.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## ===== Download methylation data from PSU's GitLab ========</span></span>
<span><span class="va">url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://git.psu.edu/genomath/MethylIT_examples"</span>,</span>
<span>            <span class="st">"/raw/master/MethylIT_testing/"</span>,</span>
<span>            <span class="st">"memoryLine_100k_samples_cg.RData"</span><span class="op">)</span></span>
<span><span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".RData"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="va">url</span>, destfile <span class="op">=</span> <span class="va">temp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">temp</span>, <span class="va">url</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">meth_samples</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "W_3_1" "W_3_2" "W_3_3" "W_3_4" "W_3_5" "M_3_1" "M_3_2" "M_3_3" "M_3_4"</span></span>
<span><span class="co">#&gt; [10] "M_3_5"</span></span></code></pre></div>
<div class="section level3">
<h3 id="gene-annotation">Gene annotation<a class="anchor" aria-label="anchor" href="#gene-annotation"></a>
</h3>
<p>Download <em>Arabidopsis thaliana</em> TAIR10 gene annotation
data</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://git.psu.edu/genomath/MethylIT_examples"</span>,</span>
<span>            <span class="st">"/raw/master/MethylIT_testing/"</span>,</span>
<span>            <span class="st">"Arabidopsis_thaliana.TAIR10.38_only_genes.RData"</span><span class="op">)</span></span>
<span><span class="va">temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".RData"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="va">url</span>, destfile <span class="op">=</span> <span class="va">temp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">temp</span>, <span class="va">url</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="the-reference-sample">The reference sample<a class="anchor" aria-label="anchor" href="#the-reference-sample"></a>
</h3>
<p>To perform the comparison between the uncertainty of methylation
levels from each group of individuals, control <span class="math inline">\((c)\)</span> and treatment <span class="math inline">\((t)\)</span>, we should estimate the uncertainty
variation with respect to the same reference individual on the mentioned
metric space. The reason to measure the uncertainty variation with
respect to the same reference resides in that even sibling individuals
follow an independent ontogenetic development and, consequently, their
developments follow independent stochastic processes. The reference
individual is estimated with function <a href="https://is.gd/qR79kW" class="external-link">poolFromGRlist</a>:</p>
<p>The centroid (of methylation levels) of the group control is used as
reference samples. Notice that <a href="https://is.gd/qR79kW" class="external-link">poolFromGRlist</a> function only uses
numerical data. Hence, we must specify the columns carrying the
read-count data.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"W"</span>, <span class="fu">names</span>(meth_samples))</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="fu">poolFromGRlist</span>(<span class="at">LR =</span> meth_samples[idx], <span class="at">stat =</span> <span class="st">"mean"</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; *** Building a unique GRanges object from the list...</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  *** Building coordinates for the new GRanges object ...</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  *** Strand information is preserved ...</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="er">|</span>                                                                      <span class="er">|</span>   <span class="dv">0</span>%</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==============</span>                                                        <span class="er">|</span>  <span class="dv">20</span>%</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  <span class="er">|============================</span>                                          <span class="er">|</span>  <span class="dv">40</span>%</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==========================================</span>                            <span class="er">|</span>  <span class="dv">60</span>%</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>  <span class="er">|========================================================</span>              <span class="er">|</span>  <span class="dv">80</span>%</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================================================|</span> <span class="dv">100</span>%</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  *** Sorting by chromosomes and start positions...</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; *** Building a virtual methylome...</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="estimation-of-information-divergences">Estimation of information divergences<a class="anchor" aria-label="anchor" href="#estimation-of-information-divergences"></a>
</h2>
<p>To perform the comparison between the uncertainty of methylation
levels from each group of individuals, control <span class="math inline">\((c)\)</span> and treatment <span class="math inline">\((t)\)</span>, the divergence between the
methylation levels of each individual is estimated with respect to the
reference sample. This is the step with the highest consumption of
computing time.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/date.html" class="external-link">date</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Wed Feb  8 18:36:18 2023"</span></span>
<span><span class="va">ptm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/proc.time.html" class="external-link">proc.time</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ml_samples</span> <span class="op">&lt;-</span> <span class="va">meth_samples</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"W_3_1"</span>,<span class="st">"W_3_2"</span>,<span class="st">"W_3_3"</span>,<span class="st">"W_3_4"</span>,<span class="st">"W_3_5"</span>, </span>
<span>                             <span class="st">"M_3_1"</span>,<span class="st">"M_3_2"</span>,<span class="st">"M_3_3"</span>,<span class="st">"M_3_4"</span>,<span class="st">"M_3_5"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">ml_hd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/estimateDivergence.html">estimateDivergence</a></span><span class="op">(</span>ref <span class="op">=</span> <span class="va">ref</span>,</span>
<span>                            indiv <span class="op">=</span> <span class="va">ml_samples</span>,</span>
<span>                            Bayesian <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            min.coverage <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">4</span><span class="op">)</span>,</span>
<span>                            and.min.cov <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                            min.meth <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                            high.coverage <span class="op">=</span> <span class="fl">500</span>,</span>
<span>                            percentile <span class="op">=</span> <span class="fl">0.999</span>,</span>
<span>                            verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/proc.time.html" class="external-link">proc.time</a></span><span class="op">(</span><span class="op">)</span> <span class="op">-</span> <span class="va">ptm</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">/</span><span class="fl">60</span> <span class="co"># in min</span></span>
<span><span class="co">#&gt; elapsed </span></span>
<span><span class="co">#&gt; 0.15765</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/date.html" class="external-link">date</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Wed Feb  8 18:36:27 2023"</span></span>
<span><span class="va">ml_hd</span><span class="op">$</span><span class="va">M_3_1</span></span>
<span><span class="co">#&gt; GRanges object with 17782 ranges and 9 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames    ranges strand |        c1        t1        c2        t2</span></span>
<span><span class="co">#&gt;              &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;       [1]        1       109      + |         3         0         4         0</span></span>
<span><span class="co">#&gt;       [2]        1       115      + |         3         0         4         0</span></span>
<span><span class="co">#&gt;       [3]        1       161      + |         4         2         5         1</span></span>
<span><span class="co">#&gt;       [4]        1       310      + |         6         6         4        12</span></span>
<span><span class="co">#&gt;       [5]        1       311      - |         1         3         3         2</span></span>
<span><span class="co">#&gt;       ...      ...       ...    ... .       ...       ...       ...       ...</span></span>
<span><span class="co">#&gt;   [17778]       Pt     84348      - |         1       210         2       186</span></span>
<span><span class="co">#&gt;   [17779]       Pt     84350      + |         2       159         0       133</span></span>
<span><span class="co">#&gt;   [17780]       Pt     84351      - |         1       202         2       175</span></span>
<span><span class="co">#&gt;   [17781]       Pt     84353      + |         1       153         0       127</span></span>
<span><span class="co">#&gt;   [17782]       Pt     84354      - |         1       194         0       164</span></span>
<span><span class="co">#&gt;                   p1         p2          TV      bay.TV       hdiv</span></span>
<span><span class="co">#&gt;            &lt;numeric&gt;  &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;  &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;       [1]   0.855530   0.626396    0.000000  -0.2291345 0.31665411</span></span>
<span><span class="co">#&gt;       [2]   0.855530   0.626396    0.000000  -0.2291345 0.31665411</span></span>
<span><span class="co">#&gt;       [3]   0.628985   0.598198    0.166667  -0.0307872 0.00699817</span></span>
<span><span class="co">#&gt;       [4]   0.491634   0.230045   -0.250000  -0.2615884 1.12163729</span></span>
<span><span class="co">#&gt;       [5]   0.274564   0.422201    0.350000   0.1476369 0.13205713</span></span>
<span><span class="co">#&gt;       ...        ...        ...         ...         ...        ...</span></span>
<span><span class="co">#&gt;   [17778] 0.00636821 0.01237291  0.00589896  0.00600470   0.199262</span></span>
<span><span class="co">#&gt;   [17779] 0.01451081 0.00266827 -0.01242236 -0.01184254   0.699579</span></span>
<span><span class="co">#&gt;   [17780] 0.00661804 0.01312918  0.00637333  0.00651113   0.211982</span></span>
<span><span class="co">#&gt;   [17781] 0.00871135 0.00279145 -0.00649351 -0.00591989   0.231223</span></span>
<span><span class="co">#&gt;   [17782] 0.00688828 0.00217286 -0.00512821 -0.00471543   0.238154</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>Function <a href="https://is.gd/0ViBO9" class="external-link">estimateDivergence</a>
returns a list of GRanges objects with the four columns of counts, the
information divergence, and additional columns:</p>
<ol style="list-style-type: decimal">
<li>The original matrix of methylated (<span class="math inline">\(c_i\)</span>) and unmethylated (<span class="math inline">\(t_i\)</span>) read counts from control (<span class="math inline">\(i=1\)</span>) and treatment (<span class="math inline">\(i=2\)</span>) samples.</li>
<li>“p1” and “p2”: methylation levels for control and treatment,
respectively.</li>
<li>“bay.TV”: total variation <span class="math inline">\(TV = p_2 -
p_1\)</span>.</li>
<li>“TV”: total variation based on simple counts: <span class="math inline">\(TV=c_1/(c_1+t_1)-c_2/(c_2+t_2)\)</span>.</li>
<li>“hdiv”: Hellinger divergence.</li>
</ol>
<p>Observe that the absolute values <span class="math inline">\(|TV|\)</span> is an information divergence as
well, which is known as total variation distance (widely used in
information theory).</p>
<div class="section level3">
<h3 id="descriptive-statistical-analysis">
<em>Descriptive statistical analysis</em><a class="anchor" aria-label="anchor" href="#descriptive-statistical-analysis"></a>
</h3>
<p>A descriptive analysis on the distribution of Hellinger divergences
of methylation levels is recommended. After the raw data filtering
introduced with <a href="https://is.gd/0ViBO9" class="external-link">estimateDivergence</a>
function, some coverage issue could be detected. Notice that Methyl-IT
pipeline assumes that the data quality is good enough.</p>
<p>Like in any other statistical analysis, descriptive statistical
analysis help us to detect potential issues in the raw data. It is the
user responsibility to perform quality check of his/her dataset before
start to apply Methyl-IT. Nevertheless, the quality checking of the raw
data is not perfect. So, it is healthy to sees for potential issues.</p>
<p>This is an optional step not included in Methyl-IT. It is up to the
user to verify the data quality.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Critical values</span></span>
<span><span class="va">critical.val</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">ml_hd</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">hd.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">hdiv</span>, <span class="fl">0.95</span><span class="op">)</span></span>
<span>    <span class="va">tv.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">bay.TV</span><span class="op">)</span>, <span class="fl">0.95</span><span class="op">)</span></span>
<span>    <span class="va">btv.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">TV</span><span class="op">)</span>, <span class="fl">0.95</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>tv <span class="op">=</span> <span class="va">tv.95</span>, hd <span class="op">=</span> <span class="va">hd.95</span>, bay.TV <span class="op">=</span> <span class="va">btv.95</span><span class="op">)</span></span>
<span>    <span class="op">)</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">critical.val</span></span>
<span><span class="co">#&gt;          tv.95%   hd.95% bay.TV.95%</span></span>
<span><span class="co">#&gt; W_3_1 0.3119338 1.003136  0.2410714</span></span>
<span><span class="co">#&gt; W_3_2 0.3359854 1.100139  0.2500000</span></span>
<span><span class="co">#&gt; W_3_3 0.2881010 1.005780  0.2142857</span></span>
<span><span class="co">#&gt; W_3_4 0.3177533 1.041646  0.2380952</span></span>
<span><span class="co">#&gt; W_3_5 0.2813523 1.075218  0.2090909</span></span>
<span><span class="co">#&gt; M_3_1 0.3806664 1.605878  0.3201389</span></span>
<span><span class="co">#&gt; M_3_2 0.3711565 1.649375  0.3000000</span></span>
<span><span class="co">#&gt; M_3_3 0.4058102 1.734889  0.3333333</span></span>
<span><span class="co">#&gt; M_3_4 0.2175861 1.446732  0.2803030</span></span>
<span><span class="co">#&gt; M_3_5 0.2281000 1.120914  0.2976190</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="quality-checking-of-the-coverage">Quality checking of the coverage<a class="anchor" aria-label="anchor" href="#quality-checking-of-the-coverage"></a>
</h3>
<p>After the raw data filtering introduced with <a href="https://is.gd/0ViBO9" class="external-link">estimateDivergence</a> function, some
coverage issue could be detected.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">ml_hd</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>                <span class="va">cov</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">c2</span> <span class="op">+</span> <span class="va">x</span><span class="op">$</span><span class="va">t2</span></span>
<span>                <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">cov</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">covr</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">q60</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.6</span><span class="op">)</span></span>
<span>    <span class="va">q9999</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.9999</span><span class="op">)</span></span>
<span>    <span class="va">idx1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="va">q60</span><span class="op">)</span></span>
<span>    <span class="va">idx2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span></span>
<span>    <span class="va">q95</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">0.95</span><span class="op">)</span></span>
<span>    <span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">idx1</span>, <span class="va">idx2</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span>,</span>
<span>        <span class="va">q60</span>, <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.95</span>, <span class="fl">0.99</span>, <span class="fl">0.999</span>, <span class="fl">0.9999</span><span class="op">)</span><span class="op">)</span>,</span>
<span>        num.siteGreater_8 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="fl">8</span><span class="op">)</span>,</span>
<span>        q60_to_500 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;=</span> <span class="va">q60</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">(</span><span class="va">x</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span>,</span>
<span>        num.siteGreater_500 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">500</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;       Min. 1st Qu. Median Mean 3rd Qu. Max. 60%     95%     99%    99.9%</span></span>
<span><span class="co">#&gt; W_3_1    0       8     20  273     494 3933  52 1203.00 2028.40 3630.120</span></span>
<span><span class="co">#&gt; W_3_2    0       7     21  330     690 2995  53 1450.75 1983.00 2662.000</span></span>
<span><span class="co">#&gt; W_3_3    0       8     22  247     434 4013  52 1059.10 1880.84 3661.840</span></span>
<span><span class="co">#&gt; W_3_4    0       7     20  293     603 3630  53 1240.30 1891.52 3261.898</span></span>
<span><span class="co">#&gt; W_3_5    0       9     21  257     407 4010  52 1149.00 2112.34 3772.608</span></span>
<span><span class="co">#&gt; M_3_1    0       7     18  291     580 3421  49 1259.95 1857.57 3131.095</span></span>
<span><span class="co">#&gt; M_3_2    0       8     18  253     421 3843  38 1155.00 2006.47 3641.188</span></span>
<span><span class="co">#&gt; M_3_3    0       7     16  284     669 2749  36 1193.00 1597.51 2401.004</span></span>
<span><span class="co">#&gt; M_3_4    0       8     19  231     399 3975  39  997.00 1968.61 3681.184</span></span>
<span><span class="co">#&gt; M_3_5    0       7     16  354     838 3398  38 1494.00 1989.78 3039.694</span></span>
<span><span class="co">#&gt;         99.99% num.siteGreater_8 q60_to_500 num.siteGreater_500</span></span>
<span><span class="co">#&gt; W_3_1 3890.672             13546       2713                4414</span></span>
<span><span class="co">#&gt; W_3_2 2879.331             12904       2392                4705</span></span>
<span><span class="co">#&gt; W_3_3 3944.623             13932       3131                4006</span></span>
<span><span class="co">#&gt; W_3_4 3550.063             13386       2520                4675</span></span>
<span><span class="co">#&gt; W_3_5 3990.634             14631       3254                4098</span></span>
<span><span class="co">#&gt; M_3_1 3378.882             13163       2475                4653</span></span>
<span><span class="co">#&gt; M_3_2 3799.164             13577       3241                4014</span></span>
<span><span class="co">#&gt; M_3_3 2688.904             12393       2492                4687</span></span>
<span><span class="co">#&gt; M_3_4 3935.676             13936       3449                3757</span></span>
<span><span class="co">#&gt; M_3_5 3323.190             13105       2494                4771</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="estimation-of-the-best-fitted-probability-distribution-model">Estimation of the best fitted probability distribution model<a class="anchor" aria-label="anchor" href="#estimation-of-the-best-fitted-probability-distribution-model"></a>
</h2>
<p>A basic requirement for the application of signal detection is the
knowledge of the probability distribution of the background noise.
Probability distribution, as a Weibull distribution model, can be
deduced on a statistical mechanical/thermodynamics basis for DNA
methylation induced by thermal fluctuations <span class="citation">(Sanchez and Mackenzie 2016)</span>.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>gof <span class="ot">&lt;-</span> <span class="fu">gofReport</span>(<span class="at">HD =</span> ml_hd,</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">model =</span> <span class="fu">c</span>(<span class="st">"Weibull2P"</span>, <span class="st">"Weibull3P"</span>,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"Gamma2P"</span>, <span class="st">"Gamma3P"</span>),</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">column =</span> <span class="dv">9</span>,</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">output =</span> <span class="st">"all"</span>,</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">alt_models =</span> <span class="cn">TRUE</span>,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">num.cores =</span> 4L,</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">task =</span> 2L,</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  <span class="er">|</span>                                                                      <span class="er">|</span>   <span class="dv">0</span>%</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==================</span>                                                    <span class="er">|</span>  <span class="dv">25</span>%</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================================</span>                                   <span class="er">|</span>  <span class="dv">50</span>%</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====================================================</span>                  <span class="er">|</span>  <span class="dv">75</span>%</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================================================|</span> <span class="dv">100</span>%</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  *** Creating report ...</span></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a>gof<span class="sc">$</span>stats</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;          w2p_AIC w2p_R.Cross.val w3p_AIC w3p_R.Cross.val   g2p_AIC</span></span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; W_3_1  -92604.45       0.9983321     Inf               0 -107319.5</span></span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; W_3_2  -94347.14       0.9985961     Inf               0 -109885.9</span></span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; W_3_3  -96466.50       0.9986545     Inf               0 -114703.9</span></span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; W_3_4  -96546.57       0.9986204     Inf               0 -114453.7</span></span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; W_3_5  -97966.02       0.9985585     Inf               0 -109813.8</span></span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; M_3_1 -106137.75       0.9992455     Inf               0 -127842.1</span></span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; M_3_2 -108524.59       0.9993022     Inf               0 -132209.9</span></span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; M_3_3 -107446.61       0.9993008     Inf               0 -132105.6</span></span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; M_3_4 -124063.33       0.9997121     Inf               0 -119186.2</span></span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; M_3_5 -114904.63       0.9994726     Inf               0 -129744.3</span></span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       g2p_R.Cross.val g3p_AIC g3p_R.Cross.val bestModel</span></span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; W_3_1       0.9992292     Inf               0       g2p</span></span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; W_3_2       0.9993784     Inf               0       g2p</span></span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; W_3_3       0.9995170     Inf               0       g2p</span></span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; W_3_4       0.9994412     Inf               0       g2p</span></span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; W_3_5       0.9991953     Inf               0       g2p</span></span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; M_3_1       0.9997532     Inf               0       g2p</span></span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; M_3_2       0.9997937     Inf               0       g2p</span></span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; M_3_3       0.9998019     Inf               0       g2p</span></span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; M_3_4       0.9995424     Inf               0       w2p</span></span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; M_3_5       0.9997344     Inf               0       g2p</span></span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a>gof<span class="sc">$</span>bestModel</span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       W_3_1       W_3_2       W_3_3       W_3_4       W_3_5       M_3_1 </span></span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   "Gamma2P"   "Gamma2P"   "Gamma2P"   "Gamma2P"   "Gamma2P"   "Gamma2P" </span></span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       M_3_2       M_3_3       M_3_4       M_3_5 </span></span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   "Gamma2P"   "Gamma2P" "Weibull2P"   "Gamma2P"</span></span></code></pre></div>
<div class="section level3">
<h3 id="histogram-and-probability-density-distribution-plot">Histogram and probability density distribution plot<a class="anchor" aria-label="anchor" href="#histogram-and-probability-density-distribution-plot"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfcol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">WT1</span> <span class="op">=</span> <span class="va">ml_hd</span><span class="op">$</span><span class="va">W_3_1</span><span class="op">$</span><span class="va">hdiv</span></span>
<span><span class="va">pars</span> <span class="op">=</span> <span class="fu"><a href="../reference/coef.html">coef</a></span><span class="op">(</span><span class="va">gof</span><span class="op">$</span><span class="va">nlms</span><span class="op">$</span><span class="va">W_3_1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">WT1</span>, <span class="fl">90</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">1</span>, family <span class="op">=</span> <span class="st">"serif"</span>,</span>
<span>     panel.first<span class="op">=</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, pch<span class="op">=</span><span class="fl">16</span>, cex<span class="op">=</span><span class="fl">1e6</span>, col<span class="op">=</span><span class="st">"grey95"</span><span class="op">)</span></span>
<span>       <span class="fu"><a href="https://rdrr.io/r/graphics/grid.html" class="external-link">grid</a></span><span class="op">(</span>col<span class="op">=</span><span class="st">"white"</span>, lty <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">}</span>, family <span class="op">=</span> <span class="st">"serif"</span>, col <span class="op">=</span> <span class="st">"cyan1"</span>,</span>
<span>     border <span class="op">=</span> <span class="st">"deepskyblue"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">2</span>, by <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">x1</span>, <span class="fu"><a href="https://rdrr.io/r/stats/GammaDist.html" class="external-link">dgamma</a></span><span class="op">(</span><span class="va">x1</span>, shape <span class="op">=</span> <span class="va">pars</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, scale <span class="op">=</span> <span class="va">pars</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">WT2</span> <span class="op">=</span> <span class="va">ml_hd</span><span class="op">$</span><span class="va">W_3_2</span><span class="op">$</span><span class="va">hdiv</span></span>
<span><span class="va">pars</span> <span class="op">=</span> <span class="fu"><a href="../reference/coef.html">coef</a></span><span class="op">(</span><span class="va">gof</span><span class="op">$</span><span class="va">nlms</span><span class="op">$</span><span class="va">W_3_2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">WT2</span>, <span class="fl">90</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">1</span>, family <span class="op">=</span> <span class="st">"serif"</span>,</span>
<span>     panel.first<span class="op">=</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, pch<span class="op">=</span><span class="fl">16</span>, cex<span class="op">=</span><span class="fl">1e6</span>, col<span class="op">=</span><span class="st">"grey95"</span><span class="op">)</span></span>
<span>       <span class="fu"><a href="https://rdrr.io/r/graphics/grid.html" class="external-link">grid</a></span><span class="op">(</span>col<span class="op">=</span><span class="st">"white"</span>, lty <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">}</span>, family <span class="op">=</span> <span class="st">"serif"</span>, col <span class="op">=</span> <span class="st">"cyan1"</span>,</span>
<span>     border <span class="op">=</span> <span class="st">"deepskyblue"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">2</span>, by <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">x1</span>, <span class="fu"><a href="https://rdrr.io/r/stats/GammaDist.html" class="external-link">dgamma</a></span><span class="op">(</span><span class="va">x1</span>, shape <span class="op">=</span> <span class="va">pars</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, scale <span class="op">=</span> <span class="va">pars</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">M1</span> <span class="op">=</span> <span class="va">ml_hd</span><span class="op">$</span><span class="va">M_3_1</span><span class="op">$</span><span class="va">hdiv</span></span>
<span><span class="va">pars</span> <span class="op">=</span> <span class="fu"><a href="../reference/coef.html">coef</a></span><span class="op">(</span><span class="va">gof</span><span class="op">$</span><span class="va">nlms</span><span class="op">$</span><span class="va">M_3_1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">M1</span>, <span class="fl">90</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">1</span>, family <span class="op">=</span> <span class="st">"serif"</span>,</span>
<span>     panel.first<span class="op">=</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, pch<span class="op">=</span><span class="fl">16</span>, cex<span class="op">=</span><span class="fl">1e6</span>, col<span class="op">=</span><span class="st">"grey95"</span><span class="op">)</span></span>
<span>       <span class="fu"><a href="https://rdrr.io/r/graphics/grid.html" class="external-link">grid</a></span><span class="op">(</span>col<span class="op">=</span><span class="st">"white"</span>, lty <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">}</span>, family <span class="op">=</span> <span class="st">"serif"</span>, col <span class="op">=</span> <span class="st">"cyan1"</span>,</span>
<span>     border <span class="op">=</span> <span class="st">"deepskyblue"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">4.5</span>, by <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">x1</span>, <span class="fu"><a href="https://rdrr.io/r/stats/GammaDist.html" class="external-link">dgamma</a></span><span class="op">(</span><span class="va">x1</span>, shape <span class="op">=</span> <span class="va">pars</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, scale <span class="op">=</span> <span class="va">pars</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">M2</span> <span class="op">=</span> <span class="va">ml_hd</span><span class="op">$</span><span class="va">M_3_2</span><span class="op">$</span><span class="va">hdiv</span></span>
<span><span class="va">pars</span> <span class="op">=</span> <span class="fu"><a href="../reference/coef.html">coef</a></span><span class="op">(</span><span class="va">gof</span><span class="op">$</span><span class="va">nlms</span><span class="op">$</span><span class="va">M_3_2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">M2</span>, <span class="fl">90</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">1</span>, family <span class="op">=</span> <span class="st">"serif"</span>,</span>
<span>     panel.first<span class="op">=</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, pch<span class="op">=</span><span class="fl">16</span>, cex<span class="op">=</span><span class="fl">1e6</span>, col<span class="op">=</span><span class="st">"grey95"</span><span class="op">)</span></span>
<span>       <span class="fu"><a href="https://rdrr.io/r/graphics/grid.html" class="external-link">grid</a></span><span class="op">(</span>col<span class="op">=</span><span class="st">"white"</span>, lty <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">}</span>, family <span class="op">=</span> <span class="st">"serif"</span>, col <span class="op">=</span> <span class="st">"cyan1"</span>,</span>
<span>     border <span class="op">=</span> <span class="st">"deepskyblue"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">4</span>, by <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">x1</span>, <span class="fu"><a href="https://rdrr.io/r/stats/GammaDist.html" class="external-link">dgamma</a></span><span class="op">(</span><span class="va">x1</span>, shape <span class="op">=</span> <span class="va">pars</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, scale <span class="op">=</span> <span class="va">pars</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<p><img src="MethylIT_files/figure-html/graph-1.png" width="1344"></p>
</div>
</div>
<div class="section level2">
<h2 id="signal-detection">Signal detection<a class="anchor" aria-label="anchor" href="#signal-detection"></a>
</h2>
<p>The information thermodynamics-based approach is postulated to
provide greater sensitivity for resolving true signal from the
thermodynamic background within the methylome <span class="citation">(Sanchez and Mackenzie 2016)</span>. Since the
biological signal created within the dynamic methylome environment
characteristic of living organisms is not free from background noise,
the approach, designated Methyl-IT, includes the application of signal
detection theory <span class="citation">(Greiner, Pfeiffer, and Smith
2000; Carter et al. 2016; Harpaz et al. 2013; Kruspe et al.
2017)</span>. Signal detection is a critical step to increase
sensitivity and resolution of methylation signal by reducing the
signal-to-noise ratio and objectively controlling the false positive
rate and prediction accuracy/risk.</p>
<div class="section level3">
<h3 id="potential-methylation-signal">
<em>Potential methylation signal</em><a class="anchor" aria-label="anchor" href="#potential-methylation-signal"></a>
</h3>
<p>The first estimation in our signal detection step is the
identification of the cytosine sites carrying potential methylation
signal <span class="math inline">\(PS\)</span>. The methylation
regulatory signal does not hold the theoretical distribution and,
consequently, for a given level of significance <span class="math inline">\(\alpha\)</span> (Type I error probability, e.g.
<span class="math inline">\(\alpha = 0.05\)</span>), cytosine positions
<span class="math inline">\(k\)</span> with information divergence <span class="math inline">\(H_k &gt;= H_{\alpha = 0.05}\)</span> can be
selected as sites carrying potential signals <span class="math inline">\(PS\)</span>. The value of <span class="math inline">\(\alpha\)</span> can be specified. For example,
potential signals with <span class="math inline">\(H_k &gt; H_{\alpha =
0.01}\)</span> can be selected. For each sample, cytosine sites are
selected based on the corresponding fitted theoretical distribution
model estimated in the previous step.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">ml_hd</span>,</span>
<span>                        nlms <span class="op">=</span> <span class="va">gof</span><span class="op">$</span><span class="va">nlms</span>,</span>
<span>                        div.col <span class="op">=</span> <span class="fl">9L</span>,</span>
<span>                        tv.col <span class="op">=</span> <span class="fl">8L</span>,</span>
<span>                        tv.cut <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>                        dist.name <span class="op">=</span> <span class="va">gof</span><span class="op">$</span><span class="va">bestModel</span><span class="op">)</span></span>
<span><span class="va">ps</span><span class="op">$</span><span class="va">M_3_1</span></span>
<span><span class="co">#&gt; GRanges object with 558 ranges and 10 metadata columns:</span></span>
<span><span class="co">#&gt;         seqnames    ranges strand |        c1        t1        c2        t2</span></span>
<span><span class="co">#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;     [1]        1       511      + |        10         5         4         8</span></span>
<span><span class="co">#&gt;     [2]        1      5459      + |         6         2         2         5</span></span>
<span><span class="co">#&gt;     [3]        1     17641      - |         5         1         1         2</span></span>
<span><span class="co">#&gt;     [4]        1     18106      - |         9         0         2         1</span></span>
<span><span class="co">#&gt;     [5]        1     24196      + |         6        14        16         0</span></span>
<span><span class="co">#&gt;     ...      ...       ...    ... .       ...       ...       ...       ...</span></span>
<span><span class="co">#&gt;   [554]        5     93916      - |         0         6         5         3</span></span>
<span><span class="co">#&gt;   [555]        5     94024      + |         0        10         3         2</span></span>
<span><span class="co">#&gt;   [556]        5     94054      + |         7         2         0         4</span></span>
<span><span class="co">#&gt;   [557]        5     94055      - |         5         1         1         5</span></span>
<span><span class="co">#&gt;   [558]        5     94058      - |         6         0         4         2</span></span>
<span><span class="co">#&gt;                p1        p2        TV    bay.TV      hdiv       wprob</span></span>
<span><span class="co">#&gt;         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;     [1]  0.650294  0.291535 -0.333333 -0.358759   1.91758 3.24985e-02</span></span>
<span><span class="co">#&gt;     [2]  0.712218  0.237111 -0.464286 -0.475107   2.04140 2.70927e-02</span></span>
<span><span class="co">#&gt;     [3]  0.773595  0.228468 -0.500000 -0.545127   1.64587 4.86429e-02</span></span>
<span><span class="co">#&gt;     [4]  0.942954  0.396115 -0.333333 -0.546839   2.32267 1.79839e-02</span></span>
<span><span class="co">#&gt;     [5]  0.303585  0.862793  0.700000  0.559208   6.73007 3.88975e-05</span></span>
<span><span class="co">#&gt;     ...       ...       ...       ...       ...       ...         ...</span></span>
<span><span class="co">#&gt;   [554] 0.0505441 0.4890867  0.625000  0.438543   2.30405  0.01847590</span></span>
<span><span class="co">#&gt;   [555] 0.0320215 0.4222008  0.600000  0.390179   2.10990  0.02450879</span></span>
<span><span class="co">#&gt;   [556] 0.7412423 0.0520884 -0.777778 -0.689154   4.10998  0.00142952</span></span>
<span><span class="co">#&gt;   [557] 0.7735955 0.1520139 -0.666667 -0.621582   3.06476  0.00620988</span></span>
<span><span class="co">#&gt;   [558] 0.9182058 0.4866520 -0.333333 -0.431554   1.77271  0.04026507</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cutpoint-estimation--dmps">Cutpoint estimation. DMPs<a class="anchor" aria-label="anchor" href="#cutpoint-estimation--dmps"></a>
</h3>
<p>Laws of statistical physics can account for background methylation, a
response to thermal fluctuations that presumably functions in DNA
stability <span class="citation">(Sanchez and Mackenzie 2016; Sanchez et
al. 2019)</span>. True signal is detected based on the optimal cutpoint
<span class="citation">(López-Ratón et al. 2014)</span>.</p>
<p>The need for the application of (what is now known as) signal
detection in cancer research was pointed out by Youden in the midst of
the last century <span class="citation">(Youden 1950)</span>. In the
next example, the simple cutpoint estimation available in Methyl-IT is
based on the application of <a href="https://en.wikipedia.org/wiki/Youden%27s_J_statistic" class="external-link">Youden
index</a> <span class="citation">(Youden 1950)</span>. Although
cutpoints are estimated for a single variable, the classification
performance can be evaluated for several variables and applying
different model classifiers. A optimal cutpoint distinguishes disease
stages from healthy individual. The performance of this classification
is given in the output of function <a href="https://is.gd/RR9iWY" class="external-link">estimateCutPoint</a>.</p>
<p>A model classifier can be requested for further predictions and its
classification performance is also provided. Below, the selected model
classifier is a quadratic discriminant analysis (QDA)
(<em>classifier1</em> = “qda”, <em>clas.perf</em> = TRUE). Four
predictor variables are available: the Hellinger divergence of
methylation levels (<em>hdiv</em>), total variation distance
(<em>TV</em>, absolute difference of methylation levels), relative
position of cytosine site in the chromosome (<em>pos</em>), and the
logarithm base two of the probability to observe a Hellinger divergence
value <span class="math inline">\(H\)</span> greater than the critical
value <span class="math inline">\(H_{\alpha = 0.05}\)</span> (values
given as probabilities in object PS, <em>wprob</em>).</p>
<p>Notice that the cutpoint can be estimated for any of the two
currently available information divergences: <em>Hellinger
divergence</em> (<em>div.col</em> = 9) or the <em>total</em>
<em>variation distance</em> (with Bayesian correction, <em>div.col</em>
= 8).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Cutpoint estimated using a model classifier</span></span>
<span><span class="va">cutpoint</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">ps</span>,</span>
<span>                            simple <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                            div.col <span class="op">=</span> <span class="fl">9</span>,</span>
<span>                            control.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"W_3_1"</span>,<span class="st">"W_3_2"</span>,<span class="st">"W_3_3"</span>,</span>
<span>                                              <span class="st">"W_3_4"</span>,<span class="st">"W_3_5"</span><span class="op">)</span>,</span>
<span>                            treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"M_3_1"</span>,<span class="st">"M_3_2"</span>,<span class="st">"M_3_3"</span>,</span>
<span>                                                <span class="st">"M_3_4"</span>,<span class="st">"M_3_5"</span><span class="op">)</span>,</span>
<span>                            column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                        wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>                            classifier1 <span class="op">=</span> <span class="st">"logistic"</span>, n.pc <span class="op">=</span> <span class="fl">4</span>,</span>
<span>                            classifier2 <span class="op">=</span> <span class="st">"pca.qda"</span>,</span>
<span>                            center <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            clas.perf <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                            verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">cutpoint</span><span class="op">$</span><span class="va">cutpoint</span></span>
<span><span class="co">#&gt; [1] 1.018108</span></span>
<span><span class="va">cutpoint</span><span class="op">$</span><span class="va">testSetPerformance</span></span>
<span><span class="co">#&gt; Confusion Matrix and Statistics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Reference</span></span>
<span><span class="co">#&gt; Prediction  CT  TT</span></span>
<span><span class="co">#&gt;         CT 391   0</span></span>
<span><span class="co">#&gt;         TT  22 696</span></span>
<span><span class="co">#&gt;                                           </span></span>
<span><span class="co">#&gt;                Accuracy : 0.9802          </span></span>
<span><span class="co">#&gt;                  95% CI : (0.9701, 0.9875)</span></span>
<span><span class="co">#&gt;     No Information Rate : 0.6276          </span></span>
<span><span class="co">#&gt;     P-Value [Acc &gt; NIR] : &lt; 2.2e-16       </span></span>
<span><span class="co">#&gt;                                           </span></span>
<span><span class="co">#&gt;                   Kappa : 0.9571          </span></span>
<span><span class="co">#&gt;                                           </span></span>
<span><span class="co">#&gt;  Mcnemar's Test P-Value : 7.562e-06       </span></span>
<span><span class="co">#&gt;                                           </span></span>
<span><span class="co">#&gt;             Sensitivity : 1.0000          </span></span>
<span><span class="co">#&gt;             Specificity : 0.9467          </span></span>
<span><span class="co">#&gt;          Pos Pred Value : 0.9694          </span></span>
<span><span class="co">#&gt;          Neg Pred Value : 1.0000          </span></span>
<span><span class="co">#&gt;              Prevalence : 0.6276          </span></span>
<span><span class="co">#&gt;          Detection Rate : 0.6276          </span></span>
<span><span class="co">#&gt;    Detection Prevalence : 0.6474          </span></span>
<span><span class="co">#&gt;       Balanced Accuracy : 0.9734          </span></span>
<span><span class="co">#&gt;                                           </span></span>
<span><span class="co">#&gt;        'Positive' Class : TT              </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="va">cutpoint</span><span class="op">$</span><span class="va">testSetModel.FDR</span></span>
<span><span class="co">#&gt; [1] 0</span></span></code></pre></div>
<p>Differentially methylated positions (DMPs) can be now identified
using <a href="https://is.gd/7cVEEd" class="external-link">selectDIMP</a> function</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dmps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectDIMP.html">selectDIMP</a></span><span class="op">(</span><span class="va">ps</span>, div.col <span class="op">=</span> <span class="fl">9</span>, cutpoint <span class="op">=</span> <span class="va">cutpoint</span><span class="op">$</span><span class="va">cutpoint</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>dmps <span class="op">=</span><span class="fu"><a href="../reference/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">dmps</span>, <span class="va">length</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;       dmps</span></span>
<span><span class="co">#&gt; W_3_1  354</span></span>
<span><span class="co">#&gt; W_3_2  342</span></span>
<span><span class="co">#&gt; W_3_3  303</span></span>
<span><span class="co">#&gt; W_3_4  383</span></span>
<span><span class="co">#&gt; W_3_5  277</span></span>
<span><span class="co">#&gt; M_3_1  558</span></span>
<span><span class="co">#&gt; M_3_2  450</span></span>
<span><span class="co">#&gt; M_3_3  471</span></span>
<span><span class="co">#&gt; M_3_4  161</span></span>
<span><span class="co">#&gt; M_3_5  252</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="differentially-methylated-genes-dmgs">Differentially methylated genes (DMGs)<a class="anchor" aria-label="anchor" href="#differentially-methylated-genes-dmgs"></a>
</h2>
<p>Differentially methylated genes (DMGs) are estimated from group
comparisons for the number of DMPs on gene-body regions between control
and treatment.</p>
<p>Function <a href="https://is.gd/WQzX5I" class="external-link">getDIMPatGenes</a> is used to
count the number of DMPs at gene-body. Nevertheless, it can be used for
any arbitrary specified genomic region as well. The operation of this
function is based on the ‘<em>findOverlaps</em>’ function from the
‘<em>GenomicRanges</em>’ Bioconductor R package. The
‘<em>findOverlaps</em>’ function has several critical parameters like,
for example, ‘<em>maxgap</em>’, ‘<em>minoverlap</em>’, and
‘<em>ignore.strand</em>’. In our function <a href="https://is.gd/WQzX5I" class="external-link">getDIMPatGenes</a>, except for setting
ignore.strand = TRUE and <em>type = “within”</em>, we preserve the rest
of default ‘findOverlaps’ parameters. In this case, these are important
parameter settings because the local mechanical effect of methylation
changes on a DNA region where a gene is located is independent of the
strand where the gene is encoded. That is, methylation changes located
in any of the two DNA strands inside the gene-body region will affect
the flexibility of the DNA molecule <span class="citation">(Choy et al.
2010; Severin et al. 2011)</span>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span></span>
<span><span class="va">dmps_at_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, GENES <span class="op">=</span> <span class="va">genes</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">dmps_at_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">dmps_at_genes</span>, columns <span class="op">=</span> <span class="fl">2L</span>,</span>
<span>                                ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>, type <span class="op">=</span> <span class="st">"equal"</span>,</span>
<span>                               verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">dmps_at_genes</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">nams</span></span>
<span><span class="va">dmps_at_genes</span></span>
<span><span class="co">#&gt; GRanges object with 70 ranges and 10 metadata columns:</span></span>
<span><span class="co">#&gt;        seqnames      ranges strand |     W_3_1     W_3_2     W_3_3     W_3_4</span></span>
<span><span class="co">#&gt;           &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;    [1]        1   3631-5899      * |         0         0         0         1</span></span>
<span><span class="co">#&gt;    [2]        1   6788-9130      * |         0         0         0         0</span></span>
<span><span class="co">#&gt;    [3]        1 23121-31227      * |        12         8        17         5</span></span>
<span><span class="co">#&gt;    [4]        1 31170-33171      * |         2         0         2         0</span></span>
<span><span class="co">#&gt;    [5]        1 33365-37871      * |         2         0         1         1</span></span>
<span><span class="co">#&gt;    ...      ...         ...    ... .       ...       ...       ...       ...</span></span>
<span><span class="co">#&gt;   [66]        5 49891-51437      * |         0         0         1         0</span></span>
<span><span class="co">#&gt;   [67]        5 53856-56052      * |         1         0         9         3</span></span>
<span><span class="co">#&gt;   [68]        5 61017-63936      * |         2         0         1         0</span></span>
<span><span class="co">#&gt;   [69]        5 84474-86275      * |         0         0         0         0</span></span>
<span><span class="co">#&gt;   [70]        5 91838-95701      * |         4         1         2         1</span></span>
<span><span class="co">#&gt;            W_3_5     M_3_1     M_3_2     M_3_3     M_3_4     M_3_5</span></span>
<span><span class="co">#&gt;        &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;    [1]         0         1         2         1         2         1</span></span>
<span><span class="co">#&gt;    [2]         0         0         0         1         0         1</span></span>
<span><span class="co">#&gt;    [3]         9        11        14        16         9         8</span></span>
<span><span class="co">#&gt;    [4]         2         1         0         0         0         0</span></span>
<span><span class="co">#&gt;    [5]         1         2         0         1         0         0</span></span>
<span><span class="co">#&gt;    ...       ...       ...       ...       ...       ...       ...</span></span>
<span><span class="co">#&gt;   [66]         0         2         2         2         2         2</span></span>
<span><span class="co">#&gt;   [67]         1         7         4         4         0         1</span></span>
<span><span class="co">#&gt;   [68]         1         6         7         2         3         4</span></span>
<span><span class="co">#&gt;   [69]         0         0         0         0         0         1</span></span>
<span><span class="co">#&gt;   [70]         1         8         2         4         5         4</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 5 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<div class="section level3">
<h3 id="setting-the-experiment-design">
<em>Setting the experiment design</em><a class="anchor" aria-label="anchor" href="#setting-the-experiment-design"></a>
</h3>
<p>The experimental design is set with <a href="https://genomaths.github.io/methylit/reference/glmDataSet.html">glmDataSet</a>
function</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">colData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WT"</span>, <span class="st">"WT"</span>, <span class="st">"WT"</span>, <span class="st">"WT"</span>, <span class="st">"WT"</span>,</span>
<span>                                            <span class="st">"ML"</span>, <span class="st">"ML"</span>, <span class="st">"ML"</span>, <span class="st">"ML"</span>, <span class="st">"ML"</span><span class="op">)</span>,</span>
<span>                                        levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WT"</span>, <span class="st">"ML"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                    <span class="va">nams</span>,</span>
<span>                    row.names <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">## A RangedGlmDataSet is created</span></span>
<span><span class="va">ds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps_at_genes</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="testing-for-dmgs">Testing for DMGs<a class="anchor" aria-label="anchor" href="#testing-for-dmgs"></a>
</h3>
<p>DMGs are detected using function <a href="https://genomaths.github.io/methylit/reference/countTest2.html">countTest2</a>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dmgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds</span>, num.cores <span class="op">=</span> <span class="fl">4L</span>,</span>
<span>                    tasks <span class="op">=</span> <span class="fl">2L</span>,</span>
<span>                    minCountPerIndv <span class="op">=</span> <span class="fl">7</span>,</span>
<span>                    CountPerBp <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>                    test <span class="op">=</span> <span class="st">"LRT"</span>,</span>
<span>                    verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; *** Number of GR after filtering counts 3</span></span>
<span><span class="co">#&gt; *** GLM...</span></span>
<span><span class="va">dmgs</span></span>
<span><span class="co">#&gt; GRanges object with 1 range and 18 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames      ranges strand |     W_3_1     W_3_2     W_3_3     W_3_4</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]        3 31145-34670      * |         3         2         3         4</span></span>
<span><span class="co">#&gt;           W_3_5     M_3_1     M_3_2     M_3_3     M_3_4     M_3_5    log2FC</span></span>
<span><span class="co">#&gt;       &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]         3         6         8         9         6         7  0.875469</span></span>
<span><span class="co">#&gt;       scaled.deviance      pvalue          model    adj.pval CT.SignalDensity</span></span>
<span><span class="co">#&gt;             &lt;numeric&gt;   &lt;numeric&gt;    &lt;character&gt;   &lt;numeric&gt;        &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]         67.8239 1.78768e-16 Neg.Binomial.W 1.78768e-16      0.000850822</span></span>
<span><span class="co">#&gt;       TT.SignalDensity SignalDensityVariation</span></span>
<span><span class="co">#&gt;              &lt;numeric&gt;              &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]       0.00204197             0.00119115</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Carter2016" class="csl-entry">
Carter, Jane V., Jianmin Pan, Shesh N. Rai, and Susan Galandiuk. 2016.
<span>“<span class="nocase">ROC-ing along: Evaluation and interpretation
of receiver operating characteristic curves</span>.”</span>
<em>Surgery</em> 159 (6): 1638–45. <a href="https://doi.org/10.1016/j.surg.2015.12.029" class="external-link">https://doi.org/10.1016/j.surg.2015.12.029</a>.
</div>
<div id="ref-Choy2010" class="csl-entry">
Choy, John S., Sijie Wei, Ju Yeon Lee, Song Tan, Steven Chu, and Tae Hee
Lee. 2010. <span>“<span class="nocase">DNA methylation increases
nucleosome compaction and rigidity</span>.”</span> <em>Journal of the
American Chemical Society</em> 132 (6): 1782–83. <a href="https://doi.org/10.1021/ja910264z" class="external-link">https://doi.org/10.1021/ja910264z</a>.
</div>
<div id="ref-Greiner2000" class="csl-entry">
Greiner, M, D Pfeiffer, and R D Smith. 2000. <span>“<span class="nocase">Principles and practical application of the
receiver-operating characteristic analysis for diagnostic
tests</span>.”</span> <em>Preventive Veterinary Medicine</em> 45 (1-2):
23–41. <a href="https://doi.org/10.1016/S0167-5877(00)00115-X" class="external-link">https://doi.org/10.1016/S0167-5877(00)00115-X</a>.
</div>
<div id="ref-Harpaz2013" class="csl-entry">
Harpaz, Rave, William DuMouchel, Paea LePendu, Anna Bauer-Mehren,
Patrick Ryan, and Nigam H Shah. 2013. <span>“<span class="nocase">Performance of Pharmacovigilance Signal Detection
Algorithms for the FDA Adverse Event Reporting System</span>.”</span>
<em>Clin Pharmacol Ther</em> 93 (6): 1–19. <a href="https://doi.org/10.1038/clpt.2013.24.Performance" class="external-link">https://doi.org/10.1038/clpt.2013.24.Performance</a>.
</div>
<div id="ref-Kruspe2017" class="csl-entry">
Kruspe, Sven, David D. Dickey, Kevin T. Urak, Giselle N. Blanco, Matthew
J. Miller, Karen C. Clark, Elliot Burghardt, et al. 2017. <span>“<span class="nocase">Rapid and Sensitive Detection of Breast Cancer Cells in
Patient Blood with Nuclease-Activated Probe Technology</span>.”</span>
<em>Molecular Therapy - Nucleic Acids</em> 8 (September): 542–57. <a href="https://doi.org/10.1016/j.omtn.2017.08.004" class="external-link">https://doi.org/10.1016/j.omtn.2017.08.004</a>.
</div>
<div id="ref-lopez2014" class="csl-entry">
López-Ratón, Mónica, Mar\backslash’\backslashia Xosé Rodríguez-Álvarez,
Carmen Cadarso-Suárez, Francisco Gude-Sampedro, and Others. 2014.
<span>“<span class="nocase">OptimalCutpoints: an R package for selecting
optimal cutpoints in diagnostic tests</span>.”</span> <em>Journal of
Statistical Software</em> 61 (8): 1–36. <a href="https://www.jstatsoft.org/article/view/v061i08" class="external-link">https://www.jstatsoft.org/article/view/v061i08</a>.
</div>
<div id="ref-Sanchez2016" class="csl-entry">
Sanchez, Robersy, and Sally A. Mackenzie. 2016. <span>“<span class="nocase">Information Thermodynamics of Cytosine DNA
Methylation</span>.”</span> Edited by Barbara Bardoni. <em>PLOS ONE</em>
11 (3): e0150427. <a href="https://doi.org/10.1371/journal.pone.0150427" class="external-link">https://doi.org/10.1371/journal.pone.0150427</a>.
</div>
<div id="ref-Sanchez2019" class="csl-entry">
Sanchez, Robersy, Xiaodong Yang, Thomas Maher, and Sally Mackenzie.
2019. <span>“<span class="nocase">Discrimination of DNA Methylation
Signal from Background Variation for Clinical
Diagnostics</span>.”</span> <em>Int. J. Mol. Sci.</em> 20 (21): 5343.
https://doi.org/<a href="https://doi.org/10.3390/ijms20215343" class="external-link">https://doi.org/10.3390/ijms20215343</a>.
</div>
<div id="ref-Severin2011" class="csl-entry">
Severin, Philip M D, Xueqing Zou, Hermann E Gaub, and Klaus Schulten.
2011. <span>“<span class="nocase">Cytosine methylation alters DNA
mechanical properties</span>.”</span> <em>Nucleic Acids Research</em> 39
(20): 8740–51. <a href="https://doi.org/10.1093/nar/gkr578" class="external-link">https://doi.org/10.1093/nar/gkr578</a>.
</div>
<div id="ref-Youden1950" class="csl-entry">
Youden, W. J. 1950. <span>“<span class="nocase">Index for rating
diagnostic tests</span>.”</span> <em>Cancer</em> 3 (1): 32–35. <a href="https://doi.org/10.1002/1097-0142(1950)3:1&lt;32::AID-CNCR2820030106&gt;3.0.CO;2-3" class="external-link">https://doi.org/10.1002/1097-0142(1950)3:1&lt;32::AID-CNCR2820030106&gt;3.0.CO;2-3</a>.
</div>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code> on the system on
which this document was compiled running pandoc 2.19.2:</p>
<pre><code><span><span class="co">#&gt; R version 4.2.2 Patched (2022-11-10 r83330)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 20.04.5 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] MethylIT_0.3.2.6     rtracklayer_1.56.1   GenomicRanges_1.48.0</span></span>
<span><span class="co">#&gt; [4] GenomeInfoDb_1.32.4  IRanges_2.30.1       S4Vectors_0.34.0    </span></span>
<span><span class="co">#&gt; [7] BiocGenerics_0.42.0  BiocStyle_2.24.0    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] colorspace_2.1-0            rjson_0.2.21               </span></span>
<span><span class="co">#&gt;   [3] class_7.3-21                rprojroot_2.0.3            </span></span>
<span><span class="co">#&gt;   [5] XVector_0.36.0              fs_1.6.1                   </span></span>
<span><span class="co">#&gt;   [7] proxy_0.4-27                rstudioapi_0.14            </span></span>
<span><span class="co">#&gt;   [9] listenv_0.9.0               bit64_4.0.5                </span></span>
<span><span class="co">#&gt;  [11] AnnotationDbi_1.58.0        prodlim_2019.11.13         </span></span>
<span><span class="co">#&gt;  [13] fansi_1.0.4                 lubridate_1.9.1            </span></span>
<span><span class="co">#&gt;  [15] codetools_0.2-19            splines_4.2.2              </span></span>
<span><span class="co">#&gt;  [17] cachem_1.0.6                knitr_1.42                 </span></span>
<span><span class="co">#&gt;  [19] jsonlite_1.8.4              pROC_1.18.0                </span></span>
<span><span class="co">#&gt;  [21] Rsamtools_2.12.0            caret_6.0-93               </span></span>
<span><span class="co">#&gt;  [23] annotate_1.74.0             png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [25] httr_1.4.4                  BiocManager_1.30.19        </span></span>
<span><span class="co">#&gt;  [27] compiler_4.2.2              Matrix_1.5-3               </span></span>
<span><span class="co">#&gt;  [29] fastmap_1.1.0               cli_3.6.0                  </span></span>
<span><span class="co">#&gt;  [31] htmltools_0.5.4             tools_4.2.2                </span></span>
<span><span class="co">#&gt;  [33] gtable_0.3.1                glue_1.6.2                 </span></span>
<span><span class="co">#&gt;  [35] GenomeInfoDbData_1.2.8      reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;  [37] dplyr_1.1.0                 Rcpp_1.0.10                </span></span>
<span><span class="co">#&gt;  [39] Biobase_2.56.0              jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [41] pkgdown_2.0.7               vctrs_0.5.2                </span></span>
<span><span class="co">#&gt;  [43] Biostrings_2.64.1           nlme_3.1-162               </span></span>
<span><span class="co">#&gt;  [45] iterators_1.0.14            timeDate_4022.108          </span></span>
<span><span class="co">#&gt;  [47] xfun_0.37                   gower_1.0.1                </span></span>
<span><span class="co">#&gt;  [49] stringr_1.5.0               proto_1.0.0                </span></span>
<span><span class="co">#&gt;  [51] globals_0.16.2              timechange_0.2.0           </span></span>
<span><span class="co">#&gt;  [53] lifecycle_1.0.3             restfulr_0.0.15            </span></span>
<span><span class="co">#&gt;  [55] XML_3.99-0.13               future_1.31.0              </span></span>
<span><span class="co">#&gt;  [57] zlibbioc_1.42.0             MASS_7.3-58.2              </span></span>
<span><span class="co">#&gt;  [59] scales_1.2.1                ipred_0.9-13               </span></span>
<span><span class="co">#&gt;  [61] ragg_1.2.5                  MatrixGenerics_1.8.1       </span></span>
<span><span class="co">#&gt;  [63] parallel_4.2.2              SummarizedExperiment_1.26.1</span></span>
<span><span class="co">#&gt;  [65] yaml_2.3.7                  memoise_2.0.1              </span></span>
<span><span class="co">#&gt;  [67] ggplot2_3.4.0               sass_0.4.5                 </span></span>
<span><span class="co">#&gt;  [69] rpart_4.1.19                RSQLite_2.2.20             </span></span>
<span><span class="co">#&gt;  [71] stringi_1.7.12              highr_0.10                 </span></span>
<span><span class="co">#&gt;  [73] genefilter_1.78.0           BiocIO_1.6.0               </span></span>
<span><span class="co">#&gt;  [75] desc_1.4.2                  randomForest_4.7-1.1       </span></span>
<span><span class="co">#&gt;  [77] foreach_1.5.2               e1071_1.7-13               </span></span>
<span><span class="co">#&gt;  [79] hardhat_1.2.0               BiocParallel_1.30.4        </span></span>
<span><span class="co">#&gt;  [81] lava_1.7.1                  rlang_1.0.6                </span></span>
<span><span class="co">#&gt;  [83] pkgconfig_2.0.3             systemfonts_1.0.4          </span></span>
<span><span class="co">#&gt;  [85] matrixStats_0.63.0          bitops_1.0-7               </span></span>
<span><span class="co">#&gt;  [87] evaluate_0.20               lattice_0.20-45            </span></span>
<span><span class="co">#&gt;  [89] purrr_1.0.1                 GenomicAlignments_1.32.1   </span></span>
<span><span class="co">#&gt;  [91] recipes_1.0.4               bit_4.0.5                  </span></span>
<span><span class="co">#&gt;  [93] tidyselect_1.2.0            parallelly_1.34.0          </span></span>
<span><span class="co">#&gt;  [95] nls2_0.3-3                  plyr_1.8.8                 </span></span>
<span><span class="co">#&gt;  [97] magrittr_2.0.3              bookdown_0.32              </span></span>
<span><span class="co">#&gt;  [99] R6_2.5.1                    generics_0.1.3             </span></span>
<span><span class="co">#&gt; [101] DBI_1.1.3                   DelayedArray_0.22.0        </span></span>
<span><span class="co">#&gt; [103] pillar_1.8.1                withr_2.5.0                </span></span>
<span><span class="co">#&gt; [105] KEGGREST_1.36.3             survival_3.5-0             </span></span>
<span><span class="co">#&gt; [107] RCurl_1.98-1.10             nnet_7.3-18                </span></span>
<span><span class="co">#&gt; [109] tibble_3.1.8                future.apply_1.10.0        </span></span>
<span><span class="co">#&gt; [111] crayon_1.5.2                utf8_1.2.3                 </span></span>
<span><span class="co">#&gt; [113] rmarkdown_2.20              grid_4.2.2                 </span></span>
<span><span class="co">#&gt; [115] minpack.lm_1.2-3            data.table_1.14.6          </span></span>
<span><span class="co">#&gt; [117] blob_1.2.3                  ModelMetrics_1.2.2.2       </span></span>
<span><span class="co">#&gt; [119] digest_0.6.31               xtable_1.8-4               </span></span>
<span><span class="co">#&gt; [121] textshaping_0.3.6           munsell_0.5.0              </span></span>
<span><span class="co">#&gt; [123] bslib_0.4.2</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
