<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Get started-with Methyl-IT • MethylIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/all.min.css" integrity="sha256-PbSmjxuVAzJ6FPvNYsrXygfGhNJYyZ2GktDbkMBqQZg=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/v4-shims.min.css" integrity="sha256-A6jcAdwFD48VMjlI3GDxUd+eCQa7/KWy6G9oe/ovaPA=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Get started-with Methyl-IT">
<meta property="og:description" content="MethylIT">
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/MethylIT/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Get started-with Methyl-IT</h1>
                        <h4 class="author"><div class="line-block">Robersy Sanchez<span class="math inline">\(^1\)</span><br><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a><br><br><span class="math inline">\(^1\)</span>Department of Biology.<br>
Pennsylvania State University, University Park, PA 16802</div></h4>
            
            <h4 class="date">23 March 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/master/vignettes/MethylIT.Rmd"><code>vignettes/MethylIT.Rmd</code></a></small>
      <div class="hidden name"><code>MethylIT.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      A fast introduction into methylation with Methyl-IT is provided here. The methylome analysis implemented in Methyl-IT based on a signal-detection and machine-learning approach.
    </div>
    
<style type="text/css">

.list-group-item.active, .list-group-item.active:focus, 
.list-group-item.active:hover {
    z-index: 2;
    color: #fff;
    background-color: #337ab7;
    border-color: #337ab7;
}

.tocify-subheader .tocify-item {
    padding-left: 25px;
    text-indent: 0;
}

.tocify-header .tocify-item {
    padding-left: 20px;
    text-indent: 0;
}

body{ /* Normal  */
    font-family: "Times New Roman", Times, serif;
    text-align: justify
}

h1.title {
    font-family: "Times New Roman", Times, serif;
    color: DarkRed;
    .subTitle {
        font-family: "Times New Roman", Times, serif;
        color: DarkRed;
    }
}

h1 { /* Header 1 */
    font-family: "Times New Roman", Times, serif;
    color: DarkBlue;
}
h2 { /* Header 2 */
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
h3 { /* Header 3 */
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
code.r{ /* Code block */
    font-size: 14px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">MethylIT</span>)</pre></body></html></div>
<div id="methyl-it-pipeline" class="section level1">
<h1 class="hasAnchor">
<a href="#methyl-it-pipeline" class="anchor"></a>Methyl-IT pipeline</h1>
<p>MethylIT is an R package for methylome analysis based on a signal-detection machine learning approach (SD-ML). This approach is postulated to provide greater sensitivity for resolving true signal from the methylation background within the methylome <span class="citation">(Sanchez and Mackenzie 2016; Sanchez et al. 2019)</span>. Because the biological signal created within the dynamic methylome environment characteristic of plants is not free from background noise, the approach, designated Methyl-IT, includes the application of signal detection theory <span class="citation">(Greiner, Pfeiffer, and Smith 2000; Carter et al. 2016; Harpaz et al. 2013; Kruspe et al. 2017)</span>. A basic requirement for the application of signal detection is the knowledge of the probability distribution of the background noise, which is used as null hypothesis in the detection of the methylation signal. This probability distribution is, in general, a member of generalized gamma distribution family, and it can be deduced on a statistical mechanical/thermodynamics basis from DNA methylation induced by thermal fluctuations <span class="citation">(Sanchez and Mackenzie 2016)</span>.</p>
<div id="load-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load data</h2>
<p>Methyl-IT uses the read counts of methylation called derived from whole-genome bisulfite sequencing (WGBS). Methylome datasets of WGBS are available at Gene Expression Omnibus (GEO DataSets). For the current example, we just load RData files containing the read-counts from several samples.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="co">#### ============ Download methylation data from PSU's GitLab =============</span>
<span class="no">url</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"https://git.psu.edu/genomath/MethylIT_examples/raw/master/"</span>,
            <span class="st">"MethylIT_testing/memoryLine_100k_samples_cg.RData"</span>)
<span class="no">temp</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span>(<span class="kw">fileext</span> <span class="kw">=</span> <span class="st">".RData"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span>(<span class="kw">url</span> <span class="kw">=</span> <span class="no">url</span>, <span class="kw">destfile</span> <span class="kw">=</span> <span class="no">temp</span>)
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="no">temp</span>)
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.remove</a></span>(<span class="no">temp</span>); <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span>(<span class="no">temp</span>, <span class="no">url</span>)
<span class="co">#&gt; [1] TRUE</span>

<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">meth_samples</span>)
<span class="co">#&gt;  [1] "W_3_1" "W_3_2" "W_3_3" "W_3_4" "W_3_5" "M_3_1" "M_3_2" "M_3_3" "M_3_4"</span>
<span class="co">#&gt; [10] "M_3_5"</span></pre></body></html></div>
<div id="gene-annotation" class="section level3">
<h3 class="hasAnchor">
<a href="#gene-annotation" class="anchor"></a>Gene annotation</h3>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">url</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"https://git.psu.edu/genomath/MethylIT_examples/raw/master/"</span>,
            <span class="st">"MethylIT_testing/Arabidopsis_thaliana.TAIR10.38_only_genes.RData"</span>)
<span class="no">temp</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span>(<span class="kw">fileext</span> <span class="kw">=</span> <span class="st">".RData"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span>(<span class="kw">url</span> <span class="kw">=</span> <span class="no">url</span>, <span class="kw">destfile</span> <span class="kw">=</span> <span class="no">temp</span>)
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="no">temp</span>)
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.remove</a></span>(<span class="no">temp</span>); <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span>(<span class="no">temp</span>, <span class="no">url</span>)
<span class="co">#&gt; [1] TRUE</span></pre></body></html></div>
</div>
</div>
<div id="the-reference-sample" class="section level2">
<h2 class="hasAnchor">
<a href="#the-reference-sample" class="anchor"></a>The reference sample</h2>
<p>To perform the comparison between the uncertainty of methylation levels from each group of individuals, control <span class="math inline">\((c)\)</span> and treatment <span class="math inline">\((t)\)</span>, we should estimate the uncertainty variation with respect to the same reference individual on the mentioned metric space. The reason to measure the uncertainty variation with respect to the same reference resides in that even sibling individuals follow an independent ontogenetic development and, consequently, their developments follow independent stochastic processes. The reference individual is estimated with function <a href="https://is.gd/qR79kW">poolFromGRlist</a>:</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">ref</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/poolFromGRlist.html">poolFromGRlist</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="no">meth_samples</span>$<span class="no">W_3_1</span>, <span class="no">meth_samples</span>$<span class="no">W_3_2</span>,
                        <span class="no">meth_samples</span>$<span class="no">W_3_3</span>), <span class="kw">stat</span> <span class="kw">=</span> <span class="st">"mean"</span>, <span class="kw">num.cores</span> <span class="kw">=</span> <span class="fl">4L</span>
)
<span class="co">#&gt; *** Building a unique GRanges object from the list...</span>
<span class="co">#&gt;  *** Building coordinates for the new GRanges object ...</span>
<span class="co">#&gt;  *** Strand information is preserved ...</span>
<span class="co">#&gt;  *** Processing GRanges for sample: #1</span>
<span class="co">#&gt;  *** Processing GRanges for sample: #2...</span>
<span class="co">#&gt;  *** Processing GRanges for sample: #3...</span>
<span class="co">#&gt;  *** Sorting by chromosomes and start positions...</span>
<span class="co">#&gt; *** Building a virtual methylome...</span></pre></body></html></div>
</div>
<div id="estimation-of-information-divergences" class="section level2">
<h2 class="hasAnchor">
<a href="#estimation-of-information-divergences" class="anchor"></a>Estimation of information divergences</h2>
<p>To perform the comparison between the uncertainty of methylation levels from each group of individuals, control <span class="math inline">\((c)\)</span> and treatment <span class="math inline">\((t)\)</span>, the divergence between the methylation levels of each individual is estimated with respect to the reference sample.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">ml_samples</span> <span class="kw">&lt;-</span> <span class="no">meth_samples</span>[<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"W_3_3"</span>, <span class="st">"W_3_4"</span>,<span class="st">"W_3_5"</span>,
                            <span class="st">"M_3_1"</span>,<span class="st">"M_3_2"</span>,<span class="st">"M_3_3"</span>,<span class="st">"M_3_4"</span>)]
<span class="no">ml_hd</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/estimateDivergence.html">estimateDivergence</a></span>(<span class="kw">ref</span> <span class="kw">=</span> <span class="no">ref</span>,
                            <span class="kw">indiv</span> <span class="kw">=</span> <span class="no">ml_samples</span>,
                            <span class="kw">Bayesian</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                            <span class="kw">min.coverage</span> <span class="kw">=</span> <span class="fl">4</span>,
                            <span class="kw">min.meth</span> <span class="kw">=</span> <span class="fl">1</span>,
                            <span class="kw">high.coverage</span> <span class="kw">=</span> <span class="fl">500</span>,
                            <span class="kw">percentile</span> <span class="kw">=</span> <span class="fl">0.999</span>,
                            <span class="kw">num.cores</span> <span class="kw">=</span> <span class="fl">4L</span>,
                            <span class="kw">tasks</span> <span class="kw">=</span> <span class="fl">2L</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="no">ml_hd</span>$<span class="no">M_3_1</span>
<span class="co">#&gt; GRanges object with 13112 ranges and 9 metadata columns:</span>
<span class="co">#&gt;           seqnames    ranges strand |        c1        t1        c2        t2</span>
<span class="co">#&gt;              &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;       [1]        1       161      + |         1         2         5         1</span>
<span class="co">#&gt;       [2]        1       310      + |         4         5         4        12</span>
<span class="co">#&gt;       [3]        1       311      - |         2         2         3         2</span>
<span class="co">#&gt;       [4]        1       500      + |         8         8         7         6</span>
<span class="co">#&gt;       [5]        1       501      - |         4         1         4         3</span>
<span class="co">#&gt;       ...      ...       ...    ... .       ...       ...       ...       ...</span>
<span class="co">#&gt;   [13108]       Pt     84348      - |         2       214         2       186</span>
<span class="co">#&gt;   [13109]       Pt     84350      + |         2       152         0       133</span>
<span class="co">#&gt;   [13110]       Pt     84351      - |         1       205         2       175</span>
<span class="co">#&gt;   [13111]       Pt     84353      + |         1       145         0       127</span>
<span class="co">#&gt;   [13112]       Pt     84354      - |         1       195         0       164</span>
<span class="co">#&gt;                            p1                  p2                   TV</span>
<span class="co">#&gt;                     &lt;numeric&gt;           &lt;numeric&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt;       [1]   0.392123103572845   0.764478583503753                  0.5</span>
<span class="co">#&gt;       [2]     0.4540512595575   0.272979343401941   -0.194444444444444</span>
<span class="co">#&gt;       [3]   0.503623099168751   0.576954596807912                  0.1</span>
<span class="co">#&gt;       [4]    0.50113182819633   0.534754757417114   0.0384615384615384</span>
<span class="co">#&gt;       [5]   0.735554399301687   0.559034434064729   -0.228571428571429</span>
<span class="co">#&gt;       ...                 ...                 ...                  ...</span>
<span class="co">#&gt;   [13108]  0.0126265121112533  0.0147758129336065  0.00137903861308117</span>
<span class="co">#&gt;   [13109]   0.017662437689319 0.00595389007120612   -0.012987012987013</span>
<span class="co">#&gt;   [13110] 0.00841476348922531  0.0156859154710009  0.00644506609620975</span>
<span class="co">#&gt;   [13111]  0.0118388367923311 0.00623170116120067 -0.00684931506849315</span>
<span class="co">#&gt;   [13112] 0.00884093144630434 0.00483925665447287 -0.00510204081632653</span>
<span class="co">#&gt;                         bay.TV               hdiv</span>
<span class="co">#&gt;                      &lt;numeric&gt;          &lt;numeric&gt;</span>
<span class="co">#&gt;       [1]    0.372355479930908  0.754596864387893</span>
<span class="co">#&gt;       [2]   -0.181071916155559  0.451488905020791</span>
<span class="co">#&gt;       [3]   0.0733314976391617 0.0295646659037349</span>
<span class="co">#&gt;       [4]    0.033622929220784 0.0173859971813955</span>
<span class="co">#&gt;       [5]   -0.176519965236958  0.236803386775921</span>
<span class="co">#&gt;       ...                  ...                ...</span>
<span class="co">#&gt;   [13108]   0.0021493008223532 0.0172923617150407</span>
<span class="co">#&gt;   [13109]  -0.0117085476181129   0.45154690308663</span>
<span class="co">#&gt;   [13110]  0.00727115198177563  0.217514462265944</span>
<span class="co">#&gt;   [13111] -0.00560713563113038  0.123141137274554</span>
<span class="co">#&gt;   [13112] -0.00400167479183146  0.108181314065562</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></pre></body></html></div>
<p>Function <a href="https://is.gd/0ViBO9">estimateDivergence</a> returns a list of GRanges objects with the four columns of counts, the information divergence, and additional columns:</p>
<pre><code>1. The original matrix of methylated ($c_i$) and unmethylated ($t_i$) read
    counts from control ($i=1$) and treatment ($i=2$) samples.
2. "p1" and "p2": methylation levels for control and treatment, 
    respectively.
3. "bay.TV": total variation TV = p2 - p1.
4. "TV": total variation based on simple counts: $TV=c1/(c1+t1)-c2/(c2+t2)$.
5. "hdiv": Hellinger divergence.</code></pre>
<p>Observe that the absolute values |TV| is an information divergence as well, which is known as total variation distance (widely used in information theory).</p>
</div>
<div id="descriptive-statistical-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#descriptive-statistical-analysis" class="anchor"></a><em>Descriptive statistical analysis</em>
</h2>
<p>A descriptive analysis on the distribution of Hellinger divergences of methylation levels is recommended. Like in any other statistical analysis, descriptive statistical analysis help us to detect potential issues in the raw data. It is the user responsibility to perform quality check of his/her dataset before start to apply Methyl-IT. Nevertheless, the quality checking of the raw data is not perfect. So, it is healthy to sees for potential issues.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="co">## Critical values</span>
<span class="no">critical.val</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span>(<span class="no">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span>(<span class="no">ml_hd</span>, <span class="kw">function</span>(<span class="no">x</span>) {
    <span class="no">hd.95</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span>(<span class="no">x</span>$<span class="no">hdiv</span>, <span class="fl">0.95</span>)
    <span class="no">tv.95</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="no">x</span>$<span class="no">bay.TV</span>), <span class="fl">0.95</span>)
    <span class="no">btv.95</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="no">x</span>$<span class="no">TV</span>), <span class="fl">0.95</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">tv</span> <span class="kw">=</span> <span class="no">tv.95</span>, <span class="kw">hd</span> <span class="kw">=</span> <span class="no">hd.95</span>, <span class="kw">TTV</span> <span class="kw">=</span> <span class="no">btv.95</span>,
            <span class="kw">num.sites.hd95</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">x</span>$<span class="no">hdiv</span> <span class="kw">&gt;</span> <span class="no">hd.95</span>),
            <span class="kw">num.sites.tv95</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">x</span>$<span class="no">bay.TV</span> <span class="kw">&gt;</span> <span class="no">tv.95</span>),
            <span class="kw">num.sites.btv.95</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">x</span>$<span class="no">TV</span> <span class="kw">&gt;</span> <span class="no">btv.95</span>)
    ))})
)
<span class="no">critical.val</span>
<span class="co">#&gt;          tv.95%    hd.95%   TTV.95% num.sites.hd95 num.sites.tv95</span>
<span class="co">#&gt; W_3_3 0.1748777 0.4966905 0.2243590            652            329</span>
<span class="co">#&gt; W_3_4 0.2543914 1.0024683 0.3333333            663            245</span>
<span class="co">#&gt; W_3_5 0.2467064 1.0291523 0.3000000            694            309</span>
<span class="co">#&gt; M_3_1 0.2947047 1.1405845 0.3636364            656            154</span>
<span class="co">#&gt; M_3_2 0.2810490 1.3483083 0.3333333            669            188</span>
<span class="co">#&gt; M_3_3 0.3051646 1.2746263 0.3888889            650            158</span>
<span class="co">#&gt; M_3_4 0.2643298 1.5618782 0.3333333            669            257</span>
<span class="co">#&gt;       num.sites.btv.95</span>
<span class="co">#&gt; W_3_3              331</span>
<span class="co">#&gt; W_3_4              248</span>
<span class="co">#&gt; W_3_5              265</span>
<span class="co">#&gt; M_3_1              159</span>
<span class="co">#&gt; M_3_2              184</span>
<span class="co">#&gt; M_3_3              172</span>
<span class="co">#&gt; M_3_4              189</span></pre></body></html></div>
<div id="quality-checking-of-the-coverage" class="section level3">
<h3 class="hasAnchor">
<a href="#quality-checking-of-the-coverage" class="anchor"></a>Quality checking of the coverage</h3>
<p>After the raw data filtering introduced with <a href="https://is.gd/0ViBO9">estimateDivergence</a> function, some coverage issue could be detected.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="no">covr</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/lapply.html">lapply</a></span>(<span class="no">ml_hd</span>, <span class="kw">function</span>(<span class="no">x</span>) {
                <span class="no">cov</span> <span class="kw">&lt;-</span> <span class="no">x</span>$<span class="no">c2</span> + <span class="no">x</span>$<span class="no">t2</span>
                <span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="no">cov</span>)
})

<span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span>(<span class="no">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span>(<span class="no">covr</span>, <span class="kw">function</span>(<span class="no">x</span>) {
    <span class="no">q60</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span>(<span class="no">x</span>, <span class="fl">0.6</span>)
    <span class="no">q9999</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span>(<span class="no">x</span>, <span class="fl">0.9999</span>)
    <span class="no">idx1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">x</span> <span class="kw">&gt;=</span> <span class="no">q60</span>)
    <span class="no">idx2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">x</span> <span class="kw">&lt;=</span> <span class="fl">500</span>)
    <span class="no">q95</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span>(<span class="no">x</span>, <span class="fl">0.95</span>)
    <span class="no">idx</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span>(<span class="no">idx1</span>, <span class="no">idx2</span>)
    <span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
        <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span>(<span class="no">x</span>)),
        <span class="no">q60</span>, <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span>(<span class="no">x</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.95</span>, <span class="fl">0.99</span>, <span class="fl">0.999</span>, <span class="fl">0.9999</span>)),
        <span class="kw">num.siteGreater_8</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">x</span> <span class="kw">&gt;=</span> <span class="fl">8</span>),
        <span class="kw">q60_to_500</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>((<span class="no">x</span> <span class="kw">&gt;=</span> <span class="no">q60</span>) <span class="kw">&amp;</span> (<span class="no">x</span> <span class="kw">&lt;=</span> <span class="fl">500</span>)),
        <span class="kw">num.siteGreater_500</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">x</span> <span class="kw">&gt;</span> <span class="fl">500</span>)
))})
)
<span class="co">#&gt;       Min. 1st Qu. Median Mean 3rd Qu. Max. 60% 95%    99%   99.9%   99.99%</span>
<span class="co">#&gt; W_3_3    0       7     12   33      40  491  15 112 351.00 424.000 466.4600</span>
<span class="co">#&gt; W_3_4    0       6     10   31      41  499  14  97 390.68 496.000 499.0000</span>
<span class="co">#&gt; W_3_5    0       7     12   34      41  495  16 113 370.20 446.480 486.6120</span>
<span class="co">#&gt; M_3_1    0       6     10   29      37  500  13  88 378.89 491.889 499.6889</span>
<span class="co">#&gt; M_3_2    0       6     11   28      30  499  14  77 357.00 457.000 496.3130</span>
<span class="co">#&gt; M_3_3    0       6      9   21      28  497  11  57 173.04 448.024 490.5020</span>
<span class="co">#&gt; M_3_4    0       7     11   28      30  494  15  81 347.00 449.000 489.6370</span>
<span class="co">#&gt;       num.siteGreater_8 q60_to_500 num.siteGreater_500</span>
<span class="co">#&gt; W_3_3              9348       5420                   0</span>
<span class="co">#&gt; W_3_4              8776       5364                   0</span>
<span class="co">#&gt; W_3_5             10133       5630                   0</span>
<span class="co">#&gt; M_3_1              8512       5288                   0</span>
<span class="co">#&gt; M_3_2              9020       5553                   0</span>
<span class="co">#&gt; M_3_3              7702       5602                   0</span>
<span class="co">#&gt; M_3_4              9390       5409                   0</span></pre></body></html></div>
</div>
</div>
<div id="estimation-of-the-best-fitted-probability-distribution-model" class="section level2">
<h2 class="hasAnchor">
<a href="#estimation-of-the-best-fitted-probability-distribution-model" class="anchor"></a>Estimation of the best fitted probability distribution model</h2>
<p>A basic requirement for the application of signal detection is the knowledge of the probability distribution of the background noise. Probability distribution, as a Weibull distribution model, can be deduced on a statistical mechanical/thermodynamics basis for DNA methylation induced by thermal fluctuations <span class="citation">(Sanchez and Mackenzie 2016)</span>.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r">gof  
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%
#&gt; 
#&gt;  *** Creating report ...
#&gt; Warning in gofReport(HD = ml_hd, model = c("Weibull2P", "Weibull3P", "Gamma2P", : The best fitted model for sample(s) M_3_4 require(s) for further analysis. 
#&gt; The model with the lowest AIC must have the highest R.Cross.val
gof$stats
#&gt;         w2p_AIC w2p_R.Cross.val w3p_AIC w3p_R.Cross.val    g2p_AIC
#&gt; W_3_3 -72384.64       0.9987526      NA              NA  -86145.27
#&gt; W_3_4 -84646.76       0.9994928      NA              NA -103110.43
#&gt; W_3_5 -86901.54       0.9994122      NA              NA -113367.18
#&gt; M_3_1 -82852.14       0.9994447      NA              NA -108854.50
#&gt; M_3_2 -90940.95       0.9996507      NA              NA -108955.49
#&gt; M_3_3 -90226.60       0.9997027      NA              NA -107175.23
#&gt; M_3_4 -94043.35       0.9997255      NA              NA  -98721.35
#&gt;       g2p_R.Cross.val    g3p_AIC g3p_R.Cross.val      bestModel
#&gt; W_3_3       0.9995184        Inf       0.0000000            g2p
#&gt; W_3_4       0.9998527        Inf       0.0000000            g2p
#&gt; W_3_5       0.9999042        Inf       0.0000000            g2p
#&gt; M_3_1       0.9999133        Inf       0.0000000            g2p
#&gt; M_3_2       0.9999000 -108998.91       0.9999015            g3p
#&gt; M_3_3       0.9999089 -107130.79       0.9999084            g2p
#&gt; M_3_4       0.9997932  -98826.68       0.9997889 Needs revision
gof$bestModel
#&gt;     W_3_3     W_3_4     W_3_5     M_3_1     M_3_2     M_3_3     M_3_4 
#&gt; "Gamma2P" "Gamma2P" "Gamma2P" "Gamma2P" "Gamma3P" "Gamma2P" "Gamma3P"</pre></body></html></div>
</div>
<div id="signal-detection" class="section level2">
<h2 class="hasAnchor">
<a href="#signal-detection" class="anchor"></a>Signal detection</h2>
<p>The information thermodynamics-based approach is postulated to provide greater sensitivity for resolving true signal from the thermodynamic background within the methylome <span class="citation">(Sanchez and Mackenzie 2016)</span>. Since the biological signal created within the dynamic methylome environment characteristic of living organisms is not free from background noise, the approach, designated Methyl-IT, includes the application of signal detection theory <span class="citation">(Greiner, Pfeiffer, and Smith 2000; Carter et al. 2016; Harpaz et al. 2013; Kruspe et al. 2017)</span>. Signal detection is a critical step to increase sensitivity and resolution of methylation signal by reducing the signal-to-noise ratio and objectively controlling the false positive rate and prediction accuracy/risk.</p>
<div id="potential-methylation-signal" class="section level3">
<h3 class="hasAnchor">
<a href="#potential-methylation-signal" class="anchor"></a><em>Potential methylation signal</em>
</h3>
<p>The first estimation in our signal detection step is the identification of the cytosine sites carrying potential methylation signal <span class="math inline">\(PS\)</span>. The methylation regulatory signal does not hold the theoretical distribution and, consequently, for a given level of significance <span class="math inline">\(\alpha\)</span> (Type I error probability, e.g. <span class="math inline">\(\alpha = 0.05\)</span>), cytosine positions <span class="math inline">\(k\)</span> with information divergence <span class="math inline">\(H_k &gt;= H_{\alpha = 0.05}\)</span> can be selected as sites carrying potential signals <span class="math inline">\(PS\)</span>. The value of <span class="math inline">\(\alpha\)</span> can be specified. For example, potential signals with <span class="math inline">\(H_k &gt; H_{\alpha = 0.01}\)</span> can be selected. For each sample, cytosine sites are selected based on the corresponding fitted theoretical distribution model estimated in the previous step.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="no">ps</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span>(<span class="kw">LR</span> <span class="kw">=</span> <span class="no">ml_hd</span>,
                        <span class="kw">nlms</span> <span class="kw">=</span> <span class="no">gof</span>$<span class="no">nlms</span>,
                        <span class="kw">div.col</span> <span class="kw">=</span> <span class="fl">9L</span>,
                        <span class="kw">tv.col</span> <span class="kw">=</span> <span class="fl">0.25</span>,
                        <span class="kw">dist.name</span> <span class="kw">=</span> <span class="no">gof</span>$<span class="no">bestModel</span>)
<span class="no">ps</span>$<span class="no">M_3_1</span>
<span class="co">#&gt; GRanges object with 737 ranges and 10 metadata columns:</span>
<span class="co">#&gt;         seqnames    ranges strand |        c1        t1        c2        t2</span>
<span class="co">#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;     [1]        1       511      + |        11         5         4         8</span>
<span class="co">#&gt;     [2]        1       648      - |         2         2         7         0</span>
<span class="co">#&gt;     [3]        1      5459      + |         6         2         2         5</span>
<span class="co">#&gt;     [4]        1      5466      - |         6         1         6         8</span>
<span class="co">#&gt;     [5]        1      7196      + |         2         3         0         9</span>
<span class="co">#&gt;     ...      ...       ...    ... .       ...       ...       ...       ...</span>
<span class="co">#&gt;   [733]       Mt     86158      + |         0        74         2        48</span>
<span class="co">#&gt;   [734]       Pt     30364      + |         0       371         4       370</span>
<span class="co">#&gt;   [735]       Pt     36502      - |         3       362         0       434</span>
<span class="co">#&gt;   [736]       Pt     46637      + |         3       487         0       494</span>
<span class="co">#&gt;   [737]       Pt     67192      - |         1       417         6       481</span>
<span class="co">#&gt;                          p1                  p2                   TV</span>
<span class="co">#&gt;                   &lt;numeric&gt;           &lt;numeric&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt;     [1]   0.673033759139014   0.353334714631113   -0.354166666666667</span>
<span class="co">#&gt;     [2]   0.503623099168751   0.908334478893389                  0.5</span>
<span class="co">#&gt;     [3]   0.713689467904528   0.326167737512289   -0.464285714285714</span>
<span class="co">#&gt;     [4]   0.798131662896332   0.436301124853528   -0.428571428571429</span>
<span class="co">#&gt;     [5]   0.425563914946981  0.0756781258028576                 -0.4</span>
<span class="co">#&gt;     ...                 ...                 ...                  ...</span>
<span class="co">#&gt;   [733] 0.00988257160661495  0.0543012507208152                 0.04</span>
<span class="co">#&gt;   [734] 0.00200202523549494  0.0127834702578034   0.0106951871657754</span>
<span class="co">#&gt;   [735]  0.0102214201503147   0.001839638935673 -0.00821917808219178</span>
<span class="co">#&gt;   [736] 0.00762161791063853 0.00161691723646952 -0.00612244897959184</span>
<span class="co">#&gt;   [737] 0.00416176037143209  0.0139203528120561   0.0099279840444868</span>
<span class="co">#&gt;                       bay.TV             hdiv              wprob</span>
<span class="co">#&gt;                    &lt;numeric&gt;        &lt;numeric&gt;          &lt;numeric&gt;</span>
<span class="co">#&gt;     [1]   -0.319699044507901 1.54766789991309 0.0187209712650864</span>
<span class="co">#&gt;     [2]    0.404711379724638  1.3579644452668 0.0276330873566829</span>
<span class="co">#&gt;     [3]   -0.387521730392239 1.32635922347908 0.0295018600053619</span>
<span class="co">#&gt;     [4]   -0.361830538042805 1.51432351745599 0.0200392118787338</span>
<span class="co">#&gt;     [5]   -0.349885789144124 1.37801067484234 0.0265120089817865</span>
<span class="co">#&gt;     ...                  ...              ...                ...</span>
<span class="co">#&gt;   [733]   0.0444186791142002  1.1148774914804 0.0459417346437029</span>
<span class="co">#&gt;   [734]   0.0107814450223084 1.75426411786179 0.0123204930412189</span>
<span class="co">#&gt;   [735] -0.00838178121464166 1.35401146205784 0.0278599176187997</span>
<span class="co">#&gt;   [736] -0.00600470067416902 1.09770327022115 0.0476450078448564</span>
<span class="co">#&gt;   [737]    0.009758592440624 1.30004029948585   0.03115842254738</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 7 sequences from an unspecified genome; no seqlengths</span></pre></body></html></div>
</div>
</div>
<div id="cutpoint-estimation--dmps" class="section level2">
<h2 class="hasAnchor">
<a href="#cutpoint-estimation--dmps" class="anchor"></a>Cutpoint estimation. DMPs</h2>
<p>Laws of statistical physics can account for background methylation, a response to thermal fluctuations that presumably functions in DNA stability <span class="citation">(Sanchez and Mackenzie 2016; Sanchez et al. 2019)</span>. True signal is detected based on the optimal cutpoint <span class="citation">(López-Ratón et al. 2014)</span>.</p>
<p>The need for the application of (what is now known as) signal detection in cancer research was pointed out by Youden in the midst of the last century <span class="citation">(Youden 1950)</span>. In the next example, the simple cutpoint estimation available in Methyl-IT is based on the application of <a href="https://en.wikipedia.org/wiki/Youden%27s_J_statistic">Youden index</a> <span class="citation">(Youden 1950)</span>. Although cutpoints are estimated for a single variable, the classification performance can be evaluated for several variables and applying different model classifiers. A optimal cutpoint distinguishes disease stages from healthy individual. The performance of this classification is given in the output of function <a href="https://is.gd/RR9iWY">estimateCutPoint</a>.</p>
<p>A model classifier can be requested for further predictions and its classification performance is also provided. Below, the selected model classifier is a quadratic discriminant analysis (QDA) (<em>classifier1</em> = “qda”, <em>clas.perf</em> = TRUE). Four predictor variables are available: the Hellinger divergence of methylation levels (<em>hdiv</em>), total variation distance (<em>TV</em>, absolute difference of methylation levels), relative position of cytosine site in the chromosome (<em>pos</em>), and the logarithm base two of the probability to observe a Hellinger divergence value <span class="math inline">\(H\)</span> greater than the critical value <span class="math inline">\(H_{\alpha = 0.05}\)</span> (values given as probabilities in object PS, <em>wprob</em>).</p>
<p>Notice that the cutpoint can be estimated for any of the two currently available information divergences: <em>Hellinger divergence</em> (<em>div.col</em> = 9) or the <em>total</em> <em>variation distance</em> (with Bayesian correction, <em>div.col</em> = 8).</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="co">### Cutpoint estimated using a model classifier</span>
<span class="no">cutpoint</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span>(<span class="kw">LR</span> <span class="kw">=</span> <span class="no">ps</span>,
                            <span class="kw">simple</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
                            <span class="kw">div.col</span> <span class="kw">=</span> <span class="fl">9</span>,
                            <span class="kw">control.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"W_3_3"</span>, <span class="st">"W_3_4"</span>, <span class="st">"W_3_5"</span>),
                            <span class="kw">treatment.names</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"M_3_1"</span>, <span class="st">"M_3_2"</span>, <span class="st">"M_3_3"</span>,
                                                <span class="st">"M_3_4"</span>, <span class="st">"M_3_5"</span>),
                            <span class="kw">column</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw">hdiv</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">bay.TV</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                                        <span class="kw">wprob</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">pos</span> <span class="kw">=</span> <span class="fl">TRUE</span>),
                            <span class="kw">classifier1</span> <span class="kw">=</span> <span class="st">"pca.qda"</span>, <span class="kw">n.pc</span> <span class="kw">=</span> <span class="fl">4</span>,
                            <span class="kw">classifier2</span> <span class="kw">=</span> <span class="st">"pca.logistic"</span>,
                            <span class="kw">center</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                            <span class="kw">scale</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                            <span class="kw">clas.perf</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
                            <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)
<span class="no">cutpoint</span>$<span class="no">cutpoint</span>
<span class="co">#&gt; [1] 0.9612333</span>
<span class="no">cutpoint</span>$<span class="no">testSetPerformance</span>
<span class="co">#&gt; Confusion Matrix and Statistics</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;           Reference</span>
<span class="co">#&gt; Prediction  CT  TT</span>
<span class="co">#&gt;         CT 237   0</span>
<span class="co">#&gt;         TT   1 544</span>
<span class="co">#&gt;                                      </span>
<span class="co">#&gt;                Accuracy : 0.9987     </span>
<span class="co">#&gt;                  95% CI : (0.9929, 1)</span>
<span class="co">#&gt;     No Information Rate : 0.6957     </span>
<span class="co">#&gt;     P-Value [Acc &gt; NIR] : &lt;2e-16     </span>
<span class="co">#&gt;                                      </span>
<span class="co">#&gt;                   Kappa : 0.997      </span>
<span class="co">#&gt;                                      </span>
<span class="co">#&gt;  Mcnemar's Test P-Value : 1          </span>
<span class="co">#&gt;                                      </span>
<span class="co">#&gt;             Sensitivity : 1.0000     </span>
<span class="co">#&gt;             Specificity : 0.9958     </span>
<span class="co">#&gt;          Pos Pred Value : 0.9982     </span>
<span class="co">#&gt;          Neg Pred Value : 1.0000     </span>
<span class="co">#&gt;              Prevalence : 0.6957     </span>
<span class="co">#&gt;          Detection Rate : 0.6957     </span>
<span class="co">#&gt;    Detection Prevalence : 0.6969     </span>
<span class="co">#&gt;       Balanced Accuracy : 0.9979     </span>
<span class="co">#&gt;                                      </span>
<span class="co">#&gt;        'Positive' Class : TT         </span>
<span class="co">#&gt; </span>
<span class="no">cutpoint</span>$<span class="no">testSetModel.FDR</span>
<span class="co">#&gt; [1] 0.001834862</span></pre></body></html></div>
<p>Differentially methylated positions (DMPs) can be now identified using <a href="https://is.gd/7cVEEd">selectDIMP</a> function</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="no">dmps</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/selectDIMP.html">selectDIMP</a></span>(<span class="no">ps</span>, <span class="kw">div.col</span> <span class="kw">=</span> <span class="fl">9</span>, <span class="kw">cutpoint</span> <span class="kw">=</span> <span class="no">cutpoint</span>$<span class="no">cutpoint</span>)
<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">dmps</span> <span class="kw">=</span><span class="fu"><a href="../reference/unlist.html">unlist</a></span>(<span class="fu"><a href="../reference/lapply.html">lapply</a></span>(<span class="no">dmps</span>, <span class="no">length</span>)))
<span class="co">#&gt;       dmps</span>
<span class="co">#&gt; W_3_3  130</span>
<span class="co">#&gt; W_3_4  709</span>
<span class="co">#&gt; W_3_5  712</span>
<span class="co">#&gt; M_3_1  737</span>
<span class="co">#&gt; M_3_2  784</span>
<span class="co">#&gt; M_3_3  738</span>
<span class="co">#&gt; M_3_4  844</span></pre></body></html></div>
</div>
<div id="differentially-methylated-genes-dmgs" class="section level2">
<h2 class="hasAnchor">
<a href="#differentially-methylated-genes-dmgs" class="anchor"></a>Differentially methylated genes (DMGs)</h2>
<p>Differentially methylated genes (DMGs) are estimated from group comparisons for the number of DMPs on gene-body regions between control and treatment.</p>
<p>Function <a href="https://is.gd/WQzX5I">getDIMPatGenes</a> is used to count the number of DMPs at gene-body. Nevertheless, it can be used for any arbitrary specified genomic region as well. The operation of this function is based on the ‘<em>findOverlaps</em>’ function from the ‘<em>GenomicRanges</em>’ Bioconductor R package. The ‘<em>findOverlaps</em>’ function has several critical parameters like, for example, ‘<em>maxgap</em>’, ‘<em>minoverlap</em>’, and ‘<em>ignore.strand</em>’. In our function <a href="https://is.gd/WQzX5I">getDIMPatGenes</a>, except for setting ignore.strand = TRUE and <em>type = “within”</em>, we preserve the rest of default ‘findOverlaps’ parameters. In this case, these are important parameter settings because the local mechanical effect of methylation changes on a DNA region where a gene is located is independent of the strand where the gene is encoded. That is, methylation changes located in any of the two DNA strands inside the gene-body region will affect the flexibility of the DNA molecule <span class="citation">(Choy et al. 2010; Severin et al. 2011)</span>.</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="no">nams</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">dmps</span>)
<span class="no">dmps_at_genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/getDIMPatGenes.html">getDIMPatGenes</a></span>(<span class="kw">GR</span> <span class="kw">=</span> <span class="no">dmps</span>, <span class="kw">GENES</span> <span class="kw">=</span> <span class="no">genes</span>, <span class="kw">ignore.strand</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="no">dmps_at_genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span>(<span class="no">dmps_at_genes</span>, <span class="kw">columns</span> <span class="kw">=</span> <span class="fl">2L</span>,
                                <span class="kw">ignore.strand</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">type</span> <span class="kw">=</span> <span class="st">"equal"</span>)
<span class="co">#&gt;  *** Building coordinates for the new GRanges object ...</span>
<span class="co">#&gt;  *** Setting strand information to * ...</span>
<span class="co">#&gt;  *** Processing GRanges for sample: #1</span>
<span class="co">#&gt;  *** Processing GRanges for sample: #2...</span>
<span class="co">#&gt;  *** Processing GRanges for sample: #3...</span>
<span class="co">#&gt;  *** Processing GRanges for sample: #4...</span>
<span class="co">#&gt;  *** Processing GRanges for sample: #5...</span>
<span class="co">#&gt;  *** Processing GRanges for sample: #6...</span>
<span class="co">#&gt;  *** Processing GRanges for sample: #7...</span>
<span class="co">#&gt;  *** Sorting by chromosomes and start positions...</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="fu">mcols</span>(<span class="no">dmps_at_genes</span>)) <span class="kw">&lt;-</span> <span class="no">nams</span>
<span class="no">dmps_at_genes</span>
<span class="co">#&gt; GRanges object with 77 ranges and 7 metadata columns:</span>
<span class="co">#&gt;        seqnames      ranges strand |     W_3_3     W_3_4     W_3_5     M_3_1</span>
<span class="co">#&gt;           &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;    [1]        1   3631-5899      * |         0         2         0         2</span>
<span class="co">#&gt;    [2]        1   6788-9130      * |         0         1         0         2</span>
<span class="co">#&gt;    [3]        1 11649-13714      * |         0         0         0         1</span>
<span class="co">#&gt;    [4]        1 23121-31227      * |         2         5         8        13</span>
<span class="co">#&gt;    [5]        1 31170-33171      * |         0         0         1         0</span>
<span class="co">#&gt;    ...      ...         ...    ... .       ...       ...       ...       ...</span>
<span class="co">#&gt;   [73]        5 49891-51437      * |         0         1         0         2</span>
<span class="co">#&gt;   [74]        5 53856-56052      * |         0         3         2         5</span>
<span class="co">#&gt;   [75]        5 61017-63936      * |         0         3         5         6</span>
<span class="co">#&gt;   [76]        5 86543-90117      * |         0         0         0         1</span>
<span class="co">#&gt;   [77]        5 91838-95701      * |         0         0         2         6</span>
<span class="co">#&gt;            M_3_2     M_3_3     M_3_4</span>
<span class="co">#&gt;        &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;    [1]         2         1         3</span>
<span class="co">#&gt;    [2]         0         1         0</span>
<span class="co">#&gt;    [3]         0         0         0</span>
<span class="co">#&gt;    [4]        15        14        12</span>
<span class="co">#&gt;    [5]         0         0         0</span>
<span class="co">#&gt;    ...       ...       ...       ...</span>
<span class="co">#&gt;   [73]         1         2         2</span>
<span class="co">#&gt;   [74]         3         2         3</span>
<span class="co">#&gt;   [75]         7         4         8</span>
<span class="co">#&gt;   [76]         0         0         0</span>
<span class="co">#&gt;   [77]         5         3         5</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 5 sequences from an unspecified genome; no seqlengths</span></pre></body></html></div>
<div id="setting-experiment-design" class="section level3">
<h3 class="hasAnchor">
<a href="#setting-experiment-design" class="anchor"></a><em>Setting experiment design</em>
</h3>
<p>The experimental design is set with <a href="https://genomaths.github.io/methylit/reference/glmDataSet.html">glmDataSet</a> function</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="no">colData</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">condition</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"WT"</span>, <span class="st">"WT"</span>, <span class="st">"WT"</span>,
                                            <span class="st">"ML"</span>, <span class="st">"ML"</span>, <span class="st">"ML"</span>, <span class="st">"ML"</span>),
                                        <span class="kw">levels</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"WT"</span>, <span class="st">"ML"</span>)),
                    <span class="no">nams</span>,
                    <span class="kw">row.names</span> <span class="kw">=</span> <span class="fl">2</span>)
<span class="co">## A RangedGlmDataSet is created</span>
<span class="no">ds</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/glmDataSet.html">glmDataSet</a></span>(<span class="kw">GR</span> <span class="kw">=</span> <span class="no">dmps_at_genes</span>, <span class="kw">colData</span> <span class="kw">=</span> <span class="no">colData</span>)</pre></body></html></div>
</div>
<div id="testing-for-dmgs" class="section level3">
<h3 class="hasAnchor">
<a href="#testing-for-dmgs" class="anchor"></a>Testing for DMGs</h3>
<p>DMGs are detected using function <a href="https://genomaths.github.io/methylit/reference/countTest2.html">countTest2</a>.</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="no">dmgs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/countTest2.html">countTest2</a></span>(<span class="kw">DS</span> <span class="kw">=</span> <span class="no">ds</span>, <span class="kw">num.cores</span> <span class="kw">=</span> <span class="fl">4L</span>,
                    <span class="kw">tasks</span> <span class="kw">=</span> <span class="fl">2L</span>,
                    <span class="kw">minCountPerIndv</span> <span class="kw">=</span> <span class="fl">7</span>,
                    <span class="kw">maxGrpCV</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1.2</span>, <span class="fl">1.2</span>),
                    <span class="kw">Minlog2FC</span> <span class="kw">=</span> <span class="fl">0.5</span>,
                    <span class="kw">CountPerBp</span> <span class="kw">=</span> <span class="fl">0.001</span>,
                    <span class="kw">test</span> <span class="kw">=</span> <span class="st">"LRT"</span>,
                    <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="co">#&gt; *** Number of GR after filtering counts 5</span>
<span class="co">#&gt; *** GLM...</span>
<span class="no">dmgs</span>
<span class="co">#&gt; GRanges object with 4 ranges and 15 metadata columns:</span>
<span class="co">#&gt;       seqnames      ranges strand |     W_3_3     W_3_4     W_3_5     M_3_1</span>
<span class="co">#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]        1 23121-31227      * |         2         5         8        13</span>
<span class="co">#&gt;   [2]        3 31145-34670      * |         2         4         3         8</span>
<span class="co">#&gt;   [3]        4 51166-53449      * |         0         3         3        14</span>
<span class="co">#&gt;   [4]        5 34538-37999      * |         1         3         1         9</span>
<span class="co">#&gt;           M_3_2     M_3_3     M_3_4            log2FC  scaled.deviance</span>
<span class="co">#&gt;       &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;         &lt;numeric&gt;        &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]        15        14        12 0.993251773010284 46.1893444462227</span>
<span class="co">#&gt;   [2]        11        12         7  1.15267950993839  6.0652836321834</span>
<span class="co">#&gt;   [3]         8         6        10  1.25276296849529 9.16216983213072</span>
<span class="co">#&gt;   [4]         7         6         6  1.43508452461667 23.8885478941796</span>
<span class="co">#&gt;                    pvalue          model             adj.pval</span>
<span class="co">#&gt;                 &lt;numeric&gt;       &lt;factor&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt;   [1] 1.0735927829192e-11 Neg.Binomial.W 4.29437113167679e-11</span>
<span class="co">#&gt;   [2]  0.0137864612185258 Neg.Binomial.W   0.0137864612185258</span>
<span class="co">#&gt;   [3] 0.00247069440811782 Neg.Binomial.W  0.00494138881623564</span>
<span class="co">#&gt;   [4] 0.00452300284262087   QuasiPoisson  0.00603067045682782</span>
<span class="co">#&gt;           CT.SignalDensity    TT.SignalDensity SignalDensityVariation</span>
<span class="co">#&gt;                  &lt;numeric&gt;           &lt;numeric&gt;              &lt;numeric&gt;</span>
<span class="co">#&gt;   [1] 0.000616750955963982 0.00166522758110275    0.00104847662513877</span>
<span class="co">#&gt;   [2] 0.000850822461712989  0.0026942711287578    0.00184344866704481</span>
<span class="co">#&gt;   [3] 0.000875656742556918 0.00415936952714536    0.00328371278458844</span>
<span class="co">#&gt;   [4] 0.000481417292509147 0.00202195262853842    0.00154053533602927</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 5 sequences from an unspecified genome; no seqlengths</span></pre></body></html></div>
</div>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-Carter2016">
<p>Carter, Jane V., Jianmin Pan, Shesh N. Rai, and Susan Galandiuk. 2016. “ROC-ing along: Evaluation and interpretation of receiver operating characteristic curves.” <em>Surgery</em> 159 (6). Mosby: 1638–45. <a href="https://doi.org/10.1016/j.surg.2015.12.029">https://doi.org/10.1016/j.surg.2015.12.029</a>.</p>
</div>
<div id="ref-Choy2010">
<p>Choy, John S., Sijie Wei, Ju Yeon Lee, Song Tan, Steven Chu, and Tae Hee Lee. 2010. “DNA methylation increases nucleosome compaction and rigidity.” <em>Journal of the American Chemical Society</em> 132 (6). American Chemical Society: 1782–3. <a href="https://doi.org/10.1021/ja910264z">https://doi.org/10.1021/ja910264z</a>.</p>
</div>
<div id="ref-Greiner2000">
<p>Greiner, M, D Pfeiffer, and R D Smith. 2000. “Principles and practical application of the receiver-operating characteristic analysis for diagnostic tests.” <em>Preventive Veterinary Medicine</em> 45 (1-2): 23–41. <a href="https://doi.org/10.1016/S0167-5877(00)00115-X">https://doi.org/10.1016/S0167-5877(00)00115-X</a>.</p>
</div>
<div id="ref-Harpaz2013">
<p>Harpaz, Rave, William DuMouchel, Paea LePendu, Anna Bauer-Mehren, Patrick Ryan, and Nigam H Shah. 2013. “Performance of Pharmacovigilance Signal Detection Algorithms for the FDA Adverse Event Reporting System.” <em>Clin Pharmacol Ther</em> 93 (6): 1–19. <a href="https://doi.org/10.1038/clpt.2013.24.Performance">https://doi.org/10.1038/clpt.2013.24.Performance</a>.</p>
</div>
<div id="ref-Kruspe2017">
<p>Kruspe, Sven, David D. Dickey, Kevin T. Urak, Giselle N. Blanco, Matthew J. Miller, Karen C. Clark, Elliot Burghardt, et al. 2017. “Rapid and Sensitive Detection of Breast Cancer Cells in Patient Blood with Nuclease-Activated Probe Technology.” <em>Molecular Therapy - Nucleic Acids</em> 8 (September). Cell Press: 542–57. <a href="https://doi.org/10.1016/j.omtn.2017.08.004">https://doi.org/10.1016/j.omtn.2017.08.004</a>.</p>
</div>
<div id="ref-lopez2014">
<p>López-Ratón, Mónica, Mar<span class="math inline">\(\backslash\)</span>’<span class="math inline">\(\backslash\)</span>ia Xosé Rodríguez-Álvarez, Carmen Cadarso-Suárez, Francisco Gude-Sampedro, and Others. 2014. “OptimalCutpoints: an R package for selecting optimal cutpoints in diagnostic tests.” <em>Journal of Statistical Software</em> 61 (8). Foundation for Open Access Statistics: 1–36. <a href="https://www.jstatsoft.org/article/view/v061i08">https://www.jstatsoft.org/article/view/v061i08</a>.</p>
</div>
<div id="ref-Sanchez2016">
<p>Sanchez, Robersy, and Sally A. Mackenzie. 2016. “Information Thermodynamics of Cytosine DNA Methylation.” Edited by Barbara Bardoni. <em>PLOS ONE</em> 11 (3). Public Library of Science: e0150427. <a href="https://doi.org/10.1371/journal.pone.0150427">https://doi.org/10.1371/journal.pone.0150427</a>.</p>
</div>
<div id="ref-Sanchez2019">
<p>Sanchez, Robersy, Xiaodong Yang, Thomas Maher, and Sally Mackenzie. 2019. “Discrimination of DNA Methylation Signal from Background Variation for Clinical Diagnostics.” <em>Int. J. Mol. Sci.</em> 20 (21): 5343. <a href="https://doi.org/https://doi.org/10.3390/ijms20215343">https://doi.org/https://doi.org/10.3390/ijms20215343</a>.</p>
</div>
<div id="ref-Severin2011">
<p>Severin, Philip M D, Xueqing Zou, Hermann E Gaub, and Klaus Schulten. 2011. “Cytosine methylation alters DNA mechanical properties.” <em>Nucleic Acids Research</em> 39 (20): 8740–51. <a href="https://doi.org/10.1093/nar/gkr578">https://doi.org/10.1093/nar/gkr578</a>.</p>
</div>
<div id="ref-Youden1950">
<p>Youden, W. J. 1950. “Index for rating diagnostic tests.” <em>Cancer</em> 3 (1): 32–35. <a href="https://doi.org/10.1002/1097-0142(1950)3:1&lt;32::AID-CNCR2820030106&gt;3.0.CO;2-3">https://doi.org/10.1002/1097-0142(1950)3:1&lt;32::AID-CNCR2820030106&gt;3.0.CO;2-3</a>.</p>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
