<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Principal Components and Linear Discriminant Analyses with Methyl-IT • MethylIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Principal Components and Linear Discriminant Analyses with Methyl-IT">
<meta property="og:description" content="MethylIT">
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/MethylIT/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="pca_lda_with_methylit_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Principal Components and Linear Discriminant Analyses with Methyl-IT</h1>
            <h3 class="subtitle"><h1>
A <a href="https://github.com/genomaths/MethylIT">Methyl-IT</a> user guide</h1></h3>
                        <h4 class="author">
<div class="line-block">Robersy Sanchez<br><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a><br>
Mackenzie’s lab</div>
<div class="line-block">Department of Biology and Plant Science.<br>
Pennsylvania State University, University Park, PA 16802<br>
Maintainer: Robersy Sanchez</div>
</h4>
            
            <h4 class="date">04 November 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/master/vignettes/pca_lda_with_methylit.Rmd"><code>vignettes/pca_lda_with_methylit.Rmd</code></a></small>
      <div class="hidden name"><code>pca_lda_with_methylit.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>When methylation analysis is intended for diagnostic/prognostic purposes, for example, in clinical applications for patient diagnostics, to know whether the patient would be in healthy or disease stage we would like to have a good predictor tool in our side. It turns out that classical machine learning (ML) tools like hierarchical clustering, principal components and linear discriminant analysis can help us to reach such a goal. The current Methyl-IT downstream analysis is equipped with the mentioned ML tools.</p>
    </div>
    
<style type="text/css">

.list-group-item.active, .list-group-item.active:focus, 
.list-group-item.active:hover {
    z-index: 2;
    color: #fff;
    background-color: #337ab7;
    border-color: #337ab7;
}

.tocify-subheader .tocify-item {
  font-size: 0.75em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify-header .tocify-item {
  font-size: 0.8em;
  padding-left: 20px;
  text-indent: 0;
}

body{ /* Normal  */
      font-size: 16px;
      font-family: "Times New Roman", Times, serif;
      text-align: justify
  }
td {  /* Table  */
  font-size: 8px;
}

h1.title {
   font-size: 36px;
   font-family: "Times New Roman", Times, serif;
   color: DarkRed;
   .subTitle {
       font-size: 24px;
       font-family: "Times New Roman", Times, serif;
       color: DarkRed;
  }
}

h1 { /* Header 1 */
  font-size: 24px;
  font-family: "Times New Roman", Times, serif;
  color: DarkBlue;
}
h2 { /* Header 2 */
    font-size: 22px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
h3 { /* Header 3 */
   font-size: 18px;
   color: DarkBlue;
   font-family: "Times New Roman", Times, serif;
}
code.r{ /* Code block */
    font-size: 12px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>
<div id="background" class="section level1">
<h1 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h1>
<p>The discrimination of the methylation signal from the stochastic methylation background resultant from the standard (non-stressful) biological processes is a critical step for the genome-wide methylation analysis. Such a discrimination requires for the knowledge of the probability distribution of the information divergence of methylation levels and a proper evaluation of the classification performance of differentially methylated positions (DMPs) into two classes: DMPs from control and DMPs from treatment.</p>
<p>Once a methylation analysis is performed to detect the genome-wide DMPs, a natural next step is to investigate whether the detected DMPs carries enough discriminatory power to discern between individual groups. Then, we must move from the methylation analysis to statistical/machine learning field (multivariate statistics). To reach our goal, each individual can be represented as a vector from the <span class="math inline">\(N\)</span>-dimensional space of DMPs, where DMPs can be expressed in terms of some suitable measurement as function of the methylation levels. These suitable measurement are information divergences, such as total variation distance (<span class="math inline">\(TV_d\)</span>, the absolute value of methylation levels) and Hellinger divergence (<span class="math inline">\(H\)</span>). It is worthy to observe that methylation levels are uncertainty measurements and, therefore, they do not carry information per se. Information is only carried on measurements expressing uncertainty variations like <span class="math inline">\(TV_d\)</span>, <span class="math inline">\(H\)</span>, Jensen-Shannon Divergence, and in general, the so called information theoretical divergences <span class="citation">(1)</span>.</p>
<p>The <span class="math inline">\(N\)</span>-dimensional space of DMPs is, usually, too big. So, a first dimension reduction is required. This can be done by splitting the chromosomes into non-overlapping genomic region and next computing some statistic like the <em>sum</em> or <em>mean</em> of some information divergences of methylation levels at DMPs inside of each region. Still we will have to deal with a multidimensional space, but acceptable for a meaningful application of hierarchical clustering (HC), principal component analysis (PCA) and linear discriminant analysis (LDA). In the current examples, Methyl-IT methylation analysis will be applied to a dataset of simulated samples to detect DMPs on then. Next, after space dimension reduction, the mentioned machine learning approached will be applied.</p>
</div>
<div id="data-generation" class="section level1">
<h1 class="hasAnchor">
<a href="#data-generation" class="anchor"></a>Data generation</h1>
<p>For the current example on methylation analysis with Methyl-IT we will use simulated data. Read-count matrices of methylated and unmethylated cytosine are generated with MethylIT.utils function <a href="https://genomaths.github.io/MethylIT_utils_HTML_Manual/simulateCounts.html">simulateCounts</a>. A basic example generating datasets is given in the web page: <a href="https://genomaths.github.io/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/MethylIT">MethylIT</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">MethylIT.utils</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ape-package.ird.fr/">ape</a></span><span class="op">)</span>

<span class="va">alpha.ct</span> <span class="op">&lt;-</span> <span class="fl">0.019</span>
<span class="va">alpha.g1</span> <span class="op">&lt;-</span> <span class="fl">0.022</span>
<span class="va">alpha.g2</span> <span class="op">&lt;-</span> <span class="fl">0.025</span>

<span class="co"># The number of cytosine sites to generate</span>
<span class="va">sites</span> <span class="op">=</span> <span class="fl">1.5e5</span> 
<span class="co"># Set a seed for pseudo-random number generation</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">124</span><span class="op">)</span>
<span class="va">control.nam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"C1"</span>, <span class="st">"C2"</span>, <span class="st">"C3"</span>, <span class="st">"C4"</span>, <span class="st">"C5"</span><span class="op">)</span>
<span class="va">treatment.nam1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"T1"</span>, <span class="st">"T2"</span>, <span class="st">"T3"</span>, <span class="st">"T4"</span>, <span class="st">"T5"</span><span class="op">)</span>
<span class="va">treatment.nam2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"T6"</span>, <span class="st">"T7"</span>, <span class="st">"T8"</span>, <span class="st">"T9"</span>, <span class="st">"T10"</span><span class="op">)</span>

<span class="co"># Reference group </span>
<span class="va">ref0</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html">simulateCounts</a></span><span class="op">(</span>num.samples <span class="op">=</span> <span class="fl">3</span>, sites <span class="op">=</span> <span class="va">sites</span>, alpha <span class="op">=</span> <span class="va">alpha.ct</span>, 
                      beta <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">50</span>, theta <span class="op">=</span> <span class="fl">4.5</span>, 
                      sample.ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"R1"</span>, <span class="st">"R2"</span>, <span class="st">"R3"</span><span class="op">)</span><span class="op">)</span>
<span class="co"># Control group </span>
<span class="va">ctrl</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html">simulateCounts</a></span><span class="op">(</span>num.samples <span class="op">=</span> <span class="fl">5</span>, sites <span class="op">=</span> <span class="va">sites</span>, alpha <span class="op">=</span> <span class="va">alpha.ct</span>, 
                      beta <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">50</span>, theta <span class="op">=</span> <span class="fl">4.5</span>, 
                      sample.ids <span class="op">=</span> <span class="va">control.nam</span><span class="op">)</span>
<span class="co"># Treatment group II</span>
<span class="va">treat</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html">simulateCounts</a></span><span class="op">(</span>num.samples <span class="op">=</span> <span class="fl">5</span>, sites <span class="op">=</span> <span class="va">sites</span>, alpha <span class="op">=</span> <span class="va">alpha.g1</span>,
                       beta <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">50</span>, theta <span class="op">=</span> <span class="fl">4.5</span>, 
                       sample.ids <span class="op">=</span> <span class="va">treatment.nam1</span><span class="op">)</span>

<span class="co"># Treatment group II</span>
<span class="va">treat2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html">simulateCounts</a></span><span class="op">(</span>num.samples <span class="op">=</span> <span class="fl">5</span>, sites <span class="op">=</span> <span class="va">sites</span>, alpha <span class="op">=</span> <span class="va">alpha.g2</span>,
                        beta <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">50</span>, theta <span class="op">=</span> <span class="fl">4.5</span>, 
                        sample.ids <span class="op">=</span> <span class="va">treatment.nam2</span><span class="op">)</span></code></pre></div>
<p>A reference sample (virtual individual) is created using individual samples from the control population using function <a href="https://genomaths.github.io/MethylIT_HTML_Manual/poolFromGRlist.html">poolFromGRlist</a>. The reference sample is further used to compute the information divergences of methylation levels, <span class="math inline">\(TV_d\)</span> and <span class="math inline">\(H\)</span>, with function <a href="https://genomaths.github.io/MethylIT_HTML_Manual/estimateDivergence.html">estimateDivergence</a>. This is a first fundamental step to remove the background noise (fluctuations) generated by the inherent stochasticity of the molecular processes in the cells.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># === Methylation level divergences ===</span>
<span class="co"># Reference sample</span>
<span class="va">ref</span> <span class="op">=</span> <span class="fu"><a href="../reference/poolFromGRlist.html">poolFromGRlist</a></span><span class="op">(</span><span class="va">ref0</span>, stat <span class="op">=</span> <span class="st">"mean"</span>, num.cores <span class="op">=</span> <span class="fl">4L</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="va">divs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/estimateDivergence.html">estimateDivergence</a></span><span class="op">(</span>ref <span class="op">=</span> <span class="va">ref</span>, indiv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">ctrl</span>, <span class="va">treat</span>, <span class="va">treat2</span><span class="op">)</span>, Bayesian <span class="op">=</span> <span class="cn">TRUE</span>, 
                           num.cores <span class="op">=</span> <span class="fl">6L</span>, percentile <span class="op">=</span> <span class="fl">1</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="co"># To remove hd == 0 to estimate. The methylation signal only is given for  </span>
<span class="va">divs</span> <span class="op">=</span> <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">divs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">div</span><span class="op">)</span> <span class="va">div</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">div</span><span class="op">$</span><span class="va">hdiv</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span> <span class="op">]</span>, keep.attr <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">divs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">divs</span><span class="op">)</span></code></pre></div>
<p>To get some statistical description about the sample is useful. Here, empirical critical values for the probability distribution of <span class="math inline">\(H\)</span> and <span class="math inline">\(TV\)</span> can is obtained using <em>quantile</em> function from the R package <em>stats</em>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">critical.val</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">divs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">x</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="va">x</span><span class="op">$</span><span class="va">hdiv</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span>
  <span class="va">hd.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">hdiv</span>, <span class="fl">0.95</span><span class="op">)</span>
  <span class="va">tv.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html">quantile</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">bay.TV</span><span class="op">)</span>, <span class="fl">0.95</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>tv <span class="op">=</span> <span class="va">tv.95</span>, hd <span class="op">=</span> <span class="va">hd.95</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span><span class="op">)</span>
<span class="va">critical.val</span></code></pre></div>
<pre><code>##        tv.95%    hd.95%
## C1  0.6406672  59.58613
## C2  0.6374645  58.41722
## C3  0.6386189  59.09187
## C4  0.7706766  92.49756
## C5  0.8076015  99.44518
## T1  0.8923509 118.55831
## T2  0.8820082 113.92496
## T3  0.8805118 114.88998
## T4  0.8865382 117.41163
## T5  0.8881189 117.81490
## T6  0.9258356 128.28320
## T7  0.9250543 129.49118
## T8  0.9200787 126.41222
## T9  0.9292491 129.89578
## T10 0.9255575 129.07053</code></pre>
</div>
<div id="modeling-the-methylation-signal" class="section level1">
<h1 class="hasAnchor">
<a href="#modeling-the-methylation-signal" class="anchor"></a>Modeling the methylation signal</h1>
<p>Here, the methylation signal is expressed in terms of Hellinger divergence of methylation levels. Here, the signal distribution is modelled by a Weibull probability distribution model. Basically, the model could be a member of the generalized gamma distribution family. For example, it could be gamma distribution, Weibull, or log-normal. To describe the signal, we may prefer a model with a cross-validations: R.Cross.val &gt; 0.95. Cross-validations for the nonlinear regressions are performed in each methylome as described in <span class="citation">(2)</span>. The non-linear fit is performed through the function <a href="https://genomaths.github.io/MethylIT_HTML_Manual/nonlinearFitDist.html">nonlinearFitDist</a></p>
<p>The best model excluding “GGamma3P” model, for which the computation is computationally expensive:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Weibull2P"</span>, <span class="st">"Weibull3P"</span>, <span class="st">"Gamma2P"</span>, <span class="st">"Gamma3P"</span><span class="op">)</span>
<span class="va">gof</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gofReport.html">gofReport</a></span><span class="op">(</span>HD <span class="op">=</span> <span class="va">divs</span>, column <span class="op">=</span> <span class="fl">9L</span>, model <span class="op">=</span> <span class="va">d</span>, num.cores <span class="op">=</span> <span class="fl">6L</span>,
                    output <span class="op">=</span> <span class="st">"all"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gof</span><span class="op">$</span><span class="va">bestModel</span></code></pre></div>
<pre><code>##          C1          C2          C3          C4          C5          T1 
## "Weibull3P" "Weibull3P" "Weibull3P" "Weibull3P" "Weibull3P" "Weibull3P" 
##          T2          T3          T4          T5          T6          T7 
## "Weibull3P" "Weibull3P" "Weibull3P" "Weibull3P" "Weibull3P" "Weibull3P" 
##          T8          T9         T10 
## "Weibull3P" "Weibull3P" "Weibull3P"</code></pre>
<p>Although we did not included the main member of the generalized gamma distribution family (“GGamma3P”), the Cross-validations correlation coefficient <em>w3p_R.Cross.val</em> values above 0.99 gives us some peace of mind on our confidence on the model prediction power/performance when confronted with external data sets (AIC and BIC are not conceived to provide such an information).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gof</span><span class="op">$</span><span class="va">stats</span></code></pre></div>
<pre><code>##       w2p_AIC w2p_R.Cross.val   w3p_AIC w3p_R.Cross.val   g2p_AIC
## C1  -173343.1       0.9994740 -174151.8       0.9994889 -165487.2
## C2  -176361.6       0.9995241 -176807.8       0.9995314 -168458.2
## C3  -177857.3       0.9995495 -178351.4       0.9995577 -170455.8
## C4  -176986.4       0.9982813 -241815.5       0.9997452 -170025.2
## C5  -173024.7       0.9979755 -236166.5       0.9996815 -166396.9
## T1  -171247.2       0.9974948 -229699.6       0.9995216 -163260.7
## T2  -170395.2       0.9974948 -231191.4       0.9995552 -163066.8
## T3  -173437.5       0.9976429 -231862.2       0.9995516 -166304.0
## T4  -171910.8       0.9975559 -229519.1       0.9995241 -164696.9
## T5  -170592.5       0.9974783 -231856.3       0.9995570 -163388.5
## T6  -175158.9       0.9973222 -230950.5       0.9994177 -166927.9
## T7  -172799.5       0.9971559 -230118.2       0.9994043 -164568.9
## T8  -173332.4       0.9972197 -230830.2       0.9994207 -164912.2
## T9  -172019.4       0.9971076 -229348.1       0.9993938 -163516.3
## T10 -173248.1       0.9971756 -229839.4       0.9993964 -165184.0
##     g2p_R.Cross.val   g3p_AIC g3p_R.Cross.val bestModel
## C1        0.9994168 -166198.9       0.9994472       w3p
## C2        0.9994744 -168881.1       0.9994902       w3p
## C3        0.9995056 -170826.3       0.9995182       w3p
## C4        0.9978091 -224251.9       0.9995298       w3p
## C5        0.9974082 -217515.6       0.9993874       w3p
## T1        0.9966440 -209442.6       0.9990548       w3p
## T2        0.9967186 -211204.4       0.9991269       w3p
## T3        0.9969232 -212608.4       0.9991339       w3p
## T4        0.9967954 -209858.8       0.9990703       w3p
## T5        0.9966969 -211370.3       0.9991181       w3p
## T6        0.9963637 -210591.2       0.9988671       w3p
## T7        0.9961477 -208990.3       0.9988197       w3p
## T8        0.9962298 -209992.3       0.9988675       w3p
## T9        0.9960448 -207860.2       0.9987920       w3p
## T10       0.9961734 -208773.2       0.9988041       w3p</code></pre>
</div>
<div id="estimation-of-an-optimal-cutpoint-to-discriminate-the-signal-induced-by-the-treatment" class="section level1">
<h1 class="hasAnchor">
<a href="#estimation-of-an-optimal-cutpoint-to-discriminate-the-signal-induced-by-the-treatment" class="anchor"></a>Estimation of an optimal cutpoint to discriminate the signal induced by the treatment</h1>
<p>The above statistical description of the dataset (evidently) suggests that there two main groups: control and treatments, while treatment group would split into two subgroups of samples. In the current case, to search for a good cutpoint, we do not need to use all the samples. The critical value <span class="math inline">\(H_{\alpha=0.05}=33.51223\)</span> suggests that any optimal cutpoint for the subset of samples T1 to T5 will be optimal for the samples T6 to T10 as well.</p>
<p>Below, we are letting the PCA+LDA model classifier to take the decision on whether a differentially methylated cytosine position is a treatment DMP. To do it, Methyl-IT function <a href="https://genomaths.github.io/MethylIT_HTML_Manual/getPotentialDIMP.html">getPotentialDIMP</a> is used to get methylation signal probabilities of the observed <span class="math inline">\(H\)</span> values for all cytosine site (alpha = 1), in accordance with the 2-parameter Weibull distribution model. Next, this information will be used to identify DMPs using Methyl-IT function <a href="https://genomaths.github.io/MethylIT_HTML_Manual/estimateCutPoint.html">estimateCutPoint</a>. Cytosine positions with <span class="math inline">\(H\)</span> values above the cutpoint are considered DMPs. Finally, a PCA + QDA model classifier will be fitted to classify DMPs into two classes: DMPs from control and those from treatment.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">divs</span>,
                         nlms <span class="op">=</span> <span class="va">gof</span><span class="op">$</span><span class="va">nlms</span>,  div.col <span class="op">=</span> <span class="fl">9L</span>, 
                         tv.cut <span class="op">=</span> <span class="fl">0.461</span>, tv.col <span class="op">=</span> <span class="fl">8</span>, alpha <span class="op">=</span> <span class="fl">0.05</span>, 
                         dist.name <span class="op">=</span> <span class="va">gof</span><span class="op">$</span><span class="va">bestModel</span><span class="op">)</span>
<span class="va">cutp</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">dmps</span>, simple <span class="op">=</span> <span class="cn">FALSE</span>,
                        column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, TV <span class="op">=</span> <span class="cn">TRUE</span>, 
                                   wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,
                        classifier1 <span class="op">=</span> <span class="st">"pca.lda"</span>, 
                        classifier2 <span class="op">=</span> <span class="st">"pca.qda"</span>, tv.cut <span class="op">=</span> <span class="fl">0.461</span>,
                        control.names <span class="op">=</span> <span class="va">control.nam</span>, 
                        treatment.names <span class="op">=</span> <span class="va">treatment.nam1</span>,
                        post.cut <span class="op">=</span> <span class="fl">0.5</span>, cut.values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">15</span>, <span class="fl">38</span>, <span class="fl">1</span><span class="op">)</span>,
                        clas.perf <span class="op">=</span> <span class="cn">TRUE</span>, prop <span class="op">=</span> <span class="fl">0.6</span>,
                        center <span class="op">=</span> <span class="cn">FALSE</span>, scale <span class="op">=</span> <span class="cn">FALSE</span>,
                        n.pc <span class="op">=</span> <span class="fl">4</span>, div.col <span class="op">=</span> <span class="fl">9L</span>, stat <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>
<span class="va">cutp</span></code></pre></div>
<pre><code>## Cutpoint estimation with 'pca.lda' classifier 
## Cutpoint search performed using model posterior probabilities 
## 
## Posterior probability used to get the cutpoint = 0.5 
## Cytosine sites with treatment PostProbCut &gt;= 0.5 have a 
## divergence value &gt;= 76.33942 
## 
## Optimized statistic: Accuracy = 0.9603 
## Cutpoint = 88.733 
## 
## Model classifier 'pca.qda' 
## 
## The accessible objects in the output list are: 
##                    Length Class           Mode     
## cutpoint           1      -none-          numeric  
## testSetPerformance 6      confusionMatrix list     
## testSetModel.FDR   1      -none-          numeric  
## model              2      pcaQDA          list     
## modelConfMatrix    6      confusionMatrix list     
## initModel          1      -none-          character
## postProbCut        1      -none-          numeric  
## postCut            1      -none-          numeric  
## classifier         1      -none-          character
## statistic          1      -none-          character
## optStatVal         1      -none-          numeric  
## cutpData           1      -none-          logical</code></pre>
<p>In this case, the cutpoint <span class="math inline">\(H_{cutpoint}=92.138\)</span> is lower from what is expected from the higher treatment empirical <span class="math inline">\(H_{\alpha = 0.05}^{TT_{Emp}}=120.45119\)</span> critical value: <span class="math inline">\(H_{cutpoint} &lt; H_{\alpha = 0.05}^{TT_{Emp}}\)</span>.</p>
<p>The model performance in in the test dataset is:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutp</span><span class="op">$</span><span class="va">testSetPerformance</span></code></pre></div>
<pre><code>## Confusion Matrix and Statistics
## 
##           Reference
## Prediction   CT   TT
##         CT 3006  160
##         TT  170 4977
##                                           
##                Accuracy : 0.9603          
##                  95% CI : (0.9559, 0.9644)
##     No Information Rate : 0.6179          
##     P-Value [Acc &gt; NIR] : &lt;2e-16          
##                                           
##                   Kappa : 0.9159          
##                                           
##  Mcnemar's Test P-Value : 0.6203          
##                                           
##             Sensitivity : 0.9689          
##             Specificity : 0.9465          
##          Pos Pred Value : 0.9670          
##          Neg Pred Value : 0.9495          
##              Prevalence : 0.6179          
##          Detection Rate : 0.5987          
##    Detection Prevalence : 0.6192          
##       Balanced Accuracy : 0.9577          
##                                           
##        'Positive' Class : TT              
## </code></pre>
</div>
<div id="representing-individual-samples-as-vectors-from-the-n-dimensional-space" class="section level1">
<h1 class="hasAnchor">
<a href="#representing-individual-samples-as-vectors-from-the-n-dimensional-space" class="anchor"></a>Representing individual samples as vectors from the <em>N</em>-dimensional space</h1>
<p>The above cutpoint can be used to identify DMPs from control and treatment. The PCA+QDA model classifier can be used any time to discriminate control DMPs from those treatment. DMPs are retrieved using <a href="https://genomaths.github.io/MethylIT_HTML_Manual/selectDIMP.html">selectDIMP</a> function (<code>selectDIMP</code>):</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectDIMP.html">selectDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">divs</span>, div.col <span class="op">=</span> <span class="fl">9L</span>, cutpoint <span class="op">=</span> <span class="va">cutp</span><span class="op">$</span><span class="va">cutpoint</span>,
                   tv.cut <span class="op">=</span> <span class="fl">0.34</span>, tv.col <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">dmps</span>,<span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code>##   C1   C2   C3   C4   C5   T1   T2   T3   T4   T5   T6   T7   T8   T9  T10 
##  431  405  395 1804 2032 2747 2586 2570 2670 2678 3174 3225 3216 3309 3249</code></pre>
<p>Next, to represent individual samples as vectors from the <em>N</em>-dimensional space, we can use <a href="https://genomaths.github.io/MethylIT_utils_HTML_Manual/getGRegionsStat-methods.html">getGRegionsStat</a> function from <a href="https://github.com/genomaths/MethylIT.utils">MethylIT.utils</a> R package. Here, the simulated “chromosome” is split into regions of 450bp non-overlapping windows. and the <em>density</em> of Hellinger divergences values is taken for each windows.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span>
<span class="va">DMRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/getGRegionsStat-methods.html">getGRegionsStat</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, win.size <span class="op">=</span> <span class="fl">150</span>, step.size <span class="op">=</span> <span class="fl">150</span>,
                        stat <span class="op">=</span> <span class="st">"density"</span>, column <span class="op">=</span> <span class="fl">9L</span>, scaling <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">DMRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">ns</span>
<span class="fu">as</span><span class="op">(</span><span class="va">DMRs</span>, <span class="st">"GRangesList"</span><span class="op">)</span></code></pre></div>
</div>
<div id="hierarchical-clustering" class="section level1">
<h1 class="hasAnchor">
<a href="#hierarchical-clustering" class="anchor"></a>Hierarchical Clustering</h1>
<p>Hierarchical clustering (HC) is an unsupervised machine learning approach. HC can provide an initial estimation of the number of possible groups and information on their members. However, the effectivity of HC will depend on the experimental dataset, the metric used, and the agglomeration algorithm applied. For an unknown reason (and based on the personal experience of the author working in numerical taxonomy), Ward’s agglomeration algorithm performs much better on biological experimental datasets than the rest of the available algorithms like UPGMA, UPGMC, etc.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmgm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">DMRs</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">dmgm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dmgm</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">dmgm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">ns</span>
<span class="va">sampleNames</span> <span class="op">&lt;-</span> <span class="va">ns</span>

<span class="va">hc</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html">hclust</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html">dist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span> <span class="va">dmgm</span> <span class="op">)</span><span class="op">)</span><span class="op">^</span><span class="fl">2</span>, method <span class="op">=</span> <span class="st">'ward.D'</span><span class="op">)</span>
<span class="va">hc.rsq</span> <span class="op">=</span> <span class="va">hc</span>
<span class="va">hc.rsq</span><span class="op">$</span><span class="va">height</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">sqrt</a></span><span class="op">(</span> <span class="va">hc</span><span class="op">$</span><span class="va">height</span> <span class="op">)</span></code></pre></div>
<div id="dendrogram" class="section level2">
<h2 class="hasAnchor">
<a href="#dendrogram" class="anchor"></a>Dendrogram</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">colors</span> <span class="op">=</span> <span class="va">sampleNames</span> 
<span class="va">colors</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"C"</span>, <span class="va">colors</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"green4"</span>
<span class="va">colors</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"T[6-9]{1}"</span>, <span class="va">colors</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"red"</span>
<span class="va">colors</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"T10"</span>, <span class="va">colors</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"red"</span>
<span class="va">colors</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"T[1-5]"</span>, <span class="va">colors</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"blue"</span>

<span class="co"># rgb(red, green, blue, alpha, names = NULL, maxColorValue = 1)</span>
<span class="va">clusters.color</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html">rgb</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.7</span>, <span class="fl">0</span>, <span class="fl">0.1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html">rgb</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">0.1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html">rgb</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0.2</span>, <span class="fl">0</span>, <span class="fl">0.1</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>font.lab<span class="op">=</span><span class="fl">2</span>,font<span class="op">=</span><span class="fl">3</span>,font.axis<span class="op">=</span><span class="fl">2</span>, mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">3</span>,<span class="fl">2</span>,<span class="fl">0</span><span class="op">)</span>, family<span class="op">=</span><span class="st">"serif"</span> , lwd <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/as.phylo.html">as.phylo</a></span><span class="op">(</span><span class="va">hc.rsq</span><span class="op">)</span>, tip.color <span class="op">=</span> <span class="va">colors</span>, label.offset <span class="op">=</span> <span class="fl">0.5</span>, font <span class="op">=</span> <span class="fl">2</span>, cex <span class="op">=</span> <span class="fl">0.9</span>,
     edge.width  <span class="op">=</span> <span class="fl">0.4</span>, direction <span class="op">=</span> <span class="st">"downwards"</span>, no.margin <span class="op">=</span> <span class="cn">FALSE</span>,
     align.tip.label <span class="op">=</span> <span class="cn">TRUE</span>, adj <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ape/man/axisPhylo.html">axisPhylo</a></span><span class="op">(</span> <span class="fl">2</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.4</span>, cex.axis <span class="op">=</span> <span class="fl">1.4</span>, hadj <span class="op">=</span> <span class="fl">0.8</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.01</span> <span class="op">)</span>
<span class="co">## cuts = c(xleft, ybottom, xright, ytop)</span>
<span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/hclust_rect.html">hclust_rect</a></span><span class="op">(</span><span class="va">hc.rsq</span>, k <span class="op">=</span> <span class="fl">3L</span>, border <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"green4"</span>, <span class="st">"blue"</span>, <span class="st">"red"</span><span class="op">)</span>, 
            color <span class="op">=</span> <span class="va">clusters.color</span>, cuts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.56</span>, <span class="fl">1.4</span>, <span class="fl">0.41</span>, <span class="fl">106</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="pca_lda_with_methylit_files/figure-html/dendrogram-1.png" width="700"></p>
<p>Here, we have use function <a href="https://www.rdocumentation.org/packages/ape/versions/5.3/topics/as.phylo">as.phylo</a> from the R package <a href="https://www.rdocumentation.org/packages/ape/versions/5.3">ape</a> for better dendrogram visualization and function <a href="https://genomaths.github.io/MethylIT_utils_HTML_Manual/hclust_rect.html">hclust_rect</a> from <a href="https://github.com/genomaths/MethylIT.utils">MethylIT.utils</a> R package to draw rectangles with background colors around the branches of a dendrogram highlighting the corresponding clusters.</p>
</div>
</div>
<div id="pca-lda" class="section level1">
<h1 class="hasAnchor">
<a href="#pca-lda" class="anchor"></a>PCA + LDA</h1>
<p>MethylIT function <a href="https://genomaths.github.io/MethylIT_HTML_Manual/pcaLDA.html">pcaLDA</a> will be used to perform the PCA and PCA + LDA analyses. The function returns a list of two objects: 1) ‘lda’: an object of class ‘lda’ from package ‘MASS.’ 2) ‘pca’: an object of class ‘prcomp’ from package ‘stats.’ For information on how to use these objects see ?lda and ?prcomp.</p>
<p>Unlike hierarchical clustering (HC), LDA is a supervised machine learning approach. So, we must provide a prior classification of the individuals, which can be derived, for example, from the HC, or from a previous exploratory analysis with PCA.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># A prior classification derived from HC</span>
<span class="va">grps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html">cutree</a></span><span class="op">(</span><span class="va">hc</span>, k <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">grps</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">grps</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"CT"</span>
<span class="va">grps</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="fl">2</span>, <span class="va">grps</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"T1"</span>
<span class="va">grps</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="fl">3</span>, <span class="va">grps</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"T2"</span>
<span class="va">grps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">grps</span><span class="op">)</span>

<span class="va">ld</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pcaLDA.html">pcaLDA</a></span><span class="op">(</span>data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">dmgm</span><span class="op">)</span>, grouping <span class="op">=</span> <span class="va">grps</span>, n.pc <span class="op">=</span> <span class="fl">3</span>, max.pc <span class="op">=</span> <span class="fl">3</span>,
             scale <span class="op">=</span> <span class="cn">FALSE</span>, center <span class="op">=</span> <span class="cn">FALSE</span>, tol <span class="op">=</span> <span class="fl">1e-6</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">ld</span><span class="op">$</span><span class="va">pca</span><span class="op">)</span></code></pre></div>
<pre><code>## Importance of first k=3 (out of 15) components:
##                             PC1      PC2      PC3
## Standard deviation     308.6113 57.35742 55.39723
## Proportion of Variance   0.7802  0.02695  0.02514
## Cumulative Proportion    0.7802  0.80718  0.83232</code></pre>
<p>We may retain enough components so that the cumulative percent of variance accounted for at least 70 to 80%. By setting <span class="math inline">\(scale=TRUE\)</span> and <span class="math inline">\(center=TRUE\)</span>, we could have different results and would improve or not our results. In particular, these settings are essentials if the <em>N</em>-dimensional space is integrated by variables from different measurement scales/units, for example, Kg and g, or Kg and Km.</p>
<div id="pca" class="section level2">
<h2 class="hasAnchor">
<a href="#pca" class="anchor"></a>PCA</h2>
<p>The individual coordinates in the principal components (PCs) are returned by function <a href="https://genomaths.github.io/MethylIT_HTML_Manual/pcaLDA.html">pcaLDA</a>. In the current case, based on the cumulative proportion of variance, the two firsts PCs carried about the 84% of the total sample variance and could split the sample into meaningful groups.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pca.coord</span> <span class="op">&lt;-</span> <span class="va">ld</span><span class="op">$</span><span class="va">pca</span><span class="op">$</span><span class="va">x</span>
<span class="va">pca.coord</span></code></pre></div>
<pre><code>##            PC1         PC2         PC3
## C1   -61.67008  -2.3733213  -1.5740632
## C2   -56.54666   0.4351670   1.9959339
## C3   -52.62494   0.8906694   0.3794566
## C4  -215.71556  -7.6261738  -3.1246417
## C5  -246.76559  -9.5185006   7.7799157
## T1  -320.91917  -7.1177724 -17.4973904
## T2  -301.37699  17.6148347 -29.8832995
## T3  -309.33731 -21.5161358 -16.7421341
## T4  -314.15070 -11.5680851 -30.1647704
## T5  -314.34414  -6.5664808 -12.5696682
## T6  -373.21980 -49.8040838 -74.8479801
## T7  -382.34810 135.0519574  79.5138524
## T8  -376.60785 -86.0682721 -28.7369947
## T9  -390.74442 -73.0496972 156.9317441
## T10 -386.22396 106.9511328 -54.4206297</code></pre>
</div>
<div id="graphic-pc1-vs-pc2" class="section level2">
<h2 class="hasAnchor">
<a href="#graphic-pc1-vs-pc2" class="anchor"></a>Graphic PC1 vs PC2</h2>
<p>Next, the graphic for individual coordinates in the two firsts PCs can be easily visualized now:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">pca.coord</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>, subgrp <span class="op">=</span> <span class="va">grps</span><span class="op">)</span>

<span class="va">p0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>
  axis.text.x  <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span> face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">18</span>, color<span class="op">=</span><span class="st">"black"</span>,
                               <span class="co"># hjust = 0.5, vjust = 0.5, </span>
                               family <span class="op">=</span> <span class="st">"serif"</span>, angle <span class="op">=</span> <span class="fl">0</span>,
                               margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0</span>, unit <span class="op">=</span> <span class="st">"pt"</span> <span class="op">)</span><span class="op">)</span>,
  axis.text.y  <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span> face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">18</span>, color<span class="op">=</span><span class="st">"black"</span>, 
                               family <span class="op">=</span> <span class="st">"serif"</span>,
                               margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span> <span class="fl">0</span>,<span class="fl">0.1</span>,<span class="fl">0</span>,<span class="fl">0</span>, unit <span class="op">=</span> <span class="st">"mm"</span> <span class="op">)</span><span class="op">)</span>,
  axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, family <span class="op">=</span> <span class="st">"serif"</span>, size <span class="op">=</span> <span class="fl">18</span>,
                              color<span class="op">=</span><span class="st">"black"</span>, vjust <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span>,
  axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, family <span class="op">=</span> <span class="st">"serif"</span>, size <span class="op">=</span> <span class="fl">18</span>,
                              color<span class="op">=</span><span class="st">"black"</span>, 
                              margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span> <span class="fl">0</span>,<span class="fl">2</span>,<span class="fl">0</span>,<span class="fl">0</span>, unit <span class="op">=</span> <span class="st">"mm"</span> <span class="op">)</span> <span class="op">)</span>,
  legend.title<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
  legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span>, family <span class="op">=</span> <span class="st">"serif"</span><span class="op">)</span>,
  legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.88</span>, <span class="fl">0.14</span><span class="op">)</span>,
  
  panel.border <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_rect</a></span><span class="op">(</span>fill<span class="op">=</span><span class="cn">NA</span>, colour <span class="op">=</span> <span class="st">"black"</span>,size<span class="op">=</span><span class="fl">0.07</span><span class="op">)</span>,
  panel.grid.minor <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_line</a></span><span class="op">(</span>color<span class="op">=</span> <span class="st">"white"</span>,size <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span>,
  axis.ticks <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_line</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>, axis.ticks.length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="st">"mm"</span><span class="op">)</span>,
  plot.margin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, <span class="st">"lines"</span><span class="op">)</span><span class="op">)</span> 

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">dt</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">PC1</span>, y <span class="op">=</span> <span class="va">PC2</span>, colour <span class="op">=</span> <span class="va">grps</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"white"</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"white"</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"green4"</span>,<span class="st">"blue"</span>,<span class="st">"brown1"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_ellipse.html">stat_ellipse</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">PC1</span>, y <span class="op">=</span> <span class="va">PC2</span>, fill <span class="op">=</span> <span class="va">subgrp</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">dt</span>, type <span class="op">=</span> <span class="st">"norm"</span>,
               geom <span class="op">=</span> <span class="st">"polygon"</span>, level <span class="op">=</span> <span class="fl">0.5</span>, alpha<span class="op">=</span><span class="fl">0.2</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"green4"</span>,<span class="st">"blue"</span>,<span class="st">"brown1"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="va">p0</span></code></pre></div>
<pre><code>## Too few points to calculate an ellipse</code></pre>
<p><img src="pca_lda_with_methylit_files/figure-html/pcaGraphic-1.png" width="700"></p>
</div>
<div id="graphic-ld1-vs-ld2" class="section level2">
<h2 class="hasAnchor">
<a href="#graphic-ld1-vs-ld2" class="anchor"></a>Graphic LD1 vs LD2</h2>
<p>In the current case, better resolution is obtained with the linear discriminant functions, which is based on the three firsts PCs. Notice that the number principal components used the LDA step must be lower than the number of individuals (<span class="math inline">\(N\)</span>) divided by 3: <span class="math inline">\(N/3\)</span>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ind.coord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html">predict</a></span><span class="op">(</span><span class="va">ld</span>, newdata <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">dmgm</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"scores"</span><span class="op">)</span>
<span class="va">dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">ind.coord</span>, subgrp <span class="op">=</span> <span class="va">grps</span><span class="op">)</span>

<span class="va">p0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>
  axis.text.x  <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span> face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">18</span>, color<span class="op">=</span><span class="st">"black"</span>,
                               <span class="co"># hjust = 0.5, vjust = 0.5, </span>
                               family <span class="op">=</span> <span class="st">"serif"</span>, angle <span class="op">=</span> <span class="fl">0</span>,
                               margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0</span>, unit <span class="op">=</span> <span class="st">"pt"</span> <span class="op">)</span><span class="op">)</span>,
  axis.text.y  <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span> face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">18</span>, color<span class="op">=</span><span class="st">"black"</span>, 
                               family <span class="op">=</span> <span class="st">"serif"</span>,
                               margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span> <span class="fl">0</span>,<span class="fl">0.1</span>,<span class="fl">0</span>,<span class="fl">0</span>, unit <span class="op">=</span> <span class="st">"mm"</span> <span class="op">)</span><span class="op">)</span>,
  axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, family <span class="op">=</span> <span class="st">"serif"</span>, size <span class="op">=</span> <span class="fl">18</span>,
                              color<span class="op">=</span><span class="st">"black"</span>, vjust <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span>,
  axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, family <span class="op">=</span> <span class="st">"serif"</span>, size <span class="op">=</span> <span class="fl">18</span>,
                              color<span class="op">=</span><span class="st">"black"</span>, 
                              margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span> <span class="fl">0</span>,<span class="fl">2</span>,<span class="fl">0</span>,<span class="fl">0</span>, unit <span class="op">=</span> <span class="st">"mm"</span> <span class="op">)</span> <span class="op">)</span>,
  legend.title<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
  legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span>, family <span class="op">=</span> <span class="st">"serif"</span><span class="op">)</span>,
  legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.15</span>, <span class="fl">0.83</span><span class="op">)</span>,
  
  panel.border <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_rect</a></span><span class="op">(</span>fill<span class="op">=</span><span class="cn">NA</span>, colour <span class="op">=</span> <span class="st">"black"</span>,size<span class="op">=</span><span class="fl">0.07</span><span class="op">)</span>,
  panel.grid.minor <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_line</a></span><span class="op">(</span>color<span class="op">=</span> <span class="st">"white"</span>,size <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span>,
  axis.ticks <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_line</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>, axis.ticks.length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="st">"mm"</span><span class="op">)</span>,
  plot.margin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, <span class="st">"lines"</span><span class="op">)</span><span class="op">)</span> 

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">dt</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">LD1</span>, y <span class="op">=</span> <span class="va">LD2</span>, colour <span class="op">=</span> <span class="va">grps</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"white"</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"white"</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"green4"</span>,<span class="st">"blue"</span>,<span class="st">"brown1"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_ellipse.html">stat_ellipse</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">LD1</span>, y <span class="op">=</span> <span class="va">LD2</span>, fill <span class="op">=</span> <span class="va">subgrp</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">dt</span>, type <span class="op">=</span> <span class="st">"norm"</span>,
               geom <span class="op">=</span> <span class="st">"polygon"</span>, level <span class="op">=</span> <span class="fl">0.5</span>, alpha<span class="op">=</span><span class="fl">0.2</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"green4"</span>,<span class="st">"blue"</span>,<span class="st">"brown1"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="va">p0</span></code></pre></div>
<pre><code>## Too few points to calculate an ellipse</code></pre>
<p><img src="pca_lda_with_methylit_files/figure-html/ldaGraphic-1.png" width="700"></p>
</div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body">
<div id="ref-Liese2006" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">F. Liese, I. Vajda, <span class="nocase">On divergences and informations in statistics and information theory</span>. <em>IEEE Transactions on Information Theory</em> <strong>52</strong>, 4394–4412 (2006).</div>
</div>
<div id="ref-Stevens2009" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">J. P. Stevens, <em><span class="nocase">Applied Multivariate Statistics for the Social Sciences</span></em>, Fifth Edit (Routledge Academic, 2009).</div>
</div>
</div>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo()</a></code> on the system on which this document was compiled running pandoc 2.11.4:</p>
<pre><code>## R version 4.1.1 (2021-08-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.3.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] ape_5.5              ggplot2_3.3.5        MethylIT.utils_0.1  
##  [4] MethylIT_0.3.2.3     rtracklayer_1.52.1   GenomicRanges_1.44.0
##  [7] GenomeInfoDb_1.28.4  IRanges_2.26.0       S4Vectors_0.30.1    
## [10] BiocGenerics_0.38.0 
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_2.0-2            rjson_0.2.20               
##   [3] modeltools_0.2-23           ellipsis_0.3.2             
##   [5] class_7.3-19                rprojroot_2.0.2            
##   [7] XVector_0.32.0              fs_1.5.0                   
##   [9] proxy_0.4-26                farver_2.1.0               
##  [11] flexmix_2.3-17              bit64_4.0.5                
##  [13] AnnotationDbi_1.54.1        prodlim_2019.11.13         
##  [15] fansi_0.5.0                 lubridate_1.7.10           
##  [17] codetools_0.2-18            splines_4.1.1              
##  [19] cachem_1.0.5                knitr_1.33                 
##  [21] Formula_1.2-4               jsonlite_1.7.2             
##  [23] pROC_1.17.0.1               Rsamtools_2.8.0            
##  [25] caret_6.0-88                annotate_1.70.0            
##  [27] png_0.1-7                   compiler_4.1.1             
##  [29] httr_1.4.2                  assertthat_0.2.1           
##  [31] Matrix_1.3-4                fastmap_1.1.0              
##  [33] htmltools_0.5.1.1           tools_4.1.1                
##  [35] gtable_0.3.0                glue_1.4.2                 
##  [37] GenomeInfoDbData_1.2.6      reshape2_1.4.4             
##  [39] dplyr_1.0.7                 Rcpp_1.0.7                 
##  [41] Biobase_2.52.0              jquerylib_0.1.4            
##  [43] pkgdown_1.6.1               vctrs_0.3.8                
##  [45] Biostrings_2.60.2           nlme_3.1-153               
##  [47] betareg_3.1-4               iterators_1.0.13           
##  [49] lmtest_0.9-38               timeDate_3043.102          
##  [51] gower_0.2.2                 xfun_0.24                  
##  [53] stringr_1.4.0               lifecycle_1.0.0            
##  [55] restfulr_0.0.13             XML_3.99-0.6               
##  [57] zoo_1.8-9                   zlibbioc_1.38.0            
##  [59] MASS_7.3-54                 scales_1.1.1               
##  [61] ipred_0.9-11                ragg_1.1.3                 
##  [63] MatrixGenerics_1.4.3        sandwich_3.0-1             
##  [65] SummarizedExperiment_1.22.0 yaml_2.2.1                 
##  [67] memoise_2.0.0               sass_0.4.0                 
##  [69] rpart_4.1-15                stringi_1.6.2              
##  [71] RSQLite_2.2.8               highr_0.9                  
##  [73] genefilter_1.74.0           BiocIO_1.2.0               
##  [75] desc_1.3.0                  foreach_1.5.1              
##  [77] e1071_1.7-7                 BiocParallel_1.26.2        
##  [79] lava_1.6.9                  rlang_0.4.11               
##  [81] pkgconfig_2.0.3             systemfonts_1.0.2          
##  [83] matrixStats_0.59.0          bitops_1.0-7               
##  [85] evaluate_0.14               lattice_0.20-45            
##  [87] purrr_0.3.4                 labeling_0.4.2             
##  [89] GenomicAlignments_1.28.0    recipes_0.1.16             
##  [91] bit_4.0.4                   tidyselect_1.1.1           
##  [93] plyr_1.8.6                  magrittr_2.0.1             
##  [95] nls2_0.2                    R6_2.5.0                   
##  [97] generics_0.1.0              DelayedArray_0.18.0        
##  [99] DBI_1.1.1                   pillar_1.6.1               
## [101] withr_2.4.2                 survival_3.2-13            
## [103] KEGGREST_1.32.0             RCurl_1.98-1.3             
## [105] nnet_7.3-16                 tibble_3.1.2               
## [107] crayon_1.4.1                utf8_1.2.1                 
## [109] rmarkdown_2.10              grid_4.1.1                 
## [111] minpack.lm_1.2-1            data.table_1.14.0          
## [113] blob_1.2.2                  ModelMetrics_1.2.2.2       
## [115] digest_0.6.27               xtable_1.8-4               
## [117] textshaping_0.3.5           munsell_0.5.0              
## [119] bslib_0.2.5.1</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
