<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Principal Components and Linear Discriminant Analyses with Methyl-IT • MethylIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Principal Components and Linear Discriminant Analyses with Methyl-IT">
<meta property="og:description" content="MethylIT">
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Principal Components and Linear Discriminant Analyses with Methyl-IT</h1>
            <h3 data-toc-skip class="subtitle"><h1>
A <a href="https://github.com/genomaths/MethylIT" class="external-link">Methyl-IT</a> user guide</h1></h3>
                        <h4 data-toc-skip class="author">
<div class="line-block">Robersy Sanchez<br><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a><br>
Mackenzie’s lab</div>
<div class="line-block">Department of Biology and Plant Science.<br>
Pennsylvania State University, University Park, PA 16802<br>
Maintainer: Robersy Sanchez</div>
</h4>
            
            <h4 data-toc-skip class="date">06 April 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/vignettes/pca_lda_with_methylit.Rmd" class="external-link"><code>vignettes/pca_lda_with_methylit.Rmd</code></a></small>
      <div class="hidden name"><code>pca_lda_with_methylit.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>When methylation analysis is intended for diagnostic/prognostic purposes, for example, in clinical applications for patient diagnostics, to know whether the patient would be in healthy or disease stage we would like to have a good predictor tool in our side. It turns out that classical machine learning (ML) tools like hierarchical clustering, principal components and linear discriminant analysis can help us to reach such a goal. The current Methyl-IT downstream analysis is equipped with the mentioned ML tools.</p>
    </div>
    
<style type="text/css">

.list-group-item.active, .list-group-item.active:focus, 
.list-group-item.active:hover {
    z-index: 2;
    color: #fff;
    background-color: #337ab7;
    border-color: #337ab7;
}

.tocify-subheader .tocify-item {
  font-size: 0.75em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify-header .tocify-item {
  font-size: 0.8em;
  padding-left: 20px;
  text-indent: 0;
}

body{ /* Normal  */
      font-size: 16px;
      font-family: "Times New Roman", Times, serif;
      text-align: justify
  }
td {  /* Table  */
  font-size: 8px;
}

h1.title {
   font-size: 36px;
   font-family: "Times New Roman", Times, serif;
   color: DarkRed;
   .subTitle {
       font-size: 24px;
       font-family: "Times New Roman", Times, serif;
       color: DarkRed;
  }
}

h1 { /* Header 1 */
  font-size: 24px;
  font-family: "Times New Roman", Times, serif;
  color: DarkBlue;
}
h2 { /* Header 2 */
    font-size: 22px;
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
h3 { /* Header 3 */
   font-size: 18px;
   color: DarkBlue;
   font-family: "Times New Roman", Times, serif;
}
code.r{ /* Code block */
    font-size: 12px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>
<div class="section level2">
<h2 id="background">Background<a class="anchor" aria-label="anchor" href="#background"></a>
</h2>
<p>The discrimination of the methylation signal from the stochastic methylation background resultant from the standard (non-stressful) biological processes is a critical step for the genome-wide methylation analysis. Such a discrimination requires for the knowledge of the probability distribution of the information divergence of methylation levels and a proper evaluation of the classification performance of differentially methylated positions (DMPs) into two classes: DMPs from control and DMPs from treatment.</p>
<p>Once a methylation analysis is performed to detect the genome-wide DMPs, a natural next step is to investigate whether the detected DMPs carries enough discriminatory power to discern between individual groups. Then, we must move from the methylation analysis to statistical/machine learning field (multivariate statistics). To reach our goal, each individual can be represented as a vector from the <span class="math inline">\(N\)</span>-dimensional space of DMPs, where DMPs can be expressed in terms of some suitable measurement as function of the methylation levels. These suitable measurement are information divergences, such as total variation distance (<span class="math inline">\(TV_d\)</span>, the absolute value of methylation levels) and Hellinger divergence (<span class="math inline">\(H\)</span>). It is worthy to observe that methylation levels are uncertainty measurements and, therefore, they do not carry information per se. Information is only carried on measurements expressing uncertainty variations like <span class="math inline">\(TV_d\)</span>, <span class="math inline">\(H\)</span>, Jensen-Shannon Divergence, and in general, the so called information theoretical divergences <span class="citation">(1)</span>.</p>
<p>The <span class="math inline">\(N\)</span>-dimensional space of DMPs is, usually, too big. So, a first dimension reduction is required. This can be done by splitting the chromosomes into non-overlapping genomic region and next computing some statistic like the <em>sum</em> or <em>mean</em> of some information divergences of methylation levels at DMPs inside of each region. Still we will have to deal with a multidimensional space, but acceptable for a meaningful application of hierarchical clustering (HC), principal component analysis (PCA) and linear discriminant analysis (LDA). In the current examples, Methyl-IT methylation analysis will be applied to a dataset of simulated samples to detect DMPs on then. Next, after space dimension reduction, the mentioned machine learning approached will be applied.</p>
</div>
<div class="section level2">
<h2 id="data-generation">Data generation<a class="anchor" aria-label="anchor" href="#data-generation"></a>
</h2>
<p>For the current example on methylation analysis with Methyl-IT we will use simulated data. Read-count matrices of methylated and unmethylated cytosine are generated with MethylIT.utils function <a href="https://genomaths.github.io/MethylIT_utils_HTML_Manual/simulateCounts.html" class="external-link">simulateCounts</a>. A basic example generating datasets is given in the web page: <a href="https://genomaths.github.io/Methylation_analysis_with_Methyl-IT.html" class="external-link">Methylation analysis with Methyl-IT</a></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/MethylIT" class="external-link">MethylIT</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MethylIT.utils</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://ape-package.ird.fr/" class="external-link">ape</a></span><span class="op">)</span>

<span class="va">alpha.ct</span> <span class="op">&lt;-</span> <span class="fl">0.019</span>
<span class="va">alpha.g1</span> <span class="op">&lt;-</span> <span class="fl">0.022</span>
<span class="va">alpha.g2</span> <span class="op">&lt;-</span> <span class="fl">0.025</span>

<span class="co"># The number of cytosine sites to generate</span>
<span class="va">sites</span> <span class="op">=</span> <span class="fl">1.5e5</span> 
<span class="co"># Set a seed for pseudo-random number generation</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">124</span><span class="op">)</span>
<span class="va">control.nam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"C1"</span>, <span class="st">"C2"</span>, <span class="st">"C3"</span>, <span class="st">"C4"</span>, <span class="st">"C5"</span><span class="op">)</span>
<span class="va">treatment.nam1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T1"</span>, <span class="st">"T2"</span>, <span class="st">"T3"</span>, <span class="st">"T4"</span>, <span class="st">"T5"</span><span class="op">)</span>
<span class="va">treatment.nam2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T6"</span>, <span class="st">"T7"</span>, <span class="st">"T8"</span>, <span class="st">"T9"</span>, <span class="st">"T10"</span><span class="op">)</span>

<span class="co"># Reference group </span>
<span class="va">ref0</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html" class="external-link">simulateCounts</a></span><span class="op">(</span>num.samples <span class="op">=</span> <span class="fl">3</span>, sites <span class="op">=</span> <span class="va">sites</span>, alpha <span class="op">=</span> <span class="va">alpha.ct</span>, 
                      beta <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">50</span>, theta <span class="op">=</span> <span class="fl">4.5</span>, 
                      sample.ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"R1"</span>, <span class="st">"R2"</span>, <span class="st">"R3"</span><span class="op">)</span><span class="op">)</span>
<span class="co"># Control group </span>
<span class="va">ctrl</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html" class="external-link">simulateCounts</a></span><span class="op">(</span>num.samples <span class="op">=</span> <span class="fl">5</span>, sites <span class="op">=</span> <span class="va">sites</span>, alpha <span class="op">=</span> <span class="va">alpha.ct</span>, 
                      beta <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">50</span>, theta <span class="op">=</span> <span class="fl">4.5</span>, 
                      sample.ids <span class="op">=</span> <span class="va">control.nam</span><span class="op">)</span>
<span class="co"># Treatment group II</span>
<span class="va">treat</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html" class="external-link">simulateCounts</a></span><span class="op">(</span>num.samples <span class="op">=</span> <span class="fl">5</span>, sites <span class="op">=</span> <span class="va">sites</span>, alpha <span class="op">=</span> <span class="va">alpha.g1</span>,
                       beta <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">50</span>, theta <span class="op">=</span> <span class="fl">4.5</span>, 
                       sample.ids <span class="op">=</span> <span class="va">treatment.nam1</span><span class="op">)</span>

<span class="co"># Treatment group II</span>
<span class="va">treat2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/simulateCounts.html" class="external-link">simulateCounts</a></span><span class="op">(</span>num.samples <span class="op">=</span> <span class="fl">5</span>, sites <span class="op">=</span> <span class="va">sites</span>, alpha <span class="op">=</span> <span class="va">alpha.g2</span>,
                        beta <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">50</span>, theta <span class="op">=</span> <span class="fl">4.5</span>, 
                        sample.ids <span class="op">=</span> <span class="va">treatment.nam2</span><span class="op">)</span></code></pre></div>
<p>A reference sample (virtual individual) is created using individual samples from the control population using function <a href="https://genomaths.github.io/MethylIT_HTML_Manual/poolFromGRlist.html" class="external-link">poolFromGRlist</a>. The reference sample is further used to compute the information divergences of methylation levels, <span class="math inline">\(TV_d\)</span> and <span class="math inline">\(H\)</span>, with function <a href="https://genomaths.github.io/MethylIT_HTML_Manual/estimateDivergence.html" class="external-link">estimateDivergence</a>. This is a first fundamental step to remove the background noise (fluctuations) generated by the inherent stochasticity of the molecular processes in the cells.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># === Methylation level divergences ===</span>
<span class="co"># Reference sample</span>
<span class="va">ref</span> <span class="op">=</span> <span class="fu"><a href="../reference/poolFromGRlist.html">poolFromGRlist</a></span><span class="op">(</span><span class="va">ref0</span>, stat <span class="op">=</span> <span class="st">"mean"</span>, num.cores <span class="op">=</span> <span class="fl">4L</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="va">divs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/estimateDivergence.html">estimateDivergence</a></span><span class="op">(</span>ref <span class="op">=</span> <span class="va">ref</span>, indiv <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">ctrl</span>, <span class="va">treat</span>, <span class="va">treat2</span><span class="op">)</span>, Bayesian <span class="op">=</span> <span class="cn">TRUE</span>, 
                           num.cores <span class="op">=</span> <span class="fl">6L</span>, percentile <span class="op">=</span> <span class="fl">1</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="co"># To remove hd == 0 to estimate. The methylation signal only is given for  </span>
<span class="va">divs</span> <span class="op">=</span> <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">divs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">div</span><span class="op">)</span> <span class="va">div</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">div</span><span class="op">$</span><span class="va">hdiv</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span> <span class="op">]</span>, keep.attr <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">divs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">divs</span><span class="op">)</span>
<span class="va">divs</span></code></pre></div>
<pre><code><span class="co">## InfDiv object of length: 15</span>
<span class="co">## ------- </span>
<span class="co">## $C1</span>
<span class="co">## GRanges object with 26720 ranges and 9 metadata columns:</span>
<span class="co">##           seqnames    ranges strand |        c1        t1        c2        t2</span>
<span class="co">##              &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">##       [1]        1         7      * |         2        42         6        38</span>
<span class="co">##       [2]        1        12      * |        23        62         0        85</span>
<span class="co">##       [3]        1        19      * |        12        27         0        39</span>
<span class="co">##       [4]        1        22      * |        13        27         0        40</span>
<span class="co">##       [5]        1        31      * |        58        49        80        27</span>
<span class="co">##       ...      ...       ...    ... .       ...       ...       ...       ...</span>
<span class="co">##   [26716]        1    149977      * |        11        35        34        12</span>
<span class="co">##   [26717]        1    149979      * |        16        34         0        50</span>
<span class="co">##   [26718]        1    149990      * |        24        54         0        78</span>
<span class="co">##   [26719]        1    149994      * |        25        51        76         0</span>
<span class="co">##   [26720]        1    149995      * |        16        63         0        79</span>
<span class="co">##                  p1         p2         TV     bay.TV      hdiv</span>
<span class="co">##           &lt;numeric&gt;  &lt;numeric&gt;  &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">##       [1] 0.0721422 0.13768432  0.0909091  0.0655421  0.526471</span>
<span class="co">##       [2] 0.2659280 0.00327293 -0.2705882 -0.2626551 19.800862</span>
<span class="co">##       [3] 0.2920976 0.00697775 -0.3076923 -0.2851199  9.313997</span>
<span class="co">##       [4] 0.3068138 0.00681017 -0.3250000 -0.3000036 10.213202</span>
<span class="co">##       [5] 0.5191743 0.73900460  0.2056075  0.2198303  5.688769</span>
<span class="co">##       ...       ...        ...        ...        ...       ...</span>
<span class="co">##   [26716]  0.235823 0.71967696   0.500000   0.483854   11.7687</span>
<span class="co">##   [26717]  0.305641 0.00549133  -0.320000  -0.300150   13.0602</span>
<span class="co">##   [26718]  0.299161 0.00356061  -0.307692  -0.295601   20.8075</span>
<span class="co">##   [26719]  0.318165 0.98256166   0.671053   0.664396   51.1029</span>
<span class="co">##   [26720]  0.203882 0.00351646  -0.202532  -0.200366   13.2064</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="co">## </span>
<span class="co">## ...</span>
<span class="co">## &lt;14 more GRanges elements&gt;</span>
<span class="co">## -------</span></code></pre>
<p>To get some statistical description about the sample is useful. Here, empirical critical values for the probability distribution of <span class="math inline">\(H\)</span> and <span class="math inline">\(TV\)</span> can is obtained using <em>quantile</em> function from the R package <em>stats</em>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">critical.val</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">divs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">x</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="va">x</span><span class="op">$</span><span class="va">hdiv</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span>
  <span class="va">hd.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">hdiv</span>, <span class="fl">0.95</span><span class="op">)</span>
  <span class="va">tv.95</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">bay.TV</span><span class="op">)</span>, <span class="fl">0.95</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>tv <span class="op">=</span> <span class="va">tv.95</span>, hd <span class="op">=</span> <span class="va">hd.95</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span><span class="op">)</span>
<span class="va">critical.val</span></code></pre></div>
<pre><code><span class="co">##        tv.95%    hd.95%</span>
<span class="co">## C1  0.6523406  62.26763</span>
<span class="co">## C2  0.6500765  60.89674</span>
<span class="co">## C3  0.6508274  61.48960</span>
<span class="co">## C4  0.4858965  56.63342</span>
<span class="co">## C5  0.4970348  58.16023</span>
<span class="co">## T1  0.9236731 126.34191</span>
<span class="co">## T2  0.9136241 120.45367</span>
<span class="co">## T3  0.9137810 121.99373</span>
<span class="co">## T4  0.9192143 124.54298</span>
<span class="co">## T5  0.9225918 124.25236</span>
<span class="co">## T6  0.9447922 132.69308</span>
<span class="co">## T7  0.9447348 134.88441</span>
<span class="co">## T8  0.9408379 131.80496</span>
<span class="co">## T9  0.9482600 134.68997</span>
<span class="co">## T10 0.9456255 133.87163</span></code></pre>
</div>
<div class="section level2">
<h2 id="modeling-the-methylation-signal">Modeling the methylation signal<a class="anchor" aria-label="anchor" href="#modeling-the-methylation-signal"></a>
</h2>
<p>Here, the methylation signal is expressed in terms of Hellinger divergence of methylation levels. Here, the signal distribution is modelled by a Weibull probability distribution model. Basically, the model could be a member of the generalized gamma distribution family. For example, it could be gamma distribution, Weibull, or log-normal. To describe the signal, we may prefer a model with a cross-validations: R.Cross.val &gt; 0.95. Cross-validations for the nonlinear regressions are performed in each methylome as described in <span class="citation">(2)</span>. The non-linear fit is performed through the function <a href="https://genomaths.github.io/MethylIT_HTML_Manual/nonlinearFitDist.html" class="external-link">nonlinearFitDist</a></p>
<p>The best model excluding “GGamma3P” model, for which the computation is computationally expensive:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Weibull2P"</span>, <span class="st">"Weibull3P"</span>, <span class="st">"Gamma2P"</span>, <span class="st">"Gamma3P"</span><span class="op">)</span>
<span class="va">gof</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gofReport.html">gofReport</a></span><span class="op">(</span>HD <span class="op">=</span> <span class="va">divs</span>, column <span class="op">=</span> <span class="fl">9L</span>, model <span class="op">=</span> <span class="va">d</span>, num.cores <span class="op">=</span> <span class="fl">6L</span>,
                    output <span class="op">=</span> <span class="st">"all"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gof</span><span class="op">$</span><span class="va">bestModel</span></code></pre></div>
<pre><code><span class="co">##          C1          C2          C3          C4          C5          T1 </span>
<span class="co">## "Weibull3P" "Weibull3P" "Weibull3P" "Weibull3P" "Weibull3P" "Weibull3P" </span>
<span class="co">##          T2          T3          T4          T5          T6          T7 </span>
<span class="co">## "Weibull3P" "Weibull3P" "Weibull3P" "Weibull3P" "Weibull3P" "Weibull3P" </span>
<span class="co">##          T8          T9         T10 </span>
<span class="co">## "Weibull3P" "Weibull3P" "Weibull3P"</span></code></pre>
<p>Although we did not included the main member of the generalized gamma distribution family (“GGamma3P”), the Cross-validations correlation coefficient <em>w3p_R.Cross.val</em> values above 0.99 gives us some peace of mind on our confidence on the model prediction power/performance when confronted with external data sets (AIC and BIC are not conceived to provide such an information).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gof</span><span class="op">$</span><span class="va">stats</span></code></pre></div>
<pre><code><span class="co">##       w2p_AIC w2p_R.Cross.val   w3p_AIC w3p_R.Cross.val   g2p_AIC</span>
<span class="co">## C1  -172642.2       0.9994637 -174557.8       0.9994971 -164820.4</span>
<span class="co">## C2  -175595.0       0.9995131 -177009.2       0.9995357 -167532.7</span>
<span class="co">## C3  -177238.5       0.9995420 -178755.4       0.9995652 -169774.2</span>
<span class="co">## C4  -219212.7       0.9994949 -264109.7       0.9998550 -206296.0</span>
<span class="co">## C5  -217448.4       0.9994364 -273042.8       0.9998800 -205264.1</span>
<span class="co">## T1  -168080.2       0.9972703 -225576.5       0.9994669 -160888.3</span>
<span class="co">## T2  -167285.4       0.9972726 -226887.3       0.9995014 -160748.1</span>
<span class="co">## T3  -170302.3       0.9974335 -227800.1       0.9995012 -163998.4</span>
<span class="co">## T4  -168551.4       0.9973218 -225182.4       0.9994660 -162169.8</span>
<span class="co">## T5  -167122.8       0.9972265 -227209.2       0.9994984 -160694.8</span>
<span class="co">## T6  -173245.9       0.9971879 -229257.5       0.9993946 -165656.8</span>
<span class="co">## T7  -170905.5       0.9970146 -228452.9       0.9993820 -163342.2</span>
<span class="co">## T8  -171372.8       0.9970759 -229287.3       0.9994006 -163649.6</span>
<span class="co">## T9  -170097.3       0.9969613 -227490.8       0.9993671 -162250.6</span>
<span class="co">## T10 -171267.1       0.9970281 -228041.8       0.9993708 -163857.5</span>
<span class="co">##     g2p_R.Cross.val   g3p_AIC g3p_R.Cross.val      bestModel</span>
<span class="co">## C1        0.9993928 -166320.4       0.9994541            w3p</span>
<span class="co">## C2        0.9994499 -168752.0       0.9994940            w3p</span>
<span class="co">## C3        0.9994861 -170987.0       0.9995261            w3p</span>
<span class="co">## C4        0.9993131 -262844.7       0.9998672 Needs revision</span>
<span class="co">## C5        0.9992280 -266936.2       0.9998725            w3p</span>
<span class="co">## T1        0.9963827 -205274.3       0.9989377            w3p</span>
<span class="co">## T2        0.9964683 -206911.8       0.9990147            w3p</span>
<span class="co">## T3        0.9966883 -208522.0       0.9990290            w3p</span>
<span class="co">## T4        0.9965314 -205649.3       0.9989540            w3p</span>
<span class="co">## T5        0.9964093 -206817.8       0.9989988            w3p</span>
<span class="co">## T6        0.9962175 -208522.1       0.9988039            w3p</span>
<span class="co">## T7        0.9959969 -206833.7       0.9987502            w3p</span>
<span class="co">## T8        0.9960768 -207922.9       0.9988034            w3p</span>
<span class="co">## T9        0.9958859 -205571.9       0.9987164            w3p</span>
<span class="co">## T10       0.9960168 -206551.8       0.9987320            w3p</span></code></pre>
</div>
<div class="section level2">
<h2 id="estimation-of-an-optimal-cutpoint-to-discriminate-the-signal-induced-by-the-treatment">Estimation of an optimal cutpoint to discriminate the signal induced by the treatment<a class="anchor" aria-label="anchor" href="#estimation-of-an-optimal-cutpoint-to-discriminate-the-signal-induced-by-the-treatment"></a>
</h2>
<p>The above statistical description of the dataset (evidently) suggests that there two main groups: control and treatments, while treatment group would split into two subgroups of samples. In the current case, to search for a good cutpoint, we do not need to use all the samples. The critical value <span class="math inline">\(H_{\alpha=0.05}=33.51223\)</span> suggests that any optimal cutpoint for the subset of samples T1 to T5 will be optimal for the samples T6 to T10 as well.</p>
<p>Below, we are letting the PCA+LDA model classifier to take the decision on whether a differentially methylated cytosine position is a treatment DMP. To do it, Methyl-IT function <a href="https://genomaths.github.io/MethylIT_HTML_Manual/getPotentialDIMP.html" class="external-link">getPotentialDIMP</a> is used to get methylation signal probabilities of the observed <span class="math inline">\(H\)</span> values for all cytosine site (alpha = 1), in accordance with the 2-parameter Weibull distribution model. Next, this information will be used to identify DMPs using Methyl-IT function <a href="https://genomaths.github.io/MethylIT_HTML_Manual/estimateCutPoint.html" class="external-link">estimateCutPoint</a>. Cytosine positions with <span class="math inline">\(H\)</span> values above the cutpoint are considered DMPs. Finally, a PCA + QDA model classifier will be fitted to classify DMPs into two classes: DMPs from control and those from treatment.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getPotentialDIMP.html">getPotentialDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">divs</span>,
                         nlms <span class="op">=</span> <span class="va">gof</span><span class="op">$</span><span class="va">nlms</span>, 
                         div.col <span class="op">=</span> <span class="fl">9L</span>, 
                         tv.cut <span class="op">=</span> <span class="fl">0.461</span>, 
                         tv.col <span class="op">=</span> <span class="fl">8</span>, 
                         alpha <span class="op">=</span> <span class="fl">0.05</span>, 
                         dist.name <span class="op">=</span> <span class="va">gof</span><span class="op">$</span><span class="va">bestModel</span><span class="op">)</span>

<span class="va">cutp</span> <span class="op">=</span> <span class="fu"><a href="../reference/estimateCutPoint.html">estimateCutPoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">dmps</span>, 
                        simple <span class="op">=</span> <span class="cn">FALSE</span>,
                        column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, TV <span class="op">=</span> <span class="cn">TRUE</span>, 
                                   wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,
                        classifier1 <span class="op">=</span> <span class="st">"pca.lda"</span>, 
                        classifier2 <span class="op">=</span> <span class="st">"pca.qda"</span>, 
                        control.names <span class="op">=</span> <span class="va">control.nam</span>, 
                        treatment.names <span class="op">=</span> <span class="va">treatment.nam1</span>,
                        post.cut <span class="op">=</span> <span class="fl">0.5</span>, 
                        cut.values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">15</span>, <span class="fl">38</span>, <span class="fl">1</span><span class="op">)</span>,
                        clas.perf <span class="op">=</span> <span class="cn">TRUE</span>,
                        prop <span class="op">=</span> <span class="fl">0.6</span>,
                        center <span class="op">=</span> <span class="cn">FALSE</span>, 
                        scale <span class="op">=</span> <span class="cn">FALSE</span>,
                        n.pc <span class="op">=</span> <span class="fl">4</span>, 
                        div.col <span class="op">=</span> <span class="fl">9L</span>, 
                        stat <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>

<span class="va">cutp</span></code></pre></div>
<pre><code><span class="co">## Cutpoint estimation with 'pca.lda' classifier </span>
<span class="co">## Cutpoint search performed using model posterior probabilities </span>
<span class="co">## </span>
<span class="co">## Posterior probability used to get the cutpoint = 0.5 </span>
<span class="co">## Cytosine sites with treatment PostProbCut &gt;= 0.5 have a </span>
<span class="co">## divergence value &gt;= 92.13664 </span>
<span class="co">## </span>
<span class="co">## Optimized statistic: Accuracy = 1 </span>
<span class="co">## Cutpoint = 92.137 </span>
<span class="co">## </span>
<span class="co">## Model classifier 'pca.qda' </span>
<span class="co">## </span>
<span class="co">## The accessible objects in the output list are: </span>
<span class="co">##                    Length Class           Mode     </span>
<span class="co">## cutpoint           1      -none-          numeric  </span>
<span class="co">## testSetPerformance 6      confusionMatrix list     </span>
<span class="co">## testSetModel.FDR   1      -none-          numeric  </span>
<span class="co">## model              2      pcaQDA          list     </span>
<span class="co">## modelConfMatrix    6      confusionMatrix list     </span>
<span class="co">## initModel          1      -none-          character</span>
<span class="co">## postProbCut        1      -none-          numeric  </span>
<span class="co">## postCut            1      -none-          numeric  </span>
<span class="co">## classifier         1      -none-          character</span>
<span class="co">## statistic          1      -none-          character</span>
<span class="co">## optStatVal         1      -none-          numeric  </span>
<span class="co">## cutpData           1      -none-          logical</span></code></pre>
<p>In this case, the cutpoint <span class="math inline">\(H_{cutpoint}=92.138\)</span> is lower from what is expected from the higher treatment empirical <span class="math inline">\(H_{\alpha = 0.05}^{TT_{Emp}}=120.45119\)</span> critical value: <span class="math inline">\(H_{cutpoint} &lt; H_{\alpha = 0.05}^{TT_{Emp}}\)</span>.</p>
<p>The model performance in in the test dataset is:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cutp</span><span class="op">$</span><span class="va">testSetPerformance</span></code></pre></div>
<pre><code><span class="co">## Confusion Matrix and Statistics</span>
<span class="co">## </span>
<span class="co">##           Reference</span>
<span class="co">## Prediction   CT   TT</span>
<span class="co">##         CT 2464    0</span>
<span class="co">##         TT    0 5278</span>
<span class="co">##                                      </span>
<span class="co">##                Accuracy : 1          </span>
<span class="co">##                  95% CI : (0.9995, 1)</span>
<span class="co">##     No Information Rate : 0.6817     </span>
<span class="co">##     P-Value [Acc &gt; NIR] : &lt; 2.2e-16  </span>
<span class="co">##                                      </span>
<span class="co">##                   Kappa : 1          </span>
<span class="co">##                                      </span>
<span class="co">##  Mcnemar's Test P-Value : NA         </span>
<span class="co">##                                      </span>
<span class="co">##             Sensitivity : 1.0000     </span>
<span class="co">##             Specificity : 1.0000     </span>
<span class="co">##          Pos Pred Value : 1.0000     </span>
<span class="co">##          Neg Pred Value : 1.0000     </span>
<span class="co">##              Prevalence : 0.6817     </span>
<span class="co">##          Detection Rate : 0.6817     </span>
<span class="co">##    Detection Prevalence : 0.6817     </span>
<span class="co">##       Balanced Accuracy : 1.0000     </span>
<span class="co">##                                      </span>
<span class="co">##        'Positive' Class : TT         </span>
<span class="co">## </span></code></pre>
</div>
<div class="section level2">
<h2 id="representing-individual-samples-as-vectors-from-the-n-dimensional-space">Representing individual samples as vectors from the <em>N</em>-dimensional space<a class="anchor" aria-label="anchor" href="#representing-individual-samples-as-vectors-from-the-n-dimensional-space"></a>
</h2>
<p>The above cutpoint can be used to identify DMPs from control and treatment. The PCA+QDA model classifier can be used any time to discriminate control DMPs from those treatment. DMPs are retrieved using <a href="https://genomaths.github.io/MethylIT_HTML_Manual/selectDIMP.html" class="external-link">selectDIMP</a> function (<code>selectDIMP</code>):</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectDIMP.html">selectDIMP</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">divs</span>, div.col <span class="op">=</span> <span class="fl">9L</span>, cutpoint <span class="op">=</span> <span class="va">cutp</span><span class="op">$</span><span class="va">cutpoint</span>,
                   tv.cut <span class="op">=</span> <span class="fl">0.34</span>, tv.col <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">dmps</span>,<span class="va">length</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##   C1   C2   C3   C4   C5   T1   T2   T3   T4   T5   T6   T7   T8   T9  T10 </span>
<span class="co">##  450  428  410  522  579 2808 2651 2632 2735 2768 3170 3233 3210 3318 3239</span></code></pre>
<p>Next, to represent individual samples as vectors from the <em>N</em>-dimensional space, we can use <a href="https://genomaths.github.io/MethylIT_utils_HTML_Manual/getGRegionsStat-methods.html" class="external-link">getGRegionsStat</a> function from <a href="https://github.com/genomaths/MethylIT.utils" class="external-link">MethylIT.utils</a> R package. Here, the simulated “chromosome” is split into regions of 450bp non-overlapping windows. and the <em>density</em> of Hellinger divergences values is taken for each windows.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span>
<span class="va">DMRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/getGRegionsStat.html" class="external-link">getGRegionsStat</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, win.size <span class="op">=</span> <span class="fl">350</span>, step.size <span class="op">=</span> <span class="fl">350</span>,
                        stat <span class="op">=</span> <span class="st">"density"</span>, column <span class="op">=</span> <span class="fl">9L</span>, scaling <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">DMRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">ns</span>
<span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">DMRs</span>, <span class="st">"GRangesList"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="hierarchical-clustering">Hierarchical Clustering<a class="anchor" aria-label="anchor" href="#hierarchical-clustering"></a>
</h2>
<p>Hierarchical clustering (HC) is an unsupervised machine learning approach. HC can provide an initial estimation of the number of possible groups and information on their members. However, the effectivity of HC will depend on the experimental dataset, the metric used, and the agglomeration algorithm applied. For an unknown reason (and based on the personal experience of the author working in numerical taxonomy), Ward’s agglomeration algorithm performs much better on biological experimental datasets than the rest of the available algorithms like UPGMA, UPGMC, etc.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dmgm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">DMRs</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">dmgm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dmgm</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dmgm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">ns</span>
<span class="va">sampleNames</span> <span class="op">&lt;-</span> <span class="va">ns</span>

<span class="va">hc</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">dist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span> <span class="va">dmgm</span> <span class="op">)</span><span class="op">)</span><span class="op">^</span><span class="fl">2</span>, method <span class="op">=</span> <span class="st">'ward.D2'</span><span class="op">)</span>
<span class="va">hc.rsq</span> <span class="op">=</span> <span class="va">hc</span>
<span class="va">hc.rsq</span><span class="op">$</span><span class="va">height</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">sqrt</a></span><span class="op">(</span> <span class="va">hc</span><span class="op">$</span><span class="va">height</span> <span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="dendrogram">Dendrogram<a class="anchor" aria-label="anchor" href="#dendrogram"></a>
</h3>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">colors</span> <span class="op">=</span> <span class="va">sampleNames</span> 
<span class="va">colors</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"C"</span>, <span class="va">colors</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"green4"</span>
<span class="va">colors</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"T[6-9]{1}"</span>, <span class="va">colors</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"red"</span>
<span class="va">colors</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"T10"</span>, <span class="va">colors</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"red"</span>
<span class="va">colors</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"T[1-5]"</span>, <span class="va">colors</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"blue"</span>

<span class="co"># rgb(red, green, blue, alpha, names = NULL, maxColorValue = 1)</span>
<span class="va">clusters.color</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.7</span>, <span class="fl">0</span>, <span class="fl">0.1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">0.1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/grDevices/rgb.html" class="external-link">rgb</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0.2</span>, <span class="fl">0</span>, <span class="fl">0.1</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>font.lab<span class="op">=</span><span class="fl">2</span>,font<span class="op">=</span><span class="fl">3</span>,font.axis<span class="op">=</span><span class="fl">2</span>, mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">4</span>,<span class="fl">2</span>,<span class="fl">0</span><span class="op">)</span>, family<span class="op">=</span><span class="st">"serif"</span> , lwd <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/as.phylo.html" class="external-link">as.phylo</a></span><span class="op">(</span><span class="va">hc.rsq</span><span class="op">)</span>, tip.color <span class="op">=</span> <span class="va">colors</span>, label.offset <span class="op">=</span> <span class="fl">0.5</span>, font <span class="op">=</span> <span class="fl">2</span>, cex <span class="op">=</span> <span class="fl">0.9</span>,
     edge.width  <span class="op">=</span> <span class="fl">0.4</span>, direction <span class="op">=</span> <span class="st">"downwards"</span>, no.margin <span class="op">=</span> <span class="cn">FALSE</span>,
     align.tip.label <span class="op">=</span> <span class="cn">TRUE</span>, adj <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ape/man/axisPhylo.html" class="external-link">axisPhylo</a></span><span class="op">(</span> <span class="fl">2</span>, las <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">0.4</span>, cex.axis <span class="op">=</span> <span class="fl">1.4</span>, hadj <span class="op">=</span> <span class="fl">0.8</span>, tck <span class="op">=</span> <span class="op">-</span><span class="fl">0.01</span> <span class="op">)</span>
<span class="co">## cuts = c(xleft, ybottom, xright, ytop)</span>
<span class="fu"><a href="https://rdrr.io/pkg/MethylIT.utils/man/hclust_rect.html" class="external-link">hclust_rect</a></span><span class="op">(</span><span class="va">hc.rsq</span>, k <span class="op">=</span> <span class="fl">2L</span>, border <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"green4"</span>, <span class="st">"blue"</span>, <span class="st">"red"</span><span class="op">)</span>, 
            color <span class="op">=</span> <span class="va">clusters.color</span>, cuts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.56</span>, <span class="fl">1.4</span>, <span class="fl">0.41</span>, <span class="fl">120</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="pca_lda_with_methylit_files/figure-html/dendrogram-1.png" width="700"></p>
<p>Here, we have use function <a href="https://www.rdocumentation.org/packages/ape/versions/5.3/topics/as.phylo" class="external-link">as.phylo</a> from the R package <a href="https://www.rdocumentation.org/packages/ape/versions/5.3" class="external-link">ape</a> for better dendrogram visualization and function <a href="https://genomaths.github.io/MethylIT_utils_HTML_Manual/hclust_rect.html" class="external-link">hclust_rect</a> from <a href="https://github.com/genomaths/MethylIT.utils" class="external-link">MethylIT.utils</a> R package to draw rectangles with background colors around the branches of a dendrogram highlighting the corresponding clusters.</p>
</div>
</div>
<div class="section level2">
<h2 id="pca-lda">PCA + LDA<a class="anchor" aria-label="anchor" href="#pca-lda"></a>
</h2>
<p>MethylIT function <a href="https://genomaths.github.io/MethylIT_HTML_Manual/pcaLDA.html" class="external-link">pcaLDA</a> will be used to perform the PCA and PCA + LDA analyses. The function returns a list of two objects: 1) ‘lda’: an object of class ‘lda’ from package ‘MASS.’ 2) ‘pca’: an object of class ‘prcomp’ from package ‘stats.’ For information on how to use these objects see ?lda and ?prcomp.</p>
<p>Unlike hierarchical clustering (HC), LDA is a supervised machine learning approach. So, we must provide a prior classification of the individuals, which can be derived, for example, from the HC, or from a previous exploratory analysis with PCA.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># A prior classification derived from HC</span>
<span class="va">grps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cutree.html" class="external-link">cutree</a></span><span class="op">(</span><span class="va">hc</span>, k <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">grps</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">grps</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"CT"</span>
<span class="va">grps</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="fl">2</span>, <span class="va">grps</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"T1"</span>
<span class="va">grps</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="fl">3</span>, <span class="va">grps</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"T2"</span>
<span class="va">grps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">grps</span><span class="op">)</span>

<span class="va">ld</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pcaLDA.html">pcaLDA</a></span><span class="op">(</span>data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">dmgm</span><span class="op">)</span>, grouping <span class="op">=</span> <span class="va">grps</span>, n.pc <span class="op">=</span> <span class="fl">3</span>, max.pc <span class="op">=</span> <span class="fl">3</span>,
             scale <span class="op">=</span> <span class="cn">FALSE</span>, center <span class="op">=</span> <span class="cn">FALSE</span>, tol <span class="op">=</span> <span class="fl">1e-6</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">ld</span><span class="op">$</span><span class="va">pca</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Importance of first k=3 (out of 15) components:</span>
<span class="co">##                             PC1      PC2      PC3</span>
<span class="co">## Standard deviation     201.2414 35.47542 28.81634</span>
<span class="co">## Proportion of Variance   0.8922  0.02773  0.01829</span>
<span class="co">## Cumulative Proportion    0.8922  0.91997  0.93826</span></code></pre>
<p>We may retain enough components so that the cumulative percent of variance accounted for at least 70 to 80%. By setting <span class="math inline">\(scale=TRUE\)</span> and <span class="math inline">\(center=TRUE\)</span>, we could have different results and would improve or not our results. In particular, these settings are essentials if the <em>N</em>-dimensional space is integrated by variables from different measurement scales/units, for example, Kg and g, or Kg and Km.</p>
<div class="section level3">
<h3 id="pca">PCA<a class="anchor" aria-label="anchor" href="#pca"></a>
</h3>
<p>The individual coordinates in the principal components (PCs) are returned by function <a href="https://genomaths.github.io/MethylIT_HTML_Manual/pcaLDA.html" class="external-link">pcaLDA</a>. In the current case, based on the cumulative proportion of variance, the two firsts PCs carried about the 84% of the total sample variance and could split the sample into meaningful groups.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pca.coord</span> <span class="op">&lt;-</span> <span class="va">ld</span><span class="op">$</span><span class="va">pca</span><span class="op">$</span><span class="va">x</span>
<span class="va">pca.coord</span></code></pre></div>
<pre><code><span class="co">##            PC1         PC2         PC3</span>
<span class="co">## C1   -35.16687  -0.9086457   3.9933638</span>
<span class="co">## C2   -32.87833  -3.0040074   1.7593098</span>
<span class="co">## C3   -27.00560  -0.7947740   0.7783422</span>
<span class="co">## C4   -38.87409   5.0258542   3.3668192</span>
<span class="co">## C5   -39.21075  -1.3709975   6.3405404</span>
<span class="co">## T1  -264.40755 -20.3387215 -36.4367326</span>
<span class="co">## T2  -222.53557  41.9563285  16.4021733</span>
<span class="co">## T3  -198.92107  -3.4770492  -2.7407996</span>
<span class="co">## T4  -214.80187 -41.8357746  -3.7670777</span>
<span class="co">## T5  -201.23338 -19.6460781   5.3147099</span>
<span class="co">## T6  -267.34036 -45.6038600  26.2379616</span>
<span class="co">## T7  -259.72994  16.7580954  14.0003693</span>
<span class="co">## T8  -241.82987  -2.7879098 -79.1083267</span>
<span class="co">## T9  -245.26878 100.7672386   6.6163448</span>
<span class="co">## T10 -240.14834 -27.1443632  52.1650787</span></code></pre>
</div>
<div class="section level3">
<h3 id="graphic-pc1-vs-pc2">Graphic PC1 vs PC2<a class="anchor" aria-label="anchor" href="#graphic-pc1-vs-pc2"></a>
</h3>
<p>Next, the graphic for individual coordinates in the two firsts PCs can be easily visualized now:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">pca.coord</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>, subgrp <span class="op">=</span> <span class="va">grps</span><span class="op">)</span>

<span class="va">p0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>
  axis.text.x  <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span> face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">18</span>, color<span class="op">=</span><span class="st">"black"</span>,
                               <span class="co"># hjust = 0.5, vjust = 0.5, </span>
                               family <span class="op">=</span> <span class="st">"serif"</span>, angle <span class="op">=</span> <span class="fl">0</span>,
                               margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">margin</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0</span>, unit <span class="op">=</span> <span class="st">"pt"</span> <span class="op">)</span><span class="op">)</span>,
  axis.text.y  <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span> face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">18</span>, color<span class="op">=</span><span class="st">"black"</span>, 
                               family <span class="op">=</span> <span class="st">"serif"</span>,
                               margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">margin</a></span><span class="op">(</span> <span class="fl">0</span>,<span class="fl">0.1</span>,<span class="fl">0</span>,<span class="fl">0</span>, unit <span class="op">=</span> <span class="st">"mm"</span> <span class="op">)</span><span class="op">)</span>,
  axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, family <span class="op">=</span> <span class="st">"serif"</span>, size <span class="op">=</span> <span class="fl">18</span>,
                              color<span class="op">=</span><span class="st">"black"</span>, vjust <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span>,
  axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, family <span class="op">=</span> <span class="st">"serif"</span>, size <span class="op">=</span> <span class="fl">18</span>,
                              color<span class="op">=</span><span class="st">"black"</span>, 
                              margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">margin</a></span><span class="op">(</span> <span class="fl">0</span>,<span class="fl">2</span>,<span class="fl">0</span>,<span class="fl">0</span>, unit <span class="op">=</span> <span class="st">"mm"</span> <span class="op">)</span> <span class="op">)</span>,
  legend.title<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
  legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span>, family <span class="op">=</span> <span class="st">"serif"</span><span class="op">)</span>,
  legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.88</span>, <span class="fl">0.14</span><span class="op">)</span>,
  
  panel.border <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_rect</a></span><span class="op">(</span>fill<span class="op">=</span><span class="cn">NA</span>, colour <span class="op">=</span> <span class="st">"black"</span>,size<span class="op">=</span><span class="fl">0.07</span><span class="op">)</span>,
  panel.grid.minor <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_line</a></span><span class="op">(</span>color<span class="op">=</span> <span class="st">"white"</span>,size <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span>,
  axis.ticks <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_line</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>, axis.ticks.length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="st">"mm"</span><span class="op">)</span>,
  plot.margin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, <span class="st">"lines"</span><span class="op">)</span><span class="op">)</span> 

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">dt</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">PC1</span>, y <span class="op">=</span> <span class="va">PC2</span>, colour <span class="op">=</span> <span class="va">grps</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"white"</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"white"</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"green4"</span>,<span class="st">"blue"</span>,<span class="st">"brown1"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_ellipse.html" class="external-link">stat_ellipse</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">PC1</span>, y <span class="op">=</span> <span class="va">PC2</span>, fill <span class="op">=</span> <span class="va">subgrp</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">dt</span>, type <span class="op">=</span> <span class="st">"norm"</span>,
               geom <span class="op">=</span> <span class="st">"polygon"</span>, level <span class="op">=</span> <span class="fl">0.5</span>, alpha<span class="op">=</span><span class="fl">0.2</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"green4"</span>,<span class="st">"blue"</span>,<span class="st">"brown1"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="va">p0</span></code></pre></div>
<p><img src="pca_lda_with_methylit_files/figure-html/pcaGraphic-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="graphic-ld1-vs-ld2">Graphic LD1 vs LD2<a class="anchor" aria-label="anchor" href="#graphic-ld1-vs-ld2"></a>
</h3>
<p>In the current case, better resolution is obtained with the linear discriminant functions, which is based on the three firsts PCs. Notice that the number principal components used the LDA step must be lower than the number of individuals (<span class="math inline">\(N\)</span>) divided by 3: <span class="math inline">\(N/3\)</span>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ind.coord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">ld</span>, newdata <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">dmgm</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"scores"</span><span class="op">)</span>
<span class="va">dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">ind.coord</span>, subgrp <span class="op">=</span> <span class="va">grps</span><span class="op">)</span>

<span class="va">p0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>
  axis.text.x  <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span> face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">18</span>, color<span class="op">=</span><span class="st">"black"</span>,
                               <span class="co"># hjust = 0.5, vjust = 0.5, </span>
                               family <span class="op">=</span> <span class="st">"serif"</span>, angle <span class="op">=</span> <span class="fl">0</span>,
                               margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">margin</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0</span>, unit <span class="op">=</span> <span class="st">"pt"</span> <span class="op">)</span><span class="op">)</span>,
  axis.text.y  <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span> face <span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">18</span>, color<span class="op">=</span><span class="st">"black"</span>, 
                               family <span class="op">=</span> <span class="st">"serif"</span>,
                               margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">margin</a></span><span class="op">(</span> <span class="fl">0</span>,<span class="fl">0.1</span>,<span class="fl">0</span>,<span class="fl">0</span>, unit <span class="op">=</span> <span class="st">"mm"</span> <span class="op">)</span><span class="op">)</span>,
  axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, family <span class="op">=</span> <span class="st">"serif"</span>, size <span class="op">=</span> <span class="fl">18</span>,
                              color<span class="op">=</span><span class="st">"black"</span>, vjust <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span>,
  axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"bold"</span>, family <span class="op">=</span> <span class="st">"serif"</span>, size <span class="op">=</span> <span class="fl">18</span>,
                              color<span class="op">=</span><span class="st">"black"</span>, 
                              margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">margin</a></span><span class="op">(</span> <span class="fl">0</span>,<span class="fl">2</span>,<span class="fl">0</span>,<span class="fl">0</span>, unit <span class="op">=</span> <span class="st">"mm"</span> <span class="op">)</span> <span class="op">)</span>,
  legend.title<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
  legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">20</span>, face <span class="op">=</span> <span class="st">"bold"</span>, family <span class="op">=</span> <span class="st">"serif"</span><span class="op">)</span>,
  legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.15</span>, <span class="fl">0.83</span><span class="op">)</span>,
  
  panel.border <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_rect</a></span><span class="op">(</span>fill<span class="op">=</span><span class="cn">NA</span>, colour <span class="op">=</span> <span class="st">"black"</span>,size<span class="op">=</span><span class="fl">0.07</span><span class="op">)</span>,
  panel.grid.minor <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_line</a></span><span class="op">(</span>color<span class="op">=</span> <span class="st">"white"</span>,size <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span>,
  axis.ticks <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_line</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>, axis.ticks.length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="st">"mm"</span><span class="op">)</span>,
  plot.margin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, <span class="st">"lines"</span><span class="op">)</span><span class="op">)</span> 

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">dt</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">LD1</span>, y <span class="op">=</span> <span class="va">LD2</span>, colour <span class="op">=</span> <span class="va">grps</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"white"</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"white"</span>, size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"green4"</span>,<span class="st">"blue"</span>,<span class="st">"brown1"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/stat_ellipse.html" class="external-link">stat_ellipse</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">LD1</span>, y <span class="op">=</span> <span class="va">LD2</span>, fill <span class="op">=</span> <span class="va">subgrp</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">dt</span>, type <span class="op">=</span> <span class="st">"norm"</span>,
               geom <span class="op">=</span> <span class="st">"polygon"</span>, level <span class="op">=</span> <span class="fl">0.5</span>, alpha<span class="op">=</span><span class="fl">0.2</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"green4"</span>,<span class="st">"blue"</span>,<span class="st">"brown1"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="va">p0</span></code></pre></div>
<pre><code><span class="co">## Too few points to calculate an ellipse</span></code></pre>
<p><img src="pca_lda_with_methylit_files/figure-html/ldaGraphic-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body">
<div id="ref-Liese2006" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">F. Liese, I. Vajda, <span class="nocase">On divergences and informations in statistics and information theory</span>. <em>IEEE Transactions on Information Theory</em> <strong>52</strong>, 4394–4412 (2006).</div>
</div>
<div id="ref-Stevens2009" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">J. P. Stevens, <em><span class="nocase">Applied Multivariate Statistics for the Social Sciences</span></em>, Fifth Edit (Routledge Academic, 2009).</div>
</div>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code> on the system on which this document was compiled running pandoc 2.14.0.3:</p>
<pre><code><span class="co">## R version 4.1.1 (2021-08-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: CentOS Linux 7 (Core)</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.3.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] ape_5.6              ggplot2_3.3.5        MethylIT.utils_0.1  </span>
<span class="co">##  [4] MethylIT_0.3.2.3     rmarkdown_2.11       rtracklayer_1.52.1  </span>
<span class="co">##  [7] GenomicRanges_1.44.0 GenomeInfoDb_1.30.0  IRanges_2.28.0      </span>
<span class="co">## [10] S4Vectors_0.32.3     BiocGenerics_0.40.0 </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] colorspace_2.0-2            rjson_0.2.21               </span>
<span class="co">##   [3] modeltools_0.2-23           ellipsis_0.3.2             </span>
<span class="co">##   [5] class_7.3-20                rprojroot_2.0.2            </span>
<span class="co">##   [7] XVector_0.34.0              fs_1.5.0                   </span>
<span class="co">##   [9] proxy_0.4-26                farver_2.1.0               </span>
<span class="co">##  [11] flexmix_2.3-17              bit64_4.0.5                </span>
<span class="co">##  [13] AnnotationDbi_1.56.2        prodlim_2019.11.13         </span>
<span class="co">##  [15] fansi_0.5.0                 lubridate_1.7.10           </span>
<span class="co">##  [17] codetools_0.2-18            splines_4.1.1              </span>
<span class="co">##  [19] cachem_1.0.5                knitr_1.33                 </span>
<span class="co">##  [21] Formula_1.2-4               jsonlite_1.7.2             </span>
<span class="co">##  [23] pROC_1.17.0.1               Rsamtools_2.8.0            </span>
<span class="co">##  [25] caret_6.0-90                annotate_1.72.0            </span>
<span class="co">##  [27] png_0.1-7                   compiler_4.1.1             </span>
<span class="co">##  [29] httr_1.4.2                  assertthat_0.2.1           </span>
<span class="co">##  [31] Matrix_1.4-1                fastmap_1.1.0              </span>
<span class="co">##  [33] htmltools_0.5.2             tools_4.1.1                </span>
<span class="co">##  [35] gtable_0.3.0                glue_1.4.2                 </span>
<span class="co">##  [37] GenomeInfoDbData_1.2.7      reshape2_1.4.4             </span>
<span class="co">##  [39] dplyr_1.0.7                 Rcpp_1.0.7                 </span>
<span class="co">##  [41] Biobase_2.54.0              jquerylib_0.1.4            </span>
<span class="co">##  [43] pkgdown_2.0.2               vctrs_0.3.8                </span>
<span class="co">##  [45] Biostrings_2.62.0           nlme_3.1-155               </span>
<span class="co">##  [47] betareg_3.1-4               iterators_1.0.13           </span>
<span class="co">##  [49] lmtest_0.9-38               timeDate_3043.102          </span>
<span class="co">##  [51] gower_0.2.2                 xfun_0.24                  </span>
<span class="co">##  [53] stringr_1.4.0               lifecycle_1.0.0            </span>
<span class="co">##  [55] restfulr_0.0.13             XML_3.99-0.6               </span>
<span class="co">##  [57] zoo_1.8-9                   zlibbioc_1.40.0            </span>
<span class="co">##  [59] MASS_7.3-56                 scales_1.1.1               </span>
<span class="co">##  [61] ipred_0.9-11                ragg_1.1.3                 </span>
<span class="co">##  [63] MatrixGenerics_1.4.3        sandwich_3.0-1             </span>
<span class="co">##  [65] parallel_4.1.1              SummarizedExperiment_1.22.0</span>
<span class="co">##  [67] yaml_2.2.1                  memoise_2.0.0              </span>
<span class="co">##  [69] sass_0.4.0                  rpart_4.1.16               </span>
<span class="co">##  [71] stringi_1.6.2               RSQLite_2.2.11             </span>
<span class="co">##  [73] highr_0.9                   genefilter_1.76.0          </span>
<span class="co">##  [75] BiocIO_1.2.0                desc_1.3.0                 </span>
<span class="co">##  [77] foreach_1.5.1               e1071_1.7-7                </span>
<span class="co">##  [79] BiocParallel_1.28.3         lava_1.6.9                 </span>
<span class="co">##  [81] rlang_0.4.11                pkgconfig_2.0.3            </span>
<span class="co">##  [83] systemfonts_1.0.2           matrixStats_0.59.0         </span>
<span class="co">##  [85] bitops_1.0-7                evaluate_0.14              </span>
<span class="co">##  [87] lattice_0.20-45             purrr_0.3.4                </span>
<span class="co">##  [89] labeling_0.4.2              GenomicAlignments_1.28.0   </span>
<span class="co">##  [91] recipes_0.1.16              bit_4.0.4                  </span>
<span class="co">##  [93] tidyselect_1.1.1            plyr_1.8.6                 </span>
<span class="co">##  [95] magrittr_2.0.1              nls2_0.2                   </span>
<span class="co">##  [97] R6_2.5.0                    generics_0.1.0             </span>
<span class="co">##  [99] DelayedArray_0.18.0         DBI_1.1.2                  </span>
<span class="co">## [101] pillar_1.6.1                withr_2.4.2                </span>
<span class="co">## [103] survival_3.3-1              KEGGREST_1.34.0            </span>
<span class="co">## [105] RCurl_1.98-1.5              nnet_7.3-17                </span>
<span class="co">## [107] tibble_3.1.2                crayon_1.4.1               </span>
<span class="co">## [109] utf8_1.2.2                  grid_4.1.1                 </span>
<span class="co">## [111] minpack.lm_1.2-1            data.table_1.14.2          </span>
<span class="co">## [113] blob_1.2.2                  ModelMetrics_1.2.2.2       </span>
<span class="co">## [115] digest_0.6.27               xtable_1.8-4               </span>
<span class="co">## [117] textshaping_0.3.5           munsell_0.5.0              </span>
<span class="co">## [119] bslib_0.3.1</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
