<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Some tips for Methyl-IT R scripts • MethylIT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Some tips for Methyl-IT R scripts">
<meta property="og:description" content="MethylIT">
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.6.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Some tips for Methyl-IT R scripts</h1>
                        <h4 data-toc-skip class="author"><div class="line-block">Robersy Sanchez<br><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a><br>
Mackenzie’s lab<br><br>
Department of Biology and Plant Science.<br>
Pennsylvania State University, University Park, PA 16802<br>
Maintainer: Robersy Sanchez</div></h4>
            
            <h4 data-toc-skip class="date">26 April 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/vignettes/some_tips_for_methylit_r_scripts.Rmd" class="external-link"><code>vignettes/some_tips_for_methylit_r_scripts.Rmd</code></a></small>
      <div class="hidden name"><code>some_tips_for_methylit_r_scripts.Rmd</code></div>

    </div>

    
    
<style type="text/css">

.list-group-item.active, .list-group-item.active:focus, 
.list-group-item.active:hover {
    z-index: 2;
    color: #fff;
    background-color: #337ab7;
    border-color: #337ab7;
}

.tocify-subheader .tocify-item {
    /* font-size: 0.80em; */ 
    padding-left: 25px;
    text-indent: 0;
}

.tocify-header .tocify-item {
    /* font-size: 0.85em; */ 
    padding-left: 20px;
    text-indent: 0;
}

body{ /* Normal  */
    /* font-size: 18px; */
    font-family: "Times New Roman", Times, serif;
    text-align: justify
}
td {  /* Table  */
    /* font-size: 8px; */
}

h1.title {
    /* font-size: 38px; */
    font-family: "Times New Roman", Times, serif;
    color: DarkRed;
    .subTitle {
        /* font-size: 24px; */
        font-family: "Times New Roman", Times, serif;
        color: DarkRed;
    }
}

h1 { /* Header 1 */
    /* font-size: 28px; */
    font-family: "Times New Roman", Times, serif;
    color: DarkBlue;
}
h2 { /* Header 2 */
    /* font-size: 22px; */
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
h3 { /* Header 3 */
    /* font-size: 18px; */
    color: DarkBlue;
    font-family: "Times New Roman", Times, serif;
}
code.r{ /* Code block */
    font-size: 14px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>
<div class="section level2">
<h2 id="reading-data-sets">Reading data sets<a class="anchor" aria-label="anchor" href="#reading-data-sets"></a>
</h2>
<p>Let’s suppose that we have the following read-count data set obtained
from Bismark ‘CX_report’ files:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">folder</span> <span class="op">&lt;-</span> <span class="st">"~/data/"</span></span>
<span><span class="va">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span>path <span class="op">=</span> <span class="va">folder</span><span class="op">)</span></span>
<span><span class="co"># [1] "~/data/Col_8-1a.fq_bismark_bt2.deduplicated.bismark.CX_report"</span></span>
<span><span class="co"># [2] "~/data/Col_8-1b.fq_bismark_bt2.deduplicated.bismark.CX_report"</span></span>
<span><span class="co"># [3] "~/data/Col_8-1c.fq_bismark_bt2.deduplicated.bismark.CX_report"</span></span></code></pre></div>
<p>This data set can be read with the following code:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/MethylIT" class="external-link">MethylIT</a></span><span class="op">)</span></span>
<span><span class="va">col0</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readCounts2GRangesList.html">readCounts2GRangesList</a></span><span class="op">(</span>filenames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">folder</span>, <span class="va">files</span><span class="op">)</span>,</span>
<span>                               sample.id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"col1"</span>, <span class="st">"col2"</span>, <span class="st">"col3"</span><span class="op">)</span>,</span>
<span>                               pattern <span class="op">=</span> <span class="st">"^[1-5]"</span>,</span>
<span>                               columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fl">1</span>, start <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                           strand <span class="op">=</span> <span class="fl">3</span>, mC <span class="op">=</span> <span class="fl">4</span>, uC <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                                           context <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Since the methylome data are from Arabidopsis thaliana (a plant), we
have added the column 6 from each Bismark ‘CX_report’ file, which for
each cytosine site carries the information of its corresponding
methylation context: CG, CHG, or CHH. Users familiar with Linux OS can
easily check which is the number of each required column by using Linux
command line ‘head’ function.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> BS-10.CX_report.txt</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>       1       +       0       0       CHH     CCC</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>       2       +       0       0       CHH     CCT</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>       3       +       0       0       CHH     CTA</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>       8       +       0       0       CHH     CCC</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>       9       +       0       0       CHH     CCT</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>       10      +       0       0       CHH     CTA</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>       15      +       0       0       CHH     CCC</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>       16      +       0       0       CHH     CCT</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>       17      +       0       0       CHH     CTA</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="ex">1</span>       22      +       0       0       CHH     CCC</span></code></pre></div>
<p>Plants methylome contains chloroplast and mitochondrial chromosomes,
which are not included in methylome analysis. In the current case, we
can exclude these chromosomes by setting:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pattern</span> <span class="op">=</span> <span class="st">"^[1-5]"</span></span></code></pre></div>
<p>That is, only rows starting with any of the integer numbers from 1 to
5 are read. Notice that <a href="https://is.gd/5pyk1N" class="external-link">readCounts2GRangesList</a> function is able
to deal with different file formats. The user just need to provide the
number of the specified columns, which for different aligner programs
can be different.</p>
<p>It is very important to notice that Methyl-IT pipeline assumes that
all the data provided have acceptable quality. It is up to the user to
guaranty the quality of its methylome data set. For example, to include
chloroplast chromosome, will artificially decrease the levels of noise
in the treatment and control group of samples, which definitely will
lead to a wrong estimation of the cutpoints in the signal detection
step.</p>
<div class="section level3">
<h3 id="analysis-per-methylation-context">Analysis per methylation context<a class="anchor" aria-label="anchor" href="#analysis-per-methylation-context"></a>
</h3>
<p>Usually, in plant epigenomic field, the methylation analysis is
performed by methylation context. After apply the script above, the
three methylation contexts were included in each sample. The <a href="https://is.gd/5pyk1N" class="external-link">readCounts2GRangesList</a> function can be
used to read only one specified methylation context, but it also
increase the number of files per sample.</p>
<p>The next code will select the methylation context that want, say
“CHG”, from the above samples:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">col0_chg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/lapply.html">lapply</a></span><span class="op">(</span><span class="va">col0</span>, <span class="kw">function</span><span class="op">(</span><span class="va">s</span><span class="op">)</span> <span class="va">s</span><span class="op">[</span><span class="va">s</span><span class="op">$</span><span class="va">context</span> <span class="op">==</span> <span class="st">"CHG"</span><span class="op">]</span>, keep.attr <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Next, to compute the reference sample, keep in mind that we have two
columns carrying integer numbers and third column carrying a character
vector, which will be not included in the computation. To prevent error
we use ‘columns = 1:2’</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ref_chg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/poolFromGRlist.html">poolFromGRlist</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">col0_chg</span>, stat <span class="op">=</span> <span class="st">"sum"</span>, columns <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,</span>
<span>                          num.cores <span class="op">=</span> <span class="fl">10L</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="dmps">DMPs<a class="anchor" aria-label="anchor" href="#dmps"></a>
</h2>
<p>Let’s suppose that we have set of DMPs estimated for each methylation
context: dmps_cg, dmp_chg, and dmps_chh. For DMG or DMR analyses we
usually include all the methylation contexts.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dmps_cg</span><span class="op">)</span></span>
<span><span class="va">dmps</span> <span class="op">&lt;-</span> <span class="va">dmps_cg</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">k</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">nams</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">dmps</span><span class="op">[[</span><span class="va">k</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">dmps_cg</span><span class="op">[[</span><span class="va">k</span><span class="op">]</span><span class="op">]</span>, <span class="va">dmps_chg</span><span class="op">[[</span><span class="va">k</span><span class="op">]</span><span class="op">]</span>, <span class="va">dmps_chh</span><span class="op">[[</span><span class="va">k</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">nams</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="annotation-file">Annotation file<a class="anchor" aria-label="anchor" href="#annotation-file"></a>
</h2>
<p>Gene annotation is needed for DMG estimation. Some organisms (not
well annotated yet) reports several “contig” sequences which are not
included between the known chromosomes. In the most frequent case, as in
the case of Arabidopsis, we do not include chloroplast and mitochondrial
chromosomes. For Arabidopsis, following code does job when the “gtf” is
used:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span></span>
<span><span class="va">AG</span> <span class="op">=</span> <span class="fu">import</span><span class="op">(</span><span class="st">"~/data/TAIR10_gff3/Arabidopsis_thaliana.TAIR10.46.gtf.gz"</span><span class="op">)</span></span>
<span><span class="va">gene</span> <span class="op">=</span> <span class="va">AG</span><span class="op">[</span> <span class="va">AG</span><span class="op">$</span><span class="va">type</span> <span class="op">==</span> <span class="st">"gene"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span> <span class="st">"gene_id"</span>, <span class="st">"gene_biotype"</span>, <span class="st">"gene_name"</span> <span class="op">)</span> <span class="op">]</span></span>
<span><span class="va">gene</span> <span class="op">=</span> <span class="va">gene</span><span class="op">[</span> <span class="va">gene</span><span class="op">$</span><span class="va">gene_biotype</span> <span class="op">==</span> <span class="st">"protein_coding"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene_id"</span>, <span class="st">"gene_name"</span><span class="op">)</span> <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">gene</span>, pruning.mode <span class="op">=</span> <span class="st">"coarse"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span><span class="op">)</span></span>
<span><span class="va">gene</span> <span class="op">=</span> <span class="fu"><a href="../reference/sortBySeqnameAndStart.html">sortBySeqnameAndStart</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span></span></code></pre></div>
<p>In fact, the critical step in the above code is:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">gene</span>, pruning.mode <span class="op">=</span> <span class="st">"coarse"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span>, <span class="st">"4"</span>, <span class="st">"5"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="general-suggestions">General suggestions<a class="anchor" aria-label="anchor" href="#general-suggestions"></a>
</h2>
<ol style="list-style-type: decimal">
<li><p>Theoretically, the nonlinear fit to estimate the probability
distribution of the information divergence and the cutpoint estimation
must be done per chromosome. However, for organisms with small
methylome, as in the case of Arbidopsis, we have not detected
significant differences, so far, when the methylome reprogramming is
genome-wide. Nevertheless, if you suspect that the methylation
reprogramming is affecting specific chromosomes, then we strongly
suggest to perform the analysis per chromosome. In the case of human
methylome, the analysis per chromosome is the right way to
proceed.</p></li>
<li><p>The cutpoint estimation step is critical. The evaluation of the
classification performance is a very important step. Typically, a
classification performance with accuracy below 95% would indicate that
either something is wrong in some previous step (not necessarily in
Methyl-IT steps) or that really there are not too much difference
between your control and treatment groups, as would be the case, for
example, in the analysis of different seed development stages. Please
read and be familiar with the vignette:<br><a href="https://is.gd/WoYFUe" class="external-link">Optimal cutpoint for the methylation
signal</a>.</p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
