<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Beta binomial posteriors — .betaBinPosteriors • MethylIT</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Beta binomial posteriors — .betaBinPosteriors" />
<meta property="og:description" content="Here, beta binomial posteriors are estimated in the context of
the methylation level estimation. The direct use of counts for methylation
level estimation is not recommended, since there is external source of noise
coming from the sample manipulation, the sequencing machine and sequence
alignment that could alter the counts and the number of methylated cytosines
with low or zero counts (mC = 0) is generally high. The low counts and noise
imply that read counts of methylated (mC) and non-methylated (uC) cytosines
must not be used directly in the estimation of methylation levels.
In a Bayesian framework, methylated read counts are modeled by a beta-
binomial distribution, which accounts for both, the biological and sampling
variations (1-3). In our case we adopted the Bayesian approach suggested in
reference (3)(Chapter 3). Naive distribution q (methylation levels). In a
Bayesian framework with uniform priors, the methylation level can be defined
as: meth_level = ( mC + 1 )/( mC + uC + 2 ). However, the most natural
statistical model for replicated BS-seq DNA methylation measurements is
beta-binomial (the beta distribution is a prior conjugate of binomial
distribution), we consider the p parameter (methylation level) in the binomial
distribution as randomly drawn from a beta distribution. The hyper-parameters
alpha ('a') and beta ('b') from the beta-binomial distribution are interpreted
as pseudo-counts." />
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/MethylIT/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Beta binomial posteriors</h1>
    <small class="dont-index">Source: <a href='https://github.com/genomaths/MethylIT/blob/master/R/betaBinPosteriors.R'><code>R/betaBinPosteriors.R</code></a></small>
    <div class="hidden name"><code>betaBinPosteriors.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Here, beta binomial posteriors are estimated in the context of
the methylation level estimation. The direct use of counts for methylation
level estimation is not recommended, since there is external source of noise
coming from the sample manipulation, the sequencing machine and sequence
alignment that could alter the counts and the number of methylated cytosines
with low or zero counts (mC = 0) is generally high. The low counts and noise
imply that read counts of methylated (mC) and non-methylated (uC) cytosines
must not be used directly in the estimation of methylation levels.</p>
<p>In a Bayesian framework, methylated read counts are modeled by a beta-
binomial distribution, which accounts for both, the biological and sampling
variations (1-3). In our case we adopted the Bayesian approach suggested in
reference (3)(Chapter 3). Naive distribution q (methylation levels). In a
Bayesian framework with uniform priors, the methylation level can be defined
as: meth_level = ( mC + 1 )/( mC + uC + 2 ). However, the most natural
statistical model for replicated BS-seq DNA methylation measurements is
beta-binomial (the beta distribution is a prior conjugate of binomial
distribution), we consider the p parameter (methylation level) in the binomial
distribution as randomly drawn from a beta distribution. The hyper-parameters
alpha ('a') and beta ('b') from the beta-binomial distribution are interpreted
as pseudo-counts.</p>
    </div>

    <pre class="usage"><span class='fu'>.betaBinPosteriors</span><span class='op'>(</span><span class='va'>success</span>, <span class='va'>trials</span>, <span class='va'>a</span>, <span class='va'>b</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>success</th>
      <td><p>number of successful events</p></td>
    </tr>
    <tr>
      <th>trials</th>
      <td><p>total number of events</p></td>
    </tr>
    <tr>
      <th>a</th>
      <td><p>previous number of successful events</p></td>
    </tr>
    <tr>
      <th>b</th>
      <td><p>previous number of unsuccessful events</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a probability</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The posterior methylation levels are estimated as
(a + success)/(a + b + trials), where 'a' and 'b' are the shape parameters
of the beta distribution, alpha and beta parameters, respectively.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    
<ol>
<li><p>Hebestreit K, Dugas M, Klein H-U (2013) Detection of
significantly differentially methylated regions in targeted bisulfite
sequencing data. Bioinformatics 29: 1647-1653. Available:
http://www.ncbi.nlm.nih.gov/pubmed/23658421.
2. Robinson MD, Kahraman A, Law CW, Lindsay H, Nowicka M, et al. (2014)
Statistical methods for detecting differentially methylated loci and
regions. Front Genet 5: 324. Available:
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4165320/.
Baldi P, Brunak S (2001) Bioinformatics: the machine learning
approach. Second. Cambridge: MIT Press. 452 p.</p></li>
</ol>


    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>MethylIT</span><span class='fu'>:::</span><span class='fu'>.betaBinPosteriors</span><span class='op'>(</span><span class='fl'>2</span>, <span class='fl'>8</span>, <span class='fl'>2</span>, <span class='fl'>8</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] 0.2222222</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


