<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Regression Test for Count — countTest2 • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Regression Test for Count — countTest2"><meta property="og:description" content="Perform Poisson and Negative Binomial regression analysis to
compare the counts from different groups, treatment and control. The
difference between functions 'countTest2' and 'countTest' resides in the
estimation of the prior weights used in Negative Binomial generalized
linear model."><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Regression Test for Count</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/countTest2.R" class="external-link"><code>R/countTest2.R</code></a></small>
    <div class="hidden name"><code>countTest2.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Perform Poisson and Negative Binomial regression analysis to
compare the counts from different groups, treatment and control. The
difference between functions 'countTest2' and 'countTest' resides in the
estimation of the prior weights used in Negative Binomial generalized
linear model.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">countTest2</span><span class="op">(</span></span>
<span>  <span class="va">DS</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  countFilter <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  CountPerBp <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  minCountPerIndv <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  maxGrpCV <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  FilterLog2FC <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>  pvalCutOff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  MVrate <span class="op">=</span> <span class="fl">0.98</span>,</span>
<span>  Minlog2FC <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  test <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Wald"</span>, <span class="st">"LRT"</span><span class="op">)</span>,</span>
<span>  scaling <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  saveAll <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>DS</dt>
<dd><p>A 'glmDataSet' object, which is created with function
<code><a href="glmDataSet.html">glmDataSet</a></code>.</p></dd>


<dt>num.cores, tasks</dt>
<dd><p>Parameters for parallel computation using package
<code><a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallel-package.html" class="external-link">BiocParallel-package</a></code>: the number of cores to
use, i.e. at most how many child processes will be run simultaneously
(see <code><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></code> and the number of tasks per job
(only for Linux OS).</p></dd>


<dt>countFilter</dt>
<dd><p>whether or not to filter the counts according to the
minimum count per region per each individual/sample, which is setting by
'minCountPerIndv'.</p></dd>


<dt>CountPerBp</dt>
<dd><p>for each group the count per bp must be equal or greater
than CountPerBp. The filter is applied if 'CountPerBp' is given and if
'x' DESeqDataSet object has the rowRanges as a GRanges object on it</p></dd>


<dt>minCountPerIndv</dt>
<dd><p>each gene or region must have more than
'minCountPerIndv' counts (on average) per individual in at least one
group.</p></dd>


<dt>maxGrpCV</dt>
<dd><p>A numerical vector. Maximum coefficient of variance for each
group. Default maxGrpCV = NULL. The numbers 'maxGrpCV1' and
'maxGrpCV2' will be taken as the maximun variances values permitted in
control and in treatment groups, respectively. If only 'maxGrpCV1' is
provided, then maxGrpCV = c('maxGrpCV1', 'maxGrpCV1'). This parameter
is addressed to prevent testing regions where intra-group variations are
very large, e.g.: control = c(1,0,1,1) and treatment = c(1, 0, 1, 40).
The coefficient of variance for the treatment group is 1.87, very high.
The generalized linear regression analysis would yield statistical
significant group differences, but evidently there is something wrong in
one of the treatment samples. We would try the application of further
statistical smoothing approach, but we prefer to leave the user decide
which regions to test.</p></dd>


<dt>FilterLog2FC</dt>
<dd><p>if TRUE, the results are filtered using the minimum
absolute value of log2FoldChanges observed to accept that a gene in the
treatment is differentially expressed in respect to the control</p></dd>


<dt>pAdjustMethod</dt>
<dd><p>method used to adjust the results; default: BH</p></dd>


<dt>pvalCutOff</dt>
<dd><p>cutoff used then a p-value adjustment is performed.</p></dd>


<dt>MVrate</dt>
<dd><p>Minimum Mean/Variance rate.</p></dd>


<dt>Minlog2FC</dt>
<dd><p>minimum logarithm base 2 of fold changes.</p></dd>


<dt>test</dt>
<dd><p>A character string matching one of 'Wald' or 'LRT'. If test =
'Wald', then the p-value of the Wald test for the coefficient of the
independent variable (<em>treatment group</em>) will be reported.
If test = 'LRT', then the p-value from a likelihood ratio test given by
<code><a href="https://rdrr.io/r/stats/anova.html" class="external-link">anova</a></code> function from <em>stats</em> packages will be
the reported p-value for the group comparison when the best fitted model
is the negative binomial. As suggested for <code><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm</a></code>, if
best fitted model is Poisson or quasi-Poisson, then the best test is
'Chi-squared' or 'F-test', respectively. So, for the sake of simplicity,
the corresponding suitable test will be applied when test = 'LRT'.</p></dd>


<dt>scaling</dt>
<dd><p>integer (default 1). Scaling factor estimate the
signal density as: scaling x 'DMP-Count-Per-Bp'. For example,
if scaling = 1000, then signal density denotes the number of DMPs in
1000 bp.</p></dd>


<dt>saveAll</dt>
<dd><p>if TRUE all the temporal results are returned</p></dd>


<dt>verbose</dt>
<dd><p>if TRUE, prints the function log to stdout</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a data frame or GRanges object (if the DS contain the GRanges
information for each gene) with the test results and original count
matrix, plus control and treatment signal densities and their variation.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>A pairwise group comparison, control versus treatment, is performed.
The experimental design settings must be introduced using function
<code><a href="glmDataSet.html">glmDataSet</a></code> to provide dataset (DS) object.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="glmDataSet.html">glmDataSet</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">133</span><span class="op">)</span> <span class="co"># Set a seed</span></span></span>
<span class="r-in"><span><span class="co">## A GRanges object with the count matrix in the metacolumns is created</span></span></span>
<span class="r-in"><span><span class="va">countData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample.int</a></span><span class="op">(</span><span class="fl">200</span>, <span class="fl">500</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">countData</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'A1'</span>,<span class="st">'A2'</span>,<span class="st">'B1'</span>,<span class="st">'B2'</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">start</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">25e4</span>, <span class="fl">2000</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">end</span> <span class="op">&lt;-</span> <span class="va">start</span> <span class="op">+</span> <span class="fl">1000</span></span></span>
<span class="r-in"><span><span class="va">chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">'chr1'</span>, <span class="fl">70</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">'chr2'</span>, <span class="fl">55</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">GR</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span>seqnames <span class="op">=</span> <span class="va">chr</span>, <span class="fu">IRanges</span><span class="op">(</span>start <span class="op">=</span> <span class="va">start</span>, end <span class="op">=</span> <span class="va">end</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">mcols</span><span class="op">(</span><span class="va">GR</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">countData</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Gene IDs</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">GR</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'gene'</span>, <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">GR</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## An experiment design is set.</span></span></span>
<span class="r-in"><span><span class="va">colData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'A'</span>,<span class="st">'A'</span>,<span class="st">'B'</span>,<span class="st">'B'</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'A1'</span>,<span class="st">'A2'</span>,<span class="st">'B1'</span>,<span class="st">'B2'</span><span class="op">)</span>, row.names <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## A RangedGlmDataSet is created</span></span></span>
<span class="r-in"><span><span class="va">ds</span> <span class="op">&lt;-</span> <span class="fu"><a href="glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">GR</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The gneralized linear model pairwise group comparison, group 'A'</span></span></span>
<span class="r-in"><span><span class="co">## ('control') versus 'B' (treatment) is performed.</span></span></span>
<span class="r-in"><span><span class="fu">countTest2</span><span class="op">(</span><span class="va">ds</span>, num.cores <span class="op">=</span> <span class="fl">1L</span>, maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.4</span>, <span class="fl">0.4</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 4 ranges and 12 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           seqnames        ranges strand |        A1        A2        B1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    gene69     chr1 136001-137001      * |        57        67       117</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    gene98     chr2 194001-195001      * |        34        39       116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   gene101     chr2 200001-201001      * |       166       168       117</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   gene114     chr2 226001-227001      * |        33        34       111</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  B2    log2FC scaled.deviance      pvalue        model</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;integer&gt; &lt;numeric&gt;       &lt;numeric&gt;   &lt;numeric&gt;  &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    gene69       102  0.568790         26.6591 4.17025e-07 Neg.Binomial</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    gene98       151  1.296789         82.6185 9.57148e-18 Neg.Binomial</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   gene101        71 -0.574699         13.0830 3.00075e-04 Neg.Binomial</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   gene114       158  1.390019         64.7083 7.92421e-15 Neg.Binomial</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              adj.pval CT.SignalDensity TT.SignalDensity SignalDensityVariation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;numeric&gt;        &lt;numeric&gt;        &lt;numeric&gt;              &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    gene69 5.56034e-07        0.0619381        0.1093906              0.0474525</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    gene98 3.82859e-17        0.0364635        0.1333666              0.0969031</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   gene101 3.00075e-04        0.1668332        0.0939061             -0.0729271</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   gene114 1.58484e-14        0.0334665        0.1343656              0.1008991</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

