<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>DMP clustering — dmpClusters • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="DMP clustering — dmpClusters"><meta property="og:description" content="Given a 'pDMP' (or 'InfDiv') object carrying DMPs (methylated
cytosines) detected in Methyl-IT downstream analysis, function
'dmpClusters' build clusters of DMPs, which can be further
tested to identify differentially methylated regions (DMRs) with
countTest2 function."><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>DMP clustering</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/dmpClusters.R" class="external-link"><code>R/dmpClusters.R</code></a></small>
    <div class="hidden name"><code>dmpClusters.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given a 'pDMP' (or 'InfDiv') object carrying DMPs (methylated
cytosines) detected in Methyl-IT downstream analysis, function
<strong><em>'dmpClusters'</em></strong> build clusters of DMPs, which can be further
tested to identify differentially methylated regions (DMRs) with
<code><a href="countTest2.html">countTest2</a></code> function.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">dmpClusters</span><span class="op">(</span></span>
<span>  <span class="va">GR</span>,</span>
<span>  maxDist <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  minNumDMPs <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  maxClustDist <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"relaxed"</span>, <span class="st">"fixed.int"</span><span class="op">)</span>,</span>
<span>  chromosomes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  columns <span class="op">=</span> <span class="fl">9L</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fu">detectCores</span><span class="op">(</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>GR</dt>
<dd><p>An object from <strong><em>'pDMP'</em></strong> class, which is returned by
<code><a href="selectDIMP.html">selectDIMP</a></code> function.</p></dd>


<dt>maxDist</dt>
<dd><p>maximum distance at which two reported bases sites from the
same cluster can be separated. Default: \(maxDist = 3\).</p></dd>


<dt>minNumDMPs</dt>
<dd><p>Minimum number of DMPs inside of each cluster.
Default: \(minNumDMPs = 1\).</p></dd>


<dt>maxClustDist</dt>
<dd><p>Clusters separated by a distance lesser than
maxClustDist' positions are merged. Default: \(maxClustDist = NULL\). If
\(0 &lt; maxClustDist &lt; maxDist\) or \(maxClustDist = NULL\), then
maxClustDist will be recalculated as \(maxClustDist = maxDist + 1\).</p></dd>


<dt>method</dt>
<dd><p>Two different approaches are implemented to clustering DMPs:</p><dl><dt><strong>"relaxed":</strong></dt>
<dd><p>DMP ranges which are separated by a distance
less than <em>'maxClustDist'</em> are merged and ranges with less
than <em>'minNumDMPs'</em> are removed.</p></dd>

<dt><strong>"fixed.int":</strong></dt>
<dd><p>It will generate a partition where the
distance between consecutive DMPs is not greater than
<em>'maxDist'</em>. Ranges with less than <em>'minNumDMPs'</em> are
removed. If, additionally, a value <em>maxClustDist &gt; 0 </em> is
provided, then the "relaxed" approach is applied to the ranges from
the first step.</p></dd>


</dl></dd>


<dt>chromosomes</dt>
<dd><p>vector of characters labeling the chromosomes included in
the analysis. Default: chromosomes = NULL (all chromosomes are included).</p></dd>


<dt>columns</dt>
<dd><p>An integer number corresponding to the specific column(s) to
use from the meta-column of each GRanges. Default values is 9 (the column
carrying to the Hellinger divergence values).</p></dd>


<dt>ignore.strand</dt>
<dd><p>Same as in
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps-methods</a>.</code></p></dd>


<dt>num.cores, tasks</dt>
<dd><p>integer(1). The number of cores to use, i.e. at most
how many child processes will be run simultaneously (see
<code><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></code> function from BiocParallel
package).The number of tasks per job. value must be a scalar integer &gt;=
0L (see MulticoreParam from BiocParallel package).</p></dd>


<dt>verbose</dt>
<dd><p>if TRUE, prints the function log to stdout.</p></dd>


<dt>...</dt>
<dd><p>Further parameters for <code><a href="uniqueGRanges.html">uniqueGRanges</a></code> function.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A GRanges object with the numbers of positions inside each cluster,
where DMPs were reported in at least one of the samples.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Two algorithmic approaches are implemented, named: "relaxed" and
"fixed.int" (see the description of parameter 'method'). The "fixed.int" is
mostly addressed to find specific methylation patterns, but the price is
the number of DMRs found is lower.</p>
<p>The number of DMPs reported in each cluster corresponds to the numbers of
sites inside the cluster where DMPs were found in at least one of the
samples (from control or from treatment). That is, <strong>dmpClusters</strong> is
only a tool to locate regions with high density of DMPs from all the
samples. It does not detect DMRs. It is assumed that only DMP coordinates
are given in the <em>'GR'</em> object. That is, all the sites provided are
considered in the computation.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    
<ol><li><p>Sanchez R, Mackenzie SA (2016) Information Thermodynamics of
Cytosine DNA Methylation. PLOS ONE 11(3): e0150427.
https://doi.org/10.1371/journal.pone.0150427</p></li>
</ol></div>
    <div id="author">
    <h2>Author</h2>
    <p>Robersy Sanchez (<a href="https://genomaths.com" class="external-link">https://genomaths.com</a>).</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Get a dataset of dmps from the package</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Build clusters of DMPs taking into account the DNA strand</span></span></span>
<span class="r-in"><span><span class="va">x1</span> <span class="op">=</span> <span class="fu">dmpClusters</span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, maxDist <span class="op">=</span> <span class="fl">7</span>, minNumDMPs <span class="op">=</span> <span class="fl">6</span>,</span></span>
<span class="r-in"><span>                 method <span class="op">=</span> <span class="st">"fixed.int"</span>, ignore.strand <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                 verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">x1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    seqnames start  end width strand dmps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1         1  1208 1229    22      -    7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2         1  2142 2167    26      +    6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3         1  2423 2449    27      +    8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4         1  3273 3305    33      -    6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5         1  3405 3427    23      -    7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6         1  4656 4688    33      +    8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7         1  4758 4785    28      -    6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8         1  5626 5645    20      -    6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9         1  5890 5909    20      -    6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10        1  5940 5973    34      -    7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11        1  6226 6242    17      -    6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12        1  7379 7405    27      +    6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13        1  9530 9557    28      +    8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14        1  9913 9935    23      +    6</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="co">## Build clusters of DMPs ignoring DNA strand and maxClustDist = 7</span></span></span>
<span class="r-in"><span><span class="va">x2</span> <span class="op">=</span> <span class="fu">dmpClusters</span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, maxDist <span class="op">=</span> <span class="fl">7</span>, minNumDMPs <span class="op">=</span> <span class="fl">6</span>,</span></span>
<span class="r-in"><span>                  maxClustDist <span class="op">=</span> <span class="fl">7</span>, method <span class="op">=</span> <span class="st">"fixed.int"</span>,</span></span>
<span class="r-in"><span>                  num.cores<span class="op">=</span><span class="fl">2L</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                  verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">DataFrame</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">x2</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The relaxed approach with method = "relaxed"</span></span></span>
<span class="r-in"><span><span class="va">x3</span> <span class="op">=</span> <span class="fu">dmpClusters</span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, minNumDMPs <span class="op">=</span> <span class="fl">6</span>, method <span class="op">=</span> <span class="st">"relaxed"</span>,</span></span>
<span class="r-in"><span>                  maxClustDist <span class="op">=</span> <span class="fl">10</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                  verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">DataFrame</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">x3</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## ==== Setting up the experiment design to test for DMRs ===</span></span></span>
<span class="r-in"><span><span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dmps</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dmps_at_clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="getDMPatRegions.html">getDMPatRegions</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps</span>, regions <span class="op">=</span> <span class="va">x3</span>,</span></span>
<span class="r-in"><span>                                    ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dmps_at_clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="uniqueGRanges.html">uniqueGRanges</a></span><span class="op">(</span><span class="va">dmps_at_clusters</span>, columns <span class="op">=</span> <span class="fl">2L</span>,</span></span>
<span class="r-in"><span>                                ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>, type <span class="op">=</span> <span class="st">'equal'</span>,</span></span>
<span class="r-in"><span>                                verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dmps_at_clusters</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">nams</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">colData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'CT'</span>, <span class="st">'CT'</span>, <span class="st">'CT'</span>,</span></span>
<span class="r-in"><span>                                        <span class="st">'TT'</span>, <span class="st">'TT'</span>, <span class="st">'TT'</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                                        levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'CT'</span>, <span class="st">'TT'</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                    <span class="va">nams</span>, row.names <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Build a RangedGlmDataSet object</span></span></span>
<span class="r-in"><span><span class="va">ds</span> <span class="op">&lt;-</span> <span class="fu"><a href="glmDataSet.html">glmDataSet</a></span><span class="op">(</span>GR <span class="op">=</span> <span class="va">dmps_at_clusters</span>, colData <span class="op">=</span> <span class="va">colData</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## ================ Testing to detect DMRs ===========</span></span></span>
<span class="r-in"><span><span class="va">dmrs</span> <span class="op">&lt;-</span> <span class="fu"><a href="countTest2.html">countTest2</a></span><span class="op">(</span>DS <span class="op">=</span> <span class="va">ds</span>, num.cores <span class="op">=</span> <span class="fl">4L</span>, minCountPerIndv <span class="op">=</span> <span class="fl">4</span>,</span></span>
<span class="r-in"><span>                maxGrpCV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, Minlog2FC <span class="op">=</span> <span class="fl">0.5</span>,</span></span>
<span class="r-in"><span>                CountPerBp <span class="op">=</span> <span class="fl">0.001</span>, test <span class="op">=</span> <span class="st">'LRT'</span>,</span></span>
<span class="r-in"><span>                verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dmrs</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

