<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>DMP cluster builder — dmpClusters • MethylIT</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/all.min.css" integrity="sha256-PbSmjxuVAzJ6FPvNYsrXygfGhNJYyZ2GktDbkMBqQZg=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/v4-shims.min.css" integrity="sha256-A6jcAdwFD48VMjlI3GDxUd+eCQa7/KWy6G9oe/ovaPA=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="DMP cluster builder — dmpClusters" />
<meta property="og:description" content="Given a 'pDMP' object carrying DMPs obtained in Methyl-IT
downstream analysis, function &quot;dmpClusters&quot; build
clusters of DMPs, which can be further tested to identify differentially
methylated regions (DMRs) with countTest2 function." />
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png" />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/MethylIT">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>DMP cluster builder</h1>
    <small class="dont-index">Source: <a href='https://github.com/genomaths/MethylIT/blob/master/R/dmpClusters.R'><code>R/dmpClusters.R</code></a></small>
    <div class="hidden name"><code>dmpClusters.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given a 'pDMP' object carrying DMPs obtained in Methyl-IT
downstream analysis, function <strong><em>"dmpClusters"</em></strong> build
clusters of DMPs, which can be further tested to identify differentially
methylated regions (DMRs) with <code><a href='countTest2.html'>countTest2</a></code> function.</p>
    </div>

    <pre class="usage"><span class='fu'>dmpClusters</span>(
  <span class='no'>GR</span>,
  <span class='kw'>maxDist</span> <span class='kw'>=</span> <span class='fl'>3</span>,
  <span class='kw'>minNumDMPs</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>chromosomes</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>num.cores</span> <span class='kw'>=</span> <span class='fl'>1L</span>,
  <span class='kw'>tasks</span> <span class='kw'>=</span> <span class='fl'>0L</span>,
  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='no'>...</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>GR</th>
      <td><p>An object from <strong><em>'pDMP'</em></strong> class, which is returned by
<code><a href='selectDIMP.html'>selectDIMP</a></code> function.</p></td>
    </tr>
    <tr>
      <th>maxDist</th>
      <td><p>maximum distance at which two reported bases sites from the
same cluster can be separated. Default: \(maxDist = 3\).</p></td>
    </tr>
    <tr>
      <th>minNumDMPs</th>
      <td><p>minimum number of marked bases inside of each cluster.
Default: \(minNumDMPs = 1\).</p></td>
    </tr>
    <tr>
      <th>chromosomes</th>
      <td><p>vector of characters labeling the chromosomes included in
the analysis. Default: chromosomes = NULL (all chromosomes are included).</p></td>
    </tr>
    <tr>
      <th>num.cores, tasks</th>
      <td><p>integer(1). The number of cores to use, i.e. at most
how many child processes will be run simultaneously (see
<code><a href='https://rdrr.io/pkg/BiocParallel/man/bplapply.html'>bplapply</a></code> function from BiocParallel
package).The number of tasks per job. value must be a scalar integer &gt;=
0L (see MulticoreParam from BiocParallel package).</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>if TRUE, prints the function log to stdout.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Further parameters for uniqueGRanges function.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>GRanges object with the numbers of positions inside each cluster
where DMPs were reported in at least one of the samples.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>DNA base coordinates provided in the <strong><em>GR</em></strong> object are
used to build a binary string of zeros and ones . The binary string of zeros
and ones is used in the detection of cluster of DMPs. Postulating that a
reported DNA base is found at the beginning and the end of a cluster,
genome-wide screening can be performed where two consecutive targeting base
positions are separated by less than a given threshold
<strong><em>maxDist</em></strong> bases (1). DMPs from all the samples included in
<strong><em>GR</em></strong> are considered to build the clusters.</p>
<p>The number of DMPs reported in each cluster corresponds to the numbers of
positions inside the cluster where DMPs were reported in at least one
of the samples.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    
<ol>
<li><p>Sanchez R, Mackenzie SA (2016) Information Thermodynamics of
Cytosine DNA Methylation. PLOS ONE 11(3): e0150427.
https://doi.org/10.1371/journal.pone.0150427</p></li>
</ol>


    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Get a dataset of potential signals and the estimated cutpoint from the</span>
<span class='co'>## package and performs cutpoint estimation</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>PS</span>, <span class='no'>cutpoint</span>)

<span class='co'>## The estimated cutpoints are used to discriminate signals from the noise.</span>
<span class='co'>## That is, DMPs are selected using the cupoints</span>
<span class='no'>dmps</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='selectDIMP.html'>selectDIMP</a></span>(<span class='no'>PS</span>, <span class='kw'>div.col</span> <span class='kw'>=</span> <span class='fl'>9L</span>, <span class='kw'>cutpoint</span> <span class='kw'>=</span> <span class='no'>cutpoint</span>$<span class='no'>cutpoint</span>)

<span class='co'>## Build clusters of DMPs</span>
<span class='no'>x1</span> <span class='kw'>=</span> <span class='fu'>dmpClusters</span>(<span class='kw'>GR</span> <span class='kw'>=</span> <span class='no'>dmps</span>, <span class='kw'>maxDist</span> <span class='kw'>=</span> <span class='fl'>7</span>, <span class='kw'>minNumDMPs</span> <span class='kw'>=</span> <span class='fl'>6</span>,<span class='kw'>num.cores</span><span class='kw'>=</span><span class='fl'>2L</span>,
                <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='no'>x1</span></div><div class='output co'>#&gt; GRanges object with 60 ranges and 1 metadata column:
#&gt;        seqnames    ranges strand |      dmps
#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;
#&gt;    [1]        1    91-119      * |         7
#&gt;    [2]        1   208-234      * |         8
#&gt;    [3]        1   552-575      * |         7
#&gt;    [4]        1   765-786      * |         9
#&gt;    [5]        1 1125-1142      * |         6
#&gt;    ...      ...       ...    ... .       ...
#&gt;   [56]        1 9409-9427      * |         6
#&gt;   [57]        1 9444-9469      * |         8
#&gt;   [58]        1 9538-9559      * |         8
#&gt;   [59]        1 9726-9742      * |         6
#&gt;   [60]        1 9911-9958      * |        14
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>## ==== Setting up the experimennt design to test for DMRs ===</span>
<span class='no'>nams</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>dmps</span>)
<span class='no'>dmps_at_clusters</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='getDIMPatGenes.html'>getDIMPatGenes</a></span>(<span class='kw'>GR</span> <span class='kw'>=</span> <span class='no'>dmps</span>, <span class='kw'>GENES</span> <span class='kw'>=</span> <span class='no'>x1</span>,
                                    <span class='kw'>ignore.strand</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)
<span class='no'>dmps_at_clusters</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='uniqueGRanges.html'>uniqueGRanges</a></span>(<span class='no'>dmps_at_clusters</span>, <span class='kw'>columns</span> <span class='kw'>=</span> <span class='fl'>2L</span>,
                                    <span class='kw'>ignore.strand</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>type</span> <span class='kw'>=</span> <span class='st'>"equal"</span>,
                                    <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span>(<span class='fu'>mcols</span>(<span class='no'>dmps_at_clusters</span>)) <span class='kw'>&lt;-</span> <span class='no'>nams</span>

<span class='no'>colData</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span>(<span class='kw'>condition</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"CT"</span>, <span class='st'>"CT"</span>, <span class='st'>"CT"</span>,
                                           <span class='st'>"TT"</span>, <span class='st'>"TT"</span>, <span class='st'>"TT"</span>),
                                         <span class='kw'>levels</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"CT"</span>, <span class='st'>"TT"</span>)),
                      <span class='no'>nams</span>,
                      <span class='kw'>row.names</span> <span class='kw'>=</span> <span class='fl'>2</span>)

<span class='co'>## Build RangedGlmDataSet is created</span>
<span class='no'>ds</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='glmDataSet.html'>glmDataSet</a></span>(<span class='kw'>GR</span> <span class='kw'>=</span> <span class='no'>dmps_at_clusters</span>, <span class='kw'>colData</span> <span class='kw'>=</span> <span class='no'>colData</span>)

<span class='co'>## ================ Testing for DMRs ===========</span>
<span class='no'>dmrs</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='countTest2.html'>countTest2</a></span>(<span class='kw'>DS</span> <span class='kw'>=</span> <span class='no'>ds</span>, <span class='kw'>num.cores</span> <span class='kw'>=</span> <span class='fl'>4L</span>,
                   <span class='kw'>minCountPerIndv</span> <span class='kw'>=</span> <span class='fl'>4</span>,
                   <span class='kw'>maxGrpCV</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>1</span>, <span class='fl'>1</span>),
                   <span class='kw'>Minlog2FC</span> <span class='kw'>=</span> <span class='fl'>0.5</span>,
                   <span class='kw'>CountPerBp</span> <span class='kw'>=</span> <span class='fl'>0.001</span>,
                   <span class='kw'>test</span> <span class='kw'>=</span> <span class='st'>"LRT"</span>,
                   <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</div><div class='output co'>#&gt; <span class='message'>*** Number of GR after filtering counts 6</span></div><div class='output co'>#&gt; <span class='message'>*** GLM...</span></div><div class='input'><span class='no'>dmrs</span></div><div class='output co'>#&gt; GRanges object with 6 ranges and 14 metadata columns:
#&gt;       seqnames    ranges strand |        C1        C2        C3        T1
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
#&gt;   [1]        1 1204-1238      * |         0         0         0         5
#&gt;   [2]        1 2142-2172      * |         0         0         0         2
#&gt;   [3]        1 2325-2359      * |         0         0         1         5
#&gt;   [4]        1 4656-4708      * |         0         2         0         8
#&gt;   [5]        1 7722-7775      * |         1         0         0         5
#&gt;   [6]        1 9911-9958      * |         0         0         0         4
#&gt;              T2        T3           log2FC  scaled.deviance              pvalue
#&gt;       &lt;numeric&gt; &lt;numeric&gt;        &lt;numeric&gt;        &lt;numeric&gt;           &lt;numeric&gt;
#&gt;   [1]         5         4 1.73460105538811 8.48613318671429 0.00357863605668028
#&gt;   [2]         6         4  1.6094379124341 6.94832058758156 0.00838979577447831
#&gt;   [3]         3         5 1.38629436111939  30.839145590753 0.00514571648076653
#&gt;   [4]         5         6 1.48160454025156 20.6669445813151  0.0104485974536701
#&gt;   [5]         1         7 1.38629436139456  7.7096310654329  0.0499909384290531
#&gt;   [6]         9         3 1.84582669049833 9.55203643495998 0.00199727710868162
#&gt;                model           adj.pval    CT.SignalDensity   TT.SignalDensity
#&gt;             &lt;factor&gt;          &lt;numeric&gt;           &lt;numeric&gt;          &lt;numeric&gt;
#&gt;   [1] Neg.Binomial.W 0.0102914329615331                   0  0.133333333333333
#&gt;   [2] Neg.Binomial.W 0.0125383169444042                   0  0.129032258064516
#&gt;   [3]   QuasiPoisson 0.0102914329615331 0.00952380952380952  0.123809523809524
#&gt;   [4]   QuasiPoisson 0.0125383169444042  0.0125786163522013  0.119496855345912
#&gt;   [5]   QuasiPoisson 0.0499909384290531 0.00617283950617284 0.0802469135802469
#&gt;   [6] Neg.Binomial.W 0.0102914329615331                   0  0.111111111111111
#&gt;       SignalDensityVariation
#&gt;                    &lt;numeric&gt;
#&gt;   [1]      0.133333333333333
#&gt;   [2]      0.129032258064516
#&gt;   [3]      0.114285714285714
#&gt;   [4]      0.106918238993711
#&gt;   [5]     0.0740740740740741
#&gt;   [6]      0.111111111111111
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#details">Details</a></li>
      <li><a href="#references">References</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Robersy Sanchez (<a href='https://genomaths.com'>https://genomaths.com</a>).</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


