<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>DMP clustering — dmpClusters • MethylIT</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="DMP clustering — dmpClusters" />
<meta property="og:description" content="Given a 'pDMP' object carrying DMPs obtained in Methyl-IT
downstream analysis, function 'dmpClusters' build
clusters of DMPs, which can be further tested to identify differentially
methylated regions (DMRs) with countTest2 function." />
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/MethylIT/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>DMP clustering</h1>
    <small class="dont-index">Source: <a href='https://github.com/genomaths/MethylIT/blob/master/R/dmpClusters.R'><code>R/dmpClusters.R</code></a></small>
    <div class="hidden name"><code>dmpClusters.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given a 'pDMP' object carrying DMPs obtained in Methyl-IT
downstream analysis, function <strong><em>'dmpClusters'</em></strong> build
clusters of DMPs, which can be further tested to identify differentially
methylated regions (DMRs) with <code><a href='countTest2.html'>countTest2</a></code> function.</p>
    </div>

    <pre class="usage"><span class='fu'>dmpClusters</span>(
  <span class='no'>GR</span>,
  <span class='kw'>maxDist</span> <span class='kw'>=</span> <span class='fl'>3</span>,
  <span class='kw'>minNumDMPs</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>maxClustDist</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>method</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"relaxed"</span>, <span class='st'>"fixed.int"</span>),
  <span class='kw'>chromosomes</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>ignore.strand</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>num.cores</span> <span class='kw'>=</span> <span class='fl'>1L</span>,
  <span class='kw'>tasks</span> <span class='kw'>=</span> <span class='fl'>0L</span>,
  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='no'>...</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>GR</th>
      <td><p>An object from <strong><em>'pDMP'</em></strong> class, which is returned by
<code><a href='selectDIMP.html'>selectDIMP</a></code> function.</p></td>
    </tr>
    <tr>
      <th>maxDist</th>
      <td><p>maximum distance at which two reported bases sites from the
same cluster can be separated. Default: \(maxDist = 3\).</p></td>
    </tr>
    <tr>
      <th>minNumDMPs</th>
      <td><p>Minimum number of DMPs inside of each cluster.
Default: \(minNumDMPs = 1\).</p></td>
    </tr>
    <tr>
      <th>maxClustDist</th>
      <td><p>Clusters separated by a distance lesser than
'maxClustDist' positions are merged. Default: \(maxClustDist = NULL\). If
\(0 &lt; maxClustDist &lt; maxDist\), then maxClustDist will be recalculated as
\(maxClustDist = maxDist + 1\).</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>Two different approaches are implemented to clustering DMPs:</p><dl'>
<dt><strong>"relaxed":</strong></dt><dd><p>DMP ranges which are separated by a distance
less than <em>'maxClustDist'</em> are merged and ranges with less
than <em>'minNumDMPs'</em> are removed.</p></dd>
<dt><strong>"fixed.int":</strong></dt><dd><p>It will gnetate a partition where the distance
between consecutive DMPs is not greater than <em>'maxDist'</em>.
Ranges with less than <em>'minNumDMPs'</em> are removed. If,
aditionally, a value <em>maxClustDist &gt; 0 </em> is provided, then the
"relaxed" approach is applied to the ranges from the first step.</p></dd>

</dl></td>
    </tr>
    <tr>
      <th>chromosomes</th>
      <td><p>vector of characters labeling the chromosomes included in
the analysis. Default: chromosomes = NULL (all chromosomes are included).</p></td>
    </tr>
    <tr>
      <th>ignore.strand</th>
      <td><p>Same as in
<code><a href='https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html'>findOverlaps-methods</a>.</code></p></td>
    </tr>
    <tr>
      <th>num.cores, tasks</th>
      <td><p>integer(1). The number of cores to use, i.e. at most
how many child processes will be run simultaneously (see
<code><a href='https://rdrr.io/pkg/BiocParallel/man/bplapply.html'>bplapply</a></code> function from BiocParallel
package).The number of tasks per job. value must be a scalar integer &gt;=
0L (see MulticoreParam from BiocParallel package).</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>if TRUE, prints the function log to stdout.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Further parameters for uniqueGRanges function.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A GRanges object with the numbers of positions inside each cluster,
where DMPs were reported in at least one of the samples.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Two algorithmic approaches are implemented, named: "relaxed" and
"fixed.int" (see the decription of parmeter 'method'). The "fixed.int" is
mostly addressed to find specific methylation patterns, but the price is
the number of DMRs found is lower.</p>
<p>The number of DMPs reported in each cluster corresponds to the numbers of
sites inside the cluster where DMPs were found in at least one of the
samples (from control or from treatment). That is, <strong>dmpClusters</strong> is
only a tool to locate regions with high density of DMPs from all the
samples. It does not detect DMRs. It is assumed that only DMP coordinates
are given in the <em>'GR'</em> object. That is, all the sites provided are
considered in the computation.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    
<ol>
<li><p>Sanchez R, Mackenzie SA (2016) Information Thermodynamics of
Cytosine DNA Methylation. PLOS ONE 11(3): e0150427.
https://doi.org/10.1371/journal.pone.0150427</p></li>
</ol>


    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Get a dataset of dmps from the package</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>dmps</span>)

<span class='co'>## Build clusters of DMPs taking into account the DNA strand</span>
<span class='no'>x1</span> <span class='kw'>=</span> <span class='fu'>dmpClusters2</span>(<span class='kw'>GR</span> <span class='kw'>=</span> <span class='no'>dmps</span>, <span class='kw'>maxDist</span> <span class='kw'>=</span> <span class='fl'>7</span>, <span class='kw'>minNumDMPs</span> <span class='kw'>=</span> <span class='fl'>6</span>,
                 <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"fixed.int"</span>, <span class='kw'>ignore.strand</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
                 <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</div><div class='output co'>#&gt; <span class='error'>Error in dmpClusters2(GR = dmps, maxDist = 7, minNumDMPs = 6, method = "fixed.int",     ignore.strand = FALSE, verbose = FALSE): could not find function "dmpClusters2"</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span>(<span class='no'>x1</span>)</div><div class='output co'>#&gt; <span class='error'>Error in data.frame(x1): object 'x1' not found</span></div><div class='input'>
<span class='co'># \donttest{</span>
<span class='co'>## Build clusters of DMPs ignoring DNA strand and maxClustDist = 7</span>
<span class='no'>x2</span> <span class='kw'>=</span> <span class='fu'>dmpClusters2</span>(<span class='kw'>GR</span> <span class='kw'>=</span> <span class='no'>dmps</span>, <span class='kw'>maxDist</span> <span class='kw'>=</span> <span class='fl'>7</span>, <span class='kw'>minNumDMPs</span> <span class='kw'>=</span> <span class='fl'>6</span>,
                  <span class='kw'>maxClustDist</span> <span class='kw'>=</span> <span class='fl'>7</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"fixed.int"</span>,
                  <span class='kw'>num.cores</span><span class='kw'>=</span><span class='fl'>2L</span>, <span class='kw'>ignore.strand</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
                  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</div><div class='output co'>#&gt; <span class='error'>Error in dmpClusters2(GR = dmps, maxDist = 7, minNumDMPs = 6, maxClustDist = 7,     method = "fixed.int", num.cores = 2L, ignore.strand = TRUE,     verbose = FALSE): could not find function "dmpClusters2"</span></div><div class='input'><span class='fu'>DataFrame</span>(<span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span>(<span class='no'>x2</span>))</div><div class='output co'>#&gt; <span class='error'>Error in data.frame(x2): object 'x2' not found</span></div><div class='input'>
<span class='co'>## The relaxed approach with method = "relaxed"</span>
<span class='no'>x3</span> <span class='kw'>=</span> <span class='fu'>dmpClusters2</span>(<span class='kw'>GR</span> <span class='kw'>=</span> <span class='no'>dmps</span>, <span class='kw'>minNumDMPs</span> <span class='kw'>=</span> <span class='fl'>6</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"relaxed"</span>,
                  <span class='kw'>maxClustDist</span> <span class='kw'>=</span> <span class='fl'>10</span>, <span class='kw'>ignore.strand</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
                  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</div><div class='output co'>#&gt; <span class='error'>Error in dmpClusters2(GR = dmps, minNumDMPs = 6, method = "relaxed", maxClustDist = 10,     ignore.strand = TRUE, verbose = FALSE): could not find function "dmpClusters2"</span></div><div class='input'><span class='fu'>DataFrame</span>(<span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span>(<span class='no'>x3</span>))</div><div class='output co'>#&gt; <span class='error'>Error in data.frame(x3): object 'x3' not found</span></div><div class='input'>
<span class='co'>## ==== Setting up the experiment design to test for DMRs ===</span>
<span class='no'>nams</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>dmps</span>)
<span class='no'>dmps_at_clusters</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='getDMPatRegions.html'>getDMPatRegions</a></span>(<span class='kw'>GR</span> <span class='kw'>=</span> <span class='no'>dmps</span>, <span class='kw'>regions</span> <span class='kw'>=</span> <span class='no'>x3</span>,
                                    <span class='kw'>ignore.strand</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</div><div class='output co'>#&gt; <span class='error'>Error in FUN(X[[i]], ...): object 'x3' not found</span></div><div class='input'><span class='no'>dmps_at_clusters</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='uniqueGRanges.html'>uniqueGRanges</a></span>(<span class='no'>dmps_at_clusters</span>, <span class='kw'>columns</span> <span class='kw'>=</span> <span class='fl'>2L</span>,
                                <span class='kw'>ignore.strand</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>type</span> <span class='kw'>=</span> <span class='st'>'equal'</span>,
                                <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</div><div class='output co'>#&gt; <span class='error'>Error in is(ListOfGranges, "list"): object 'dmps_at_clusters' not found</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span>(<span class='fu'>mcols</span>(<span class='no'>dmps_at_clusters</span>)) <span class='kw'>&lt;-</span> <span class='no'>nams</span></div><div class='output co'>#&gt; <span class='error'>Error in colnames(mcols(dmps_at_clusters)) &lt;- nams: object 'dmps_at_clusters' not found</span></div><div class='input'>
<span class='no'>colData</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span>(<span class='kw'>condition</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>'CT'</span>, <span class='st'>'CT'</span>, <span class='st'>'CT'</span>,
                                        <span class='st'>'TT'</span>, <span class='st'>'TT'</span>, <span class='st'>'TT'</span>),
                                        <span class='kw'>levels</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>'CT'</span>, <span class='st'>'TT'</span>)),
                    <span class='no'>nams</span>, <span class='kw'>row.names</span> <span class='kw'>=</span> <span class='fl'>2</span>)

<span class='co'>## Build a RangedGlmDataSet object</span>
<span class='no'>ds</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='glmDataSet.html'>glmDataSet</a></span>(<span class='kw'>GR</span> <span class='kw'>=</span> <span class='no'>dmps_at_clusters</span>, <span class='kw'>colData</span> <span class='kw'>=</span> <span class='no'>colData</span>)</div><div class='output co'>#&gt; <span class='error'>Error in glmDataSet(GR = dmps_at_clusters, colData = colData): object 'dmps_at_clusters' not found</span></div><div class='input'>
<span class='co'>## ================ Testing to detect DMRs ===========</span>
<span class='no'>dmrs</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='countTest2.html'>countTest2</a></span>(<span class='kw'>DS</span> <span class='kw'>=</span> <span class='no'>ds</span>, <span class='kw'>num.cores</span> <span class='kw'>=</span> <span class='fl'>4L</span>, <span class='kw'>minCountPerIndv</span> <span class='kw'>=</span> <span class='fl'>4</span>,
                <span class='kw'>maxGrpCV</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>1</span>, <span class='fl'>1</span>), <span class='kw'>Minlog2FC</span> <span class='kw'>=</span> <span class='fl'>0.5</span>,
                <span class='kw'>CountPerBp</span> <span class='kw'>=</span> <span class='fl'>0.001</span>, <span class='kw'>test</span> <span class='kw'>=</span> <span class='st'>'LRT'</span>,
                <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</div><div class='output co'>#&gt; <span class='message'>*** Number of GR after filtering counts 92</span></div><div class='output co'>#&gt; <span class='message'>*** GLM...</span></div><div class='input'><span class='no'>dmrs</span></div><div class='output co'>#&gt; GRanges object with 22 ranges and 12 metadata columns:
#&gt;           seqnames        ranges strand |        A1        A2        B1
#&gt;              &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
#&gt;    gene22     chr1   42001-43001      * |        44        53       187
#&gt;    gene26     chr1   50001-51001      * |       174       139        32
#&gt;    gene29     chr1   56001-57001      * |        12        19       129
#&gt;    gene44     chr1   86001-87001      * |       110       173        23
#&gt;    gene57     chr1 112001-113001      * |        25        94       187
#&gt;       ...      ...           ...    ... .       ...       ...       ...
#&gt;   gene106     chr2 210001-211001      * |       122        85         3
#&gt;   gene107     chr2 212001-213001      * |       181       111        76
#&gt;   gene113     chr2 224001-225001      * |       142       172        59
#&gt;   gene114     chr2 226001-227001      * |        33        34       111
#&gt;   gene117     chr2 232001-233001      * |        55        77        13
#&gt;                  B2             log2FC  scaled.deviance               pvalue
#&gt;           &lt;integer&gt;          &lt;numeric&gt;        &lt;numeric&gt;            &lt;numeric&gt;
#&gt;    gene22        85   1.03109108779261 11.5705078723797 0.000670061144147836
#&gt;    gene26        62  -1.20290840827015 30.9662853450191 2.62549774873693e-08
#&gt;    gene29        64   1.82870298441974 29.4987592669987 5.59525099643946e-08
#&gt;    gene44        88 -0.935916696330904  3.9493596280495   0.0468892079882497
#&gt;    gene57       174   1.10975446522135 6.43211707492669    0.011207491350226
#&gt;       ...       ...                ...              ...                  ...
#&gt;   gene106        16  -2.38827981409893 30.9821950561452 2.60406378697818e-08
#&gt;   gene107        21   -1.1020428237649 5.75602794152791   0.0164321785717638
#&gt;   gene113       127 -0.523646312195051 3.84537303089257   0.0498834241817064
#&gt;   gene114       158   1.39001876021087 15.1218903312962 0.000100787798473054
#&gt;   gene117        43 -0.857450231851222 4.67402216233158   0.0306221309485589
#&gt;                    model             adj.pval   CT.SignalDensity
#&gt;              &lt;character&gt;            &lt;numeric&gt;          &lt;numeric&gt;
#&gt;    gene22   Neg.Binomial   0.0012284454309377 0.0484515484515484
#&gt;    gene26   Neg.Binomial 1.92536501574041e-07  0.156343656343656
#&gt;    gene29   Neg.Binomial 3.07738804804171e-07 0.0154845154845155
#&gt;    gene44   Neg.Binomial   0.0491220274162616  0.141358641358641
#&gt;    gene57   Neg.Binomial   0.0164376539803314 0.0594405594405594
#&gt;       ...            ...                  ...                ...
#&gt;   gene106   Neg.Binomial 1.92536501574041e-07  0.103396603396603
#&gt;   gene107   Neg.Binomial   0.0212651722693414  0.145854145854146
#&gt;   gene113   Neg.Binomial   0.0498834241817064  0.156843156843157
#&gt;   gene114 Neg.Binomial.W 0.000256868046468425 0.0334665334665335
#&gt;   gene117   Neg.Binomial   0.0374270489371276 0.0659340659340659
#&gt;              TT.SignalDensity SignalDensityVariation
#&gt;                     &lt;numeric&gt;              &lt;numeric&gt;
#&gt;    gene22   0.135864135864136     0.0874125874125874
#&gt;    gene26   0.046953046953047     -0.109390609390609
#&gt;    gene29  0.0964035964035964     0.0809190809190809
#&gt;    gene44  0.0554445554445554    -0.0859140859140859
#&gt;    gene57    0.18031968031968      0.120879120879121
#&gt;       ...                 ...                    ...
#&gt;   gene106 0.00949050949050949    -0.0939060939060939
#&gt;   gene107  0.0484515484515484    -0.0974025974025974
#&gt;   gene113  0.0929070929070929    -0.0639360639360639
#&gt;   gene114   0.134365634365634      0.100899100899101
#&gt;   gene117   0.027972027972028     -0.037962037962038
#&gt;   -------
#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</div><div class='input'># }

</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


