<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Find Differentially Methylated Regions (DMRs) — dmrfinder • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Find Differentially Methylated Regions (DMRs) — dmrfinder"><meta property="og:description" content="A function addressed to find DMRs, assuming that differentially methylated
position (DMPs) are already identified and carried in 'obj'.
Following MethylIT pipeline, DMRs can be identified in control and
in treatment groups. The concept of DMR is subjective in some extend.
That is, while MethylIT pipeline identify cytosine positions with high
probability to be differential methylated in the treatment group with
respect to the control group, it will correspond to the user to define
whether to say, e.g., a region 100 bp long carrying 3 DMPs is a DMR. There
is not theoretical ground to decide whether such a DMR would be biologically
relevant or not for a given species. Only the experimentation or further
downstream analyses including, for example, the network enrichment analysis
of DMR-associated genes would gives us an answer."><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Find Differentially Methylated Regions (DMRs)</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/dmrfinder.R" class="external-link"><code>R/dmrfinder.R</code></a></small>
    <div class="hidden name"><code>dmrfinder.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A function addressed to find DMRs, assuming that differentially methylated
position (DMPs) are already identified and carried in 'obj'.</p>
<p>Following MethylIT pipeline, DMRs can be identified in control and
in treatment groups. The concept of DMR is subjective in some extend.
That is, while MethylIT pipeline identify cytosine positions with high
probability to be differential methylated in the treatment group with
respect to the control group, it will correspond to the user to define
whether to say, e.g., a region 100 bp long carrying 3 DMPs is a DMR. There
is not theoretical ground to decide whether such a DMR would be biologically
relevant or not for a given species. Only the experimentation or further
downstream analyses including, for example, the network enrichment analysis
of DMR-associated genes would gives us an answer.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">dmrfinder</span><span class="op">(</span><span class="va">obj</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for list</span></span>
<span><span class="fu">dmrfinder</span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  hypo <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  hyper <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  min.signal <span class="op">=</span> <span class="fl">2L</span>,</span>
<span>  win.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  step.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  grfeatures <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"count"</span>, <span class="st">"denCount"</span>, <span class="st">"density"</span>, <span class="st">"mean"</span>, <span class="st">"sum"</span>, <span class="st">"gmean"</span>, <span class="st">"median"</span><span class="op">)</span>,</span>
<span>  column <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tv.colum <span class="op">=</span> <span class="fl">8L</span>,</span>
<span>  absolute <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  select.strand <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  maxgap <span class="op">=</span> <span class="op">-</span><span class="fl">1L</span>,</span>
<span>  minoverlap <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  select <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"within"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"equal"</span>, <span class="st">"any"</span><span class="op">)</span>,</span>
<span>  scaling <span class="op">=</span> <span class="fl">1000L</span>,</span>
<span>  logbase <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  missings <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  naming <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  plots <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  plot.chr <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  num.plot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span>,</span>
<span>  xlab <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ylab <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  mfcol <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cex.axis <span class="op">=</span> <span class="fl">1.1</span>,</span>
<span>  cex.lab <span class="op">=</span> <span class="fl">1.1</span>,</span>
<span>  mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for pDMP</span></span>
<span><span class="fu">dmrfinder</span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  hypo <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  hyper <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  min.signal <span class="op">=</span> <span class="fl">2L</span>,</span>
<span>  win.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  step.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  grfeatures <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sum"</span>, <span class="st">"mean"</span>, <span class="st">"gmean"</span>, <span class="st">"median"</span>, <span class="st">"density"</span>, <span class="st">"count"</span>, <span class="st">"denCount"</span><span class="op">)</span>,</span>
<span>  column <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tv.colum <span class="op">=</span> <span class="fl">8L</span>,</span>
<span>  absolute <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  select.strand <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  maxgap <span class="op">=</span> <span class="op">-</span><span class="fl">1L</span>,</span>
<span>  minoverlap <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  select <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"within"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"equal"</span>, <span class="st">"any"</span><span class="op">)</span>,</span>
<span>  scaling <span class="op">=</span> <span class="fl">1000L</span>,</span>
<span>  logbase <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  missings <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  naming <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  plots <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  plot.chr <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  num.plot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span>,</span>
<span>  xlab <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ylab <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  mfcol <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cex.axis <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  cex.lab <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for InfDiv</span></span>
<span><span class="fu">dmrfinder</span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  hypo <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  hyper <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  min.signal <span class="op">=</span> <span class="fl">2L</span>,</span>
<span>  win.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  step.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  grfeatures <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sum"</span>, <span class="st">"mean"</span>, <span class="st">"gmean"</span>, <span class="st">"median"</span>, <span class="st">"density"</span>, <span class="st">"count"</span>, <span class="st">"denCount"</span><span class="op">)</span>,</span>
<span>  column <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tv.colum <span class="op">=</span> <span class="fl">8L</span>,</span>
<span>  absolute <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  select.strand <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  maxgap <span class="op">=</span> <span class="op">-</span><span class="fl">1L</span>,</span>
<span>  minoverlap <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  select <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"within"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"equal"</span>, <span class="st">"any"</span><span class="op">)</span>,</span>
<span>  scaling <span class="op">=</span> <span class="fl">1000L</span>,</span>
<span>  logbase <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  missings <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  naming <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  plots <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  plot.chr <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  num.plot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span>,</span>
<span>  xlab <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ylab <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  mfcol <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cex.axis <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  cex.lab <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>obj</dt>
<dd><p>An object from pDMP or InfDiv class or simply a list of
GRanges-class.</p></dd>


<dt>...</dt>
<dd><p>If plots == TRUE, then further parameter for plot can be passed.</p></dd>


<dt>hypo</dt>
<dd><p>Logical. If TRUE then only hypo-methylated positions are
considered.</p></dd>


<dt>hyper</dt>
<dd><p>Logical. If TRUE then only hyper-methylated positions are
considered.</p></dd>


<dt>min.signal</dt>
<dd><p>A number denoting minimun number of differentially
methylated positions (DMPs) or signal value (if stat != "count") that must
be in region to be considered as a potential differentially methylated
region (DMR).</p></dd>


<dt>win.size, step.size, grfeatures, stat, column</dt>
<dd><p>Described in
<a href="getGRangesStat.html">getGRangesStat</a>.</p></dd>


<dt>tv.colum</dt>
<dd><p>Only if hypo or hyper is TRUE. An integer denoting whether
the column carrying the difference of methylation levels (total variance,
TV) is locate on each <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> object from
'obj'.</p></dd>


<dt>absolute, select.strand, maxgap, minoverlap</dt>
<dd><p>Described
in <a href="getGRangesStat.html">getGRangesStat</a>.</p></dd>


<dt>select, ignore.strand, type, scaling, logbase</dt>
<dd><p>Described
in <a href="getGRangesStat.html">getGRangesStat</a>.</p></dd>


<dt>missings, naming, na.rm, num.cores, tasks</dt>
<dd><p>Described
in <a href="getGRangesStat.html">getGRangesStat</a>.</p></dd>


<dt>plots</dt>
<dd><p>Logical. Whether to plot a graphic showing the identified DMRs
(if any) on a specified chromosome coordinates.</p></dd>


<dt>plot.chr</dt>
<dd><p>DMRs found on different chromosomes would have the same
numerical coordinate and, consequently to show the graphics, we must
specify the chromosome.</p></dd>


<dt>num.plot</dt>
<dd><p>If plots = TRUE, an integer denoting the number of plots.</p></dd>


<dt>xlab, ylab, mfcol, cex.lab, mar, verbose</dt>
<dd><p>Described in <a href="getGRangesStat.html">getGRangesStat</a>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A GRanges carrying all the identified DMRs. The indexes of the regions
holding the condition sets by 'min.signal' can be retrieved by typing
'metadata(x)'.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>There is not a unique setting to find DMRs. Under different settings
clusterization algorithm can yield different set of DMRs. The users are
responsible to find which are the best setting for them. A DMR found in the
treatment group does not implies that the same DMR is not in the control
group. Indeed, for example, a DMR found in the treatment group would be also
a DMR in the control group, where some DMPs can share and some not, or
simply the DMR can be mostly hypo-methylated in control group and mostly
hyper-methylated in the treatment group.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Load a DMPs example data set</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">dmps</span>, package <span class="op">=</span> <span class="st">"MethylIT"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Search hypo-methylated DMRs with at least two DMPs on 200 bp.</span></span></span>
<span class="r-in"><span><span class="co">## The graphics of samples 2 to 5 are returned and</span></span></span>
<span class="r-in"><span><span class="co">## they indicate that there is not hypo-DMRs with such</span></span></span>
<span class="r-in"><span><span class="co">## features in neither of the groups, control and treatment.</span></span></span>
<span class="r-in"><span><span class="va">hyp_dmrs</span> <span class="op">&lt;-</span> <span class="fu">dmrfinder</span><span class="op">(</span></span></span>
<span class="r-in"><span>    obj <span class="op">=</span> <span class="va">dmps</span>,</span></span>
<span class="r-in"><span>    hypo <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    min.signal <span class="op">=</span> <span class="fl">2L</span>,</span></span>
<span class="r-in"><span>    win.size <span class="op">=</span> <span class="fl">200</span>,</span></span>
<span class="r-in"><span>    step.size <span class="op">=</span> <span class="fl">50</span>,</span></span>
<span class="r-in"><span>    stat <span class="op">=</span> <span class="st">"count"</span>,</span></span>
<span class="r-in"><span>    tv.colum <span class="op">=</span> <span class="fl">8L</span>,</span></span>
<span class="r-in"><span>    naming <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    plots <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    mfcol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    num.plot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="dmrfinder-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Search hyper-methylated DMRs with at least two DMPs on 200 bp.</span></span></span>
<span class="r-in"><span><span class="co">## Only the treatment group carries hyper-DMRs with such features.</span></span></span>
<span class="r-in"><span><span class="va">hyp_dmrs</span> <span class="op">&lt;-</span> <span class="fu">dmrfinder</span><span class="op">(</span></span></span>
<span class="r-in"><span>    obj <span class="op">=</span> <span class="va">dmps</span>,</span></span>
<span class="r-in"><span>    hyper <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    min.signal <span class="op">=</span> <span class="fl">10L</span>,</span></span>
<span class="r-in"><span>    win.size <span class="op">=</span> <span class="fl">200</span>,</span></span>
<span class="r-in"><span>    step.size <span class="op">=</span> <span class="fl">50</span>,</span></span>
<span class="r-in"><span>    stat <span class="op">=</span> <span class="st">"count"</span>,</span></span>
<span class="r-in"><span>    tv.colum <span class="op">=</span> <span class="fl">8L</span>,</span></span>
<span class="r-in"><span>    naming <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    plots <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    mfcol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    num.plot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="dmrfinder-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

