<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>J Information Divergence of Methylation Levels — estimateJDiv • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="J Information Divergence of Methylation Levels — estimateJDiv"><meta property="og:description" content="Given a the methylation levels from two individuals at a given
cytosine site, this function computes the J information divergence
(JD) between methylation levels. The motivation to introduce
JD in Methyl-IT is founded on:
It is a symmetrised form of Kullback–Leibler divergence
(\(D_{KL}\)). Kullback and Leibler themselves actually defined
the divergence as: \(D_{KL}(P || Q) + D_{KL}(Q || P)\), which
is symmetric and nonnegative, where the probability distributions
P and Q are defined on the same probability space
(see reference (1) and Wikipedia).
In general, JD is highly correlated with Hellinger
divergence, which is the main divergence currently used in
Methyl-IT (see examples for function
estimateDivergence.
By construction, the unit of measurement of JD is:
bit of information, which set the basis for further
information-thermodynamics analyses.

"><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>J Information Divergence of Methylation Levels</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/estimateJDiv.R" class="external-link"><code>R/estimateJDiv.R</code></a></small>
    <div class="hidden name"><code>estimateJDiv.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given a the methylation levels from two individuals at a given
cytosine site, this function computes the J information divergence
(<em>JD</em>) between methylation levels. The motivation to introduce
<em>JD</em> in Methyl-IT is founded on:</p><ol><li><p>It is a symmetrised form of Kullback–Leibler divergence
(\(D_{KL}\)). Kullback and Leibler themselves actually defined
the divergence as: \(D_{KL}(P || Q) + D_{KL}(Q || P)\), which
is symmetric and nonnegative, where the probability distributions
<em>P</em> and <em>Q</em> are defined on the same probability space
(see reference (1) and <a href="https://is.gd/oS8Bwv" class="external-link">Wikipedia</a>).</p></li>
<li><p>In general, <em>JD</em> is highly correlated with Hellinger
divergence, which is the main divergence currently used in
Methyl-IT (see examples for function
<code><a href="estimateDivergence.html">estimateDivergence</a></code>.</p></li>
<li><p>By construction, the unit of measurement of <em>JD</em> is:
bit of information, which set the basis for further
information-thermodynamics analyses.</p></li>
</ol></div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">estimateJDiv</span><span class="op">(</span></span>
<span>  <span class="va">p</span>,</span>
<span>  logbase <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  stat <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  n <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"single"</span>, <span class="st">"all"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>p</dt>
<dd><p>A numerical vector of the methylation levels \(p = (p1, p2)\)
from individuals 1 and 2.</p></dd>


<dt>logbase</dt>
<dd><p>Logarithm base used to compute the JD. Logarithm base 2 is
used as default. Use logbase = exp(1) for natural logarithm.</p></dd>


<dt>stat</dt>
<dd><p>logical(1). Whether to compute the statistic with asymptotic
Chi-squared distribution with one degree of freedom (details).</p></dd>


<dt>n</dt>
<dd><p>An integer vector \(n = (n_1, n_2)\) carrying the coverage used
in the computation of the methylation levels \(p = (p_1, p_2)\). Default
is NULL. If provided, then the coverages are used to compute J divergence
statistic as reported in reference (3), which follows Chi-squared probablity
distribution.</p></dd>


<dt>output</dt>
<dd><p>Optional. If 'stat = TRUE', whether to return \(JD\) and its
asymptotic chi-squared statistic.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>The J divergence value for the given methylation levels is
returned</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The methylation levels \(p_{ij}\), at a given cytosine site
<em>i</em> from an individual <em>j</em>, lead to the probability vectors
\(p^{ij} = (p_{ij}, 1 - p_{ij})\). Then, the J-information divergence
between the methylation levels \(p_{ij}\) and \(q^i = (q_i, 1 - q_i)\),
used as reference, is given by (1-2):</p>
<p>$$JD(p^{ij}, q^i) = ((p_{ij} - q_i) * log(p_{ij}/q_i) +
                        (q_i - p_{ij}) * log((1 - p_{ij})/(1 - q_i)))/2$$</p>
<p>Missing methylation levels, reported as NA or NaN, are replaced with zero.</p>
<p>The statistic with asymptotic Chi-squared distribution is based on the
statistic suggested by Kupperman (1957) (2) for \(D_{KL}\) and commented in
references (3-4). That is,</p>
<p>$$ 2*(n_{1} + 1)*(n_{2} + 1)*JD(p^{ij}, q^i)/(n_1 + n_2 + 2)$$</p>
<p>Where \(n_{1}\) and \(n_{2}\) are the total counts (coverage in the case
of methylation) used to compute the probabilities \(p_{ij}\) and
\(q_i\). A basic Bayesian correction is added to prevent zero counts.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    
<ol><li><p>Kullback S, Leibler RA. On Information and Sufficiency.
Ann Math Stat. 1951;22: 79–86. doi:10.1214/aoms/1177729694.</p></li>
<li><p>Kupperman, M., 1957. Further application to information theory to
multivariate analysis and statistical inference. Ph.D.
Dissertation, George Washington University.</p></li>
<li><p>Salicrú M, Morales D, Menéndez ML, Pardo L. On the applications of
divergence type measures in testing statistical hypotheses.
Journal of Multivariate Analysis. 1994. pp. 372–391.
doi:10.1006/jmva.1994.1068.</p></li>
<li><p>Basu A, Mandal A, Pardo L. Hypothesis testing for two discrete
populations based on the Hellinger distance. Stat Probab Lett.
Elsevier B.V.; 2010;80: 206–214. doi:10.1016/j.spl.2009.10.008.</p></li>
<li><p>J. K. Chung, P. L. Kannappan, C. T. Ng, P. K. Sahoo, Measures of
distance between probability distributions. J. Math. Anal. Appl.
138, 280–292 (1989).</p></li>
<li><p>Lin J. Divergence Measures Based on the Shannon Entropy. IEEE
Trans Inform Theory, 1991, 37:145–51.</p></li>
<li><p>Sanchez R, Mackenzie SA. Information thermodynamics of cytosine
DNA methylation. PLoS One, 2016, 11:e0150427.</p></li>
</ol></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><a href="https://en.wikipedia.org/wiki/Kullback-Leibler_divergence" class="external-link">https://en.wikipedia.org/wiki/Kullback-Leibler_divergence</a>
for more details, and <code><a href="estimateDivergence.html">estimateDivergence</a></code> for an example of
using it.</p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Robersy Sanchez 11/27/2019 <a href="https://github.com/genomaths" class="external-link">https://github.com/genomaths</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.5</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">estimateJDiv</span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## A numerical trick is implemented. The J-divergence values are the</span></span></span>
<span class="r-in"><span><span class="co">## same for the following vectors:</span></span></span>
<span class="r-in"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.9999999999999999</span>, <span class="fl">0</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">q</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">estimateJDiv</span><span class="op">(</span><span class="va">p</span><span class="op">)</span> <span class="op">==</span> <span class="fu">estimateJDiv</span><span class="op">(</span><span class="va">q</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] FALSE</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

