<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Information divergence estimator — estimateBayesianDivergence • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Information divergence estimator — estimateBayesianDivergence"><meta property="og:description" content="The Information divergence of methylation levels is estimated
using the direct estimation or a Bayesian approach of the methylation levels.
if 'meth.level = FALSE', Hellinger divergence is computed as given in
reference (1). The Bayesian approach followed is described in reference (2)."><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Information divergence estimator</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/estimateBayesianDivergence.R" class="external-link"><code>R/estimateBayesianDivergence.R</code></a></small>
    <div class="hidden name"><code>estimateBayesianDivergence.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The Information divergence of methylation levels is estimated
using the direct estimation or a Bayesian approach of the methylation levels.
if 'meth.level = FALSE', Hellinger divergence is computed as given in
reference (1). The Bayesian approach followed is described in reference (2).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">estimateBayesianDivergence</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  Bayesian <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  init.pars <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  via.optim <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  JD <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  jd.stat <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>mC1 <span class="op">=</span> <span class="fl">1</span>, uC1 <span class="op">=</span> <span class="fl">2</span>, mC2 <span class="op">=</span> <span class="fl">3</span>, uC2 <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>,</span>
<span>  meth.level <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  preserve.gr <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  loss.fun <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"linear"</span>, <span class="st">"huber"</span>, <span class="st">"smooth"</span>, <span class="st">"cauchy"</span>, <span class="st">"arctg"</span><span class="op">)</span>,</span>
<span>  logbase <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>x</dt>
<dd><p>A matrix of counts or GRanges object with the table of counts in
the meta-columns (methylated mC and unmethylated uC cytosines). Unless
specified in the parameter 'columns', the methylation counts must be given in
the first four columns: 'mC1' and 'uC1' methylated and unmethylated counts
for control sample, and 'mC2' and 'uC2' methylated and unmethylated counts
for treatment sample, respectively.</p></dd>


<dt>Bayesian</dt>
<dd><p>logical. Whether to perform the estimations based on
posterior estimations of methylation levels.</p></dd>


<dt>init.pars</dt>
<dd><p>initial parameter values. Defaults is NULL and an initial
guess is estimated using <code><a href="https://rdrr.io/r/stats/optim.html" class="external-link">optim</a></code> function. If the initial
guessing fails initial parameter values are to alpha = 1 &amp;
beta = 1, which imply the parsimony pseudo-counts greater than zero.</p></dd>


<dt>via.optim</dt>
<dd><p>Logical. Whether to estimate beta distribution parameters
via <code><a href="https://rdrr.io/r/stats/optim.html" class="external-link">optim</a></code> or <code><a href="https://rdrr.io/pkg/minpack.lm/man/nls.lm.html" class="external-link">nls.lm</a></code>. If any of
this approaches fail then parameters used <em>init.pars</em> will be returned.</p></dd>


<dt>JD</dt>
<dd><p>Logic (Default:FALSE). Option on whether to add a column with
values of J-information divergence (see <code><a href="estimateJDiv.html">estimateJDiv</a></code>).
It is only compute if JD = TRUE and meth.level = FALSE.</p></dd>


<dt>jd.stat</dt>
<dd><p>logical(1). Whether to compute the \(JD\) statistic with
asymptotic Chi-squared distribution with one degree of freedom (see
<code><a href="estimateJDiv.html">estimateJDiv</a></code>).</p></dd>


<dt>columns</dt>
<dd><p>Vector of integer numbers of the columns where the counts
'mC1', 'uC1', 'mC2', and 'uC2' are in the matrix (default 1 to 4). That is,
the input could have more than 4 columns, but only 4 columns with the counts
are used. if 'meth.level == TRUE', then columns = \(c('p1', 'p2')\).</p></dd>


<dt>meth.level</dt>
<dd><p>logical(1). Methylation levels can be provided in place of
counts.</p></dd>


<dt>preserve.gr</dt>
<dd><p>logical(1). Option of whether to preserve all
the metadata from the original GRanges object.</p></dd>


<dt>loss.fun</dt>
<dd><p>Loss function(s) used in the estimation of the best fitted
model to beta distribution (only applied when Bayesian=TRUE; see
<a href="https://en.wikipedia.org/wiki/Loss_function" class="external-link">(Loss function)</a>). This
fitting uses the approach followed in in the R package
<a href="https://github.com/genomaths/usefr" class="external-link">usefr</a>. After
\(z =  1/2 * sum((f(x) - y)^2)\) we have:</p><ol><li><p>"linear": linear function which gives a standard least squares:
\(loss(z) = z\).</p></li>
<li><p>"huber": Huber loss,
\(loss(z) = ifelse(z \leq 1, z, sqrt(z) -1)\).</p></li>
<li><p>"smooth": Smooth approximation to the sum of residues absolute
values: \(loss(z) = 2*(sqrt(z + 1) - 1)\).</p></li>
<li><p>"cauchy": Cauchy loss: \(loss(z) = log(z + 1)\).</p></li>
<li><p>"arctg": arc-tangent loss function: \(loss(x) = atan(z)\).</p></li>
</ol><p>Loss 'linear' function works well for most of the methylation datasets with
acceptable quality.</p></dd>


<dt>logbase</dt>
<dd><p>Logarithm base used to compute the JD (if JD = TRUE).
Logarithm base 2 is used as default (bit unit). Use logbase = exp(1) for
natural logarithm.</p></dd>


<dt>verbose</dt>
<dd><p>if TRUE, prints the function log to stdout</p></dd>


<dt>...</dt>
<dd><p>Optional parameter values for: maxiter, ftol, ptol, and gradtol
from <code><a href="https://rdrr.io/pkg/minpack.lm/man/nlsLM.html" class="external-link">nlsLM</a></code> and <code><a href="https://rdrr.io/r/stats/nlm.html" class="external-link">nlm</a></code> functions.</p></dd>


<dt>num.cores</dt>
<dd><p>The number of cores to use, i.e. at most how many child
processes will be run simultaneously (see 'bplapply' function from
BiocParallel package).</p></dd>


<dt>tasks</dt>
<dd><p>integer(1). The number of tasks per job. value must be a scalar
integer &gt;= 0L. In this documentation a job is defined as a single call to a
function, such as bplapply, bpmapply etc. A task is the division of the
\(X\) argument into chunks. When tasks == 0 (default), \(X\) is divided
as evenly as possible over the number of workers (see MulticoreParam from
BiocParallel package).</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>The input matrix or GRanges object with the four columns of counts
and additional columns. If Bayessian = TRUE, the results are based on
the posterior estimations of methylation levels. The basic additional columns
are:</p><dl><dt>1)</dt>
<dd><p>The original matrix of methylated \(c_{ij}\) and unmethylated
\(t_{ij}\) read counts from control \(j=1\) and treatment \(j=2\)
samples at positions \(i\).</p></dd>

<dt>2)</dt>
<dd><p>'p1' and 'p2': methylation levels for control and treatment,
respectively. If 'meth.level = FALSE' and 'Bayesian = TRUE' (recommended),
'p1' and 'p2' are estimated following the Bayesian approach described in
reference (1).</p></dd>

<dt>3)</dt>
<dd><p>'bay.TV': total variation TV = p2 - p1</p></dd>

<dt>4)</dt>
<dd><p>'TV': total variation based on simple counts:
\(TV=c_1/(c_1+t_1)-c_2/(c_2+t_2)\), where \(c_i\) and \(t_i\) denote
methylated and unmethylated read counts, respectively.</p></dd>

<dt>5)</dt>
<dd><p>'hdiv': Hellinger divergence. If Bayesian = TRUE, the results are
based on the posterior estimations of methylation levels. if 'meth.level =
FALSE', then 'hdiv' is computed as given in reference (1), otherwise as:
$$hdiv = (sqrt(p_1) - sqrt(p_2))^2 + (sqrt(1 -p_1) - sqrt(1 -
p_2))^2$$</p></dd>


</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>For the current version, the Information divergence of methylation
levels is estimated based on Hellinger divergence. If read counts are
provided, then Hellinger divergence is computed as given in the first formula
from Theorem 1 from reference 1. In the present case,</p>
<p>$$H = 2 (n_1 + 1) (n_2 + 1)*((sqrt(p_1) - sqrt(p_2))^2 +
(sqrt(1-p_2) - sqrt(1-p_2))^2)/(n_1 + n_2 + 2)$$</p>
<p>where \(n_1\) and \(n_2\) are the coverage for the control and
treatment, respectively. Notice that each row from the matrix of counts
correspond to a single cytosine position and has four values corresponding to
'mC1' and 'uC1' (control), and mC2' and 'uC2' for treatment.</p>
<p>If the methylation levels are provided in place of counts, then
Hellinger divergence is computed as:
$$H = (sqrt(p_1) - sqrt(p_2))^2 + (sqrt(1 - p_1) - sqrt(1 - p_2))^2$$</p>
<p>This formula assumes that the probability vectors derived from the
methylation levels \(p_i = c(p_{i2}, 1 - p_{i2})\) (see function
<code><a href="estimateHellingerDiv.html">estimateHellingerDiv</a></code>) are an unbiased estimation of the
expected one. The Bayesian approach followed here is described in reference
(2).</p>
    </div>
    <div id="references">
    <h2>References</h2>
    
<ol><li><p>Basu  A., Mandal  A., Pardo L. Hypothesis testing for two
discrete populations based on the Hellinger distance. Stat. Probab.
Lett. 2010, 80: 206-214.</p></li>
<li><p>Sanchez R, Yang X, Maher T, Mackenzie S. Discrimination of DNA
Methylation Signal from Background Variation for Clinical Diagnostics. Int.
J. Mol Sci, 2019, 20:5343.</p></li>
</ol></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="estimateDivergence.html">estimateDivergence</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Robersy Sanchez (<a href="https://genomaths.com" class="external-link">https://genomaths.com</a>)</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## The read count data are created</span></span></span>
<span class="r-in"><span><span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="st">'chr1'</span>, start <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, end <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,</span></span>
<span class="r-in"><span>                strand <span class="op">=</span> <span class="st">'*'</span>,</span></span>
<span class="r-in"><span>                mC1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/NegBinomial.html" class="external-link">rnbinom</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span>, mu <span class="op">=</span> <span class="fl">4</span>, n <span class="op">=</span> <span class="fl">500</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                uC1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/NegBinomial.html" class="external-link">rnbinom</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span>, mu <span class="op">=</span> <span class="fl">4</span>, n <span class="op">=</span> <span class="fl">500</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                mC2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/NegBinomial.html" class="external-link">rnbinom</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span>, mu <span class="op">=</span> <span class="fl">4</span>, n <span class="op">=</span> <span class="fl">500</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                uC2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/NegBinomial.html" class="external-link">rnbinom</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span>, mu <span class="op">=</span> <span class="fl">4</span>, n <span class="op">=</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu">makeGRangesFromDataFrame</span><span class="op">(</span><span class="va">gr</span>, keep.extra.columns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Estimation of the information divergences</span></span></span>
<span class="r-in"><span><span class="va">hd</span> <span class="op">&lt;-</span> <span class="fu">estimateBayesianDivergence</span><span class="op">(</span><span class="va">gr</span>, JD <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in estimateHellingerDiv(p = y[3:4], n = y[1:2]):</span> *** Vector p has values out of the range [0, 1]</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Keep in mind that Hellinger and J divergences are, in general,</span></span></span>
<span class="r-in"><span><span class="co">## correlated</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html" class="external-link">cor.test</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">hd</span><span class="op">$</span><span class="va">hdiv</span><span class="op">)</span>, y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">hd</span><span class="op">$</span><span class="va">jdiv</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        method <span class="op">=</span> <span class="st">'kendall'</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in cor.test(x = as.numeric(hd$hdiv), y = as.numeric(hd$jdiv), method = "kendall"):</span> object 'hd' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## An example with methylation levels</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">sites</span> <span class="op">=</span> <span class="fl">10</span></span></span>
<span class="r-in"><span><span class="va">dat</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="st">'chr1'</span>, start <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="va">sites</span>, end <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="va">sites</span>,strand <span class="op">=</span> <span class="st">'*'</span>,</span></span>
<span class="r-in"><span>                m1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span>n <span class="op">=</span> <span class="va">sites</span><span class="op">)</span>, m2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span>n <span class="op">=</span> <span class="va">sites</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">dat</span> <span class="op">=</span>  <span class="fu">makeGRangesFromDataFrame</span><span class="op">(</span><span class="va">dat</span>, keep.extra.columns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Transforming the list of data frames into a GRanges object</span></span></span>
<span class="r-in"><span><span class="va">hd</span> <span class="op">&lt;-</span> <span class="fu">estimateBayesianDivergence</span><span class="op">(</span>x <span class="op">=</span> <span class="va">dat</span>, columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>p1 <span class="op">=</span> <span class="fl">1</span>, p2 <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                                meth.level <span class="op">=</span> <span class="cn">TRUE</span>, preserve.gr <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in bplapply(seq_len(nrow(x)), function(i) {    estimateHellingerDiv(p = as.numeric(x[i, ]))}, BPPARAM = bpparam):</span> object 'bpparam' not found</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

