<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Estimate cutpoints to distinguish the treatment methylation signal
from the control — estimateCutPoint • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Estimate cutpoints to distinguish the treatment methylation signal
from the control — estimateCutPoint"><meta property="og:description" content="Given a list of two GRanges objects, control and treatment,
carrying the potential signals (prior classification) from controls and
treatments in terms of an information divergence (given the metacolumns),
the function estimates the cutpoints of the control group versus treatment
group."><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Estimate cutpoints to distinguish the treatment methylation signal
from the control</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/estimateCutPoint.R" class="external-link"><code>R/estimateCutPoint.R</code></a></small>
    <div class="hidden name"><code>estimateCutPoint.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given a list of two GRanges objects, control and treatment,
carrying the potential signals (prior classification) from controls and
treatments in terms of an information divergence (given the metacolumns),
the function estimates the cutpoints of the control group versus treatment
group.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">estimateCutPoint</span><span class="op">(</span></span>
<span>  <span class="va">LR</span>,</span>
<span>  <span class="va">control.names</span>,</span>
<span>  <span class="va">treatment.names</span>,</span>
<span>  simple <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, jdiv <span class="op">=</span> <span class="cn">TRUE</span>, jdiv.stat <span class="op">=</span> <span class="cn">FALSE</span>, TV <span class="op">=</span> <span class="cn">TRUE</span>, bay.TV <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  classifier1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"logistic"</span>, <span class="st">"pca.logistic"</span>, <span class="st">"lda"</span>, <span class="st">"qda"</span>, <span class="st">"pca.lda"</span>, <span class="st">"pca.qda"</span>,</span>
<span>    <span class="st">"random_forest"</span><span class="op">)</span>,</span>
<span>  classifier2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tv.cut <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  tv.col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  div.col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  clas.perf <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  post.cut <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  prop <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>  center <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  n.pc <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  interactions <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cut.values <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  cutp_data <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  maxnodes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ntree <span class="op">=</span> <span class="fl">400</span>,</span>
<span>  nsplit <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>LR</dt>
<dd><p>An object from 'pDMP' class. This object is previously obtained
with function <code><a href="getPotentialDIMP.html">getPotentialDIMP</a></code>.</p></dd>


<dt>control.names, treatment.names</dt>
<dd><p>Names/IDs of the control and
treatment samples, which must be include in the variable LR.</p></dd>


<dt>simple</dt>
<dd><p>Logic (default, TRUE). If TRUE, then Youden Index is used to
estimate the cutpoint. If FALSE, the minimum information divergence value
with posterior classification probability greater than <em>post.cut</em>
(usually <em>post.cut</em> = 0.5) as estimated by <em>classifier1</em> will be
the reported cutpoint, except if a better cutpoint is found in the set of
values provided by the user in the parameter <em>cut.values</em>.</p></dd>


<dt>column</dt>
<dd><p>a logical vector for column names for the predictor variables
to be used: Hellinger divergence 'hdiv', total variation 'TV', probability
of potential DMP 'wprob', and the relative cytosine site position 'pos' in
respect to the chromosome where it is located. The relative position is
estimated as (x - x.min)/(x.max - x), where x.min and x.max are the maximum
and minimum for the corresponding chromosome, respectively. If 'wprob =
TRUE', then Logarithm base-10 of 'wprob' will be used as predictor in place
of 'wprob'.</p></dd>


<dt>classifier1, classifier2</dt>
<dd><p>Classification model to use. Option
'logistic' applies a logistic regression model; option 'lda' applies a
Linear Discriminant Analysis (LDA); 'qda' applies a Quadratic Discriminant
Analysis (QDA), 'pca.logistic' applies logistic regression model using the
Principal Component (PCs) estimated with Principal Component Analysis (PCA)
as predictor variables. pca.lda' applies LDA using PCs as predictor
variables, and the option 'pca.qda' applies a Quadratic Discriminant
Analysis (QDA) using PCs as predictor variables. If classifier2 is not NULL,
then it will be used to evaluate the classification performance, and the
corresponding best fitted model will be returned.</p></dd>


<dt>tv.cut</dt>
<dd><p>A cutoff for the total variation distance to be applied to
each site/range. Only sites/ranges <em>k</em> with \(TVD_{k} &gt; tv.cut\) are
are used in the analysis. Its value must be a number. \(0 &lt; tv.cut &lt; 1\).
Default is \(tv.cut = 0.25\).</p></dd>


<dt>tv.col</dt>
<dd><p>Column number for the total variation to be used for filtering
cytosine positions (if provided).</p></dd>


<dt>div.col</dt>
<dd><p>Column number for divergence variable for which the
estimation of the cutpoint will be performed.</p></dd>


<dt>clas.perf</dt>
<dd><p>Logic. Whether to evaluate the classification performance
for the estimated cutpoint using a model classifier when 'simple=TRUE'.
Default, FALSE.</p></dd>


<dt>post.cut</dt>
<dd><p>If 'simple=FALSE', this is posterior probability to decide
whether a DMPs belong to treatment group. Default <em>post.cut</em> = 0.5.</p></dd>


<dt>prop</dt>
<dd><p>Proportion to split the dataset used in the logistic regression:
group versus divergence (at DMPs) into two subsets, training and testing.</p></dd>


<dt>center</dt>
<dd><p>A logical value indicating whether the variables should be
shifted to be zero centered.</p></dd>


<dt>scale</dt>
<dd><p>A logical value indicating whether the variables should be</p></dd>


<dt>n.pc</dt>
<dd><p>Number of principal components (PCs) to use if the classifier is
not 'logistic'. In the current case, the maximun number of PCs is 4.</p></dd>


<dt>interactions</dt>
<dd><p>If a logistic classifier is used. Variable interactions
to consider in a logistic regression model. Any pairwise combination of the
variable 'hdiv', 'TV', 'wprob', and 'pos' can be provided. For example:
'hdiv:TV', 'wprob:pos', 'wprob:TV', etc.</p></dd>


<dt>cut.values</dt>
<dd><p>Cut values of the information divergence (ID) specified
in <em>div.col</em> where to check the classification performance (0 &lt;
<em>cut.interval</em> &lt; max ID). If provided, the search for a cutpoint will
include these values.</p></dd>


<dt>stat</dt>
<dd><p>An integer number indicating the statistic to be used in the
testing when <em>simple</em> = FALSE The mapping for statistic names are:</p><ul><li><p>0 = 'Accuracy'</p></li>
<li><p>1 = 'Sensitivity'</p></li>
<li><p>2 = 'Specificity'</p></li>
<li><p>3 = 'Pos Pred Value'</p></li>
<li><p>4 = 'Neg Pred Value'</p></li>
<li><p>5 = 'Precision'</p></li>
<li><p>6 = 'Recall'</p></li>
<li><p>7 = 'F1'</p></li>
<li><p>8 = 'Prevalence'</p></li>
<li><p>9 = 'Detection Rate'</p></li>
<li><p>10 = 'Detection Prevalence'</p></li>
<li><p>11 = 'Balanced Accuracy'</p></li>
<li><p>12 = 'FDR'</p></li>
</ul></dd>


<dt>cutp_data</dt>
<dd><p>logical(1) (optional). If TRUE, and simple = TRUE, then a
data frame for further analysis or estimation of the optimal cutpoint
based  only on the selected divergence is provided.</p></dd>


<dt>maxnodes, ntree</dt>
<dd><p>Only for Random Forest classifier
(<code><a href="https://rdrr.io/pkg/randomForest/man/randomForest.html" class="external-link">randomForest</a></code>). Maximum number maxnodes of
terminal nodes trees in the forest can have. If not given, trees are grown
to the maximum possible. Parameter ntree stands for the number of trees to
grow. This should not be set to too small a number, to ensure that every
input row gets predicted at least a few times.</p></dd>


<dt>nsplit</dt>
<dd><p>Only for Random Forest classifier. The Random Forest
(<code><a href="https://rdrr.io/pkg/randomForest/man/randomForest.html" class="external-link">randomForest</a></code>) package uses a C+Fortran
implementation which only supports integer indexes, so any dataframe/data
table/matrix with &gt;2^31 elements (limit for integers) gives an error. The
option nsplit is applied to train \(ntrees=floor(ntree/nsplit)\) models
with rep(ntrees,nsplit) which are finally combined to obtain a forest with
ntree. Each model would contain ntrees.</p></dd>


<dt>num.cores, tasks</dt>
<dd><p>Parameters for parallel computation using package
<code><a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallel-package.html" class="external-link">BiocParallel-package</a></code>: the number of cores to
use, i.e. at most how many child processes will be run simultaneously (see
<code><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></code> and the number of tasks per job (only
for Linux OS).</p></dd>


<dt>...</dt>
<dd><p>arguments passed to or from other methods.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Depending the parameter setting will return the following list with
elements:</p><ol><li><p>cutpoint: Cutpoint estimated.</p></li>
<li><p>testSetPerformance: Performance evaluation on the test set.</p></li>
<li><p>testSetModel.FDR: False discovery rate on the test set.</p></li>
<li><p>model: Model used in the performance evaluation.</p></li>
<li><p>modelConfMatrix: Confusion matrix for the whole dataset
derived applying the model classifier used in the performance
evaluation.</p></li>
<li><p>initModel: Initial classifier model applied to estimate
posterior classifications used in the cutpoint estimation.</p></li>
<li><p>postProbCut: Posterior probability used to estimate the
cutpoint</p></li>
<li><p>classifier: Name of the model classifier used in the
performance evaluation.</p></li>
<li><p>statistic: Name of the performance statistic used to find the
cutpoint when <em>simple</em> =  FALSE.</p></li>
<li><p>optStatVal: Value of the performance statistic at the
cutpoint.</p></li>
</ol></div>
    <div id="details">
    <h2>Details</h2>
    <p>The function performs an estimation of the optimal cutpoint for the
classification of the differentially methylated (cytosines) positions into
two classes: DMPs from control and DMPs from treatment. The simplest approach
to estimate the cutpoint is based on the application of Youden Index. More
complexes approach based in several machine learning model are provided as
well.</p>
<p>Results of the classification performance resulting from the estimated
cutpoint are normally given, with the exception of those extreme situations
where the statistics to evaluate performance cannot be estimated. More than
one classifier model can be applied. For example, one classifier (logistic
model) can be used to estimate the posterior classification probabilities of
DMP into those from control and those from treatment. These probabilities are
then used to estimate the cutpoint in range of values from, say, 0.5 to 0.8.
Next, a different classifier can be used to evaluate the classification
performance. Different classifier models would yield different performances.
Models are returned and can be used in further prediction with new datasets
from the same batch experiment. This is a machine learning approach to
discriminate the biological regulatory signal naturally generated in the
control from that one induced by the treatment.</p>
<p>Notice that the estimation of an optimal cutpoint based on the application
Youden Index (simple = TRUE) only uses the information provided by the
selected information divergence. As a result, classification results based
only in one variable can be poor or can fail. However, option simple =
FALSE, uses the information from several variables following a
machine-learning (ML) approach.</p>
<p>Nevertheless, when simple = TRUE, still a ML model classifier can be built
using the optimal cutpoint estimated and setting clas.perf = TRUE. Such a ML
model can be used for predictions in further analyses with function
<code><a href="predictDIMPclass.html">predictDIMPclass</a></code>.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="evaluateDIMPclass.html">evaluateDIMPclass</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Get a dataset of potential signals and the estimates cutpoint</span></span></span>
<span class="r-in"><span><span class="co">## from the package and performs cutpoint estimation</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PS</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">cutp</span> <span class="op">&lt;-</span> <span class="fu"><a href="mlCutpoint.html">mlCutpoint</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">PS</span>,</span></span>
<span class="r-in"><span>                   column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, TV <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                              wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                   classifier1 <span class="op">=</span> <span class="st">'qda'</span>, n.pc <span class="op">=</span> <span class="fl">4</span>,</span></span>
<span class="r-in"><span>                   control.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'C1'</span>, <span class="st">'C2'</span>, <span class="st">'C3'</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                   treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'T1'</span>, <span class="st">'T2'</span>, <span class="st">'T3'</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                   tv.cut <span class="op">=</span> <span class="fl">0.68</span>, prop <span class="op">=</span> <span class="fl">0.6</span>,</span></span>
<span class="r-in"><span>                   cut.values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">114</span>, <span class="fl">118</span>, <span class="fl">1</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                   div.col <span class="op">=</span> <span class="fl">9L</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found more than one class "pDMP" in cache; using the first, from namespace 'MethylIT'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Also defined by ‘MethylIT.utils’</span>
<span class="r-in"><span><span class="va">cutp</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cutpoint estimation with 'qda' classifier </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cutpoint search performed using model posterior probabilities </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Posterior probability used to get the cutpoint = 0.5 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cytosine sites with treatment PostProbCut &gt;= 0.5 have a </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> divergence value &gt;= 117.0424 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Optimized statistic: Accuracy = 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Cutpoint = 117.04 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Model classifier 'qda' </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> The accessible objects in the output list are: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                    Length Class           Mode     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> cutpoint            1     -none-          numeric  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> testSetPerformance  6     confusionMatrix list     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> testSetModel.FDR    1     -none-          numeric  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> model              13     qdaDMP          list     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> modelConfMatrix     6     confusionMatrix list     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> initModel           1     -none-          character</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> postProbCut         1     -none-          numeric  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> postCut             1     -none-          numeric  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> classifier          1     -none-          character</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> statistic           1     -none-          character</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> optStatVal          1     -none-          numeric  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> cutpData            1     -none-          logical  </span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

