<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Information Divergences of Methylation Levels — estimateDivergence • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Information Divergences of Methylation Levels — estimateDivergence"><meta property="og:description" content="This function prepares the data for the estimation of
information divergences and works as a wrapper calling the functions that
compute selected information divergences of methylation levels. In the
downstream analysis, the probability distribution of a given information
divergence is used in Methyl-IT as the null hypothesis of the noise
distribution, which permits, in a further signal detection step, the
discrimination of the methylation regulatory signal from the background
noise.
For the current version, two information divergences of methylation levels
are computed by default: 1) Hellinger divergence (H) and 2) the total
variation distance (TVD). In the context of methylation analysis
TVD corresponds to the absolute difference of methylation levels.
Here, although the variable reported is the total variation (TV), the
variable actually used for the downstream analysis is TVD. Once a
differentially methylated position (DMP) is identified in the downstream
analysis, TV is the standard indicator of whether the cytosine
position is hyper- or hypo-methylated.
The option to compute the J-information divergence (JD) is currently
provided. The motivation to introduce this divergence is given in the help
of function estimateJDiv."><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Information Divergences of Methylation Levels</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/estimateDivergence.R" class="external-link"><code>R/estimateDivergence.R</code></a></small>
    <div class="hidden name"><code>estimateDivergence.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function prepares the data for the estimation of
information divergences and works as a wrapper calling the functions that
compute selected information divergences of methylation levels. In the
downstream analysis, the probability distribution of a given information
divergence is used in Methyl-IT as the null hypothesis of the noise
distribution, which permits, in a further signal detection step, the
discrimination of the methylation regulatory signal from the background
noise.</p>
<p>For the current version, two information divergences of methylation levels
are computed by default: 1) Hellinger divergence (<em>H</em>) and 2) the total
variation distance (<em>TVD</em>). In the context of methylation analysis
<em>TVD</em> corresponds to the absolute difference of methylation levels.
Here, although the variable reported is the total variation (<em>TV</em>), the
variable actually used for the downstream analysis is <em>TVD</em>. Once a
differentially methylated position (DMP) is identified in the downstream
analysis, <em>TV</em> is the standard indicator of whether the cytosine
position is hyper- or hypo-methylated.</p>
<p>The option to compute the J-information divergence (JD) is currently
provided. The motivation to introduce this divergence is given in the help
of function <code><a href="estimateJDiv.html">estimateJDiv</a></code>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">estimateDivergence</span><span class="op">(</span></span>
<span>  <span class="va">ref</span>,</span>
<span>  <span class="va">indiv</span>,</span>
<span>  Bayesian <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  init.pars <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  via.optim <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>mC <span class="op">=</span> <span class="fl">1</span>, uC <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,</span>
<span>  min.coverage <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  min.meth <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  min.umeth <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  min.sitecov <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  high.coverage <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  percentile <span class="op">=</span> <span class="fl">0.999</span>,</span>
<span>  JD <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  jd.stat <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fu">multicoreWorkers</span><span class="op">(</span><span class="op">)</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  meth.level <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  loss.fun <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"linear"</span>, <span class="st">"huber"</span>, <span class="st">"smooth"</span>, <span class="st">"cauchy"</span>, <span class="st">"arctg"</span><span class="op">)</span>,</span>
<span>  logbase <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>ref</dt>
<dd><p>The GRanges object of the reference individual that will be used
in the estimation of the information divergence.</p></dd>


<dt>indiv</dt>
<dd><p>A list of GRanges objects from the individuals that will be
used in the estimation of the information divergence.</p></dd>


<dt>Bayesian</dt>
<dd><p>logical(1). Whether to perform the estimations based on
posterior estimations of methylation levels.</p></dd>


<dt>init.pars</dt>
<dd><p>initial parameter values. Defaults is NULL and an initial
guess is estimated using <code><a href="https://rdrr.io/r/stats/optim.html" class="external-link">optim</a></code> function. If the initial
guessing fails initial parameter values are to alpha = 1 and
beta = 1, which imply the parsimony pseudo-counts greater than zero.</p></dd>


<dt>via.optim</dt>
<dd><p>Optional. Only used if <em>Bayesian = TRUE</em> Whether to
estimate beta distribution parameters via <code><a href="https://rdrr.io/r/stats/optim.html" class="external-link">optim</a></code> or
<code><a href="https://rdrr.io/pkg/minpack.lm/man/nls.lm.html" class="external-link">nls.lm</a></code>. If any of this approaches fail then
parameters used <em>init.pars</em> will be returned.</p></dd>


<dt>columns</dt>
<dd><p>Vector of one or two integer numbers denoting the indexes of
the columns where the methylated and unmethylated read counts are found or,
if meth.level = TRUE, the columns corresponding to the methylation levels.
If columns = NULL and meth.level = FALSE, then columns = c(1,2) is assumed.
If columns = NULL and meth.level = TRUE, then columns = 1 is assumed.</p></dd>


<dt>min.coverage</dt>
<dd><p>An integer or an integer vector of length 2. Cytosine
sites where the coverage in both samples, 'x' and 'y', are less than
'min.coverage' are discarded. The cytosine site is preserved, however, if
the coverage is greater than 'min.coverage' in at least one sample. If
'min.coverage' is an integer vector, then the corresponding min coverage is
applied to each sample.</p></dd>


<dt>min.meth</dt>
<dd><p>An integer or an integer vector of length 2. Cytosine sites
where the numbers of read counts of methylated cytosine in both samples, '1'
and '2', are less than 'min.meth' are discarded. If 'min.meth' is an integer
vector, then the corresponding min number of reads is applied to each
sample. Default is min.meth = 4.</p></dd>


<dt>min.umeth</dt>
<dd><p>An integer or an integer vector of length 2
(\(min.umeth = c(min.umeth1, min.umeth2)\)). Min number of reads to
consider cytosine position. Specifically cytosine positions where \((uC
\leq min.umeth) and (mC &gt; 0) and (mC \leq min.meth)\) hold will be removed,
where mC and uC stand for the numbers of methylated and unmethylated reads.
Default is \(min.umeth = 0\).</p></dd>


<dt>min.sitecov</dt>
<dd><p>An integer. The minimum total coverage. Only sites where
the total coverage \((cov1 + cov2)\) is greater than 'min.sitecov' are
considered for downstream analysis, where cov1 and cov2 are the coverages
for samples 1 and 2, respectively.</p></dd>


<dt>high.coverage</dt>
<dd><p>An integer for read counts. Cytosine sites having
higher coverage than this are discarded.</p></dd>


<dt>percentile</dt>
<dd><p>Threshold to remove the outliers from each file and all
files stacked.</p></dd>


<dt>JD</dt>
<dd><p>Logic (Default:FALSE). Option on whether to add a column with
values of J-information divergence (see <code><a href="estimateJDiv.html">estimateJDiv</a></code>). It is
only compute if JD = TRUE and meth.level = FALSE.</p></dd>


<dt>jd.stat</dt>
<dd><p>logical(1). Whether to compute the \(JD\) statistic with
asymptotic Chi-squared distribution with one degree of freedom (see
<code><a href="estimateJDiv.html">estimateJDiv</a></code>).</p></dd>


<dt>ignore.strand</dt>
<dd><p>When set to TRUE, the strand information is ignored in
the overlap of <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> objects. This is a
parameter passed to <code><a href="uniqueGRanges.html">uniqueGRanges</a></code> function. Default value:
FALSE.</p></dd>


<dt>num.cores</dt>
<dd><p>The number of cores to use, i.e. at most how many child
processes will be run simultaneously (see 'bplapply' function from
BiocParallel package).</p></dd>


<dt>tasks</dt>
<dd><p>integer(1). The number of tasks per job. value must be a scalar
\(integer &gt;= 0L\). In this documentation a job is defined as a single call
to a function, such as bplapply, bpmapply etc. A task is the division of the
X argument into chunks. When tasks == 0 (default), X is divided as evenly as
possible over the number of workers (see MulticoreParam from BiocParallel
package).</p></dd>


<dt>meth.level</dt>
<dd><p>logical(1) Whether methylation levels are given in place
of counts. Default is FALSE.</p></dd>


<dt>loss.fun</dt>
<dd><p>Loss function(s) used in the estimation of the best fitted
model to beta distribution (only applied when Bayesian=TRUE; see
<a href="https://en.wikipedia.org/wiki/Loss_function" class="external-link">(Loss function)</a>). This
fitting uses the approach followed in in the R package
<a href="https://github.com/genomaths/usefr" class="external-link">usefr</a>. After
\(z =  1/2 * sum((f(x) - y)^2)\) we have:</p><ol><li><p>"linear": linear function which gives a standard least squares:
\(loss(z) = z\).</p></li>
<li><p>"huber": Huber loss,
\(loss(z) = ifelse(z \leq 1, z, sqrt(z) -1)\).</p></li>
<li><p>"smooth": Smooth approximation to the sum of residues absolute
values: \(loss(z) = 2*(sqrt(z + 1) - 1)\).</p></li>
<li><p>"cauchy": Cauchy loss: \(loss(z) = log(z + 1)\).</p></li>
<li><p>"arctg": arc-tangent loss function: \(loss(x) = atan(z)\).</p></li>
</ol><p>Loss 'linear' function works well for most of the methylation datasets with
acceptable quality.</p></dd>


<dt>logbase</dt>
<dd><p>Logarithm base used to compute the JD (if JD = TRUE).
Logarithm base 2 is used as default (bit unit). Use logbase = \(exp(1)\)
for natural logarithm.</p></dd>


<dt>verbose</dt>
<dd><p>if TRUE, prints the function log to stdout</p></dd>


<dt>...</dt>
<dd><p>Optional parameters for <code><a href="uniqueGRanges.html">uniqueGRanges</a></code> function.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>An object from 'infDiv' class with the four columns of counts, the
information divergence, and additional columns:</p><dl><dt>1) <strong>A matrix:</strong></dt>
<dd><p>The original matrix of methylated
\(c_{ij}\) and unmethylated \(t_{ij}\) read counts from control
\(j=1\) and treatment \(j=2\) samples at positions \(i\).</p></dd>


<dt>2) <strong>'p1' and 'p2':</strong></dt>
<dd><p>methylation levels for control and
treatment, respectively. If 'meth.level = FALSE' and 'Bayesian = TRUE'
(recommended), 'p1' and 'p2' are estimated following the Bayesian approach
described in reference (1).</p></dd>


<dt>3) <strong>'bay.TV':</strong></dt>
<dd><p>total variation TV = p2 - p1</p></dd>


<dt>4) <strong>'TV':</strong></dt>
<dd><p>total variation based on simple counts:
\(TV=c_1/(c_1+t_1)-c_2/(c_2+t_2)\), where \(c_i\) and \(t_i\) denote
methylated and unmethylated read counts, respectively.</p></dd>


<dt>5) <strong>Hellinger divergence, 'hdiv':</strong></dt>
<dd><p>If Bayesian = TRUE, the
results are based on the posterior estimations of methylation levels. if
meth.level = FALSE', then 'hdiv' is computed as given in reference (2),
otherwise as: $$hdiv = (sqrt(p_1) - sqrt(p_2))^2 + (sqrt(1 -p_1) -
sqrt(1 - p_2))^2$$</p></dd>


</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>If read counts are provided, then Hellinger divergence is computed
as given in the first formula from Theorem 1 from reference (1). In the
present case:</p>
<p>$$H = 2 (n_1 + 1) (n_2 + 1)*((\sqrt(p_1) - \sqrt(p_2))^2 +
(\sqrt(1-p_2) - \sqrt(1-p_2))^2)/(n_1 + n_2 + 2)$$</p>
<p>where \(n_1\) and \(n_2\) are the coverage for the control and
treatment, respectively. Notice that each row from the matrix of counts
correspond to a single cytosine position and has four values corresponding
to 'mC1' and 'uC1' (control), and mC2' and 'uC2' for treatment.</p>
<p>According with the above equation, to estimate Hellinger divergence, not
only the methylation levels are considered in the estimation of H, but also
the control and treatment coverage at each given cytosine site. At this
point, it is worthy to do mention that if the reference sample is derived
with function <code><a href="poolFromGRlist.html">poolFromGRlist</a></code> using the 'sum' of read counts to
compute a methylation pool, then 'min.coverage' parameter value must be used
to prevent an over estimation of the divergence for low coverage cytosines
sites. For example, if a reference sample is derived as the methylation pool
of read count sum from 3 individuals and we want to consider only
methylation sites with minimum coverage of 4, then we can set min.coverage =
c(12, 4), where the number 12 (3 x 4) is the minimum coverage requested for
the each cytosine site in the reference sample.</p>
<p>If the methylation levels are provided in place of counts, then the
Hellinger divergence is computed as:
$$H = (\sqrt(p_1) - \sqrt(p_2))^2 + (\sqrt(1 - p_1) - \sqrt(1 - p_2))^2$$</p>
<p>This formula assumes that the probability vectors derived from the
methylation levels  \(p_i = c(p_{i1}, 1 - p_{i2}\)) (see
<code><a href="estimateHellingerDiv.html">estimateHellingerDiv</a></code> are an unbiased estimation of the
expected one. The function applies a pairwise filtering after building a
single GRanges from the two GRanges objects. Experimentally available
cytosine sites are paired using the function <code><a href="uniqueGRanges.html">uniqueGRanges</a></code>.</p>
<p>It is important to observe that several filtering conditions are provided to
select biological meaningful cytosine positions, which prevent to carry
experimental errors in the downstream analyses. By filtering the read count
we try to remove bad quality data, which would be in the edge of the
experimental error originated by the BS-seq sequencing. It is user
responsibility to check whether cytosine positions used in the analysis are
biological meaningful. For example, a cytosine position with counts mC1 = 10
and uC1 = 20 in the 'ref' sample and mC2 = 1 and uC2 = 0 in an 'indv' sample
will lead to methylation levels p1 = 0.333 and p2 = 1, respectively, and TV
= p2 - p1 = 0.667, which apparently indicates a hypermethylated site.
However, there are not enough reads supporting p2 = 1. A Bayesian estimation
of TV will reveal that this site would be, in fact, hypomethylated. So, the
best practice will be the removing of sites like that. This particular case
is removed under the default settings: min.coverage = 4, min.meth = 4, and
min.umeth = 0 (see example for function <code><a href="uniqueGRfilterByCov.html">uniqueGRfilterByCov</a></code>,
called by 'estimateDivergence').</p>
    </div>
    <div id="references">
    <h2>References</h2>
    
<ol><li><p>Sanchez R, Yang X, Maher T, Mackenzie S. Discrimination of DNA
Methylation Signal from Background Variation for Clinical Diagnostics. Int.
J. Mol Sci, 2019, 20:5343.</p></li>
<li><p>Basu  A., Mandal  A., Pardo L. Hypothesis testing for two
discrete populations based on the Hellinger distance. Stat. Probab.
Lett. 2010, 80: 206-214.</p></li>
</ol></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="estimateBayesianDivergence.html">estimateBayesianDivergence</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Robersy Sanchez (<a href="https://genomaths.com" class="external-link">https://genomaths.com</a>)</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Load a dataset of simulated read counts.</span></span></span>
<span class="r-in"><span><span class="co">## This is a list of GRanges objects</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"rcounts"</span>, package <span class="op">=</span> <span class="st">"MethylIT"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The first GRanges object is the referenc sample</span></span></span>
<span class="r-in"><span><span class="va">rcounts</span><span class="op">$</span><span class="va">ref</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 10000 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           seqnames    ranges strand |        mC        uC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [1]        1         1      + |        18        46</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [2]        1         2      + |         2       140</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [3]        1         3      - |         4        24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [4]        1         4      + |        41        55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [5]        1         5      + |        36       145</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       ...      ...       ...    ... .       ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [9996]        1      9996      - |        49        49</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [9997]        1      9997      + |        24        95</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [9998]        1      9998      + |        18        47</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [9999]        1      9999      - |        32        95</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [10000]        1     10000      - |        28       116</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## For the safe of time, we will use just the first 500 cytosine positions</span></span></span>
<span class="r-in"><span><span class="va">rc</span> <span class="op">&lt;-</span> <span class="fu"><a href="lapply.html">lapply</a></span><span class="op">(</span><span class="va">rcounts</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span> <span class="fl">1</span><span class="op">:</span><span class="fl">500</span> <span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">rc</span><span class="op">$</span><span class="va">C1</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 500 ranges and 2 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames    ranges strand |        mC        uC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]        1         1      + |        57         7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]        1         2      + |         0       142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]        1         3      - |         0        28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]        1         4      + |        64        32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]        1         5      + |         0       181</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...       ...    ... .       ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [496]        1       496      + |       123        20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [497]        1       497      - |        53         8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [498]        1       498      + |         0       129</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [499]        1       499      + |         0        28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [500]        1       500      - |         6       135</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The are three control and three treat samples</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">rc</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "ref" "C1"  "C2"  "C3"  "T1"  "T2"  "T3" </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The estimation of information divergences includes both available</span></span></span>
<span class="r-in"><span><span class="co">## divergences: Hellinger and J divergences.</span></span></span>
<span class="r-in"><span><span class="va">hd</span> <span class="op">&lt;-</span> <span class="fu">estimateDivergence</span><span class="op">(</span></span></span>
<span class="r-in"><span>    ref <span class="op">=</span> <span class="va">rc</span><span class="op">$</span><span class="va">ref</span>,</span></span>
<span class="r-in"><span>    indiv <span class="op">=</span> <span class="va">rc</span><span class="op">[</span> <span class="op">-</span><span class="fl">1</span> <span class="op">]</span>,</span></span>
<span class="r-in"><span>    Bayesian <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    num.cores <span class="op">=</span> <span class="fl">6L</span>,</span></span>
<span class="r-in"><span>    percentile <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>    JD <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">hd</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> InfDiv object of length: 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $C1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 323 ranges and 10 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames    ranges strand |        c1        t1        c2        t2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]        1         1      + |        18        46        57         7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]        1         3      - |         4        24         0        28</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]        1         4      + |        41        55        64        32</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]        1         5      + |        36       145         0       181</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]        1         6      - |        16        97        64        49</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...       ...    ... .       ...       ...       ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [319]        1       495      + |        16        78         0        94</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [320]        1       496      + |        31       112       123        20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [321]        1       497      - |        13        48        53         8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [322]        1       498      + |        57        72         0       129</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [323]        1       500      - |         2       140         6       135</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                p1         p2         TV     bay.TV      hdiv       jdiv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         &lt;numeric&gt;  &lt;numeric&gt;  &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]  0.272555 0.86505099   0.609375   0.592496  26.14515   1.213627</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]  0.161637 0.01061095  -0.142857  -0.151026   2.77443   0.314745</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]  0.407564 0.65457939   0.239583   0.247015   6.03698   0.180549</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]  0.199746 0.00175225  -0.198895  -0.197994  31.85233   0.708001</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]  0.147470 0.55803651   0.424779   0.410567  22.64077   0.588705</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...       ...        ...        ...        ...       ...        ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [319] 0.1744742 0.00333588 -0.1702128 -0.1711383 13.073338 0.51175476</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [320] 0.2167029 0.84894900  0.6433566  0.6322461 65.407536 1.37339109</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [321] 0.2134473 0.84284463  0.6557377  0.6293973 27.809084 1.35469532</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [322] 0.4259443 0.00244640 -0.4418605 -0.4234979 54.855382 1.74503697</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [323] 0.0270367 0.04412242  0.0284687  0.0170857  0.307412 0.00625465</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;5 more GRanges elements&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Keep in mind that Hellinger and J divergences are, in general,</span></span></span>
<span class="r-in"><span><span class="co">## correlated</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html" class="external-link">cor.test</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">hd</span><span class="op">$</span><span class="va">T1</span><span class="op">$</span><span class="va">hdiv</span>, y <span class="op">=</span> <span class="va">hd</span><span class="op">$</span><span class="va">T1</span><span class="op">$</span><span class="va">jdiv</span>, method <span class="op">=</span> <span class="st">'kendall'</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Kendall's rank correlation tau</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> data:  hd$T1$hdiv and hd$T1$jdiv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> z = 23.802, p-value &lt; 2.2e-16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> alternative hypothesis: true tau is not equal to 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sample estimates:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       tau </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 0.7979042 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html" class="external-link">cor.test</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">hd</span><span class="op">$</span><span class="va">C1</span><span class="op">$</span><span class="va">hdiv</span>, y <span class="op">=</span> <span class="va">hd</span><span class="op">$</span><span class="va">C1</span><span class="op">$</span><span class="va">jdiv</span>, method <span class="op">=</span> <span class="st">'pearson'</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Pearson's product-moment correlation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> data:  hd$C1$hdiv and hd$C1$jdiv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> t = 34.446, df = 321, p-value &lt; 2.2e-16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> alternative hypothesis: true correlation is not equal to 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 95 percent confidence interval:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  0.8614380 0.9083527</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sample estimates:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       cor </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 0.8871664 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

