<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Nonlinear fit of Log-Normal CDF (LogNorm) — fitLogNormDist • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Nonlinear fit of Log-Normal CDF (LogNorm) — fitLogNormDist"><meta property="og:description" content="This function performs the nonlinear fit of GGamma CDF of a
variable x"><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Nonlinear fit of Log-Normal CDF (LogNorm)</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/fitLogNormDist.R" class="external-link"><code>R/fitLogNormDist.R</code></a></small>
    <div class="hidden name"><code>fitLogNormDist.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function performs the nonlinear fit of GGamma CDF of a
variable x</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">fitLogNormDist</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">probability.x</span>,</span>
<span>  <span class="va">parameter.values</span>,</span>
<span>  summarized.data <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  sample.size <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  npoints <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  maxiter <span class="op">=</span> <span class="fl">1024</span>,</span>
<span>  ftol <span class="op">=</span> <span class="fl">1e-12</span>,</span>
<span>  ptol <span class="op">=</span> <span class="fl">1e-12</span>,</span>
<span>  maxfev <span class="op">=</span> <span class="fl">1e+05</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>x</dt>
<dd><p>numerical vector</p></dd>


<dt>probability.x</dt>
<dd><p>probability vector of x. If not provided, the values
are estimated using the empirical cumulative distribution function
('ecdf') from 'stats' R package.</p></dd>


<dt>parameter.values</dt>
<dd><p>initial parameter values for the nonlinear fit. If
the locator parameter is included (mu != 0), this must be given as
parameter.values = list(alpha = 'value', scale = 'value', mu = 'value')
or if mu = 0, as: parameter.values = list(alpha = 'value',
scale = 'value'). If not provided, then an initial guess is provided.</p></dd>


<dt>summarized.data</dt>
<dd><p>Logic value. If TRUE (default: FALSE), summarized
data based on 'npoints' are used to perform the nonlinear fit.</p></dd>


<dt>sample.size</dt>
<dd><p>size of the sample.</p></dd>


<dt>npoints</dt>
<dd><p>number of points used in the fit.</p></dd>


<dt>maxiter</dt>
<dd><p>positive integer. Termination occurs when the number of
iterations reaches maxiter. Default value: 1024.</p></dd>


<dt>ftol</dt>
<dd><p>non-negative numeric. Termination occurs when both the actual
and predicted relative reductions in the sum of squares are at most ftol.
Therefore, ftol measures the relative error desired in the sum of
squares. Default value: 1e-12</p></dd>


<dt>ptol</dt>
<dd><p>non-negative numeric. Termination occurs when the relative error
between two consecutive iterates is at most ptol. Therefore, ptol
measures the relative error desired in the approximate solution.
Default value: 1e-12.</p></dd>


<dt>maxfev</dt>
<dd><p>integer; termination occurs when the number of calls to fn has
reached maxfev. Note that nls.lm sets the value of maxfev to
100*(length(par) + 1) if maxfev = integer(), where par is the list or
vector of parameters to be optimized.</p></dd>


<dt>verbose</dt>
<dd><p>if TRUE, prints the function log to stdout</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Model table with coefficients and goodness-of-fit results:
Adj.R.Square, deviance, AIC, R.Cross.val, and rho, as well as, the
coefficient covariance matrix.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The algorithm tries to fit the two-parameter LogNorm CDF
using a modification of Levenberg-Marquardt algorithm implemented in function
'nls.lm' from 'minpack.lm' package that is used to perform the nonlinear fit.
Cross-validations for the nonlinear regressions (R.Cross.val) were performed
in each methylome as described in reference (1). In addition, Stein's formula
for adjusted R squared (rho) was used as an estimator of the average
cross-validation predictive power (1).</p>
<p>If the number of values to fit is &gt;10^6, the fitting to a LogNorm CDF would
be a time consuming task. To reduce the computational time, the option
'summarized.data' can be set 'TRUE'. If summarized.data = TRUE, the original
variable values are summarized into 'npoint' bins and their midpoints are
used as the new predictors. In this case, only the goodness-of-fit indicators
AIC and R.Cross.val are estimated based on all the original variable x
values.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    
<ol><li><p>Stevens JP. Applied Multivariate Statistics for the Social
Sciences. Fifth Edit. Routledge Academic; 2009.</p></li>
</ol></div>
    <div id="author">
    <h2>Author</h2>
    <p>Robersy Sanchez - 04/09/2019</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">126</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Lognormal.html" class="external-link">rlnorm</a></span><span class="op">(</span><span class="fl">1000</span>, meanlog <span class="op">=</span> <span class="fl">1.03</span>, sdlog <span class="op">=</span> <span class="fl">2.1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">fitLogNormDist</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> *** Trying nonlinear fit to a Log-Normal Dist.distribution model ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *** Performing nonlinear regression model  crossvalidation...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         Estimate   Std. Error  t value Pr(&gt;|t|))      Adj.R.Square</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> meanlog 1.013025 0.0007225363 1402.041         0 0.999999988821257</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sdlog   1.997010 0.0012082324 1652.836         0                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       rho       R.Cross.val                  DEV</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> meanlog 0.999999988798855 0.999886282714612 9.30628464170656e-07</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sdlog                                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       AIC               BIC     COV.mean       COV.sd  df</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> meanlog -7979.03725044855 -7959.40622933263 5.220586e-07 1.296256e-08 998</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sdlog                                       1.296256e-08 1.459826e-06 998</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

