<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Count DMPs at gene-body — getDIMPatGenes • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Count DMPs at gene-body — getDIMPatGenes"><meta property="og:description" content="The function counts DMPs overlapping with gene-body. In fact,
this function also can be used to count DMPs overlapping with any set of
regions given in a GRanges object."><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Count DMPs at gene-body</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/getDIMPatGenes.R" class="external-link"><code>R/getDIMPatGenes.R</code></a></small>
    <div class="hidden name"><code>getDIMPatGenes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The function counts DMPs overlapping with gene-body. In fact,
this function also can be used to count DMPs overlapping with any set of
regions given in a GRanges object.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getDIMPatGenes</span><span class="op">(</span></span>
<span>  <span class="va">GR</span>,</span>
<span>  <span class="va">GENES</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"within"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  only.hypo <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  only.hyper <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"list"</span>, <span class="st">"GRanges"</span><span class="op">)</span>,</span>
<span>  by.coord <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  gene_id_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_name_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for default</span></span>
<span><span class="fu">getDIMPatGenes</span><span class="op">(</span></span>
<span>  <span class="va">GR</span>,</span>
<span>  <span class="va">GENES</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"within"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  only.hypo <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  only.hyper <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  output <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  by.coord <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  gene_id_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_name_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for GRanges</span></span>
<span><span class="fu">getDIMPatGenes</span><span class="op">(</span></span>
<span>  <span class="va">GR</span>,</span>
<span>  <span class="va">GENES</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"within"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  only.hypo <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  only.hyper <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  output <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  by.coord <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  gene_id_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_name_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for pDMP</span></span>
<span><span class="fu">getDIMPatGenes</span><span class="op">(</span></span>
<span>  <span class="va">GR</span>,</span>
<span>  <span class="va">GENES</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"within"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  only.hypo <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  only.hyper <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"list"</span>, <span class="st">"GRanges"</span><span class="op">)</span>,</span>
<span>  by.coord <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  gene_id_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_name_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for InfDiv</span></span>
<span><span class="fu">getDIMPatGenes</span><span class="op">(</span></span>
<span>  <span class="va">GR</span>,</span>
<span>  <span class="va">GENES</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"within"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  only.hypo <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  only.hyper <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"list"</span>, <span class="st">"GRanges"</span><span class="op">)</span>,</span>
<span>  by.coord <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  gene_id_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_name_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for list</span></span>
<span><span class="fu">getDIMPatGenes</span><span class="op">(</span></span>
<span>  <span class="va">GR</span>,</span>
<span>  <span class="va">GENES</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"within"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  only.hypo <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  only.hyper <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"list"</span>, <span class="st">"GRanges"</span><span class="op">)</span>,</span>
<span>  by.coord <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  gene_id_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_name_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>GR</dt>
<dd><p>An objects object from the any of the classes: 'pDMP', 'InfDiv',
GRangesList, GRanges or a list of GRanges.</p></dd>


<dt>GENES</dt>
<dd><p>A GRanges object with gene coordinates and gene IDs. A
column named <strong>'gene_id'</strong> carrying the gene ids should be included in
the metacolumns. If the meta-column named 'gene_id' is not provided, then
gene (region) ids will be created using the gene (region) coordinates.</p></dd>


<dt>ignore.strand, type</dt>
<dd><p>Same as for
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps-methods</a></code>.</p></dd>


<dt>only.hypo, only.hyper</dt>
<dd><p>logical(1). Whether to select only
hypo-methylated or hyper-methylated cytosine sites.</p></dd>


<dt>output</dt>
<dd><p>Class of the object to be returned, a "list", or a "GRanges"
object.</p></dd>


<dt>by.coord</dt>
<dd><p>logical(1). If TRUE, then the DMP are count per coordinate
and not per gene id.</p></dd>


<dt>gene_id_col</dt>
<dd><p>Optional. An integer denoting the column from the
<em>GENES</em> metacolumn where the gene ids are given.</p></dd>


<dt>gene_name_col</dt>
<dd><p>Optional. An integer denoting the column from the
<em>GENES</em> metacolumn where the gene 'name' are given.</p></dd>


<dt>...</dt>
<dd><p>optional arguments for
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps-methods</a></code>. Users must evaluate
whether specific setting makes sense on each particular context.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A a list GRanges object.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>If <strong><em>by.coord == FALSE</em></strong> and "gene_id" is provided
in <strong><em>GENES</em></strong> argument, then DMP counts are made per gene-id.
Hence, DMPs from different regions with the same gene-id, say e.g. exons,
will be pooled in the count as they bear the same id.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="getDMPatRegions.html">getDMPatRegions</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Gene annotation</span></span></span>
<span class="r-in"><span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span>seqnames <span class="op">=</span> <span class="st">'1'</span>,</span></span>
<span class="r-in"><span>ranges <span class="op">=</span> <span class="fu">IRanges</span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3631</span>, <span class="fl">6788</span>, <span class="fl">11649</span><span class="op">)</span>, end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5899</span>, <span class="fl">9130</span>, <span class="fl">13714</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'+'</span>, <span class="st">'-'</span>, <span class="st">'-'</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">mcols</span><span class="op">(</span><span class="va">genes</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'AT1G01010'</span>, <span class="st">'AT1G01020'</span>,</span></span>
<span class="r-in"><span><span class="st">'AT1G01030'</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Get a dataset of potential signals and the estimated cutpoint from the</span></span></span>
<span class="r-in"><span><span class="co">## package</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PS</span>, <span class="va">cutpoint</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The estimated cutpoints are used to discriminate signals from the noise.</span></span></span>
<span class="r-in"><span><span class="co">## That is, DMPs are selected using the cupoints</span></span></span>
<span class="r-in"><span><span class="va">DIMPs</span> <span class="op">&lt;-</span> <span class="fu"><a href="selectDIMP.html">selectDIMP</a></span><span class="op">(</span><span class="va">PS</span>, div.col <span class="op">=</span> <span class="fl">9L</span>, cutpoint <span class="op">=</span> <span class="va">cutpoint</span><span class="op">$</span><span class="va">cutpoint</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Finally DMPs found on genes</span></span></span>
<span class="r-in"><span><span class="va">DIMR</span> <span class="op">&lt;-</span> <span class="fu">getDIMPatGenes</span><span class="op">(</span>GR <span class="op">=</span> <span class="va">DIMPs</span><span class="op">$</span><span class="va">T1</span>, GENES <span class="op">=</span> <span class="va">genes</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

