<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Statistic of Genomic Regions — getGRangesStat • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Statistic of Genomic Regions — getGRangesStat"><meta property="og:description" content="A function to estimate the summarized measures of a specified
variable given in a GRanges object (a column from the metacolums of the
GRanges object) after split the GRanges object into intervals."><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Statistic of Genomic Regions</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/getGRangesStat.R" class="external-link"><code>R/getGRangesStat.R</code></a></small>
    <div class="hidden name"><code>getGRangesStat.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A function to estimate the summarized measures of a specified
variable given in a GRanges object (a column from the metacolums of the
GRanges object) after split the GRanges object into intervals.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getGRangesStat</span><span class="op">(</span></span>
<span>  <span class="va">GR</span>,</span>
<span>  win.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  step.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  grfeatures <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sum"</span>, <span class="st">"mean"</span>, <span class="st">"gmean"</span>, <span class="st">"median"</span>, <span class="st">"density"</span>, <span class="st">"count"</span>, <span class="st">"denCount"</span><span class="op">)</span>,</span>
<span>  stat2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat3 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  column <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  absolute <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  absolute2 <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  absolute3 <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  select.strand <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  maxgap <span class="op">=</span> <span class="op">-</span><span class="fl">1L</span>,</span>
<span>  minoverlap <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  select <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"within"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"equal"</span>, <span class="st">"any"</span><span class="op">)</span>,</span>
<span>  scaling <span class="op">=</span> <span class="fl">1000L</span>,</span>
<span>  logbase <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  missings <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  naming <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for pDMP</span></span>
<span><span class="fu">getGRangesStat</span><span class="op">(</span></span>
<span>  <span class="va">GR</span>,</span>
<span>  win.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  step.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  grfeatures <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sum"</span>, <span class="st">"mean"</span>, <span class="st">"gmean"</span>, <span class="st">"median"</span>, <span class="st">"density"</span>, <span class="st">"count"</span>, <span class="st">"denCount"</span><span class="op">)</span>,</span>
<span>  stat2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat3 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  column <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  absolute <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  absolute2 <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  absolute3 <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  select.strand <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  maxgap <span class="op">=</span> <span class="op">-</span><span class="fl">1L</span>,</span>
<span>  minoverlap <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  select <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"within"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"equal"</span>, <span class="st">"any"</span><span class="op">)</span>,</span>
<span>  scaling <span class="op">=</span> <span class="fl">1000L</span>,</span>
<span>  logbase <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  missings <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  naming <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for InfDiv</span></span>
<span><span class="fu">getGRangesStat</span><span class="op">(</span></span>
<span>  <span class="va">GR</span>,</span>
<span>  win.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  step.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  grfeatures <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sum"</span>, <span class="st">"mean"</span>, <span class="st">"gmean"</span>, <span class="st">"median"</span>, <span class="st">"density"</span>, <span class="st">"count"</span>, <span class="st">"denCount"</span><span class="op">)</span>,</span>
<span>  stat2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat3 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  column <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  absolute <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  absolute2 <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  absolute3 <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  select.strand <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  maxgap <span class="op">=</span> <span class="op">-</span><span class="fl">1L</span>,</span>
<span>  minoverlap <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  select <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"within"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"equal"</span>, <span class="st">"any"</span><span class="op">)</span>,</span>
<span>  scaling <span class="op">=</span> <span class="fl">1000L</span>,</span>
<span>  logbase <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  missings <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  naming <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for list</span></span>
<span><span class="fu">getGRangesStat</span><span class="op">(</span></span>
<span>  <span class="va">GR</span>,</span>
<span>  win.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  step.size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  grfeatures <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sum"</span>, <span class="st">"mean"</span>, <span class="st">"gmean"</span>, <span class="st">"median"</span>, <span class="st">"density"</span>, <span class="st">"count"</span>, <span class="st">"denCount"</span><span class="op">)</span>,</span>
<span>  stat2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat3 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  column <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  absolute <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  absolute2 <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  absolute3 <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  select.strand <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  maxgap <span class="op">=</span> <span class="op">-</span><span class="fl">1L</span>,</span>
<span>  minoverlap <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  select <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"within"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"equal"</span>, <span class="st">"any"</span><span class="op">)</span>,</span>
<span>  scaling <span class="op">=</span> <span class="fl">1000L</span>,</span>
<span>  logbase <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  missings <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  naming <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>GR</dt>
<dd><p>A <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> object or a
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList-class</a></code> object carrying the
variables of interest in the GRanges metacolumn(s).</p></dd>


<dt>win.size</dt>
<dd><p>An integer for the size of the windows/regions size of the
intervals of genomics regions.</p></dd>


<dt>step.size</dt>
<dd><p>Interval at which the regions/windows must be defined</p></dd>


<dt>grfeatures</dt>
<dd><p>A GRanges object corresponding to an annotated genomic
feature. For example, gene region, transposable elements, exons, intergenic
region, etc. If provided, then parameters 'win.size' and step.size are
ignored and the statistics are estimated for 'grfeatures'.</p></dd>


<dt>stat</dt>
<dd><p>Statistic used to estimate the summarized value of the variable
of interest in each interval/window. Posible options are:</p><dl><dt><strong>'mean':</strong></dt>
<dd><p>The mean of values inside each region.</p></dd>

<dt><strong>'gmean':</strong></dt>
<dd><p>The geometric mean of values inside each region.</p></dd>

<dt><strong>'median':</strong></dt>
<dd><p>The median of values inside each region.</p></dd>

<dt><strong>'density':</strong></dt>
<dd><p>The density of values inside each region. That
is, the sum of values found in each region divided by the width of
the region.</p></dd>

<dt><strong>'count':</strong></dt>
<dd><p>Compute the number/count of positions with values
greater than zero inside each regions.</p></dd>

<dt><strong>'denCount':</strong></dt>
<dd><p>The number of sites with value &gt; 0 inside each
region divided by the width of the region.</p></dd>

<dt><strong>'sum':</strong></dt>
<dd><p>The sum of values inside each region.</p></dd>


</dl><p>If <strong>GR</strong> have zero metacolum, then it is set <em>stat = "count"</em> and
all the sites are included in the computation.</p></dd>


<dt>stat2, stat3</dt>
<dd><p>The same as for 'stat' argument. If provided, the
statistic selected in 'stat2' and stat3 will be also reported.</p></dd>


<dt>column</dt>
<dd><p>Integer number denoting the column where the variable of
interest is located in the metacolumn of the GRanges object. Default is 1L
if the number of columns is greater than 1, otherwise NULL.</p></dd>


<dt>absolute</dt>
<dd><p>Optional. Logic (default: FALSE). Whether to use the
absolute values of the variable provided. For example, the difference of
methylation levels could take negative values (TV) and we would be
interested on the sum of abs(TV), which is sum of the total variation
distance.</p></dd>


<dt>absolute2, absolute3</dt>
<dd><p>The same as for 'absolute' argument, but applied
when 'stat2' and 'stat3' are not null, respectively.</p></dd>


<dt>select.strand</dt>
<dd><p>Optional. If provided,'+' or '-', then the summarized
statistic is computed only for the specified DNA chain.</p></dd>


<dt>maxgap, minoverlap, type</dt>
<dd><p>See
<code><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps-methods</a></code> in the
<strong>IRanges</strong> package for a description of these arguments.</p></dd>


<dt>ignore.strand</dt>
<dd><p>When set to TRUE, the strand information is ignored in
the overlap calculations.</p></dd>


<dt>scaling</dt>
<dd><p>integer (default 1). Scaling factor to be used when
stat = 'density'. For example, if scaling = 1000, then density * scaling
denotes the sum of values in 1000 bp.</p></dd>


<dt>logbase</dt>
<dd><p>A positive number: the base with respect to which logarithms
are computed when parameter 'entropy = TRUE' (default: logbase = 2).</p></dd>


<dt>missings</dt>
<dd><p>Whether to write '0' or 'NA' on regions where there is not
data to compute the statistic.</p></dd>


<dt>naming</dt>
<dd><p>Logical value. If TRUE, the rows GRanges object will be
given the names(grfeatures). Default is FALSE.</p></dd>


<dt>na.rm</dt>
<dd><p>Logical value. If TRUE, the NA values will be removed.</p></dd>


<dt>num.cores, tasks</dt>
<dd><p>Parameters for parallel computation using package
<code><a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallel-package.html" class="external-link">BiocParallel-package</a></code>: the number of cores to
use, i.e. at most how many child processes will be run simultaneously (see
<code><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></code> and the number of tasks per job (only
for Linux OS). Only used when signal is a list of
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> object.</p></dd>


<dt>verbose</dt>
<dd><p>Logical. Default is TRUE. If TRUE, then the progress of the
computational tasks is given.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> object or a
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList-class</a></code> object with the new genomic
regions and their corresponding summarized statistic.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function split a Grange object into intervals genomic regions
(GRs) of fixed size A summarized statistic (mean, median, geometric mean or
sum) is calculated for the specified variable values from each region.
Notice that if win.size == step.size, then non-overlapping windows are
obtained.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="getDMPatRegions.html">getDMPatRegions</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Robersy Sanchez (<a href="https://github.com/genomaths" class="external-link">https://github.com/genomaths</a>).</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html" class="external-link">Rle</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'chr1'</span>, <span class="st">'chr2'</span>, <span class="st">'chr3'</span>, <span class="st">'chr4'</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>            <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">5</span>, <span class="fl">5</span>, <span class="fl">5</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>            ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">20</span>, end <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>            strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'+'</span>, <span class="st">'-'</span><span class="op">)</span>, <span class="fl">10</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>            A <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0</span>, length <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gr</span><span class="op">$</span><span class="va">B</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fl">20</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">grs</span> <span class="op">&lt;-</span> <span class="fu">getGRangesStat</span><span class="op">(</span><span class="va">gr</span>, win.size <span class="op">=</span> <span class="fl">4</span>, step.size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |=                                                                     |   1%  |                                                                              |==================                                                    |  25%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%</span>
<span class="r-in"><span><span class="va">grs</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 8 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |       sum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1       1-4      * |  3.684211</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1       5-8      * |  0.789474</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr2       6-9      * |  2.631579</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr2     10-13      * |  0.526316</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr3     11-14      * |  1.578947</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [6]     chr3     15-18      * |  0.263158</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [7]     chr4     16-19      * |  0.526316</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [8]     chr4     20-23      * |  0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 4 sequences from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Selecting the positive strand</span></span></span>
<span class="r-in"><span><span class="va">grs</span> <span class="op">&lt;-</span> <span class="fu">getGRangesStat</span><span class="op">(</span><span class="va">gr</span>, win.size <span class="op">=</span> <span class="fl">4</span>,</span></span>
<span class="r-in"><span>             step.size <span class="op">=</span> <span class="fl">4</span>, select.strand <span class="op">=</span> <span class="st">'+'</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |=                                                                     |   1%  |                                                                              |==================                                                    |  25%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%</span>
<span class="r-in"><span><span class="va">grs</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 4 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |       sum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1       1-4      + |  1.894737</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1       5-8      + |  0.789474</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr3     11-14      + |  0.842105</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr3     15-18      + |  0.263158</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Selecting the negative strand</span></span></span>
<span class="r-in"><span><span class="va">grs</span> <span class="op">&lt;-</span> <span class="fu">getGRangesStat</span><span class="op">(</span><span class="va">gr</span>, win.size <span class="op">=</span> <span class="fl">4</span>,</span></span>
<span class="r-in"><span>             step.size <span class="op">=</span> <span class="fl">4</span>, select.strand <span class="op">=</span> <span class="st">'-'</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |=                                                                     |   1%  |                                                                              |==================                                                    |  25%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%</span>
<span class="r-in"><span><span class="va">grs</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 4 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |       sum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr2       6-9      - |  1.368421</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr2     10-13      - |  0.526316</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr4     16-19      - |  0.315789</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr4     20-23      - |  0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

