<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Potential methylation signal — getPotentialDIMP • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Potential methylation signal — getPotentialDIMP"><meta property="og:description" content="This function perform a selection of the cytosine sites
carrying the potential methylation signal. The potential signals from
controls and treatments are used as prior classification in further step
of signal detection."><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Potential methylation signal</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/getPotentialDIMP.R" class="external-link"><code>R/getPotentialDIMP.R</code></a></small>
    <div class="hidden name"><code>getPotentialDIMP.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function perform a selection of the cytosine sites
carrying the potential methylation signal. The potential signals from
controls and treatments are used as prior classification in further step
of signal detection.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getPotentialDIMP</span><span class="op">(</span></span>
<span>  <span class="va">LR</span>,</span>
<span>  nlms <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">div.col</span>,</span>
<span>  dist.name <span class="op">=</span> <span class="st">"Weibull2P"</span>,</span>
<span>  absolute <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  pval.col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tv.col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tv.cut <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  idv.col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  idv.cut <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min.coverage <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  hdiv.col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  hdiv.cut <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pAdjustMethod <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>LR</dt>
<dd><p>An object from 'InfDiv' or 'testDMP' class. These objects are
previously obtained with function <code><a href="estimateDivergence.html">estimateDivergence</a></code> or
<code><a href="FisherTest.html">FisherTest</a></code>.</p></dd>


<dt>nlms</dt>
<dd><p>A list of distribution fitted models (output of
<code><a href="gofReport.html">gofReport</a></code> function) or NULL. If NULL, then empirical
cumulative distribution function is used to get the potential DMPs.</p></dd>


<dt>div.col</dt>
<dd><p>Column number for divergence variable is located in the
meta-column.</p></dd>


<dt>dist.name</dt>
<dd><p>Name of the fitted distribution. This could be the name of
one distribution or a characters vector of length(nlms). Default is the
two parameters Weibull distribution: 'Weibull2P'. The available options
are:</p><dl><dt><strong>"Weibull2P"</strong></dt>
<dd><p>Weibull with two-parameters.</p></dd>

<dt><strong>"Weibull3P"</strong></dt>
<dd><p>Weibull with three-parameters.</p></dd>

<dt><strong>"Gamma2P"</strong></dt>
<dd><p>Gamma with two-parameters.</p></dd>

<dt><strong>"Gamma3P"</strong></dt>
<dd><p>Gamma with three-parameters.</p></dd>

<dt><strong>"GGamma3P"</strong></dt>
<dd><p>Generalized gamma with three-parameters.</p></dd>

<dt><strong>"GGamma4P"</strong></dt>
<dd><p>Generalized gamma with four-parameters.</p></dd>

<dt><strong>"ECDF"</strong></dt>
<dd><p>The empirical cumulative distribution function.</p></dd>

<dt><strong>"None"</strong></dt>
<dd><p>No distribution.</p></dd>


</dl><p>If <strong>dist.name != 'None'</strong>, and <strong>nlms != NULL</strong>, then a column
named 'wprob' with a probability vector derived from the application of
model 'nlms' will be returned.</p></dd>


<dt>absolute</dt>
<dd><p>Logic (default, FALSE). Total variation (TV, the difference
of methylation levels) is normally an output in the downstream MethylIT
analysis. If 'absolute = TRUE', then TV is transformed into |TV|, which is
an information divergence that can be fitted to Weibull or to Generalized
Gamma distribution. So, if the nonlinear fit was performed for |TV|, then
absolute must be set to TRUE.</p></dd>


<dt>alpha</dt>
<dd><p>A numerical value (usually \(\alpha \leq 0.05\)) used to
select cytosine sites \(k\) with information divergence (\(DIV_k\)) for
which the the probabilities hold: \(P(DIV_k &gt; DIV(\alpha))\).</p></dd>


<dt>pval.col</dt>
<dd><p>An integer denoting a column from each GRanges object from
LR where p-values are provided when <strong>dist.name == 'None'</strong> and
<strong>nlms == NULL</strong>. Default is NULL. If NUll and <strong>dist.name ==
'None'</strong> and <strong>nlms == NULL</strong>, then a column named <strong>adj.pval</strong>
will used to select the potential DMPs.</p></dd>


<dt>tv.col</dt>
<dd><p>Column number for the total variation to be used for
filtering cytosine positions (if provided).</p></dd>


<dt>tv.cut</dt>
<dd><p>If tv.cut and tv.col are provided, then cytosine sites k with
\(abs(TV_k) &lt; tv.cut\) are removed before to perform the ROC analysis.</p></dd>


<dt>min.coverage</dt>
<dd><p>Cytosine sites with coverage less than min.coverage are
discarded. Default: 0</p></dd>


<dt>hdiv.col</dt>
<dd><p>Optional. A column number for the Hellinger distance to be
used for filtering cytosine positions. Default is NULL.</p></dd>


<dt>hdiv.cut</dt>
<dd><p>If hdiv.cut and hdiv.col are provided, then cytosine sites
\(k\) with hdiv &lt; hdiv.cut are removed.</p></dd>


<dt>pAdjustMethod</dt>
<dd><p>method used to adjust the p-values from other
approaches like Fisher's exact test, which involve multiple comparisons
Default is NULL. Do not apply it when a probability distribution model is
used (<strong>when nlms is given</strong>), since it makes not sense.</p></dd>


<dt>idiv.col</dt>
<dd><p>Optional. A column number for any of the available
information divergences: \(TV, bay.TV, hdiv, or jdiv\).
used for filtering cytosine positions. Default is NULL.</p></dd>


<dt>idiv.cut</dt>
<dd><p>If hdiv.cut and hdiv.col are provided, then cytosine sites
\(k\) with hdiv &lt; hdiv.cut are removed.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A list of GRanges objects, each GRanges object carrying the
selected cytosine sites and the probabilities that the specified divergence
values can be greater than the critical value specified by \(\alpha\):
\(P(DIV_k &gt; DIV(\alpha))\).</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The potential signals are cytosine sites k with information
divergence (DIV_k) values greater than the DIV(alpha = 0.05). The value
of alpha can be specified. For example, potential signals with
DIV_k &gt; DIV(alpha = 0.01) can be selected. For each sample, cytosine
sites are selected based on the corresponding nonlinear fitted
distribution model that has been supplied.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Get a dataset of Hellinger divergency of methylation levels and their</span></span></span>
<span class="r-in"><span><span class="co">## corresponding best nonlinear fit distribution models from the package</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">HD</span>, <span class="va">gof</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">PS</span> <span class="op">&lt;-</span> <span class="fu">getPotentialDIMP</span><span class="op">(</span>LR <span class="op">=</span> <span class="va">HD</span>, nlms <span class="op">=</span> <span class="va">gof</span><span class="op">$</span><span class="va">nlms</span>, dist.name <span class="op">=</span> <span class="va">gof</span><span class="op">$</span><span class="va">bestModel</span>,</span></span>
<span class="r-in"><span>                    div.col <span class="op">=</span> <span class="fl">9L</span>, alpha <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

