<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Report the Best Fitted Probability Distribution Model — gofReport • MethylIT</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Report the Best Fitted Probability Distribution Model — gofReport" />
<meta property="og:description" content="In Methyl-IT,  the signal detection step requires for the
knowledge of the probability distribution of the methylation noise, which
is the null hypothesis used to discriminate the signal from the noise.
Both, signal and noise, are expressed in terms of an information
divergence of methylation levels, which (currently in Methyl-IT) could be
the Hellinger divergence or the total variation distance.
As shown in reference (1), on statistical physics basis, the probability
distribution of the noise is a member of the generalized gamma
distribution. In particular, if the methylation changes on the DNA
molecule follow a Poisson process, then the noise follows a Weibull
probability distribution.
Function 'gofReport' search for the best fitted model between the set of
models requested by the user. Two goodness-of-fit (GoF) criteria are applied
to select the best fitted model: Akaike's information criterion (AIC) and the
correlation coefficient of cross-validations for the nonlinear regressions
(R.Cross.val) (2). These criteria evaluate different information inferred
from the models. AIC deals with the trade-off between the goodness of fit of
the model and the complexity of the model, while R.Cross.val provides
information on the prediction power/performance of the model when confronted
with external dataset.
Although the numerical algorithms to accomplish the nonlinear fit are not
perfect, in general, the model with the lowest AIC must have the highest
R.Cross.val. If the model with the lowest AIC has not the highest
R.Cross.val, then further analyzes are required. These analyzes could include
the visualization of the graphics for the density distribution, evaluation of
whether the parameter values can be meaningful or not, etc. Nevertheless, the
best model will, in general, lead to the identification of the greater amount
of potential DMPs and DMPs, as well as, the highest classification accuracy
estimated with functions estimateCutPoint and
evaluateDIMPclass. In the worse scenario, these observations
can ultimately lead to a post-hoc decision on which the best model is." />
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/MethylIT/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Report the Best Fitted Probability Distribution Model</h1>
    <small class="dont-index">Source: <a href='https://github.com/genomaths/MethylIT/blob/master/R/gofReport.R'><code>R/gofReport.R</code></a></small>
    <div class="hidden name"><code>gofReport.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>In Methyl-IT,  the signal detection step requires for the
knowledge of the probability distribution of the methylation noise, which
is the null hypothesis used to discriminate the signal from the noise.
Both, signal and noise, are expressed in terms of an information
divergence of methylation levels, which (currently in Methyl-IT) could be
the Hellinger divergence or the total variation distance.
As shown in reference (1), on statistical physics basis, the probability
distribution of the noise is a member of the generalized gamma
distribution. In particular, if the methylation changes on the DNA
molecule follow a Poisson process, then the noise follows a Weibull
probability distribution.</p>
<p>Function 'gofReport' search for the best fitted model between the set of
models requested by the user. Two goodness-of-fit (GoF) criteria are applied
to select the best fitted model: Akaike's information criterion (AIC) and the
correlation coefficient of cross-validations for the nonlinear regressions
(R.Cross.val) (2). These criteria evaluate different information inferred
from the models. AIC deals with the trade-off between the goodness of fit of
the model and the complexity of the model, while R.Cross.val provides
information on the prediction power/performance of the model when confronted
with external dataset.</p>
<p>Although the numerical algorithms to accomplish the nonlinear fit are not
perfect, in general, the model with the lowest AIC must have the highest
R.Cross.val. If the model with the lowest AIC has not the highest
R.Cross.val, then further analyzes are required. These analyzes could include
the visualization of the graphics for the density distribution, evaluation of
whether the parameter values can be meaningful or not, etc. Nevertheless, the
best model will, in general, lead to the identification of the greater amount
of potential DMPs and DMPs, as well as, the highest classification accuracy
estimated with functions <code><a href='estimateCutPoint.html'>estimateCutPoint</a></code> and
<code><a href='evaluateDIMPclass.html'>evaluateDIMPclass</a></code>. In the worse scenario, these observations
can ultimately lead to a post-hoc decision on which the best model is.</p>
    </div>

    <pre class="usage"><span class='fu'>gofReport</span><span class='op'>(</span>
  <span class='va'>HD</span>,
  model <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Weibull2P"</span>, <span class='st'>"Weibull3P"</span>, <span class='st'>"Gamma2P"</span>, <span class='st'>"Gamma3P"</span><span class='op'>)</span>,
  column <span class='op'>=</span> <span class='fl'>9</span>,
  absolute <span class='op'>=</span> <span class='cn'>FALSE</span>,
  output <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"best.model"</span>, <span class='st'>"all"</span><span class='op'>)</span>,
  alt_models <span class='op'>=</span> <span class='cn'>FALSE</span>,
  r.cv <span class='op'>=</span> <span class='cn'>FALSE</span>,
  npoints <span class='op'>=</span> <span class='cn'>NULL</span>,
  num.cores <span class='op'>=</span> <span class='fl'>1L</span>,
  verbose <span class='op'>=</span> <span class='cn'>FALSE</span>,
  <span class='va'>...</span>
<span class='op'>)</span>

<span class='co'># S3 method for ProbDistrList</span>
<span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>HD</th>
      <td><p>An 'InfDiv' object returned by function
<code><a href='estimateDivergence.html'>estimateDivergence</a></code>.</p></td>
    </tr>
    <tr>
      <th>model</th>
      <td><p>A character vector naming the models to fit. Default is
model = c('Weibull2P', 'Weibull3P', 'Gamma2P', 'Gamma3P'). See
<code><a href='nonlinearFitDist.html'>nonlinearFitDist</a></code> for more options.</p></td>
    </tr>
    <tr>
      <th>column</th>
      <td><p>An integer number denoting the index of the GRanges column where
the information divergence is given. Default column = 9, which is the
column where the Hellinger divergence values are reported (by default) by
function <code><a href='estimateDivergence.html'>estimateDivergence</a></code>.</p></td>
    </tr>
    <tr>
      <th>absolute</th>
      <td><p>Logic (default, FALSE). Total variation (TV, the difference
of methylation levels) is normally an output in the downstream MethylIT
analysis. If 'absolute = TRUE', then TV is transformed into |TV|, which
is an information divergence that can be fitted as well.</p></td>
    </tr>
    <tr>
      <th>output</th>
      <td><p>If output == 'all', the table with the GoF statistics is
returned in a list together with the best fitted model and the
corresponding statistics. Default is 'best.model'</p></td>
    </tr>
    <tr>
      <th>alt_models</th>
      <td><p>logical(1). If TRUE, then the best model based on highest
R.Cross.val is returned for those samples where the model(s) with lowest
AIC has not the highest R.Cross.val.</p></td>
    </tr>
    <tr>
      <th>r.cv</th>
      <td><p>logical(1). Whether to select the best model based on the highest
R.Cross.val (2) (see details section).</p></td>
    </tr>
    <tr>
      <th>npoints</th>
      <td><p>number of points to be used in the fit. Default is NULL.</p></td>
    </tr>
    <tr>
      <th>num.cores</th>
      <td><p>The number of cores to use in the nonlinear fit step, i.e.
at most how many child processes will be run simultaneously (see
<code><a href='https://rdrr.io/pkg/BiocParallel/man/bplapply.html'>bplapply</a></code> function from BiocParallel package).</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>If TRUE, prints the function log to stdout</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Further arguments to pass to <code><a href='nonlinearFitDist.html'>nonlinearFitDist</a></code>.</p></td>
    </tr>
    <tr>
      <th>x</th>
      <td><p>An object from 'ProbDistrList' class</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>If 'output = 'best.model'', a character vector with the name of the
best fitted model for each sample and model statistics, which can be used the
next step to get the potential DMPs with function
<code><a href='getPotentialDIMP.html'>getPotentialDIMP</a></code>. Otherwise, it will return a list with the
table carrying the GoF values and the previously mentioned data.</p>
<p>The best model selection is based on the lowest AIC. However, if
\(alt_models = TRUE\) and \(r.cv = FALSE\), then the returned list will
contain two sublists named 'alt_models'  and 'alt_nlms' with the best model
selected based on the highest R.Cross.val. If \(r.cv = TRUE\), best model
selection is based on the highest R.Cross.val and sublist named 'alt_models'
will carry the model selected based on the lowest AIC. This sublist is
returned only if at least there is one sample where the model with the lowest
AIC has not the highest R.Cross.val. This report is useful to analyze any
conflict between models. For example, some times the best model selected
based on AIC has a R.Cross.val = 0.99945, while the highest R.Cross.val is
0.99950. In such a situation the model with lowest AIC is still fine.
However, some times the model with the best AIC has some meaningless
parameter value. For example, scale = 0.0000000001 in 'GGamma3P' or in
'Weibull2P' models. This last situation can result from the numerical
algorithm used in the parameter estimation. The numerical algorithms for
nonlinear fit estimation are not perfect!</p>
<p>For distribution models that include a location parameter ('Weibull3P',
'Gamma3P', and 'GGamma4P') there is an additional important constraint, which
cannot be evaluated with 'AIC' or 'R.Cross.val': \(\mu &gt; 0\). That is, the
information divergences are strictly positive magnitudes, therefore, any best
model in terms of AIC with values \(\mu &lt; 0\) are meaningless and rejected.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    
<ol>
<li><p>R. Sanchez and S. A. Mackenzie, “Information Thermodynamics of
Cytosine DNA Methylation,” PLoS One, vol. 11, no. 3, p. e0150427,
Mar. 2016.</p></li>
<li><p>Stevens JP. Applied Multivariate Statistics for the Social
Sciences. Fifth Edit. Routledge Academic; 2009.</p></li>
</ol>

    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='nonlinearFitDist.html'>nonlinearFitDist</a></code></p></div>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Robersy Sanchez 11/25/2019 <a href='https://github.com/genomaths'>https://github.com/genomaths</a></p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Loading information divergence dataset</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>HD</span><span class='op'>)</span>

<span class='co'>## Subsetting object HD (for the sake of runnig a faster example)</span>
<span class='va'>hd</span> <span class='op'>&lt;-</span> <span class='fu'><a href='lapply.html'>lapply</a></span><span class='op'>(</span><span class='va'>HD</span>, <span class='kw'>function</span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='va'>x</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_len</a></span><span class='op'>(</span><span class='fl'>100</span><span class='op'>)</span><span class='op'>]</span>, keep.attr <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>

<span class='co'>## The GoF report</span>
<span class='va'>dt</span> <span class='op'>&lt;-</span> <span class='fu'>gofReport</span><span class='op'>(</span><span class='va'>hd</span><span class='op'>)</span>
</div><div class='output co'>#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |==================                                                    |  25%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%
#&gt; 
#&gt;  *** Creating report ... </div><div class='output co'>#&gt; <span class='warning'>Warning: </span>
#&gt; <span class='warning'> The best fitted model for sample(s) C2 require(s) for further analysis. </span>
#&gt; <span class='warning'>The model with the lowest AIC must have the highest R.Cross.val</span></div><div class='output co'>#&gt;      w2p_AIC w2p_R.Cross.val   w3p_AIC w3p_R.Cross.val   g2p_AIC
#&gt; C1 -444.9345       0.9963494        NA              NA -442.7779
#&gt; C2 -404.4685       0.9942247 -402.6029       0.9940469 -390.2555
#&gt; C3 -491.5246       0.9978235        NA              NA -512.3279
#&gt; T1 -482.0144       0.9973647 -487.2492       0.9973930 -450.5389
#&gt; T2 -469.6186       0.9969509        NA              NA -473.5559
#&gt; T3 -529.6873       0.9983691 -523.4597       0.9983520 -503.6232
#&gt;    g2p_R.Cross.val   g3p_AIC g3p_R.Cross.val      bestModel
#&gt; C1       0.9960281       Inf       0.0000000            w2p
#&gt; C2       0.9940825 -387.3422       0.9942271 Needs revision
#&gt; C3       0.9981350       Inf       0.0000000            g2p
#&gt; T1       0.9966888 -477.7403       0.9972774            w3p
#&gt; T2       0.9972196       Inf       0.0000000            g2p
#&gt; T3       0.9979191 -513.8786       0.9981566            w2p
#&gt; </div><div class='input'>
<span class='co'>## Report the model where AIC and R.Cross.val are in conflict.</span>
<span class='va'>dt</span> <span class='op'>&lt;-</span> <span class='fu'>gofReport</span><span class='op'>(</span>HD <span class='op'>=</span> <span class='va'>hd</span>, output <span class='op'>=</span> <span class='st'>"all"</span>, alt_models <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
</div><div class='output co'>#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |==================                                                    |  25%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%
#&gt; 
#&gt;  *** Creating report ... </div><div class='output co'>#&gt; <span class='warning'>Warning: </span>
#&gt; <span class='warning'> The best fitted model for sample(s) T3 require(s) for further analysis. </span>
#&gt; <span class='warning'>The model with the lowest AIC must have the highest R.Cross.val</span></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


