<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Compute methylation levels — meth_levels • MethylIT</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Compute methylation levels — meth_levels" />
<meta property="og:description" content="This function computes the" />
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/MethylIT/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Compute methylation levels</h1>
    <small class="dont-index">Source: <a href='https://github.com/genomaths/MethylIT/blob/master/R/meth_levels.R'><code>R/meth_levels.R</code></a></small>
    <div class="hidden name"><code>meth_levels.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function computes the</p>
    </div>

    <pre class="usage"><span class='fu'>meth_levels</span><span class='op'>(</span>
  <span class='va'>GR</span>,
  <span class='va'>x</span>,
  columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>mC1 <span class='op'>=</span> <span class='fl'>1</span>, uC1 <span class='op'>=</span> <span class='fl'>2</span>, mC2 <span class='op'>=</span> <span class='cn'>NULL</span>, uC2 <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span>,
  Bayesian <span class='op'>=</span> <span class='cn'>FALSE</span>,
  init.pars <span class='op'>=</span> <span class='cn'>NULL</span>,
  min.coverage <span class='op'>=</span> <span class='fl'>4</span>,
  tv <span class='op'>=</span> <span class='cn'>FALSE</span>,
  bay.tv <span class='op'>=</span> <span class='cn'>FALSE</span>,
  filter <span class='op'>=</span> <span class='cn'>FALSE</span>,
  preserve.dt <span class='op'>=</span> <span class='cn'>FALSE</span>,
  loss.fun <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"linear"</span>, <span class='st'>"huber"</span>, <span class='st'>"smooth"</span>, <span class='st'>"cauchy"</span>, <span class='st'>"arctg"</span><span class='op'>)</span>,
  num.cores <span class='op'>=</span> <span class='fl'>1</span>,
  tasks <span class='op'>=</span> <span class='fl'>0L</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>,
  <span class='va'>...</span>
<span class='op'>)</span>

<span class='co'># S4 method for ANY,data.frame</span>
<span class='fu'>meth_levels</span><span class='op'>(</span>
  <span class='va'>GR</span>,
  <span class='va'>x</span>,
  columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>mC1 <span class='op'>=</span> <span class='fl'>1</span>, uC1 <span class='op'>=</span> <span class='fl'>2</span>, mC2 <span class='op'>=</span> <span class='cn'>NULL</span>, uC2 <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span>,
  Bayesian <span class='op'>=</span> <span class='cn'>FALSE</span>,
  init.pars <span class='op'>=</span> <span class='cn'>NULL</span>,
  min.coverage <span class='op'>=</span> <span class='fl'>4</span>,
  tv <span class='op'>=</span> <span class='cn'>FALSE</span>,
  bay.tv <span class='op'>=</span> <span class='cn'>FALSE</span>,
  filter <span class='op'>=</span> <span class='cn'>FALSE</span>,
  preserve.dt <span class='op'>=</span> <span class='cn'>FALSE</span>,
  loss.fun <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"linear"</span>, <span class='st'>"huber"</span>, <span class='st'>"smooth"</span>, <span class='st'>"cauchy"</span>, <span class='st'>"arctg"</span><span class='op'>)</span>,
  num.cores <span class='op'>=</span> <span class='fl'>1</span>,
  tasks <span class='op'>=</span> <span class='fl'>0L</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>,
  <span class='va'>...</span>
<span class='op'>)</span>

<span class='co'># S4 method for GRanges,ANY</span>
<span class='fu'>meth_levels</span><span class='op'>(</span>
  <span class='va'>GR</span>,
  <span class='va'>x</span>,
  columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>mC1 <span class='op'>=</span> <span class='fl'>1</span>, uC1 <span class='op'>=</span> <span class='fl'>2</span>, mC2 <span class='op'>=</span> <span class='cn'>NULL</span>, uC2 <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span>,
  Bayesian <span class='op'>=</span> <span class='cn'>FALSE</span>,
  init.pars <span class='op'>=</span> <span class='cn'>NULL</span>,
  min.coverage <span class='op'>=</span> <span class='fl'>4</span>,
  tv <span class='op'>=</span> <span class='cn'>FALSE</span>,
  bay.tv <span class='op'>=</span> <span class='cn'>FALSE</span>,
  filter <span class='op'>=</span> <span class='cn'>FALSE</span>,
  preserve.dt <span class='op'>=</span> <span class='cn'>FALSE</span>,
  loss.fun <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"linear"</span>, <span class='st'>"huber"</span>, <span class='st'>"smooth"</span>, <span class='st'>"cauchy"</span>, <span class='st'>"arctg"</span><span class='op'>)</span>,
  num.cores <span class='op'>=</span> <span class='fl'>1</span>,
  tasks <span class='op'>=</span> <span class='fl'>0L</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>,
  <span class='va'>...</span>
<span class='op'>)</span>

<span class='co'># S4 method for list,ANY</span>
<span class='fu'>meth_levels</span><span class='op'>(</span>
  <span class='va'>GR</span>,
  x <span class='op'>=</span> <span class='cn'>NULL</span>,
  columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>mC1 <span class='op'>=</span> <span class='fl'>1</span>, uC1 <span class='op'>=</span> <span class='fl'>2</span>, mC2 <span class='op'>=</span> <span class='fl'>0</span>, uC2 <span class='op'>=</span> <span class='fl'>0</span><span class='op'>)</span>,
  Bayesian <span class='op'>=</span> <span class='cn'>FALSE</span>,
  init.pars <span class='op'>=</span> <span class='cn'>NULL</span>,
  min.coverage <span class='op'>=</span> <span class='fl'>4</span>,
  tv <span class='op'>=</span> <span class='cn'>FALSE</span>,
  bay.tv <span class='op'>=</span> <span class='cn'>FALSE</span>,
  filter <span class='op'>=</span> <span class='cn'>FALSE</span>,
  preserve.dt <span class='op'>=</span> <span class='cn'>FALSE</span>,
  loss.fun <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"linear"</span>, <span class='st'>"huber"</span>, <span class='st'>"smooth"</span>, <span class='st'>"cauchy"</span>, <span class='st'>"arctg"</span><span class='op'>)</span>,
  num.cores <span class='op'>=</span> <span class='fl'>1</span>,
  tasks <span class='op'>=</span> <span class='fl'>0L</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>GR, x</th>
      <td><p>A <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges-class</a></code> object
(<strong><em>GR</em></strong>) or 'data.frame' (<strong><em>x</em></strong>) with a matrix of
counts in the meta-columns (methylated mC and unmethylated uC cytosines) or a
list of <code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges-class</a></code> objects.</p></td>
    </tr>
    <tr>
      <th>columns</th>
      <td><p>Vector of one or two integer numbers denoting the indexes of
the columns where the methylated and unmethylated read counts are found.
Unless specified in the parameter 'columns', the methylation counts must be
given in the first four columns: 'mC1' and 'uC1' methylated and unmethylated
counts for control sample, and 'mC2' and 'uC2' methylated and unmethylated
counts for treatment sample, respectively.</p></td>
    </tr>
    <tr>
      <th>Bayesian</th>
      <td><p>logical(1). Whether to perform the estimations based on
posterior estimations of methylation levels.</p></td>
    </tr>
    <tr>
      <th>init.pars</th>
      <td><p>initial parameter values. Defaults is NULL and an initial
guess is estimated using <code><a href='https://rdrr.io/r/stats/optim.html'>optim</a></code> function. If the initial
guessing fails initial parameter values are to alpha = 1 &amp;
beta = 1, which imply the parsimony pseudo-counts greater than zero.</p></td>
    </tr>
    <tr>
      <th>min.coverage</th>
      <td><p>An integer or an integer vector of length 2. Cytosine
sites where the coverage in both samples, 'x' and 'y', are less than
min.coverage' are discarded. The cytosine site is preserved, however, if the
coverage is greater than 'min.coverage' in at least one sample. If
'min.coverage' is an integer vector, then the corresponding min coverage is
applied to each sample.</p></td>
    </tr>
    <tr>
      <th>tv</th>
      <td><p>logical(1). Whether to compute the total variation distance at each
cytosine site. That is, the difference of methylation levels.</p></td>
    </tr>
    <tr>
      <th>bay.tv</th>
      <td><p>logical(1). Whether to compute the total variation distance at
each cytosine site based on Bayesian estimation of methylation levels.</p></td>
    </tr>
    <tr>
      <th>filter</th>
      <td><p>logical(1). Optional. If TRUE, then only cytosine sites with
\(coverages &gt; min.coverage\) are including in the computation.</p></td>
    </tr>
    <tr>
      <th>preserve.dt</th>
      <td><p>logical(1). Option of whether to preserve all
the metadata from the original 'data.frame' or
<code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges-class</a></code> object.</p></td>
    </tr>
    <tr>
      <th>loss.fun</th>
      <td><p>Described in <code><a href='estimateBetaDist.html'>estimateBetaDist</a></code>.</p></td>
    </tr>
    <tr>
      <th>num.cores, tasks</th>
      <td><p>Parameters for parallel computation using package
<code><a href='https://rdrr.io/pkg/BiocParallel/man/BiocParallel-package.html'>BiocParallel-package</a></code>: the number of cores to use,
i.e. at most how many child processes will be run simultaneously (see
<code><a href='https://rdrr.io/pkg/BiocParallel/man/bplapply.html'>bplapply</a></code> and the number of tasks per job (only
for Linux OS). These parameters will be passed to
<code><a href='uniqueGRanges.html'>uniqueGRanges</a></code>.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>if TRUE, prints the function log to stdout</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Optional parameter values for: maxiter, ftol, ptol, and gradtol
from <code><a href='https://rdrr.io/pkg/minpack.lm/man/nlsLM.html'>nlsLM</a></code> and <code><a href='https://rdrr.io/r/stats/nlm.html'>nlm</a></code> functions.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Robersy Sanchez (<a href='https://genomaths.com'>https://genomaths.com</a>)</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## The read count data are created</span>
<span class='va'>num.samples</span> <span class='op'>&lt;-</span> <span class='fl'>250</span>
<span class='va'>s</span> <span class='op'>&lt;-</span> <span class='fl'>1</span><span class='op'>:</span><span class='va'>num.samples</span>
<span class='va'>gr</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>chr <span class='op'>=</span> <span class='st'>'chr1'</span>, start <span class='op'>=</span> <span class='va'>s</span>, end <span class='op'>=</span> <span class='va'>s</span>,
                strand <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/sample.html'>sample</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"+"</span>, <span class='st'>"-"</span><span class='op'>)</span>, <span class='va'>num.samples</span>, replace <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>,
                mCc <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/NegBinomial.html'>rnbinom</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='va'>num.samples</span>, mu <span class='op'>=</span> <span class='fl'>4</span>, n <span class='op'>=</span> <span class='fl'>500</span><span class='op'>)</span>,
                uCc <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/NegBinomial.html'>rnbinom</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='va'>num.samples</span>, mu <span class='op'>=</span> <span class='fl'>4</span>, n <span class='op'>=</span> <span class='fl'>500</span><span class='op'>)</span>,
                mCt <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/NegBinomial.html'>rnbinom</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='va'>num.samples</span>, mu <span class='op'>=</span> <span class='fl'>4</span>, n <span class='op'>=</span> <span class='fl'>500</span><span class='op'>)</span>,
                uCt <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/stats/NegBinomial.html'>rnbinom</a></span><span class='op'>(</span>size <span class='op'>=</span> <span class='va'>num.samples</span>, mu <span class='op'>=</span> <span class='fl'>4</span>, n <span class='op'>=</span> <span class='fl'>500</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>gr</span> <span class='op'>&lt;-</span> <span class='fu'>makeGRangesFromDataFrame</span><span class='op'>(</span><span class='va'>gr</span>, keep.extra.columns <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>

<span class='va'>gr</span> <span class='op'>&lt;-</span> <span class='fu'>meth_levels</span><span class='op'>(</span>GR <span class='op'>=</span> <span class='va'>gr</span>,
                columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>mC1 <span class='op'>=</span> <span class='fl'>1</span>, uC1 <span class='op'>=</span> <span class='fl'>2</span>,
                            mC2 <span class='op'>=</span> <span class='fl'>3</span>, uC2 <span class='op'>=</span> <span class='fl'>4</span><span class='op'>)</span>,
                preserve.dt <span class='op'>=</span> <span class='cn'>TRUE</span>,
                Bayesian <span class='op'>=</span> <span class='cn'>TRUE</span>, tv <span class='op'>=</span> <span class='cn'>TRUE</span>, bay.tv <span class='op'>=</span> <span class='cn'>TRUE</span>,
                num.cores <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>
</div><div class='output co'>#&gt; 
#&gt; *** Estimating betaBinomial-posteriors... 
#&gt; 
#&gt; *** Estimating betaBinomial-posteriors... </div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


