<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Machine-learning Approach for Cutpoint Estimation — mlCutpoint • MethylIT</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Machine-learning Approach for Cutpoint Estimation — mlCutpoint" />
<meta property="og:description" content="Internal function to estimate cutpoint following a
machine-learning approach" />
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/MethylIT/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Machine-learning Approach for Cutpoint Estimation</h1>
    <small class="dont-index">Source: <a href='https://github.com/genomaths/MethylIT/blob/master/R/mlCutpoint.R'><code>R/mlCutpoint.R</code></a></small>
    <div class="hidden name"><code>mlCutpoint.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Internal function to estimate cutpoint following a
machine-learning approach</p>
    </div>

    <pre class="usage"><span class='fu'>mlCutpoint</span><span class='op'>(</span>
  <span class='va'>LR</span>,
  <span class='va'>control.names</span>,
  <span class='va'>treatment.names</span>,
  <span class='va'>column</span>,
  <span class='va'>div.col</span>,
  tv.col <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>tv.cut</span>,
  post.cut <span class='op'>=</span> <span class='fl'>0.5</span>,
  <span class='va'>classifier1</span>,
  classifier2 <span class='op'>=</span> <span class='cn'>NULL</span>,
  interactions <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>n.pc</span>,
  prop <span class='op'>=</span> <span class='fl'>0.6</span>,
  stat <span class='op'>=</span> <span class='fl'>0L</span>,
  cut.values <span class='op'>=</span> <span class='cn'>NULL</span>,
  num.cores <span class='op'>=</span> <span class='fl'>1L</span>,
  tasks <span class='op'>=</span> <span class='fl'>0L</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>LR, res, control.names, treatment.names, column, div.col</th>
      <td><p>Same as in
<code><a href='estimateCutPoint.html'>estimateCutPoint</a></code></p></td>
    </tr>
    <tr>
      <th>column, div.col, tv.col, tv.cut, cut.values, stat</th>
      <td><p>Same as in
<code><a href='estimateCutPoint.html'>estimateCutPoint</a></code></p></td>
    </tr>
    <tr>
      <th>classifier1, classifier2, n.pc, prop, post.cut</th>
      <td><p>Same as in
<code><a href='estimateCutPoint.html'>estimateCutPoint</a></code></p></td>
    </tr>
    <tr>
      <th>interactions</th>
      <td><p>If a logistic classifier is used. Variable interactions to
consider in a logistic regression model. Any pairwise combination of the
variable 'hdiv', 'TV', 'wprob', and 'pos' can be provided. For example:
'hdiv:TV', 'wprob:pos', 'wprob:TV', etc.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments for <code><a href='evaluateDIMPclass.html'>evaluateDIMPclass</a></code> function</p></td>
    </tr>
    <tr>
      <th>cutp_data, num.cores, tasks</th>
      <td><p>Same as in <code><a href='estimateCutPoint.html'>estimateCutPoint</a></code></p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Specified in function <code><a href='estimateCutPoint.html'>estimateCutPoint</a></code> for parameter
setting <em>simple = FALSE</em></p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function is called by function <code><a href='estimateCutPoint.html'>estimateCutPoint</a></code>.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Get a set of potential DMPS (PS)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>PS</span>, package <span class='op'>=</span> <span class='st'>'MethylIT'</span><span class='op'>)</span>

<span class='va'>cutp</span> <span class='op'>&lt;-</span> <span class='fu'>mlCutpoint</span><span class='op'>(</span>LR <span class='op'>=</span> <span class='va'>PS</span>,
                 column <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>hdiv <span class='op'>=</span> <span class='cn'>TRUE</span>, TV <span class='op'>=</span> <span class='cn'>TRUE</span>,
                            wprob <span class='op'>=</span> <span class='cn'>TRUE</span>, pos <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>,
                 classifier1 <span class='op'>=</span> <span class='st'>'qda'</span>, n.pc <span class='op'>=</span> <span class='fl'>4</span>,
                 control.names <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'C1'</span>, <span class='st'>'C2'</span>, <span class='st'>'C3'</span><span class='op'>)</span>,
                 treatment.names <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'T1'</span>, <span class='st'>'T2'</span>, <span class='st'>'T3'</span><span class='op'>)</span>,
                 tv.cut <span class='op'>=</span> <span class='fl'>0.68</span>, prop <span class='op'>=</span> <span class='fl'>0.6</span>,
                 div.col <span class='op'>=</span> <span class='fl'>9L</span><span class='op'>)</span>
<span class='va'>cutp</span>
</div><div class='output co'>#&gt; Cutpoint estimation with 'qda' classifier 
#&gt; Cutpoint search performed using model posterior probabilities 
#&gt; 
#&gt; Posterior probability used to get the cutpoint = 0.5 
#&gt; Cytosine sites with treatment PostProbCut &gt;= 0.5 have a 
#&gt; divergence value &gt;= 117.0424 
#&gt; 
#&gt; Optimized statistic: Accuracy = 1 
#&gt; Cutpoint = 117.04 
#&gt; 
#&gt; Model classifier 'qda' 
#&gt; 
#&gt; The accessible objects in the output list are: 
#&gt;                    Length Class           Mode     
#&gt; cutpoint            1     -none-          numeric  
#&gt; testSetPerformance  6     confusionMatrix list     
#&gt; testSetModel.FDR    1     -none-          numeric  
#&gt; model              10     qdaDMP          list     
#&gt; modelConfMatrix     6     confusionMatrix list     
#&gt; initModel           1     -none-          character
#&gt; postProbCut         1     -none-          numeric  
#&gt; postCut             1     -none-          numeric  
#&gt; classifier          1     -none-          character
#&gt; statistic           1     -none-          character
#&gt; optStatVal          1     -none-          numeric  
#&gt; cutpData            1     -none-          logical  </div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


