<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Quadratic Discriminant Analysis (QDA) using Principal Component
Analysis (PCA) — pcaQDA • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Quadratic Discriminant Analysis (QDA) using Principal Component
Analysis (PCA) — pcaQDA"><meta property="og:description" content="The principal components (PCs) for predictor variables provided
as input data are estimated and then the individual coordinates in the
selected PCs are used as predictors in the qda
Predict using a PCA-LDA model built with function 'pcaLDA'"><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Quadratic Discriminant Analysis (QDA) using Principal Component
Analysis (PCA)</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/pcaQDA.R" class="external-link"><code>R/pcaQDA.R</code></a></small>
    <div class="hidden name"><code>pcaQDA.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The principal components (PCs) for predictor variables provided
as input data are estimated and then the individual coordinates in the
selected PCs are used as predictors in the qda</p>
<p>Predict using a PCA-LDA model built with function 'pcaLDA'</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">pcaQDA</span><span class="op">(</span></span>
<span>  formula <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  data <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  grouping <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  n.pc <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  center <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  tol <span class="op">=</span> <span class="fl">1e-04</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"moment"</span>,</span>
<span>  max.pc <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for pcaQDA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  <span class="va">newdata</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"qda.pred"</span>, <span class="st">"class"</span>, <span class="st">"posterior"</span>, <span class="st">"pca.ind.coord"</span>, <span class="st">"all"</span><span class="op">)</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>formula</dt>
<dd><p>Same as in <code><a href="https://rdrr.io/pkg/MASS/man/qda.html" class="external-link">qda</a></code> from package 'MASS'.</p></dd>


<dt>data</dt>
<dd><p>Same as in <code><a href="https://rdrr.io/pkg/MASS/man/qda.html" class="external-link">qda</a></code> from package 'MASS'.</p></dd>


<dt>grouping</dt>
<dd><p>Same as in <code><a href="https://rdrr.io/pkg/MASS/man/qda.html" class="external-link">qda</a></code> from package 'MASS'.</p></dd>


<dt>n.pc</dt>
<dd><p>Number of principal components to use in the qda.</p></dd>


<dt>scale</dt>
<dd><p>Same as in <code><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></code> from package 'stats'.</p></dd>


<dt>center</dt>
<dd><p>Same as in <code><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></code> from package 'stats'.</p></dd>


<dt>tol</dt>
<dd><p>Same as in <code><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></code> from package 'stats'.</p></dd>


<dt>method</dt>
<dd><p>Same as in <code><a href="https://rdrr.io/pkg/MASS/man/qda.html" class="external-link">qda</a></code> from package 'MASS'.</p></dd>


<dt>max.pc</dt>
<dd><p>Same as in parameter 'rank.' from
<code><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></code> from package 'stats'.</p></dd>


<dt>object</dt>
<dd><p>To use with function 'predict'. A 'pcaQDA' object containing a
list of two objects: 1) an object of class inheriting from 'qda' and 2)
an object of class inheriting from 'prcomp'.</p></dd>


<dt>newdata</dt>
<dd><p>To use with function 'predict'. New data for classification
prediction.</p></dd>


<dt>type</dt>
<dd><p>To use with function 'predict'. The type of prediction
required. The default is 'all' basic predictions: classes and posterior
classification probabilities. Option 'qda.pred' returns the
object given by function 'predict.qda' from MASS package: 'class',
'posterior', 'scores' (cases scores on discriminant variables,
see <code><a href="https://rdrr.io/pkg/MASS/man/qda.html" class="external-link">qda</a></code>.</p></dd>


<dt>...</dt>
<dd><p>Not in use.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Function 'pcaQDA' returns an object ('pcaQDA') consisting of a list with
two objects:</p><ol><li><p>'qda': an object of class <code><a href="https://rdrr.io/pkg/MASS/man/qda.html" class="external-link">qda</a></code> from package
'MASS'.</p></li>
<li><p>'pca': an object of class <code><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></code> from package
'stats'.</p></li>
</ol><p>For information on how to use these objects see ?qda and ?prcomp.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The principal components (PCs) are obtained using the function
'prcomp' from R package 'stats', while the qda is performed using the
'qda' function from R package 'MASS'. The current application only uses
basic functionalities of mentioned functions. As shown in the example,
'pcaQDA' function can be used in general classification problems.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="pcaLDA.html">pcaLDA</a></code>, <code><a href="https://rdrr.io/pkg/MASS/man/qda.html" class="external-link">qda</a></code> and
<code><a href="https://rdrr.io/pkg/MASS/man/predict.lda.html" class="external-link">predict.lda</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">iris</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">qd1</span> <span class="op">&lt;-</span> <span class="fu">pcaQDA</span><span class="op">(</span>formula <span class="op">=</span> <span class="va">Species</span> <span class="op">~</span> <span class="va">Petal.Length</span> <span class="op">+</span> <span class="va">Sepal.Length</span> <span class="op">+</span> <span class="va">Sepal.Width</span>,</span></span>
<span class="r-in"><span>data <span class="op">=</span> <span class="va">iris</span>, n.pc <span class="op">=</span> <span class="fl">1</span>, max.pc <span class="op">=</span> <span class="fl">2</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>, center <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">## === Prediction === ##</span></span></span>
<span class="r-in"><span><span class="va">qd2</span> <span class="op">&lt;-</span> <span class="fu">pcaQDA</span><span class="op">(</span>formula <span class="op">=</span> <span class="va">Species</span> <span class="op">~</span><span class="va">.</span>, data <span class="op">=</span> <span class="va">iris</span>, n.pc <span class="op">=</span> <span class="fl">1</span>, max.pc <span class="op">=</span> <span class="fl">2</span>,</span></span>
<span class="r-in"><span>scale <span class="op">=</span> <span class="cn">TRUE</span>, center <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Set a random seed to reproduce the sampling</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample.int</a></span><span class="op">(</span><span class="fl">150</span>, <span class="fl">40</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">newdata</span> <span class="op">&lt;-</span> <span class="va">iris</span><span class="op">[</span><span class="va">idx</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">newdata.prediction</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">qd2</span>, newdata <span class="op">=</span> <span class="va">newdata</span>, type <span class="op">=</span> <span class="st">'all'</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The confusion matrix</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>TRUE.class <span class="op">=</span> <span class="va">iris</span><span class="op">$</span><span class="va">Species</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>PRED.class <span class="op">=</span> <span class="va">newdata.prediction</span><span class="op">$</span><span class="va">class</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             PRED.class</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TRUE.class   setosa versicolor virginica</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   setosa         12          0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   versicolor      0         12         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   virginica       0          1        13</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

