<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Parallel Job Split — pjob_split • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Parallel Job Split — pjob_split"><meta property="og:description" content="This function is addressed to set parallel jobs in a multicore
server in analogous way as in a computer cluster. The purpose is to prevent
nested parallel loops. So, each job can be able to implement independent
parallel computation.
This function will be useful when working with methylomes from species with
large chromosomes, e.g., human methylomes, where the computation must be
done by chromosome. So, 'pjob_split' will permit the application of a
'divide-and-conquer' strategy.
The maximum number of cores in use will be the number of screen sessions
multiplied by the number of cores used by the functions working in the
execution of the file given in 'filename' argument."><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Parallel Job Split</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/pjob_split.R" class="external-link"><code>R/pjob_split.R</code></a></small>
    <div class="hidden name"><code>pjob_split.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function is addressed to set parallel jobs in a multicore
server in analogous way as in a computer cluster. The purpose is to prevent
nested parallel loops. So, each job can be able to implement independent
parallel computation.</p>
<p>This function will be useful when working with methylomes from species with
large chromosomes, e.g., human methylomes, where the computation must be
done by chromosome. So, 'pjob_split' will permit the application of a
'divide-and-conquer' strategy.</p>
<p>The maximum number of cores in use will be the number of screen sessions
multiplied by the number of cores used by the functions working in the
execution of the file given in 'filename' argument.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">pjob_split</span><span class="op">(</span><span class="va">filename</span>, num.jobs <span class="op">=</span> <span class="fl">1L</span>, args <span class="op">=</span> <span class="cn">NULL</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>filename</dt>
<dd><p>The path to the file where the script is found.</p></dd>


<dt>num.jobs</dt>
<dd><p>An integer. The number of jobs.</p></dd>


<dt>args</dt>
<dd><p>Additional arguments to pass to the script function. If NULL,
still the job index is passed and users can use it in the downstream
analysis.</p></dd>


<dt>verbose</dt>
<dd><p>If TRUE it will report the progress of the job.</p></dd>

</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>This function is set only for Linux OS, so far.
The user must check the status of its job in a Linux terminal. For example,
typing 'top' or 'htop', and searching for R job linked to their user names.</p>
<p>The Linux 'screen' app must be installed. Users can create their own
scripts and must test whether the script is working or not before start any
big computation.</p>
<p>For further details, please, the example.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Save the script below in some place, say, '~/script4pjob_split.R'</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="co">## Argument passed from the system terminal to the script execution</span></span></span>
<span class="r-in"><span>    <span class="co">## In the current case chromosomes</span></span></span>
<span class="r-in"><span>    <span class="va">arg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/commandArgs.html" class="external-link">commandArgs</a></span><span class="op">(</span>trailingOnly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">k</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="va">arg</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="va">arg</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/MethylIT" class="external-link">MethylIT</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="co">## Most of the time chromosome names are just numbers (as frequentely</span></span></span>
<span class="r-in"><span>    <span class="co">## found in Arabidopsis). In human there are numbers and letters ('X'</span></span></span>
<span class="r-in"><span>    <span class="co">## and 'Y')</span></span></span>
<span class="r-in"><span>    <span class="va">chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr3"</span>, <span class="st">"chr4"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="co">## Loading information divergence dataset provided with MethylIT</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">hdiv</span><span class="op">)</span> <span class="co"># This would be just the path to a folder</span></span></span>
<span class="r-in"><span>    <span class="va">paths</span> <span class="op">&lt;-</span> <span class="st">"~"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Weibull2P"</span>, <span class="st">"Gamma2P"</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="kw">for</span> <span class="op">(</span><span class="va">h</span> <span class="kw">in</span> <span class="va">k</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>        <span class="co">## A plausible example:</span></span></span>
<span class="r-in"><span>        <span class="co">## load(paste0(paths, "/filename.RData"))</span></span></span>
<span class="r-in"><span>        <span class="co">## -------------------------------------</span></span></span>
<span class="r-in"><span>        <span class="co">## Here, we want to compute the best fitted model per chromosome.</span></span></span>
<span class="r-in"><span>        <span class="co">## In the current case, we assume that each sample come from</span></span></span>
<span class="r-in"><span>        <span class="co">## Arabidopsis with five chromosomes.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>        <span class="co">## Select chromosome 'h'</span></span></span>
<span class="r-in"><span>        <span class="va">hd</span> <span class="op">&lt;-</span> <span class="fu"><a href="lapply.html">lapply</a></span><span class="op">(</span><span class="va">hdiv</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>            <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">x</span>, pruning.mode <span class="op">=</span> <span class="st">"coarse"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">chr</span><span class="op">[</span><span class="va">h</span><span class="op">]</span></span></span>
<span class="r-in"><span>            <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></span>
<span class="r-in"><span>        <span class="op">}</span>, keep.attr <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>        <span class="va">gof_jd</span> <span class="op">&lt;-</span> <span class="fu"><a href="gofReport.html">gofReport</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>            HD <span class="op">=</span> <span class="va">hd</span>,</span></span>
<span class="r-in"><span>            model <span class="op">=</span> <span class="va">d</span>,</span></span>
<span class="r-in"><span>            column <span class="op">=</span> <span class="fl">10L</span>,</span></span>
<span class="r-in"><span>            num.cores <span class="op">=</span> <span class="fl">10L</span>,</span></span>
<span class="r-in"><span>            alt_models <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>            r.cv <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>            output <span class="op">=</span> <span class="st">"all"</span>,</span></span>
<span class="r-in"><span>            verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>        <span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">gof_jd</span>,</span></span>
<span class="r-in"><span>            file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">paths</span>, <span class="st">"/gof_jd_test_"</span>, <span class="va">chr</span><span class="op">[</span><span class="va">h</span><span class="op">]</span>,<span class="st">".RData"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="op">}</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Next, make sure that your script works by typing in a</span></span></span>
<span class="r-in"><span><span class="co">## terminal: 'Rscript ~/script4pjob_split.R 1'. Next, run:</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span> <span class="fu">pjob_split</span><span class="op">(</span> filename <span class="op">=</span> <span class="st">"~/script4pjob_split.R"</span>,</span></span>
<span class="r-in"><span>             args <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>             num.jobs <span class="op">=</span> <span class="fl">4L</span>,</span></span>
<span class="r-in"><span>             verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span><span class="co">## Four 'RData' files with the names 'gof_jd_test_chr[i].RData'</span></span></span>
<span class="r-in"><span><span class="co">## (i= 1,...4) must be created.</span></span></span>
<span class="r-in"><span><span class="co">## If the computation is taken enough time, then in the terminal you can</span></span></span>
<span class="r-in"><span><span class="co">## type: 'screen -r' to see the computations taking place. If all the</span></span></span>
<span class="r-in"><span><span class="co">## sessions are still working, then you will see them named as:</span></span></span>
<span class="r-in"><span><span class="co">## PID.s18 to PID.s23, where PID is process id. A specific screen session</span></span></span>
<span class="r-in"><span><span class="co">## can be retrieved typing, e.g.: 'screen -r PID.s18'.</span></span></span>
<span class="r-in"><span><span class="co">## To remove the saved files:</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span> <span class="va">k</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="va">job.start</span>, <span class="va">job.end</span>, <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span> <span class="kw">for</span> <span class="op">(</span><span class="va">h</span> <span class="kw">in</span> <span class="va">k</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">paths</span>, <span class="st">"/gof_jd_test_"</span>, <span class="va">chr</span><span class="op">[</span><span class="va">h</span><span class="op">]</span>, <span class="st">".RData"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span> <span class="op">}</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

