<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Prediction Method of Random Forest for 'pDMP' Objects — predict • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Prediction Method of Random Forest for 'pDMP' Objects — predict"><meta property="og:description" content="This function applies
predict.randomForest to a 'pDMP' object
provided in the argument \(newdata\)."><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.6.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Prediction Method of Random Forest for 'pDMP' Objects</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/predict.R" class="external-link"><code>R/predict.R</code></a></small>
    <div class="hidden name"><code>predict.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function applies
<code><a href="https://rdrr.io/pkg/randomForest/man/predict.randomForest.html" class="external-link">predict.randomForest</a></code> to a 'pDMP' object
provided in the argument \(newdata\).</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">predict</span><span class="op">(</span><span class="va">object</span>, newdata <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for RandomForest,GRanges</span></span>
<span><span class="fu">predict</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  <span class="va">newdata</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"class"</span>, <span class="st">"response"</span>, <span class="st">"prob"</span>, <span class="st">"votes"</span><span class="op">)</span>,</span>
<span>  predict.all <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  keep.data <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for RandomForest,pDMP</span></span>
<span><span class="fu">predict</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  <span class="va">newdata</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"class"</span>, <span class="st">"response"</span>, <span class="st">"prob"</span>, <span class="st">"votes"</span><span class="op">)</span>,</span>
<span>  predict.all <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  keep.data <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for randomForestformula,GRanges</span></span>
<span><span class="fu">predict</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  <span class="va">newdata</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"class"</span>, <span class="st">"response"</span>, <span class="st">"prob"</span>, <span class="st">"votes"</span><span class="op">)</span>,</span>
<span>  predict.all <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  keep.data <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for randomForestformulaList,GRangesList</span></span>
<span><span class="fu">predict</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  <span class="va">newdata</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"class"</span>, <span class="st">"response"</span>, <span class="st">"prob"</span>, <span class="st">"votes"</span><span class="op">)</span>,</span>
<span>  predict.all <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  keep.data <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>object</dt>
<dd><p>an object of 'RandomForest-class', as that created by the
function <code><a href="evaluateDIMPclass.html">evaluateDIMPclass</a></code>. If \(newdata\) is a
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code>, then 'newdata' must be an
element of a pDMP-class object, i.e., it must have the same structure as
the GRanges-class elements in a pDMP-class object.</p></dd>


<dt>newdata</dt>
<dd><p>A 'pDMP' object generated with function
<code><a href="selectDIMP.html">selectDIMP</a></code>.</p></dd>


<dt>predict.all, type, ...</dt>
<dd><p>The same as in
<code><a href="https://rdrr.io/pkg/randomForest/man/predict.randomForest.html" class="external-link">predict.randomForest</a></code>.</p></dd>


<dt>keep.data</dt>
<dd><p>Logical. Whether to preserve the original class
from object \('newdata'\). If TRUE, then the predicted class and
the posterior probability of the treatment class are added as a
metacolumns of \('newdata'\).</p></dd>


<dt>num.cores, tasks</dt>
<dd><p>Parameters for parallel computation using package
<code><a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallel-package.html" class="external-link">BiocParallel-package</a></code>: the number of cores to
use, i.e. at most how many child processes will be run simultaneously (see
<code><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></code> and the number of tasks per job (only
for Linux OS).</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>The same results as given by function
<code><a href="https://rdrr.io/pkg/randomForest/man/predict.randomForest.html" class="external-link">predict.randomForest</a></code>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The generic function just call function
<code><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></code> from 'stats' R package.</p>
<p>If \(newdata\) is a <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code>, then
'newdata' must be an element of a pDMP-class object, i.e., it must have
the same structure as the GRanges-class elements in a pDMP-class object.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Robersy Sanchez <a href="https://genomaths.com" class="external-link">https://genomaths.com</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Load a DMP data set</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">dmps</span>, package <span class="op">=</span> <span class="st">'MethylIT'</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Let's accomplish the classification by using Random Forest</span></span></span>
<span class="r-in"><span><span class="co">## algorithm</span></span></span>
<span class="r-in"><span><span class="va">perf</span> <span class="op">&lt;-</span> <span class="fu"><a href="evaluateDIMPclass.html">evaluateDIMPclass</a></span><span class="op">(</span>LR <span class="op">=</span> <span class="va">dmps</span>,</span></span>
<span class="r-in"><span>                          column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>hdiv <span class="op">=</span> <span class="cn">TRUE</span>, TV <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                                     wprob <span class="op">=</span> <span class="cn">TRUE</span>, pos <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                          classifier <span class="op">=</span> <span class="st">'random_forest'</span>,</span></span>
<span class="r-in"><span>                          n.pc <span class="op">=</span> <span class="fl">4L</span>,</span></span>
<span class="r-in"><span>                          control.names <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'C1'</span>, <span class="st">'C2'</span>, <span class="st">'C3'</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                          treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'T1'</span>, <span class="st">'T2'</span>, <span class="st">'T3'</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                          center <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                          scale <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                          prop <span class="op">=</span> <span class="fl">0.6</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">predict</span><span class="op">(</span>object <span class="op">=</span> <span class="va">perf</span><span class="op">$</span><span class="va">model</span>, newdata <span class="op">=</span> <span class="va">dmps</span>, keep.data <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pDMP object of length: 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $C1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 45 ranges and 13 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        seqnames    ranges strand |        c1        t1        c2        t2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]        1       233      - |        40       122       162         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]        1       732      + |        50       154       200         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]        1      1045      - |        45       115       160         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]        1      1409      - |        44       132       176         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]        1      1739      - |        48       145       193         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...      ...       ...    ... .       ...       ...       ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [41]        1      8945      - |        37       111       148         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [42]        1      9205      + |        47       144       188         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [43]        1      9238      - |       156        55         0       211</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [44]        1      9249      + |        52       138       187         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [45]        1      9383      - |        40       122       162         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               p1         p2        TV    bay.TV      hdiv      jdiv       wprob</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]  0.245757   0.989442  0.753086  0.743685   136.153   3.03720 0.001431041</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]  0.244258   0.972183  0.735294  0.727925   150.759   2.45917 0.000747885</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]  0.277937   0.989312  0.718750  0.711375   124.864   2.81337 0.002371176</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]  0.248754   0.990272  0.750000  0.741518   148.040   3.06399 0.000843622</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]  0.247631   0.991120  0.751295  0.743489   164.066   3.12473 0.000415743</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...       ...        ...       ...       ...       ...       ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [41]  0.248534 0.98845689  0.750000  0.739923   122.543   2.96573 0.002631713</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [42]  0.245127 0.97548911  0.738220  0.730362   143.991   2.53338 0.001009617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [43]  0.714772 0.00156955 -0.739336 -0.713202   183.531   3.79371 0.000177204</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [44]  0.271268 0.97536149  0.710526  0.704094   134.321   2.37020 0.001552899</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [45]  0.245757 0.98944187  0.753086  0.743685   136.153   3.03720 0.001431041</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           class posterior</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;factor&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]       CT    0.1100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]       CT    0.0050</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]       CT    0.0725</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]       CT    0.0275</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]       CT    0.0050</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...      ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [41]       CT    0.1225</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [42]       CT    0.0325</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [43]       CT    0.0300</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [44]       CT    0.0550</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [45]       CT    0.1075</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;5 more GRanges elements&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ------- </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

