<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Predict DIMP class — predictDIMPclass • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Predict DIMP class — predictDIMPclass"><meta property="og:description" content="This function classify each DMP as a control or a treatment
DMP"><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Predict DIMP class</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/predictDIMPclass.R" class="external-link"><code>R/predictDIMPclass.R</code></a></small>
    <div class="hidden name"><code>predictDIMPclass.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function classify each DMP as a control or a treatment
DMP</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">predictDIMPclass</span><span class="op">(</span></span>
<span>  <span class="va">LR</span>,</span>
<span>  <span class="va">model</span>,</span>
<span>  conf.matrix <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  control.names <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  treatment.names <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>LR</dt>
<dd><p>A list of GRanges objects obtained through the through MethylIT
downstream analysis. Basically, this object is a list of GRanges containing
only differentially methylated position (DMPs). The metacolumn of each
GRanges must contain the column: Hellinger divergence 'hdiv', total
variation 'TV', the probability of potential DMP 'wprob', which naturally
are added in the downstream analysis of MethylIT.</p></dd>


<dt>model</dt>
<dd><p>A classifier model obtained with the function
'evaluateDIMPclass'.</p></dd>


<dt>conf.matrix</dt>
<dd><p>Optional. Logic, whether a confusion matrix should be
returned (default, FALSE, see below).</p></dd>


<dt>control.names</dt>
<dd><p>Optional. Names/IDs of the control samples, which must
be include in the variable LR (default, NULL).</p></dd>


<dt>treatment.names</dt>
<dd><p>Optional. Names/IDs of the treatment samples, which
must be include in the variable LR (default, NULL).</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>The same LR object with tow new columns named 'class' and 'posterior'
added to each GRanges object from LR (default). Based on the model
prediction each DMP is labeled as control 'CT' or as treatment 'TT' in
column 'class'. Column 'posterior' provides, for each DMP, the posterior
probability that the given DMP can be classified as induced by the
'treatment' (a treatment DMP).</p>


<p>Control DMPs classified as 'treatment' are false positives. However, if the
same cytosine position is classified as 'treatment DMP' in both groups,
control and treatment, but with higher posterior probability in the
treatment group, then this would indicate a reinforcement of the methylation
status in such a position induced by the treatment.</p>


<p>If 'conf.matrix' is TRUE and the arguments control.names and treatment.names
are provided, then the overall confusion matrix is returned.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Predictions only makes sense if the query DMPs belong to same
methylation context and derive from an experiment accomplished under the
same condition set for the DMPs used to build the model.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### Load dataset from the package</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">logit_perf</span>, <span class="va">dmps</span>, package <span class="op">=</span> <span class="st">'MethylIT'</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### Select a random subset (70%) from each DMP sample</span></span></span>
<span class="r-in"><span><span class="va">DMPs</span> <span class="op">&lt;-</span> <span class="fu"><a href="lapply.html">lapply</a></span><span class="op">(</span><span class="va">dmps</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>            <span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">*</span> <span class="fl">0.7</span></span></span>
<span class="r-in"><span>            <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample.int</a></span><span class="op">(</span><span class="va">idx</span><span class="op">)</span>  <span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="op">}</span>, keep.attr <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### To accomplish the prediction for logistic model</span></span></span>
<span class="r-in"><span><span class="va">predclass.dmps</span> <span class="op">&lt;-</span> <span class="fu">predictDIMPclass</span><span class="op">(</span></span></span>
<span class="r-in"><span>    LR <span class="op">=</span> <span class="va">DMPs</span>,</span></span>
<span class="r-in"><span>    model <span class="op">=</span> <span class="va">logit_perf</span><span class="op">$</span><span class="va">model</span>,</span></span>
<span class="r-in"><span>    conf.matrix <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    control.names <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'C1'</span>, <span class="st">'C2'</span>, <span class="st">'C3'</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'T1'</span>, <span class="st">'T2'</span>, <span class="st">'T3'</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found more than one class "pDMP" in cache; using the first, from namespace 'MethylIT'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Also defined by ‘MethylIT.utils’</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">predclass.dmps</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Confusion Matrix and Statistics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           Reference</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Prediction   CT   TT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         CT   91 1066</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         TT    0    0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                Accuracy : 0.0787          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  95% CI : (0.0638, 0.0957)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     No Information Rate : 0.9213          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     P-Value [Acc &gt; NIR] : 1               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   Kappa : 0               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Mcnemar's Test P-Value : &lt;2e-16          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             Sensitivity : 0.00000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             Specificity : 1.00000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          Pos Pred Value :     NaN         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          Neg Pred Value : 0.07865         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              Prevalence : 0.92135         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          Detection Rate : 0.00000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    Detection Prevalence : 0.00000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       Balanced Accuracy : 0.50000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        'Positive' Class : TT              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### To accomplish the prediction PCA-QDA model</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"pcaQda_perf"</span>, package <span class="op">=</span> <span class="st">'MethylIT'</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">predclass.dmps</span> <span class="op">&lt;-</span> <span class="fu">predictDIMPclass</span><span class="op">(</span></span></span>
<span class="r-in"><span>    LR <span class="op">=</span> <span class="va">DMPs</span>,</span></span>
<span class="r-in"><span>    model <span class="op">=</span> <span class="va">logit_perf</span><span class="op">$</span><span class="va">model</span>,</span></span>
<span class="r-in"><span>    conf.matrix <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    control.names <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'C1'</span>, <span class="st">'C2'</span>, <span class="st">'C3'</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'T1'</span>, <span class="st">'T2'</span>, <span class="st">'T3'</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">predclass.dmps</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Confusion Matrix and Statistics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           Reference</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Prediction   CT   TT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         CT   91 1066</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         TT    0    0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                Accuracy : 0.0787          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  95% CI : (0.0638, 0.0957)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     No Information Rate : 0.9213          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     P-Value [Acc &gt; NIR] : 1               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   Kappa : 0               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Mcnemar's Test P-Value : &lt;2e-16          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             Sensitivity : 0.00000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             Specificity : 1.00000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          Pos Pred Value :     NaN         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          Neg Pred Value : 0.07865         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              Prevalence : 0.92135         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          Detection Rate : 0.00000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    Detection Prevalence : 0.00000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       Balanced Accuracy : 0.50000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        'Positive' Class : TT              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### To accomplish the prediction PCA-LDA model</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"pcaLda_perf"</span>, package <span class="op">=</span> <span class="st">'MethylIT'</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">predclass.dmps</span> <span class="op">&lt;-</span> <span class="fu">predictDIMPclass</span><span class="op">(</span></span></span>
<span class="r-in"><span>    LR <span class="op">=</span> <span class="va">DMPs</span>,</span></span>
<span class="r-in"><span>    model <span class="op">=</span> <span class="va">logit_perf</span><span class="op">$</span><span class="va">model</span>,</span></span>
<span class="r-in"><span>    conf.matrix <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    control.names <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'C1'</span>, <span class="st">'C2'</span>, <span class="st">'C3'</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    treatment.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'T1'</span>, <span class="st">'T2'</span>, <span class="st">'T3'</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">predclass.dmps</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Confusion Matrix and Statistics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           Reference</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Prediction   CT   TT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         CT   91 1066</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         TT    0    0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                Accuracy : 0.0787          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  95% CI : (0.0638, 0.0957)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     No Information Rate : 0.9213          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     P-Value [Acc &gt; NIR] : 1               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   Kappa : 0               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Mcnemar's Test P-Value : &lt;2e-16          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             Sensitivity : 0.00000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             Specificity : 1.00000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          Pos Pred Value :     NaN         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          Neg Pred Value : 0.07865         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              Prevalence : 0.92135         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          Detection Rate : 0.00000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    Detection Prevalence : 0.00000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       Balanced Accuracy : 0.50000         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        'Positive' Class : TT              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                           </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

