<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Read files of methylation count tables — readCounts2GRangesList • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Read files of methylation count tables — readCounts2GRangesList"><meta property="og:description" content="This function is addressed to read files with methylation count
table data commonly generated after the alignment of BS-seq data or found in
GEO database"><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Read files of methylation count tables</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/readCounts2GRangesList.R" class="external-link"><code>R/readCounts2GRangesList.R</code></a></small>
    <div class="hidden name"><code>readCounts2GRangesList.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function is addressed to read files with methylation count
table data commonly generated after the alignment of BS-seq data or found in
GEO database</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">readCounts2GRangesList</span><span class="op">(</span></span>
<span>  filenames <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sample.id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pattern <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  remove <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="cn">NULL</span>, start <span class="op">=</span> <span class="cn">NULL</span>, end <span class="op">=</span> <span class="cn">NULL</span>, strand <span class="op">=</span> <span class="cn">NULL</span>, fraction <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    percent <span class="op">=</span> <span class="cn">NULL</span>, mC <span class="op">=</span> <span class="cn">NULL</span>, uC <span class="op">=</span> <span class="cn">NULL</span>, coverage <span class="op">=</span> <span class="cn">NULL</span>, context <span class="op">=</span> <span class="cn">NULL</span>, signal <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>,</span>
<span>  chromosome.names <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  chromosomes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>filenames</dt>
<dd><p>Character vector with the file names</p></dd>


<dt>sample.id</dt>
<dd><p>Character vector with the names of the samples
corresponding to each file</p></dd>


<dt>pattern</dt>
<dd><p>Chromosome name pattern. Users working on Linux OS can
specify the reading of specific lines from each file by using regular
expressions.</p></dd>


<dt>remove</dt>
<dd><p>Logic (TRUE). Usually the supplementary files from GEO
datasets are 'gz' compressed. File datasets must be decompressed to be
read. The decompressed files are removed after read if this is set
'TRUE'.</p></dd>


<dt>columns</dt>
<dd><p>Vector of integer numbers denoting the table columns that
must be read. The numbers for 'seqnames' (chromosomes), 'start', and 'end'
(if different from 'start') columns must be given. The possible fields are:
'seqnames' (chromosomes),'start', 'end', 'strand', 'fraction', percent'
(methylation percentage), 'mC' (methylates cytosine), 'uC' (non methylated
cytosine), 'coverage', and 'context' (methylation context). These column
headers are not required to be in the files. An optional column named
<em>'signal'</em> can be used to include a relevant information about the
methylation signal.</p></dd>


<dt>chromosome.names</dt>
<dd><p>If provided, for each GRanges object, chromosome
names will be changed to those provided in 'chromosome.names' applying
seqlevels(x) &lt;- chromosome.names'. This option permits to use all the
functionality of the function 'seqlevels' defined from package
'GenomeInfoDb', which rename, add, and reorder the seqlevels all at once (see
?seqlevels).</p></dd>


<dt>chromosomes</dt>
<dd><p>If provided, it must be a character vector with the names
of the chromosomes that you want to include in the final GRanges objects.</p></dd>


<dt>verbose</dt>
<dd><p>If TRUE, prints the function log to stdout</p></dd>


<dt>...</dt>
<dd><p>Additional parameters for 'fread' function from 'data.table'
package</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A list of GRanges objects</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Read tables from files with a table methylation count data using
the function fread from the package 'data.table' and and yields a list of
GRanges objects with the information provided.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Create a cov file with it's file name including 'gz'</span></span></span>
<span class="r-in"><span><span class="co">## 'gz' (tarball extension)</span></span></span>
<span class="r-in"><span><span class="va">filename</span> <span class="op">&lt;-</span> <span class="st">'./file.cov'</span></span></span>
<span class="r-in"><span><span class="va">gr1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'chr1'</span>, <span class="st">'chr1'</span><span class="op">)</span>, post <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'+'</span>, <span class="st">'-'</span><span class="op">)</span>, ratio <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.9</span>, <span class="fl">0.5</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                context <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'CG'</span>, <span class="st">'CG'</span><span class="op">)</span>, CT <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">20</span>, <span class="fl">30</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">filename</span> <span class="op">&lt;-</span> <span class="st">'./file.cov'</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">gr1</span><span class="op">)</span>, file <span class="op">=</span> <span class="va">filename</span>,</span></span>
<span class="r-in"><span>            col.names <span class="op">=</span> <span class="cn">TRUE</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span>, quote <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Read the file. It does not work. Typing mistake: 'fractions'</span></span></span>
<span class="r-in"><span><span class="va">LR</span> <span class="op">&lt;-</span> <span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="fu">readCounts2GRangesList</span><span class="op">(</span>filenames <span class="op">=</span> <span class="va">filename</span>, remove <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                            sample.id <span class="op">=</span> <span class="st">'test'</span>,</span></span>
<span class="r-in"><span>                            columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fl">1</span>, start <span class="op">=</span> <span class="fl">2</span>,</span></span>
<span class="r-in"><span>                                    strand <span class="op">=</span> <span class="fl">3</span>, fractions <span class="op">=</span> <span class="fl">4</span>,</span></span>
<span class="r-in"><span>                                    context <span class="op">=</span> <span class="fl">5</span>, coverage <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                                    silent <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="va">filename</span><span class="op">)</span> <span class="co"># Remove the file</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

