<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Selection of DMPs — selectDIMP • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Selection of DMPs — selectDIMP"><meta property="og:description" content="For a given cutpoint (previously estimated with the function
estimateCutPoint), 'selectDIMP' will return the differentially
informative methylated positions (DMPs). DMPs are cytosine positions
for which the divergence is greater than the cutpoint."><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Selection of DMPs</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/selectDIMP.R" class="external-link"><code>R/selectDIMP.R</code></a></small>
    <div class="hidden name"><code>selectDIMP.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>For a given cutpoint (previously estimated with the function
estimateCutPoint), 'selectDIMP' will return the differentially
informative methylated positions (DMPs). DMPs are cytosine positions
for which the divergence is greater than the cutpoint.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">selectDIMP</span><span class="op">(</span></span>
<span>  <span class="va">LR</span>,</span>
<span>  div.col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pval.col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  absolute <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">cutpoint</span>,</span>
<span>  tv.col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tv.cut <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for InfDiv</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for pDMP</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for testDMP</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>LR</dt>
<dd><p>An object from 'pDMP' class.</p></dd>


<dt>div.col</dt>
<dd><p>Number of the column where the divergence variable (i.e.,
Hellinger divergence or total variation) is located in the GRanges
meta-columns.</p></dd>


<dt>pval.col</dt>
<dd><p>If the cutpoints is a p-value, then the column number for
p-values should be provided. Default: NULL. Notice that one of the
parameter values div.col or pval.col must be given.</p></dd>


<dt>absolute</dt>
<dd><p>Logic (default, FALSE). Total variation (TV, the difference
of methylation levels) is normally an output in the downstream MethylIT
analysis. If 'absolute = TRUE', then TV is transformed into
\(|TV|\), which is an information divergence that can be fitted to
Weibull or to Generalized Gamma distribution. So, if the nonlinear fit
was performed for \(|TV|\), then here absolute must be set to TRUE.</p></dd>


<dt>cutpoint</dt>
<dd><p>Cutpoint to select DMPs. Cytosine positions with divergence
greater than 'cutpoint' will selected as DMPs. Cutpoints are estimated
with the function 'estimateCutPoint'.</p></dd>


<dt>tv.col</dt>
<dd><p>Column number for the total variation to be used for filtering
cytosine positions (if provided).</p></dd>


<dt>tv.cut</dt>
<dd><p>If tv.cut and tv.col are provided, then cytosine sites \(k\)
with \(|TV| &lt; tv.cut\) are removed.</p></dd>


<dt>x</dt>
<dd><p>An object from 'pDMP_OR_InfDiv' class</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>An object from 'pDMP' class containing only differentially
informative position (DMPs).</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Theoretically a DMP denotes a cytosine position with high
probability to be differentially methylated. That is, in the statistical
molecular-biophysics context, a DMP must be considered only in a
probabilistic term and not as an absolute deterministic experimental output.</p>
<p>The uncertainty and dynamics of the DNA methylation process, the continuous
action of the omnipresent thermal fluctuations, as well as, the inherent
stochasticity of the biochemical reactions make it impossible to ensure
whether a specific cytosine position is methylated in an absolutely
deterministic sense. Notice that the concept of DMP is not applicable to a
single cell (if we use an instrumentation/protocol to directly measure
methylation at the molecular level, and not via PCR), since a concrete,
single DNA cytosine position in a single cell is methylated or not
methylated.</p>
<p>However, when pooling DNA extracted from a tissue, the previous reasonings
about uncertainty hold plus an additional uncertainty factor: cells from the
same tissue are not synchronized but are found in different stages of their
ontogenetic developments. Hence, the DMP concept holds in the mentioned
circumstances where the uncertainty of methylation is present.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Get a dataset of potential signals and the estimated cutpoint from the</span></span></span>
<span class="r-in"><span><span class="co">## package</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PS</span>, <span class="va">cutpoint</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The estimated cutpoints are used to discriminate signals from the noise.</span></span></span>
<span class="r-in"><span><span class="co">## That is, DMPs are selected using the cupoints</span></span></span>
<span class="r-in"><span><span class="va">DMPs</span> <span class="op">&lt;-</span> <span class="fu">selectDIMP</span><span class="op">(</span><span class="va">PS</span>, div.col <span class="op">=</span> <span class="fl">9L</span>, cutpoint <span class="op">=</span> <span class="va">cutpoint</span><span class="op">$</span><span class="va">cutpoint</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

