<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Unique genomic ranges from a list of GRanges objects — uniqueGRanges • MethylIT</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Unique genomic ranges from a list of GRanges objects — uniqueGRanges" />
<meta property="og:description" content="Build an unique GRanges object from a list of Granges objects." />
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/MethylIT/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Unique genomic ranges from a list of GRanges objects</h1>
    <small class="dont-index">Source: <a href='https://github.com/genomaths/MethylIT/blob/master/R/uniqueGRanges.R'><code>R/uniqueGRanges.R</code></a></small>
    <div class="hidden name"><code>uniqueGRanges.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Build an unique GRanges object from a list of Granges objects.</p>
    </div>

    <pre class="usage"><span class='fu'>uniqueGRanges</span><span class='op'>(</span>
  <span class='va'>ListOfGranges</span>,
  ncols <span class='op'>=</span> <span class='cn'>NULL</span>,
  columns <span class='op'>=</span> <span class='cn'>NULL</span>,
  chromosomes <span class='op'>=</span> <span class='cn'>NULL</span>,
  maxgap <span class='op'>=</span> <span class='op'>-</span><span class='fl'>1L</span>,
  minoverlap <span class='op'>=</span> <span class='fl'>1L</span>,
  missing <span class='op'>=</span> <span class='fl'>0</span>,
  type <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"any"</span>, <span class='st'>"start"</span>, <span class='st'>"end"</span>, <span class='st'>"within"</span>, <span class='st'>"equal"</span><span class='op'>)</span>,
  select <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"all"</span>, <span class='st'>"first"</span>, <span class='st'>"last"</span>, <span class='st'>"arbitrary"</span><span class='op'>)</span>,
  ignore.strand <span class='op'>=</span> <span class='cn'>FALSE</span>,
  keep.strand <span class='op'>=</span> <span class='op'>!</span><span class='va'>ignore.strand</span>,
  num.cores <span class='op'>=</span> <span class='fl'>1</span>,
  tasks <span class='op'>=</span> <span class='fl'>0L</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ListOfGranges</th>
      <td><p>Objects to combine. A list of
<code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges-class</a></code> object or a
<code><a href='https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html'>GRangesList-class</a></code> object.</p></td>
    </tr>
    <tr>
      <th>ncols</th>
      <td><p>integer. Number of columns to use from the meta-column of each
GRanges object. Default value: NULL. If NULL, all the columns (from
column 1 to ncols) from each GRanges will be present in the
\(uniqueGRanges\) output.</p></td>
    </tr>
    <tr>
      <th>columns</th>
      <td><p>integer number(s) corresponding to the specific column(s) to
use from the meta-column of each GRanges. Default value: NULL. if
provided, the metacolumn from the uniqueGRanges output will contain the
specified columns.</p></td>
    </tr>
    <tr>
      <th>chromosomes</th>
      <td><p>Chromosomes used Default value: NULL</p></td>
    </tr>
    <tr>
      <th>maxgap, minoverlap, ignore.strand, select, type</th>
      <td><p>The same as in
<code><a href='https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html'>findOverlaps-methods</a></code>.</p></td>
    </tr>
    <tr>
      <th>missing</th>
      <td><p>A numerical value (default 0) or NA to write in ranges with
missing values. For example, suppose that we want to build a
uniqueGRanges object from the GRanges objects X and Y. If a given range
k from the GRanges object X with metacolumn value x is missing in the
GRanges object Y, then the metacolumn of range k from
uniqueGRanges(list(X,Y)) object will be the row vector (x,0) or (x,NA)
if missing = NA.</p></td>
    </tr>
    <tr>
      <th>keep.strand</th>
      <td><p>When set to TRUE, the strand information is preserved on
the objects even if ignore.strand is set to TRUE.  This makes it possible
to ignore the strand during overlap calculations but to preserve the
strand information and not overwrite with *. Default value is
keep.strand = !ignore.strand.</p></td>
    </tr>
    <tr>
      <th>num.cores</th>
      <td><p>The number of cores to use, i.e. at most how many child
processes will be run simultaneously (see bplapply function from
BiocParallel package).</p></td>
    </tr>
    <tr>
      <th>tasks</th>
      <td><p>integer(1). The number of tasks per job. value must be a scalar
integer &gt;= 0L. In this documentation a job is defined as a single call
to a function, such as bplapply, bpmapply etc. A task is the division of
the X argument into chunks. When tasks == 0 (default), X is divided as
evenly as possible over the number of workers (see MulticoreParam from
BiocParallel package).</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>if TRUE, prints the function log to stdout</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a GRanges object</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The metadata of each one of these GRanges must have one or more
columns to yield a unique GRanges object with metadata columns from the
original GRanges objects. Otherwise, a unique GRanges object will be
created  without metadata columns. Additionally, all metadata must be the
same class, e.g. all numeric or all characters, or all factor</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Robersy Sanchez (<a href='https://genomaths.com'>https://genomaths.com</a>).</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>dfChr1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>chr <span class='op'>=</span> <span class='st'>'chr1'</span>, start <span class='op'>=</span> <span class='fl'>11</span><span class='op'>:</span><span class='fl'>15</span>, end <span class='op'>=</span> <span class='fl'>11</span><span class='op'>:</span><span class='fl'>15</span>,
                    strand <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'+'</span>,<span class='st'>'-'</span>,<span class='st'>'+'</span>,<span class='st'>'*'</span>,<span class='st'>'.'</span><span class='op'>)</span>, score <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>5</span><span class='op'>)</span>
<span class='va'>dfChr2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>chr <span class='op'>=</span> <span class='st'>'chr1'</span>, start <span class='op'>=</span> <span class='fl'>11</span><span class='op'>:</span><span class='fl'>15</span>, end <span class='op'>=</span> <span class='fl'>11</span><span class='op'>:</span><span class='fl'>15</span>,
                    strand <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'+'</span>,<span class='st'>'-'</span>,<span class='st'>'+'</span>,<span class='st'>'*'</span>,<span class='st'>'.'</span><span class='op'>)</span>, score <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>5</span><span class='op'>)</span>
<span class='va'>dfChr3</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>chr <span class='op'>=</span> <span class='st'>'chr1'</span>, start <span class='op'>=</span> <span class='fl'>11</span><span class='op'>:</span><span class='fl'>15</span>, end <span class='op'>=</span> <span class='fl'>11</span><span class='op'>:</span><span class='fl'>15</span>,
                    strand <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'+'</span>,<span class='st'>'-'</span>,<span class='st'>'+'</span>,<span class='st'>'*'</span>,<span class='st'>'.'</span><span class='op'>)</span>, score <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>5</span><span class='op'>)</span>

<span class='va'>gr1</span> <span class='op'>&lt;-</span> <span class='fu'>makeGRangesFromDataFrame</span><span class='op'>(</span><span class='va'>dfChr1</span>, keep.extra.columns <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>gr2</span> <span class='op'>&lt;-</span> <span class='fu'>makeGRangesFromDataFrame</span><span class='op'>(</span><span class='va'>dfChr2</span>, keep.extra.columns <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>gr3</span> <span class='op'>&lt;-</span> <span class='fu'>makeGRangesFromDataFrame</span><span class='op'>(</span><span class='va'>dfChr3</span>, keep.extra.columns <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>

<span class='va'>grList</span> <span class='op'>&lt;-</span> <span class='fu'>GRangesList</span><span class='op'>(</span><span class='st'>'gr1'</span> <span class='op'>=</span> <span class='va'>gr1</span>, <span class='st'>'gr2'</span> <span class='op'>=</span> <span class='va'>gr2</span>, <span class='st'>'gr3'</span> <span class='op'>=</span> <span class='va'>gr3</span><span class='op'>)</span>

<span class='fu'>uniqueGRanges</span><span class='op'>(</span><span class='va'>grList</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'> *** Building coordinates for the new GRanges object ...</span></div><div class='output co'>#&gt; <span class='message'> *** Strand information is preserved ...</span></div><div class='output co'>#&gt; <span class='message'> *** Processing GRanges for sample: #1</span></div><div class='output co'>#&gt; <span class='message'> *** Processing GRanges for sample: #2...</span></div><div class='output co'>#&gt; <span class='message'> *** Processing GRanges for sample: #3...</span></div><div class='output co'>#&gt; <span class='message'> *** Sorting by chromosomes and start positions...</span></div><div class='output co'>#&gt; GRanges object with 5 ranges and 3 metadata columns:
#&gt;       seqnames    ranges strand |     score   score.1   score.2
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
#&gt;   [1]     chr1        11      + |         1         1         1
#&gt;   [2]     chr1        12      - |         2         2         2
#&gt;   [3]     chr1        13      + |         3         3         3
#&gt;   [4]     chr1        14      * |         4         4         4
#&gt;   [5]     chr1        15      * |         5         5         5
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


