<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Unique GRanges of methylation read counts filtered by coverages — uniqueGRfilterByCov • MethylIT</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Unique GRanges of methylation read counts filtered by coverages — uniqueGRfilterByCov" />
<meta property="og:description" content="Given two GRanges objects, samples '1' and '2', this function
will filter by coverage each cytosine site from each GRanges object." />
<meta property="og:image" content="https://genomaths.github.io/methylit/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/MethylIT/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Unique GRanges of methylation read counts filtered by coverages</h1>
    <small class="dont-index">Source: <a href='https://github.com/genomaths/MethylIT/blob/master/R/uniqueGRfilterByCov.R'><code>R/uniqueGRfilterByCov.R</code></a></small>
    <div class="hidden name"><code>uniqueGRfilterByCov.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given two GRanges objects, samples '1' and '2', this function
will filter by coverage each cytosine site from each GRanges object.</p>
    </div>

    <pre class="usage"><span class='fu'>uniqueGRfilterByCov</span><span class='op'>(</span>
  <span class='va'>x</span>,
  y <span class='op'>=</span> <span class='cn'>NULL</span>,
  min.coverage <span class='op'>=</span> <span class='fl'>4</span>,
  min.meth <span class='op'>=</span> <span class='fl'>0</span>,
  min.umeth <span class='op'>=</span> <span class='fl'>0</span>,
  min.sitecov <span class='op'>=</span> <span class='fl'>4</span>,
  percentile <span class='op'>=</span> <span class='fl'>0.9999</span>,
  high.coverage <span class='op'>=</span> <span class='cn'>NULL</span>,
  columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>mC <span class='op'>=</span> <span class='fl'>1</span>, uC <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span>,
  num.cores <span class='op'>=</span> <span class='fl'>1L</span>,
  ignore.strand <span class='op'>=</span> <span class='cn'>FALSE</span>,
  tasks <span class='op'>=</span> <span class='fl'>0L</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>,
  <span class='va'>...</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>An object from the classes 'GRanges', 'InfDiv', or 'pDMP' with
methylated and unmethylated counts in its meta-column. If the argument
'y' is not given, then it is assumed that the first four columns of the
GRanges metadata 'x' are counts: methylated and unmethylated counts for
samples '1' and '2'.</p></td>
    </tr>
    <tr>
      <th>y</th>
      <td><p>A GRanges object with methylated and unmethylated counts in its
meta-column. Default is NULL. If x is a 'InfDiv', or 'pDMP', then 'y' is
not needed, since samples '1' and '2' are the first four columns of
these objects.</p></td>
    </tr>
    <tr>
      <th>min.coverage</th>
      <td><p>An integer or an integer vector of length 2. Cytosine
sites where the coverage in both samples, 'x' and 'y', are less than
'min.coverage' are discarded. The cytosine site is preserved, however, if
the coverage is greater than 'min.coverage' in at least one sample. If
'min.coverage' is an integer vector, then the corresponding min coverage is
applied to each sample.</p></td>
    </tr>
    <tr>
      <th>min.meth</th>
      <td><p>An integer or an integer vector of length 2. Cytosine sites
where the numbers of read counts of methylated cytosine in both samples,
'1' and '2', are less than 'min.meth' are discarded. If 'min.meth' is an
integer vector, then the corresponding min number of reads is applied to
each sample.</p></td>
    </tr>
    <tr>
      <th>min.umeth</th>
      <td><p>An integer or an integer vector of length 2. Min number of
reads to consider cytosine position. Specifically cytosine positions
where (uC &lt;= min.umeth) &amp; (mC &gt; 0) &amp; (mC &lt; min.meth) hold will be
removed, where mC and uC stand for the numbers of methylated and
unmethylated reads. Default is min.umeth = 0.</p></td>
    </tr>
    <tr>
      <th>min.sitecov</th>
      <td><p>An integer. The minimum total coverage. Only sites where
the total coverage (cov1 + cov2) is greater than 'min.sitecov' are
considered for downstream analysis, where cov1 and cov2 are the coverages
for samples 1 and 2, respectively.</p></td>
    </tr>
    <tr>
      <th>percentile</th>
      <td><p>Threshold to remove the outliers (PCR bias) from each file
and all files stacked. If 'high.coverage = NULL', then the threshold
\(q\) will be computed as:
$$q1 = quantile(cov1, probs=percentile)$$
$$q2 = quantile(cov2, probs=percentile)$$
$$q = min(q1, q2)$$</p><pre>where \eqn{cov1} and \eqn{cov2} are the coverage vectors from samples 1
and 2, respectively.
</pre></td>
    </tr>
    <tr>
      <th>high.coverage</th>
      <td><p>An integer for read counts. Cytosine sites having
higher coverage than this are discarded. Default is NULL. If
<strong>high.coverage</strong> is not NULL, then the <strong>percentile</strong> argument is
disregarded and <strong>high.coverage</strong> is used as threshold to remove the
PCR bias. If percentile is not null, then q = max(q, high.coverage).</p></td>
    </tr>
    <tr>
      <th>columns</th>
      <td><p>Vector of integer numbers of the columns (from each GRanges
meta-column) where the methylated and unmethylated counts are provided.
If not provided, then the methylated and unmethylated counts are assumed
to be at columns 1 and 2, respectively.</p></td>
    </tr>
    <tr>
      <th>num.cores</th>
      <td><p>The number of cores to use, i.e. at most how many child
processes will be run simultaneously (see bplapply function from
BiocParallel package).</p></td>
    </tr>
    <tr>
      <th>ignore.strand</th>
      <td><p>When set to TRUE, the strand information is ignored in
the overlap calculations. Default value: TRUE</p></td>
    </tr>
    <tr>
      <th>tasks</th>
      <td><p>Integer(1). The number of tasks per job. value must be a scalar
integer &gt;= 0L. In this documentation a job is defined as a single call to a
function, such as bplapply, bpmapply etc. A task is the division of the X
argument into chunks. When tasks == 0 (default), X is divided as evenly as
possible over the number of workers (see MulticoreParam from BiocParallel
package).</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>if TRUE, prints the function log to stdout</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional parameters for <code><a href='uniqueGRanges.html'>uniqueGRanges</a></code> function.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A GRanges object with the columns of methylated and unmethylated
counts filtered for each cytosine position.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Cytosine sites with 'coverage' &gt; 'min.coverage' in at least one of
the samples are preserved. Positions with 'coverage' &lt; 'min.coverage' in
both samples, 'x' and 'y', are removed. Positions with 'coverage' &gt;
'percentile' (e.g., 99.9 percentile) are removed as well. It is expected
that the columns of methylated and unmethylated counts are given.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>### Create new data</span>
<span class='va'>df1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>chr <span class='op'>=</span> <span class='st'>'chr1'</span>, start <span class='op'>=</span> <span class='fl'>11</span><span class='op'>:</span><span class='fl'>16</span>, end <span class='op'>=</span> <span class='fl'>11</span><span class='op'>:</span><span class='fl'>16</span>,
                mC <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>2</span>,<span class='fl'>10</span>,<span class='fl'>7</span>,<span class='fl'>9</span>,<span class='fl'>1</span>,<span class='fl'>10</span><span class='op'>)</span>, uC <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>30</span>,<span class='fl'>20</span>,<span class='fl'>4</span>,<span class='fl'>8</span>,<span class='fl'>0</span>,<span class='fl'>10</span><span class='op'>)</span><span class='op'>)</span>

<span class='va'>df2</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>chr <span class='op'>=</span> <span class='st'>'chr1'</span>, start <span class='op'>=</span> <span class='fl'>12</span><span class='op'>:</span><span class='fl'>18</span>, end <span class='op'>=</span> <span class='fl'>12</span><span class='op'>:</span><span class='fl'>18</span>,
                mC2 <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>7</span>, uC2 <span class='op'>=</span> <span class='fl'>0</span><span class='op'>:</span><span class='fl'>6</span><span class='op'>)</span>

<span class='va'>gr1</span> <span class='op'>&lt;-</span> <span class='fu'>makeGRangesFromDataFrame</span><span class='op'>(</span><span class='va'>df1</span>, keep.extra.columns <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>gr2</span> <span class='op'>&lt;-</span> <span class='fu'>makeGRangesFromDataFrame</span><span class='op'>(</span><span class='va'>df2</span>, keep.extra.columns <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>

<span class='co'>## Filtering</span>
<span class='va'>r1</span> <span class='op'>&lt;-</span> <span class='fu'>uniqueGRfilterByCov</span><span class='op'>(</span><span class='va'>gr1</span>, <span class='va'>gr2</span>, ignore.strand <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'> *** Building coordinates for the new GRanges object ...</span></div><div class='output co'>#&gt; <span class='message'> *** Setting strand information to * ...</span></div><div class='output co'>#&gt; <span class='message'> *** Processing GRanges for sample: #1</span></div><div class='output co'>#&gt; <span class='message'> *** Processing GRanges for sample: #2...</span></div><div class='output co'>#&gt; <span class='message'> *** Sorting by chromosomes and start positions...</span></div><div class='input'><span class='va'>r1</span>
</div><div class='output co'>#&gt; GRanges object with 3 ranges and 4 metadata columns:
#&gt;       seqnames    ranges strand |        mC        uC       mC2       uC2
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
#&gt;   [1]     chr1        13      * |         7         4         2         1
#&gt;   [2]     chr1        15      * |         1         0         4         3
#&gt;   [3]     chr1        17      * |         0         0         6         5
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>## Cytosine position with coordinate 15 (rows #2) can pass the</span>
<span class='co'>## filtering conditions of min.coverage = 4 and lead to meaningless</span>
<span class='co'>## situations with methylation levels p = 1/(1 + 0) = 1</span>
<span class='va'>r1</span><span class='op'>[</span><span class='fl'>2</span><span class='op'>]</span>
</div><div class='output co'>#&gt; GRanges object with 1 range and 4 metadata columns:
#&gt;       seqnames    ranges strand |        mC        uC       mC2       uC2
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
#&gt;   [1]     chr1        15      * |         1         0         4         3
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div><div class='input'>
<span class='co'>## The last situation can be prevent, in this case, by setting</span>
<span class='co'>## min.meth = 1:</span>
<span class='va'>r1</span> <span class='op'>&lt;-</span> <span class='fu'>uniqueGRfilterByCov</span><span class='op'>(</span><span class='va'>gr1</span>, <span class='va'>gr2</span>, min.meth <span class='op'>=</span> <span class='fl'>1</span>, ignore.strand <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'> *** Building coordinates for the new GRanges object ...</span></div><div class='output co'>#&gt; <span class='message'> *** Setting strand information to * ...</span></div><div class='output co'>#&gt; <span class='message'> *** Processing GRanges for sample: #1</span></div><div class='output co'>#&gt; <span class='message'> *** Processing GRanges for sample: #2...</span></div><div class='output co'>#&gt; <span class='message'> *** Sorting by chromosomes and start positions...</span></div><div class='input'><span class='va'>r1</span>
</div><div class='output co'>#&gt; GRanges object with 2 ranges and 4 metadata columns:
#&gt;       seqnames    ranges strand |        mC        uC       mC2       uC2
#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
#&gt;   [1]     chr1        13      * |         7         4         2         1
#&gt;   [2]     chr1        17      * |         0         0         6         5
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


