<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Unique GRanges-class of methylation
read counts filtered by coverage. — uniqueGRfilterByCov • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Unique GRanges-class of methylation
read counts filtered by coverage. — uniqueGRfilterByCov"><meta property="og:description" content="Given two GRanges-class objects,
samples '1' and '2', carrying the counts of methylated (mC) and unmethylated
(uC) cytosines in their metacolumns, this function will filter by coverage
each cytosine site from each GRanges-class
object."><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Unique <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> of methylation
read counts filtered by coverage.</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/uniqueGRfilterByCov.R" class="external-link"><code>R/uniqueGRfilterByCov.R</code></a></small>
    <div class="hidden name"><code>uniqueGRfilterByCov.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given two <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> objects,
samples '1' and '2', carrying the counts of methylated (mC) and unmethylated
(uC) cytosines in their metacolumns, this function will filter by coverage
each cytosine site from each <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code>
object.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">uniqueGRfilterByCov</span><span class="op">(</span>
  <span class="va">x</span>,
  y <span class="op">=</span> <span class="cn">NULL</span>,
  min.coverage <span class="op">=</span> <span class="fl">4</span>,
  min.meth <span class="op">=</span> <span class="fl">0</span>,
  min.umeth <span class="op">=</span> <span class="fl">0</span>,
  min.sitecov <span class="op">=</span> <span class="fl">4</span>,
  percentile <span class="op">=</span> <span class="fl">0.9999</span>,
  high.coverage <span class="op">=</span> <span class="cn">NULL</span>,
  columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>mC <span class="op">=</span> <span class="fl">1</span>, uC <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,
  ignore.strand <span class="op">=</span> <span class="cn">FALSE</span>,
  num.cores <span class="op">=</span> <span class="fl">1L</span>,
  tasks <span class="op">=</span> <span class="fl">0L</span>,
  verbose <span class="op">=</span> <span class="cn">TRUE</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>x</dt>
<dd><p>An object from the classes 'GRanges', 'InfDiv', or 'pDMP' with
methylated and unmethylated counts in its meta-column. If the argument 'y' is
not given, then it is assumed that the first four columns of the
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> metadata 'x' are counts:
methylated and unmethylated counts for samples '1' and '2'. If x is a
'InfDiv', or 'pDMP', then 'y' is not needed, since samples '1' and '2' are
the first four columns of these objects.</p></dd>
<dt>y</dt>
<dd><p>A <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> or a
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList-class</a></code> object with methylated and
unmethylated counts in its meta-column. Default is NULL. If 'y' is a
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList-class</a></code> object, then the first two
metacolumns from 'x' are used in the pair-wise filtering toguether with the
first two metacolumns from each <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code>
element from 'y'.</p></dd>
<dt>min.coverage</dt>
<dd><p>An integer or an integer vector of length 2. Cytosine
sites where the coverage in both samples, 'x' and 'y', are less than
'min.coverage' are discarded. The cytosine site is preserved, however, if the
coverage is greater than 'min.coverage' in at least one sample. If
'min.coverage' is an integer vector, then the corresponding min coverage is
applied to each sample.</p></dd>
<dt>min.meth</dt>
<dd><p>An integer or an integer vector of length 2. Cytosine sites
where the numbers of read counts of methylated cytosine in both samples, '1'
and '2', are less than 'min.meth' are discarded. If 'min.meth' is an integer
vector, then the corresponding min number of reads is applied to each sample.</p></dd>
<dt>min.umeth</dt>
<dd><p>An integer or an integer vector of length 2. Min number of
reads to consider cytosine position. Specifically cytosine positions where
\(uC \leq min.umeth and mC &gt; 0 and mC &lt; min.meth\) hold will be removed,
where mC and uC stand for the numbers of methylated and unmethylated reads.
Default is \(min.umeth = 0\).</p></dd>
<dt>min.sitecov</dt>
<dd><p>An integer. The minimum total coverage. Only sites where
the total coverage \(cov1 + cov2\) is greater than 'min.sitecov' are
considered for downstream analysis, where cov1 and cov2 are the coverages
for samples 1 and 2, respectively.</p></dd>
<dt>percentile</dt>
<dd><p>Threshold to remove the outliers (PCR bias) from each file
and all files stacked. If 'high.coverage = NULL', then the threshold
\(q\) will be computed as:
$$q1 = quantile(cov1, probs=percentile)$$
$$q2 = quantile(cov2, probs=percentile)$$
$$q = min(q1, q2)$$</p><div class="sourceCode"><pre><code>where \eqn{cov1} and \eqn{cov2} are the coverage vectors from samples 1
and 2, respectively.
</code></pre></div></dd>
<dt>high.coverage</dt>
<dd><p>An integer for read counts. Cytosine sites having
higher coverage than this are discarded. Default is NULL. If
<strong>high.coverage</strong> is not NULL, then the <strong>percentile</strong> argument is
disregarded and <strong>high.coverage</strong> is used as threshold to remove the
PCR bias. If percentile is not null, then q = max(q, high.coverage).</p></dd>
<dt>columns</dt>
<dd><p>Vector of integer numbers of the columns (from each GRanges
meta-column) where the methylated and unmethylated counts are provided. If
not provided, then the methylated and unmethylated counts are assumed to be
at columns 1 and 2, respectively.</p></dd>
<dt>ignore.strand</dt>
<dd><p>When set to TRUE, the strand information is ignored in
the overlap of <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> objects. This is a
parameter passed to <code><a href="uniqueGRanges.html">uniqueGRanges</a></code> function. Default value:
FALSE.</p></dd>
<dt>num.cores, tasks</dt>
<dd><p>Parameters for parallel computation using package
<code><a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallel-package.html" class="external-link">BiocParallel-package</a></code>: the number of cores to use,
i.e. at most how many child processes will be run simultaneously (see
<code><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></code> and the number of tasks per job (only
for Linux OS).</p></dd>
<dt>verbose</dt>
<dd><p>if TRUE, prints the function log to stdout</p></dd>
<dt>...</dt>
<dd><p>Additional parameters for <code><a href="uniqueGRanges.html">uniqueGRanges</a></code> function.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>if 'x' and 'y' are <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> object,
then a <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> with the columns
of methylated and unmethylated counts filtered for each cytosine position.
A <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList-class</a></code> object will returned, if
'y' is a <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList-class</a></code> object of same
length as length(y) and named as names(y).</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Cytosine sites with 'coverage' &gt; 'min.coverage' in at least one of
the samples are preserved. Positions with 'coverage' &lt; 'min.coverage' in both
samples, 'x' and 'y', are removed. Positions with 'coverage' &gt; 'percentile'
(e.g., 99.9 percentile) are removed as well. It is expected that the columns
of methylated and unmethylated counts are given.</p>
<p>This function is addressed to create pair-wise
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> object with four metacolumns of
count: samples '1' and '2', carrying the counts of methylated (mC) and
unmethylated (uC) cytosines in their metacolumns, respectively. Counts from
sample 1 are typically used as reference counts in computing information
divergences in the downtstream analysis.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co">### Create new data</span></span>
<span class="r-in"><span class="va">df1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="st">'chr1'</span>, start <span class="op">=</span> <span class="fl">11</span><span class="op">:</span><span class="fl">16</span>, end <span class="op">=</span> <span class="fl">11</span><span class="op">:</span><span class="fl">16</span>,</span>
<span class="r-in">                mC <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">10</span>,<span class="fl">7</span>,<span class="fl">9</span>,<span class="fl">1</span>,<span class="fl">10</span><span class="op">)</span>, uC <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">30</span>,<span class="fl">20</span>,<span class="fl">4</span>,<span class="fl">8</span>,<span class="fl">0</span>,<span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">df2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="st">'chr1'</span>, start <span class="op">=</span> <span class="fl">12</span><span class="op">:</span><span class="fl">18</span>, end <span class="op">=</span> <span class="fl">12</span><span class="op">:</span><span class="fl">18</span>,</span>
<span class="r-in">                mC2 <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">7</span>, uC2 <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">gr1</span> <span class="op">&lt;-</span> <span class="fu">makeGRangesFromDataFrame</span><span class="op">(</span><span class="va">df1</span>, keep.extra.columns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-in"><span class="va">gr2</span> <span class="op">&lt;-</span> <span class="fu">makeGRangesFromDataFrame</span><span class="op">(</span><span class="va">df2</span>, keep.extra.columns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Filtering</span></span>
<span class="r-in"><span class="va">r1</span> <span class="op">&lt;-</span> <span class="fu">uniqueGRfilterByCov</span><span class="op">(</span><span class="va">gr1</span>, <span class="va">gr2</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Building coordinates for the new GRanges object ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Setting strand information to * ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Sorting by chromosomes and start positions...</span>
<span class="r-in"><span class="va">r1</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 3 ranges and 4 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        mC        uC       mC2       uC2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1        13      * |         7         4         2         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1        15      * |         1         0         4         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1        17      * |         0         0         6         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Cytosine position with coordinate 15 (rows #2) can pass the</span></span>
<span class="r-in"><span class="co">## filtering conditions of min.coverage = 4 and lead to meaningless</span></span>
<span class="r-in"><span class="co">## situations with methylation levels p = 1/(1 + 0) = 1</span></span>
<span class="r-in"><span class="va">r1</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 1 range and 4 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        mC        uC       mC2       uC2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1        15      * |         1         0         4         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## The last situation can be prevent, in this case, by setting</span></span>
<span class="r-in"><span class="co">## min.meth = 1:</span></span>
<span class="r-in"><span class="va">r1</span> <span class="op">&lt;-</span> <span class="fu">uniqueGRfilterByCov</span><span class="op">(</span><span class="va">gr1</span>, <span class="va">gr2</span>, min.meth <span class="op">=</span> <span class="fl">1</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Building coordinates for the new GRanges object ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Setting strand information to * ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Sorting by chromosomes and start positions...</span>
<span class="r-in"><span class="va">r1</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 2 ranges and 4 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |        mC        uC       mC2       uC2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1        13      * |         7         4         2         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1        17      * |         0         0         6         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

