<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Unique genomic ranges from a list of GRanges objects — uniqueGRanges • MethylIT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Unique genomic ranges from a list of GRanges objects — uniqueGRanges"><meta property="og:description" content="Build an unique GRanges object from a list of Granges objects."><meta property="og:image" content="https://genomaths.github.io/methylit/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MethylIT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.3.2.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MethylIT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/MethylIT.html">MethylIT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Methylation_analysis_with_Methyl-IT.html">Methylation analysis with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/cancer_example.html">Example of Methylome Analysis with MethylIT using Cancer Datasets</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/Cutpoint_estimation_with_Methyl-IT.html">Optimal cutpoint for the methylation signal</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/pca_lda_with_methylit.html">Principal Components and Linear Discriminant Analyses with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/GenExp_Methylation_association_hyper_down-regulated.html">Association Between Gene Expression and Cytosine DNA Methylation at gene-body</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/dmr_at_memory.html">DMR detection with Methyl-IT</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/some_tips_for_methylit_r_scripts.html">Some tips for Methyl-IT R scripts</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/genomaths/MethylIT/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Unique genomic ranges from a list of GRanges objects</h1>
    <small class="dont-index">Source: <a href="https://github.com/genomaths/MethylIT/blob/HEAD/R/uniqueGRanges.R" class="external-link"><code>R/uniqueGRanges.R</code></a></small>
    <div class="hidden name"><code>uniqueGRanges.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Build an unique GRanges object from a list of Granges objects.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">uniqueGRanges</span><span class="op">(</span></span>
<span>  <span class="va">ListOfGranges</span>,</span>
<span>  ncolns <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  columns <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  chromosomes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  maxgap <span class="op">=</span> <span class="op">-</span><span class="fl">1L</span>,</span>
<span>  minoverlap <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  missing <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"any"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"within"</span>, <span class="st">"equal"</span><span class="op">)</span>,</span>
<span>  select <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"all"</span>, <span class="st">"first"</span>, <span class="st">"last"</span>, <span class="st">"arbitrary"</span><span class="op">)</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  num.cores <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>  tasks <span class="op">=</span> <span class="fl">0L</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>ListOfGranges</dt>
<dd><p>Objects to combine. A list of
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a></code> object or a
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList-class</a></code> object.</p></dd>


<dt>ncolns</dt>
<dd><p>integer. Number of columns to use from the meta-column of each
GRanges object. Default value: NULL. If NULL, all the columns (from
column 1 to ncolns) from each GRanges will be present in the
\(uniqueGRanges\) output. If \(ncolns\) = NULL and \(columns != NULL\),
then the function will set \(ncolns = length(columns)\).</p></dd>


<dt>columns</dt>
<dd><p>integer number(s) corresponding to the specific column(s) to
use from the meta-column of each GRanges. Default value: NULL. if provided,
the metacolumn from the uniqueGRanges output will contain the specified
columns.</p></dd>


<dt>chromosomes</dt>
<dd><p>Chromosomes used Default value: NULL</p></dd>


<dt>maxgap, minoverlap, ignore.strand, select, type</dt>
<dd><p>The same as in
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps-methods</a></code>.</p></dd>


<dt>missing</dt>
<dd><p>A numerical value (default 0) or NA to write in ranges with
missing values. For example, suppose that we want to build a uniqueGRanges
object from the GRanges objects X and Y. If a given range k from the GRanges
object X with metacolumn value x is missing in the GRanges object Y, then the
metacolumn of range k from uniqueGRanges(list(X,Y)) object will be the row
vector (x,0) or (x,NA) if missing = NA.</p></dd>


<dt>num.cores, tasks</dt>
<dd><p>Parameters for parallel computation using package
<code><a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallel-package.html" class="external-link">BiocParallel-package</a></code>: the number of cores to use,
i.e. at most how many child processes will be run simultaneously (see
<code><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html" class="external-link">bplapply</a></code> and the number of tasks per job (only
for Linux OS).</p></dd>


<dt>verbose</dt>
<dd><p>if TRUE, prints the function log to stdout</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a GRanges object</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The metadata of each one of these GRanges must have one or more
columns to yield a unique GRanges object with metadata columns from the
original GRanges objects. Otherwise, a unique GRanges object will be created
without metadata columns. Additionally, all metadata must be the same class,
e.g. all numeric or all characters, or all factor</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Robersy Sanchez (<a href="https://genomaths.com" class="external-link">https://genomaths.com</a>).</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## 'GRanges' object construction</span></span></span>
<span class="r-in"><span><span class="va">dfChr1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="st">'chr1'</span>, start <span class="op">=</span> <span class="fl">11</span><span class="op">:</span><span class="fl">15</span>, end <span class="op">=</span> <span class="fl">11</span><span class="op">:</span><span class="fl">15</span>,</span></span>
<span class="r-in"><span>                     strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'+'</span>,<span class="st">'-'</span>,<span class="st">'+'</span>,<span class="st">'*'</span>,<span class="st">'.'</span><span class="op">)</span>, score <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dfChr2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="st">'chr1'</span>, start <span class="op">=</span> <span class="fl">11</span><span class="op">:</span><span class="fl">15</span>, end <span class="op">=</span> <span class="fl">11</span><span class="op">:</span><span class="fl">15</span>,</span></span>
<span class="r-in"><span>                     strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'+'</span>,<span class="st">'-'</span>,<span class="st">'+'</span>,<span class="st">'*'</span>,<span class="st">'.'</span><span class="op">)</span>, score <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dfChr3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="st">'chr1'</span>, start <span class="op">=</span> <span class="fl">11</span><span class="op">:</span><span class="fl">15</span>, end <span class="op">=</span> <span class="fl">11</span><span class="op">:</span><span class="fl">15</span>,</span></span>
<span class="r-in"><span>                     strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'+'</span>,<span class="st">'-'</span>,<span class="st">'+'</span>,<span class="st">'*'</span>,<span class="st">'+'</span><span class="op">)</span>, score <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">gr1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html" class="external-link">makeGRangesFromDataFrame</a></span><span class="op">(</span><span class="va">dfChr1</span>, keep.extra.columns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gr2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html" class="external-link">makeGRangesFromDataFrame</a></span><span class="op">(</span><span class="va">dfChr2</span>, keep.extra.columns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gr3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html" class="external-link">makeGRangesFromDataFrame</a></span><span class="op">(</span><span class="va">dfChr3</span>, keep.extra.columns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## A 'GRangesList' object:</span></span></span>
<span class="r-in"><span><span class="va">grList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList</a></span><span class="op">(</span><span class="st">'gr1'</span> <span class="op">=</span> <span class="va">gr1</span>, <span class="st">'gr2'</span> <span class="op">=</span> <span class="va">gr2</span>, <span class="st">'gr3'</span> <span class="op">=</span> <span class="va">gr3</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">grList</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRangesList object of length 3:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $gr1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 5 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1        11      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1        12      - |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1        13      + |         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1        14      * |         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr1        15      * |         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $gr2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 5 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1        11      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1        12      - |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1        13      + |         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1        14      * |         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr1        15      * |         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $gr3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 5 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1        11      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1        12      - |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1        13      + |         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1        14      * |         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr1        15      + |         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The default setting does not ignore strands value (ignore.strand = FALSE).</span></span></span>
<span class="r-in"><span><span class="co">## In object 'gr1' and 'gr2', the strand at the 5th range is unknown.</span></span></span>
<span class="r-in"><span><span class="co">## However, in object 'gr3' the 5th range has "+" strand. The defaul setting</span></span></span>
<span class="r-in"><span><span class="co">## preserve this information. Two ranges with start = 15 are yielded: 1) the</span></span></span>
<span class="r-in"><span><span class="co">## first range the first range with strand = '*' carries the value 5 in the</span></span></span>
<span class="r-in"><span><span class="co">## columns derived from 'gr1' and 'gr2', and 0 (missing = 0) in the column</span></span></span>
<span class="r-in"><span><span class="co">## derived from 'gr3'.</span></span></span>
<span class="r-in"><span><span class="fu">uniqueGRanges</span><span class="op">(</span><span class="va">grList</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Building coordinates for the new GRanges object ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Strand information is preserved ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |=======================                                               |  33%  |                                                                              |===============================================                       |  67%  |                                                                              |======================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Sorting by chromosomes and start positions...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 6 ranges and 3 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |     score   score.1   score.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1        11      + |         1         1         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1        12      - |         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1        13      + |         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1        14      * |         4         4         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr1        15      * |         5         5         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [6]     chr1        15      + |         0         0         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## A new GRanges object is included with values in different strands</span></span></span>
<span class="r-in"><span><span class="va">dfChr4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="st">'chr1'</span>, start <span class="op">=</span> <span class="fl">11</span><span class="op">:</span><span class="fl">15</span>, end <span class="op">=</span> <span class="fl">11</span><span class="op">:</span><span class="fl">15</span>,</span></span>
<span class="r-in"><span>                     strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'-'</span>,<span class="st">'-'</span>,<span class="st">'+'</span>,<span class="st">'*'</span>,<span class="st">'-'</span><span class="op">)</span>, score <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gr4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html" class="external-link">makeGRangesFromDataFrame</a></span><span class="op">(</span><span class="va">dfChr4</span>, keep.extra.columns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">grList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html" class="external-link">GRangesList</a></span><span class="op">(</span><span class="st">'gr1'</span> <span class="op">=</span> <span class="va">gr1</span>, <span class="st">'gr2'</span> <span class="op">=</span> <span class="va">gr2</span>, <span class="st">'gr3'</span> <span class="op">=</span> <span class="va">gr3</span>, gr4 <span class="op">=</span> <span class="va">gr4</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">grList</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRangesList object of length 4:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $gr1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 5 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1        11      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1        12      - |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1        13      + |         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1        14      * |         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr1        15      * |         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $gr2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 5 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1        11      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1        12      - |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1        13      + |         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1        14      * |         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr1        15      * |         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $gr3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 5 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1        11      + |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1        12      - |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1        13      + |         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1        14      * |         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr1        15      + |         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $gr4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 5 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |     score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1        11      - |         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1        12      - |         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1        13      + |         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1        14      * |         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr1        15      - |         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The default setting preserve the information adding new ranges.</span></span></span>
<span class="r-in"><span><span class="fu">uniqueGRanges</span><span class="op">(</span><span class="va">grList</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Building coordinates for the new GRanges object ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Strand information is preserved ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |==================                                                    |  25%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Sorting by chromosomes and start positions...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 8 ranges and 4 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |     score   score.1   score.2   score.3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1        11      + |         1         1         1         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1        11      - |         0         0         0         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1        12      - |         2         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1        13      + |         3         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr1        14      * |         4         4         4         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [6]     chr1        15      * |         5         5         0         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [7]     chr1        15      + |         0         0         5         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [8]     chr1        15      - |         0         0         0         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The setting: 'ignore.strand = TRUE' ignores the strand information</span></span></span>
<span class="r-in"><span><span class="co">## and only the information on the base position is preserved.</span></span></span>
<span class="r-in"><span><span class="fu">uniqueGRanges</span><span class="op">(</span><span class="va">grList</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Building coordinates for the new GRanges object ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Setting strand information to * ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |==================                                                    |  25%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  *** Sorting by chromosomes and start positions...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 5 ranges and 4 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand |     score   score.1   score.2   score.3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1        11      * |         1         1         1         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1        12      * |         2         2         2         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]     chr1        13      * |         3         3         3         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]     chr1        14      * |         4         4         4         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]     chr1        15      * |         5         5         5         5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

