<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>tableBoots — tableBoots • usefr</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="tableBoots — tableBoots"><meta property="og:description" content="Parametric Bootstrap of \(n x m\) Contingency independence
test. It is assumed that the provided contingency table summarizes the
results where two or more groups are compared and the outcome is a
categorical variable, i.e., disease vs. no disease, pass vs. fail, treated
patient vs. control group, etc.
The goodness of fit statistic is the root-mean-square statistic (RMST)
or Hellinger divergence, as proposed by Perkins et al. [1, 2]. Hellinger
divergence (HD) is computed as proposed in [3]."><meta property="og:image" content="https://genomaths.github.io/usefr/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">usefr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/usefr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/usefr.html">usefr</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/fisher-exact-test-failure-can-lead-to-biassed-results.html">Fisher's exact test failure can lead to biased results</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>tableBoots</h1>
    
    <div class="hidden name"><code>tableBoots.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Parametric Bootstrap of \(n x m\) Contingency independence
test. It is assumed that the provided contingency table summarizes the
results where two or more groups are compared and the outcome is a
categorical variable, i.e., disease vs. no disease, pass vs. fail, treated
patient vs. control group, etc.</p>
<p>The goodness of fit statistic is the root-mean-square statistic (RMST)
or Hellinger divergence, as proposed by Perkins et al. [1, 2]. Hellinger
divergence (HD) is computed as proposed in [3].</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">tableBoots</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  stat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rmst"</span>, <span class="st">"hd"</span>, <span class="st">"chisq"</span>, <span class="st">"all"</span><span class="op">)</span>,</span>
<span>  out.stat <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  num.permut <span class="op">=</span> <span class="fl">100</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>x</dt>
<dd><p>A numerical matrix corresponding to cross tabulation \(n x m\)
table (contingency table).</p></dd>


<dt>stat</dt>
<dd><p>Statistic to be used in the testing: 'rmst','hdiv', or 'all'.</p></dd>


<dt>out.stat</dt>
<dd><p>logical(1). Whether to return the values of the statistics
used: the bootstrap mean and the original value estimated.</p></dd>


<dt>num.permut</dt>
<dd><p>Number of permutations.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A p-value probability</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>For goodness-of-fit the following null hypothesis is tested
    \(H_\theta: p = p(\theta)\)
    To conduct a single simulation, we perform the following three-step
     procedure [1,2]:</p><ol><li><p>To generate m i.i.d. draws according to the model distribution
          \(p(\theta)\), where \(\theta'\) is the estimate calculated
          from the experimental data,</p></li>
<li><p>To estimate the parameter \(\theta\) from the data generated in
          Step 1, obtaining a new estimate \(\theta\)est.</p></li>
<li><p>To calculate the statistic under consideration (HD,
          RMST), using the data generated in Step 1 and taking the model
          distribution to be \(\theta\)est, where \(\theta\)est is the
          estimate calculated in Step 2 from the data generated in Step 1.</p></li>
</ol><p>After conducting many such simulations, the confidence level for
    rejecting the null hypothesis is the fraction of the statistics
    calculated in step 3 that are less than the statistic calculated from
    the empirical data. The significance level \(\alpha\) is the same as a
    confidence level of \(1-\alpha\).</p>
    </div>
    <div id="references">
    <h2>References</h2>
    
<ol><li><p>Perkins W, Tygert M, Ward R. Chi^2 and Classical Exact Tests
          Often Wildly Misreport Significance; the Remedy Lies in Computers
          [Internet]. Uploaded to ArXiv. 2011. Report No.:
          arXiv:1108.4126v2.</p></li>
<li><p>Perkins, W., Tygert, M. &amp; Ward, R. Computing the confidence
          levels or a root-mean square test of goodness-of-fit. 217,
          9072-9084 (2011).</p></li>
<li><p>Basu, A., Mandal, A. &amp; Pardo, L. Hypothesis testing for two
          discrete populations based on the Hellinger distance. Stat.
          Probab. Lett. 80, 206-214 (2010).</p></li>
</ol></div>
    <div id="author">
    <h2>Author</h2>
    <p>Robersy Sanchez (<a href="https://genomaths.com" class="external-link">https://genomaths.com</a>).</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">## The numbers of methylated and unmethylted "read-counts" targeting DNA</span></span></span>
<span class="r-in"><span><span class="co">## cytosine sites are the typical raw data resulting from bisulfate</span></span></span>
<span class="r-in"><span><span class="co">## sequencing experiments (BiSeq). Methylation analysis requires for the</span></span></span>
<span class="r-in"><span><span class="co">## application of some statistical tests to identify differentially</span></span></span>
<span class="r-in"><span><span class="co">## methylated sites/ positions (DMSs or DMPs). BiSeq experiments on humans</span></span></span>
<span class="r-in"><span><span class="co">## are relatively expensive (still in 2020) and some researcher is willing</span></span></span>
<span class="r-in"><span><span class="co">## to accept a site coverage (total) of 4 read-counts as threshold for a</span></span></span>
<span class="r-in"><span><span class="co">## site to be included in the analysis. Let's suppose that at a given site</span></span></span>
<span class="r-in"><span><span class="co">## we have the following contingency table:</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">site_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">3</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    nrow <span class="op">=</span> <span class="fl">2</span>,</span></span>
<span class="r-in"><span>    dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ctrl"</span>, <span class="st">"treat"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        treat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"meth"</span>, <span class="st">"unmeth"</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">site_res</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        treat</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> status  meth unmeth</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ctrl     3      1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   treat    1      3</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## The application of classical test variants yield the following results</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/fisher.test.html" class="external-link">fisher.test</a></span><span class="op">(</span><span class="va">site_res</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.4857143</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/fisher.test.html" class="external-link">fisher.test</a></span><span class="op">(</span><span class="va">site_res</span>, alternative <span class="op">=</span> <span class="st">"greater"</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.2428571</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/fisher.test.html" class="external-link">fisher.test</a></span><span class="op">(</span><span class="va">site_res</span>, simulate.p.value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.4857143</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/chisq.test.html" class="external-link">chisq.test</a></span><span class="op">(</span><span class="va">site_res</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.4795001</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/chisq.test.html" class="external-link">chisq.test</a></span><span class="op">(</span><span class="va">site_res</span>, simulate.p.value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.4832584</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## That is, these test did not detect differences between the read-counts</span></span></span>
<span class="r-in"><span><span class="co">## for control and the treated patients. 'tableBoots' function suggests</span></span></span>
<span class="r-in"><span><span class="co">## that the site is borderline.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">tableBoots</span><span class="op">(</span><span class="va">site_res</span>, stat <span class="op">=</span> <span class="st">"all"</span>, num.permut <span class="op">=</span> <span class="fl">1e3</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  rmst.p.value  hdiv.p.value chisq.p.value </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    0.09390609    0.04095904    0.14585415 </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## A slight change in the read counts identify a meaningful difference</span></span></span>
<span class="r-in"><span><span class="co">## not detected by the classicla tests</span></span></span>
<span class="r-in"><span><span class="va">site_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">4</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    nrow <span class="op">=</span> <span class="fl">2</span>,</span></span>
<span class="r-in"><span>    dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ctrl"</span>, <span class="st">"treat"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        treat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"meth"</span>, <span class="st">"unmeth"</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">site_res</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        treat</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> status  meth unmeth</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ctrl     3      1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   treat    0      4</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">23</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">tableBoots</span><span class="op">(</span><span class="va">site_res</span>, stat <span class="op">=</span> <span class="st">"all"</span>, num.permut <span class="op">=</span> <span class="fl">1e3</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  rmst.p.value  hdiv.p.value chisq.p.value </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   0.007992008   0.008991009   0.010989011 </span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/fisher.test.html" class="external-link">fisher.test</a></span><span class="op">(</span><span class="va">site_res</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.1428571</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/fisher.test.html" class="external-link">fisher.test</a></span><span class="op">(</span><span class="va">site_res</span>, alternative <span class="op">=</span> <span class="st">"greater"</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.07142857</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/fisher.test.html" class="external-link">fisher.test</a></span><span class="op">(</span><span class="va">site_res</span>, simulate.p.value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.1428571</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/chisq.test.html" class="external-link">chisq.test</a></span><span class="op">(</span><span class="va">site_res</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.144127</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/chisq.test.html" class="external-link">chisq.test</a></span><span class="op">(</span><span class="va">site_res</span>, simulate.p.value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.1324338</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Now, let's suppose that we want to test whether methylation status from</span></span></span>
<span class="r-in"><span><span class="co">## two DNA sites the same differentially methylated region (DMR)</span></span></span>
<span class="r-in"><span><span class="co">## are independent, i.e. whether the treatment affected the methylation</span></span></span>
<span class="r-in"><span><span class="co">## status of the two sites. In this case, the counts grouped into four</span></span></span>
<span class="r-in"><span><span class="co">## categories.</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">site_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">4</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    nrow <span class="op">=</span> <span class="fl">2</span>, byrow <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>    dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ctrl"</span>, <span class="st">"treat"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        treat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>            <span class="st">"meth.1"</span>, <span class="st">"unmeth.1"</span>,</span></span>
<span class="r-in"><span>            <span class="st">"meth.2"</span>, <span class="st">"unmeth.2"</span></span></span>
<span class="r-in"><span>        <span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">site_res</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        treat</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> status  meth.1 unmeth.1 meth.2 unmeth.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ctrl       3        1      3        1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   treat      1        3      0        4</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Chi-squared from the R package 'stats'</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/chisq.test.html" class="external-link">chisq.test</a></span><span class="op">(</span><span class="va">site_res</span><span class="op">)</span><span class="op">$</span><span class="va">p.bvalue</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> NULL</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/chisq.test.html" class="external-link">chisq.test</a></span><span class="op">(</span><span class="va">site_res</span>, simulate.p.value <span class="op">=</span> <span class="cn">TRUE</span>, B <span class="op">=</span> <span class="fl">2e3</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.1244378</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">tableBoots</span><span class="op">(</span><span class="va">site_res</span>, stat <span class="op">=</span> <span class="st">"all"</span>, num.permut <span class="op">=</span> <span class="fl">999</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  rmst.p.value  hdiv.p.value chisq.p.value </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         0.065         0.087         0.068 </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Results above are in border. If we include, third site,</span></span></span>
<span class="r-in"><span><span class="co">## then sinces would different.</span></span></span>
<span class="r-in"><span><span class="va">site_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">4</span>, <span class="fl">4</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">4</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    nrow <span class="op">=</span> <span class="fl">2</span>, byrow <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>    dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ctrl"</span>, <span class="st">"treat"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        treat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>            <span class="st">"meth.1"</span>, <span class="st">"unmeth.1"</span>,</span></span>
<span class="r-in"><span>            <span class="st">"meth.2"</span>, <span class="st">"unmeth.2"</span>,</span></span>
<span class="r-in"><span>            <span class="st">"meth.3"</span>, <span class="st">"unmeth.3"</span></span></span>
<span class="r-in"><span>        <span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">site_res</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        treat</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> status  meth.1 unmeth.1 meth.2 unmeth.2 meth.3 unmeth.3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ctrl       3        1      3        1      4        1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   treat      1        3      0        4      0        4</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## That is, we have not reason to believe that the observed methylation</span></span></span>
<span class="r-in"><span><span class="co">## levels in the treatment are not independent</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/chisq.test.html" class="external-link">chisq.test</a></span><span class="op">(</span><span class="va">site_res</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.02764776</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/chisq.test.html" class="external-link">chisq.test</a></span><span class="op">(</span><span class="va">site_res</span>, simulate.p.value <span class="op">=</span> <span class="cn">TRUE</span>, B <span class="op">=</span> <span class="fl">2e3</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.017991</span>
<span class="r-in"><span><span class="fu">tableBoots</span><span class="op">(</span><span class="va">site_res</span>, stat <span class="op">=</span> <span class="st">"all"</span>, num.permut <span class="op">=</span> <span class="fl">999</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  rmst.p.value  hdiv.p.value chisq.p.value </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         0.014         0.016         0.010 </span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Robersy Sanchez.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

