<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analysis of Automorphisms on a MSA of Primate BRCA1 Gene • GenomAutomorphism</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Analysis of Automorphisms on a MSA of Primate BRCA1 Gene">
<meta property="og:description" content="GenomAutomorphism">
<meta property="og:image" content="https://genomaths.github.io/GenomAutomorphism/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomAutomorphism</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/GenomAutomorphism.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/GenomAutomorphism.html">GenomAutomorphism</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/GenomAutomorphism/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="automorphism_on_msa_brca1_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</h1>
                        <h4 class="author">Robersy Sanchez</h4>
            <address class="author_afil">
      Department of Biology. Pennsylvania State University, University Park, PA 16802<br><a class="author_email" href="mailto:#"></a><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a>
      </address>
                  
            <h4 class="date">28 February 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/master/vignettes/automorphism_on_msa_brca1.Rmd"><code>vignettes/automorphism_on_msa_brca1.Rmd</code></a></small>
      <div class="hidden name"><code>automorphism_on_msa_brca1.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Analysis of DNA mutational events on DNA Multiple Sequence Alignment (MSA) by means of automorphisms between pairwise DNA sequences algebraically represented as Abelian finite group. Herein, we show a simple example on how to apply the analysis on MSA of primate BRCA1 gene.</p>
    </div>
    
<style type="text/css">

.watch-out {
  max-width: 1820px !important;
  margin-left: auto;
  margin-right: auto;
  background-color: #c9feed;
  font-weight: bold
  font-size: 100%;
  padding: 2px;
}

pre, code {
  max-width: 1820px !important;
  margin-left: auto;
  margin-right: auto;
  background-color: #c9feed;
  font-weight: bold
  font-size: 100px;
  padding: 2px;
}

code {
    font-size: 100%;
    font-family: Consolas,Monaco,"Courier New",monospace;
    border-radius: 0;
}

code.r{
  font-size: 100%;
}
</style>
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>This is a R package to compute the autimorphisms between pairwise aligned DNA sequences represented as elements from a Genomic Abelian group as described in reference (<a href="#1">1</a>). In a general scenario, whole chromosomes or genomic regions from a population (from any species or close related species) can be algebraically represented as a direct sum of cyclic groups or more specifically Abelian <em>p</em>-groups. Basically, we propose the representation of multiple sequence alignments (MSA) of length <em>N</em> as a finite Abelian group created by the direct sum of homocyclic Abelian groups of <em>prime-power order</em>:</p>
<p><span class="math display">\[
\qquad G = (\mathbb{Z}_{p^{\alpha_{1}}_1})^{n_1} \oplus (\mathbb{Z}_{p^{\alpha_{2}}_1})^{n_2} \oplus \dots \oplus (\mathbb{Z}_{p^{\alpha_{k}}_k})^{n_k} 
\]</span></p>
<p>Where, the <span class="math inline">\(p_i\)</span>’s are prime numbers, <span class="math inline">\(\alpha_i \in \mathbb{N}\)</span> and <span class="math inline">\(\mathbb{Z}_{p^{\alpha_{i}}_i}\)</span> is the group of integer modulo <span class="math inline">\(p^{\alpha_{i}}_i\)</span>.</p>
<p>For the purpose of estimating the automorphism between two aligned DNA sequences, <span class="math inline">\(p^{\alpha_{i}}_i \in \{5, 2^6, 5^3 \}\)</span>.</p>
<div id="automorphisms" class="section level2">
<h2 class="hasAnchor">
<a href="#automorphisms" class="anchor"></a>Automorphisms</h2>
<p>Herein, automorphisms are considered algebraic descriptions of mutational event observed in codon sequences represented on different Abelian groups. In particular, as described in references (<a href="#3">3-4</a>), for each representation of the codon set on a defined Abelian group there are 24 possible isomorphic Abelian groups. These Abelian groups can be labeled based on the DNA base-order used to generate them. The set of 24 Abelian groups can be described as a group isomorphic to the symmetric group of degree four (<span class="math inline">\(S_4\)</span>, see reference (<a href="#4">4</a>)).</p>
<p>For further support about the symmetric group on the 24 Abelian group of genetic-code cubes, users can also see <a href="https://github.com/genomaths/GenomeAlgebra_SymmetricGroup">Symmetric Group of the Genetic-Code Cubes.</a>, specifically the Mathematica notebook <em>IntroductionToZ5GeneticCodeVectorSpace.nb</em> and interact with it using Wolfram Player, freely available (for Windows and Linux OS) at, <a href="https://www.wolfram.com/player/" class="uri">https://www.wolfram.com/player/</a>.</p>
</div>
</div>
<div id="automorphisms-on-mathbbz_64" class="section level1">
<h1 class="hasAnchor">
<a href="#automorphisms-on-mathbbz_64" class="anchor"></a>Automorphisms on <span class="math inline">\(\mathbb{Z}_{64}\)</span>
</h1>
<p>First, we proceed to load the R package required for our analysis</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/GenomAutomorphism">GenomAutomorphism</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings">Biostrings</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/omarwagih/ggseqlogo">ggseqlogo</a></span><span class="op">)</span></code></pre></div>
<p>Next, we proceed to check the DNA multiple sequence alignment (MSA) file. This ia a FASTA file carrying the MSA of primates BRCA1 gene. Notice that we are familiar with the <em>FASTA</em> file, then it is better to directly read it with function automorphism.</p>
<p>However, for the current example, this step can be bypassed, since the MSA is provided provided together with <em><em>GenomAutomorphism</em></em> R package</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Do not run it. This is included with package</span>
<span class="va">URL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"https://github.com/genomaths/seqalignments/raw/master/BRCA1/"</span>,
              <span class="st">"brca1_primates_dna_repair_20_sequences.fasta"</span><span class="op">)</span>

<span class="va">brca1_aln</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html">readDNAMultipleAlignment</a></span><span class="op">(</span>filepath <span class="op">=</span> <span class="va">URL</span><span class="op">)</span>
<span class="va">brca1_aln</span></code></pre></div>
<p>Load MSA available in the package</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"brca1_aln"</span><span class="op">)</span>
<span class="va">brca1_aln</span>
<span class="co">#&gt; DNAMultipleAlignment with 20 rows and 2283 columns</span>
<span class="co">#&gt;       aln                                                                                           names               </span>
<span class="co">#&gt;  [1] ATGGATTTATCTGCTCTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGACACCTACCTGATACCCCAGATCCCCCACAGCCACTACTGA NM_007298.3:20-22...</span>
<span class="co">#&gt;  [2] ATGGATTTATCTGCTCTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGACACCTACCTGATACCCCAGATCCCCCACAGCCACTACTGA U64805.1:1-2280_H...</span>
<span class="co">#&gt;  [3] ATGGATTTATCTGCTCTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGACACCTACCTGATACCCCAGATCCCCCACAGCCACTACTGA XM_031011560.1:23...</span>
<span class="co">#&gt;  [4] ATGGATTTATCTGCTCTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGACACCTACCTGATACCCCAGATCCCCCACAGCCACTACTGA XM_031011561.1:23...</span>
<span class="co">#&gt;  [5] ATGGATTTATCTGCTCTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGACACCTACCTGATACCCCAGATCCCCCACAGCCACTACTGA XM_031011562.1:16...</span>
<span class="co">#&gt;  [6] ATGGATTTATCTGCTCTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGACACCTACCTGATACCTCAGATCCCCCACAGCCACTACTGA XM_009432101.3:27...</span>
<span class="co">#&gt;  [7] ATGGATTTATCTGCTCTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGACACCTACCTGATACCTCAGATCCCCCACAGCCACTACTGA XM_009432104.3:37...</span>
<span class="co">#&gt;  [8] ATGGATTTATCTGCTCTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGACACCTACCTGATACCTCAGATCCCCCACAGCCACTACTGA XM_016930487.2:37...</span>
<span class="co">#&gt;  [9] ATGGATTTATCTGCTCTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGACACCTACCTGATACCTCAGATCCCCCACAGCCACTACTGA XM_009432099.3:37...</span>
<span class="co">#&gt;  ... ...</span>
<span class="co">#&gt; [12] ATGGATTTATCTGCTCTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGACACCTACCTGATACCTCAGATCCCCCACAGCCACTACTGA XM_034941185.1:24...</span>
<span class="co">#&gt; [13] ATGGATTTATCTGCTCTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGACACCTACCTGATACCTCAGATCCCCCACAGCCACTACTGA XM_034941182.1:25...</span>
<span class="co">#&gt; [14] ATGGATTTATCTGCTGTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGATACCTACCTGATACCCCAGATCCCCCACAGCCACTACTGA XM_032163757.1:14...</span>
<span class="co">#&gt; [15] ATGGATTTATCTGCTGTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGATACCTACCTGATACCCCAGATCCCCCACAGCCACTACTGA XM_032163756.1:14...</span>
<span class="co">#&gt; [16] ATGGATTTATCTGCTGTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGATACCTACCTGATACCCCAGATCCCCCACAGCCACTACTGA XM_032163758.1:13...</span>
<span class="co">#&gt; [17] ATGGATTTATCTGCTCTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGACGCCTACCTGATACCCCAGATCCCCCACAGCCACTACTGA XM_030923119.1:18...</span>
<span class="co">#&gt; [18] ATGGATTTATCTGCTCTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGACGCCTACCTGATACCCCAGATCCCCCACAGCCACTACTGA XM_030923118.1:18...</span>
<span class="co">#&gt; [19] ATGGATTTACCTGCTGTTCGCGTTGAAGAAGTACAAAATGTCATT...CTGGACACCTACCTGATACCCCAGATCCCCCACAGCCACTACTGA XM_025363316.1:14...</span>
<span class="co">#&gt; [20] ATGGATTTATCTGCTGTTCGTGTTGAAGAAGTGCAAAATGTCCTT...CTGGACACCTACCTGATACCCCAGATCCCTCACAGCCACTACTGA XM_039475995.1:49...</span></code></pre></div>
<p>The sequence names</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/strtrim.html">strtrim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">brca1_aln</span><span class="op">)</span>, <span class="fl">100</span><span class="op">)</span>
<span class="co">#&gt;  [1] "NM_007298.3:20-2299_Homo_sapiens_BRCA1_DNA_repair_associated_(BRCA1)_transcript_variant_4_mRNA"      </span>
<span class="co">#&gt;  [2] "U64805.1:1-2280_Homo_sapiens_Brca1-delta11b_(Brca1)_mRNA_complete_cds"                               </span>
<span class="co">#&gt;  [3] "XM_031011560.1:233-2515_PREDICTED:_Gorilla_gorilla_gorilla_BRCA1_DNA_repair_associated_(BRCA1)_trans"</span>
<span class="co">#&gt;  [4] "XM_031011561.1:233-2512_PREDICTED:_Gorilla_gorilla_gorilla_BRCA1_DNA_repair_associated_(BRCA1)_trans"</span>
<span class="co">#&gt;  [5] "XM_031011562.1:163-2442_PREDICTED:_Gorilla_gorilla_gorilla_BRCA1_DNA_repair_associated_(BRCA1)_trans"</span>
<span class="co">#&gt;  [6] "XM_009432101.3:276-2555_PREDICTED:_Pan_troglodytes_BRCA1_DNA_repair_associated_(BRCA1)_transcript_va"</span>
<span class="co">#&gt;  [7] "XM_009432104.3:371-2650_PREDICTED:_Pan_troglodytes_BRCA1_DNA_repair_associated_(BRCA1)_transcript_va"</span>
<span class="co">#&gt;  [8] "XM_016930487.2:371-2650_PREDICTED:_Pan_troglodytes_BRCA1_DNA_repair_associated_(BRCA1)_transcript_va"</span>
<span class="co">#&gt;  [9] "XM_009432099.3:371-2653_PREDICTED:_Pan_troglodytes_BRCA1_DNA_repair_associated_(BRCA1)_transcript_va"</span>
<span class="co">#&gt; [10] "XM_034941183.1:254-2533_PREDICTED:_Pan_paniscus_BRCA1_DNA_repair_associated_(BRCA1)_transcript_varia"</span>
<span class="co">#&gt; [11] "XM_034941184.1:254-2533_PREDICTED:_Pan_paniscus_BRCA1_DNA_repair_associated_(BRCA1)_transcript_varia"</span>
<span class="co">#&gt; [12] "XM_034941185.1:248-2527_PREDICTED:_Pan_paniscus_BRCA1_DNA_repair_associated_(BRCA1)_transcript_varia"</span>
<span class="co">#&gt; [13] "XM_034941182.1:254-2536_PREDICTED:_Pan_paniscus_BRCA1_DNA_repair_associated_(BRCA1)_transcript_varia"</span>
<span class="co">#&gt; [14] "XM_032163757.1:145-2418_PREDICTED:_Hylobates_moloch_BRCA1_DNA_repair_associated_(BRCA1)_transcript_v"</span>
<span class="co">#&gt; [15] "XM_032163756.1:145-2421_PREDICTED:_Hylobates_moloch_BRCA1_DNA_repair_associated_(BRCA1)_transcript_v"</span>
<span class="co">#&gt; [16] "XM_032163758.1:139-2412_PREDICTED:_Hylobates_moloch_BRCA1_DNA_repair_associated_(BRCA1)_transcript_v"</span>
<span class="co">#&gt; [17] "XM_030923119.1:184-2463_PREDICTED:_Rhinopithecus_roxellana_BRCA1_DNA_repair_associated_(BRCA1)_trans"</span>
<span class="co">#&gt; [18] "XM_030923118.1:183-2465_PREDICTED:_Rhinopithecus_roxellana_BRCA1_DNA_repair_associated_(BRCA1)_trans"</span>
<span class="co">#&gt; [19] "XM_025363316.1:147-2426_PREDICTED:_Theropithecus_gelada_BRCA1_DNA_repair_associated_(BRCA1)_transcri"</span>
<span class="co">#&gt; [20] "XM_039475995.1:49-2328_PREDICTED:_Saimiri_boliviensis_boliviensis_BRCA1_DNA_repair_associated_(BRCA1"</span></code></pre></div>
<p>Next, function automorphism will be applied to represent the codon sequence in the Abelian group <span class="math inline">\(\mathbb{Z}_{64}\)</span> (i.e., the set of integers remainder modulo 64). The codon coordinates are requested on the cube ACGT. Following reference (<a href="#4">4</a>)), cubes are labeled based on the order of DNA bases used to define the sum operation.</p>
<p>In Z64, automorphisms are described as functions <span class="math inline">\(f(x) = k\,x \quad mod\,64\)</span>, where <span class="math inline">\(k\)</span> and <span class="math inline">\(x\)</span> are elements from the set of integers modulo 64. Below, in function <em>automorphism</em> three important arguments are given values: <em>group = “Z64”</em>, <em>cube = c(“ACGT”, “TGCA”)</em>, and <em>cube_alt = c(“CATG”, “GTAC”)</em>. Setting for group specifies on which group the automorphisms will be computed. These groups can be: “Z5”, “Z64”, “Z125”, and “Z5^3”.</p>
<p>In groups “Z64” and “Z125” not all the mutational events can be described as automorphisms from a given cube. So, a character string denoting pairs of “dual” the genetic-code cubes, as given in references (<a href="#2">1-4</a>)), is given as argument for <em>cube</em>. That is, the base pairs from the given cubes must be complementary each other. Such a cube pair are call dual cubes and, as shown in reference (<a href="#4">4</a>)), each pair integrates group. If automorphisms are not found in first set of dual cubes, then the algorithm search for automorphism in a alternative set of dual cubes.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Do not run it. This is included with package</span>

<span class="va">nams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"human_1"</span>,<span class="st">"human_2"</span>,<span class="st">"gorilla_1"</span>,<span class="st">"gorilla_2"</span>,<span class="st">"gorilla_3"</span>,
        <span class="st">"chimpanzee_1"</span>,<span class="st">"chimpanzee_2"</span>,<span class="st">"chimpanzee_3"</span>,<span class="st">"chimpanzee_4"</span>,
        <span class="st">"bonobos_1"</span>,<span class="st">"bonobos_2"</span>,<span class="st">"bonobos_3"</span>,<span class="st">"bonobos_4"</span>,<span class="st">"silvery_gibbon_1"</span>,
        <span class="st">"silvery_gibbon_1"</span>,<span class="st">"silvery_gibbon_3"</span>,<span class="st">"golden_monkey_1"</span>,
        <span class="st">"golden_monkey_2"</span>,<span class="st">"gelada_baboon"</span>,<span class="st">"bolivian_monkey"</span><span class="op">)</span>

<span class="va">brca1_autm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphism.html">automorphism</a></span><span class="op">(</span>
                      seqs <span class="op">=</span> <span class="va">brca1_aln</span>, 
                      group <span class="op">=</span> <span class="st">"Z64"</span>, 
                      cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"TGCA"</span><span class="op">)</span>,
                      cube_alt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CATG"</span>, <span class="st">"GTAC"</span><span class="op">)</span>,
                      nms <span class="op">=</span> <span class="va">nams</span>,
                      verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>Object <em>brca1_autm</em> is included with package and can be load typing:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">brca1_autm</span><span class="op">)</span>
<span class="va">brca1_autm</span>
<span class="co">#&gt; AutomorphismList object of length: 190</span>
<span class="co">#&gt; names(190): human_1.human_2 human_1.gorilla_1 human_1.gorilla_2 ... golden_monkey_2.gelada_baboon golden_monkey_2.bolivian_monkey gelada_baboon.bolivian_monkey </span>
<span class="co">#&gt; ------- </span>
<span class="co">#&gt; Automorphism object with 761 ranges and 6 metadata columns:</span>
<span class="co">#&gt;         seqnames    ranges strand |        seq1        seq2    coord1    coord2      autm        cube</span>
<span class="co">#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt;</span>
<span class="co">#&gt;     [1]        1         1      + |         ATG         ATG        50        50         1        ACGT</span>
<span class="co">#&gt;     [2]        1         2      + |         GAT         GAT        11        11         1        ACGT</span>
<span class="co">#&gt;     [3]        1         3      + |         TTA         TTA        60        60         1        ACGT</span>
<span class="co">#&gt;     [4]        1         4      + |         TCT         TCT        31        31         1        ACGT</span>
<span class="co">#&gt;     [5]        1         5      + |         GCT         GCT        27        27         1        ACGT</span>
<span class="co">#&gt;     ...      ...       ...    ... .         ...         ...       ...       ...       ...         ...</span>
<span class="co">#&gt;   [757]        1       757      + |         CAC         CAC         5         5         1        ACGT</span>
<span class="co">#&gt;   [758]        1       758      + |         AGC         AGC        33        33         1        ACGT</span>
<span class="co">#&gt;   [759]        1       759      + |         CAC         CAC         5         5         1        ACGT</span>
<span class="co">#&gt;   [760]        1       760      + |         TAC         TAC        13        13         1        ACGT</span>
<span class="co">#&gt;   [761]        1       761      + |         TGA         TGA        44        44         1        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; ...</span>
<span class="co">#&gt; &lt;189 more DFrame element(s)&gt;</span>
<span class="co">#&gt; Two slots: 'DataList' &amp; 'SeqRanges'</span>
<span class="co">#&gt; -------</span></code></pre></div>
<p>An AutomorphismList class object is returned. The codon sequences (<em>seq1</em> and <em>seq2</em>) with their corresponding coordinates (left) are returned, as well as the coordinated representation on <span class="math inline">\(\mathbb{Z}_{64}\)</span> (<em>coord1</em> and <em>coord2</em>). Observe that two new columns were added, the automorphism coefficient <span class="math inline">\(k\)</span> (named as <em>brca1_autm</em>) and the genetic-code cube where the automorphism was found. By convention the DNA sequence is given for the positive strand. Since the <em>dual cube</em> of <strong>“ACGT”</strong> corresponds to the complementary base order <strong>TGCA</strong>, automorphisms described by the cube <strong>TGCA</strong> represent mutational events affecting the DNA negative strand (-).</p>
<p>The last result can be summarized by gene regions as follow:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">brca1_autm_range</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphismByRanges.html">automorphismByRanges</a></span><span class="op">(</span><span class="va">brca1_autm</span>, 
                                  min.len <span class="op">=</span> <span class="fl">2</span>,
                                  num.cores <span class="op">=</span> <span class="fl">10L</span>,
                                  verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">brca1_autm_range</span>
<span class="co">#&gt; GRangesList object of length 189:</span>
<span class="co">#&gt; $human_1.human_2</span>
<span class="co">#&gt; GRanges object with 3 ranges and 1 metadata column:</span>
<span class="co">#&gt;       seqnames    ranges strand |        cube</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;   [1]        1     1-349      + |        ACGT</span>
<span class="co">#&gt;   [2]        1       350      + |        Gaps</span>
<span class="co">#&gt;   [3]        1   351-761      + |        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $human_1.gorilla_1</span>
<span class="co">#&gt; GRanges object with 7 ranges and 1 metadata column:</span>
<span class="co">#&gt;       seqnames    ranges strand |        cube</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;   [1]        1     1-233      + |        ACGT</span>
<span class="co">#&gt;   [2]        1       234      + |        TGCA</span>
<span class="co">#&gt;   [3]        1   235-349      + |        ACGT</span>
<span class="co">#&gt;   [4]        1       350      + |        Trnl</span>
<span class="co">#&gt;   [5]        1   351-622      + |        ACGT</span>
<span class="co">#&gt;   [6]        1       623      + |        TGCA</span>
<span class="co">#&gt;   [7]        1   624-761      + |        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $human_1.gorilla_2</span>
<span class="co">#&gt; GRanges object with 7 ranges and 1 metadata column:</span>
<span class="co">#&gt;       seqnames    ranges strand |        cube</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;   [1]        1     1-233      + |        ACGT</span>
<span class="co">#&gt;   [2]        1       234      + |        TGCA</span>
<span class="co">#&gt;   [3]        1   235-349      + |        ACGT</span>
<span class="co">#&gt;   [4]        1       350      + |        Gaps</span>
<span class="co">#&gt;   [5]        1   351-622      + |        ACGT</span>
<span class="co">#&gt;   [6]        1       623      + |        TGCA</span>
<span class="co">#&gt;   [7]        1   624-761      + |        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ...</span>
<span class="co">#&gt; &lt;186 more elements&gt;</span></code></pre></div>
<p>That is, function <em>automorphismByRanges</em> permits the classification of the pairwise alignment of protein-coding sub-regions based on the mutational events observed on it quantitatively represented as automorphisms on genetic-code cubes.</p>
<p>Searching for automorphisms on <span class="math inline">\(\mathbb{Z}_{64}\)</span> permits us a quantitative differentiation between mutational events at different codon positions from a given DNA protein-encoding region. As shown in reference (<a href="#4">4</a>)) a set of different cubes can be applied to describe the best evolutionary aminoacid scale highly correlated with aminoacid physicochemical properties describing the observed evolutionary process in a given protein.</p>
<p>More information about this subject can be found in the supporting material from reference (<a href="#4">4</a>)) at GitHub <a href="https://github.com/genomaths/GenomeAlgebra_SymmetricGroup">GenomeAlgebra_SymmetricGroup</a>, particularly by interacting with the Mathematica notebook <a href="https://github.com/genomaths/GenomeAlgebra_SymmetricGroup/blob/master/Genetic-Code-Scales_of_Amino-Acids.nb">Genetic-Code-Scales_of_Amino-Acids.nb</a>.</p>
<div id="bar-plot-automorphism-distribution-by-cubes" class="section level2">
<h2 class="hasAnchor">
<a href="#bar-plot-automorphism-distribution-by-cubes" class="anchor"></a>Bar plot automorphism distribution by cubes</h2>
<p>The automorphism distribution by cubes can be summarized in the bar-plot graphic</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/unlist.html">unlist</a></span><span class="op">(</span><span class="va">brca1_autm_range</span><span class="op">)</span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">cts</span><span class="op">$</span><span class="va">cube</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, cex <span class="op">=</span> <span class="fl">1</span>, font <span class="op">=</span> <span class="fl">2</span>, las <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html">barplot</a></span><span class="op">(</span><span class="va">counts</span>, main<span class="op">=</span><span class="st">"Automorphism distribution"</span>,
        xlab<span class="op">=</span><span class="st">"Genetic-code cube representation"</span>,
        col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"darkblue"</span>,<span class="st">"red"</span>, <span class="st">"darkgreen"</span>, <span class="st">"magenta"</span><span class="op">)</span>, 
        border <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>,line <span class="op">=</span> <span class="op">-</span><span class="fl">6</span>, at <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.7</span>, <span class="fl">1.9</span>, <span class="fl">3.1</span>, <span class="fl">4.3</span><span class="op">)</span>,
      text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span> <span class="va">counts</span> <span class="op">)</span>, 
      col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"white"</span>, <span class="st">"black"</span>, <span class="st">"yellow"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="automorphism_on_msa_brca1_files/figure-html/barplot-1.png" width="576"></p>
</div>
<div id="grouping-automorphism-by-automorphisms-coefficients" class="section level2">
<h2 class="hasAnchor">
<a href="#grouping-automorphism-by-automorphisms-coefficients" class="anchor"></a>Grouping automorphism by automorphism’s coefficients</h2>
<p>Automorphisms with the same automorphism’s coefficients can be grouped.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">autby_coef</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphismByCoef.html">automorphismByCoef</a></span><span class="op">(</span><span class="va">brca1_autm</span>, 
                                 verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">autby_coef</span>
<span class="co">#&gt; AutomorphismByCoefList object of length 190:</span>
<span class="co">#&gt; $human_1.human_2</span>
<span class="co">#&gt; AutomorphismByCoef object with 239 ranges and 5 metadata columns:</span>
<span class="co">#&gt;         seqnames    ranges strand |        seq1        seq2      autm    mut_type        cube</span>
<span class="co">#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;     [1]        1     1-238      + |         ATG         ATG         1         HHH        ACGT</span>
<span class="co">#&gt;     [2]        1     1-238      + |         GAT         GAT         1         HHH        ACGT</span>
<span class="co">#&gt;     [3]        1     1-238      + |         TTA         TTA         1         HHH        ACGT</span>
<span class="co">#&gt;     [4]        1     1-238      + |         TCT         TCT         1         HHH        ACGT</span>
<span class="co">#&gt;     [5]        1     1-238      + |         GCT         GCT         1         HHH        ACGT</span>
<span class="co">#&gt;     ...      ...       ...    ... .         ...         ...       ...         ...         ...</span>
<span class="co">#&gt;   [235]        1   511-761      + |         CCC         CCC         1         HHH        ACGT</span>
<span class="co">#&gt;   [236]        1   511-761      + |         CTT         CTT         1         HHH        ACGT</span>
<span class="co">#&gt;   [237]        1   511-761      + |         CCT         CCT         1         HHH        ACGT</span>
<span class="co">#&gt;   [238]        1   511-761      + |         ATA         ATA         1         HHH        ACGT</span>
<span class="co">#&gt;   [239]        1   511-761      + |         TGA         TGA         1         HHH        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ...</span>
<span class="co">#&gt; &lt;189 more elements&gt;</span></code></pre></div>
</div>
<div id="conserved-and-non-conserved-regions" class="section level2">
<h2 class="hasAnchor">
<a href="#conserved-and-non-conserved-regions" class="anchor"></a>Conserved and non-conserved regions</h2>
<p>A specific function is provided to get the coordinates of conserved and non-conserved regions, which can be used in further downstream analyses.</p>
<p>The conserved regions from each pairwise comparison are retrieved typing:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/conserved_regions.html">conserved_regions</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">autby_coef</span><span class="op">)</span>
<span class="co">#&gt; AutomorphismByCoef object with 80625 ranges and 5 metadata columns:</span>
<span class="co">#&gt;                                   seqnames    ranges strand |        seq1        seq2      autm    mut_type        cube</span>
<span class="co">#&gt;                                      &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;             human_1.gelada_baboon        1       1-3      + |         ATG         ATG         1         HHH        ACGT</span>
<span class="co">#&gt;             human_1.gelada_baboon        1       1-3      + |         GAT         GAT         1         HHH        ACGT</span>
<span class="co">#&gt;             human_1.gelada_baboon        1       1-3      + |         TTA         TTA         1         HHH        ACGT</span>
<span class="co">#&gt;             human_2.gelada_baboon        1       1-3      + |         ATG         ATG         1         HHH        ACGT</span>
<span class="co">#&gt;             human_2.gelada_baboon        1       1-3      + |         GAT         GAT         1         HHH        ACGT</span>
<span class="co">#&gt;                               ...      ...       ...    ... .         ...         ...       ...         ...         ...</span>
<span class="co">#&gt;   golden_monkey_2.bolivian_monkey        1   757-761      + |         TGA         TGA         1         HHH        ACGT</span>
<span class="co">#&gt;     gelada_baboon.bolivian_monkey        1   757-761      + |         CAC         CAC         1         HHH        ACGT</span>
<span class="co">#&gt;     gelada_baboon.bolivian_monkey        1   757-761      + |         AGC         AGC         1         HHH        ACGT</span>
<span class="co">#&gt;     gelada_baboon.bolivian_monkey        1   757-761      + |         TAC         TAC         1         HHH        ACGT</span>
<span class="co">#&gt;     gelada_baboon.bolivian_monkey        1   757-761      + |         TGA         TGA         1         HHH        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
<p>The non-conserved regions from each pairwise comparison are obtained with the same function but different settings:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ncs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/conserved_regions.html">conserved_regions</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">autby_coef</span>, conserved <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">ncs</span>
<span class="co">#&gt; AutomorphismByCoef object with 8956 ranges and 5 metadata columns:</span>
<span class="co">#&gt;                                    seqnames    ranges strand |        seq1        seq2      autm    mut_type</span>
<span class="co">#&gt;                                       &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;character&gt;</span>
<span class="co">#&gt;              human_1.gelada_baboon        1         4      + |         TCT         CCT         9         YHH</span>
<span class="co">#&gt;              human_2.gelada_baboon        1         4      + |         TCT         CCT         9         YHH</span>
<span class="co">#&gt;            gorilla_1.gelada_baboon        1         4      + |         TCT         CCT         9         YHH</span>
<span class="co">#&gt;            gorilla_2.gelada_baboon        1         4      + |         TCT         CCT         9         YHH</span>
<span class="co">#&gt;            gorilla_3.gelada_baboon        1         4      + |         TCT         CCT         9         YHH</span>
<span class="co">#&gt;                                ...      ...       ...    ... .         ...         ...       ...         ...</span>
<span class="co">#&gt;   silvery_gibbon_1.bolivian_monkey        1       756      + |         CCC         CCT        59         HHY</span>
<span class="co">#&gt;   silvery_gibbon_3.bolivian_monkey        1       756      + |         CCC         CCT        59         HHY</span>
<span class="co">#&gt;    golden_monkey_1.bolivian_monkey        1       756      + |         CCC         CCT        59         HHY</span>
<span class="co">#&gt;    golden_monkey_2.bolivian_monkey        1       756      + |         CCC         CCT        59         HHY</span>
<span class="co">#&gt;      gelada_baboon.bolivian_monkey        1       756      + |         CCC         CCT        59         HHY</span>
<span class="co">#&gt;                                           cube</span>
<span class="co">#&gt;                                    &lt;character&gt;</span>
<span class="co">#&gt;              human_1.gelada_baboon        ACGT</span>
<span class="co">#&gt;              human_2.gelada_baboon        ACGT</span>
<span class="co">#&gt;            gorilla_1.gelada_baboon        ACGT</span>
<span class="co">#&gt;            gorilla_2.gelada_baboon        ACGT</span>
<span class="co">#&gt;            gorilla_3.gelada_baboon        ACGT</span>
<span class="co">#&gt;                                ...         ...</span>
<span class="co">#&gt;   silvery_gibbon_1.bolivian_monkey        ACGT</span>
<span class="co">#&gt;   silvery_gibbon_3.bolivian_monkey        ACGT</span>
<span class="co">#&gt;    golden_monkey_1.bolivian_monkey        ACGT</span>
<span class="co">#&gt;    golden_monkey_2.bolivian_monkey        ACGT</span>
<span class="co">#&gt;      gelada_baboon.bolivian_monkey        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
</div>
</div>
<div id="subsetting-regions-of-interest" class="section level1">
<h1 class="hasAnchor">
<a href="#subsetting-regions-of-interest" class="anchor"></a>Subsetting regions of interest</h1>
<p>The application of some of the available bioinformatic tools is straightforward. For example, knowing the positions for the conserved and non-conserved regions, we can subset into the alignment region of potential interest.</p>
<p>The coordinates carrying mutational events are:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">ncs</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;   [1]   4   6   7  11  15  20  25  29  45  52  60  63  71  74  77  85  86  91  94 100 101 104 113 122 124 128 132 138</span>
<span class="co">#&gt;  [29] 139 142 143 146 148 151 152 155 159 161 164 167 170 172 175 176 181 182 190 191 193 194 196 205 214 216 225 227</span>
<span class="co">#&gt;  [57] 231 233 234 239 240 241 244 245 247 251 252 253 254 267 269 270 275 276 279 281 282 286 287 292 293 299 300 301</span>
<span class="co">#&gt;  [85] 310 318 319 323 324 326 328 329 330 331 333 339 340 350 357 359 360 361 363 369 370 374 375 376 378 394 396 399</span>
<span class="co">#&gt; [113] 402 406 407 410 417 418 421 425 428 431 433 435 438 442 443 444 445 447 453 458 461 466 472 476 477 478 480 482</span>
<span class="co">#&gt; [141] 483 485 486 487 489 499 501 502 503 505 507 509 510 511 516 518 521 522 523 525 526 532 535 536 537 545 549 550</span>
<span class="co">#&gt; [169] 552 554 560 564 568 569 572 576 578 584 585 590 597 601 602 605 623 647 653 654 659 660 665 669 678 681 683 685</span>
<span class="co">#&gt; [197] 693 701 702 703 704 716 726 728 730 731 748 749 753 756</span></code></pre></div>
<p>We will use function <a href="https://www.rdocumentation.org/packages/Biostrings/versions/2.40.2/topics/XStringSet-class">DNAStringSet</a> from the Bioconductor R package <a href="https://kasperdanielhansen.github.io/genbioconductor/html/Biostrings.html">Biostrings</a> to subset the MSA of primate cytochrome c. </p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">brca1_aln_region</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html">DNAStringSet</a></span><span class="op">(</span><span class="va">brca1_aln</span>, start <span class="op">=</span> <span class="fl">37</span><span class="op">*</span><span class="fl">3</span> <span class="op">-</span> <span class="fl">2</span>, end <span class="op">=</span> <span class="fl">42</span><span class="op">*</span><span class="fl">3</span><span class="op">)</span>
<span class="va">brca1_aln_region</span>
<span class="co">#&gt; DNAStringSet object of length 20:</span>
<span class="co">#&gt;      width seq                                                                                      names               </span>
<span class="co">#&gt;  [1]    18 ACAAAGTGTGACCACATA                                                                       NM_007298.3:20-22...</span>
<span class="co">#&gt;  [2]    18 ACAAAGTGTGACCACATA                                                                       U64805.1:1-2280_H...</span>
<span class="co">#&gt;  [3]    18 ACAAAGTGTGACCACATA                                                                       XM_031011560.1:23...</span>
<span class="co">#&gt;  [4]    18 ACAAAGTGTGACCACATA                                                                       XM_031011561.1:23...</span>
<span class="co">#&gt;  [5]    18 ACAAAGTGTGACCACATA                                                                       XM_031011562.1:16...</span>
<span class="co">#&gt;  ...   ... ...</span>
<span class="co">#&gt; [16]    18 ACAAAGTGTGACCACATA                                                                       XM_032163758.1:13...</span>
<span class="co">#&gt; [17]    18 ACAAAGTGTGACCACATA                                                                       XM_030923119.1:18...</span>
<span class="co">#&gt; [18]    18 ACAAAGTGTGACCACATA                                                                       XM_030923118.1:18...</span>
<span class="co">#&gt; [19]    18 ACAAAGTGTGACCACATA                                                                       XM_025363316.1:14...</span>
<span class="co">#&gt; [20]    18 ACAAAGTGTGACCACATA                                                                       XM_039475995.1:49...</span></code></pre></div>
<p>Next, function <a href="https://rdrr.io/cran/ggseqlogo/man/ggseqlogo.html">ggseqlogo</a> from the R package <a href="https://rdrr.io/cran/ggseqlogo/">ggseqlogo</a> will be applied to get the logo sequence from the selected MSA region.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggseqlogo/man/ggseqlogo.html">ggseqlogo</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">brca1_aln_region</span><span class="op">)</span> <span class="op">)</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; = "none")` instead.</span></code></pre></div>
<p><img src="automorphism_on_msa_brca1_files/figure-html/seqlogo-1.png" width="700"></p>
<p>The corresponding aminoacid sequence will be</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">brca1_aln_region_aa</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/translate.html">translate</a></span><span class="op">(</span><span class="va">brca1_aln_region</span><span class="op">)</span>
<span class="va">brca1_aln_region_aa</span>
<span class="co">#&gt; AAStringSet object of length 20:</span>
<span class="co">#&gt;      width seq                                                                                      names               </span>
<span class="co">#&gt;  [1]     6 TKCDHI                                                                                   NM_007298.3:20-22...</span>
<span class="co">#&gt;  [2]     6 TKCDHI                                                                                   U64805.1:1-2280_H...</span>
<span class="co">#&gt;  [3]     6 TKCDHI                                                                                   XM_031011560.1:23...</span>
<span class="co">#&gt;  [4]     6 TKCDHI                                                                                   XM_031011561.1:23...</span>
<span class="co">#&gt;  [5]     6 TKCDHI                                                                                   XM_031011562.1:16...</span>
<span class="co">#&gt;  ...   ... ...</span>
<span class="co">#&gt; [16]     6 TKCDHI                                                                                   XM_032163758.1:13...</span>
<span class="co">#&gt; [17]     6 TKCDHI                                                                                   XM_030923119.1:18...</span>
<span class="co">#&gt; [18]     6 TKCDHI                                                                                   XM_030923118.1:18...</span>
<span class="co">#&gt; [19]     6 TKCDHI                                                                                   XM_025363316.1:14...</span>
<span class="co">#&gt; [20]     6 TKCDHI                                                                                   XM_039475995.1:49...</span></code></pre></div>
<p>And the corresponding aminoacid sequence logo will is</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggseqlogo/man/ggseqlogo.html">ggseqlogo</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">brca1_aln_region_aa</span><span class="op">)</span> <span class="op">)</span></code></pre></div>
<p><img src="automorphism_on_msa_brca1_files/figure-html/aa_logo-1.png" width="700"></p>
<p>That is, the first mutational event reported in the codon sequence correspond to a synonymous mutation.</p>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p>1.<a name="1"></a> Sanchez R, Morgado E, Grau R. Gene algebra from a genetic code algebraic structure. J Math Biol. 2005 Oct;51(4):431-57. doi: 10.1007/s00285-005-0332-8. Epub 2005 Jul 13. PMID: 16012800. ( <a href="https://arxiv.org/pdf/q-bio/0412033.pdf">PDF</a>).</p>
<p>2.<a name="2"></a> Robersy Sanchez, Jesús Barreto (2021) Genomic Abelian Finite Groups. <a href="https://doi.org/10.1101/2021.06.01.446543">doi: 10.1101/2021.06.01.446543</a>.</p>
<p>3.<a name="3"></a> M. V José, E.R. Morgado, R. Sánchez, T. Govezensky, The 24 possible algebraic representations of the standard genetic code in six or in three dimensions, Adv. Stud. Biol. 4 (2012) 119–152.<a href="https://is.gd/na9eap">PDF</a>.</p>
<p>4.<a name="4"></a> R. Sanchez. Symmetric Group of the Genetic–Code Cubes. Effect of the Genetic–Code Architecture on the Evolutionary Process MATCH Commun. Math. Comput. Chem. 79 (2018) 527-560. <a href="https://bit.ly/2Z9mjM7">PDF</a>.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://genomaths.com">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
