<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Automorphism analysis on COVID-19 data • GenomAutomorphism</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Automorphism analysis on COVID-19 data">
<meta property="og:description" content="GenomAutomorphism">
<meta property="og:image" content="https://genomaths.github.io/GenomAutomorphism/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomAutomorphism</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/GenomAutomorphism.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/GenomAutomorphism.html">GenomAutomorphism</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/GenomAutomorphism/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="covid_19_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Automorphism analysis on COVID-19 data</h1>
                        <h4 class="author">Robersy Sanchez</h4>
            <address class="author_afil">
      Department of Biology. Pennsylvania State University, University Park, PA 16802<br><a class="author_email" href="mailto:#"></a><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a>
      </address>
                  
            <h4 class="date">04 March 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/master/vignettes/covid_19.Rmd"><code>vignettes/covid_19.Rmd</code></a></small>
      <div class="hidden name"><code>covid_19.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Analysis of COVID-19 mutational events in terms of automorphisms.</p>
    </div>
    
<style type="text/css">

div.main-container {
  max-width: 2200px !important;
  padding: 2px;
}

.main-container {
  max-width: 2200px !important;
  padding: 2px;
}

.watch-out {
  max-width: 2200px !important;
  margin-left: auto;
  margin-right: auto;
  background-color: #c9feed;
  font-weight: bold
  font-size: 100%;
  padding: 2px;
}

pre, code {
  max-width: 2200px !important;
  margin-left: auto;
  margin-right: auto;
  background-color: #c9feed;
  font-weight: bold
  font-size: 100px;
  padding: 2px;
}

code {
    font-size: 100%;
    font-family: Consolas,Monaco,"Courier New",monospace;
    border-radius: 0;
}

body {
  max-width: 2200px !important;
  margin: auto;
}  

code.r{
  font-size: 100%;
}
</style>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings">Biostrings</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames, dirname, do.call, duplicated, eval,</span>
<span class="co">#&gt;     evalq, Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget, order,</span>
<span class="co">#&gt;     paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span>
<span class="co">#&gt;     table, tapply, union, unique, unsplit, which.max, which.min</span>
<span class="co">#&gt; Loading required package: S4Vectors</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid, I, unname</span>
<span class="co">#&gt; Loading required package: IRanges</span>
<span class="co">#&gt; Loading required package: XVector</span>
<span class="co">#&gt; Loading required package: GenomeInfoDb</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'Biostrings'</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     strsplit</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/GenomAutomorphism">GenomAutomorphism</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">usefr</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'usefr'</span>
<span class="co">#&gt; The following object is masked from 'package:BiocGenerics':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     unlist</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     unlist</span></code></pre></div>
<div id="sars-coronavirus-gz02-vs-bat-sl-covzc45" class="section level1">
<h1 class="hasAnchor">
<a href="#sars-coronavirus-gz02-vs-bat-sl-covzc45" class="anchor"></a>SARS coronavirus GZ02 vs bat-SL-CoVZC45</h1>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">URL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"https://github.com/genomaths/seqalignments/raw/master/"</span>, 
        <span class="st">"COVID-19/AY390556.1_265-13398_13398-21485_RNA-POL_SARS_COVI_GZ02.fas"</span><span class="op">)</span>

<span class="va">covid_aln2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html">readDNAMultipleAlignment</a></span><span class="op">(</span>filepath <span class="op">=</span> <span class="va">URL</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">covid_aut</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphism.html">automorphism</a></span><span class="op">(</span>
                    seq <span class="op">=</span> <span class="va">covid_aln2</span>,
                    group <span class="op">=</span> <span class="st">"Z64"</span>,
                    cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"TGCA"</span><span class="op">)</span>,
                    cube_alt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CATG"</span>, <span class="st">"GTAC"</span><span class="op">)</span>,
                    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="va">covid_aut</span>
<span class="co">#&gt; Automorphism object with 7100 ranges and 6 metadata columns:</span>
<span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2    coord1    coord2      autm        cube</span>
<span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt;</span>
<span class="co">#&gt;      [1]        1         1      + |         ATG         ATG        50        50         1        ACGT</span>
<span class="co">#&gt;      [2]        1         2      + |         GAG         GAG        10        10         1        ACGT</span>
<span class="co">#&gt;      [3]        1         3      + |         AGC         AGC        33        33         1        ACGT</span>
<span class="co">#&gt;      [4]        1         4      + |         CTT         CTT        55        55         1        ACGT</span>
<span class="co">#&gt;      [5]        1         5      + |         GTT         GTC        59        57        27        ACGT</span>
<span class="co">#&gt;      ...      ...       ...    ... .         ...         ...       ...       ...       ...         ...</span>
<span class="co">#&gt;   [7096]        1      7096      + |         CTT         CTT        55        55         1        ACGT</span>
<span class="co">#&gt;   [7097]        1      7097      + |         GTT         GTT        59        59         1        ACGT</span>
<span class="co">#&gt;   [7098]        1      7098      + |         AAT         AAC         3         1        43        ACGT</span>
<span class="co">#&gt;   [7099]        1      7099      + |         AAC         AAC         1         1         1        ACGT</span>
<span class="co">#&gt;   [7100]        1      7100      + |         TAA         TAA        12        12         1        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">covid_aut</span><span class="op">$</span><span class="va">cube</span><span class="op">[</span> <span class="va">covid_aut</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="fl">1</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">covid_aut</span><span class="op">$</span><span class="va">autm</span><span class="op">)</span> <span class="op">]</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, cex <span class="op">=</span> <span class="fl">0.9</span>, font <span class="op">=</span> <span class="fl">2</span>, mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">6</span>,<span class="fl">4</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html">barplot</a></span><span class="op">(</span><span class="va">counts</span>, main<span class="op">=</span><span class="st">"Automorphism distribution"</span>,
        xlab<span class="op">=</span><span class="st">"Genetic-code cube representation"</span>,
        ylab<span class="op">=</span><span class="st">"Fixed mutational events"</span>,
        col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"darkblue"</span>,<span class="st">"red"</span>, <span class="st">"darkgreen"</span><span class="op">)</span>, 
        border <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co">#ylim = c(0, 2000), </span>
        cex.lab <span class="op">=</span> <span class="fl">2</span>, cex.main <span class="op">=</span> <span class="fl">1.5</span>, cex.names <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">2</span>, at <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">500</span>, <span class="fl">1000</span>, <span class="fl">1500</span>, <span class="fl">2000</span><span class="op">)</span>, cex.axis <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>,line <span class="op">=</span> <span class="op">-</span><span class="fl">1.5</span>, at <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.7</span>, <span class="fl">1.9</span>, <span class="fl">3.1</span>, <span class="fl">4.3</span>, <span class="fl">5.5</span><span class="op">)</span>,
      text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span> <span class="va">counts</span> <span class="op">)</span>, cex <span class="op">=</span> <span class="fl">1.4</span>,
      col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"white"</span>,<span class="st">"yellow"</span>, <span class="st">"black"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="covid_19_files/figure-html/barplot-1.png" width="576"></p>
</div>
<div id="analysis-of-conserved-regions" class="section level1">
<h1 class="hasAnchor">
<a href="#analysis-of-conserved-regions" class="anchor"></a>Analysis of conserved regions</h1>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">conserv2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/conserved_regions.html">conserved_regions</a></span><span class="op">(</span><span class="va">covid_aut</span><span class="op">)</span>
<span class="va">conserv2</span>
<span class="co">#&gt; AutomorphismByCoef object with 3813 ranges and 5 metadata columns:</span>
<span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2      autm    mut_type        cube</span>
<span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;      [1]        1       1-4      + |         ATG         ATG         1         HHH        ACGT</span>
<span class="co">#&gt;      [2]        1       1-4      + |         GAG         GAG         1         HHH        ACGT</span>
<span class="co">#&gt;      [3]        1       1-4      + |         AGC         AGC         1         HHH        ACGT</span>
<span class="co">#&gt;      [4]        1       1-4      + |         CTT         CTT         1         HHH        ACGT</span>
<span class="co">#&gt;      [5]        1         7      + |         GGT         GGT         1         HHH        ACGT</span>
<span class="co">#&gt;      ...      ...       ...    ... .         ...         ...       ...         ...         ...</span>
<span class="co">#&gt;   [3809]        1 7093-7094      + |         GAT         GAT         1         HHH        ACGT</span>
<span class="co">#&gt;   [3810]        1 7096-7097      + |         CTT         CTT         1         HHH        ACGT</span>
<span class="co">#&gt;   [3811]        1 7096-7097      + |         GTT         GTT         1         HHH        ACGT</span>
<span class="co">#&gt;   [3812]        1 7099-7100      + |         AAC         AAC         1         HHH        ACGT</span>
<span class="co">#&gt;   [3813]        1 7099-7100      + |         TAA         TAA         1         HHH        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Next, function <a href="https://rdrr.io/github/genomaths/usefr/man/fitCDF.html">fitCDF</a> is applied to find the best fitted probability function to the the sizes of conserved regions.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">widths</span> <span class="op">&lt;-</span> <span class="fu">width</span><span class="op">(</span><span class="va">conserv2</span><span class="op">)</span>
<span class="va">dist2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/fitCDF.html">fitCDF</a></span><span class="op">(</span><span class="va">widths</span>, distNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">7</span>, <span class="fl">10</span>, <span class="fl">11</span>, <span class="fl">19</span>, <span class="fl">20</span><span class="op">)</span>, plot <span class="op">=</span> <span class="cn">TRUE</span>, 
               loss.fun <span class="op">=</span> <span class="st">"cauchy"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Log-normal distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Half-Normal distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Gamma distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Generalized 3P Gamma distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Weibull distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Exponential distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting 2P Exponential distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt;  * Estimating Studentized residuals for Generalized 3P Gamma distribution</span>
<span class="co">#&gt;  * Plots for Generalized 3P Gamma distribution...</span></code></pre></div>
<p><img src="covid_19_files/figure-html/dist-1.png" width="1152"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dist2</span>
<span class="co">#&gt; ggamma CDF model</span>
<span class="co">#&gt; ------</span>
<span class="co">#&gt; Parameters:</span>
<span class="co">#&gt;          Estimate  Std. Error   t value   Pr(&gt;|t|)    </span>
<span class="co">#&gt; alpha 0.621461074 0.002930067 212.09793 &lt; 2.22e-16 ***</span>
<span class="co">#&gt; scale 0.970537071 0.021586713  44.95993 &lt; 2.22e-16 ***</span>
<span class="co">#&gt; psi   2.441111337 0.021649209 112.75753 &lt; 2.22e-16 ***</span>
<span class="co">#&gt; ---</span>
<span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residual standard error: 1.973562e-05 on 3810 degrees of freedom</span>
<span class="co">#&gt; Number of iterations to termination: 29 </span>
<span class="co">#&gt; Reason for termination: Relative error in the sum of squares is at most `ftol'. </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Goodness of fit:</span>
<span class="co">#&gt;     Adj.R.Square rho R.Cross.val       AIC</span>
<span class="co">#&gt; gof            1   1   0.9998381 -29782.22</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>lwd <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">widths</span>, <span class="fl">14</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">1</span>, family <span class="op">=</span> <span class="st">"serif"</span>,
family <span class="op">=</span> <span class="st">"serif"</span>, col <span class="op">=</span> <span class="st">"cyan1"</span>, cex.main <span class="op">=</span> <span class="fl">0.9</span>,
main <span class="op">=</span> <span class="st">"Histogram and best fitted CDF model for conserved region sizes"</span>,
xlab <span class="op">=</span> <span class="st">"Conserved region size"</span><span class="op">)</span>
<span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">150</span>, by <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span><span class="op">(</span><span class="va">x1</span>, <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/ggamma.html">dggamma</a></span><span class="op">(</span><span class="va">x1</span>,
                  alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">dist2</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,
                  scale <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">dist2</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,
                  psi <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">dist2</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>
                  <span class="op">)</span>,
      col <span class="op">=</span> <span class="st">"red"</span>, lwd <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="covid_19_files/figure-html/hist-1.png" width="576"></p>
<p>After apply Bootstrap test for Goodness of fit, tt seems to be that we have no reasons to reject the hypothesis that the sizes of conserved regions follows Generalized Gamma distribution.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/usefr/man/mcgoftest.html">mcgoftest</a></span><span class="op">(</span> 
          varobj <span class="op">=</span> <span class="va">widths</span>, 
          model <span class="op">=</span> <span class="va">dist2</span>,
          stat <span class="op">=</span> <span class="st">"ks"</span><span class="op">)</span>
<span class="co">#&gt; *** Permutation GoF testing based on Kolmogorov-Smirnov statistic ( parametric approach )  ...</span>
<span class="co">#&gt;       KS.stat.D      mc_p.value KS.stat.p.value     sample.size       num.sampl </span>
<span class="co">#&gt;       0.1671919       1.0000000       0.0000000    3813.0000000     999.0000000</span>

<span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\n===========================\n"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ===========================</span>
<span class="fu"><a href="https://rdrr.io/pkg/usefr/man/mcgoftest.html">mcgoftest</a></span><span class="op">(</span>
          varobj <span class="op">=</span> <span class="va">widths</span>,
          model <span class="op">=</span> <span class="va">dist2</span>,
          stat <span class="op">=</span> <span class="st">"rmse"</span>,
          breaks <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>
<span class="co">#&gt; *** Permutation GoF testing based on Root Mean Square statistic ( parametric approach )  ...</span>
<span class="co">#&gt;        rmse  mc_p.value sample.size   num.sampl </span>
<span class="co">#&gt;    13.75682     1.00000  3813.00000   999.00000</span></code></pre></div>
<div id="grouping-automorphism-by-automorphisms-coefficients--types-of-mutations" class="section level2">
<h2 class="hasAnchor">
<a href="#grouping-automorphism-by-automorphisms-coefficients--types-of-mutations" class="anchor"></a>Grouping automorphism by automorphism’s coefficients. Types of mutations</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">autby_coef2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphismByCoef.html">automorphismByCoef</a></span><span class="op">(</span><span class="va">covid_aut</span><span class="op">)</span>
<span class="va">autby_coef2</span> <span class="op">&lt;-</span> <span class="va">autby_coef2</span><span class="op">[</span> <span class="va">autby_coef2</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">autby_coef2</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="op">-</span><span class="fl">1</span>  <span class="op">]</span></code></pre></div>
<p>Alignment gaps denoting indel mutations are labeled as “—”.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">autby_coef2</span><span class="op">$</span><span class="va">mut_type</span><span class="op">)</span>
<span class="va">counts2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="va">counts2</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">count.</span> <span class="op">&lt;-</span> <span class="va">counts2</span><span class="op">[</span> <span class="va">counts2</span> <span class="op">&gt;</span> <span class="fl">9</span> <span class="op">]</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, cex.axis <span class="op">=</span> <span class="fl">2</span>, font <span class="op">=</span> <span class="fl">2</span>, las <span class="op">=</span> <span class="fl">1</span>, 
    cex.main <span class="op">=</span> <span class="fl">1.4</span>, mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">3</span>,<span class="fl">4</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html">barplot</a></span><span class="op">(</span><span class="va">count.</span>, main<span class="op">=</span><span class="st">"Automorphism distribution per Mutation type"</span>,
        col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"yellow"</span>, <span class="st">"blue"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">36</span><span class="op">)</span>, 
        border <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">FALSE</span>,las<span class="op">=</span><span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>,  cex.axis <span class="op">=</span> <span class="fl">2</span>, line <span class="op">=</span> <span class="op">-</span><span class="fl">1.8</span> <span class="op">)</span></code></pre></div>
<p><img src="covid_19_files/figure-html/barplot2-1.png" width="1344"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; HHY HHW HHR HHK HHM YHH RHH YHW HHS HRH RHW KHH YHK MHH --- YHR WHH MHW RHY RYH SHH HWH HYW MHK RHK HRR HMH KHW WHW YHM </span>
<span class="co">#&gt; 841 481 447 119  96  60  55  42  36  33  33  30  30  29  27  23  22  21  20  20  20  18  16  15  15  14  13  13  13  13 </span>
<span class="co">#&gt; YHS HRY HYH HSH MHR HMW HWY HYY MHY SHR WHR HKY HMY RHR SHK WHM KHY KSH MHM RMH RRW RRY WSH KYH RMW RRH SHM SHW SHY YHY </span>
<span class="co">#&gt;  13  12  11  10  10   9   9   9   9   9   9   7   7   7   7   7   6   6   6   6   6   6   6   5   5   5   5   5   5   5 </span>
<span class="co">#&gt; HKH HMR HSW HYR MRW RHM RYM WHK HKW HWK HWW HYM KHK KHM KHR KHS MMR MYH RHS RMR RRK RWW RYW RYY SMH WHY WSM HMK HMM HRK </span>
<span class="co">#&gt;   4   4   4   4   4   4   4   4   3   3   3   3   3   3   3   3   3   3   3   3   3   3   3   3   3   3   3   2   2   2 </span>
<span class="co">#&gt; HSY HWM KMH KSM KWH MHS MMH MMM MMW MRH MRK RMY RSY RWK RWR RYK SYH WHS WKK WYK WYW YSH HMS HRM HRW HSK HWR HYK HYS KKK </span>
<span class="co">#&gt;   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   1   1   1   1   1   1   1   1 </span>
<span class="co">#&gt; KMW KSW KWW KWY KYK KYW KYY MKH MSM MWH MWM MYK MYM MYR MYW RKH RKW RMK RRS RSH RSW RWY RYR SHS SKW SMK SMS SMW SRH SRM </span>
<span class="co">#&gt;   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1 </span>
<span class="co">#&gt; SWK SWM SWS SWW SYW WKH WKM WMH WMK WMW WMY WRH WSW WSY WWH WWY WYM WYR WYS YKH YKR YMH YMM YMS YMW YRH YWH YWM YWS YWY </span>
<span class="co">#&gt;   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1 </span>
<span class="co">#&gt; YYH </span>
<span class="co">#&gt;   1</span></code></pre></div>
</div>
</div>
<div id="sars-coronavirus-gz02-vs-bat-sl-covzc45-and-bat-sars-like-coronavirus-isolate-rs7327" class="section level1">
<h1 class="hasAnchor">
<a href="#sars-coronavirus-gz02-vs-bat-sl-covzc45-and-bat-sars-like-coronavirus-isolate-rs7327" class="anchor"></a>SARS coronavirus GZ02 vs bat-SL-CoVZC45 and Bat SARS-like coronavirus isolate Rs7327</h1>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">covid_autm</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span>
<span class="va">covid_autm</span>
<span class="co">#&gt; Automorphism object with 9722 ranges and 6 metadata columns:</span>
<span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2    coord1    coord2      autm        cube</span>
<span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt;</span>
<span class="co">#&gt;      [1]        1         1      + |         ATG         ATG        50        50         1        ACGT</span>
<span class="co">#&gt;      [2]        1         2      + |         GAG         GAG        10        10         1        ACGT</span>
<span class="co">#&gt;      [3]        1         3      + |         AGC         AGC        33        33         1        ACGT</span>
<span class="co">#&gt;      [4]        1         4      + |         CTT         CTT        55        55         1        ACGT</span>
<span class="co">#&gt;      [5]        1         5      + |         GTT         GTT        59        59         1        ACGT</span>
<span class="co">#&gt;      ...      ...       ...    ... .         ...         ...       ...       ...       ...         ...</span>
<span class="co">#&gt;   [9718]        1      9718      + |         TCA         TCA        28        28         1        ACGT</span>
<span class="co">#&gt;   [9719]        1      9719      + |         ACT         ACT        19        19         1        ACGT</span>
<span class="co">#&gt;   [9720]        1      9720      + |         CAG         CAG         6         6         1        ACGT</span>
<span class="co">#&gt;   [9721]        1      9721      + |         GCA         GCA        24        24         1        ACGT</span>
<span class="co">#&gt;   [9722]        1      9722      + |         TAA         TAA        12        12         1        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">conserv</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/conserved_regions.html">conserved_regions</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">)</span>
<span class="va">conserv</span>
<span class="co">#&gt; AutomorphismByCoef object with 6348 ranges and 5 metadata columns:</span>
<span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2      autm    mut_type        cube</span>
<span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;      [1]        1      1-19      + |         ATG         ATG         1         HHH        ACGT</span>
<span class="co">#&gt;      [2]        1      1-19      + |         GAG         GAG         1         HHH        ACGT</span>
<span class="co">#&gt;      [3]        1      1-19      + |         AGC         AGC         1         HHH        ACGT</span>
<span class="co">#&gt;      [4]        1      1-19      + |         CTT         CTT         1         HHH        ACGT</span>
<span class="co">#&gt;      [5]        1      1-19      + |         GTT         GTT         1         HHH        ACGT</span>
<span class="co">#&gt;      ...      ...       ...    ... .         ...         ...       ...         ...         ...</span>
<span class="co">#&gt;   [6344]        1 9683-9722      + |         GGA         GGA         1         HHH        ACGT</span>
<span class="co">#&gt;   [6345]        1 9683-9722      + |         TCT         TCT         1         HHH        ACGT</span>
<span class="co">#&gt;   [6346]        1 9683-9722      + |         TCA         TCA         1         HHH        ACGT</span>
<span class="co">#&gt;   [6347]        1 9683-9722      + |         GCA         GCA         1         HHH        ACGT</span>
<span class="co">#&gt;   [6348]        1 9683-9722      + |         TAA         TAA         1         HHH        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">consvr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">conserv</span>, <span class="va">conserv2</span><span class="op">)</span>
<span class="va">widths</span> <span class="op">&lt;-</span> <span class="fu">width</span><span class="op">(</span><span class="va">consvr</span><span class="op">)</span>
<span class="va">dist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/fitCDF.html">fitCDF</a></span><span class="op">(</span><span class="va">widths</span>, distNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">7</span>, <span class="fl">10</span>, <span class="fl">11</span>, <span class="fl">19</span>, <span class="fl">20</span><span class="op">)</span>, plot <span class="op">=</span> <span class="cn">TRUE</span>, 
               loss.fun <span class="op">=</span> <span class="st">"cauchy"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Log-normal distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Half-Normal distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Gamma distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Generalized 3P Gamma distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Weibull distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Exponential distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting 2P Exponential distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt;  * Estimating Studentized residuals for Generalized 3P Gamma distribution</span>
<span class="co">#&gt;  * Plots for Generalized 3P Gamma distribution...</span></code></pre></div>
<p><img src="covid_19_files/figure-html/consv-1.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dist</span>
<span class="co">#&gt; ggamma CDF model</span>
<span class="co">#&gt; ------</span>
<span class="co">#&gt; Parameters:</span>
<span class="co">#&gt;          Estimate  Std. Error   t value   Pr(&gt;|t|)    </span>
<span class="co">#&gt; alpha 0.381621931 0.002040595 187.01507 &lt; 2.22e-16 ***</span>
<span class="co">#&gt; scale 0.492128654 0.022620932  21.75545 &lt; 2.22e-16 ***</span>
<span class="co">#&gt; psi   3.452723567 0.035559333  97.09753 &lt; 2.22e-16 ***</span>
<span class="co">#&gt; ---</span>
<span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residual standard error: 6.55254e-05 on 10158 degrees of freedom</span>
<span class="co">#&gt; Number of iterations to termination: 41 </span>
<span class="co">#&gt; Reason for termination: Relative error in the sum of squares is at most `ftol'. </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Goodness of fit:</span>
<span class="co">#&gt;     Adj.R.Square       rho R.Cross.val      AIC</span>
<span class="co">#&gt; gof    0.9999999 0.9999999   0.9995317 -67874.9</span></code></pre></div>
<p>Joining the datasets increases the sample size and improved predictions</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>lwd <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">widths</span>, <span class="fl">14</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">1</span>, family <span class="op">=</span> <span class="st">"serif"</span>,
family <span class="op">=</span> <span class="st">"serif"</span>, col <span class="op">=</span> <span class="st">"cyan1"</span>, cex.main <span class="op">=</span> <span class="fl">0.9</span>,
main <span class="op">=</span> <span class="st">"Histogram and best fitted CDF model for conserved region sizes"</span>,
xlab <span class="op">=</span> <span class="st">"Conserved region size"</span><span class="op">)</span>
<span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">150</span>, by <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span><span class="op">(</span><span class="va">x1</span>, <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/ggamma.html">dggamma</a></span><span class="op">(</span><span class="va">x1</span>,
                  alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">dist</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,
                  scale <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">dist</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,
                  psi <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">dist</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>
                  <span class="op">)</span>,
      col <span class="op">=</span> <span class="st">"red"</span>, lwd <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="covid_19_files/figure-html/hist2-1.png" width="576"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/usefr/man/mcgoftest.html">mcgoftest</a></span><span class="op">(</span> 
          varobj <span class="op">=</span> <span class="va">widths</span>, 
          model <span class="op">=</span> <span class="va">dist</span>,
          stat <span class="op">=</span> <span class="st">"ks"</span><span class="op">)</span>
<span class="co">#&gt; *** Permutation GoF testing based on Kolmogorov-Smirnov statistic ( parametric approach )  ...</span>
<span class="co">#&gt;       KS.stat.D      mc_p.value KS.stat.p.value     sample.size       num.sampl </span>
<span class="co">#&gt;    8.707602e-02    1.000000e+00    0.000000e+00    1.016100e+04    9.990000e+02</span>

<span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\n===========================\n"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ===========================</span>
<span class="fu"><a href="https://rdrr.io/pkg/usefr/man/mcgoftest.html">mcgoftest</a></span><span class="op">(</span>
          varobj <span class="op">=</span> <span class="va">widths</span>,
          model <span class="op">=</span> <span class="va">dist</span>,
          stat <span class="op">=</span> <span class="st">"rmse"</span>,
          breaks <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>
<span class="co">#&gt; *** Permutation GoF testing based on Root Mean Square statistic ( parametric approach )  ...</span>
<span class="co">#&gt;        rmse  mc_p.value sample.size   num.sampl </span>
<span class="co">#&gt;    29.19551     1.00000 10161.00000   999.00000</span></code></pre></div>
<div id="grouping-automorphism-by-automorphisms-coefficients" class="section level2">
<h2 class="hasAnchor">
<a href="#grouping-automorphism-by-automorphisms-coefficients" class="anchor"></a>Grouping automorphism by automorphism’s coefficients</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">autby_coef</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphismByCoef.html">automorphismByCoef</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">)</span>
<span class="va">autby_coef</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">autby_coef</span>, <span class="va">autby_coef2</span><span class="op">)</span>
<span class="va">autby_coef</span> <span class="op">&lt;-</span> <span class="va">autby_coef</span><span class="op">[</span> <span class="va">autby_coef</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">autby_coef</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="op">-</span><span class="fl">1</span>  <span class="op">]</span></code></pre></div>
<p>Alignment gaps denoting indel mutations are labeled as “—”.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">autby_coef</span><span class="op">$</span><span class="va">mut_type</span><span class="op">)</span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="va">counts</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">count.</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span> <span class="va">counts</span> <span class="op">&gt;</span> <span class="fl">9</span> <span class="op">]</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, cex.axis <span class="op">=</span> <span class="fl">2</span>, font <span class="op">=</span> <span class="fl">2</span>, las <span class="op">=</span> <span class="fl">1</span>, 
    cex.main <span class="op">=</span> <span class="fl">1.4</span>, cex.lab <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html">barplot</a></span><span class="op">(</span><span class="va">count.</span>, main<span class="op">=</span><span class="st">"Automorphism distribution per Mutation type"</span>,
        col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"yellow"</span>, <span class="st">"blue"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">36</span><span class="op">)</span>, 
        border <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">TRUE</span>,las<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="covid_19_files/figure-html/barplot_3-1.png" width="1344"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  HHY  HHR  HHW  HHK  HHM  YHH  RHH  HRH  HHS  YHW  ---  MHH  KHH  RHW  WHH  YHK  HYH  YHR  RHY  MHW  RYH  SHH  HWH  HYW </span>
<span class="co">#&gt; 1266  636  570  151  121   94   79   51   49   47   43   41   36   35   33   33   29   26   24   23   22   22   21   21 </span>
<span class="co">#&gt;  HMH  RHK  HRR  MHK  WHW  YHM  KHW  HRY  YHS  HMW  MHR  HSH  HWY  HYY  MHY  RHR  SHR  WHR  YHY  HMY  KHY  RMH  RRY  HKH </span>
<span class="co">#&gt;   20   19   17   17   17   17   16   15   14   11   11   10   10   10    9    9    9    9    9    8    8    8    8    7 </span>
<span class="co">#&gt;  HKY  HMR  KSH  RMW  SHK  SHW  WHM  MHM  RHM  RRW  RYY  SHM  SHY  WSH  HYR  KYH  RRH  RYW  HSW  HWW  KHK  KHR  MRH  MRW </span>
<span class="co">#&gt;    7    7    7    7    7    7    7    6    6    6    6    6    6    6    5    5    5    5    4    4    4    4    4    4 </span>
<span class="co">#&gt;  RRK  RYM  WHK  WHY  HKW  HMK  HWK  HYK  HYM  KHM  KHS  KMH  KSW  MHS  MMM  MMR  MYH  RHS  RKW  RMR  RMY  RSW  RWW  RYK </span>
<span class="co">#&gt;    4    4    4    4    3    3    3    3    3    3    3    3    3    3    3    3    3    3    3    3    3    3    3    3 </span>
<span class="co">#&gt;  SMH  SRH  SYH  WMH  WSM  WSY  WYW  YWH  HMM  HRK  HSY  HWM  KMW  KSM  KWH  MMH  MMW  MRK  MWH  MWM  RSY  RWK  RWR  SMW </span>
<span class="co">#&gt;    3    3    3    3    3    3    3    3    2    2    2    2    2    2    2    2    2    2    2    2    2    2    2    2 </span>
<span class="co">#&gt;  SRM  WHS  WKK  WMS  WRH  WWH  WWY  WYK  WYM  YKH  YSH  YWM  YWY  YYH  HKK  HKR  HMS  HRM  HRW  HSK  HSR  HWR  HWS  HYS </span>
<span class="co">#&gt;    2    2    2    2    2    2    2    2    2    2    2    2    2    2    1    1    1    1    1    1    1    1    1    1 </span>
<span class="co">#&gt;  KKK  KKS  KRK  KRY  KWW  KWY  KYK  KYW  KYY  MKH  MKM  MRM  MSM  MWS  MWW  MYK  MYM  MYR  MYW  MYY  RKH  RMK  RRM  RRS </span>
<span class="co">#&gt;    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 </span>
<span class="co">#&gt;  RSH  RSK  RWY  RYR  SHS  SKW  SKY  SMK  SMS  SWK  SWM  SWR  SWS  SWW  SWY  SYW  WKH  WKM  WMK  WMW  WMY  WRY  WSW  WWK </span>
<span class="co">#&gt;    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 </span>
<span class="co">#&gt;  WWS  WYH  WYR  WYS  YKR  YKY  YMH  YMM  YMS  YMW  YRH  YRY  YWS  YWW  YYR </span>
<span class="co">#&gt;    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://genomaths.com">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
