<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Automorphism analysis on COVID-19 data • GenomAutomorphism</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Automorphism analysis on COVID-19 data">
<!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GenomAutomorphism</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.8.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../articles/GenomAutomorphism.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Package functions</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/GenomAutomorphism.html">GenomAutomorphism</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/automorphism_and_decision_tree_II.html">A Short Introduction to Algebraic Taxonomy on Genes Regions II</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/Modular_Matrix_Operation_of_Mutational_Events.html">Modular Matrix Operations of Mutational Events</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Automorphism analysis on COVID-19 data</h1>
                        <h4 data-toc-skip class="author">Robersy
Sanchez</h4>
            <address class="author_afil">
      Department of Biology. Pennsylvania State University, University
Park, PA 16802<br><a class="author_email" href="mailto:#"></a><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">28 December 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/vignettes/covid_19.Rmd" class="external-link"><code>vignettes/covid_19.Rmd</code></a></small>
      <div class="d-none name"><code>covid_19.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The analysis of COVID-19 mutational events in terms of
      automorphisms is accomplished here. The analysis is accomplished
      in a pairwise sequence alignments of SARS coronaviruses SARS
      coronavirus GZ02 (GenBank: AY390556.1: 265-13398_13398-21485) and
      Bat SARS-like coronavirus isolate bat-SL-CoVZC45 (GenBank:
      MG772933.1:265-1345513455-21542), nonstructural polyprotein. The
      analysis indicate that the distribution of the conserved sites
      according to their sizes can be modeled by generalized gamma
      distribution.</p>
    </div>
    
<style type="text/css">

div.main-container {
  max-width: 2200px !important;
  padding: 2px;
}

.main-container {
  max-width: 2200px !important;
  padding: 2px;
}

.watch-out {
  max-width: 2200px !important;
  margin-left: auto;
  margin-right: auto;
  background-color: #c9feed;
  font-weight: bold
  font-size: 100%;
  padding: 2px;
}

pre, code {
  max-width: 2200px !important;
  margin-left: auto;
  margin-right: auto;
  background-color: #c9feed;
  font-weight: bold
  font-size: 100px;
  padding: 2px;
}

code {
    font-size: 100%;
    font-family: Consolas,Monaco,"Courier New",monospace;
    border-radius: 0;
}

body {
  max-width: 2200px !important;
  margin: auto;
}  

code.r{
  font-size: 100%;
}
</style>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">Biostrings</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://genomaths.github.io/genomautomorphism">GenomAutomorphism</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/usefr" class="external-link">usefr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://www.stefanom.io/latex2exp/" class="external-link">latex2exp</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="sars-coronavirus-gz02-vs-bat-sl-covzc45">SARS coronavirus GZ02 vs bat-SL-CoVZC45<a class="anchor" aria-label="anchor" href="#sars-coronavirus-gz02-vs-bat-sl-covzc45"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">URL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"https://github.com/genomaths/seqalignments/raw/master/"</span>, </span>
<span>        <span class="st">"COVID-19/AY390556.1_265-13398_13398-21485_RNA-POL_SARS_COVI_GZ02.fas"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">covid_aln2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html" class="external-link">readDNAMultipleAlignment</a></span><span class="op">(</span>filepath <span class="op">=</span> <span class="va">URL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in .Call2("fasta_index", filexp_list, nrec, skip, seek.first.rec, : reading FASTA file</span></span>
<span><span class="co">#&gt; https://github.com/genomaths/seqalignments/raw/master/COVID-19/AY390556.1_265-13398_13398-21485_RNA-POL_SARS_COVI_GZ02.fas:</span></span>
<span><span class="co">#&gt; ignored 2 invalid one-letter sequence codes</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">covid_aut</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphisms.html">automorphisms</a></span><span class="op">(</span></span>
<span>                    seq <span class="op">=</span> <span class="va">covid_aln2</span>,</span>
<span>                    group <span class="op">=</span> <span class="st">"Z64"</span>,</span>
<span>                    cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"TGCA"</span><span class="op">)</span>,</span>
<span>                    cube_alt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CATG"</span>, <span class="st">"GTAC"</span><span class="op">)</span>,</span>
<span>                    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">covid_aut</span></span>
<span><span class="co">#&gt; Automorphism object with 7100 ranges and 8 metadata columns:</span></span>
<span><span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2         aa1         aa2    coord1    coord2      autm</span></span>
<span><span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;      [1]        1         1      + |         ATG         ATG           M           M        50        50         1</span></span>
<span><span class="co">#&gt;      [2]        1         2      + |         GAG         GAG           E           E        10        10         1</span></span>
<span><span class="co">#&gt;      [3]        1         3      + |         AGC         AGC           S           S        33        33         1</span></span>
<span><span class="co">#&gt;      [4]        1         4      + |         CTT         CTT           L           L        55        55         1</span></span>
<span><span class="co">#&gt;      [5]        1         5      + |         GTT         GTC           V           V        59        57        27</span></span>
<span><span class="co">#&gt;      ...      ...       ...    ... .         ...         ...         ...         ...       ...       ...       ...</span></span>
<span><span class="co">#&gt;   [7096]        1      7096      + |         CTT         CTT           L           L        55        55         1</span></span>
<span><span class="co">#&gt;   [7097]        1      7097      + |         GTT         GTT           V           V        59        59         1</span></span>
<span><span class="co">#&gt;   [7098]        1      7098      + |         AAT         AAC           N           N         3         1        43</span></span>
<span><span class="co">#&gt;   [7099]        1      7099      + |         AAC         AAC           N           N         1         1         1</span></span>
<span><span class="co">#&gt;   [7100]        1      7100      + |         TAA         TAA           *           *        12        12         1</span></span>
<span><span class="co">#&gt;                 cube</span></span>
<span><span class="co">#&gt;          &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1]        ACGT</span></span>
<span><span class="co">#&gt;      [2]        ACGT</span></span>
<span><span class="co">#&gt;      [3]        ACGT</span></span>
<span><span class="co">#&gt;      [4]        ACGT</span></span>
<span><span class="co">#&gt;      [5]        ACGT</span></span>
<span><span class="co">#&gt;      ...         ...</span></span>
<span><span class="co">#&gt;   [7096]        ACGT</span></span>
<span><span class="co">#&gt;   [7097]        ACGT</span></span>
<span><span class="co">#&gt;   [7098]        ACGT</span></span>
<span><span class="co">#&gt;   [7099]        ACGT</span></span>
<span><span class="co">#&gt;   [7100]        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">covid_aut</span><span class="op">$</span><span class="va">cube</span><span class="op">[</span> <span class="va">covid_aut</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="fl">1</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">covid_aut</span><span class="op">$</span><span class="va">autm</span><span class="op">)</span> <span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, cex <span class="op">=</span> <span class="fl">0.9</span>, font <span class="op">=</span> <span class="fl">2</span>, mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">6</span>,<span class="fl">4</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="va">counts</span>, main<span class="op">=</span><span class="st">"Automorphism distribution"</span>,</span>
<span>        xlab<span class="op">=</span><span class="st">"Genetic-code cube representation"</span>,</span>
<span>        ylab<span class="op">=</span><span class="st">"Fixed mutational events"</span>,</span>
<span>        col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"darkblue"</span>,<span class="st">"red"</span>, <span class="st">"darkgreen"</span><span class="op">)</span>, </span>
<span>        border <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co">#ylim = c(0, 2000), </span></span>
<span>        cex.lab <span class="op">=</span> <span class="fl">2</span>, cex.main <span class="op">=</span> <span class="fl">1.5</span>, cex.names <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></span><span class="op">(</span><span class="fl">2</span>, at <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">500</span>, <span class="fl">1000</span>, <span class="fl">1500</span>, <span class="fl">2000</span><span class="op">)</span>, cex.axis <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html" class="external-link">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>,line <span class="op">=</span> <span class="op">-</span><span class="fl">1.5</span>, at <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.7</span>, <span class="fl">1.9</span>, <span class="fl">3.1</span>, <span class="fl">4.3</span>, <span class="fl">5.5</span><span class="op">)</span>,</span>
<span>      text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span> <span class="va">counts</span> <span class="op">)</span>, cex <span class="op">=</span> <span class="fl">1.4</span>,</span>
<span>      col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"white"</span>,<span class="st">"yellow"</span>, <span class="st">"black"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="covid_19_files/figure-html/barplot-1.png" width="576"></p>
</div>
<div class="section level2">
<h2 id="analysis-of-conserved-regions">Analysis of conserved regions<a class="anchor" aria-label="anchor" href="#analysis-of-conserved-regions"></a>
</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">conserv2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/conserved_regions.html">conserved_regions</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">covid_aut</span><span class="op">)</span></span>
<span><span class="va">conserv2</span></span>
<span><span class="co">#&gt; AutomorphismByCoef object with 3813 ranges and 7 metadata columns:</span></span>
<span><span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2         aa1         aa2      autm    mut_type        cube</span></span>
<span><span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1]        1       1-4      + |         ATG         ATG           M           M         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      [2]        1       1-4      + |         GAG         GAG           E           E         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      [3]        1       1-4      + |         AGC         AGC           S           S         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      [4]        1       1-4      + |         CTT         CTT           L           L         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      [5]        1         7      + |         GGT         GGT           G           G         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      ...      ...       ...    ... .         ...         ...         ...         ...       ...         ...         ...</span></span>
<span><span class="co">#&gt;   [3809]        1 7093-7094      + |         GAT         GAT           D           D         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   [3810]        1 7096-7097      + |         CTT         CTT           L           L         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   [3811]        1 7096-7097      + |         GTT         GTT           V           V         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   [3812]        1 7099-7100      + |         AAC         AAC           N           N         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   [3813]        1 7099-7100      + |         TAA         TAA           *           *         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>Next, function <a href="https://rdrr.io/github/genomaths/usefr/man/fitCDF.html" class="external-link">fitCDF</a>
is applied to find the best fitted probability function to the the sizes
of conserved regions.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">widths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">conserv2</span><span class="op">)</span></span>
<span><span class="va">dist2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/fitCDF.html" class="external-link">fitCDF</a></span><span class="op">(</span><span class="va">widths</span>, distNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">7</span>, <span class="fl">9</span>, <span class="fl">10</span>, <span class="fl">11</span>, <span class="fl">12</span>, <span class="fl">19</span>, <span class="fl">20</span><span class="op">)</span>, plot <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>               loss.fun <span class="op">=</span> <span class="st">"cauchy"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting Half-Normal distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting Gamma distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting Generalized 4P Gamma distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting Generalized 3P Gamma distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting Weibull distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting 3P Weibull distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting Exponential distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting 2P Exponential distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt;  * Estimating Studentized residuals for Generalized 4P Gamma distribution</span></span>
<span><span class="co">#&gt;  * Plots for Generalized 4P Gamma distribution...</span></span></code></pre></div>
<p><img src="covid_19_files/figure-html/dist-1.png" width="1152"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dist2</span></span>
<span><span class="co">#&gt; ggamma CDF model</span></span>
<span><span class="co">#&gt; ------</span></span>
<span><span class="co">#&gt; Parameters:</span></span>
<span><span class="co">#&gt;          Estimate  Std. Error   t value   Pr(&gt;|t|)    </span></span>
<span><span class="co">#&gt; alpha  0.38596938  0.01174701  32.85681 &lt; 2.22e-16 ***</span></span>
<span><span class="co">#&gt; scale  0.02195697  0.00742671   2.95649  0.0031308 ** </span></span>
<span><span class="co">#&gt; mu    -0.39511486  0.01856885 -21.27837 &lt; 2.22e-16 ***</span></span>
<span><span class="co">#&gt; psi    7.52471426  0.52104576  14.44156 &lt; 2.22e-16 ***</span></span>
<span><span class="co">#&gt; ---</span></span>
<span><span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Residual standard error: 1.809493e-05 on 3809 degrees of freedom</span></span>
<span><span class="co">#&gt; Number of iterations to termination: 222 </span></span>
<span><span class="co">#&gt; Reason for termination: Relative error in the sum of squares is at most `ftol'. </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Goodness of fit:</span></span>
<span><span class="co">#&gt;     Adj.R.Square rho R.Cross.val       AIC</span></span>
<span><span class="co">#&gt; gof            1   1   0.9998617 -30582.48</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>lwd <span class="op">=</span> <span class="fl">0.5</span>, cex.axis <span class="op">=</span> <span class="fl">2</span>, cex.lab <span class="op">=</span><span class="fl">1.4</span>,</span>
<span>    cex.main <span class="op">=</span> <span class="fl">2</span>, mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>,<span class="fl">6</span>,<span class="fl">4</span>,<span class="fl">4</span><span class="op">)</span>, family <span class="op">=</span> <span class="st">"serif"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">widths</span>, <span class="fl">20</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">1</span>, family <span class="op">=</span> <span class="st">"serif"</span>,</span>
<span>col <span class="op">=</span> <span class="st">"cyan1"</span>, cex.main <span class="op">=</span> <span class="fl">0.9</span>,</span>
<span>main <span class="op">=</span> <span class="st">"Histogram and best fitted CDF model for conserved region sizes"</span>,</span>
<span>xlab <span class="op">=</span> <span class="st">"Conserved region size (bp)"</span>, yaxt <span class="op">=</span> <span class="st">"n"</span>, ylab<span class="op">=</span><span class="st">""</span>, cex.axis <span class="op">=</span> <span class="fl">1.4</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>, cex.axis <span class="op">=</span> <span class="fl">1.4</span>, las <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html" class="external-link">mtext</a></span><span class="op">(</span><span class="st">"Density"</span>, side <span class="op">=</span> <span class="fl">2</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, line <span class="op">=</span> <span class="fl">3.5</span><span class="op">)</span></span>
<span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">35</span>, by <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">txt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://www.stefanom.io/latex2exp/reference/TeX.html" class="external-link">TeX</a></span><span class="op">(</span><span class="st">r'($\textit{f}(\textit{x}) = \frac{\alpha}{\beta\Gamma(\gamma)}</span></span>
<span><span class="st">  {(\frac{\textit{x - \mu}}{\beta})}^{\alpha\delta-1}</span></span>
<span><span class="st">  e^{(-\frac{\textit{x - \mu}}{\beta})^\alpha}$)'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">x1</span>, <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/ggamma.html" class="external-link">dggamma</a></span><span class="op">(</span><span class="va">x1</span>,</span>
<span>                  alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">dist2</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>                  scale <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">dist2</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,</span>
<span>                  mu <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">dist2</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>,</span>
<span>                  psi <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">dist2</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span></span>
<span>                  <span class="op">)</span>,</span>
<span>      col <span class="op">=</span> <span class="st">"red"</span>, lwd <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html" class="external-link">mtext</a></span><span class="op">(</span><span class="va">txt</span>, side <span class="op">=</span> <span class="fl">3</span>, line <span class="op">=</span> <span class="op">-</span><span class="fl">4</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, adj <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span></span></code></pre></div>
<p><img src="covid_19_files/figure-html/hist-1.png" width="672"></p>
<p>After apply Bootstrap test for Goodness of fit, tt seems to be that
we have no reasons to reject the hypothesis that the sizes of conserved
regions follows Generalized Gamma distribution.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/usefr/man/mcgoftest.html" class="external-link">mcgoftest</a></span><span class="op">(</span> </span>
<span>          varobj <span class="op">=</span> <span class="va">widths</span>, </span>
<span>          model <span class="op">=</span> <span class="va">dist2</span>,</span>
<span>          stat <span class="op">=</span> <span class="st">"ks"</span><span class="op">)</span></span>
<span><span class="co">#&gt; *** Permutation GoF testing based on Kolmogorov-Smirnov statistic ( parametric approach )  ...</span></span>
<span><span class="co">#&gt;       KS.stat.D      mc_p.value KS.stat.p.value     sample.size       num.sampl </span></span>
<span><span class="co">#&gt;    1.733499e-01    1.000000e+00   5.983316e-100    3.813000e+03    9.990000e+02</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/usefr/man/mcgoftest.html" class="external-link">mcgoftest</a></span><span class="op">(</span></span>
<span>          varobj <span class="op">=</span> <span class="va">widths</span>,</span>
<span>          model <span class="op">=</span> <span class="va">dist2</span>,</span>
<span>          stat <span class="op">=</span> <span class="st">"ad"</span><span class="op">)</span></span>
<span><span class="co">#&gt; *** Permutation GoF testing based on Anderson–Darling statistic ( parametric approach )  ...</span></span>
<span><span class="co">#&gt;     AD.stat  mc_p.value sample.size   num.sampl </span></span>
<span><span class="co">#&gt;    119.2847      1.0000   3813.0000    999.0000</span></span></code></pre></div>
<div class="section level3">
<h3 id="grouping-automorphism-by-automorphisms-coefficients--types-of-mutations">Grouping automorphism by automorphism’s coefficients. Types of
mutations<a class="anchor" aria-label="anchor" href="#grouping-automorphism-by-automorphisms-coefficients--types-of-mutations"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">autby_coef2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphism_bycoef.html">automorphism_bycoef</a></span><span class="op">(</span><span class="va">covid_aut</span><span class="op">)</span></span>
<span><span class="va">autby_coef2</span> <span class="op">&lt;-</span> <span class="va">autby_coef2</span><span class="op">[</span> <span class="va">autby_coef2</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">autby_coef2</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="op">-</span><span class="fl">1</span>  <span class="op">]</span></span></code></pre></div>
<p>Alignment gaps denoting indel mutations are labeled as “—”.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">autby_coef2</span><span class="op">$</span><span class="va">mut_type</span><span class="op">)</span></span>
<span><span class="va">counts2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">counts2</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">count.</span> <span class="op">&lt;-</span> <span class="va">counts2</span><span class="op">[</span> <span class="va">counts2</span> <span class="op">&gt;</span> <span class="fl">9</span> <span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, cex.axis <span class="op">=</span> <span class="fl">2</span>, font <span class="op">=</span> <span class="fl">2</span>, las <span class="op">=</span> <span class="fl">1</span>, </span>
<span>    cex.main <span class="op">=</span> <span class="fl">1.4</span>, mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">3</span>,<span class="fl">4</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="va">count.</span>, main<span class="op">=</span><span class="st">"Automorphism distribution per Mutation type"</span>,</span>
<span>        col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"yellow"</span>, <span class="st">"blue"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">36</span><span class="op">)</span>, </span>
<span>        border <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">FALSE</span>,las<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>,  cex.axis <span class="op">=</span> <span class="fl">2</span>, line <span class="op">=</span> <span class="op">-</span><span class="fl">1.8</span> <span class="op">)</span></span></code></pre></div>
<p><img src="covid_19_files/figure-html/barplot2-1.png" width="1344"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; HHY HHW HHR HHK HHM YHH RHH YHW HHS HRH RHW KHH YHK MHH --- YHR WHH MHW RHY RYH SHH HWH HYW MHK RHK HRR HMH KHW WHW YHM </span></span>
<span><span class="co">#&gt; 841 481 447 119  96  60  55  42  36  33  33  30  30  29  27  23  22  21  20  20  20  18  16  15  15  14  13  13  13  13 </span></span>
<span><span class="co">#&gt; YHS HRY HYH HSH MHR HMW HWY HYY MHY SHR WHR HKY HMY RHR SHK WHM KHY KSH MHM RMH RRW RRY WSH KYH RMW RRH SHM SHW SHY YHY </span></span>
<span><span class="co">#&gt;  13  12  11  10  10   9   9   9   9   9   9   7   7   7   7   7   6   6   6   6   6   6   6   5   5   5   5   5   5   5 </span></span>
<span><span class="co">#&gt; HKH HMR HSW HYR MRW RHM RYM WHK HKW HWK HWW HYM KHK KHM KHR KHS MMR MYH RHS RMR RRK RWW RYW RYY SMH WHY WSM HMK HMM HRK </span></span>
<span><span class="co">#&gt;   4   4   4   4   4   4   4   4   3   3   3   3   3   3   3   3   3   3   3   3   3   3   3   3   3   3   3   2   2   2 </span></span>
<span><span class="co">#&gt; HSY HWM KMH KSM KWH MHS MMH MMM MMW MRH MRK RMY RSY RWK RWR RYK SYH WHS WKK WYK WYW YSH HMS HRM HRW HSK HWR HYK HYS KKK </span></span>
<span><span class="co">#&gt;   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   1   1   1   1   1   1   1   1 </span></span>
<span><span class="co">#&gt; KMW KSW KWW KWY KYK KYW KYY MKH MSM MWH MWM MYK MYM MYR MYW RKH RKW RMK RRS RSH RSW RWY RYR SHS SKW SMK SMS SMW SRH SRM </span></span>
<span><span class="co">#&gt;   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1 </span></span>
<span><span class="co">#&gt; SWK SWM SWS SWW SYW WKH WKM WMH WMK WMW WMY WRH WSW WSY WWH WWY WYM WYR WYS YKH YKR YMH YMM YMS YMW YRH YWH YWM YWS YWY </span></span>
<span><span class="co">#&gt;   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1 </span></span>
<span><span class="co">#&gt; YYH </span></span>
<span><span class="co">#&gt;   1</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="sars-coronavirus-gz02-vs-bat-sl-covzc45-and-bat-sars-like-coronavirus-isolate-rs7327">SARS coronavirus GZ02 vs bat-SL-CoVZC45 and Bat SARS-like
coronavirus isolate Rs7327<a class="anchor" aria-label="anchor" href="#sars-coronavirus-gz02-vs-bat-sl-covzc45-and-bat-sars-like-coronavirus-isolate-rs7327"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">covid_autm</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span></span>
<span><span class="va">covid_autm</span></span>
<span><span class="co">#&gt; Automorphism object with 9722 ranges and 8 metadata columns:</span></span>
<span><span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2         aa1         aa2    coord1    coord2      autm</span></span>
<span><span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;      [1]        1         1      + |         ATG         ATG           M           M        50        50         1</span></span>
<span><span class="co">#&gt;      [2]        1         2      + |         GAG         GAG           E           E        10        10         1</span></span>
<span><span class="co">#&gt;      [3]        1         3      + |         AGC         AGC           S           S        33        33         1</span></span>
<span><span class="co">#&gt;      [4]        1         4      + |         CTT         CTT           L           L        55        55         1</span></span>
<span><span class="co">#&gt;      [5]        1         5      + |         GTT         GTT           V           V        59        59         1</span></span>
<span><span class="co">#&gt;      ...      ...       ...    ... .         ...         ...         ...         ...       ...       ...       ...</span></span>
<span><span class="co">#&gt;   [9718]        1      9718      + |         TCA         TCA           S           S        28        28         1</span></span>
<span><span class="co">#&gt;   [9719]        1      9719      + |         ACT         ACT           T           T        19        19         1</span></span>
<span><span class="co">#&gt;   [9720]        1      9720      + |         CAG         CAG           Q           Q         6         6         1</span></span>
<span><span class="co">#&gt;   [9721]        1      9721      + |         GCA         GCA           A           A        24        24         1</span></span>
<span><span class="co">#&gt;   [9722]        1      9722      + |         TAA         TAA           *           *        12        12         1</span></span>
<span><span class="co">#&gt;                 cube</span></span>
<span><span class="co">#&gt;          &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1]        ACGT</span></span>
<span><span class="co">#&gt;      [2]        ACGT</span></span>
<span><span class="co">#&gt;      [3]        ACGT</span></span>
<span><span class="co">#&gt;      [4]        ACGT</span></span>
<span><span class="co">#&gt;      [5]        ACGT</span></span>
<span><span class="co">#&gt;      ...         ...</span></span>
<span><span class="co">#&gt;   [9718]        ACGT</span></span>
<span><span class="co">#&gt;   [9719]        ACGT</span></span>
<span><span class="co">#&gt;   [9720]        ACGT</span></span>
<span><span class="co">#&gt;   [9721]        ACGT</span></span>
<span><span class="co">#&gt;   [9722]        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">conserv</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/conserved_regions.html">conserved_regions</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">)</span></span>
<span><span class="va">conserv</span></span>
<span><span class="co">#&gt; AutomorphismByCoef object with 6348 ranges and 7 metadata columns:</span></span>
<span><span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2         aa1         aa2      autm    mut_type        cube</span></span>
<span><span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1]        1      1-19      + |         ATG         ATG           M           M         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      [2]        1      1-19      + |         GAG         GAG           E           E         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      [3]        1      1-19      + |         AGC         AGC           S           S         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      [4]        1      1-19      + |         CTT         CTT           L           L         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      [5]        1      1-19      + |         GTT         GTT           V           V         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      ...      ...       ...    ... .         ...         ...         ...         ...       ...         ...         ...</span></span>
<span><span class="co">#&gt;   [6344]        1 9683-9722      + |         GGA         GGA           G           G         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   [6345]        1 9683-9722      + |         TCT         TCT           S           S         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   [6346]        1 9683-9722      + |         TCA         TCA           S           S         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   [6347]        1 9683-9722      + |         GCA         GCA           A           A         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   [6348]        1 9683-9722      + |         TAA         TAA           *           *         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">consvr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">conserv</span>, <span class="va">conserv2</span><span class="op">)</span></span>
<span><span class="va">widths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">consvr</span><span class="op">)</span></span>
<span><span class="va">dist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/fitCDF.html" class="external-link">fitCDF</a></span><span class="op">(</span><span class="va">widths</span>, distNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">7</span>, <span class="fl">9</span>, <span class="fl">10</span>, <span class="fl">11</span>, <span class="fl">19</span>, <span class="fl">20</span><span class="op">)</span>, plot <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>               loss.fun <span class="op">=</span> <span class="st">"cauchy"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting Log-normal distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting Half-Normal distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting Gamma distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting Generalized 4P Gamma distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting Generalized 3P Gamma distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting Weibull distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting Exponential distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; *** Fitting 2P Exponential distribution ...</span></span>
<span><span class="co">#&gt; .Fitting Done.</span></span>
<span><span class="co">#&gt;  * Estimating Studentized residuals for Generalized 4P Gamma distribution</span></span>
<span><span class="co">#&gt;  * Plots for Generalized 4P Gamma distribution...</span></span></code></pre></div>
<p><img src="covid_19_files/figure-html/consv-1.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dist</span></span>
<span><span class="co">#&gt; ggamma CDF model</span></span>
<span><span class="co">#&gt; ------</span></span>
<span><span class="co">#&gt; Parameters:</span></span>
<span><span class="co">#&gt;          Estimate  Std. Error   t value   Pr(&gt;|t|)    </span></span>
<span><span class="co">#&gt; alpha 0.591610381 0.003539813 167.13041 &lt; 2.22e-16 ***</span></span>
<span><span class="co">#&gt; scale 8.652891132 0.218974837  39.51546 &lt; 2.22e-16 ***</span></span>
<span><span class="co">#&gt; mu    0.549870141 0.005491076 100.13888 &lt; 2.22e-16 ***</span></span>
<span><span class="co">#&gt; psi   1.352270767 0.015700233  86.13062 &lt; 2.22e-16 ***</span></span>
<span><span class="co">#&gt; ---</span></span>
<span><span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Residual standard error: 5.502277e-05 on 10157 degrees of freedom</span></span>
<span><span class="co">#&gt; Number of iterations to termination: 24 </span></span>
<span><span class="co">#&gt; Reason for termination: Relative error in the sum of squares is at most `ftol'. </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Goodness of fit:</span></span>
<span><span class="co">#&gt;     Adj.R.Square rho R.Cross.val       AIC</span></span>
<span><span class="co">#&gt; gof            1   1   0.9996438 -69784.96</span></span></code></pre></div>
<p>Joining the datasets increases the sample size and improved
predictions in respect to the comparison SARS coronavirus GZ02 vs Bat
SARS-like coronavirus alone.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>lwd <span class="op">=</span> <span class="fl">0.5</span>, cex.axis <span class="op">=</span> <span class="fl">2</span>, cex.lab <span class="op">=</span><span class="fl">1.4</span>,</span>
<span>    cex.main <span class="op">=</span> <span class="fl">2</span>, mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>,<span class="fl">6</span>,<span class="fl">4</span>,<span class="fl">4</span><span class="op">)</span>, family <span class="op">=</span> <span class="st">"serif"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">widths</span>, <span class="fl">14</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">1</span>, family <span class="op">=</span> <span class="st">"serif"</span>,</span>
<span>col <span class="op">=</span> <span class="st">"cyan1"</span>, cex.main <span class="op">=</span> <span class="fl">0.9</span>,</span>
<span>main <span class="op">=</span> <span class="st">"Histogram and best fitted CDF model for conserved region sizes"</span>,</span>
<span>xlab <span class="op">=</span> <span class="st">"Conserved region size (bp)"</span>, yaxt <span class="op">=</span> <span class="st">"n"</span>, ylab<span class="op">=</span><span class="st">""</span>, cex.axis <span class="op">=</span> <span class="fl">1.4</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>, cex.axis <span class="op">=</span> <span class="fl">1.4</span>, las <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html" class="external-link">mtext</a></span><span class="op">(</span><span class="st">"Density"</span>, side <span class="op">=</span> <span class="fl">2</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, line <span class="op">=</span> <span class="fl">3.5</span><span class="op">)</span></span>
<span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">150</span>, by <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">txt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://www.stefanom.io/latex2exp/reference/TeX.html" class="external-link">TeX</a></span><span class="op">(</span><span class="st">r'($\textit{f}(\textit{x}) = \frac{\alpha}{\beta\Gamma(\gamma)}</span></span>
<span><span class="st">  {(\frac{\textit{x - \mu}}{\beta})}^{\alpha\delta-1}</span></span>
<span><span class="st">  e^{(-\frac{\textit{x - \mu}}{\beta})^\alpha}$)'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">x1</span>, <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/ggamma.html" class="external-link">dggamma</a></span><span class="op">(</span><span class="va">x1</span>,</span>
<span>                  alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">dist</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>                  scale <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">dist</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,</span>
<span>                  psi <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html" class="external-link">coef</a></span><span class="op">(</span><span class="va">dist</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span></span>
<span>                  <span class="op">)</span>,</span>
<span>      col <span class="op">=</span> <span class="st">"red"</span>, lwd <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html" class="external-link">mtext</a></span><span class="op">(</span><span class="va">txt</span>, side <span class="op">=</span> <span class="fl">3</span>, line <span class="op">=</span> <span class="op">-</span><span class="fl">4</span>, cex <span class="op">=</span> <span class="fl">1.4</span>, adj <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span></span></code></pre></div>
<p><img src="covid_19_files/figure-html/hist2-1.png" width="576"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/usefr/man/mcgoftest.html" class="external-link">mcgoftest</a></span><span class="op">(</span> </span>
<span>          varobj <span class="op">=</span> <span class="va">widths</span>, </span>
<span>          model <span class="op">=</span> <span class="va">dist</span>,</span>
<span>          stat <span class="op">=</span> <span class="st">"ks"</span><span class="op">)</span></span>
<span><span class="co">#&gt; *** Permutation GoF testing based on Kolmogorov-Smirnov statistic ( parametric approach )  ...</span></span>
<span><span class="co">#&gt;       KS.stat.D      mc_p.value KS.stat.p.value     sample.size       num.sampl </span></span>
<span><span class="co">#&gt;    8.628456e-02    1.000000e+00    3.919415e-66    1.016100e+04    9.990000e+02</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\n===========================\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ===========================</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/usefr/man/mcgoftest.html" class="external-link">mcgoftest</a></span><span class="op">(</span></span>
<span>          varobj <span class="op">=</span> <span class="va">widths</span>,</span>
<span>          model <span class="op">=</span> <span class="va">dist</span>,</span>
<span>          stat <span class="op">=</span> <span class="st">"ad"</span><span class="op">)</span></span>
<span><span class="co">#&gt; *** Permutation GoF testing based on Anderson–Darling statistic ( parametric approach )  ...</span></span>
<span><span class="co">#&gt;     AD.stat  mc_p.value sample.size   num.sampl </span></span>
<span><span class="co">#&gt;    70.64955     1.00000 10161.00000   999.00000</span></span></code></pre></div>
<div class="section level3">
<h3 id="grouping-automorphism-by-automorphisms-coefficients">Grouping automorphism by automorphism’s coefficients<a class="anchor" aria-label="anchor" href="#grouping-automorphism-by-automorphisms-coefficients"></a>
</h3>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">autby_coef</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphism_bycoef.html">automorphism_bycoef</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">)</span></span>
<span><span class="va">autby_coef</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">autby_coef</span>, <span class="va">autby_coef2</span><span class="op">)</span></span>
<span><span class="va">autby_coef</span> <span class="op">&lt;-</span> <span class="va">autby_coef</span><span class="op">[</span> <span class="va">autby_coef</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">autby_coef</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="op">-</span><span class="fl">1</span>  <span class="op">]</span></span></code></pre></div>
<p>Alignment gaps denoting indel mutations are labeled as “—”.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">autby_coef</span><span class="op">$</span><span class="va">mut_type</span><span class="op">)</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">counts</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">count.</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span> <span class="va">counts</span> <span class="op">&gt;</span> <span class="fl">9</span> <span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, cex.axis <span class="op">=</span> <span class="fl">2</span>, font <span class="op">=</span> <span class="fl">2</span>, las <span class="op">=</span> <span class="fl">1</span>, </span>
<span>    cex.main <span class="op">=</span> <span class="fl">1.4</span>, cex.lab <span class="op">=</span> <span class="fl">2</span>, mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">4</span>,<span class="fl">4</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="va">count.</span>, main<span class="op">=</span><span class="st">"Automorphism distribution per Mutation type"</span>,</span>
<span>        col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"yellow"</span>, <span class="st">"blue"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">36</span><span class="op">)</span>, </span>
<span>        border <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">FALSE</span>,las<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>,  cex.axis <span class="op">=</span> <span class="fl">2</span>, line <span class="op">=</span> <span class="op">-</span><span class="fl">1.8</span> <span class="op">)</span></span></code></pre></div>
<p><img src="covid_19_files/figure-html/barplot_3-1.png" width="1440"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  HHY  HHR  HHW  HHK  HHM  YHH  RHH  HRH  HHS  YHW  ---  MHH  KHH  RHW  WHH  YHK  HYH  YHR  RHY  MHW  RYH  SHH  HWH  HYW </span></span>
<span><span class="co">#&gt; 1266  636  570  151  121   94   79   51   49   47   43   41   36   35   33   33   29   26   24   23   22   22   21   21 </span></span>
<span><span class="co">#&gt;  HMH  RHK  HRR  MHK  WHW  YHM  KHW  HRY  YHS  HMW  MHR  HSH  HWY  HYY  MHY  RHR  SHR  WHR  YHY  HMY  KHY  RMH  RRY  HKH </span></span>
<span><span class="co">#&gt;   20   19   17   17   17   17   16   15   14   11   11   10   10   10    9    9    9    9    9    8    8    8    8    7 </span></span>
<span><span class="co">#&gt;  HKY  HMR  KSH  RMW  SHK  SHW  WHM  MHM  RHM  RRW  RYY  SHM  SHY  WSH  HYR  KYH  RRH  RYW  HSW  HWW  KHK  KHR  MRH  MRW </span></span>
<span><span class="co">#&gt;    7    7    7    7    7    7    7    6    6    6    6    6    6    6    5    5    5    5    4    4    4    4    4    4 </span></span>
<span><span class="co">#&gt;  RRK  RYM  WHK  WHY  HKW  HMK  HWK  HYK  HYM  KHM  KHS  KMH  KSW  MHS  MMM  MMR  MYH  RHS  RKW  RMR  RMY  RSW  RWW  RYK </span></span>
<span><span class="co">#&gt;    4    4    4    4    3    3    3    3    3    3    3    3    3    3    3    3    3    3    3    3    3    3    3    3 </span></span>
<span><span class="co">#&gt;  SMH  SRH  SYH  WMH  WSM  WSY  WYW  YWH  HMM  HRK  HSY  HWM  KMW  KSM  KWH  MMH  MMW  MRK  MWH  MWM  RSY  RWK  RWR  SMW </span></span>
<span><span class="co">#&gt;    3    3    3    3    3    3    3    3    2    2    2    2    2    2    2    2    2    2    2    2    2    2    2    2 </span></span>
<span><span class="co">#&gt;  SRM  WHS  WKK  WMS  WRH  WWH  WWY  WYK  WYM  YKH  YSH  YWM  YWY  YYH  HKK  HKR  HMS  HRM  HRW  HSK  HSR  HWR  HWS  HYS </span></span>
<span><span class="co">#&gt;    2    2    2    2    2    2    2    2    2    2    2    2    2    2    1    1    1    1    1    1    1    1    1    1 </span></span>
<span><span class="co">#&gt;  KKK  KKS  KRK  KRY  KWW  KWY  KYK  KYW  KYY  MKH  MKM  MRM  MSM  MWS  MWW  MYK  MYM  MYR  MYW  MYY  RKH  RMK  RRM  RRS </span></span>
<span><span class="co">#&gt;    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 </span></span>
<span><span class="co">#&gt;  RSH  RSK  RWY  RYR  SHS  SKW  SKY  SMK  SMS  SWK  SWM  SWR  SWS  SWW  SWY  SYW  WKH  WKM  WMK  WMW  WMY  WRY  WSW  WWK </span></span>
<span><span class="co">#&gt;    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1 </span></span>
<span><span class="co">#&gt;  WWS  WYH  WYR  WYS  YKR  YKY  YMH  YMM  YMS  YMW  YRH  YRY  YWS  YWW  YYR </span></span>
<span><span class="co">#&gt;    1    1    1    1    1    1    1    1    1    1    1    1    1    1    1</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
