<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Get started-with GenomAutomorphism • GenomAutomorphism</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Get started-with GenomAutomorphism">
<meta property="og:description" content="GenomAutomorphism">
<meta property="og:image" content="https://genomaths.github.io/GenomAutomorphism/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomAutomorphism</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/GenomAutomorphism.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/GenomAutomorphism.html">GenomAutomorphism</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/GenomAutomorphism/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="GenomAutomorphism_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Get started-with GenomAutomorphism</h1>
                        <h4 class="author">Robersy Sanchez</h4>
            <address class="author_afil">
      Department of Biology. Pennsylvania State University, University Park, PA 16802<br><a class="author_email" href="mailto:#"></a><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a>
      </address>
                  
            <h4 class="date">28 February 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/master/vignettes/GenomAutomorphism.Rmd"><code>vignettes/GenomAutomorphism.Rmd</code></a></small>
      <div class="hidden name"><code>GenomAutomorphism.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>A fast introduction into the analysis of DNA mutational events by means of automorphisms between two DNA sequences algebraically represented as Abelian finite group.</p>
    </div>
    
<style type="text/css">

.watch-out {
  max-width: 1820px !important;
  margin-left: auto;
  margin-right: auto;
  background-color: #c9feed;
  font-weight: bold
  font-size: 100%;
  padding: 2px;
}

pre, code {
  max-width: 1820px !important;
  margin-left: auto;
  margin-right: auto;
  background-color: #c9feed;
  font-weight: bold
  font-size: 100px;
  padding: 2px;
}

code {
    font-size: 100%;
    font-family: Consolas,Monaco,"Courier New",monospace;
    border-radius: 0;
}

code.r{
  font-size: 100%;
}
</style>
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>This is a R package to compute the autimorphisms between pairwise aligned DNA sequences represented as elements from a Genomic Abelian group as described in reference (<a href="#1">1</a>). In a general scenario, whole chromosomes or genomic regions from a population (from any species or close related species) can be algebraically represented as a direct sum of cyclic groups or more specifically Abelian <em>p</em>-groups. Basically, we propose the representation of multiple sequence alignments (MSA) of length <em>N</em> as a finite Abelian group created by the direct sum of Abelian group of <em>prime-power order</em>:</p>
<p><span class="math display">\[
\qquad G = (\mathbb{Z}_{p^{\alpha_{1}}_1})^{n_1} \oplus (\mathbb{Z}_{p^{\alpha_{2}}_1})^{n_2} \oplus \dots \oplus (\mathbb{Z}_{p^{\alpha_{k}}_k})^{n_k} 
\]</span></p>
<p>Where, the <span class="math inline">\(p_i\)</span>’s are prime numbers, <span class="math inline">\(\alpha_i \in \mathbb{N}\)</span> and <span class="math inline">\(\mathbb{Z}_{p^{\alpha_{i}}_i}\)</span> is the group of integer modulo <span class="math inline">\(p^{\alpha_{i}}_i\)</span>.</p>
<p>For the purpose of estimating the automorphism between two aligned DNA sequences, <span class="math inline">\(p^{\alpha_{i}}_i \in \{5, 2^6, 5^3 \}\)</span>.</p>
<div id="automorphisms" class="section level2">
<h2 class="hasAnchor">
<a href="#automorphisms" class="anchor"></a>Automorphisms</h2>
<p>Herein, automorphisms are considered algebraic descriptions of mutational event observed in codon sequences represented on different Abelian groups. In particular, as described in references (<a href="#3">3-4</a>), for each representation of the codon set on a defined Abelian group there are 24 possible isomorphic Abelian groups. These Abelian groups can be labeled based on the DNA base-order used to generate them. The set of 24 Abelian groups can be described as a group isomorphic to the symmetric group of degree four (<span class="math inline">\(S_4\)</span>, see reference (<a href="#4">4</a>)).</p>
<p>For further support about the symmetric group on the 24 Abelian group of genetic-code cubes, users can also see <a href="https://github.com/genomaths/GenomeAlgebra_SymmetricGroup">Symmetric Group of the Genetic-Code Cubes.</a>, specifically the Mathematica notebook <em>IntroductionToZ5GeneticCodeVectorSpace.nb</em> and interact with it using Wolfram Player, freely available (for Windows and Linux OS) at, <a href="https://www.wolfram.com/player/" class="uri">https://www.wolfram.com/player/</a>.</p>
</div>
<div id="load-the-r-libraries" class="section level2">
<h2 class="hasAnchor">
<a href="#load-the-r-libraries" class="anchor"></a>Load the R libraries</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings">Biostrings</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames, dirname, do.call, duplicated, eval,</span>
<span class="co">#&gt;     evalq, Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget, order,</span>
<span class="co">#&gt;     paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span>
<span class="co">#&gt;     table, tapply, union, unique, unsplit, which.max, which.min</span>
<span class="co">#&gt; Loading required package: S4Vectors</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid, I, unname</span>
<span class="co">#&gt; Loading required package: IRanges</span>
<span class="co">#&gt; Loading required package: XVector</span>
<span class="co">#&gt; Loading required package: GenomeInfoDb</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'Biostrings'</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     strsplit</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/GenomAutomorphism">GenomAutomorphism</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">usefr</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'usefr'</span>
<span class="co">#&gt; The following object is masked from 'package:BiocGenerics':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     unlist</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     unlist</span></code></pre></div>
</div>
</div>
<div id="read-the-alignment-fasta-and-encode-the-sequences" class="section level1">
<h1 class="hasAnchor">
<a href="#read-the-alignment-fasta-and-encode-the-sequences" class="anchor"></a>Read the alignment <em>FASTA</em> and encode the sequences</h1>
<p>The FASTA file carries the pairwise sequence alignment of protein coding regions SARS coronavirus GZ02 (GenBank: AY390556.1) and Bat SARS-like coronavirus isolate Rs7327 (GenBank: KY417151.1).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">URL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"https://github.com/genomaths/seqalignments/raw/master/"</span>, 
        <span class="st">"COVID-19/AY390556.1_and_KY417151.1_aligned_protein-coding.fas"</span><span class="op">)</span>

<span class="va">covid_aln</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/MultipleAlignment-class.html">readDNAMultipleAlignment</a></span><span class="op">(</span>filepath <span class="op">=</span> <span class="va">URL</span><span class="op">)</span>
<span class="va">covid_aln</span>
<span class="co">#&gt; DNAMultipleAlignment with 2 rows and 29166 columns</span>
<span class="co">#&gt;      aln                                                                                            names               </span>
<span class="co">#&gt; [1] ATGGAGAGCCTTGTTCTTGGTGTCAACGAGAAAACACACGTCCAAC...CAAAATTCCATGAGTGGAGCTTCTGCTGATTCAACTCAGGCATAA lcl|AY390556.1_cd...</span>
<span class="co">#&gt; [2] ATGGAGAGCCTTGTTCTTGGTGTCAACGAGAAAACACACGTCCAAC...CAAAATTCCATGAGTGGAGCTTCTGCTGATTCAACTCAGGCATAA lcl|KY417151.1_cd...</span></code></pre></div>
<p>The DNA sequence is read and the corresponding codon sequences are represented in the Abelian group <span class="math inline">\(\mathbb{Z}_{64}\)</span> (i.e., the set of integers remainder modulo 64). The codon coordinates are requested on the cube ACGT. Following reference (<a href="#4">4</a>)), cubes are labeled based on the order of DNA bases used to define the sum operation.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">covid_aln</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span>
<span class="va">codons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/codon_coord.html">codon_coord</a></span><span class="op">(</span>codon <span class="op">=</span> <span class="va">covid_aln</span>, 
                      cube <span class="op">=</span> <span class="st">"ACGT"</span>, 
                      group <span class="op">=</span> <span class="st">"Z64"</span>, 
                      chr <span class="op">=</span> <span class="fl">1L</span>,
                      strand <span class="op">=</span> <span class="st">"+"</span>,
                      start <span class="op">=</span> <span class="fl">1</span>,
                      end <span class="op">=</span> <span class="fl">750</span><span class="op">)</span>
<span class="va">codons</span>
<span class="co">#&gt; CodonGroup object with 250 ranges and 4 metadata columns:</span>
<span class="co">#&gt;         seqnames    ranges strand |        seq1        seq2      coord1      coord2</span>
<span class="co">#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;     [1]        1         1      + |         ATG         ATG          50          50</span>
<span class="co">#&gt;     [2]        1         2      + |         GAG         GAG          10          10</span>
<span class="co">#&gt;     [3]        1         3      + |         AGC         AGC          33          33</span>
<span class="co">#&gt;     [4]        1         4      + |         CTT         CTT          55          55</span>
<span class="co">#&gt;     [5]        1         5      + |         GTT         GTT          59          59</span>
<span class="co">#&gt;     ...      ...       ...    ... .         ...         ...         ...         ...</span>
<span class="co">#&gt;   [246]        1       246      + |         GAT         GAT          11          11</span>
<span class="co">#&gt;   [247]        1       247      + |         AAG         AAG           2           2</span>
<span class="co">#&gt;   [248]        1       248      + |         AGC         AGC          33          33</span>
<span class="co">#&gt;   [249]        1       249      + |         TAC         TAT          13          15</span>
<span class="co">#&gt;   [250]        1       250      + |         GAG         GAG          10          10</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
<p>The codon sequences (<em>seq1</em> and <em>seq2</em>) with their corresponding coordinates (left) are returned, as well as the coordinated representation on <span class="math inline">\(\mathbb{Z}_{64}\)</span> (<em>coord1</em> and <em>coord2</em>).</p>
</div>
<div id="automorphisms-on-mathbbz_64" class="section level1">
<h1 class="hasAnchor">
<a href="#automorphisms-on-mathbbz_64" class="anchor"></a>Automorphisms on <span class="math inline">\(\mathbb{Z}_{64}\)</span>
</h1>
<p>Automorphisms can be computed starting directly from the <em>FASTA</em> file. Notice that we can work only with genomic regions of our interest by giving the <em>start</em> and <em>end</em> alignment coordinates. In Z64 automorphisms are described as functions <span class="math inline">\(f(x) = k\,x \quad mod\,64\)</span>, where <span class="math inline">\(k\)</span> and <span class="math inline">\(x\)</span> are elements from the set of integers modulo 64. Below, in function <em>automorphism</em> three important arguments are given values: <em>group = “Z64”</em>, <em>cube = c(“ACGT”, “TGCA”)</em>, and <em>cube_alt = c(“CATG”, “GTAC”)</em>. Setting for group specifies on which group the automorphisms will be computed. These groups can be: “Z5”, “Z64”, “Z125”, and “Z5^3”.</p>
<p>In groups “Z64” and “Z125” not all the mutational events can be described as automorphisms from a given cube. So, a character string denoting pairs of “dual” the genetic-code cubes, as given in references (<a href="#2">1-4</a>)), is given as argument for <em>cube</em>. That is, the base pairs from the given cubes must be complementary each other. Such a cube pair are call dual cubes and, as shown in reference (<a href="#4">4</a>)), each pair integrates group. If automorphisms are not found in first set of dual cubes, then the algorithm search for automorphism in a alternative set of dual cubes.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">autm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphism.html">automorphism</a></span><span class="op">(</span>
                      seqs <span class="op">=</span> <span class="va">covid_aln</span>,
                      group <span class="op">=</span> <span class="st">"Z64"</span>,
                      cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"TGCA"</span><span class="op">)</span>,
                      cube_alt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CATG"</span>, <span class="st">"GTAC"</span><span class="op">)</span>,
                      start <span class="op">=</span> <span class="fl">1</span>,
                      end <span class="op">=</span> <span class="fl">750</span>, 
                      verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">autm</span>
<span class="co">#&gt; Automorphism object with 250 ranges and 6 metadata columns:</span>
<span class="co">#&gt;         seqnames    ranges strand |        seq1        seq2    coord1    coord2      autm        cube</span>
<span class="co">#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt;</span>
<span class="co">#&gt;     [1]        1         1      + |         ATG         ATG        50        50         1        ACGT</span>
<span class="co">#&gt;     [2]        1         2      + |         GAG         GAG        10        10         1        ACGT</span>
<span class="co">#&gt;     [3]        1         3      + |         AGC         AGC        33        33         1        ACGT</span>
<span class="co">#&gt;     [4]        1         4      + |         CTT         CTT        55        55         1        ACGT</span>
<span class="co">#&gt;     [5]        1         5      + |         GTT         GTT        59        59         1        ACGT</span>
<span class="co">#&gt;     ...      ...       ...    ... .         ...         ...       ...       ...       ...         ...</span>
<span class="co">#&gt;   [246]        1       246      + |         GAT         GAT        11        11         1        ACGT</span>
<span class="co">#&gt;   [247]        1       247      + |         AAG         AAG         2         2         1        ACGT</span>
<span class="co">#&gt;   [248]        1       248      + |         AGC         AGC        33        33         1        ACGT</span>
<span class="co">#&gt;   [249]        1       249      + |         TAC         TAT        13        15        11        ACGT</span>
<span class="co">#&gt;   [250]        1       250      + |         GAG         GAG        10        10         1        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Observe that two new columns were added, the automorphism coefficient <span class="math inline">\(k\)</span> (named as <em>autm</em>) and the genetic-code cube where the automorphism was found. By convention the DNA sequence is given for the positive strand. Since the <em>dual cube</em> of <strong>“ACGT”</strong> corresponds to the complementary base order <strong>TGCA</strong>, automorphisms described by the cube <strong>TGCA</strong> represent mutational events affecting the DNA negative strand (-).</p>
<p>The last result can be summarized by gene regions as follow:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aut_range</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphismByRanges.html">automorphismByRanges</a></span><span class="op">(</span><span class="va">autm</span><span class="op">)</span>
<span class="va">aut_range</span>
<span class="co">#&gt; GRanges object with 9 ranges and 1 metadata column:</span>
<span class="co">#&gt;       seqnames    ranges strand |        cube</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;   [1]        1     1-102      + |        ACGT</span>
<span class="co">#&gt;   [2]        1       103      - |        TGCA</span>
<span class="co">#&gt;   [3]        1   104-105      + |        ACGT</span>
<span class="co">#&gt;   [4]        1       106      - |        TGCA</span>
<span class="co">#&gt;   [5]        1   107-201      + |        ACGT</span>
<span class="co">#&gt;   [6]        1       202      - |        TGCA</span>
<span class="co">#&gt;   [7]        1   203-205      + |        ACGT</span>
<span class="co">#&gt;   [8]        1       206      - |        TGCA</span>
<span class="co">#&gt;   [9]        1   207-250      + |        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
<p>That is, function <em>automorphismByRanges</em> permits the classification of the pairwise alignment of protein-coding sub-regions based on the mutational events observed on it quantitatively represented as automorphisms on genetic-code cubes.</p>
<p>Searching for automorphisms on <span class="math inline">\(\mathbb{Z}_{64}\)</span> permits us a quantitative differentiation between mutational events at different codon positions from a given DNA protein-encoding region. As shown in reference (<a href="#4">4</a>)) a set of different cubes can be applied to describe the best evolutionary aminoacid scale highly correlated with aminoacid physicochemical properties describing the observed evolutionary process in a given protein.</p>
<p>More information about this subject can be found in the supporting material from reference (<a href="#4">4</a>)) at GitHub <a href="https://github.com/genomaths/GenomeAlgebra_SymmetricGroup">GenomeAlgebra_SymmetricGroup</a>, particularly by interacting with the Mathematica notebook <a href="https://github.com/genomaths/GenomeAlgebra_SymmetricGroup/blob/master/Genetic-Code-Scales_of_Amino-Acids.nb">Genetic-Code-Scales_of_Amino-Acids.nb</a>.</p>
</div>
<div id="automorphisms-between-whole-genomes-of-sars-cov-2-related-coronaviruses" class="section level1">
<h1 class="hasAnchor">
<a href="#automorphisms-between-whole-genomes-of-sars-cov-2-related-coronaviruses" class="anchor"></a>Automorphisms between whole genomes of SARS-CoV-2 related coronaviruses</h1>
<p>Next, the automorphism for the whole pairwise alignment of SARS-CoV-2 related coronaviruses:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Do not need to run it. </span>
<span class="va">covid_autm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphism.html">automorphism</a></span><span class="op">(</span>
                    seq <span class="op">=</span> <span class="va">covid_aln</span>,
                    group <span class="op">=</span> <span class="st">"Z64"</span>,
                    cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"TGCA"</span><span class="op">)</span>,
                    cube_alt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CATG"</span>, <span class="st">"GTAC"</span><span class="op">)</span>,
                    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="co">## This data is available with the package</span>
<span class="co"># data(covid_autm, package = "GenomAutomorphism")</span>
<span class="va">covid_autm</span></code></pre></div>
<p>And the summary by range</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aut_range</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphismByRanges.html">automorphismByRanges</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">)</span>
<span class="va">aut_range</span>
<span class="co">#&gt; GRanges object with 460 ranges and 1 metadata column:</span>
<span class="co">#&gt;         seqnames    ranges strand |        cube</span>
<span class="co">#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;     [1]        1     1-102      + |        ACGT</span>
<span class="co">#&gt;     [2]        1       103      - |        TGCA</span>
<span class="co">#&gt;     [3]        1   104-105      + |        ACGT</span>
<span class="co">#&gt;     [4]        1       106      - |        TGCA</span>
<span class="co">#&gt;     [5]        1   107-201      + |        ACGT</span>
<span class="co">#&gt;     ...      ...       ...    ... .         ...</span>
<span class="co">#&gt;   [456]        1 9577-9608      + |        ACGT</span>
<span class="co">#&gt;   [457]        1      9609      - |        TGCA</span>
<span class="co">#&gt;   [458]        1 9610-9674      + |        ACGT</span>
<span class="co">#&gt;   [459]        1      9675      - |        TGCA</span>
<span class="co">#&gt;   [460]        1 9676-9722      + |        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Regions no described by automorphism can be described as translations (labeled “Trnl”) and they can be shown as follow:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">idx</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">$</span><span class="va">cube</span> <span class="op">==</span> <span class="st">"Trnl"</span><span class="op">)</span>
<span class="va">covid_autm</span><span class="op">[</span> <span class="va">idx</span> <span class="op">]</span>
<span class="co">#&gt; Automorphism object with 19 ranges and 6 metadata columns:</span>
<span class="co">#&gt;        seqnames    ranges strand |        seq1        seq2    coord1    coord2      autm        cube</span>
<span class="co">#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt;</span>
<span class="co">#&gt;    [1]        1       916      + |         AAC         AAA         1         0         0        Trnl</span>
<span class="co">#&gt;    [2]        1       944      + |         ---         CCG        NA        22         0        Trnl</span>
<span class="co">#&gt;    [3]        1       971      + |         ACA         ---        16        NA         0        Trnl</span>
<span class="co">#&gt;    [4]        1      7077      + |         ---         AAA        NA         0         0        Trnl</span>
<span class="co">#&gt;    [5]        1      9109      + |         AAT         AAA         3         0         0        Trnl</span>
<span class="co">#&gt;    ...      ...       ...    ... .         ...         ...       ...       ...       ...         ...</span>
<span class="co">#&gt;   [15]        1      9277      + |         CTT         ---        55        NA         0        Trnl</span>
<span class="co">#&gt;   [16]        1      9278      + |         CAT         ---         7        NA         0        Trnl</span>
<span class="co">#&gt;   [17]        1      9279      + |         GAA         ---         8        NA         0        Trnl</span>
<span class="co">#&gt;   [18]        1      9295      + |         AAA         ---         0        NA         0        Trnl</span>
<span class="co">#&gt;   [19]        1      9296      + |         CGA         ---        36        NA         0        Trnl</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
<p>These codon positions cover insertion-deletion (<em>indel</em>) mutational events. The wholes regions can be summarized typing:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">idx</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">aut_range</span><span class="op">$</span><span class="va">cube</span> <span class="op">==</span> <span class="st">"Trnl"</span><span class="op">)</span>
<span class="va">aut_range</span><span class="op">[</span> <span class="va">idx</span> <span class="op">]</span>
<span class="co">#&gt; GRanges object with 13 ranges and 1 metadata column:</span>
<span class="co">#&gt;        seqnames    ranges strand |        cube</span>
<span class="co">#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;    [1]        1       916      + |        Trnl</span>
<span class="co">#&gt;    [2]        1       944      + |        Trnl</span>
<span class="co">#&gt;    [3]        1       971      + |        Trnl</span>
<span class="co">#&gt;    [4]        1      7077      + |        Trnl</span>
<span class="co">#&gt;    [5]        1      9109      + |        Trnl</span>
<span class="co">#&gt;    ...      ...       ...    ... .         ...</span>
<span class="co">#&gt;    [9]        1 9244-9245      + |        Trnl</span>
<span class="co">#&gt;   [10]        1      9247      + |        Trnl</span>
<span class="co">#&gt;   [11]        1      9257      + |        Trnl</span>
<span class="co">#&gt;   [12]        1 9277-9279      + |        Trnl</span>
<span class="co">#&gt;   [13]        1 9295-9296      + |        Trnl</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Only one indel mutation was found in the region where the spike glycoprotein is located: 7076 - 8331.</p>
<p>That is, the pairwise alignment of SARS coronavirus GZ02 and Bat SARS-like coronavirus (bat-SL-CoVZC45) reveals four single <em>indel</em> mutational events and four regions with multiple <em>indel</em> mutations. In general, <em>indel</em> mutational event can be modeled as translations on <span class="math inline">\(\mathbb{Z}_{64}\)</span>.</p>
<div id="bar-plot-automorphism-distribution-by-cubes" class="section level2">
<h2 class="hasAnchor">
<a href="#bar-plot-automorphism-distribution-by-cubes" class="anchor"></a>Bar plot automorphism distribution by cubes</h2>
<p>The automorphism distribution by cubes can be summarized in the bar-plot graphic</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">$</span><span class="va">cube</span><span class="op">[</span> <span class="va">covid_autm</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="fl">1</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">$</span><span class="va">autm</span><span class="op">)</span> <span class="op">]</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, cex <span class="op">=</span> <span class="fl">0.9</span>, font <span class="op">=</span> <span class="fl">2</span>, mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">6</span>,<span class="fl">4</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html">barplot</a></span><span class="op">(</span><span class="va">counts</span>, main<span class="op">=</span><span class="st">"Automorphism distribution"</span>,
        xlab<span class="op">=</span><span class="st">"Genetic-code cube representation"</span>,
        ylab<span class="op">=</span><span class="st">"Fixed mutational events"</span>,
        col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"darkblue"</span>,<span class="st">"red"</span>, <span class="st">"darkgreen"</span><span class="op">)</span>, 
        border <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co">#ylim = c(0, 2000), </span>
        cex.lab <span class="op">=</span> <span class="fl">2</span>, cex.main <span class="op">=</span> <span class="fl">1.5</span>, cex.names <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">2</span>, at <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">200</span>, <span class="fl">400</span>, <span class="fl">600</span>, <span class="fl">800</span><span class="op">)</span>, cex.axis <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>,line <span class="op">=</span> <span class="op">-</span><span class="fl">1.5</span>, at <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.7</span>, <span class="fl">1.9</span>, <span class="fl">3.1</span>, <span class="fl">4.3</span>, <span class="fl">5.5</span><span class="op">)</span>,
      text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span> <span class="va">counts</span> <span class="op">)</span>, cex <span class="op">=</span> <span class="fl">1.4</span>,
      col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"white"</span>,<span class="st">"yellow"</span>, <span class="st">"black"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="GenomAutomorphism_files/figure-html/barplot-1.png" width="576"></p>
</div>
<div id="conserved-and-non-conserved-regions" class="section level2">
<h2 class="hasAnchor">
<a href="#conserved-and-non-conserved-regions" class="anchor"></a>Conserved and non-conserved regions</h2>
<div id="conserved-regions" class="section level3">
<h3 class="hasAnchor">
<a href="#conserved-regions" class="anchor"></a>Conserved regions</h3>
<p>Conserved and non-conserved gene regions can be easily observed in most of MSA editing bioinformatic tools. However, here were interesting into get the regions coordinates for further downstream analysis.</p>
<p>Conserved regions from pairwise comparisons are obtain with function conserved_regions:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">conserv</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/conserved_regions.html">conserved_regions</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">)</span>
<span class="va">conserv</span>
<span class="co">#&gt; AutomorphismByCoef object with 6348 ranges and 5 metadata columns:</span>
<span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2      autm    mut_type        cube</span>
<span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;      [1]        1      1-19      + |         ATG         ATG         1         HHH        ACGT</span>
<span class="co">#&gt;      [2]        1      1-19      + |         GAG         GAG         1         HHH        ACGT</span>
<span class="co">#&gt;      [3]        1      1-19      + |         AGC         AGC         1         HHH        ACGT</span>
<span class="co">#&gt;      [4]        1      1-19      + |         CTT         CTT         1         HHH        ACGT</span>
<span class="co">#&gt;      [5]        1      1-19      + |         GTT         GTT         1         HHH        ACGT</span>
<span class="co">#&gt;      ...      ...       ...    ... .         ...         ...       ...         ...         ...</span>
<span class="co">#&gt;   [6344]        1 9683-9722      + |         GGA         GGA         1         HHH        ACGT</span>
<span class="co">#&gt;   [6345]        1 9683-9722      + |         TCT         TCT         1         HHH        ACGT</span>
<span class="co">#&gt;   [6346]        1 9683-9722      + |         TCA         TCA         1         HHH        ACGT</span>
<span class="co">#&gt;   [6347]        1 9683-9722      + |         GCA         GCA         1         HHH        ACGT</span>
<span class="co">#&gt;   [6348]        1 9683-9722      + |         TAA         TAA         1         HHH        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Several regions are similar for more than one comparison.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">conserv_unique</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/conserved_regions.html">conserved_regions</a></span><span class="op">(</span><span class="va">covid_autm</span>, output <span class="op">=</span> <span class="st">"unique"</span><span class="op">)</span>
<span class="va">conserv_unique</span>
<span class="co">#&gt; GRanges object with 744 ranges and 2 metadata columns:</span>
<span class="co">#&gt;         seqnames    ranges strand |      autm        cube</span>
<span class="co">#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;character&gt;</span>
<span class="co">#&gt;     [1]        1      1-19      + |         1        ACGT</span>
<span class="co">#&gt;     [2]        1     21-44      + |         1        ACGT</span>
<span class="co">#&gt;     [3]        1     47-52      + |         1        ACGT</span>
<span class="co">#&gt;     [4]        1     54-60      + |         1        ACGT</span>
<span class="co">#&gt;     [5]        1     62-80      + |         1        ACGT</span>
<span class="co">#&gt;     ...      ...       ...    ... .       ...         ...</span>
<span class="co">#&gt;   [740]        1 9610-9614      + |         1        ACGT</span>
<span class="co">#&gt;   [741]        1 9616-9648      + |         1        ACGT</span>
<span class="co">#&gt;   [742]        1 9650-9674      + |         1        ACGT</span>
<span class="co">#&gt;   [743]        1 9676-9681      + |         1        ACGT</span>
<span class="co">#&gt;   [744]        1 9683-9722      + |         1        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
</div>
<div id="the-best-fitted-cdf-model-for-the-sizes-of-conserved-regions" class="section level3">
<h3 class="hasAnchor">
<a href="#the-best-fitted-cdf-model-for-the-sizes-of-conserved-regions" class="anchor"></a>The best fitted CDF model for the sizes of conserved regions</h3>
<p>Next, function <a href="https://rdrr.io/github/genomaths/usefr/man/fitCDF.html">fitCDF</a> is applied to find the best fitted probability function to the the sizes of conserved regions.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">widths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/partykit/man/partynode-methods.html">width</a></span><span class="op">(</span><span class="va">conserv</span><span class="op">)</span>
<span class="va">dist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/fitCDF.html">fitCDF</a></span><span class="op">(</span><span class="va">widths</span>, distNames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">7</span>, <span class="fl">10</span>, <span class="fl">11</span>, <span class="fl">19</span>, <span class="fl">20</span><span class="op">)</span>, plot <span class="op">=</span> <span class="cn">TRUE</span>, 
               loss.fun <span class="op">=</span> <span class="st">"cauchy"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Log-normal distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Half-Normal distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Gamma distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Generalized 3P Gamma distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Weibull distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting Exponential distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; *** Fitting 2P Exponential distribution ...</span>
<span class="co">#&gt; .Fitting Done.</span>
<span class="co">#&gt;  * Estimating Studentized residuals for Generalized 3P Gamma distribution</span>
<span class="co">#&gt;  * Plots for Generalized 3P Gamma distribution...</span></code></pre></div>
<p><img src="GenomAutomorphism_files/figure-html/dist-1.png" width="1152"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dist</span>
<span class="co">#&gt; ggamma CDF model</span>
<span class="co">#&gt; ------</span>
<span class="co">#&gt; Parameters:</span>
<span class="co">#&gt;           Estimate   Std. Error   t value   Pr(&gt;|t|)    </span>
<span class="co">#&gt; alpha  0.715996194  0.005430414 131.84928 &lt; 2.22e-16 ***</span>
<span class="co">#&gt; scale 11.575282670  0.315824764  36.65097 &lt; 2.22e-16 ***</span>
<span class="co">#&gt; psi    1.773290037  0.024022540  73.81776 &lt; 2.22e-16 ***</span>
<span class="co">#&gt; ---</span>
<span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Residual standard error: 0.0001224475 on 6345 degrees of freedom</span>
<span class="co">#&gt; Number of iterations to termination: 18 </span>
<span class="co">#&gt; Reason for termination: Relative error in the sum of squares is at most `ftol'. </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Goodness of fit:</span>
<span class="co">#&gt;     Adj.R.Square       rho R.Cross.val       AIC</span>
<span class="co">#&gt; gof    0.9999998 0.9999998   0.9992046 -37955.95</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>lwd <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">widths</span>, <span class="fl">14</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, las <span class="op">=</span> <span class="fl">1</span>, family <span class="op">=</span> <span class="st">"serif"</span>,
family <span class="op">=</span> <span class="st">"serif"</span>, col <span class="op">=</span> <span class="st">"cyan1"</span>, cex.main <span class="op">=</span> <span class="fl">0.9</span>,
main <span class="op">=</span> <span class="st">"Histogram and best fitted CDF model for conserved region sizes"</span>,
xlab <span class="op">=</span> <span class="st">"Conserved region size"</span><span class="op">)</span>
<span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">150</span>, by <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span><span class="op">(</span><span class="va">x1</span>, <span class="fu"><a href="https://rdrr.io/pkg/usefr/man/ggamma.html">dggamma</a></span><span class="op">(</span><span class="va">x1</span>,
                  alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">dist</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,
                  scale <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">dist</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,
                  psi <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/coef.html">coef</a></span><span class="op">(</span><span class="va">dist</span><span class="op">$</span><span class="va">bestfit</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>
                  <span class="op">)</span>,
      col <span class="op">=</span> <span class="st">"red"</span>, lwd <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="GenomAutomorphism_files/figure-html/hist-1.png" width="576"></p>
<p>After apply Bootstrap test for Goodness of fit, tt seems to be that we have no reasons to reject the hypothesis that the sizes of conserved regions follows Generalized Gamma distribution.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/usefr/man/mcgoftest.html">mcgoftest</a></span><span class="op">(</span> 
          varobj <span class="op">=</span> <span class="va">widths</span>, 
          model <span class="op">=</span> <span class="va">dist</span>,
          stat <span class="op">=</span> <span class="st">"ks"</span><span class="op">)</span>
<span class="co">#&gt; *** Permutation GoF testing based on Kolmogorov-Smirnov statistic ( parametric approach )  ...</span>
<span class="co">#&gt;       KS.stat.D      mc_p.value KS.stat.p.value     sample.size       num.sampl </span>
<span class="co">#&gt;    4.775274e-02    1.000000e+00    5.342393e-13    6.348000e+03    9.990000e+02</span>

<span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\n===========================\n"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ===========================</span>
<span class="fu"><a href="https://rdrr.io/pkg/usefr/man/mcgoftest.html">mcgoftest</a></span><span class="op">(</span>
          varobj <span class="op">=</span> <span class="va">widths</span>,
          model <span class="op">=</span> <span class="va">dist</span>,
          stat <span class="op">=</span> <span class="st">"rmse"</span>,
          breaks <span class="op">=</span> <span class="fl">12</span><span class="op">)</span>
<span class="co">#&gt; *** Permutation GoF testing based on Root Mean Square statistic ( parametric approach )  ...</span>
<span class="co">#&gt;        rmse  mc_p.value sample.size   num.sampl </span>
<span class="co">#&gt;    24.98844     1.00000  6348.00000   999.00000</span></code></pre></div>
<p>Nevertheless, the data comes from only one pairwise DNA sequence alignment and the distribution would differ when consider a large DNA multiple sequence alignment.</p>
<p>Short summary for the other CDFs analyzed are shown typing:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dist</span><span class="op">$</span><span class="va">fit</span>
<span class="co">#&gt; $`Generalized 3P Gamma`</span>
<span class="co">#&gt; Nonlinear regression via the Levenberg-Marquardt algorithm</span>
<span class="co">#&gt; parameter estimates: 0.715996194439718, 11.5752826701165, 1.77329003663246 </span>
<span class="co">#&gt; residual sum-of-squares: 9.513e-05</span>
<span class="co">#&gt; reason terminated: Relative error in the sum of squares is at most `ftol'.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Gamma</span>
<span class="co">#&gt; Nonlinear regression via the Levenberg-Marquardt algorithm</span>
<span class="co">#&gt; parameter estimates: 1.01644779433848, 28.1326662667232 </span>
<span class="co">#&gt; residual sum-of-squares: 0.0001594</span>
<span class="co">#&gt; reason terminated: Relative error in the sum of squares is at most `ftol'.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Weibull</span>
<span class="co">#&gt; Nonlinear regression via the Levenberg-Marquardt algorithm</span>
<span class="co">#&gt; parameter estimates: 1.00730503428026, 28.7027570491589 </span>
<span class="co">#&gt; residual sum-of-squares: 0.0001635</span>
<span class="co">#&gt; reason terminated: Relative error in the sum of squares is at most `ftol'.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Exponential</span>
<span class="co">#&gt; Nonlinear regression via the Levenberg-Marquardt algorithm</span>
<span class="co">#&gt; parameter estimates: 0.0347995442981667 </span>
<span class="co">#&gt; residual sum-of-squares: 0.0001674</span>
<span class="co">#&gt; reason terminated: Relative error in the sum of squares is at most `ftol'.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Half-Normal`</span>
<span class="co">#&gt; Nonlinear regression via the Levenberg-Marquardt algorithm</span>
<span class="co">#&gt; parameter estimates: 0.0351040019336635, -2.52184152197332 </span>
<span class="co">#&gt; residual sum-of-squares: 0.004265</span>
<span class="co">#&gt; reason terminated: Relative error in the sum of squares is at most `ftol'.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`2P Exponential`</span>
<span class="co">#&gt; Nonlinear regression via the Levenberg-Marquardt algorithm</span>
<span class="co">#&gt; parameter estimates: 1, 14.7879025598451 </span>
<span class="co">#&gt; residual sum-of-squares: 16.31</span>
<span class="co">#&gt; reason terminated: Relative error in the sum of squares is at most `ftol'.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`Log-normal`</span>
<span class="co">#&gt; Nonlinear regression via the Levenberg-Marquardt algorithm</span>
<span class="co">#&gt; parameter estimates: 1.06355468285382, 0.341608314254582 </span>
<span class="co">#&gt; residual sum-of-squares: 0.006118</span>
<span class="co">#&gt; reason terminated: Relative error in the sum of squares is at most `ftol'.</span></code></pre></div>
</div>
</div>
</div>
<div id="automorphisms-on-mathbbz_125" class="section level1">
<h1 class="hasAnchor">
<a href="#automorphisms-on-mathbbz_125" class="anchor"></a>Automorphisms on <span class="math inline">\(\mathbb{Z}_{125}\)</span>
</h1>
<p>Alternatively, we can use the algebraic representation on on <span class="math inline">\(\mathbb{Z}_{125}\)</span>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">covid_autm_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphism.html">automorphism</a></span><span class="op">(</span>
                      seq <span class="op">=</span> <span class="va">covid_aln</span>, 
                      group <span class="op">=</span> <span class="st">"Z125"</span>, 
                      cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"TGCA"</span><span class="op">)</span>,
                      cube_alt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CATG"</span>, <span class="st">"GTAC"</span><span class="op">)</span>,
                      verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">covid_autm_2</span>
<span class="co">#&gt; Automorphism object with 9722 ranges and 6 metadata columns:</span>
<span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2    coord1    coord2      autm        cube</span>
<span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt;</span>
<span class="co">#&gt;      [1]        1         1      + |         ATG         ATG       108       108         1        ACGT</span>
<span class="co">#&gt;      [2]        1         2      + |         GAG         GAG        43        43         1        ACGT</span>
<span class="co">#&gt;      [3]        1         3      + |         AGC         AGC        82        82         1        ACGT</span>
<span class="co">#&gt;      [4]        1         4      + |         CTT         CTT       114       114         1        ACGT</span>
<span class="co">#&gt;      [5]        1         5      + |         GTT         GTT       119       119         1        ACGT</span>
<span class="co">#&gt;      ...      ...       ...    ... .         ...         ...       ...       ...       ...         ...</span>
<span class="co">#&gt;   [9718]        1      9718      + |         TCA         TCA        71        71         1        ACGT</span>
<span class="co">#&gt;   [9719]        1      9719      + |         ACT         ACT        59        59         1        ACGT</span>
<span class="co">#&gt;   [9720]        1      9720      + |         CAG         CAG        38        38         1        ACGT</span>
<span class="co">#&gt;   [9721]        1      9721      + |         GCA         GCA        66        66         1        ACGT</span>
<span class="co">#&gt;   [9722]        1      9722      + |         TAA         TAA        46        46         1        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
<p>And the summary by range</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aut_range_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphismByRanges.html">automorphismByRanges</a></span><span class="op">(</span><span class="va">covid_autm_2</span><span class="op">)</span>
<span class="va">aut_range_2</span>
<span class="co">#&gt; GRanges object with 21 ranges and 1 metadata column:</span>
<span class="co">#&gt;        seqnames    ranges strand |        cube</span>
<span class="co">#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;    [1]        1     1-943      + |        ACGT</span>
<span class="co">#&gt;    [2]        1       944      + |        Trnl</span>
<span class="co">#&gt;    [3]        1   945-970      + |        ACGT</span>
<span class="co">#&gt;    [4]        1       971      + |        Trnl</span>
<span class="co">#&gt;    [5]        1  972-7076      + |        ACGT</span>
<span class="co">#&gt;    ...      ...       ...    ... .         ...</span>
<span class="co">#&gt;   [17]        1 9258-9276      + |        ACGT</span>
<span class="co">#&gt;   [18]        1 9277-9279      + |        Trnl</span>
<span class="co">#&gt;   [19]        1 9280-9294      + |        ACGT</span>
<span class="co">#&gt;   [20]        1 9295-9296      + |        Trnl</span>
<span class="co">#&gt;   [21]        1 9297-9722      + |        ACGT</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></code></pre></div>
<p>The whole genome can be described by automorphisms on <span class="math inline">\(\mathbb{Z}_{125}\)</span>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">covid_autm_2</span><span class="op">$</span><span class="va">cube</span><span class="op">[</span> <span class="va">covid_autm_2</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="fl">1</span> <span class="op">]</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, cex <span class="op">=</span> <span class="fl">1</span>, font <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html">barplot</a></span><span class="op">(</span><span class="va">counts</span>, main<span class="op">=</span><span class="st">"Automorphism distribution"</span>,
        xlab<span class="op">=</span><span class="st">"Genetic-code cube representation"</span>,
        ylab<span class="op">=</span><span class="st">"Fixed mutational events"</span>,
        col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"darkblue"</span>,<span class="st">"red"</span><span class="op">)</span>, 
        ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1100</span><span class="op">)</span>,
        border <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>,line <span class="op">=</span> <span class="op">-</span><span class="fl">2</span>, at <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.7</span>, <span class="fl">1.9</span>, <span class="fl">3.1</span><span class="op">)</span>,
      text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span> <span class="va">counts</span> <span class="op">)</span>, cex <span class="op">=</span> <span class="fl">1.4</span>,
      col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"white"</span>,<span class="st">"red"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="GenomAutomorphism_files/figure-html/barplot_3-1.png" width="384"></p>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p>1.<a name="1"></a> Sanchez R, Morgado E, Grau R. Gene algebra from a genetic code algebraic structure. J Math Biol. 2005 Oct;51(4):431-57. doi: 10.1007/s00285-005-0332-8. Epub 2005 Jul 13. PMID: 16012800. ( <a href="https://arxiv.org/pdf/q-bio/0412033.pdf">PDF</a>).</p>
<p>2.<a name="2"></a> Robersy Sanchez, Jesús Barreto (2021) Genomic Abelian Finite Groups. <a href="https://doi.org/10.1101/2021.06.01.446543">doi: 10.1101/2021.06.01.446543</a>.</p>
<p>3.<a name="3"></a> M. V José, E.R. Morgado, R. Sánchez, T. Govezensky, The 24 possible algebraic representations of the standard genetic code in six or in three dimensions, Adv. Stud. Biol. 4 (2012) 119–152.<a href="https://is.gd/na9eap">PDF</a>.</p>
<p>4.<a name="4"></a> R. Sanchez. Symmetric Group of the Genetic–Code Cubes. Effect of the Genetic–Code Architecture on the Evolutionary Process MATCH Commun. Math. Comput. Chem. 79 (2018) 527-560. <a href="https://bit.ly/2Z9mjM7">PDF</a>.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://genomaths.com">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
