<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="GenomAutomorphism">
<title>Get started-with GenomAutomorphism • GenomAutomorphism</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Get started-with GenomAutomorphism">
<meta property="og:description" content="GenomAutomorphism">
<meta property="og:image" content="https://genomaths.github.io/genomautomorphism/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GenomAutomorphism</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.5.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../articles/GenomAutomorphism.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Package functions</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/GenomAutomorphism.html">GenomAutomorphism</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/automorphism_and_decision_tree_II.html">A Short Introduction to Algebraic Taxonomy on Genes Regions II</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/covid_19.html">Automorphism analysis on COVID-19 data</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/Modular_Matrix_Operation_of_Mutational_Events.html">Modular Matrix Operations of Mutational Events</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Get started-with GenomAutomorphism</h1>
                        <h4 data-toc-skip class="author">Robersy
Sanchez</h4>
            <address class="author_afil">
      Department of Biology. Pennsylvania State University, University
Park, PA 16802<br><a class="author_email" href="mailto:#"></a><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">28 April 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/genomaths/GenomAutomorphism/blob/HEAD/vignettes/GenomAutomorphism.Rmd" class="external-link"><code>vignettes/GenomAutomorphism.Rmd</code></a></small>
      <div class="d-none name"><code>GenomAutomorphism.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>A fast introduction into the analysis of DNA mutational events
      by means of automorphisms between two DNA sequences algebraically
      represented as Abelian finite group.</p>
    </div>
    
<div class="section level2">
<h2 id="genomautomorphism-overview">
<code>GenomAutomorphism</code> Overview<a class="anchor" aria-label="anchor" href="#genomautomorphism-overview"></a>
</h2>
<p>This is a R package to compute the automorphisms between pairwise
aligned DNA sequences represented as elements from a Genomic Abelian
group as described in reference (<a href="#id_1">1</a>). In a general
scenario, whole chromosomes or genomic regions from a population (from
any species or close related species) can be algebraically represented
as a direct sum of cyclic groups or more specifically Abelian
<em>p</em>-groups. Basically, we propose the representation of multiple
sequence alignments (MSA) of length <em>N</em> as a finite Abelian group
created by the direct sum of Abelian group of <em>prime-power
order</em>:</p>
<p><span class="math display">\[
\qquad G = (\mathbb{Z}_{p^{\alpha_{1}}_1})^{n_1} \oplus
(\mathbb{Z}_{p^{\alpha_{2}}_1})^{n_2} \oplus \dots \oplus
(\mathbb{Z}_{p^{\alpha_{k}}_k})^{n_k}
\]</span></p>
<p>Where, the <span class="math inline">\(p_i\)</span>’s are prime
numbers, <span class="math inline">\(\alpha_i \in \mathbb{N}\)</span>
and <span class="math inline">\(\mathbb{Z}_{p^{\alpha_{i}}_i}\)</span>
is the group of integer modulo <span class="math inline">\(p^{\alpha_{i}}_i\)</span>.</p>
<p>For the purpose of estimating the automorphism between two aligned
DNA sequences, <span class="math inline">\(p^{\alpha_{i}}_i \in \{5,
2^6, 5^3 \}\)</span>.</p>
</div>
<div class="section level2">
<h2 id="installing-genomautomorphism">Installing <code>GenomAutomorphism</code><a class="anchor" aria-label="anchor" href="#installing-genomautomorphism"></a>
</h2>
<p><code>GenomAutomorphism</code> uses several R dependencies, which can
be installed as follows:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span><span class="op">)</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Biostrings"</span>, <span class="st">"GenomicRanges"</span>, <span class="st">"S4Vectors"</span>,</span>
<span>        <span class="st">"BiocParallel"</span>, <span class="st">"GenomeInfoDb"</span>, <span class="st">"BiocGenerics"</span>, <span class="st">"numbers"</span>, <span class="st">"devtools"</span>,</span>
<span>        <span class="st">"doParallel"</span>, <span class="st">"data.table"</span>, <span class="st">"foreach"</span>,<span class="st">"parallel"</span><span class="op">)</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><code>GenomAutomorphism</code> can be installed as follows:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"GenomAutomorphism"</span><span class="op">)</span></span></code></pre></div>
<p>You can install <code>GenomAutomorphism</code> package from GitHub as
follow</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">'genomaths/GenomAutomorphism'</span><span class="op">)</span></span></code></pre></div>
<p>You can install <code>GenomAutomorphism</code> ‘beta’ package with
last updating (which would differ from Bioconductor version) from GitHub
as follow:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">'genomaths/GenomAutomorphism_beta'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="automorphisms">Automorphisms<a class="anchor" aria-label="anchor" href="#automorphisms"></a>
</h2>
<p>Herein, automorphisms are considered algebraic descriptions of
mutational event observed in codon sequences represented on different
Abelian groups. In particular, as described in references (<a href="#id_3">3-4</a>), for each representation of the codon set on a
defined Abelian group there are 24 possible isomorphic Abelian groups.
These Abelian groups can be labeled based on the DNA base-order used to
generate them. The set of 24 Abelian groups can be described as a group
isomorphic to the symmetric group of degree four (<span class="math inline">\(S_4\)</span>, see reference (<a href="#id_4">4</a>)).</p>
<p>For further support about the symmetric group on the 24 Abelian group
of genetic-code cubes, users can also see <a href="https://is.gd/pTl8Js" class="external-link">Symmetric Group of the
Genetic-CodeCubes.</a>, specifically the Mathematica notebook
<em>IntroductionToZ5GeneticCodeVectorSpace.nb</em> and interact with it
using Wolfram Player, freely available (for Windows and Linux OS) at, <a href="https://www.wolfram.com/player/" class="external-link uri">https://www.wolfram.com/player/</a>.</p>
<div class="section level3">
<h3 id="load-the-r-libraries">Load the R libraries<a class="anchor" aria-label="anchor" href="#load-the-r-libraries"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">Biostrings</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/genomaths/GenomAutomorphism" class="external-link">GenomAutomorphism</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="read-the-alignment-fasta-and-encode-the-sequences">Read the alignment <em>FASTA</em> and encode the sequences<a class="anchor" aria-label="anchor" href="#read-the-alignment-fasta-and-encode-the-sequences"></a>
</h3>
<p>A pairwise sequence alignment of protein coding regions SARS
coronavirus GZ02 (GenBank: AY390556.1) and Bat SARS-like coronavirus
isolate Rs7327 (GenBank: KY417151.1) is provided with the package.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">covid_aln</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span></span>
<span><span class="va">covid_aln</span></span>
<span><span class="co">#&gt; DNAMultipleAlignment with 2 rows and 29166 columns</span></span>
<span><span class="co">#&gt;      aln                                                                                            names               </span></span>
<span><span class="co">#&gt; [1] <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span> lcl|AY390556.1_cd...</span></span>
<span><span class="co">#&gt; [2] <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span> lcl|KY417151.1_cd...</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="group-representations">Group representations<a class="anchor" aria-label="anchor" href="#group-representations"></a>
</h2>
<p>Group operations defined on the sets of DNA bases and codons are
associated to physicochemical or/and biophysical relationships between
DNA bases and between codons and aminoacids. In other words, a proper
definition of a group operation on the set of bases or on the set of
codons will encode the physicochemical or/and biophysical relationships
between the set’s elements. Thus, <em>by group operations defined on the
set of bases or on the set of codons, we understand an encoding applied
to represent specified physicochemical or/and biophysical relationships
as group operations between the elements of the set</em>. Then, we shall
say that <em><em>such an encoding permits the representation of DNA
bases, codons, genes, and genomic sequences as elements from algebraic
structures</em></em>.</p>
<p>The DNA base set can be represented in 24 possible base orders, which
leads to 24 possible representations of the genetic code. Each genetic
code representation base-triplets on the Galois field <em>GF(4)</em> (or
in <em>GF(5)</em>) leads to genetic code vector 3D-space, which is
mathematically equivalent to a cube inserted in the 3D space (1). Each
cube is denoted according to the corresponding base order.</p>
<p>Given a base-order, say ‘ACGT’, the Abelian group defined on this
ordered set is isomorphic to the Abelian group defined on the set of
integers modulo 4 (<span class="math inline">\(\mathbb{Z}_{4}\)</span>).
In practical terms, this is equivalent to replace each DNA base by the
corresponding integer element. The base replacement in cube “ACGT and
group”Z4” (<span class="math inline">\(\mathbb{Z}_{4}\)</span>) is:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/base2int.html">base2int</a></span><span class="op">(</span><span class="st">"ACGT"</span>, group <span class="op">=</span> <span class="st">"Z4"</span>, cube <span class="op">=</span> <span class="st">"ACGT"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0 1 2 3</span></span></code></pre></div>
<p>The base replacement in cube “ACGT and group ‘Z5’ (<span class="math inline">\(\mathbb{Z}_{5}\)</span>):</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/base2int.html">base2int</a></span><span class="op">(</span><span class="st">"ACGT"</span>, group <span class="op">=</span> <span class="st">"Z5"</span>, cube <span class="op">=</span> <span class="st">"ACGT"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1 2 3 4</span></span></code></pre></div>
<p>After the DNA sequence is read, the corresponding codon sequences can
be represented in the Abelian group <span class="math inline">\(\mathbb{Z}_{64}\)</span> (i.e., the set of
integers remainder modulo 64). The codon coordinates are requested on
the cube ACGT. Following reference (<a href="#id_4">4</a>)), cubes are
labeled based on the order of DNA bases used to define the sum
operation.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">codons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/codon_coord.html">codon_coord</a></span><span class="op">(</span></span>
<span>                    codon <span class="op">=</span> <span class="va">covid_aln</span>, </span>
<span>                    cube <span class="op">=</span> <span class="st">"ACGT"</span>, </span>
<span>                    group <span class="op">=</span> <span class="st">"Z64"</span>, </span>
<span>                    chr <span class="op">=</span> <span class="fl">1L</span>,</span>
<span>                    strand <span class="op">=</span> <span class="st">"+"</span>,</span>
<span>                    start <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                    end <span class="op">=</span> <span class="fl">750</span><span class="op">)</span></span>
<span><span class="va">codons</span></span>
<span><span class="co">#&gt; CodonGroup object with 250 ranges and 4 metadata columns:</span></span>
<span><span class="co">#&gt;         seqnames    ranges strand |        seq1        seq2      coord1      coord2</span></span>
<span><span class="co">#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;     [1]        1         1      + |         ATG         ATG          50          50</span></span>
<span><span class="co">#&gt;     [2]        1         2      + |         GAG         GAG          10          10</span></span>
<span><span class="co">#&gt;     [3]        1         3      + |         AGC         AGC          33          33</span></span>
<span><span class="co">#&gt;     [4]        1         4      + |         CTT         CTT          55          55</span></span>
<span><span class="co">#&gt;     [5]        1         5      + |         GTT         GTT          59          59</span></span>
<span><span class="co">#&gt;     ...      ...       ...    ... .         ...         ...         ...         ...</span></span>
<span><span class="co">#&gt;   [246]        1       246      + |         GAT         GAT          11          11</span></span>
<span><span class="co">#&gt;   [247]        1       247      + |         AAG         AAG           2           2</span></span>
<span><span class="co">#&gt;   [248]        1       248      + |         AGC         AGC          33          33</span></span>
<span><span class="co">#&gt;   [249]        1       249      + |         TAC         TAT          13          15</span></span>
<span><span class="co">#&gt;   [250]        1       250      + |         GAG         GAG          10          10</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>The codon sequences (<em>seq1</em> and <em>seq2</em>) with their
corresponding coordinates (left) are returned, as well as the
coordinated representation on <span class="math inline">\(\mathbb{Z}_{64}\)</span> (<em>coord1</em> and
<em>coord2</em>).</p>
<div class="section level3">
<h3 id="dual-genetic-code-cubes">
<em>“Dual”</em> genetic-code cubes<a class="anchor" aria-label="anchor" href="#dual-genetic-code-cubes"></a>
</h3>
<p>The particular interest are the coordinate representation on
<em>“dual”</em> genetic-code cubes. These are cubes where codons with
complementary base pairs have the same coordinates in the corresponding
cubes, as shown in reference (<a href="#id_4">4</a>)). Each pair of
<em>“dual”</em> cubes integrates a group.</p>
<p>For example, let’s consider the complementary codons “ACG” and “TGC”,
with complementary base pairs: A::T, C:::G, and G:::C, where symbol “:”
denotes the hydrogen bonds between the bases.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACG"</span>, <span class="st">"TGC"</span><span class="op">)</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></span><span class="op">(</span><span class="va">x0</span><span class="op">)</span></span>
<span><span class="va">x1</span></span>
<span><span class="co">#&gt; DNAStringSet object of length 2:</span></span>
<span><span class="co">#&gt;     width seq</span></span>
<span><span class="co">#&gt; [1]     3 <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span></span>
<span><span class="co">#&gt; [2]     3 <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span></span></span></code></pre></div>
<p>Their representations on the dual cubes “ACGT” and “TGCA” on <span class="math inline">\(\mathbb{Z}_{4}\)</span> are:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/base_methods.html">base_coord</a></span><span class="op">(</span><span class="va">x1</span>, cube <span class="op">=</span> <span class="st">"ACGT"</span><span class="op">)</span></span>
<span><span class="va">x2</span></span>
<span><span class="co">#&gt; BaseGroup object with 3 ranges and 4 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand |        seq1        seq2    coord1    coord2</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]        1         1      + |           A           T         0         3</span></span>
<span><span class="co">#&gt;   [2]        1         2      + |           C           G         1         2</span></span>
<span><span class="co">#&gt;   [3]        1         3      + |           G           C         2         1</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span>
<span></span>
<span><span class="va">x2.</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/base_methods.html">base_coord</a></span><span class="op">(</span><span class="va">x1</span>, cube <span class="op">=</span> <span class="st">"TGCA"</span><span class="op">)</span></span>
<span><span class="va">x2.</span></span>
<span><span class="co">#&gt; BaseGroup object with 3 ranges and 4 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand |        seq1        seq2    coord1    coord2</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]        1         1      + |           A           T         3         0</span></span>
<span><span class="co">#&gt;   [2]        1         2      + |           C           G         2         1</span></span>
<span><span class="co">#&gt;   [3]        1         3      + |           G           C         1         2</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>The sum of base coordinates modulo <span class="math inline">\(\mathbb{Z}_{4}\)</span> is 3.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## cube "ACGT"</span></span>
<span><span class="op">(</span><span class="va">x2</span><span class="op">$</span><span class="va">coord1</span> <span class="op">+</span> <span class="va">x2</span><span class="op">$</span><span class="va">coord2</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/Arithmetic.html" class="external-link">%%</a></span> <span class="fl">4</span>   </span>
<span><span class="co">#&gt; [1] 3 3 3</span></span>
<span></span>
<span><span class="co">## cube "TGCA"</span></span>
<span><span class="op">(</span><span class="va">x2.</span><span class="op">$</span><span class="va">coord1</span> <span class="op">+</span> <span class="va">x2.</span><span class="op">$</span><span class="va">coord2</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/Arithmetic.html" class="external-link">%%</a></span> <span class="fl">4</span>   </span>
<span><span class="co">#&gt; [1] 3 3 3</span></span></code></pre></div>
<p>The same result for the same codon on different cubes</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Codon ACG</span></span>
<span><span class="op">(</span><span class="va">x2</span><span class="op">$</span><span class="va">coord1</span> <span class="op">+</span> <span class="va">x2.</span><span class="op">$</span><span class="va">coord1</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/Arithmetic.html" class="external-link">%%</a></span> <span class="fl">4</span> </span>
<span><span class="co">#&gt; [1] 3 3 3</span></span>
<span></span>
<span><span class="co">## Codon TGC</span></span>
<span><span class="op">(</span><span class="va">x2</span><span class="op">$</span><span class="va">coord2</span> <span class="op">+</span> <span class="va">x2.</span><span class="op">$</span><span class="va">coord2</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/Arithmetic.html" class="external-link">%%</a></span> <span class="fl">4</span> </span>
<span><span class="co">#&gt; [1] 3 3 3</span></span></code></pre></div>
<p>Their codon representation on <span class="math inline">\(\mathbb{Z}_{64}\)</span> are:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## cube ACGT</span></span>
<span><span class="va">x3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/codon_coord.html">codon_coord</a></span><span class="op">(</span>codon <span class="op">=</span> <span class="va">x2</span>, group <span class="op">=</span> <span class="st">"Z64"</span><span class="op">)</span> </span>
<span><span class="va">x3</span></span>
<span><span class="co">#&gt; CodonGroup object with 1 range and 4 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand |        seq1        seq2      coord1      coord2</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;   [1]        1         1      + |         ACG         TGC          18          45</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span>
<span></span>
<span><span class="co">## cube TGCA</span></span>
<span><span class="va">x3.</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/codon_coord.html">codon_coord</a></span><span class="op">(</span>codon <span class="op">=</span> <span class="va">x2.</span>, group <span class="op">=</span> <span class="st">"Z64"</span><span class="op">)</span> </span>
<span><span class="va">x3.</span></span>
<span><span class="co">#&gt; CodonGroup object with 1 range and 4 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand |        seq1        seq2      coord1      coord2</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;   [1]        1         1      + |         ACG         TGC          45          18</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>The sum of base coordinates modulo <span class="math inline">\(\mathbb{Z}_{64}\)</span> is 63.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## cube "ACGT"</span></span>
<span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">x3</span><span class="op">$</span><span class="va">coord1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">x3</span><span class="op">$</span><span class="va">coord2</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/Arithmetic.html" class="external-link">%%</a></span> <span class="fl">64</span>  </span>
<span><span class="co">#&gt; [1] 63</span></span>
<span></span>
<span><span class="co">## cube "TGCA"</span></span>
<span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">x3.</span><span class="op">$</span><span class="va">coord1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">x3.</span><span class="op">$</span><span class="va">coord2</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/Arithmetic.html" class="external-link">%%</a></span> <span class="fl">64</span>   </span>
<span><span class="co">#&gt; [1] 63</span></span></code></pre></div>
<p>The same result for the same codon on different cubes</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Codon ACG</span></span>
<span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">x3</span><span class="op">$</span><span class="va">coord1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">x3.</span><span class="op">$</span><span class="va">coord1</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/Arithmetic.html" class="external-link">%%</a></span> <span class="fl">64</span> </span>
<span><span class="co">#&gt; [1] 63</span></span>
<span></span>
<span><span class="co">## Codon TGC</span></span>
<span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">x3</span><span class="op">$</span><span class="va">coord2</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">x3.</span><span class="op">$</span><span class="va">coord2</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/Arithmetic.html" class="external-link">%%</a></span> <span class="fl">64</span> </span>
<span><span class="co">#&gt; [1] 63</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="automorphisms-on-mathbbz_64">Automorphisms on <span class="math inline">\(\mathbb{Z}_{64}\)</span><a class="anchor" aria-label="anchor" href="#automorphisms-on-mathbbz_64"></a>
</h2>
<p>Automorphisms can be computed starting directly from the
<em>FASTA</em> file. Notice that we can work only with genomic regions
of our interest by giving the <em>start</em> and <em>end</em> alignment
coordinates. In <span class="math inline">\(\mathbb{Z}_{64}\)</span>
automorphisms are described as functions <span class="math inline">\(f(x) = k\,x\quad mod\,64\)</span>, where <span class="math inline">\(k\)</span> and <span class="math inline">\(x\)</span> are elements from the set of integers
modulo 64. Below, in function <a href="https://is.gd/WMxUKz" class="external-link">automorphism</a> three important arguments
are given values: <em>group = “Z64”</em>, <em>cube = c(“ACGT”,
“TGCA”)</em>, and <em>cube_alt = c(“CATG”, “GTAC”)</em>.</p>
<p>In groups “Z64” and “Z125” not all the mutational events can be
described as automorphisms from a given cube. The analysis of
automorphisms is then accomplished in the set of <em>dual</em>
genetic-code cubes. A character string denoting pairs of <em>dual</em>
genetic-code cubes, is given as argument for <em>cube</em>. Setting for
group specifies on which group the automorphisms will be computed. These
groups can be: “Z5”, “Z64”, “Z125”, and “Z5^3”.</p>
<p>If automorphisms are not found in first set of dual cubes, then the
algorithm search for automorphisms in a alternative set of dual
cubes.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">autm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphisms.html">automorphisms</a></span><span class="op">(</span></span>
<span>                    seqs <span class="op">=</span> <span class="va">covid_aln</span>,</span>
<span>                    group <span class="op">=</span> <span class="st">"Z64"</span>,</span>
<span>                    cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"TGCA"</span><span class="op">)</span>,</span>
<span>                    cube_alt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CATG"</span>, <span class="st">"GTAC"</span><span class="op">)</span>,</span>
<span>                    start <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                    end <span class="op">=</span> <span class="fl">750</span>, </span>
<span>                    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">autm</span></span>
<span><span class="co">#&gt; Automorphism object with 250 ranges and 8 metadata columns:</span></span>
<span><span class="co">#&gt;         seqnames    ranges strand |        seq1        seq2         aa1         aa2    coord1    coord2      autm</span></span>
<span><span class="co">#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;     [1]        1         1      + |         ATG         ATG           M           M        50        50         1</span></span>
<span><span class="co">#&gt;     [2]        1         2      + |         GAG         GAG           E           E        10        10         1</span></span>
<span><span class="co">#&gt;     [3]        1         3      + |         AGC         AGC           S           S        33        33         1</span></span>
<span><span class="co">#&gt;     [4]        1         4      + |         CTT         CTT           L           L        55        55         1</span></span>
<span><span class="co">#&gt;     [5]        1         5      + |         GTT         GTT           V           V        59        59         1</span></span>
<span><span class="co">#&gt;     ...      ...       ...    ... .         ...         ...         ...         ...       ...       ...       ...</span></span>
<span><span class="co">#&gt;   [246]        1       246      + |         GAT         GAT           D           D        11        11         1</span></span>
<span><span class="co">#&gt;   [247]        1       247      + |         AAG         AAG           K           K         2         2         1</span></span>
<span><span class="co">#&gt;   [248]        1       248      + |         AGC         AGC           S           S        33        33         1</span></span>
<span><span class="co">#&gt;   [249]        1       249      + |         TAC         TAT           Y           Y        13        15        11</span></span>
<span><span class="co">#&gt;   [250]        1       250      + |         GAG         GAG           E           E        10        10         1</span></span>
<span><span class="co">#&gt;                cube</span></span>
<span><span class="co">#&gt;         &lt;character&gt;</span></span>
<span><span class="co">#&gt;     [1]        ACGT</span></span>
<span><span class="co">#&gt;     [2]        ACGT</span></span>
<span><span class="co">#&gt;     [3]        ACGT</span></span>
<span><span class="co">#&gt;     [4]        ACGT</span></span>
<span><span class="co">#&gt;     [5]        ACGT</span></span>
<span><span class="co">#&gt;     ...         ...</span></span>
<span><span class="co">#&gt;   [246]        ACGT</span></span>
<span><span class="co">#&gt;   [247]        ACGT</span></span>
<span><span class="co">#&gt;   [248]        ACGT</span></span>
<span><span class="co">#&gt;   [249]        ACGT</span></span>
<span><span class="co">#&gt;   [250]        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>Observe that two new columns were added, the automorphism coefficient
<span class="math inline">\(k\)</span> (named as <em>autm</em>) and the
genetic-code cube where the automorphism was found. By convention the
DNA sequence is given for the positive strand. Since the <em>dual
cube</em> of <strong>“ACGT”</strong> corresponds to the complementary
base order <strong>TGCA</strong>, automorphisms described by the cube
<strong>TGCA</strong> represent mutational events affecting the DNA
negative strand (-).</p>
<p>The last result can be summarized by gene regions as follow:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aut_range</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphismByRanges.html">automorphismByRanges</a></span><span class="op">(</span><span class="va">autm</span><span class="op">)</span></span>
<span><span class="va">aut_range</span></span>
<span><span class="co">#&gt; GRanges object with 9 ranges and 1 metadata column:</span></span>
<span><span class="co">#&gt;       seqnames    ranges strand |        cube</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span></span>
<span><span class="co">#&gt;   [1]        1     1-102      + |        ACGT</span></span>
<span><span class="co">#&gt;   [2]        1       103      - |        TGCA</span></span>
<span><span class="co">#&gt;   [3]        1   104-105      + |        ACGT</span></span>
<span><span class="co">#&gt;   [4]        1       106      - |        TGCA</span></span>
<span><span class="co">#&gt;   [5]        1   107-201      + |        ACGT</span></span>
<span><span class="co">#&gt;   [6]        1       202      - |        TGCA</span></span>
<span><span class="co">#&gt;   [7]        1   203-205      + |        ACGT</span></span>
<span><span class="co">#&gt;   [8]        1       206      - |        TGCA</span></span>
<span><span class="co">#&gt;   [9]        1   207-250      + |        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>That is, function <a href="https://is.gd/5RWHFF" class="external-link">automorphismByRanges</a> permits the
classification of the pairwise alignment of protein-coding sub-regions
based on the mutational events observed on it quantitatively represented
as automorphisms on genetic-code cubes.</p>
<p>Searching for automorphisms on <span class="math inline">\(\mathbb{Z}_{64}\)</span> permits us a quantitative
differentiation between mutational events at different codon positions
from a given DNA protein-encoding region. As shown in reference (<a href="#id_4">4</a>) a set of different cubes can be applied to describe
the best evolutionary aminoacid scale highly correlated with aminoacid
physicochemical properties describing the observed evolutionary process
in a given protein.</p>
<p>More information about this subject can be found in the supporting
material from reference (<a href="#id_4">4</a>)) at GitHub <a href="https://is.gd/pTl8Js" class="external-link">GenomeAlgebra_SymmetricGroup</a>,
particularly by interacting with the Mathematica notebook <a href="https://is.gd/LVd06G" class="external-link">Genetic-Code-Scales_of_Amino-Acids.nb</a>.</p>
</div>
<div class="section level2">
<h2 id="automorphisms-between-whole-genomes-of-sars-cov-2-related-coronaviruses">Automorphisms between whole genomes of SARS-CoV-2 related
coronaviruses<a class="anchor" aria-label="anchor" href="#automorphisms-between-whole-genomes-of-sars-cov-2-related-coronaviruses"></a>
</h2>
<p>Next, the automorphism for the whole pairwise alignment of SARS-CoV-2
related coronaviruses:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Do not need to run it. </span></span>
<span><span class="va">covid_autm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphisms.html">automorphisms</a></span><span class="op">(</span></span>
<span>                    seq <span class="op">=</span> <span class="va">covid_aln</span>,</span>
<span>                    group <span class="op">=</span> <span class="st">"Z64"</span>,</span>
<span>                    cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"TGCA"</span><span class="op">)</span>,</span>
<span>                    cube_alt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CATG"</span>, <span class="st">"GTAC"</span><span class="op">)</span>,</span>
<span>                    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>This data is available with the package</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">covid_autm</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span></span>
<span><span class="va">covid_autm</span></span>
<span><span class="co">#&gt; Automorphism object with 9722 ranges and 8 metadata columns:</span></span>
<span><span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2         aa1         aa2    coord1    coord2      autm</span></span>
<span><span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;      [1]        1         1      + |         ATG         ATG           M           M        50        50         1</span></span>
<span><span class="co">#&gt;      [2]        1         2      + |         GAG         GAG           E           E        10        10         1</span></span>
<span><span class="co">#&gt;      [3]        1         3      + |         AGC         AGC           S           S        33        33         1</span></span>
<span><span class="co">#&gt;      [4]        1         4      + |         CTT         CTT           L           L        55        55         1</span></span>
<span><span class="co">#&gt;      [5]        1         5      + |         GTT         GTT           V           V        59        59         1</span></span>
<span><span class="co">#&gt;      ...      ...       ...    ... .         ...         ...         ...         ...       ...       ...       ...</span></span>
<span><span class="co">#&gt;   [9718]        1      9718      + |         TCA         TCA           S           S        28        28         1</span></span>
<span><span class="co">#&gt;   [9719]        1      9719      + |         ACT         ACT           T           T        19        19         1</span></span>
<span><span class="co">#&gt;   [9720]        1      9720      + |         CAG         CAG           Q           Q         6         6         1</span></span>
<span><span class="co">#&gt;   [9721]        1      9721      + |         GCA         GCA           A           A        24        24         1</span></span>
<span><span class="co">#&gt;   [9722]        1      9722      + |         TAA         TAA           *           *        12        12         1</span></span>
<span><span class="co">#&gt;                 cube</span></span>
<span><span class="co">#&gt;          &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1]        ACGT</span></span>
<span><span class="co">#&gt;      [2]        ACGT</span></span>
<span><span class="co">#&gt;      [3]        ACGT</span></span>
<span><span class="co">#&gt;      [4]        ACGT</span></span>
<span><span class="co">#&gt;      [5]        ACGT</span></span>
<span><span class="co">#&gt;      ...         ...</span></span>
<span><span class="co">#&gt;   [9718]        ACGT</span></span>
<span><span class="co">#&gt;   [9719]        ACGT</span></span>
<span><span class="co">#&gt;   [9720]        ACGT</span></span>
<span><span class="co">#&gt;   [9721]        ACGT</span></span>
<span><span class="co">#&gt;   [9722]        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>And the summary by range</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aut_range</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphismByRanges.html">automorphismByRanges</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">)</span></span>
<span><span class="va">aut_range</span></span>
<span><span class="co">#&gt; GRanges object with 460 ranges and 1 metadata column:</span></span>
<span><span class="co">#&gt;         seqnames    ranges strand |        cube</span></span>
<span><span class="co">#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span></span>
<span><span class="co">#&gt;     [1]        1     1-102      + |        ACGT</span></span>
<span><span class="co">#&gt;     [2]        1       103      - |        TGCA</span></span>
<span><span class="co">#&gt;     [3]        1   104-105      + |        ACGT</span></span>
<span><span class="co">#&gt;     [4]        1       106      - |        TGCA</span></span>
<span><span class="co">#&gt;     [5]        1   107-201      + |        ACGT</span></span>
<span><span class="co">#&gt;     ...      ...       ...    ... .         ...</span></span>
<span><span class="co">#&gt;   [456]        1 9577-9608      + |        ACGT</span></span>
<span><span class="co">#&gt;   [457]        1      9609      - |        TGCA</span></span>
<span><span class="co">#&gt;   [458]        1 9610-9674      + |        ACGT</span></span>
<span><span class="co">#&gt;   [459]        1      9675      - |        TGCA</span></span>
<span><span class="co">#&gt;   [460]        1 9676-9722      + |        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>Regions no described by automorphism can be described as translations
(labeled “Trnl”) and they can be shown as follow:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">idx</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">$</span><span class="va">cube</span> <span class="op">==</span> <span class="st">"Trnl"</span><span class="op">)</span></span>
<span><span class="va">covid_autm</span><span class="op">[</span> <span class="va">idx</span> <span class="op">]</span></span>
<span><span class="co">#&gt; Automorphism object with 19 ranges and 8 metadata columns:</span></span>
<span><span class="co">#&gt;        seqnames    ranges strand |        seq1        seq2         aa1         aa2    coord1    coord2      autm</span></span>
<span><span class="co">#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;    [1]        1       916      + |         AAC         AAA           N           K         1         0         0</span></span>
<span><span class="co">#&gt;    [2]        1       944      + |         ---         CCG           -           P        NA        22         0</span></span>
<span><span class="co">#&gt;    [3]        1       971      + |         ACA         ---           T           -        16        NA         0</span></span>
<span><span class="co">#&gt;    [4]        1      7077      + |         ---         AAA           -           K        NA         0         0</span></span>
<span><span class="co">#&gt;    [5]        1      9109      + |         AAT         AAA           N           K         3         0         0</span></span>
<span><span class="co">#&gt;    ...      ...       ...    ... .         ...         ...         ...         ...       ...       ...       ...</span></span>
<span><span class="co">#&gt;   [15]        1      9277      + |         CTT         ---           L           -        55        NA         0</span></span>
<span><span class="co">#&gt;   [16]        1      9278      + |         CAT         ---           H           -         7        NA         0</span></span>
<span><span class="co">#&gt;   [17]        1      9279      + |         GAA         ---           E           -         8        NA         0</span></span>
<span><span class="co">#&gt;   [18]        1      9295      + |         AAA         ---           K           -         0        NA         0</span></span>
<span><span class="co">#&gt;   [19]        1      9296      + |         CGA         ---           R           -        36        NA         0</span></span>
<span><span class="co">#&gt;               cube</span></span>
<span><span class="co">#&gt;        &lt;character&gt;</span></span>
<span><span class="co">#&gt;    [1]        Trnl</span></span>
<span><span class="co">#&gt;    [2]        Trnl</span></span>
<span><span class="co">#&gt;    [3]        Trnl</span></span>
<span><span class="co">#&gt;    [4]        Trnl</span></span>
<span><span class="co">#&gt;    [5]        Trnl</span></span>
<span><span class="co">#&gt;    ...         ...</span></span>
<span><span class="co">#&gt;   [15]        Trnl</span></span>
<span><span class="co">#&gt;   [16]        Trnl</span></span>
<span><span class="co">#&gt;   [17]        Trnl</span></span>
<span><span class="co">#&gt;   [18]        Trnl</span></span>
<span><span class="co">#&gt;   [19]        Trnl</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>These codon positions cover insertion-deletion (<em>indel</em>)
mutational events. The wholes regions can be summarized typing:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">idx</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">aut_range</span><span class="op">$</span><span class="va">cube</span> <span class="op">==</span> <span class="st">"Trnl"</span><span class="op">)</span></span>
<span><span class="va">aut_range</span><span class="op">[</span> <span class="va">idx</span> <span class="op">]</span></span>
<span><span class="co">#&gt; GRanges object with 13 ranges and 1 metadata column:</span></span>
<span><span class="co">#&gt;        seqnames    ranges strand |        cube</span></span>
<span><span class="co">#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span></span>
<span><span class="co">#&gt;    [1]        1       916      + |        Trnl</span></span>
<span><span class="co">#&gt;    [2]        1       944      + |        Trnl</span></span>
<span><span class="co">#&gt;    [3]        1       971      + |        Trnl</span></span>
<span><span class="co">#&gt;    [4]        1      7077      + |        Trnl</span></span>
<span><span class="co">#&gt;    [5]        1      9109      + |        Trnl</span></span>
<span><span class="co">#&gt;    ...      ...       ...    ... .         ...</span></span>
<span><span class="co">#&gt;    [9]        1 9244-9245      + |        Trnl</span></span>
<span><span class="co">#&gt;   [10]        1      9247      + |        Trnl</span></span>
<span><span class="co">#&gt;   [11]        1      9257      + |        Trnl</span></span>
<span><span class="co">#&gt;   [12]        1 9277-9279      + |        Trnl</span></span>
<span><span class="co">#&gt;   [13]        1 9295-9296      + |        Trnl</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>Only one indel mutation was found in the region where the spike
glycoprotein is located: 7076 - 8331. That is, the pairwise alignment of
SARS coronavirus GZ02 and Bat SARS-like coronavirus (bat-SL-CoVZC45)
reveals 8 single <em>indel</em> mutational events, four regions with two
<em>indel</em> mutations and one region with 3 <em>indel</em>
mutations.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">aut_range</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;    seqnames start  end width strand cube</span></span>
<span><span class="co">#&gt; 1         1   916  916     1      + Trnl</span></span>
<span><span class="co">#&gt; 2         1   944  944     1      + Trnl</span></span>
<span><span class="co">#&gt; 3         1   971  971     1      + Trnl</span></span>
<span><span class="co">#&gt; 4         1  7077 7077     1      + Trnl</span></span>
<span><span class="co">#&gt; 5         1  9109 9109     1      + Trnl</span></span>
<span><span class="co">#&gt; 6         1  9188 9189     2      + Trnl</span></span>
<span><span class="co">#&gt; 7         1  9230 9230     1      + Trnl</span></span>
<span><span class="co">#&gt; 8         1  9234 9235     2      + Trnl</span></span>
<span><span class="co">#&gt; 9         1  9244 9245     2      + Trnl</span></span>
<span><span class="co">#&gt; 10        1  9247 9247     1      + Trnl</span></span>
<span><span class="co">#&gt; 11        1  9257 9257     1      + Trnl</span></span>
<span><span class="co">#&gt; 12        1  9277 9279     3      + Trnl</span></span>
<span><span class="co">#&gt; 13        1  9295 9296     2      + Trnl</span></span>
<span></span>
<span><span class="co">## region width</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">aut_range</span><span class="op">[</span> <span class="va">idx</span> <span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] 1 1 1 1 1 2 1 2 2 1 1 3 2</span></span></code></pre></div>
<p>In general, <em>indel</em> mutational event can be modeled as
translations on <span class="math inline">\(\mathbb{Z}_{64}\)</span>.</p>
<div class="section level3">
<h3 id="bar-plot-automorphism-distribution-by-cubes">Bar plot automorphism distribution by cubes<a class="anchor" aria-label="anchor" href="#bar-plot-automorphism-distribution-by-cubes"></a>
</h3>
<p>The automorphism distribution by cubes can be summarized in the
bar-plot graphic</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">$</span><span class="va">cube</span><span class="op">[</span> <span class="va">covid_autm</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="fl">1</span> <span class="op">|</span> </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">$</span><span class="va">autm</span><span class="op">)</span> <span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, cex <span class="op">=</span> <span class="fl">0.9</span>, font <span class="op">=</span> <span class="fl">2</span>, mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">6</span>,<span class="fl">4</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="va">counts</span>, main<span class="op">=</span><span class="st">"Automorphism distribution"</span>,</span>
<span>        xlab<span class="op">=</span><span class="st">"Genetic-code cube representation"</span>,</span>
<span>        ylab<span class="op">=</span><span class="st">"Fixed mutational events"</span>,</span>
<span>        col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"darkblue"</span>,<span class="st">"red"</span>, <span class="st">"darkgreen"</span><span class="op">)</span>, </span>
<span>        border <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>        cex.lab <span class="op">=</span> <span class="fl">2</span>, cex.main <span class="op">=</span> <span class="fl">1.5</span>, cex.names <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></span><span class="op">(</span><span class="fl">2</span>, at <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">200</span>, <span class="fl">400</span>, <span class="fl">600</span>, <span class="fl">800</span><span class="op">)</span>, cex.axis <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html" class="external-link">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>,line <span class="op">=</span> <span class="op">-</span><span class="fl">1.5</span>, at <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.7</span>, <span class="fl">1.9</span>, <span class="fl">3.1</span>, <span class="fl">4.3</span>, <span class="fl">5.5</span><span class="op">)</span>,</span>
<span>    text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span> <span class="va">counts</span> <span class="op">)</span>, cex <span class="op">=</span> <span class="fl">1.4</span>,</span>
<span>    col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"white"</span>,<span class="st">"yellow"</span>, <span class="st">"black"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomAutomorphism_files/figure-html/barplot-1.png" width="576"></p>
</div>
<div class="section level3">
<h3 id="grouping-automorphism-by-automorphisms-coefficients--types-of-mutations">Grouping automorphism by automorphism’s coefficients. Types of
mutations<a class="anchor" aria-label="anchor" href="#grouping-automorphism-by-automorphisms-coefficients--types-of-mutations"></a>
</h3>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">autby_coef</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphism_bycoef.html">automorphism_bycoef</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">)</span></span>
<span><span class="va">autby_coef</span> <span class="op">&lt;-</span> <span class="va">autby_coef</span><span class="op">[</span> <span class="va">autby_coef</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">autby_coef</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="op">-</span><span class="fl">1</span>  <span class="op">]</span></span></code></pre></div>
<p>Barplot of frequency of mutation types greater than 2.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">autby_coef</span><span class="op">$</span><span class="va">mut_type</span><span class="op">)</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">counts</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">count.</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span> <span class="va">counts</span> <span class="op">&gt;</span> <span class="fl">2</span> <span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, cex.axis <span class="op">=</span> <span class="fl">2</span>, font <span class="op">=</span> <span class="fl">2</span>, las <span class="op">=</span> <span class="fl">1</span>, </span>
<span>    cex.main <span class="op">=</span> <span class="fl">1.4</span>, mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">2</span>,<span class="fl">4</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="va">count.</span>, main<span class="op">=</span><span class="st">"Automorphism distribution per Mutation type"</span>,</span>
<span>        col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"yellow"</span>, <span class="st">"blue"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">36</span><span class="op">)</span>, </span>
<span>        border <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">FALSE</span>,las<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">2</span>,  cex.axis <span class="op">=</span> <span class="fl">2</span>, line <span class="op">=</span> <span class="op">-</span><span class="fl">1.8</span> <span class="op">)</span></span></code></pre></div>
<p><img src="GenomAutomorphism_files/figure-html/barplot_2-1.png" width="1344"></p>
<p>Every single base mutational event across the MSA was classified
according IUPAC nomenclature: 1) According to the number of hydrogen
bonds (on DNA/RNA double helix): strong S={C, G} (three hydrogen bonds)
and weak W={A, U} (two hydrogen bonds). According to the chemical type:
purines R={A, G} and pyrimidines Y={C, U}. 3). According to the presence
of amino or keto groups on the base rings: amino M={C, A} and keto K={G,
T}. Constant (hold) base positions were labeled with letter H. So, codon
positions labeled as HKH means that the first and third bases remains
constant and mutational events between bases G and T were found in the
MSA.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; HHY HHR HHW YHH HHK HHM RHH HRH HYH --- HHS MHH WHH HMH KHH HYW YHW RHK RHY WHW YHM YHY HKH HMR HRR HRY HWH KHW RYY YHK </span></span>
<span><span class="co">#&gt; 425 189  89  34  32  25  24  18  18  16  13  12  11   7   6   5   5   4   4   4   4   4   3   3   3   3   3   3   3   3 </span></span>
<span><span class="co">#&gt; YHR HMW HYK KHY KSW MHK MHW MRH RHM RHR RHW RKW RMH RMW RRY RSW RYH RYW SHH SHW SRH WMH WMS WSY YWH HKK HKR HMK HMY HSR </span></span>
<span><span class="co">#&gt;   3   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   2   1   1   1   1   1 </span></span>
<span><span class="co">#&gt; HWS HWW HWY HYR HYY KHK KHR KKS KMH KMW KRK KRY KSH MHR MHS MKM MMM MRM MWH MWM MWS MWW MYY RMY RRK RRM RSK RYK SHM SHY </span></span>
<span><span class="co">#&gt;   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1 </span></span>
<span><span class="co">#&gt; SKY SMW SRM SWR SWY SYH WHY WRH WRY WWH WWK WWS WWY WYH WYM WYW YHS YKH YKY YRY YWM YWW YWY YYH YYR </span></span>
<span><span class="co">#&gt;   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1   1</span></span></code></pre></div>
<p>The analysis of the frequency of mutational events (automorphisms,
COVID: human SARS coronavirus GZ02 vs Bat SARS-like coronavirus isolate
at-SL-CoVZC45) by mutation types is shown in the last figure. Results
are consistent with the well-known observation highlighted by Crick: the
highest mutational rate is found in the third base of the codon (HHY:
425, HHR: 189, HHW: 88), followed by YHH: 34 in the first base, and the
lowest rate is found in the second one (<a href="#id_5">5</a>).</p>
</div>
<div class="section level3">
<h3 id="conserved-and-non-conserved-regions">Conserved and non-conserved regions<a class="anchor" aria-label="anchor" href="#conserved-and-non-conserved-regions"></a>
</h3>
<div class="section level4">
<h4 id="conserved-regions">Conserved regions<a class="anchor" aria-label="anchor" href="#conserved-regions"></a>
</h4>
<p>Conserved and non-conserved gene regions can be easily observed in
most of MSA editing bioinformatic tools. However, here were interesting
into get the regions coordinates for further downstream analysis.</p>
<p>Conserved regions from pairwise comparisons are obtain with function
conserved_regions:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">conserv</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/conserved_regions.html">conserved_regions</a></span><span class="op">(</span><span class="va">covid_autm</span><span class="op">)</span></span>
<span><span class="va">conserv</span></span>
<span><span class="co">#&gt; AutomorphismByCoef object with 6348 ranges and 7 metadata columns:</span></span>
<span><span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2         aa1         aa2      autm    mut_type        cube</span></span>
<span><span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1]        1      1-19      + |         ATG         ATG           M           M         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      [2]        1      1-19      + |         GAG         GAG           E           E         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      [3]        1      1-19      + |         AGC         AGC           S           S         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      [4]        1      1-19      + |         CTT         CTT           L           L         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      [5]        1      1-19      + |         GTT         GTT           V           V         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;      ...      ...       ...    ... .         ...         ...         ...         ...       ...         ...         ...</span></span>
<span><span class="co">#&gt;   [6344]        1 9683-9722      + |         GGA         GGA           G           G         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   [6345]        1 9683-9722      + |         TCT         TCT           S           S         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   [6346]        1 9683-9722      + |         TCA         TCA           S           S         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   [6347]        1 9683-9722      + |         GCA         GCA           A           A         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   [6348]        1 9683-9722      + |         TAA         TAA           *           *         1         HHH        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>Several regions are similar for more than one comparison.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">conserv_unique</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/conserved_regions.html">conserved_regions</a></span><span class="op">(</span><span class="va">covid_autm</span>, output <span class="op">=</span> <span class="st">"unique"</span><span class="op">)</span></span>
<span><span class="va">conserv_unique</span></span>
<span><span class="co">#&gt; GRanges object with 744 ranges and 2 metadata columns:</span></span>
<span><span class="co">#&gt;         seqnames    ranges strand |      autm        cube</span></span>
<span><span class="co">#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;     [1]        1      1-19      + |         1        ACGT</span></span>
<span><span class="co">#&gt;     [2]        1     21-44      + |         1        ACGT</span></span>
<span><span class="co">#&gt;     [3]        1     47-52      + |         1        ACGT</span></span>
<span><span class="co">#&gt;     [4]        1     54-60      + |         1        ACGT</span></span>
<span><span class="co">#&gt;     [5]        1     62-80      + |         1        ACGT</span></span>
<span><span class="co">#&gt;     ...      ...       ...    ... .       ...         ...</span></span>
<span><span class="co">#&gt;   [740]        1 9610-9614      + |         1        ACGT</span></span>
<span><span class="co">#&gt;   [741]        1 9616-9648      + |         1        ACGT</span></span>
<span><span class="co">#&gt;   [742]        1 9650-9674      + |         1        ACGT</span></span>
<span><span class="co">#&gt;   [743]        1 9676-9681      + |         1        ACGT</span></span>
<span><span class="co">#&gt;   [744]        1 9683-9722      + |         1        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="automorphisms-on-mathbbz_125">Automorphisms on <span class="math inline">\(\mathbb{Z}_{125}\)</span><a class="anchor" aria-label="anchor" href="#automorphisms-on-mathbbz_125"></a>
</h2>
<p>Alternatively, we can use the algebraic representation on on <span class="math inline">\(\mathbb{Z}_{125}\)</span>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">autm_z125</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphisms.html">automorphisms</a></span><span class="op">(</span></span>
<span>                    seq <span class="op">=</span> <span class="va">covid_aln</span>, </span>
<span>                    group <span class="op">=</span> <span class="st">"Z125"</span>, </span>
<span>                    cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"TGCA"</span><span class="op">)</span>,</span>
<span>                    cube_alt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CATG"</span>, <span class="st">"GTAC"</span><span class="op">)</span>,</span>
<span>                    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>For the sake of reducing computational time in this example,
‘autm_z125’ is available with the package.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">autm_z125</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span></span>
<span><span class="va">autm_z125</span></span>
<span><span class="co">#&gt; Automorphism object with 9722 ranges and 8 metadata columns:</span></span>
<span><span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2         aa1         aa2    coord1    coord2      autm</span></span>
<span><span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;      [1]        1         1      + |         ATG         ATG           M           M       108       108         1</span></span>
<span><span class="co">#&gt;      [2]        1         2      + |         GAG         GAG           E           E        43        43         1</span></span>
<span><span class="co">#&gt;      [3]        1         3      + |         AGC         AGC           S           S        82        82         1</span></span>
<span><span class="co">#&gt;      [4]        1         4      + |         CTT         CTT           L           L       114       114         1</span></span>
<span><span class="co">#&gt;      [5]        1         5      + |         GTT         GTT           V           V       119       119         1</span></span>
<span><span class="co">#&gt;      ...      ...       ...    ... .         ...         ...         ...         ...       ...       ...       ...</span></span>
<span><span class="co">#&gt;   [9718]        1      9718      + |         TCA         TCA           S           S        71        71         1</span></span>
<span><span class="co">#&gt;   [9719]        1      9719      + |         ACT         ACT           T           T        59        59         1</span></span>
<span><span class="co">#&gt;   [9720]        1      9720      + |         CAG         CAG           Q           Q        38        38         1</span></span>
<span><span class="co">#&gt;   [9721]        1      9721      + |         GCA         GCA           A           A        66        66         1</span></span>
<span><span class="co">#&gt;   [9722]        1      9722      + |         TAA         TAA           *           *        46        46         1</span></span>
<span><span class="co">#&gt;                 cube</span></span>
<span><span class="co">#&gt;          &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1]        ACGT</span></span>
<span><span class="co">#&gt;      [2]        ACGT</span></span>
<span><span class="co">#&gt;      [3]        ACGT</span></span>
<span><span class="co">#&gt;      [4]        ACGT</span></span>
<span><span class="co">#&gt;      [5]        ACGT</span></span>
<span><span class="co">#&gt;      ...         ...</span></span>
<span><span class="co">#&gt;   [9718]        ACGT</span></span>
<span><span class="co">#&gt;   [9719]        ACGT</span></span>
<span><span class="co">#&gt;   [9720]        ACGT</span></span>
<span><span class="co">#&gt;   [9721]        ACGT</span></span>
<span><span class="co">#&gt;   [9722]        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>And the summary by range</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aut_range_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphismByRanges.html">automorphismByRanges</a></span><span class="op">(</span><span class="va">autm_z125</span><span class="op">)</span></span>
<span><span class="va">aut_range_2</span></span>
<span><span class="co">#&gt; GRanges object with 21 ranges and 1 metadata column:</span></span>
<span><span class="co">#&gt;        seqnames    ranges strand |        cube</span></span>
<span><span class="co">#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span></span>
<span><span class="co">#&gt;    [1]        1     1-943      + |        ACGT</span></span>
<span><span class="co">#&gt;    [2]        1       944      + |        Trnl</span></span>
<span><span class="co">#&gt;    [3]        1   945-970      + |        ACGT</span></span>
<span><span class="co">#&gt;    [4]        1       971      + |        Trnl</span></span>
<span><span class="co">#&gt;    [5]        1  972-7076      + |        ACGT</span></span>
<span><span class="co">#&gt;    ...      ...       ...    ... .         ...</span></span>
<span><span class="co">#&gt;   [17]        1 9258-9276      + |        ACGT</span></span>
<span><span class="co">#&gt;   [18]        1 9277-9279      + |        Trnl</span></span>
<span><span class="co">#&gt;   [19]        1 9280-9294      + |        ACGT</span></span>
<span><span class="co">#&gt;   [20]        1 9295-9296      + |        Trnl</span></span>
<span><span class="co">#&gt;   [21]        1 9297-9722      + |        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>The whole genome can be described by automorphisms on <span class="math inline">\(\mathbb{Z}_{125}\)</span>.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">autm_z125</span><span class="op">$</span><span class="va">cube</span><span class="op">[</span> <span class="va">autm_z125</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="fl">1</span> <span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, cex <span class="op">=</span> <span class="fl">1</span>, font <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="va">counts</span>, main<span class="op">=</span><span class="st">"Automorphism distribution"</span>,</span>
<span>        xlab<span class="op">=</span><span class="st">"Genetic-code cube representation"</span>,</span>
<span>        ylab<span class="op">=</span><span class="st">"Fixed mutational events"</span>,</span>
<span>        col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"darkblue"</span>,<span class="st">"red"</span><span class="op">)</span>, </span>
<span>        ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1300</span><span class="op">)</span>,</span>
<span>        border <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html" class="external-link">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>,line <span class="op">=</span> <span class="op">-</span><span class="fl">2</span>, at <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.7</span>, <span class="fl">1.9</span>, <span class="fl">3.1</span><span class="op">)</span>,</span>
<span>    text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span> <span class="va">counts</span> <span class="op">)</span>, cex <span class="op">=</span> <span class="fl">1.4</span>,</span>
<span>    col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"white"</span>,<span class="st">"red"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomAutomorphism_files/figure-html/barplot_3-1.png" width="288"></p>
</div>
<div class="section level2">
<h2 id="automorphisms-on-the-genetic-code-cube-representation-on-gf5">Automorphisms on the Genetic-code Cube Representation on GF(5)<a class="anchor" aria-label="anchor" href="#automorphisms-on-the-genetic-code-cube-representation-on-gf5"></a>
</h2>
<p>The Genetic-code Cube Representations on the Galois Field GF(5) were
studied in (<a href="#id_4">4</a>). Each codon is represented by each
coordinate in the 3D space.</p>
<p>Automorphisms are represented by diagonal matrices, with elements
<span class="math inline">\(x\)</span> in <span class="math inline">\(x
\in \mathbb{Z}_5\)</span>.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">autm_3d</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphisms.html">automorphisms</a></span><span class="op">(</span></span>
<span>                    seq <span class="op">=</span> <span class="va">covid_aln</span>, </span>
<span>                    group <span class="op">=</span> <span class="st">"Z5^3"</span>, </span>
<span>                    cube <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACGT"</span>, <span class="st">"TGCA"</span><span class="op">)</span>,</span>
<span>                    cube_alt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CATG"</span>, <span class="st">"GTAC"</span><span class="op">)</span>,</span>
<span>                    verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>The result is available with package</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">autm_3d</span>, package <span class="op">=</span> <span class="st">"GenomAutomorphism"</span><span class="op">)</span></span>
<span><span class="va">autm_3d</span></span>
<span><span class="co">#&gt; Automorphism object with 9722 ranges and 8 metadata columns:</span></span>
<span><span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2         aa1         aa2   coord1   coord2        autm</span></span>
<span><span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;matrix&gt; &lt;matrix&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1]        1         1      + |         ATG         ATG           M           M    1:4:3    1:4:3       1,1,1</span></span>
<span><span class="co">#&gt;      [2]        1         2      + |         GAG         GAG           E           E    3:1:3    3:1:3       1,1,1</span></span>
<span><span class="co">#&gt;      [3]        1         3      + |         AGC         AGC           S           S    1:3:2    1:3:2       1,1,1</span></span>
<span><span class="co">#&gt;      [4]        1         4      + |         CTT         CTT           L           L    2:4:4    2:4:4       1,1,1</span></span>
<span><span class="co">#&gt;      [5]        1         5      + |         GTT         GTT           V           V    3:4:4    3:4:4       1,1,1</span></span>
<span><span class="co">#&gt;      ...      ...       ...    ... .         ...         ...         ...         ...      ...      ...         ...</span></span>
<span><span class="co">#&gt;   [9718]        1      9718      + |         TCA         TCA           S           S    4:2:1    4:2:1       1,1,1</span></span>
<span><span class="co">#&gt;   [9719]        1      9719      + |         ACT         ACT           T           T    1:2:4    1:2:4       1,1,1</span></span>
<span><span class="co">#&gt;   [9720]        1      9720      + |         CAG         CAG           Q           Q    2:1:3    2:1:3       1,1,1</span></span>
<span><span class="co">#&gt;   [9721]        1      9721      + |         GCA         GCA           A           A    3:2:1    3:2:1       1,1,1</span></span>
<span><span class="co">#&gt;   [9722]        1      9722      + |         TAA         TAA           *           *    4:1:1    4:1:1       1,1,1</span></span>
<span><span class="co">#&gt;                 cube</span></span>
<span><span class="co">#&gt;          &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1]        ACGT</span></span>
<span><span class="co">#&gt;      [2]        ACGT</span></span>
<span><span class="co">#&gt;      [3]        ACGT</span></span>
<span><span class="co">#&gt;      [4]        ACGT</span></span>
<span><span class="co">#&gt;      [5]        ACGT</span></span>
<span><span class="co">#&gt;      ...         ...</span></span>
<span><span class="co">#&gt;   [9718]        ACGT</span></span>
<span><span class="co">#&gt;   [9719]        ACGT</span></span>
<span><span class="co">#&gt;   [9720]        ACGT</span></span>
<span><span class="co">#&gt;   [9721]        ACGT</span></span>
<span><span class="co">#&gt;   [9722]        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<div class="section level3">
<h3 id="grouping-automorphism-by-automorphisms-coefficients">Grouping automorphism by automorphism’s coefficients<a class="anchor" aria-label="anchor" href="#grouping-automorphism-by-automorphisms-coefficients"></a>
</h3>
<p>Automorphisms that preserved codons (DNA base-triplets) are
represented by the identity matrix, i.e., the matrix with diagonal
elements “1,1,1”.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">autby_coef_3d</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/automorphism_bycoef.html">automorphism_bycoef</a></span><span class="op">(</span><span class="va">autm_3d</span><span class="op">)</span></span>
<span><span class="va">autby_coef_3d</span> <span class="op">&lt;-</span> <span class="va">autby_coef_3d</span><span class="op">[</span> <span class="va">autby_coef_3d</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="st">"1,1,1"</span> <span class="op">]</span></span>
<span><span class="va">autby_coef_3d</span></span>
<span><span class="co">#&gt; AutomorphismByCoef object with 1083 ranges and 7 metadata columns:</span></span>
<span><span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2         aa1         aa2        autm    mut_type</span></span>
<span><span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1]        1        20      + |         GTC         GTT           V           V       1,1,2         HHY</span></span>
<span><span class="co">#&gt;      [2]        1     45-46      + |         CAC         CAT           H           H       1,1,2         HHY</span></span>
<span><span class="co">#&gt;      [3]        1     45-46      + |         CTC         CTT           L           L       1,1,2         HHY</span></span>
<span><span class="co">#&gt;      [4]        1        53      + |         CTA         TTA           L           L       2,1,1         YHH</span></span>
<span><span class="co">#&gt;      [5]        1        61      + |         CTG         TTG           L           L       2,1,1         YHH</span></span>
<span><span class="co">#&gt;      ...      ...       ...    ... .         ...         ...         ...         ...         ...         ...</span></span>
<span><span class="co">#&gt;   [1079]        1      9609      + |         CCA         CCG           P           P       1,1,3         HHR</span></span>
<span><span class="co">#&gt;   [1080]        1      9615      + |         TTT         TTC           F           F       1,1,3         HHY</span></span>
<span><span class="co">#&gt;   [1081]        1      9649      + |         AAC         AGC           N           S       1,3,1         HRH</span></span>
<span><span class="co">#&gt;   [1082]        1      9675      + |         AAA         AAG           K           K       1,1,3         HHR</span></span>
<span><span class="co">#&gt;   [1083]        1      9682      + |         TTG         TTA           L           L       1,1,2         HHR</span></span>
<span><span class="co">#&gt;                 cube</span></span>
<span><span class="co">#&gt;          &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1]        ACGT</span></span>
<span><span class="co">#&gt;      [2]        ACGT</span></span>
<span><span class="co">#&gt;      [3]        ACGT</span></span>
<span><span class="co">#&gt;      [4]        ACGT</span></span>
<span><span class="co">#&gt;      [5]        ACGT</span></span>
<span><span class="co">#&gt;      ...         ...</span></span>
<span><span class="co">#&gt;   [1079]        ACGT</span></span>
<span><span class="co">#&gt;   [1080]        ACGT</span></span>
<span><span class="co">#&gt;   [1081]        ACGT</span></span>
<span><span class="co">#&gt;   [1082]        ACGT</span></span>
<span><span class="co">#&gt;   [1083]        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>Conserved regions from pairwise comparisons are obtain with function
conserved_regions:</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">conserv</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/conserved_regions.html">conserved_regions</a></span><span class="op">(</span><span class="va">autm_3d</span><span class="op">)</span></span>
<span><span class="va">conserv</span></span>
<span><span class="co">#&gt; AutomorphismByCoef object with 6348 ranges and 7 metadata columns:</span></span>
<span><span class="co">#&gt;          seqnames    ranges strand |        seq1        seq2         aa1         aa2        autm    mut_type</span></span>
<span><span class="co">#&gt;             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1]        1      1-19      + |         ATG         ATG           M           M       1,1,1         HHH</span></span>
<span><span class="co">#&gt;      [2]        1      1-19      + |         GAG         GAG           E           E       1,1,1         HHH</span></span>
<span><span class="co">#&gt;      [3]        1      1-19      + |         AGC         AGC           S           S       1,1,1         HHH</span></span>
<span><span class="co">#&gt;      [4]        1      1-19      + |         CTT         CTT           L           L       1,1,1         HHH</span></span>
<span><span class="co">#&gt;      [5]        1      1-19      + |         GTT         GTT           V           V       1,1,1         HHH</span></span>
<span><span class="co">#&gt;      ...      ...       ...    ... .         ...         ...         ...         ...         ...         ...</span></span>
<span><span class="co">#&gt;   [6344]        1 9683-9722      + |         GGA         GGA           G           G       1,1,1         HHH</span></span>
<span><span class="co">#&gt;   [6345]        1 9683-9722      + |         TCT         TCT           S           S       1,1,1         HHH</span></span>
<span><span class="co">#&gt;   [6346]        1 9683-9722      + |         TCA         TCA           S           S       1,1,1         HHH</span></span>
<span><span class="co">#&gt;   [6347]        1 9683-9722      + |         GCA         GCA           A           A       1,1,1         HHH</span></span>
<span><span class="co">#&gt;   [6348]        1 9683-9722      + |         TAA         TAA           *           *       1,1,1         HHH</span></span>
<span><span class="co">#&gt;                 cube</span></span>
<span><span class="co">#&gt;          &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1]        ACGT</span></span>
<span><span class="co">#&gt;      [2]        ACGT</span></span>
<span><span class="co">#&gt;      [3]        ACGT</span></span>
<span><span class="co">#&gt;      [4]        ACGT</span></span>
<span><span class="co">#&gt;      [5]        ACGT</span></span>
<span><span class="co">#&gt;      ...         ...</span></span>
<span><span class="co">#&gt;   [6344]        ACGT</span></span>
<span><span class="co">#&gt;   [6345]        ACGT</span></span>
<span><span class="co">#&gt;   [6346]        ACGT</span></span>
<span><span class="co">#&gt;   [6347]        ACGT</span></span>
<span><span class="co">#&gt;   [6348]        ACGT</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>The whole genome mutational events represented as automorphisms on
the 3D space <span class="math inline">\(\mathbb{Z}_{5}^3\)</span>,
specifically on the cube ACGT (see <a href="#id_4">4</a>).</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">autby_coef_3d</span><span class="op">$</span><span class="va">cube</span><span class="op">[</span> <span class="va">autby_coef_3d</span><span class="op">$</span><span class="va">autm</span> <span class="op">!=</span> <span class="st">"1,1,1"</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>family <span class="op">=</span> <span class="st">"serif"</span>, cex <span class="op">=</span> <span class="fl">1</span>, font <span class="op">=</span> <span class="fl">2</span>, cex.main <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="va">counts</span>, main<span class="op">=</span><span class="st">"Automorphism distribution"</span>,</span>
<span>        xlab<span class="op">=</span><span class="st">"Genetic-code cube representation"</span>,</span>
<span>        ylab<span class="op">=</span><span class="st">"Fixed mutational events"</span>,</span>
<span>        col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"darkblue"</span>,<span class="st">"red"</span><span class="op">)</span>, </span>
<span>        ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1300</span><span class="op">)</span>, </span>
<span>        border <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html" class="external-link">mtext</a></span><span class="op">(</span>side <span class="op">=</span> <span class="fl">1</span>,line <span class="op">=</span> <span class="op">-</span><span class="fl">2</span>, at <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.7</span>, <span class="fl">1.9</span><span class="op">)</span>,</span>
<span>    text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span> <span class="va">counts</span> <span class="op">)</span>, cex <span class="op">=</span> <span class="fl">1.4</span>,</span>
<span>    col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"white"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="GenomAutomorphism_files/figure-html/barplot_4-1.png" width="288"></p>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<pre><code>1. R. Sanchez. Symmetric Group of the Genetic-Code Cubes.
Effect of the Genetic-Code Architecture on the Evolutionary Process MATCH
Commun. Math. Comput. Chem. 79 (2018) 527-560. 
[PDF](https://bit.ly/2Z9mjM7).

2. Sanchez R, Morgado E, Grau R. Gene algebra from a 
genetic code algebraic structure. J Math Biol. 2005 Oct;51(4):431-57. 
doi:10.1007/s00285-005-0332-8. Epub 2005 Jul 13. PMID: 16012800.
([PDF](https://arxiv.org/pdf/q-bio/0412033.pdf)).

3. Robersy Sanchez, Jesus Barreto (2021) Genomic Abelian
Finite Groups. 
[doi:10.1101/2021.06.01.446543](https://doi.org/10.1101/2021.06.01.446543)

4. M. V Jose, E.R. Morgado, R. Sanchez, T. Govezensky,
The 24 possible algebraic representations of the standard genetic code in
six or in three dimensions, Adv. Stud. Biol. 4 (2012) 119-152.
[PDF](https://is.gd/na9eap).

5. Crick FHC. The Origin of the Genetic Code. J Mol Biol.
1968;38: 367–379.</code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code> on the system on
which this document was compiled:</p>
<pre><code><span><span class="co">#&gt; R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/New_York</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] GenomAutomorphism_1.5.1 Biostrings_2.71.6       GenomeInfoDb_1.39.14    XVector_0.43.1         </span></span>
<span><span class="co">#&gt; [5] IRanges_2.37.1          S4Vectors_0.41.7        BiocGenerics_0.49.1     BiocStyle_2.31.0       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] utf8_1.2.4              sass_0.4.9              generics_0.1.3          numbers_0.8-5          </span></span>
<span><span class="co">#&gt;  [5] digest_0.6.35           magrittr_2.0.3          evaluate_0.23           bookdown_0.39          </span></span>
<span><span class="co">#&gt;  [9] iterators_1.0.14        fastmap_1.1.1           foreach_1.5.2           doParallel_1.0.17      </span></span>
<span><span class="co">#&gt; [13] jsonlite_1.8.8          BiocManager_1.30.22     httr_1.4.7              fansi_1.0.6            </span></span>
<span><span class="co">#&gt; [17] purrr_1.0.2             UCSC.utils_0.99.7       codetools_0.2-20        textshaping_0.3.7      </span></span>
<span><span class="co">#&gt; [21] jquerylib_0.1.4         cli_3.6.2               rlang_1.1.3             crayon_1.5.2           </span></span>
<span><span class="co">#&gt; [25] cachem_1.0.8            yaml_2.3.8              tools_4.4.0             parallel_4.4.0         </span></span>
<span><span class="co">#&gt; [29] BiocParallel_1.37.1     memoise_2.0.1           dplyr_1.1.4             GenomeInfoDbData_1.2.12</span></span>
<span><span class="co">#&gt; [33] vctrs_0.6.5             R6_2.5.1                matrixStats_1.3.0       lifecycle_1.0.4        </span></span>
<span><span class="co">#&gt; [37] zlibbioc_1.49.3         fs_1.6.4                htmlwidgets_1.6.4       ragg_1.3.0             </span></span>
<span><span class="co">#&gt; [41] pkgconfig_2.0.3         desc_1.4.3              pillar_1.9.0            pkgdown_2.0.9          </span></span>
<span><span class="co">#&gt; [45] bslib_0.7.0             data.table_1.15.4       glue_1.7.0              systemfonts_1.0.5      </span></span>
<span><span class="co">#&gt; [49] highr_0.10              tidyselect_1.2.1        tibble_3.2.1            xfun_0.43              </span></span>
<span><span class="co">#&gt; [53] GenomicRanges_1.55.4    rstudioapi_0.16.0       knitr_1.46              htmltools_0.5.8.1      </span></span>
<span><span class="co">#&gt; [57] rmarkdown_2.26          compiler_4.4.0</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
