<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Beta binomial posteriors</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="/media/sf_D/<a href='https://genomaths.github.io/MethylIT_HTML_Manual/MethylIT_Manual.html'>MethylIT</a>_HTML_Manual/R.css" />
</head><body>

<table width="100%" summary="page for .betaBinPosteriors {<a href='https://genomaths.github.io/MethylIT_HTML_Manual/MethylIT_Manual.html'>MethylIT</a>}"><tr><td>.betaBinPosteriors {<a href='https://genomaths.github.io/MethylIT_HTML_Manual/MethylIT_Manual.html'>MethylIT</a>}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Beta binomial posteriors</h2>

<h3>Description</h3>

<p>Here, beta binomial posteriors are estimated in the context of
the methylation level estimation. The direct use of counts for
methylation level estimation is not recommended, since there is external
source of noise coming from the sample manipulation, the sequencing
machine and sequence alignment that could alter the counts and the
number of methylated cytosines with low or zero counts (mC = 0) is
generally high. The low counts and noise imply that read counts of
methylated (mC) and non-methylated (uC) cytosines must not be used
directly in the estimation of methylation levels.
</p>
<p>In a Bayesian framework, methylated read counts are modeled by a beta-
binomial distribution, which accounts for both, the biological and
sampling variations [1-3]. In our case we adopted the Bayesian approach
suggested in reference [3](Chapter 3). Naive distribution q
(methylation levels). In a Bayesian framework with uniform priors, the
methylation level can be defined as:
meth_level = ( mC + 1 )/( mC + uC + 2 ). However, the most
natural statistical model for replicated BS-seq DNA methylation
measurements is beta-binomial (the beta distribution is a prior
conjugate of binomial distribution), we consider the p parameter
(methylation level) in the binomial distribution as randomly drawn from
a beta distribution. The hyper-parameters alpha ('a') and beta ('b')
from the beta-binomial distribution are interpreted as pseudo-counts.
</p>


<h3>Usage</h3>

<pre>
.betaBinPosteriors(success, trials, a, b)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>success</code></td>
<td>
<p>number of successful events</p>
</td></tr>
<tr valign="top"><td><code>trials</code></td>
<td>
<p>total number of events</p>
</td></tr>
<tr valign="top"><td><code>a</code></td>
<td>
<p>previous number of successful events</p>
</td></tr>
<tr valign="top"><td><code>b</code></td>
<td>
<p>previous number of unsuccessful events</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The posterior methylation levels are estimated as
(a + success)/(a + b + trials), where 'a' and 'b' are the shape
parameters of the beta distribution, alpha and beta parameters,
respectively.
</p>


<h3>Value</h3>

<p>a probability
</p>


<h3>References</h3>

<p>1. Hebestreit K, Dugas M, Klein H-U (2013) Detection of
significantly differentially methylated regions in targeted bisulfite
sequencing data. Bioinformatics 29: 1647-1653. Available:
http://www.ncbi.nlm.nih.gov/pubmed/23658421. Accessed 4 February 2014.
2. Robinson MD, Kahraman A, Law CW, Lindsay H, Nowicka M, et al. (2014)
Statistical methods for detecting differentially methylated loci and
regions. Front Genet 5: 324. Available:
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4165320/ Accessed 26 August
2015. 3. Baldi P, Brunak S (2001) Bioinformatics: the machine learning
approach. Second. Cambridge: MIT Press. 452 p.
</p>


<h3>Examples</h3>

<pre>
    <a href='https://genomaths.github.io/MethylIT_HTML_Manual/MethylIT_Manual.html'>MethylIT</a>:::.betaBinPosteriors(2, 8, 2, 8)

</pre>

<hr /><div style="text-align: center;">[Package <em><a href='https://genomaths.github.io/MethylIT_HTML_Manual/MethylIT_Manual.html'>MethylIT</a></em> version 0.3.1 ]</div>
</body></html>
