<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Evaluate DIMPs Classification</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="/media/sf_D/<a href='https://genomaths.github.io/MethylIT_HTML_Manual/MethylIT_Manual.html'>MethylIT</a>_HTML_Manual/R.css" />
</head><body>

<table width="100%" summary="page for evaluateDIMPclass {<a href='https://genomaths.github.io/MethylIT_HTML_Manual/MethylIT_Manual.html'>MethylIT</a>}"><tr><td>evaluateDIMPclass {<a href='https://genomaths.github.io/MethylIT_HTML_Manual/MethylIT_Manual.html'>MethylIT</a>}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Evaluate DIMPs Classification</h2>

<h3>Description</h3>

<p>For a given cutpoint (previously estimated with the function
estimateCutPoint), 'evaluateDIMPclass' will return the evaluation of the
classification of DIMPs into two clases: DIMPS from control and DIMPs
from treatment samples.
</p>


<h3>Usage</h3>

<pre>
evaluateDIMPclass(LR, control.names, treatment.names, column = c(hdiv =
  FALSE, TV = FALSE, wprob = FALSE, pos = FALSE),
  classifier = c("logistic", "pca.logistic", "lda", "qda", "pca.lda",
  "pca.qda"), pval.col = NULL, n.pc = 1, center = FALSE,
  scale = FALSE, interaction = NULL, output = "conf.mat",
  prop = 0.6, num.boot = 100, num.cores = 1L, tasks = 0L,
  seed = 1234, verbose = TRUE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>LR</code></td>
<td>
<p>A list of GRanges objects (LR) including control and treatment
GRanges containing divergence values for each DIMP in the meta-column.
LR is generated by the function 'selectDIMP' Each GRanges object must
correspond to a sample. For example, if  a sample is named 's1', then
this sample can be accessed in the list of GRanges objects as LR$s1.</p>
</td></tr>
<tr valign="top"><td><code>control.names</code></td>
<td>
<p>Names/IDs of the control samples, which must be include
in thr variable LR.</p>
</td></tr>
<tr valign="top"><td><code>treatment.names</code></td>
<td>
<p>Names/IDs of the treatment samples, which must be
included in the variable LR.</p>
</td></tr>
<tr valign="top"><td><code>column</code></td>
<td>
<p>a logical vector for column names for the predictor variables
to be used: Hellinger divergence &quot;hdiv&quot;, total variation &quot;TV&quot;,
probability of potential DIMP &quot;wprob&quot;, and the relative cytosine site
position &quot;pos&quot; in respect to the chromosome where it is located. The
relative position is estimated as (x - x.min)/(x.max - x), where x.min
and x.max are the maximum and minimum for the corresponding chromosome,
repectively. If &quot;wprob = TRUE&quot;, then Logarithm base-10 of &quot;wprob&quot; will
be used as predictor in place of &quot;wprob&quot;.</p>
</td></tr>
<tr valign="top"><td><code>classifier</code></td>
<td>
<p>Classification model to use. Option &quot;logistic&quot; applies a
logistic regression model; option &quot;lda&quot; applies a Linear Discriminant
Analysis (LDA); &quot;qda&quot; applies a Quadratic Discriminant Analysis (QDA),
&quot;pca.logistic&quot; applies logistic regression model using the Principal
Component (PCs) estimated with Principal Component Analysis (PCA) as
predictor variables. pca.lda&quot; applies LDA using PCs as predictor
variables, and the option &quot;pca.qda&quot; applies a Quadratic Discriminant
Analysis (QDA) using PCs as predictor variables.</p>
</td></tr>
<tr valign="top"><td><code>pval.col</code></td>
<td>
<p>Column number for p-value used in the performance
analysis and estimation of the cutpoints. Default: NULL. If NULL it is
assumed that the column is named &quot;wprob&quot;.</p>
</td></tr>
<tr valign="top"><td><code>n.pc</code></td>
<td>
<p>Number of principal components (PCs) to use if the classifier is
not 'logistic'. In the current case, the maximun number of PCs is 4.</p>
</td></tr>
<tr valign="top"><td><code>center</code></td>
<td>
<p>A logical value indicating whether the variables should be
shifted to be zero centered (same as in 'prcomp' prcomp). Only used if
classifier = &quot;pcaLDA&quot;.</p>
</td></tr>
<tr valign="top"><td><code>scale</code></td>
<td>
<p>A logical value indicating whether the variables should be
scaled to have unit variance before the analysis takes place (same as in
'prcomp' prcomp). Only used if classifier = &quot;pcaLDA&quot;.</p>
</td></tr>
<tr valign="top"><td><code>interaction</code></td>
<td>
<p>Variable interactions to consider in a logistic
regression model. Any pairwise combination of the variable &quot;hdiv&quot;, &quot;TV&quot;,
&quot;wprob&quot;, and &quot;pos&quot; can be provided. For example: &quot;hdiv:TV&quot;, &quot;wprob:pos&quot;,
&quot;wprob:TV&quot;, etc.</p>
</td></tr>
<tr valign="top"><td><code>output</code></td>
<td>
<p>Type of output to request: output = c(&quot;conf.mat&quot;, &quot;mc.val&quot;,
&quot;boot.all&quot;, &quot;all&quot;). See below.</p>
</td></tr>
<tr valign="top"><td><code>prop</code></td>
<td>
<p>Proportion to split the dataset used in the logistic regression:
group versus divergence (at DIMPs) into two subsets, training and
testing.</p>
</td></tr>
<tr valign="top"><td><code>num.boot</code></td>
<td>
<p>Number of bootstrap validations to perform in the evaluation
of the logistic regression: group versus divergence (at DIMPs).</p>
</td></tr>
<tr valign="top"><td><code>num.cores, tasks</code></td>
<td>
<p>Paramaters for parallele computation using package
<code>BiocParallel-package</code>: the number of cores to
use, i.e. at most how many child processes will be run simultaneously
(see <code>bplapply</code> and the number of tasks per job
(only for Linux OS).</p>
</td></tr>
<tr valign="top"><td><code>seed</code></td>
<td>
<p>Random seed used for random number generation.</p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>if TRUE, prints the function log to stdout</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The regulatory methylation signal is also an output from a natural
process that continuously takes place across the ontogenetic development
of the organisms. So, we expect to see methylation signal on natural
ordinary conditions. Here, to distinguish a control methylation signal
from a treatment, three classification models are provided: 1) logistic,
2) Linear Discriminant Analysis (LDA) and 3) Quadratic Discriminant
Analysis (QDA). In particular, four predictor variables can be used:
Hellinger divergence &quot;hdiv&quot;, total variation &quot;TV&quot;, probability of
potential DIMP &quot;wprob&quot; and DIMP genomic coordinated &quot;pos&quot;. Principal
component analysis (PCA) is used to convert a set of observations of
possibly correlated predictor variables into a set of values of linearly
uncorrelated variables (principal components, PCs). The PCs are used as
new, uncorrelated predictor variables for LDA, QDA, and logistic
classifiers.
</p>
<p>A classification result with low accuracy and compromising values from
other classification performance indicators (see below) suggest that the
treatment does not induce a significant regulatory signal different
from control.
</p>


<h3>Value</h3>

<p>output = &quot;conf.mat&quot; will perform a logistic regression group versus
divergence (at DIMPs) to evaluate the discrimination between
control-DIMPs and treatment-DIMPs. The evaluation of this classification
is provided through the function 'confusionMatrix' from R package
'caret'. &quot;mc.val&quot; will perform a 'num.boot'-times Monte Carlo
(bootstrap) validation and return a summary. By default function
'confusionMatrix' from R package caret' randomly splits the sample into
two subsets, training and testing, according to the supplied proportion
'prop' (i.e., prop = 0.6). After selecting output = &quot;mc.val&quot;, the
function 'confusionMatrix' will be executed 'num.boot'-times, each time
performing a different random split of the sample. &quot;boot.all&quot; same as
&quot;mc.val&quot; plus a matrix with statistcs reported by 'confusionMatrix'.
&quot;all&quot; return a list with all the mentioned outputs.
</p>


<h3>Examples</h3>

<pre>
set.seed(123) ## To set a seed for random number generation
## GRanges object of the reference with methylation levels in
## its matacolumn
num.points &lt;- 5000
Ref &lt;- makeGRangesFromDataFrame(
    data.frame(chr = '1',
            start = 1:num.points,
            end = 1:num.points,
            strand = '*',
            p1 = rbeta(num.points, shape1 = 1, shape2 = 1.5)),
    keep.extra.columns = TRUE)

## List of Granges objects of individuals methylation levels
Indiv &lt;- GRangesList(
    sample11 = makeGRangesFromDataFrame(
        data.frame(chr = '1',
            start = 1:num.points,
            end = 1:num.points,
            strand = '*',
            p2 = rbeta(num.points, shape1 = 1.5, shape2 = 2)),
         keep.extra.columns = TRUE),
    sample12 = makeGRangesFromDataFrame(
        data.frame(chr = '1',
            start = 1:num.points,
            end = 1:num.points,
            strand = '*',
            p2 = rbeta(num.points, shape1 = 1.6, shape2 = 2)),
        keep.extra.columns = TRUE),
    sample21 = makeGRangesFromDataFrame(
        data.frame(chr = '1',
            start = 1:num.points,
            end = 1:num.points,
            strand = '*',
            p2 = rbeta(num.points, shape1 = 40, shape2 = 4)),
        keep.extra.columns = TRUE),
    sample22 = makeGRangesFromDataFrame(
        data.frame(chr = '1',
            start = 1:num.points,
            end = 1:num.points,
            strand = '*',
            p2 = rbeta(num.points, shape1 = 41, shape2 = 4)),
        keep.extra.columns = TRUE))
## To estimate Hellinger divergence using only the methylation levels.
HD &lt;- estimateDivergence(ref = Ref, indiv = Indiv, meth.level = TRUE,
                            columns = 1)
## To perform the nonlinear regression analysis
nlms &lt;- nonlinearFitDist(HD, column = 4, verbose = FALSE)

## Next, the potential signal can be estimated
PS &lt;- getPotentialDIMP(LR = HD, nlms = nlms, div.col = 4, alpha = 0.05)

## The cutpoint estimation used to discriminate the signal from the noise
cutpoints &lt;- estimateCutPoint(PS, control.names = c("sample11", "sample12"),
                            treatment.names = c("sample21", "sample22"),
                            div.col = 4, verbose = TRUE)
## DIMPs are selected using the cupoints
DIMPs &lt;- selectDIMP(PS, div.col = 4, cutpoint = min(cutpoints$cutpoint))

## Classification of DIMPs into two clases: DIMPS from control and DIMPs from
## treatment samples and evaluation of the classifier performance (for more
## details see ?evaluateDIMPclass).
conf.mat &lt;- evaluateDIMPclass(DIMPs,
                            column = c(hdiv = TRUE, TV = TRUE,
                            wprob = TRUE, pos = FALSE),
                            control.names = c("sample11", "sample12"),
                            treatment.names = c("sample21", "sample22"))
confusion.matrix &lt;- conf.mat$conf.mat
model.fit &lt;- summary(conf.mat$model)
</pre>

<hr /><div style="text-align: center;">[Package <em><a href='https://genomaths.github.io/MethylIT_HTML_Manual/MethylIT_Manual.html'>MethylIT</a></em> version 0.3.1 ]</div>
</body></html>
