<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Selection of DMPs</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="/media/sf_D/<a href='https://genomaths.github.io/MethylIT_HTML_Manual/MethylIT_Manual.html'>MethylIT</a>_HTML_Manual/R.css" />
</head><body>

<table width="100%" summary="page for selectDIMP {<a href='https://genomaths.github.io/MethylIT_HTML_Manual/MethylIT_Manual.html'>MethylIT</a>}"><tr><td>selectDIMP {<a href='https://genomaths.github.io/MethylIT_HTML_Manual/MethylIT_Manual.html'>MethylIT</a>}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Selection of DMPs</h2>

<h3>Description</h3>

<p>For a given cutpoint (previously estimated with the function
estimateCutPoint), 'selectDIMP' will return the differentially
informative methylated positions (DMPs). DMPs are cytosine positions
for which the divergence is greater than the cutpoint.
</p>


<h3>Usage</h3>

<pre>
selectDIMP(LR, div.col = NULL, pval.col = NULL, absolute = FALSE,
  cutpoint, tv.col = NULL, tv.cut = NULL)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>LR</code></td>
<td>
<p>An object from &quot;pDMP&quot; class.</p>
</td></tr>
<tr valign="top"><td><code>div.col</code></td>
<td>
<p>Number of the column where the divergence variable (i.e.,
Hellinger divergence or total variation) is located in the GRanges
meta-columns.</p>
</td></tr>
<tr valign="top"><td><code>pval.col</code></td>
<td>
<p>If the cutpoints is a p-value, then the column number for
p-values should be provided. Default: NULL. Notice that one of the
parameter values div.col or pval.col must be given.</p>
</td></tr>
<tr valign="top"><td><code>absolute</code></td>
<td>
<p>Logic (default, FALSE). Total variation (TV, the difference
of methylation levels) is normally an output in the downstream <a href='https://genomaths.github.io/MethylIT_HTML_Manual/MethylIT_Manual.html'>MethylIT</a>
analysis. If 'absolute = TRUE', then TV is tranformed into |TV|, which is
an information divergence that can be fitted to Weibull or to Generalized
Gamma distribution. So, if the nonlinear fit was performed for |TV|, then
here absolute must be set to TRUE.</p>
</td></tr>
<tr valign="top"><td><code>cutpoint</code></td>
<td>
<p>Cutpoint to select DMPs. Cytosine positions with divergence
greater than 'cutpoint' will selected as DMPs. Cutpoints are estimated
with the function 'estimateCutPoint'.</p>
</td></tr>
<tr valign="top"><td><code>tv.col</code></td>
<td>
<p>Column number for the total variation to be used for filtering
cytosine positions (if provided).</p>
</td></tr>
<tr valign="top"><td><code>tv.cut</code></td>
<td>
<p>If tv.cut and tv.col are provided, then cytosine sites k with
abs(TV_k) &lt; tv.cut are removed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Theoretically a DMP denotes a cytosine position with high
probability to be differentially methylated. That is, in the statistical
molecular-biophysics context, a DMP must be considered only in a
probabilistic term and not as an absolute deterministic experimental
output.
</p>
<p>The uncertainty and dynamics of the DNA methylation process, the
continuous action of the omnipresent thermal fluctuations, as well as,
the inherent stochasticity of the biochemical reactions make it
impossible to ensure whether a specific cytosine position is methylated
in an absolutely deterministic sense. Notice that the concept of DMP is
not applicable to a single cell (if we use an instrumentation/protocol
to directly measure methylation at the molecular level, and not via PCR),
since a concrete, single DNA cytosine position in a single cell is
methylated or not methylated.
</p>
<p>However, when pooling DNA extracted from a tissue, the previous
reasonings about uncertainty hold plus an additional uncertainty factor:
cells from the same tissue are not synchronized but are found in
different stages of their ontogenetic developments. Hence, the DMP
concept holds in the mentioned circumstances where the uncertainty of
methylation is present.
</p>


<h3>Value</h3>

<p>An object from &quot;pDMP&quot; class containing only differentially
informative position (DMPs).
</p>


<h3>Examples</h3>

<pre>
## Get a dataset of potential signals and the estimated cutpoint from the
## package
data(PS, cutpoint)

## The estimated cutpoints are used to discriminate signals from the noise.
## That is, DMPs are selected using the cupoints
DMPs &lt;- selectDIMP(PS, div.col = 9L, cutpoint = cutpoint$cutpoint)

</pre>

<hr /><div style="text-align: center;">[Package <em><a href='https://genomaths.github.io/MethylIT_HTML_Manual/MethylIT_Manual.html'>MethylIT</a></em> version 0.3.1 ]</div>
</body></html>
