<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>What the Package Does (One Line, Title Case) • GenomAutomorphism</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="What the Package Does (One Line, Title Case)">
<meta property="og:description" content="This is a R package to compute the autimorphisms between pairwise
             aligned DNA sequences represented as elements from an Abelian
             group.">
<meta property="og:image" content="https://genomaths.github.io/GenomAutomorphism/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">GenomAutomorphism</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/GenomAutomorphism.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Package functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/GenomAutomorphism.html">GenomAutomorphism</a>
    </li>
    <li class="divider">
    <li>
      <a href="articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a>
    </li>
    <li class="divider">
    <li>
      <a href="articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a>
    </li>
    <li class="divider">
    <li>
      <a href="articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a>
    </li>
    <li class="divider">
    <li>
      <a href="articles/covid_19.html">Automorphism analysis on COVID-19 data</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/genomaths/GenomAutomorphism/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="http://twitter.com/robersysr">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="genomautomorphism-" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#genomautomorphism-" class="anchor"></a>GenomAutomorphism <a href="https://genomaths.github.io/genomautomorphism"><img src="reference/figures/logo.png" align="right"></a>
</h1></div>
<p>Robersy Sanchez<br>
Department of Biology. Eberly College of Science.<br>
Pennsylvania State University, University Park, PA 16802<br><a href="mailto:rus547@psu.edu" class="email">rus547@psu.edu</a><br><a href="https://orcid.org/0000-0002-5246-1453">ORCID: orcid.org/0000-0002-5246-1453</a></p>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>This is a R package to compute the autimorphisms between pairwise aligned DNA sequences represented as elements from a Genomic Abelian group as described in the paper <a href="https://www.biorxiv.org/content/10.1101/2021.06.01.446543v2">Genomic Abelian Finite Groups</a>. In a general scenario, whole chromosomes or genomic regions from a population (from any species or close related species) can be algebraically represented as a direct sum of cyclic groups or more specifically Abelian <em>p</em>-groups. Basically, we propose the representation of multiple sequence alignments (MSA) of length <em>N</em> as a finite Abelian group created by the direct sum of homocyclic Abelian group of <em>prime-power order</em>:</p>
<p>   <em>G</em> = (ℤ<sub><em>p</em><sub>1</sub><sup><em>α</em><sub>1</sub></sup></sub>)<sup><em>n</em><sub>1</sub></sup> ⊕ (ℤ<sub><em>p</em><sub>1</sub><sup><em>α</em><sub>2</sub></sup></sub>)<sup><em>n</em><sub>2</sub></sup> ⊕ … ⊕ (ℤ<sub><em>p</em><sub><em>k</em></sub><sup><em>α</em><sub><em>k</em></sub></sup></sub>)<sup><em>n</em><sub><em>k</em></sub></sup></p>
<p>Where, the <em>p</em><sub><em>i</em></sub>’s are prime numbers, <em>α</em><sub><em>i</em></sub> ∈ ℕ and ℤ<sub><em>p</em><sub><em>i</em></sub><sup><em>α</em><sub><em>i</em></sub></sup></sub> is the group of integer modulo <em>p</em><sub><em>i</em></sub><sup><em>α</em><sub><em>i</em></sub></sup>.</p>
<p>For the purpose of automorphism between two aligned DNA sequences, <em>p</em><sub><em>i</em></sub><sup><em>α</em><sub><em>i</em></sub></sup> ∈ {5, 2<sup>6</sup>, 5<sup>3</sup>}.</p>
<hr>
</div>
<div id="status" class="section level2">
<h2 class="hasAnchor">
<a href="#status" class="anchor"></a>Status</h2>
<p>This application is under development. Watch this repo or check for updates.</p>
<hr>
</div>
<div id="tutotials" class="section level2">
<h2 class="hasAnchor">
<a href="#tutotials" class="anchor"></a>Tutotials</h2>
<p>There are several tutorials on how to use the package at <a href="https://genomaths.github.io/genomautomorphism">GenomAutomorphism</a> website <a href="https://genomaths.github.io/genomautomorphism"><img src="reference/figures/logo.png" align="middle" width="32" height="32"></a></p>
</div>
<div id="dependences" class="section level2">
<h2 class="hasAnchor">
<a href="#dependences" class="anchor"></a>Dependences</h2>
<p>This package depends, so far, from: <em>Biostrings</em>, <em>GenomicRanges</em>, <em>numbers</em>, and <em>S4Vectors</em>.</p>
<hr>
</div>
<div id="installation-of-r-dependencies" class="section level2">
<h2 class="hasAnchor">
<a href="#installation-of-r-dependencies" class="anchor"></a>Installation of R dependencies:</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>)) <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>    BiocManager<span class="sc">::</span><span class="fu">install</span>()</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>    BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="fu">c</span>(<span class="st">"Biostrings"</span>, <span class="st">"GenomicRanges"</span>, <span class="st">"S4Vectors"</span>))</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="fu">c</span>(<span class="st">"numbers"</span>, <span class="st">"devtools"</span>), <span class="at">dependencies=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<hr>
</div>
<div id="you-can-install-genomautomorphism-package-from-github" class="section level2">
<h2 class="hasAnchor">
<a href="#you-can-install-genomautomorphism-package-from-github" class="anchor"></a>You can install <strong>GenomAutomorphism</strong> package from GitHub</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>   devtools<span class="sc">::</span><span class="fu">install_git</span>(<span class="st">"https://github.com/genomaths/GenomAutomorphism.git"</span>)</span></code></pre></div>
<hr>
</div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<ol>
<li><p>Sanchez R, Morgado E, Grau R. Gene algebra from a genetic code algebraic structure. J Math Biol. 2005 Oct;51(4):431-57. doi: 10.1007/s00285-005-0332-8. Epub 2005 Jul 13. PMID: 16012800. ( <a href="https://arxiv.org/pdf/q-bio/0412033.pdf">PDF</a>).</p></li>
<li><p>Sanchez R, Grau R, Morgado E. A novel Lie algebra of the genetic code over the Galois field of four DNA bases. Math Biosci. 2006;202: 156–174. <a href="doi:10.1016/j.mbs.2006.03.017" class="uri">doi:10.1016/j.mbs.2006.03.017</a></p></li>
<li><p>Sanchez R, Grau R. An algebraic hypothesis about the primeval genetic code architecture. Math Biosci. 2009/07/18. 2009;221: 60–76. <a href="doi:S0025-5564(09)00114-X" class="uri">doi:S0025-5564(09)00114-X</a> <span class="math display"><em>p</em><em>i</em><em>i</em></span> 10.1016/j.mbs.2009.07.001</p></li>
<li><p>Robersy Sanchez, Jesús Barreto (2021) Genomic Abelian Finite Groups. <a href="https://doi.org/10.1101/2021.06.01.446543">doi: 10.1101/2021.06.01.446543</a>.</p></li>
<li><p>M. V José, E.R. Morgado, R. Sánchez, T. Govezensky, The 24 possible algebraic representations of the standard genetic code in six or in three dimensions, Adv. Stud. Biol. 4 (2012) 119–152.<a href="https://is.gd/na9eap">PDF</a>.</p></li>
<li><p>R. Sanchez. Symmetric Group of the Genetic–Code Cubes. Effect of the Genetic–Code Architecture on the Evolutionary Process MATCH Commun. Math. Comput. Chem. 79 (2018) 527-560. <a href="https://bit.ly/2Z9mjM7">PDF</a>.</p></li>
</ol>
<div id="see-also" class="section level2">
<h2 class="hasAnchor">
<a href="#see-also" class="anchor"></a>See also</h2>
<p><a href="https://github.com/genomaths/GenomeAlgebra_SymmetricGroup">Symmetric Group of the Genetic-Code Cubes</a></p>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/genomaths/GenomAutomorphism/">https://​github.com/​genomaths/​GenomAutomorphism/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/genomaths/GenomAutomorphism/issues">https://​github.com/​genomaths/​GenomAutomorphism/​issues</a>
</li>
<li>Author's website at <br><a href="https://genomaths.com">https://​genomaths.com</a>
</li>
<li>Affiliation (<img src="https://git.psu.edu/genomath/MethylIT_data/raw/master/penn-state-lion.png" height="24"> PennState) at <br><a href="https://bio.psu.edu/directory/rus547">https://​bio.psu.edu/​directory/​rus547</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://genomaths.com">Robersy Sanchez</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-5246-1453" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://genomaths.com">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
