<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Compute the automorphisms between DNA's Abelian group representations • GenomAutomorphism</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Compute the automorphisms between DNA's Abelian group representations">
<meta name="description" content="This is a R package to compute the automorphisms between pairwise aligned DNA sequences represented as elements from a Genomic Abelian group. In a general scenario, from genomic regions till the whole genomes from a given population (from any species or close related species) can be algebraically represented as a direct sum of cyclic groups or more specifically Abelian p-groups. Basically, we propose the representation of multiple sequence alignments of length N bp as element of a finite Abelian group created by the direct sum of homocyclic Abelian group of prime-power order.">
<meta property="og:description" content="This is a R package to compute the automorphisms between pairwise aligned DNA sequences represented as elements from a Genomic Abelian group. In a general scenario, from genomic regions till the whole genomes from a given population (from any species or close related species) can be algebraically represented as a direct sum of cyclic groups or more specifically Abelian p-groups. Basically, we propose the representation of multiple sequence alignments of length N bp as element of a finite Abelian group created by the direct sum of homocyclic Abelian group of prime-power order.">
<meta property="og:image" content="https://genomaths.github.io/genomautomorphism/logo.png">
<!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-115082821-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115082821-1');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">GenomAutomorphism</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.8.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="index.html"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="articles/GenomAutomorphism.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Package functions</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/GenomAutomorphism.html">GenomAutomorphism</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="articles/automorphism_on_msa.html">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="articles/automorphism_on_msa_brca1.html">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="articles/automorphism_and_decision_tree.html">A Short Introduction to Algebraic Taxonomy on Genes Regions</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="articles/automorphism_and_decision_tree_II.html">A Short Introduction to Algebraic Taxonomy on Genes Regions II</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="articles/covid_19.html">Automorphism analysis on COVID-19 data</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="articles/Modular_Matrix_Operation_of_Mutational_Events.html">Modular Matrix Operations of Mutational Events</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="news/index.html">News</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/genomaths/GenomAutomorphism/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="genomautomorphism-">GenomAutomorphism 
<a class="anchor" aria-label="anchor" href="#genomautomorphism-"></a>
</h1>
</div>
<p>Robersy Sanchez<br>
Department of Biology. Eberly College of Science.<br>
Pennsylvania State University, University Park, PA 16802<br><a href="mailto:genomicmath@gmail.com" class="email">genomicmath@gmail.com</a><br><a href="https://orcid.org/0000-0002-5246-1453" class="external-link">ORCID: orcid.org/0000-0002-5246-1453</a></p>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>This repository contains a ‘beta’ version of the <a href="https://github.com/genomaths/GenomAutomorphism/" class="external-link">GenomAutomorphism</a> R package.</p>
<p>This is a R package to compute the automorphisms between pairwise aligned DNA sequences represented as elements from a Genomic Abelian group as described in the paper <a href="https://www.biorxiv.org/content/10.1101/2021.06.01.446543v2" class="external-link">Genomic Abelian Finite Groups</a>. In a general scenario, whole chromosomes or genomic regions from a population (from any species or close related species) can be algebraically represented as a direct sum of cyclic groups or more specifically Abelian <em>p</em>-groups. Basically, we propose the representation of multiple sequence alignments (MSA) of length <em>N</em> as a finite Abelian group created by the direct sum of homocyclic Abelian group of <em>prime-power order</em>:</p>
<p>   <em>G</em> = (ℤ<sub><em>p</em><sub>1</sub><sup><em>α</em><sub>1</sub></sup></sub>)<sup><em>n</em><sub>1</sub></sup> ⊕ (ℤ<sub><em>p</em><sub>1</sub><sup><em>α</em><sub>2</sub></sup></sub>)<sup><em>n</em><sub>2</sub></sup> ⊕ … ⊕ (ℤ<sub><em>p</em><sub><em>k</em></sub><sup><em>α</em><sub><em>k</em></sub></sup></sub>)<sup><em>n</em><sub><em>k</em></sub></sup></p>
<p>Where, the <em>p</em><sub><em>i</em></sub>’s are prime numbers, <em>α</em><sub><em>i</em></sub> ∈ ℕ and ℤ<sub><em>p</em><sub><em>i</em></sub><sup><em>α</em><sub><em>i</em></sub></sup></sub> is the group of integer modulo <em>p</em><sub><em>i</em></sub><sup><em>α</em><sub><em>i</em></sub></sup>.</p>
<p>For the purpose of automorphism between two aligned DNA sequences, <em>p</em><sub><em>i</em></sub><sup><em>α</em><sub><em>i</em></sub></sup> ∈ {5, 2<sup>6</sup>, 5<sup>3</sup>}.</p>
<hr>
</div>
<div class="section level2">
<h2 id="status">Status<a class="anchor" aria-label="anchor" href="#status"></a>
</h2>
<p>This application is currently available in Bioconductor (version 3.18) <a href="https://doi.org/doi:10.18129/B9.bioc.GenomAutomorphism" class="external-link uri">https://doi.org/doi:10.18129/B9.bioc.GenomAutomorphism</a>. Watch this repo or check for updates.</p>
<hr>
</div>
<div class="section level2">
<h2 id="tutorials">Tutorials<a class="anchor" aria-label="anchor" href="#tutorials"></a>
</h2>
<p>There are several tutorials on how to use the package at <a href="https://genomaths.github.io/genomautomorphism">GenomAutomorphism</a> website <a href="https://genomaths.github.io/genomautomorphism"><img src="reference/figures/logo.png" align="middle" width="32" height="32"></a></p>
<ul>
<li><a href="https://is.gd/zwRaUw" target="_blank" rel="noopener" class="external-link">Get started-with GenomAutomorphism</a></li>
<li><a href="https://is.gd/A03Fkl" target="_blank" rel="noopener" class="external-link">Analysis of Automorphisms on a DNA Multiple Sequence Alignment</a></li>
<li><a href="https://is.gd/gsZcuj" target="_blank" rel="noopener" class="external-link">Analysis of Automorphisms on a MSA of Primate BRCA1 Gene</a></li>
<li><a href="https://is.gd/87wlbL" target="_blank" rel="noopener" class="external-link">A Short Introduction to Algebraic Taxonomy on Genes Regions</a></li>
<li><a href="https://is.gd/836uas" target="_blank" rel="noopener" class="external-link">Automorphism analysis on COVID-19 data</a></li>
<li><a href="https://is.gd/wlyzhr" target="_blank" rel="noopener" class="external-link">Modular Matrix Operations of Mutational Events</a></li>
</ul>
</div>
<div class="section level2">
<h2 id="dependences">Dependences<a class="anchor" aria-label="anchor" href="#dependences"></a>
</h2>
<p>This package depends, so far, from: <em>Biostrings</em>, <em>GenomicRanges</em>, <em>numbers</em>, and <em>S4Vectors</em>.</p>
<hr>
</div>
<div class="section level2">
<h2 id="installation-of-r-dependencies">Installation of R dependencies:<a class="anchor" aria-label="anchor" href="#installation-of-r-dependencies"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>)) <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>    BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="fu">c</span>(<span class="st">"Biostrings"</span>, <span class="st">"GenomicRanges"</span>, <span class="st">"S4Vectors"</span>,</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>    <span class="st">"BiocParallel"</span>, <span class="st">"GenomeInfoDb"</span>, <span class="st">"BiocGenerics"</span>, <span class="st">"numbers"</span>, <span class="st">"devtools"</span>,</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>    <span class="st">"doParallel"</span>, <span class="st">"data.table"</span>, <span class="st">"foreach"</span>,<span class="st">"parallel"</span>), <span class="at">dependencies =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="you-can-install-genomautomorphism-package-from-github">You can install <strong>GenomAutomorphism</strong> package from GitHub<a class="anchor" aria-label="anchor" href="#you-can-install-genomautomorphism-package-from-github"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>   BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">'genomaths/GenomAutomorphism'</span>)</span></code></pre></div>
<hr>
</div>
</div>
<div class="section level1">
<h1 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h1>
<ol style="list-style-type: decimal">
<li><p>Sanchez R, Morgado E, Grau R. Gene algebra from a genetic code algebraic structure. J Math Biol. 2005 Oct;51(4):431-57. doi: 10.1007/s00285-005-0332-8. Epub 2005 Jul 13. PMID: 16012800. ( <a href="https://arxiv.org/pdf/q-bio/0412033.pdf" class="external-link">PDF</a>).</p></li>
<li><p>Sanchez R, Grau R, Morgado E. A novel Lie algebra of the genetic code over the Galois field of four DNA bases. Math Biosci. 2006;202: 156–174. <a href="doi:10.1016/j.mbs.2006.03.017" class="uri">doi:10.1016/j.mbs.2006.03.017</a></p></li>
<li><p>Sanchez R, Grau R. An algebraic hypothesis about the primeval genetic code architecture. Math Biosci. 2009/07/18. 2009;221: 60–76. <a href="https://doi.org/10.1016/j.mbs.2009.07.001" class="external-link">doi:10.1016/j.mbs.2009.07.001</a></p></li>
<li><p>Robersy Sanchez, Jesús Barreto (2021) Genomic Abelian Finite Groups. <a href="https://doi.org/10.1101/2021.06.01.446543" class="external-link">doi: 10.1101/2021.06.01.446543</a>.</p></li>
<li><p>M. V José, E.R. Morgado, R. Sánchez, T. Govezensky, The 24 possible algebraic representations of the standard genetic code in six or in three dimensions, Adv. Stud. Biol. 4 (2012) 119–152.<a href="https://is.gd/na9eap" class="external-link">PDF</a>.</p></li>
<li><p>R. Sanchez. Symmetric Group of the Genetic–Code Cubes. Effect of the Genetic–Code Architecture on the Evolutionary Process MATCH Commun. Math. Comput. Chem. 79 (2018) 527-560. <a href="https://bit.ly/2Z9mjM7" class="external-link">PDF</a>.</p></li>
<li><p>Sanchez, R., 2014. Evolutionary Analysis of DNA-protein-coding regions based on a genetic code cube metric. Current Topics in Medicinal Chemistry, 14(3), pp.407-417. <a href="https://doi.org/10.2174/1568026613666131204110022" class="external-link uri">https://doi.org/10.2174/1568026613666131204110022</a>.</p></li>
</ol>
<div class="section level2">
<h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a>
</h2>
<p><a href="https://github.com/genomaths/GenomeAlgebra_SymmetricGroup" class="external-link">Symmetric Group of the Genetic-Code Cubes</a></p>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/GenomAutomorphism" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/genomaths/GenomAutomorphism/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/genomaths/GenomAutomorphism/issues" class="external-link">Report a bug</a></li>
<li><a href="https://genomaths.com" class="external-link">Author's website</a></li>
<li><a href="https://bio.psu.edu/directory/rus547" class="external-link">Affiliation (<img src="https://git.psu.edu/genomath/MethylIT_data/raw/master/penn-state-lion.png" height="24"> PennState)</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0" class="external-link">Artistic-2.0</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing GenomAutomorphism</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://genomaths.com" class="external-link">Robersy Sanchez</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-5246-1453" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://genomaths.com" class="external-link">Robersy Sanchez</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
